--- a/net/minecraft/realms/RealmsConnect.java
+++ b/net/minecraft/realms/RealmsConnect.java
@@ -48,7 +_,7 @@
                   return;
                }
 
-               RealmsConnect.this.f_120690_.m_129505_(new ClientHandshakePacketListenerImpl(RealmsConnect.this.f_120690_, minecraft, RealmsConnect.this.f_120688_, (p_120726_) -> {
+               RealmsConnect.this.f_120690_.m_129505_(new ClientHandshakePacketListenerImpl(RealmsConnect.this.f_120690_, minecraft, RealmsConnect.this.f_120688_, (p_209500_0_) -> {
                }));
                if (RealmsConnect.this.f_120689_) {
                   return;
