--- a/net/minecraft/advancements/critereon/BeeNestDestroyedTrigger.java
+++ b/net/minecraft/advancements/critereon/BeeNestDestroyedTrigger.java
@@ -37,8 +_,8 @@
    }
 
    public void m_17477_(ServerPlayer p_17478_, Block p_17479_, ItemStack p_17480_, int p_17481_) {
-      this.m_66234_(p_17478_, (p_17486_) -> {
-         return p_17486_.m_17508_(p_17479_, p_17480_, p_17481_);
+      this.m_66234_(p_17478_, (p_226220_3_) -> {
+         return p_226220_3_.m_17508_(p_17479_, p_17480_, p_17481_);
       });
    }
 
