--- a/net/minecraft/advancements/critereon/ChanneledLightningTrigger.java
+++ b/net/minecraft/advancements/critereon/ChanneledLightningTrigger.java
@@ -23,11 +_,11 @@
    }
 
    public void m_21721_(ServerPlayer p_21722_, Collection<? extends Entity> p_21723_) {
-      List<LootContext> list = p_21723_.stream().map((p_21720_) -> {
-         return EntityPredicate.m_36616_(p_21722_, p_21720_);
+      List<LootContext> list = p_21723_.stream().map((p_233674_1_) -> {
+         return EntityPredicate.m_36616_(p_21722_, p_233674_1_);
       }).collect(Collectors.toList());
-      this.m_66234_(p_21722_, (p_21730_) -> {
-         return p_21730_.m_21744_(list);
+      this.m_66234_(p_21722_, (p_233673_1_) -> {
+         return p_233673_1_.m_21744_(list);
       });
    }
 
@@ -40,8 +_,8 @@
       }
 
       public static ChanneledLightningTrigger.TriggerInstance m_21746_(EntityPredicate... p_21747_) {
-         return new ChanneledLightningTrigger.TriggerInstance(EntityPredicate.Composite.f_36667_, Stream.of(p_21747_).map(EntityPredicate.Composite::m_36673_).toArray((p_21741_) -> {
-            return new EntityPredicate.Composite[p_21741_];
+         return new ChanneledLightningTrigger.TriggerInstance(EntityPredicate.Composite.f_36667_, Stream.of(p_21747_).map(EntityPredicate.Composite::m_36673_).toArray((p_233675_0_) -> {
+            return new EntityPredicate.Composite[p_233675_0_];
          }));
       }
 
