--- a/net/minecraft/advancements/critereon/ChangeDimensionTrigger.java
+++ b/net/minecraft/advancements/critereon/ChangeDimensionTrigger.java
@@ -23,8 +_,8 @@
    }
 
    public void m_19757_(ServerPlayer p_19758_, ResourceKey<Level> p_19759_, ResourceKey<Level> p_19760_) {
-      this.m_66234_(p_19758_, (p_19768_) -> {
-         return p_19768_.m_19784_(p_19759_, p_19760_);
+      this.m_66234_(p_19758_, (p_233550_2_) -> {
+         return p_233550_2_.m_19784_(p_19759_, p_19760_);
       });
    }
 
