--- a/net/minecraft/advancements/critereon/ConsumeItemTrigger.java
+++ b/net/minecraft/advancements/critereon/ConsumeItemTrigger.java
@@ -21,8 +_,8 @@
    }
 
    public void m_23682_(ServerPlayer p_23683_, ItemStack p_23684_) {
-      this.m_66234_(p_23683_, (p_23687_) -> {
-         return p_23687_.m_23701_(p_23684_);
+      this.m_66234_(p_23683_, (p_226325_1_) -> {
+         return p_226325_1_.m_23701_(p_23684_);
       });
    }
 
