--- a/net/minecraft/advancements/critereon/EntityHurtPlayerTrigger.java
+++ b/net/minecraft/advancements/critereon/EntityHurtPlayerTrigger.java
@@ -18,8 +_,8 @@
    }
 
    public void m_35174_(ServerPlayer p_35175_, DamageSource p_35176_, float p_35177_, float p_35178_, boolean p_35179_) {
-      this.m_66234_(p_35175_, (p_35186_) -> {
-         return p_35186_.m_35200_(p_35175_, p_35176_, p_35177_, p_35178_, p_35179_);
+      this.m_66234_(p_35175_, (p_226603_5_) -> {
+         return p_226603_5_.m_35200_(p_35175_, p_35176_, p_35177_, p_35178_, p_35179_);
       });
    }
 
