--- a/net/minecraft/advancements/critereon/EntityPredicate.java
+++ b/net/minecraft/advancements/critereon/EntityPredicate.java
@@ -176,6 +_,83 @@
       return (new LootContext.Builder(p_36617_.m_9236_())).m_78972_(LootContextParams.f_81455_, p_36618_).m_78972_(LootContextParams.f_81460_, p_36617_.m_20182_()).m_78977_(p_36617_.m_21187_()).m_78975_(LootContextParamSets.f_81419_);
    }
 
+   public static class Composite {
+      public static final EntityPredicate.Composite f_36667_ = new EntityPredicate.Composite(new LootItemCondition[0]);
+      private final LootItemCondition[] f_36668_;
+      private final Predicate<LootContext> f_36669_;
+
+      private Composite(LootItemCondition[] p_36672_) {
+         this.f_36668_ = p_36672_;
+         this.f_36669_ = LootItemConditions.m_81834_(p_36672_);
+      }
+
+      public static EntityPredicate.Composite m_36690_(LootItemCondition... p_36691_) {
+         return new EntityPredicate.Composite(p_36691_);
+      }
+
+      public static EntityPredicate.Composite m_36677_(JsonObject p_36678_, String p_36679_, DeserializationContext p_36680_) {
+         JsonElement jsonelement = p_36678_.get(p_36679_);
+         return m_36683_(p_36679_, p_36680_, jsonelement);
+      }
+
+      public static EntityPredicate.Composite[] m_36692_(JsonObject p_36693_, String p_36694_, DeserializationContext p_36695_) {
+         JsonElement jsonelement = p_36693_.get(p_36694_);
+         if (jsonelement != null && !jsonelement.isJsonNull()) {
+            JsonArray jsonarray = GsonHelper.m_13924_(jsonelement, p_36694_);
+            EntityPredicate.Composite[] aentitypredicate$composite = new EntityPredicate.Composite[jsonarray.size()];
+
+            for(int i = 0; i < jsonarray.size(); ++i) {
+               aentitypredicate$composite[i] = m_36683_(p_36694_ + "[" + i + "]", p_36695_, jsonarray.get(i));
+            }
+
+            return aentitypredicate$composite;
+         } else {
+            return new EntityPredicate.Composite[0];
+         }
+      }
+
+      private static EntityPredicate.Composite m_36683_(String p_36684_, DeserializationContext p_36685_, @Nullable JsonElement p_36686_) {
+         if (p_36686_ != null && p_36686_.isJsonArray()) {
+            LootItemCondition[] alootitemcondition = p_36685_.m_25874_(p_36686_.getAsJsonArray(), p_36685_.m_25873_().toString() + "/" + p_36684_, LootContextParamSets.f_81419_);
+            return new EntityPredicate.Composite(alootitemcondition);
+         } else {
+            EntityPredicate entitypredicate = EntityPredicate.m_36614_(p_36686_);
+            return m_36673_(entitypredicate);
+         }
+      }
+
+      public static EntityPredicate.Composite m_36673_(EntityPredicate p_36674_) {
+         if (p_36674_ == EntityPredicate.f_36550_) {
+            return f_36667_;
+         } else {
+            LootItemCondition lootitemcondition = LootItemEntityPropertyCondition.m_81867_(LootContext.EntityTarget.THIS, p_36674_).m_6409_();
+            return new EntityPredicate.Composite(new LootItemCondition[]{lootitemcondition});
+         }
+      }
+
+      public boolean m_36681_(LootContext p_36682_) {
+         return this.f_36669_.test(p_36682_);
+      }
+
+      public JsonElement m_36675_(SerializationContext p_36676_) {
+         return (JsonElement)(this.f_36668_.length == 0 ? JsonNull.INSTANCE : p_36676_.m_64772_(this.f_36668_));
+      }
+
+      public static JsonElement m_36687_(EntityPredicate.Composite[] p_36688_, SerializationContext p_36689_) {
+         if (p_36688_.length == 0) {
+            return JsonNull.INSTANCE;
+         } else {
+            JsonArray jsonarray = new JsonArray();
+
+            for(EntityPredicate.Composite entitypredicate$composite : p_36688_) {
+               jsonarray.add(entitypredicate$composite.m_36675_(p_36689_));
+            }
+
+            return jsonarray;
+         }
+      }
+   }
+
    public static class Builder {
       private EntityTypePredicate f_36619_ = EntityTypePredicate.f_37636_;
       private DistancePredicate f_36620_ = DistancePredicate.f_26241_;
@@ -277,83 +_,6 @@
 
       public EntityPredicate m_36662_() {
          return new EntityPredicate(this.f_36619_, this.f_36620_, this.f_36621_, this.f_36622_, this.f_36623_, this.f_36624_, this.f_36625_, this.f_36626_, this.f_36627_, this.f_36628_, this.f_36629_, this.f_36630_, this.f_36631_);
-      }
-   }
-
-   public static class Composite {
-      public static final EntityPredicate.Composite f_36667_ = new EntityPredicate.Composite(new LootItemCondition[0]);
-      private final LootItemCondition[] f_36668_;
-      private final Predicate<LootContext> f_36669_;
-
-      private Composite(LootItemCondition[] p_36672_) {
-         this.f_36668_ = p_36672_;
-         this.f_36669_ = LootItemConditions.m_81834_(p_36672_);
-      }
-
-      public static EntityPredicate.Composite m_36690_(LootItemCondition... p_36691_) {
-         return new EntityPredicate.Composite(p_36691_);
-      }
-
-      public static EntityPredicate.Composite m_36677_(JsonObject p_36678_, String p_36679_, DeserializationContext p_36680_) {
-         JsonElement jsonelement = p_36678_.get(p_36679_);
-         return m_36683_(p_36679_, p_36680_, jsonelement);
-      }
-
-      public static EntityPredicate.Composite[] m_36692_(JsonObject p_36693_, String p_36694_, DeserializationContext p_36695_) {
-         JsonElement jsonelement = p_36693_.get(p_36694_);
-         if (jsonelement != null && !jsonelement.isJsonNull()) {
-            JsonArray jsonarray = GsonHelper.m_13924_(jsonelement, p_36694_);
-            EntityPredicate.Composite[] aentitypredicate$composite = new EntityPredicate.Composite[jsonarray.size()];
-
-            for(int i = 0; i < jsonarray.size(); ++i) {
-               aentitypredicate$composite[i] = m_36683_(p_36694_ + "[" + i + "]", p_36695_, jsonarray.get(i));
-            }
-
-            return aentitypredicate$composite;
-         } else {
-            return new EntityPredicate.Composite[0];
-         }
-      }
-
-      private static EntityPredicate.Composite m_36683_(String p_36684_, DeserializationContext p_36685_, @Nullable JsonElement p_36686_) {
-         if (p_36686_ != null && p_36686_.isJsonArray()) {
-            LootItemCondition[] alootitemcondition = p_36685_.m_25874_(p_36686_.getAsJsonArray(), p_36685_.m_25873_().toString() + "/" + p_36684_, LootContextParamSets.f_81419_);
-            return new EntityPredicate.Composite(alootitemcondition);
-         } else {
-            EntityPredicate entitypredicate = EntityPredicate.m_36614_(p_36686_);
-            return m_36673_(entitypredicate);
-         }
-      }
-
-      public static EntityPredicate.Composite m_36673_(EntityPredicate p_36674_) {
-         if (p_36674_ == EntityPredicate.f_36550_) {
-            return f_36667_;
-         } else {
-            LootItemCondition lootitemcondition = LootItemEntityPropertyCondition.m_81867_(LootContext.EntityTarget.THIS, p_36674_).m_6409_();
-            return new EntityPredicate.Composite(new LootItemCondition[]{lootitemcondition});
-         }
-      }
-
-      public boolean m_36681_(LootContext p_36682_) {
-         return this.f_36669_.test(p_36682_);
-      }
-
-      public JsonElement m_36675_(SerializationContext p_36676_) {
-         return (JsonElement)(this.f_36668_.length == 0 ? JsonNull.INSTANCE : p_36676_.m_64772_(this.f_36668_));
-      }
-
-      public static JsonElement m_36687_(EntityPredicate.Composite[] p_36688_, SerializationContext p_36689_) {
-         if (p_36688_.length == 0) {
-            return JsonNull.INSTANCE;
-         } else {
-            JsonArray jsonarray = new JsonArray();
-
-            for(EntityPredicate.Composite entitypredicate$composite : p_36688_) {
-               jsonarray.add(entitypredicate$composite.m_36675_(p_36689_));
-            }
-
-            return jsonarray;
-         }
       }
    }
 }
