--- a/net/minecraft/advancements/critereon/ItemUsedOnBlockTrigger.java
+++ b/net/minecraft/advancements/critereon/ItemUsedOnBlockTrigger.java
@@ -23,8 +_,8 @@
 
    public void m_45482_(ServerPlayer p_45483_, BlockPos p_45484_, ItemStack p_45485_) {
       BlockState blockstate = p_45483_.m_9236_().m_8055_(p_45484_);
-      this.m_66234_(p_45483_, (p_45491_) -> {
-         return p_45491_.m_45510_(blockstate, p_45483_.m_9236_(), p_45484_, p_45485_);
+      this.m_66234_(p_45483_, (p_226694_4_) -> {
+         return p_226694_4_.m_45510_(blockstate, p_45483_.m_9236_(), p_45484_, p_45485_);
       });
    }
 
