--- a/net/minecraft/advancements/critereon/KilledByCrossbowTrigger.java
+++ b/net/minecraft/advancements/critereon/KilledByCrossbowTrigger.java
@@ -35,8 +_,8 @@
          list.add(EntityPredicate.m_36616_(p_46872_, entity));
       }
 
-      this.m_66234_(p_46872_, (p_46881_) -> {
-         return p_46881_.m_46897_(list, set.size());
+      this.m_66234_(p_46872_, (p_234940_2_) -> {
+         return p_234940_2_.m_46897_(list, set.size());
       });
    }
 
