--- a/net/minecraft/advancements/critereon/LevitationTrigger.java
+++ b/net/minecraft/advancements/critereon/LevitationTrigger.java
@@ -19,8 +_,8 @@
    }
 
    public void m_49116_(ServerPlayer p_49117_, Vec3 p_49118_, int p_49119_) {
-      this.m_66234_(p_49117_, (p_49124_) -> {
-         return p_49124_.m_49140_(p_49117_, p_49118_, p_49119_);
+      this.m_66234_(p_49117_, (p_226852_3_) -> {
+         return p_226852_3_.m_49140_(p_49117_, p_49118_, p_49119_);
       });
    }
 
