--- a/net/minecraft/advancements/critereon/MinMaxBounds.java
+++ b/net/minecraft/advancements/critereon/MinMaxBounds.java
@@ -145,16 +_,6 @@
       return (T)(p_55324_ == null ? null : p_55325_.apply(p_55324_));
    }
 
-   @FunctionalInterface
-   public interface BoundsFactory<T extends Number, R extends MinMaxBounds<T>> {
-      R m_55329_(@Nullable T p_55330_, @Nullable T p_55331_);
-   }
-
-   @FunctionalInterface
-   public interface BoundsFromReaderFactory<T extends Number, R extends MinMaxBounds<T>> {
-      R m_55332_(StringReader p_55333_, @Nullable T p_55334_, @Nullable T p_55335_) throws CommandSyntaxException;
-   }
-
    public static class Floats extends MinMaxBounds<Float> {
       public static final MinMaxBounds.Floats f_55336_ = new MinMaxBounds.Floats((Float)null, (Float)null);
       private final Double f_55337_;
@@ -204,8 +_,8 @@
       }
 
       public static MinMaxBounds.Floats m_55347_(StringReader p_55348_) throws CommandSyntaxException {
-         return m_55353_(p_55348_, (p_55361_) -> {
-            return p_55361_;
+         return m_55353_(p_55348_, (p_211358_0_) -> {
+            return p_211358_0_;
          });
       }
 
@@ -214,6 +_,16 @@
       }
    }
 
+   @FunctionalInterface
+   public interface BoundsFactory<T extends Number, R extends MinMaxBounds<T>> {
+      R m_55329_(@Nullable T p_55330_, @Nullable T p_55331_);
+   }
+
+   @FunctionalInterface
+   public interface BoundsFromReaderFactory<T extends Number, R extends MinMaxBounds<T>> {
+      R m_55332_(StringReader p_55333_, @Nullable T p_55334_, @Nullable T p_55335_) throws CommandSyntaxException;
+   }
+
    public static class Ints extends MinMaxBounds<Integer> {
       public static final MinMaxBounds.Ints f_55364_ = new MinMaxBounds.Ints((Integer)null, (Integer)null);
       private final Long f_55365_;
@@ -259,8 +_,8 @@
       }
 
       public static MinMaxBounds.Ints m_55375_(StringReader p_55376_) throws CommandSyntaxException {
-         return m_55381_(p_55376_, (p_55389_) -> {
-            return p_55389_;
+         return m_55381_(p_55376_, (p_211346_0_) -> {
+            return p_211346_0_;
          });
       }
 
