--- a/net/minecraft/advancements/critereon/PlacedBlockTrigger.java
+++ b/net/minecraft/advancements/critereon/PlacedBlockTrigger.java
@@ -24,8 +_,8 @@
       Block block = m_59482_(p_59485_);
       StatePropertiesPredicate statepropertiespredicate = StatePropertiesPredicate.m_67679_(p_59485_.get("state"));
       if (block != null) {
-         statepropertiespredicate.m_67672_(block.m_49965_(), (p_59475_) -> {
-            throw new JsonSyntaxException("Block " + block + " has no property " + p_59475_ + ":");
+         statepropertiespredicate.m_67672_(block.m_49965_(), (p_226948_1_) -> {
+            throw new JsonSyntaxException("Block " + block + " has no property " + p_226948_1_ + ":");
          });
       }
 
@@ -48,8 +_,8 @@
 
    public void m_59469_(ServerPlayer p_59470_, BlockPos p_59471_, ItemStack p_59472_) {
       BlockState blockstate = p_59470_.m_9236_().m_8055_(p_59471_);
-      this.m_66234_(p_59470_, (p_59481_) -> {
-         return p_59481_.m_59507_(blockstate, p_59471_, p_59470_.m_9236_(), p_59472_);
+      this.m_66234_(p_59470_, (p_226949_4_) -> {
+         return p_226949_4_.m_59507_(blockstate, p_59471_, p_59470_.m_9236_(), p_59472_);
       });
    }
 
