--- a/net/minecraft/advancements/critereon/PlayerHurtEntityTrigger.java
+++ b/net/minecraft/advancements/critereon/PlayerHurtEntityTrigger.java
@@ -22,8 +_,8 @@
 
    public void m_60112_(ServerPlayer p_60113_, Entity p_60114_, DamageSource p_60115_, float p_60116_, float p_60117_, boolean p_60118_) {
       LootContext lootcontext = EntityPredicate.m_36616_(p_60113_, p_60114_);
-      this.m_66234_(p_60113_, (p_60126_) -> {
-         return p_60126_.m_60142_(p_60113_, lootcontext, p_60115_, p_60116_, p_60117_, p_60118_);
+      this.m_66234_(p_60113_, (p_226956_6_) -> {
+         return p_226956_6_.m_60142_(p_60113_, lootcontext, p_60115_, p_60116_, p_60117_, p_60118_);
       });
    }
 
