--- a/net/minecraft/advancements/critereon/PlayerPredicate.java
+++ b/net/minecraft/advancements/critereon/PlayerPredicate.java
@@ -44,9 +_,9 @@
       } else {
          Object2BooleanMap<String> object2booleanmap = new Object2BooleanOpenHashMap<>();
          JsonObject jsonobject = GsonHelper.m_13918_(p_62290_, "criterion data");
-         jsonobject.entrySet().forEach((p_62288_) -> {
-            boolean flag1 = GsonHelper.m_13877_(p_62288_.getValue(), "criterion test");
-            object2booleanmap.put(p_62288_.getKey(), flag1);
+         jsonobject.entrySet().forEach((p_227003_1_) -> {
+            boolean flag1 = GsonHelper.m_13877_(p_227003_1_.getValue(), "criterion test");
+            object2booleanmap.put(p_227003_1_.getKey(), flag1);
          });
          return new PlayerPredicate.AdvancementCriterionsPredicate(object2booleanmap);
       }
@@ -180,11 +_,11 @@
 
          if (!this.f_62247_.isEmpty()) {
             JsonArray jsonarray = new JsonArray();
-            this.f_62247_.forEach((p_62274_, p_62275_) -> {
+            this.f_62247_.forEach((p_226999_1_, p_226999_2_) -> {
                JsonObject jsonobject3 = new JsonObject();
-               jsonobject3.addProperty("type", Registry.f_122867_.m_7981_(p_62274_.m_12859_()).toString());
-               jsonobject3.addProperty("stat", m_62265_(p_62274_).toString());
-               jsonobject3.add("value", p_62275_.m_55328_());
+               jsonobject3.addProperty("type", Registry.f_122867_.m_7981_(p_226999_1_.m_12859_()).toString());
+               jsonobject3.addProperty("stat", m_62265_(p_226999_1_).toString());
+               jsonobject3.add("value", p_226999_2_.m_55328_());
                jsonarray.add(jsonobject3);
             });
             jsonobject.add("stats", jsonarray);
@@ -192,16 +_,16 @@
 
          if (!this.f_62248_.isEmpty()) {
             JsonObject jsonobject1 = new JsonObject();
-            this.f_62248_.forEach((p_62284_, p_62285_) -> {
-               jsonobject1.addProperty(p_62284_.toString(), p_62285_);
+            this.f_62248_.forEach((p_227002_1_, p_227002_2_) -> {
+               jsonobject1.addProperty(p_227002_1_.toString(), p_227002_2_);
             });
             jsonobject.add("recipes", jsonobject1);
          }
 
          if (!this.f_62249_.isEmpty()) {
             JsonObject jsonobject2 = new JsonObject();
-            this.f_62249_.forEach((p_62280_, p_62281_) -> {
-               jsonobject2.add(p_62280_.toString(), p_62281_.m_7943_());
+            this.f_62249_.forEach((p_227001_1_, p_227001_2_) -> {
+               jsonobject2.add(p_227001_1_.toString(), p_227001_2_.m_7943_());
             });
             jsonobject.add("advancements", jsonobject2);
          }
@@ -210,6 +_,22 @@
       }
    }
 
+   static class AdvancementDonePredicate implements PlayerPredicate.AdvancementPredicate {
+      private final boolean f_62299_;
+
+      public AdvancementDonePredicate(boolean p_62301_) {
+         this.f_62299_ = p_62301_;
+      }
+
+      public JsonElement m_7943_() {
+         return new JsonPrimitive(this.f_62299_);
+      }
+
+      public boolean test(AdvancementProgress p_62304_) {
+         return p_62304_.m_8193_() == this.f_62299_;
+      }
+   }
+
    static class AdvancementCriterionsPredicate implements PlayerPredicate.AdvancementPredicate {
       private final Object2BooleanMap<String> f_62291_;
 
@@ -235,26 +_,6 @@
       }
    }
 
-   static class AdvancementDonePredicate implements PlayerPredicate.AdvancementPredicate {
-      private final boolean f_62299_;
-
-      public AdvancementDonePredicate(boolean p_62301_) {
-         this.f_62299_ = p_62301_;
-      }
-
-      public JsonElement m_7943_() {
-         return new JsonPrimitive(this.f_62299_);
-      }
-
-      public boolean test(AdvancementProgress p_62304_) {
-         return p_62304_.m_8193_() == this.f_62299_;
-      }
-   }
-
-   interface AdvancementPredicate extends Predicate<AdvancementProgress> {
-      JsonElement m_7943_();
-   }
-
    public static class Builder {
       private MinMaxBounds.Ints f_62307_ = MinMaxBounds.Ints.f_55364_;
       private GameType f_62308_ = GameType.NOT_SET;
@@ -265,5 +_,9 @@
       public PlayerPredicate m_62313_() {
          return new PlayerPredicate(this.f_62307_, this.f_62308_, this.f_62309_, this.f_62310_, this.f_62311_);
       }
+   }
+
+   interface AdvancementPredicate extends Predicate<AdvancementProgress> {
+      JsonElement m_7943_();
    }
 }
