--- a/net/minecraft/advancements/critereon/SimpleCriterionTrigger.java
+++ b/net/minecraft/advancements/critereon/SimpleCriterionTrigger.java
@@ -17,7 +_,7 @@
    private final Map<PlayerAdvancements, Set<CriterionTrigger.Listener<T>>> f_66232_ = Maps.newIdentityHashMap();
 
    public final void m_6467_(PlayerAdvancements p_66243_, CriterionTrigger.Listener<T> p_66244_) {
-      this.f_66232_.computeIfAbsent(p_66243_, (p_66252_) -> {
+      this.f_66232_.computeIfAbsent(p_66243_, (p_227072_0_) -> {
          return Sets.newHashSet();
       }).add(p_66244_);
    }
