--- a/net/minecraft/advancements/critereon/SummonedEntityTrigger.java
+++ b/net/minecraft/advancements/critereon/SummonedEntityTrigger.java
@@ -20,8 +_,8 @@
 
    public void m_68256_(ServerPlayer p_68257_, Entity p_68258_) {
       LootContext lootcontext = EntityPredicate.m_36616_(p_68257_, p_68258_);
-      this.m_66234_(p_68257_, (p_68265_) -> {
-         return p_68265_.m_68279_(lootcontext);
+      this.m_66234_(p_68257_, (p_227229_1_) -> {
+         return p_227229_1_.m_68279_(lootcontext);
       });
    }
 
