--- a/net/minecraft/util/profiling/ActiveProfiler.java
+++ b/net/minecraft/util/profiling/ActiveProfiler.java
@@ -128,7 +_,7 @@
 
    private ActiveProfiler.PathEntry m_18406_() {
       if (this.f_18379_ == null) {
-         this.f_18379_ = this.f_18372_.computeIfAbsent(this.f_18377_, (p_18405_) -> {
+         this.f_18379_ = this.f_18372_.computeIfAbsent(this.f_18377_, (p_230080_0_) -> {
             return new ActiveProfiler.PathEntry();
          });
       }
