--- a/net/minecraft/util/profiling/FilledProfileResults.java
+++ b/net/minecraft/util/profiling/FilledProfileResults.java
@@ -40,8 +_,8 @@
       }
    };
    private static final Splitter f_18454_ = Splitter.on('\u001e');
-   private static final Comparator<Entry<String, FilledProfileResults.CounterCollector>> f_18455_ = Entry.<String, FilledProfileResults.CounterCollector>comparingByValue(Comparator.comparingLong((p_18489_) -> {
-      return p_18489_.f_18538_;
+   private static final Comparator<Entry<String, FilledProfileResults.CounterCollector>> f_18455_ = Entry.<String, FilledProfileResults.CounterCollector>comparingByValue(Comparator.comparingLong((p_230096_0_) -> {
+      return p_230096_0_.f_18538_;
    })).reversed();
    private final Map<String, ? extends ProfilerPathEntry> f_18456_;
    private final long f_18457_;
@@ -119,14 +_,14 @@
 
    private Map<String, FilledProfileResults.CounterCollector> m_18531_() {
       Map<String, FilledProfileResults.CounterCollector> map = Maps.newTreeMap();
-      this.f_18456_.forEach((p_18512_, p_18513_) -> {
-         Object2LongMap<String> object2longmap = p_18513_.m_7446_();
+      this.f_18456_.forEach((p_230101_1_, p_230101_2_) -> {
+         Object2LongMap<String> object2longmap = p_230101_2_.m_7446_();
          if (!object2longmap.isEmpty()) {
-            List<String> list = f_18454_.splitToList(p_18512_);
-            object2longmap.forEach((p_18521_, p_18522_) -> {
-               map.computeIfAbsent(p_18521_, (p_18529_) -> {
+            List<String> list = f_18454_.splitToList(p_230101_1_);
+            object2longmap.forEach((p_230103_2_, p_230103_3_) -> {
+               map.computeIfAbsent(p_230103_2_, (p_230105_0_) -> {
                   return new FilledProfileResults.CounterCollector();
-               }).m_18547_(list.iterator(), p_18522_);
+               }).m_18547_(list.iterator(), p_230103_3_);
             });
          }
 
@@ -205,8 +_,8 @@
    private void m_18481_(int p_18482_, String p_18483_, StringBuilder p_18484_) {
       List<ResultField> list = this.m_6412_(p_18483_);
       Object2LongMap<String> object2longmap = ObjectUtils.firstNonNull(this.f_18456_.get(p_18483_), f_18453_).m_7446_();
-      object2longmap.forEach((p_18508_, p_18509_) -> {
-         m_18497_(p_18484_, p_18482_).append('#').append(p_18508_).append(' ').append((Object)p_18509_).append('/').append(p_18509_ / (long)this.f_18461_).append('\n');
+      object2longmap.forEach((p_230100_3_, p_230100_4_) -> {
+         m_18497_(p_18484_, p_18482_).append('#').append(p_230100_3_).append(' ').append((Object)p_230100_4_).append('/').append(p_230100_4_ / (long)this.f_18461_).append('\n');
       });
       if (list.size() >= 3) {
          for(int i = 1; i < list.size(); ++i) {
@@ -226,15 +_,15 @@
 
    private void m_18475_(int p_18476_, String p_18477_, FilledProfileResults.CounterCollector p_18478_, int p_18479_, StringBuilder p_18480_) {
       m_18497_(p_18480_, p_18476_).append(p_18477_).append(" total:").append(p_18478_.f_18537_).append('/').append(p_18478_.f_18538_).append(" average: ").append(p_18478_.f_18537_ / (long)p_18479_).append('/').append(p_18478_.f_18538_ / (long)p_18479_).append('\n');
-      p_18478_.f_18539_.entrySet().stream().sorted(f_18455_).forEach((p_18474_) -> {
-         this.m_18475_(p_18476_ + 1, p_18474_.getKey(), p_18474_.getValue(), p_18479_, p_18480_);
+      p_18478_.f_18539_.entrySet().stream().sorted(f_18455_).forEach((p_230094_4_) -> {
+         this.m_18475_(p_18476_ + 1, p_230094_4_.getKey(), p_230094_4_.getValue(), p_18479_, p_18480_);
       });
    }
 
    private void m_18514_(Map<String, FilledProfileResults.CounterCollector> p_18515_, StringBuilder p_18516_, int p_18517_) {
-      p_18515_.forEach((p_18503_, p_18504_) -> {
-         p_18516_.append("-- Counter: ").append(p_18503_).append(" --\n");
-         this.m_18475_(0, "root", p_18504_.f_18539_.get("root"), p_18517_, p_18516_);
+      p_18515_.forEach((p_230099_3_, p_230099_4_) -> {
+         p_18516_.append("-- Counter: ").append(p_230099_3_).append(" --\n");
+         this.m_18475_(0, "root", p_230099_4_.f_18539_.get("root"), p_18517_, p_18516_);
          p_18516_.append("\n\n");
       });
    }
@@ -266,7 +_,7 @@
          if (!p_18548_.hasNext()) {
             this.f_18537_ += p_18549_;
          } else {
-            this.f_18539_.computeIfAbsent(p_18548_.next(), (p_18546_) -> {
+            this.f_18539_.computeIfAbsent(p_18548_.next(), (p_230111_0_) -> {
                return new FilledProfileResults.CounterCollector();
             }).m_18547_(p_18548_, p_18549_);
          }
