--- a/net/minecraft/data/tags/TagsProvider.java
+++ b/net/minecraft/data/tags/TagsProvider.java
@@ -30,11 +_,29 @@
    private static final Gson f_126542_ = (new GsonBuilder()).setPrettyPrinting().create();
    protected final DataGenerator f_126539_;
    protected final Registry<T> f_126540_;
-   private final Map<ResourceLocation, Tag.Builder> f_126543_ = Maps.newLinkedHashMap();
+   protected final Map<ResourceLocation, Tag.Builder> f_126543_ = Maps.newLinkedHashMap();
+   protected final String modId;
+   protected final String folder;
+   protected final net.minecraftforge.common.data.ExistingFileHelper existingFileHelper;
+   private final net.minecraftforge.common.data.ExistingFileHelper.IResourceType resourceType;
 
+   @Deprecated//Forge, Use ModID version.
    protected TagsProvider(DataGenerator p_126546_, Registry<T> p_126547_) {
-      this.f_126539_ = p_126546_;
-      this.f_126540_ = p_126547_;
+      this(p_126546_, p_126547_, "vanilla", null);
+   }
+   protected TagsProvider(DataGenerator p_i49827_1_, Registry<T> p_i49827_2_, String modId, @javax.annotation.Nullable net.minecraftforge.common.data.ExistingFileHelper existingFileHelper) {
+      this(p_i49827_1_, p_i49827_2_, modId, existingFileHelper, null);
+   }
+   protected TagsProvider(DataGenerator p_i49827_1_, Registry<T> p_i49827_2_, String modId, @javax.annotation.Nullable net.minecraftforge.common.data.ExistingFileHelper existingFileHelper, @javax.annotation.Nullable String folder) {
+      this.f_126539_ = p_i49827_1_;
+      this.f_126540_ = p_i49827_2_;
+      this.modId = modId;
+      this.existingFileHelper = existingFileHelper;
+      if (folder == null) {//TODO - 1.17: Remove getTagFolder as a method, and inline the default implementation here
+         folder = getTagFolder();
+      }
+      this.folder = folder;
+      this.resourceType = new net.minecraftforge.common.data.ExistingFileHelper.ResourceType(net.minecraft.server.packs.PackType.SERVER_DATA, ".json", "tags/" + this.folder);
    }
 
    protected abstract void m_6577_();
@@ -43,19 +_,21 @@
       this.f_126543_.clear();
       this.m_6577_();
       Tag<T> tag = SetTag.m_13216_();
-      Function<ResourceLocation, Tag<T>> function = (p_126552_) -> {
-         return this.f_126543_.containsKey(p_126552_) ? tag : null;
-      };
-      Function<ResourceLocation, T> function1 = (p_126567_) -> {
-         return this.f_126540_.m_6612_(p_126567_).orElse((T)null);
-      };
-      this.f_126543_.forEach((p_126559_, p_126560_) -> {
-         List<Tag.BuilderEntry> list = p_126560_.m_13331_(function, function1).collect(Collectors.toList());
+      Function<ResourceLocation, Tag<T>> function = (p_240523_2_) -> {
+         return this.f_126543_.containsKey(p_240523_2_) ? tag : null;
+      };
+      Function<ResourceLocation, T> function1 = (p_240527_1_) -> {
+         return this.f_126540_.m_6612_(p_240527_1_).orElse((T)null);
+      };
+      this.f_126543_.forEach((p_240524_4_, p_240524_5_) -> {
+         // FORGE: Add validation via existing resources
+         List<Tag.BuilderEntry> list = p_240524_5_.m_13331_(function, function1).filter(this::missing).collect(Collectors.toList());
          if (!list.isEmpty()) {
-            throw new IllegalArgumentException(String.format("Couldn't define tag %s as it is missing following references: %s", p_126559_, list.stream().map(Objects::toString).collect(Collectors.joining(","))));
+            throw new IllegalArgumentException(String.format("Couldn't define tag %s as it is missing following references: %s", p_240524_4_, list.stream().map(Objects::toString).collect(Collectors.joining(","))));
          } else {
-            JsonObject jsonobject = p_126560_.m_13334_();
-            Path path = this.m_6648_(p_126559_);
+            JsonObject jsonobject = p_240524_5_.m_13334_();
+            Path path = this.m_6648_(p_240524_4_);
+            if (path == null) return; //Forge: Allow running this data provider without writing it. Recipe provider needs valid tags.
 
             try {
                String s = f_126542_.toJson((JsonElement)jsonobject);
@@ -77,20 +_,36 @@
       });
    }
 
+   private boolean missing(Tag.BuilderEntry reference) {
+      Tag.Entry entry = reference.m_13347_();
+      // We only care about non-optional tag entries, this is the only type that can reference a resource and needs validation
+      // Optional tags should not be validated
+      if (entry instanceof Tag.TagEntry) {
+         return existingFileHelper == null || !existingFileHelper.exists(((Tag.TagEntry)entry).getId(), resourceType);
+      }
+      return false;
+   }
+
+   @SuppressWarnings({ "unchecked", "rawtypes" })
+   protected String getTagFolder() {
+      return ((Registry)Registry.f_122897_).m_7981_(f_126540_).m_135815_() + "s";
+   }
+
    protected abstract Path m_6648_(ResourceLocation p_126561_);
 
    protected TagsProvider.TagAppender<T> m_126548_(Tag.Named<T> p_126549_) {
       Tag.Builder tag$builder = this.m_126562_(p_126549_);
-      return new TagsProvider.TagAppender<>(tag$builder, this.f_126540_, "vanilla");
+      return new TagsProvider.TagAppender<>(tag$builder, this.f_126540_, modId);
    }
 
    protected Tag.Builder m_126562_(Tag.Named<T> p_126563_) {
-      return this.f_126543_.computeIfAbsent(p_126563_.m_6979_(), (p_126565_) -> {
+      return this.f_126543_.computeIfAbsent(p_126563_.m_6979_(), (p_240526_0_) -> {
+         existingFileHelper.trackGenerated(p_240526_0_, resourceType);
          return new Tag.Builder();
       });
    }
 
-   public static class TagAppender<T> {
+   public static class TagAppender<T> implements net.minecraftforge.common.extensions.IForgeTagBuilder<T> {
       private final Tag.Builder f_126568_;
       private final Registry<T> f_126569_;
       private final String f_126570_;
@@ -113,10 +_,23 @@
 
       @SafeVarargs
       public final TagsProvider.TagAppender<T> m_126584_(T... p_126585_) {
-         Stream.<T>of(p_126585_).map(this.f_126569_::m_7981_).forEach((p_126587_) -> {
-            this.f_126568_.m_13327_(p_126587_, this.f_126570_);
+         Stream.<T>of(p_126585_).map(this.f_126569_::m_7981_).forEach((p_240533_1_) -> {
+            this.f_126568_.m_13327_(p_240533_1_, this.f_126570_);
          });
          return this;
+      }
+
+      public TagsProvider.TagAppender<T> add(Tag.Entry tag) {
+          f_126568_.m_13307_(tag, f_126570_);
+          return this;
+      }
+
+      public Tag.Builder getInternalBuilder() {
+          return f_126568_;
+      }
+
+      public String getModID() {
+          return f_126570_;
       }
    }
 }
