--- a/net/minecraft/client/renderer/debug/BeeDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/BeeDebugRenderer.java
@@ -68,23 +_,23 @@
    }
 
    private void m_113126_() {
-      this.f_113050_.entrySet().removeIf((p_113132_) -> {
-         return this.f_113048_.f_91073_.m_6815_((p_113132_.getValue()).f_113158_) == null;
+      this.f_113050_.entrySet().removeIf((p_228984_1_) -> {
+         return this.f_113048_.f_91073_.m_6815_((p_228984_1_.getValue()).f_113158_) == null;
       });
    }
 
    private void m_113136_() {
       long i = this.f_113048_.f_91073_.m_46467_() - 20L;
-      this.f_113049_.entrySet().removeIf((p_113057_) -> {
-         return (p_113057_.getValue()).f_113185_ < i;
+      this.f_113049_.entrySet().removeIf((p_228962_2_) -> {
+         return (p_228962_2_.getValue()).f_113185_ < i;
       });
    }
 
    private void m_113141_() {
       BlockPos blockpos = this.m_113154_().m_90588_();
-      this.f_113050_.values().forEach((p_113153_) -> {
-         if (this.m_113147_(p_113153_)) {
-            this.m_113137_(p_113153_);
+      this.f_113050_.values().forEach((p_228994_1_) -> {
+         if (this.m_113147_(p_228994_1_)) {
+            this.m_113137_(p_228994_1_);
          }
 
       });
@@ -97,16 +_,16 @@
       }
 
       Map<BlockPos, Set<UUID>> map = this.m_113146_();
-      this.f_113049_.values().forEach((p_113098_) -> {
-         if (blockpos.m_123314_(p_113098_.f_113180_, 30.0D)) {
-            Set<UUID> set = map.get(p_113098_.f_113180_);
-            this.m_113073_(p_113098_, (Collection<UUID>)(set == null ? Sets.newHashSet() : set));
+      this.f_113049_.values().forEach((p_228973_3_) -> {
+         if (blockpos.m_123314_(p_228973_3_.f_113180_, 30.0D)) {
+            Set<UUID> set = map.get(p_228973_3_.f_113180_);
+            this.m_113073_(p_228973_3_, (Collection<UUID>)(set == null ? Sets.newHashSet() : set));
          }
 
       });
-      this.m_113155_().forEach((p_113090_, p_113091_) -> {
-         if (blockpos.m_123314_(p_113090_, 30.0D)) {
-            this.m_113092_(p_113090_, p_113091_);
+      this.m_113155_().forEach((p_228971_2_, p_228971_3_) -> {
+         if (blockpos.m_123314_(p_228971_2_, 30.0D)) {
+            this.m_113092_(p_228971_2_, p_228971_3_);
          }
 
       });
@@ -114,11 +_,11 @@
 
    private Map<BlockPos, Set<UUID>> m_113146_() {
       Map<BlockPos, Set<UUID>> map = Maps.newHashMap();
-      this.f_113050_.values().forEach((p_113135_) -> {
-         p_113135_.f_113165_.forEach((p_113125_) -> {
-            map.computeIfAbsent(p_113125_, (p_113150_) -> {
+      this.f_113050_.values().forEach((p_228985_1_) -> {
+         p_228985_1_.f_113165_.forEach((p_228986_2_) -> {
+            map.computeIfAbsent(p_228986_2_, (p_241727_0_) -> {
                return Sets.newHashSet();
-            }).add(p_113135_.m_113174_());
+            }).add(p_228985_1_.m_113174_());
          });
       });
       return map;
@@ -126,14 +_,14 @@
 
    private void m_113151_() {
       Map<BlockPos, Set<UUID>> map = Maps.newHashMap();
-      this.f_113050_.values().stream().filter(BeeDebugRenderer.BeeInfo::m_113178_).forEach((p_113121_) -> {
-         map.computeIfAbsent(p_113121_.f_113162_, (p_113145_) -> {
+      this.f_113050_.values().stream().filter(BeeDebugRenderer.BeeInfo::m_113178_).forEach((p_241722_1_) -> {
+         map.computeIfAbsent(p_241722_1_.f_113162_, (p_241726_0_) -> {
             return Sets.newHashSet();
-         }).add(p_113121_.m_113174_());
+         }).add(p_241722_1_.m_113174_());
       });
-      map.entrySet().forEach((p_113118_) -> {
-         BlockPos blockpos = p_113118_.getKey();
-         Set<UUID> set = p_113118_.getValue();
+      map.entrySet().forEach((p_228978_0_) -> {
+         BlockPos blockpos = p_228978_0_.getKey();
+         Set<UUID> set = p_228978_0_.getValue();
          Set<String> set1 = set.stream().map(DebugEntityNameGenerator::m_133668_).collect(Collectors.toSet());
          int i = 1;
          m_113110_(set1.toString(), blockpos, i++, -256);
@@ -273,8 +_,8 @@
    }
 
    private Collection<UUID> m_113129_(BlockPos p_113130_) {
-      return this.f_113050_.values().stream().filter((p_113087_) -> {
-         return p_113087_.m_113175_(p_113130_);
+      return this.f_113050_.values().stream().filter((p_228970_1_) -> {
+         return p_228970_1_.m_113175_(p_113130_);
       }).map(BeeDebugRenderer.BeeInfo::m_113174_).collect(Collectors.toSet());
    }
 
@@ -283,7 +_,7 @@
 
       for(BeeDebugRenderer.BeeInfo beedebugrenderer$beeinfo : this.f_113050_.values()) {
          if (beedebugrenderer$beeinfo.f_113161_ != null && !this.f_113049_.containsKey(beedebugrenderer$beeinfo.f_113161_)) {
-            map.computeIfAbsent(beedebugrenderer$beeinfo.f_113161_, (p_113140_) -> {
+            map.computeIfAbsent(beedebugrenderer$beeinfo.f_113161_, (p_241725_0_) -> {
                return Lists.newArrayList();
             }).add(beedebugrenderer$beeinfo.m_113177_());
          }
@@ -293,8 +_,8 @@
    }
 
    private void m_113156_() {
-      DebugRenderer.m_113448_(this.f_113048_.m_91288_(), 8).ifPresent((p_113059_) -> {
-         this.f_113051_ = p_113059_.m_20148_();
+      DebugRenderer.m_113448_(this.f_113048_.m_91288_(), 8).ifPresent((p_228963_1_) -> {
+         this.f_113051_ = p_228963_1_.m_20148_();
       });
    }
 
