--- a/net/minecraft/client/renderer/debug/ChunkDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/ChunkDebugRenderer.java
@@ -5,7 +_,6 @@
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.Map;
-import java.util.Map.Entry;
 import java.util.concurrent.CompletableFuture;
 import javax.annotation.Nullable;
 import net.minecraft.Util;
@@ -56,7 +_,7 @@
          Map<ChunkPos, String> map = this.f_113366_.f_113379_.getNow((Map<ChunkPos, String>)null);
          double d1 = this.f_113363_.f_91063_.m_109153_().m_90583_().f_82480_ * 0.85D;
 
-         for(Entry<ChunkPos, String> entry : this.f_113366_.f_113378_.entrySet()) {
+         for(Map.Entry<ChunkPos, String> entry : this.f_113366_.f_113378_.entrySet()) {
             ChunkPos chunkpos = entry.getKey();
             String s = entry.getValue();
             if (map != null) {
