--- a/net/minecraft/client/renderer/debug/DebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/DebugRenderer.java
@@ -109,8 +_,8 @@
          Vec3 vec32 = vec3.m_82549_(vec31);
          AABB aabb = p_113449_.m_20191_().m_82369_(vec31).m_82400_(1.0D);
          int i = p_113450_ * p_113450_;
-         Predicate<Entity> predicate = (p_113447_) -> {
-            return !p_113447_.m_5833_() && p_113447_.m_6087_();
+         Predicate<Entity> predicate = (p_217727_0_) -> {
+            return !p_217727_0_.m_5833_() && p_217727_0_.m_6087_();
          };
          EntityHitResult entityhitresult = ProjectileUtil.m_37287_(p_113449_, vec3, vec32, aabb, predicate, (double)i);
          if (entityhitresult == null) {
