--- a/net/minecraft/client/renderer/debug/NeighborsUpdateRenderer.java
+++ b/net/minecraft/client/renderer/debug/NeighborsUpdateRenderer.java
@@ -28,7 +_,7 @@
    }
 
    public void m_113596_(long p_113597_, BlockPos p_113598_) {
-      Map<BlockPos, Integer> map = this.f_113593_.computeIfAbsent(p_113597_, (p_113606_) -> {
+      Map<BlockPos, Integer> map = this.f_113593_.computeIfAbsent(p_113597_, (p_241730_0_) -> {
          return Maps.newHashMap();
       });
       int i = map.getOrDefault(p_113598_, 0);
