--- a/com/mojang/realmsclient/util/task/DownloadTask.java
+++ b/com/mojang/realmsclient/util/task/DownloadTask.java
@@ -42,7 +_,7 @@
                return;
             }
 
-            m_90405_(new RealmsDownloadLatestWorldScreen(this.f_90317_, worlddownload, this.f_90318_, (p_90325_) -> {
+            m_90405_(new RealmsDownloadLatestWorldScreen(this.f_90317_, worlddownload, this.f_90318_, (p_238115_0_) -> {
             }));
             return;
          } catch (RetryCallException retrycallexception) {
