--- a/com/mojang/realmsclient/util/task/GetServerDetailsTask.java
+++ b/com/mojang/realmsclient/util/task/GetServerDetailsTask.java
@@ -90,12 +_,12 @@
             if (realmsserveraddress1.f_87566_ != null && realmsserveraddress1.f_87567_ != null) {
                Component component = new TranslatableComponent("mco.configure.world.resourcepack.question.line1");
                Component component1 = new TranslatableComponent("mco.configure.world.resourcepack.question.line2");
-               m_90405_(new RealmsLongConfirmationScreen((p_90340_) -> {
+               m_90405_(new RealmsLongConfirmationScreen((p_238121_2_) -> {
                   try {
-                     if (p_90340_) {
-                        Function<Throwable, Void> function = (p_90342_) -> {
+                     if (p_238121_2_) {
+                        Function<Throwable, Void> function = (p_238122_1_) -> {
                            Minecraft.m_91087_().m_91100_().m_118586_();
-                           f_90394_.error(p_90342_);
+                           f_90394_.error(p_238122_1_);
                            m_90405_(new RealmsGenericErrorScreen(new TextComponent("Failed to download resource pack!"), this.f_90328_));
                            return null;
                         };
