--- a/net/minecraft/ChatFormatting.java
+++ b/net/minecraft/ChatFormatting.java
@@ -36,10 +_,10 @@
    ITALIC("ITALIC", 'o', true),
    RESET("RESET", 'r', -1, (Integer)null);
 
-   private static final Map<String, ChatFormatting> f_126619_ = Arrays.stream(values()).collect(Collectors.toMap((p_126660_) -> {
-      return m_126662_(p_126660_.f_126621_);
-   }, (p_126652_) -> {
-      return p_126652_;
+   private static final Map<String, ChatFormatting> f_126619_ = Arrays.stream(values()).collect(Collectors.toMap((p_199746_0_) -> {
+      return m_126662_(p_199746_0_.f_126621_);
+   }, (p_199747_0_) -> {
+      return p_199747_0_;
    }));
    private static final Pattern f_126620_ = Pattern.compile("(?i)\u00a7[0-9A-FK-OR]");
    private final String f_126621_;
