--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -140,7 +_,6 @@
 import net.minecraft.world.level.block.entity.JigsawBlockEntity;
 import net.minecraft.world.level.block.entity.SignBlockEntity;
 import net.minecraft.world.level.block.entity.StructureBlockEntity;
-import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.phys.AABB;
 import net.minecraft.world.phys.BlockHitResult;
@@ -148,7 +_,6 @@
 import net.minecraft.world.phys.shapes.BooleanOp;
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
-import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
@@ -290,7 +_,7 @@
    }
 
    public void m_9942_(Component p_9943_) {
-      this.f_9742_.m_129514_(new ClientboundDisconnectPacket(p_9943_), (p_9828_) -> {
+      this.f_9742_.m_129514_(new ClientboundDisconnectPacket(p_9943_), (p_210161_2_) -> {
          this.f_9742_.m_129507_(p_9943_);
       });
       this.f_9742_.m_129540_();
@@ -299,9 +_,9 @@
 
    private <T> void m_9801_(T p_9802_, Consumer<T> p_9803_, BiFunction<TextFilter, T, CompletableFuture<Optional<T>>> p_9804_) {
       BlockableEventLoop<?> blockableeventloop = this.f_9743_.m_9236_().m_7654_();
-      Consumer<T> consumer = (p_9941_) -> {
+      Consumer<T> consumer = (p_244545_2_) -> {
          if (this.m_6198_().m_129536_()) {
-            p_9803_.accept(p_9941_);
+            p_9803_.accept(p_244545_2_);
          } else {
             f_9744_.debug("Ignoring packet due to disconnection");
          }
@@ -309,8 +_,8 @@
       };
       TextFilter textfilter = this.f_9743_.m_8967_();
       if (textfilter != null) {
-         p_9804_.apply(textfilter, p_9802_).thenAcceptAsync((p_9823_) -> {
-            p_9823_.ifPresent(consumer);
+         p_9804_.apply(textfilter, p_9802_).thenAcceptAsync((p_244539_1_) -> {
+            p_244539_1_.ifPresent(consumer);
          }, blockableeventloop);
       } else {
          blockableeventloop.execute(() -> {
@@ -392,9 +_,11 @@
             }
 
             entity.m_19890_(d3, d4, d5, f, f1);
+            this.f_9743_.m_19890_(d3, d4, d5, this.f_9743_.f_19857_, this.f_9743_.f_19858_); // Forge - Resync player position on vehicle moving
             boolean flag2 = serverlevel.m_45756_(entity, entity.m_20191_().m_82406_(0.0625D));
             if (flag && (flag1 || !flag2)) {
                entity.m_19890_(d0, d1, d2, f, f1);
+               this.f_9743_.m_19890_(d3, d4, d5, this.f_9743_.f_19857_, this.f_9743_.f_19858_); // Forge - Resync player position on vehicle moving
                this.f_9742_.m_129512_(new ClientboundMoveVehiclePacket(entity));
                return;
             }
@@ -411,7 +_,7 @@
    }
 
    private boolean m_9793_(Entity p_9794_) {
-      return p_9794_.f_19853_.m_45556_(p_9794_.m_20191_().m_82400_(0.0625D).m_82363_(0.0D, -0.55D, 0.0D)).allMatch(BlockBehaviour.BlockStateBase::m_60795_);
+      return BlockPos.m_121921_(p_9794_.m_20191_().m_82400_(0.0625D).m_82363_(0.0D, -0.55D, 0.0D)).allMatch(b -> p_9794_.f_19853_.m_8055_(b).isAir(p_9794_.f_19853_, b));
    }
 
    public void m_7376_(ServerboundAcceptTeleportationPacket p_9835_) {
@@ -460,8 +_,8 @@
       }
 
       ParseResults<CommandSourceStack> parseresults = this.f_9745_.m_129892_().m_82094_().parse(stringreader, this.f_9743_.m_20203_());
-      this.f_9745_.m_129892_().m_82094_().getCompletionSuggestions(parseresults).thenAccept((p_9850_) -> {
-         this.f_9742_.m_129512_(new ClientboundCommandSuggestionsPacket(p_9847_.m_133901_(), p_9850_));
+      this.f_9745_.m_129892_().m_82094_().getCompletionSuggestions(parseresults).thenAccept((p_195519_2_) -> {
+         this.f_9742_.m_129512_(new ClientboundCommandSuggestionsPacket(p_9847_.m_133901_(), p_195519_2_));
       });
    }
 
@@ -691,10 +_,10 @@
 
             int j = p_9862_.m_134013_();
             if (Inventory.m_36045_(j) || j == 40) {
-               this.m_9815_(list, flag ? (p_9936_) -> {
-                  this.m_9805_(p_9936_.get(0), p_9936_.subList(1, p_9936_.size()), j);
-               } : (p_9789_) -> {
-                  this.m_9812_(p_9789_, j);
+               this.m_9815_(list, flag ? (p_244543_2_) -> {
+                  this.m_9805_(p_244543_2_.get(0), p_244543_2_.subList(1, p_244543_2_.size()), j);
+               } : (p_244531_2_) -> {
+                  this.m_9812_(p_244531_2_, j);
                });
             }
          }
@@ -863,12 +_,12 @@
    }
 
    private boolean m_9795_(LevelReader p_9796_, AABB p_9797_) {
-      Stream<VoxelShape> stream = p_9796_.m_7786_(this.f_9743_, this.f_9743_.m_20191_().m_82406_((double)1.0E-5F), (p_9938_) -> {
+      Stream<VoxelShape> stream = p_9796_.m_7786_(this.f_9743_, this.f_9743_.m_20191_().m_82406_((double)1.0E-5F), (p_241167_0_) -> {
          return true;
       });
       VoxelShape voxelshape = Shapes.m_83064_(p_9797_.m_82406_((double)1.0E-5F));
-      return stream.anyMatch((p_9800_) -> {
-         return !Shapes.m_83157_(p_9800_, voxelshape, BooleanOp.f_82689_);
+      return stream.anyMatch((p_241164_1_) -> {
+         return !Shapes.m_83157_(p_241164_1_, voxelshape, BooleanOp.f_82689_);
       });
    }
 
@@ -951,7 +_,9 @@
       Direction direction = blockhitresult.m_82434_();
       this.f_9743_.m_9243_();
       if (blockpos.m_123342_() < this.f_9745_.m_129917_()) {
-         if (this.f_9766_ == null && this.f_9743_.m_20275_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_() + 0.5D, (double)blockpos.m_123343_() + 0.5D) < 64.0D && serverlevel.m_7966_(this.f_9743_, blockpos)) {
+         double dist = f_9743_.m_21051_(net.minecraftforge.common.ForgeMod.REACH_DISTANCE.get()).m_22135_() + 3;
+         dist *= dist;
+         if (this.f_9766_ == null && this.f_9743_.m_20275_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_() + 0.5D, (double)blockpos.m_123343_() + 0.5D) < dist && serverlevel.m_7966_(this.f_9743_, blockpos)) {
             InteractionResult interactionresult = this.f_9743_.f_8941_.m_7179_(this.f_9743_, serverlevel, itemstack, interactionhand, blockhitresult);
             if (direction == Direction.UP && !interactionresult.m_19077_() && blockpos.m_123342_() >= this.f_9745_.m_129917_() - 1 && m_9790_(this.f_9743_, itemstack)) {
                Component component = (new TranslatableComponent("build.tooHigh", this.f_9745_.m_129917_())).m_130940_(ChatFormatting.RED);
@@ -1072,7 +_,7 @@
    }
 
    public void m_7388_(ServerboundChatPacket p_9841_) {
-      String s = StringUtils.normalizeSpace(p_9841_.m_133837_());
+      String s = org.apache.commons.lang3.StringUtils.normalizeSpace(p_9841_.m_133837_());
       if (s.startsWith("/")) {
          PacketUtils.m_131359_(p_9841_, this, this.f_9743_.m_9236_());
          this.m_9954_(s);
@@ -1098,7 +_,9 @@
          if (p_9955_.startsWith("/")) {
             this.m_9957_(p_9955_);
          } else {
-            Component component = new TranslatableComponent("chat.type.text", this.f_9743_.m_5446_(), p_9955_);
+            Component component = new TranslatableComponent("chat.type.text", this.f_9743_.m_5446_(), net.minecraftforge.common.ForgeHooks.newChatWithLinks(p_9955_));
+            component = net.minecraftforge.common.ForgeHooks.onServerChatEvent(this, p_9955_, component);
+            if (component == null) return;
             this.f_9745_.m_6846_().m_11264_(component, ChatType.CHAT, this.f_9743_.m_20148_());
          }
 
@@ -1188,6 +_,7 @@
             if (p_9866_.m_134056_() == ServerboundInteractPacket.Action.INTERACT) {
                optional = Optional.of(this.f_9743_.m_36157_(entity, interactionhand));
             } else if (p_9866_.m_134056_() == ServerboundInteractPacket.Action.INTERACT_AT) {
+               if (net.minecraftforge.common.ForgeHooks.onInteractEntityAt(f_9743_, entity, p_9866_.m_134060_(), interactionhand) != null) return;
                optional = Optional.of(entity.m_7111_(this.f_9743_, p_9866_.m_134060_(), interactionhand));
             } else if (p_9866_.m_134056_() == ServerboundInteractPacket.Action.ATTACK) {
                if (entity instanceof ItemEntity || entity instanceof ExperienceOrb || entity instanceof AbstractArrow || entity == this.f_9743_) {
@@ -1285,8 +_,8 @@
       PacketUtils.m_131359_(p_9882_, this, this.f_9743_.m_9236_());
       this.f_9743_.m_9243_();
       if (!this.f_9743_.m_5833_() && this.f_9743_.f_36096_.f_38840_ == p_9882_.m_134249_() && this.f_9743_.f_36096_.m_38949_(this.f_9743_) && this.f_9743_.f_36096_ instanceof RecipeBookMenu) {
-         this.f_9745_.m_129894_().m_44043_(p_9882_.m_134252_()).ifPresent((p_9885_) -> {
-            ((RecipeBookMenu)this.f_9743_.f_36096_).m_6951_(p_9882_.m_134253_(), p_9885_, this.f_9743_);
+         this.f_9745_.m_129894_().m_44043_(p_9882_.m_134252_()).ifPresent((p_241165_2_) -> {
+            ((RecipeBookMenu)this.f_9743_.f_36096_).m_6951_(p_9882_.m_134253_(), p_241165_2_, this.f_9743_);
          });
       }
    }
@@ -1349,8 +_,8 @@
 
    public void m_5527_(ServerboundSignUpdatePacket p_9921_) {
       List<String> list = Stream.of(p_9921_.m_134663_()).map(ChatFormatting::m_126649_).collect(Collectors.toList());
-      this.m_9815_(list, (p_9952_) -> {
-         this.m_9922_(p_9921_, p_9952_);
+      this.m_9815_(list, (p_244547_2_) -> {
+         this.m_9922_(p_9921_, p_244547_2_);
       });
    }
 
@@ -1403,6 +_,8 @@
    }
 
    public void m_7423_(ServerboundCustomPayloadPacket p_9860_) {
+      PacketUtils.m_131359_(p_9860_, this, this.f_9743_.m_9236_());
+      net.minecraftforge.fml.network.NetworkHooks.onCustomPayload(p_9860_, this.f_9742_);
    }
 
    public void m_7477_(ServerboundChangeDifficultyPacket p_9839_) {
