--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -392,9 +_,11 @@
             }
 
             entity.m_19890_(d3, d4, d5, f, f1);
+            this.f_9743_.m_19890_(d3, d4, d5, this.f_9743_.f_19857_, this.f_9743_.f_19858_); // Forge - Resync player position on vehicle moving
             boolean flag2 = serverlevel.m_45756_(entity, entity.m_20191_().m_82406_(0.0625D));
             if (flag && (flag1 || !flag2)) {
                entity.m_19890_(d0, d1, d2, f, f1);
+               this.f_9743_.m_19890_(d3, d4, d5, this.f_9743_.f_19857_, this.f_9743_.f_19858_); // Forge - Resync player position on vehicle moving
                this.f_9742_.m_129512_(new ClientboundMoveVehiclePacket(entity));
                return;
             }
@@ -411,7 +_,7 @@
    }
 
    private boolean m_9793_(Entity p_9794_) {
-      return p_9794_.f_19853_.m_45556_(p_9794_.m_20191_().m_82400_(0.0625D).m_82363_(0.0D, -0.55D, 0.0D)).allMatch(BlockBehaviour.BlockStateBase::m_60795_);
+      return BlockPos.m_121921_(p_9794_.m_20191_().m_82400_(0.0625D).m_82363_(0.0D, -0.55D, 0.0D)).allMatch(b -> p_9794_.f_19853_.m_8055_(b).isAir(p_9794_.f_19853_, b));
    }
 
    public void m_7376_(ServerboundAcceptTeleportationPacket p_9835_) {
@@ -951,7 +_,9 @@
       Direction direction = blockhitresult.m_82434_();
       this.f_9743_.m_9243_();
       if (blockpos.m_123342_() < this.f_9745_.m_129917_()) {
-         if (this.f_9766_ == null && this.f_9743_.m_20275_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_() + 0.5D, (double)blockpos.m_123343_() + 0.5D) < 64.0D && serverlevel.m_7966_(this.f_9743_, blockpos)) {
+         double dist = f_9743_.m_21051_(net.minecraftforge.common.ForgeMod.REACH_DISTANCE.get()).m_22135_() + 3;
+         dist *= dist;
+         if (this.f_9766_ == null && this.f_9743_.m_20275_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_() + 0.5D, (double)blockpos.m_123343_() + 0.5D) < dist && serverlevel.m_7966_(this.f_9743_, blockpos)) {
             InteractionResult interactionresult = this.f_9743_.f_8941_.m_7179_(this.f_9743_, serverlevel, itemstack, interactionhand, blockhitresult);
             if (direction == Direction.UP && !interactionresult.m_19077_() && blockpos.m_123342_() >= this.f_9745_.m_129917_() - 1 && m_9790_(this.f_9743_, itemstack)) {
                Component component = (new TranslatableComponent("build.tooHigh", this.f_9745_.m_129917_())).m_130940_(ChatFormatting.RED);
@@ -1072,7 +_,7 @@
    }
 
    public void m_7388_(ServerboundChatPacket p_9841_) {
-      String s = StringUtils.normalizeSpace(p_9841_.m_133837_());
+      String s = org.apache.commons.lang3.StringUtils.normalizeSpace(p_9841_.m_133837_());
       if (s.startsWith("/")) {
          PacketUtils.m_131359_(p_9841_, this, this.f_9743_.m_9236_());
          this.m_9954_(s);
@@ -1098,7 +_,9 @@
          if (p_9955_.startsWith("/")) {
             this.m_9957_(p_9955_);
          } else {
-            Component component = new TranslatableComponent("chat.type.text", this.f_9743_.m_5446_(), p_9955_);
+            Component component = new TranslatableComponent("chat.type.text", this.f_9743_.m_5446_(), net.minecraftforge.common.ForgeHooks.newChatWithLinks(p_9955_));
+            component = net.minecraftforge.common.ForgeHooks.onServerChatEvent(this, p_9955_, component);
+            if (component == null) return;
             this.f_9745_.m_6846_().m_11264_(component, ChatType.CHAT, this.f_9743_.m_20148_());
          }
 
@@ -1188,6 +_,7 @@
             if (p_9866_.m_134056_() == ServerboundInteractPacket.Action.INTERACT) {
                optional = Optional.of(this.f_9743_.m_36157_(entity, interactionhand));
             } else if (p_9866_.m_134056_() == ServerboundInteractPacket.Action.INTERACT_AT) {
+               if (net.minecraftforge.common.ForgeHooks.onInteractEntityAt(f_9743_, entity, p_9866_.m_134060_(), interactionhand) != null) return;
                optional = Optional.of(entity.m_7111_(this.f_9743_, p_9866_.m_134060_(), interactionhand));
             } else if (p_9866_.m_134056_() == ServerboundInteractPacket.Action.ATTACK) {
                if (entity instanceof ItemEntity || entity instanceof ExperienceOrb || entity instanceof AbstractArrow || entity == this.f_9743_) {
@@ -1403,6 +_,8 @@
    }
 
    public void m_7423_(ServerboundCustomPayloadPacket p_9860_) {
+      PacketUtils.m_131359_(p_9860_, this, this.f_9743_.m_9236_());
+      net.minecraftforge.fml.network.NetworkHooks.onCustomPayload(p_9860_, this.f_9742_);
    }
 
    public void m_7477_(ServerboundChangeDifficultyPacket p_9839_) {
