--- a/net/minecraft/util/ClassInstanceMultiMap.java
+++ b/net/minecraft/util/ClassInstanceMultiMap.java
@@ -56,10 +_,10 @@
       if (!this.f_13528_.isAssignableFrom(p_13534_)) {
          throw new IllegalArgumentException("Don't know how to search for " + p_13534_);
       } else {
-         List<T> list = this.f_13527_.computeIfAbsent(p_13534_, (p_13538_) -> {
-            return this.f_13529_.stream().filter(p_13538_::isInstance).collect(Collectors.toList());
+         List<T> list = this.f_13527_.computeIfAbsent(p_13534_, (p_219791_1_) -> {
+            return this.f_13529_.stream().filter(p_219791_1_::isInstance).collect(Collectors.toList());
          });
-         return (Collection<S>)Collections.unmodifiableCollection(list);
+         return (Collection<S>) Collections.unmodifiableCollection(list);
       }
    }
 
