--- a/net/minecraft/util/FormattedCharSequence.java
+++ b/net/minecraft/util/FormattedCharSequence.java
@@ -9,7 +_,7 @@
 
 @FunctionalInterface
 public interface FormattedCharSequence {
-   FormattedCharSequence f_13691_ = (p_13704_) -> {
+   FormattedCharSequence f_13691_ = (p_242236_0_) -> {
       return true;
    };
 
@@ -18,29 +_,29 @@
 
    @OnlyIn(Dist.CLIENT)
    static FormattedCharSequence m_13693_(int p_13694_, Style p_13695_) {
-      return (p_13730_) -> {
-         return p_13730_.m_6411_(0, p_13695_, p_13694_);
+      return (p_242243_2_) -> {
+         return p_242243_2_.m_6411_(0, p_13695_, p_13694_);
       };
    }
 
    @OnlyIn(Dist.CLIENT)
    static FormattedCharSequence m_13714_(String p_13715_, Style p_13716_) {
-      return p_13715_.isEmpty() ? f_13691_ : (p_13739_) -> {
-         return StringDecomposer.m_14317_(p_13715_, p_13716_, p_13739_);
+      return p_13715_.isEmpty() ? f_13691_ : (p_242245_2_) -> {
+         return StringDecomposer.m_14317_(p_13715_, p_13716_, p_242245_2_);
       };
    }
 
    @OnlyIn(Dist.CLIENT)
    static FormattedCharSequence m_13740_(String p_13741_, Style p_13742_, Int2IntFunction p_13743_) {
-      return p_13741_.isEmpty() ? f_13691_ : (p_13721_) -> {
-         return StringDecomposer.m_14337_(p_13741_, p_13742_, m_13705_(p_13721_, p_13743_));
+      return p_13741_.isEmpty() ? f_13691_ : (p_242240_3_) -> {
+         return StringDecomposer.m_14337_(p_13741_, p_13742_, m_13705_(p_242240_3_, p_13743_));
       };
    }
 
    @OnlyIn(Dist.CLIENT)
    static FormattedCharSink m_13705_(FormattedCharSink p_13706_, Int2IntFunction p_13707_) {
-      return (p_13711_, p_13712_, p_13713_) -> {
-         return p_13706_.m_6411_(p_13711_, p_13712_, p_13707_.apply(Integer.valueOf(p_13713_)));
+      return (p_242238_2_, p_242238_3_, p_242238_4_) -> {
+         return p_13706_.m_6411_(p_242238_2_, p_242238_3_, p_13707_.apply(Integer.valueOf(p_242238_4_)));
       };
    }
 
@@ -66,16 +_,16 @@
 
    @OnlyIn(Dist.CLIENT)
    static FormattedCharSequence m_13733_(FormattedCharSequence p_13734_, FormattedCharSequence p_13735_) {
-      return (p_13702_) -> {
-         return p_13734_.m_13731_(p_13702_) && p_13735_.m_13731_(p_13702_);
+      return (p_242235_2_) -> {
+         return p_13734_.m_13731_(p_242235_2_) && p_13735_.m_13731_(p_242235_2_);
       };
    }
 
    @OnlyIn(Dist.CLIENT)
    static FormattedCharSequence m_13744_(List<FormattedCharSequence> p_13745_) {
-      return (p_13726_) -> {
+      return (p_242242_1_) -> {
          for(FormattedCharSequence formattedcharsequence : p_13745_) {
-            if (!formattedcharsequence.m_13731_(p_13726_)) {
+            if (!formattedcharsequence.m_13731_(p_242242_1_)) {
                return false;
             }
          }
