--- a/net/minecraft/util/StringDecomposer.java
+++ b/net/minecraft/util/StringDecomposer.java
@@ -129,15 +_,15 @@
    }
 
    public static boolean m_14328_(FormattedText p_14329_, Style p_14330_, FormattedCharSink p_14331_) {
-      return !p_14329_.m_7451_((p_14302_, p_14303_) -> {
-         return m_14306_(p_14303_, 0, p_14302_, p_14331_) ? Optional.empty() : f_14298_;
+      return !p_14329_.m_7451_((p_238337_1_, p_238337_2_) -> {
+         return m_14306_(p_238337_2_, 0, p_238337_1_, p_14331_) ? Optional.empty() : f_14298_;
       }, p_14330_).isPresent();
    }
 
    public static String m_14304_(String p_14305_) {
       StringBuilder stringbuilder = new StringBuilder();
-      m_14317_(p_14305_, Style.f_131099_, (p_14343_, p_14344_, p_14345_) -> {
-         stringbuilder.appendCodePoint(p_14345_);
+      m_14317_(p_14305_, Style.f_131099_, (p_238342_1_, p_238342_2_, p_238342_3_) -> {
+         stringbuilder.appendCodePoint(p_238342_3_);
          return true;
       });
       return stringbuilder.toString();
@@ -145,8 +_,8 @@
 
    public static String m_14326_(FormattedText p_14327_) {
       StringBuilder stringbuilder = new StringBuilder();
-      m_14328_(p_14327_, Style.f_131099_, (p_14323_, p_14324_, p_14325_) -> {
-         stringbuilder.appendCodePoint(p_14325_);
+      m_14328_(p_14327_, Style.f_131099_, (p_244781_1_, p_244781_2_, p_244781_3_) -> {
+         stringbuilder.appendCodePoint(p_244781_3_);
          return true;
       });
       return stringbuilder.toString();
