--- a/net/minecraft/util/StringUtil.java
+++ b/net/minecraft/util/StringUtil.java
@@ -4,7 +_,6 @@
 import javax.annotation.Nullable;
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
-import org.apache.commons.lang3.StringUtils;
 
 public class StringUtil {
    private static final Pattern f_14402_ = Pattern.compile("(?i)\\u00A7[0-9A-FK-OR]");
@@ -23,6 +_,6 @@
    }
 
    public static boolean m_14408_(@Nullable String p_14409_) {
-      return StringUtils.isEmpty(p_14409_);
+      return org.apache.commons.lang3.StringUtils.isEmpty(p_14409_);
    }
 }
