--- a/net/minecraft/util/StringRepresentable.java
+++ b/net/minecraft/util/StringRepresentable.java
@@ -19,8 +_,8 @@
 
    static <E extends Enum<E> & StringRepresentable> Codec<E> m_14350_(Supplier<E[]> p_14351_, Function<? super String, ? extends E> p_14352_) {
       E[] ae = p_14351_.get();
-      return m_14353_(Enum::ordinal, (p_14361_) -> {
-         return ae[p_14361_];
+      return m_14353_(Enum::ordinal, (p_233026_1_) -> {
+         return ae[p_233026_1_];
       }, p_14352_);
    }
 
@@ -31,18 +_,18 @@
          }
 
          public <T> DataResult<Pair<E, T>> decode(DynamicOps<T> p_14390_, T p_14391_) {
-            return p_14390_.compressMaps() ? p_14390_.getNumberValue(p_14391_).flatMap((p_14385_) -> {
-               return Optional.ofNullable(p_14355_.apply(p_14385_.intValue())).map(DataResult::success).orElseGet(() -> {
-                  return DataResult.error("Unknown element id: " + p_14385_);
-               });
-            }).map((p_14388_) -> {
-               return Pair.of(p_14388_, p_14390_.empty());
-            }) : p_14390_.getStringValue(p_14391_).flatMap((p_14382_) -> {
-               return Optional.ofNullable(p_14356_.apply(p_14382_)).map(DataResult::success).orElseGet(() -> {
-                  return DataResult.error("Unknown element name: " + p_14382_);
-               });
-            }).map((p_14375_) -> {
-               return Pair.of(p_14375_, p_14390_.empty());
+            return p_14390_.compressMaps() ? p_14390_.getNumberValue(p_14391_).flatMap((p_233034_1_) -> {
+               return Optional.ofNullable(p_14355_.apply(p_233034_1_.intValue())).map(DataResult::success).orElseGet(() -> {
+                  return DataResult.error("Unknown element id: " + p_233034_1_);
+               });
+            }).map((p_233035_1_) -> {
+               return Pair.of(p_233035_1_, p_14390_.empty());
+            }) : p_14390_.getStringValue(p_14391_).flatMap((p_233033_1_) -> {
+               return Optional.ofNullable(p_14356_.apply(p_233033_1_)).map(DataResult::success).orElseGet(() -> {
+                  return DataResult.error("Unknown element name: " + p_233033_1_);
+               });
+            }).map((p_233030_1_) -> {
+               return Pair.of(p_233030_1_, p_14390_.empty());
             });
          }
 
