--- a/net/minecraft/client/renderer/block/model/multipart/Condition.java
+++ b/net/minecraft/client/renderer/block/model/multipart/Condition.java
@@ -10,13 +_,13 @@
 @FunctionalInterface
 @OnlyIn(Dist.CLIENT)
 public interface Condition {
-   Condition f_111922_ = (p_111932_) -> {
-      return (p_111930_) -> {
+   Condition f_111922_ = (p_223290_0_) -> {
+      return (p_223289_0_) -> {
          return true;
       };
    };
-   Condition f_111923_ = (p_111928_) -> {
-      return (p_111926_) -> {
+   Condition f_111923_ = (p_223288_0_) -> {
+      return (p_223287_0_) -> {
          return false;
       };
    };
