--- a/net/minecraft/client/renderer/block/model/multipart/MultiPart.java
+++ b/net/minecraft/client/renderer/block/model/multipart/MultiPart.java
@@ -72,14 +_,14 @@
    }
 
    public Collection<ResourceLocation> m_7970_() {
-      return this.m_111967_().stream().flatMap((p_111969_) -> {
-         return p_111969_.m_112020_().m_7970_().stream();
+      return this.m_111967_().stream().flatMap((p_209563_0_) -> {
+         return p_209563_0_.m_112020_().m_7970_().stream();
       }).collect(Collectors.toSet());
    }
 
    public Collection<Material> m_5500_(Function<ResourceLocation, UnbakedModel> p_111976_, Set<Pair<String, String>> p_111977_) {
-      return this.m_111967_().stream().flatMap((p_111981_) -> {
-         return p_111981_.m_112020_().m_5500_(p_111976_, p_111977_).stream();
+      return this.m_111967_().stream().flatMap((p_228832_2_) -> {
+         return p_228832_2_.m_112020_().m_5500_(p_111976_, p_111977_).stream();
       }).collect(Collectors.toSet());
    }
 
