--- a/net/minecraft/client/renderer/block/model/multipart/Selector.java
+++ b/net/minecraft/client/renderer/block/model/multipart/Selector.java
@@ -71,13 +_,13 @@
             throw new JsonParseException("No elements found in selector");
          } else if (set.size() == 1) {
             if (p_112034_.has("OR")) {
-               List<Condition> list1 = Streams.stream(GsonHelper.m_13933_(p_112034_, "OR")).map((p_112038_) -> {
-                  return m_112033_(p_112038_.getAsJsonObject());
+               List<Condition> list1 = Streams.stream(GsonHelper.m_13933_(p_112034_, "OR")).map((p_200692_0_) -> {
+                  return m_112033_(p_200692_0_.getAsJsonObject());
                }).collect(Collectors.toList());
                return new OrCondition(list1);
             } else if (p_112034_.has("AND")) {
-               List<Condition> list = Streams.stream(GsonHelper.m_13933_(p_112034_, "AND")).map((p_112028_) -> {
-                  return m_112033_(p_112028_.getAsJsonObject());
+               List<Condition> list = Streams.stream(GsonHelper.m_13933_(p_112034_, "AND")).map((p_200691_0_) -> {
+                  return m_112033_(p_200691_0_.getAsJsonObject());
                }).collect(Collectors.toList());
                return new AndCondition(list);
             } else {
