--- a/net/minecraft/core/dispenser/DispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/DispenseItemBehavior.java
@@ -375,8 +_,9 @@
                level.m_46597_(blockpos, BaseFireBlock.m_49245_(level, blockpos));
             } else if (CampfireBlock.m_51321_(blockstate)) {
                level.m_46597_(blockpos, blockstate.m_61124_(BlockStateProperties.f_61443_, Boolean.valueOf(true)));
-            } else if (blockstate.m_60734_() instanceof TntBlock) {
-               TntBlock.m_57433_(level, blockpos);
+            } else if (blockstate.isFlammable(level, blockpos, p_123412_.m_6414_().m_61143_(DispenserBlock.f_52659_).m_122424_())) {
+               blockstate.catchFire(level, blockpos, p_123412_.m_6414_().m_61143_(DispenserBlock.f_52659_).m_122424_(), null);
+               if (blockstate.m_60734_() instanceof TntBlock)
                level.m_7471_(blockpos, false);
             } else {
                this.m_123573_(false);
