--- a/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -83,8 +_,8 @@
 
    private List<Entity> m_58759_(BlockState p_58760_, BeehiveBlockEntity.BeeReleaseStatus p_58761_) {
       List<Entity> list = Lists.newArrayList();
-      this.f_58732_.removeIf((p_58766_) -> {
-         return this.m_58754_(p_58760_, p_58766_, list, p_58761_);
+      this.f_58732_.removeIf((p_226966_4_) -> {
+         return this.m_58754_(p_58760_, p_226966_4_, list, p_58761_);
       });
       return list;
    }
@@ -147,8 +_,8 @@
          if (flag && p_58758_ != BeehiveBlockEntity.BeeReleaseStatus.EMERGENCY) {
             return false;
          } else {
-            Entity entity = EntityType.m_20645_(compoundtag, this.f_58857_, (p_58740_) -> {
-               return p_58740_;
+            Entity entity = EntityType.m_20645_(compoundtag, this.f_58857_, (p_226960_0_) -> {
+               return p_226960_0_;
             });
             if (entity != null) {
                if (!entity.m_6095_().m_20609_(EntityTypeTags.f_13122_)) {
