--- a/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
@@ -60,8 +_,8 @@
             int j = this.f_59043_[i]++;
             if (this.f_59043_[i] >= this.f_59044_[i]) {
                Container container = new SimpleContainer(itemstack);
-               ItemStack itemstack1 = this.f_58857_.m_7465_().m_44015_(RecipeType.f_44111_, container, this.f_58857_).map((p_59050_) -> {
-                  return p_59050_.m_5874_(container);
+               ItemStack itemstack1 = this.f_58857_.m_7465_().m_44015_(RecipeType.f_44111_, container, this.f_58857_).map((p_213979_1_) -> {
+                  return p_213979_1_.m_5874_(container);
                }).orElse(itemstack);
                BlockPos blockpos = this.m_58899_();
                Containers.m_18992_(this.f_58857_, (double)blockpos.m_123341_(), (double)blockpos.m_123342_(), (double)blockpos.m_123343_(), itemstack1);
