--- a/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
@@ -130,10 +_,8 @@
                   BlockPos blockpos1 = this.f_58858_.m_7918_(j1, k1, l1);
                   BlockState blockstate = this.f_58857_.m_8055_(blockpos1);
 
-                  for(Block block : f_59184_) {
-                     if (blockstate.m_60713_(block)) {
-                        this.f_59188_.add(blockpos1);
-                     }
+                  if (blockstate.isConduitFrame(this.f_58857_, blockpos1, m_58899_())) {
+                     this.f_59188_.add(blockpos1);
                   }
                }
             }
