--- a/net/minecraft/world/level/block/entity/HopperBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/HopperBlockEntity.java
@@ -133,6 +_,7 @@
    }
 
    private boolean m_59405_() {
+      if (net.minecraftforge.items.VanillaInventoryCodeHooks.insertHook(this)) return true;
       Container container = this.m_59406_();
       if (container == null) {
          return false;
@@ -177,6 +_,8 @@
    }
 
    public static boolean m_59352_(Hopper p_59353_) {
+      Boolean ret = net.minecraftforge.items.VanillaInventoryCodeHooks.extractHook(p_59353_);
+      if (ret != null) return ret;
       Container container = m_59393_(p_59353_);
       if (container != null) {
          Direction direction = Direction.DOWN;
@@ -325,7 +_,7 @@
       Block block = blockstate.m_60734_();
       if (block instanceof WorldlyContainerHolder) {
          container = ((WorldlyContainerHolder)block).m_5840_(blockstate, p_59348_, blockpos);
-      } else if (block.m_60588_()) {
+      } else if (blockstate.hasTileEntity()) {
          BlockEntity blockentity = p_59348_.m_7702_(blockpos);
          if (blockentity instanceof Container) {
             container = (Container)blockentity;
@@ -403,5 +_,14 @@
 
    protected AbstractContainerMenu m_6555_(int p_59312_, Inventory p_59313_) {
       return new HopperMenu(p_59312_, p_59313_, this);
+   }
+
+   @Override
+   protected net.minecraftforge.items.IItemHandler createUnSidedHandler() {
+      return new net.minecraftforge.items.VanillaHopperItemHandler(this);
+   }
+
+   public long getLastUpdateTime() {
+      return this.f_59303_;
    }
 }
