--- a/net/minecraft/world/level/block/entity/JigsawBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/JigsawBlockEntity.java
@@ -152,8 +_,8 @@
       }
 
       public static Optional<JigsawBlockEntity.JointType> m_59457_(String p_59458_) {
-         return Arrays.stream(values()).filter((p_59461_) -> {
-            return p_59461_.m_7912_().equals(p_59458_);
+         return Arrays.stream(values()).filter((p_235674_1_) -> {
+            return p_235674_1_.m_7912_().equals(p_59458_);
          }).findFirst();
       }
    }
