--- a/net/minecraft/world/level/block/entity/StructureBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/StructureBlockEntity.java
@@ -313,8 +_,8 @@
    }
 
    private List<StructureBlockEntity> m_59870_(List<StructureBlockEntity> p_59871_) {
-      Predicate<StructureBlockEntity> predicate = (p_59856_) -> {
-         return p_59856_.f_59819_ == StructureMode.CORNER && Objects.equals(this.f_59812_, p_59856_.f_59812_);
+      Predicate<StructureBlockEntity> predicate = (p_200665_1_) -> {
+         return p_200665_1_.f_59819_ == StructureMode.CORNER && Objects.equals(this.f_59812_, p_200665_1_.f_59812_);
       };
       return p_59871_.stream().filter(predicate).collect(Collectors.toList());
    }
