--- a/net/minecraft/world/entity/npc/VillagerData.java
+++ b/net/minecraft/world/entity/npc/VillagerData.java
@@ -8,18 +_,18 @@
 
 public class VillagerData {
    private static final int[] f_35551_ = new int[]{0, 10, 70, 150, 250};
-   public static final Codec<VillagerData> f_35550_ = RecordCodecBuilder.create((p_35570_) -> {
-      return p_35570_.group(Registry.f_122868_.fieldOf("type").orElseGet(() -> {
+   public static final Codec<VillagerData> f_35550_ = RecordCodecBuilder.create((p_234556_0_) -> {
+      return p_234556_0_.group(Registry.f_122868_.fieldOf("type").orElseGet(() -> {
          return VillagerType.f_35821_;
-      }).forGetter((p_35580_) -> {
-         return p_35580_.f_35552_;
+      }).forGetter((p_234558_0_) -> {
+         return p_234558_0_.f_35552_;
       }), Registry.f_122869_.fieldOf("profession").orElseGet(() -> {
          return VillagerProfession.f_35585_;
-      }).forGetter((p_35575_) -> {
-         return p_35575_.f_35553_;
-      }), Codec.INT.fieldOf("level").orElse(1).forGetter((p_35564_) -> {
-         return p_35564_.f_35554_;
-      })).apply(p_35570_, VillagerData::new);
+      }).forGetter((p_234557_0_) -> {
+         return p_234557_0_.f_35553_;
+      }), Codec.INT.fieldOf("level").orElse(1).forGetter((p_234555_0_) -> {
+         return p_234555_0_.f_35554_;
+      })).apply(p_234556_0_, VillagerData::new);
    });
    private final VillagerType f_35552_;
    private final VillagerProfession f_35553_;
