--- a/net/minecraft/world/entity/npc/WanderingTraderSpawner.java
+++ b/net/minecraft/world/entity/npc/WanderingTraderSpawner.java
@@ -87,7 +_,7 @@
          BlockPos blockpos = player.m_20183_();
          int i = 48;
          PoiManager poimanager = p_35916_.m_8904_();
-         Optional<BlockPos> optional = poimanager.m_27186_(PoiType.f_27347_.m_27392_(), (p_35933_) -> {
+         Optional<BlockPos> optional = poimanager.m_27186_(PoiType.f_27347_.m_27392_(), (p_221241_0_) -> {
             return true;
          }, blockpos, 48, PoiManager.Occupancy.ANY);
          BlockPos blockpos1 = optional.orElse(blockpos);
