--- a/net/minecraft/world/level/newbiome/layer/traits/AreaTransformer1.java
+++ b/net/minecraft/world/level/newbiome/layer/traits/AreaTransformer1.java
@@ -8,9 +_,9 @@
    default <R extends Area> AreaFactory<R> m_77002_(BigContext<R> p_77003_, AreaFactory<R> p_77004_) {
       return () -> {
          R r = p_77004_.m_76488_();
-         return p_77003_.m_6678_((p_77008_, p_77009_) -> {
-            p_77003_.m_6687_((long)p_77008_, (long)p_77009_);
-            return this.m_7591_(p_77003_, r, p_77008_, p_77009_);
+         return p_77003_.m_6678_((p_202711_3_, p_202711_4_) -> {
+            p_77003_.m_6687_((long)p_202711_3_, (long)p_202711_4_);
+            return this.m_7591_(p_77003_, r, p_202711_3_, p_202711_4_);
          }, r);
       };
    }
