--- a/net/minecraft/world/level/newbiome/layer/traits/AreaTransformer2.java
+++ b/net/minecraft/world/level/newbiome/layer/traits/AreaTransformer2.java
@@ -10,9 +_,9 @@
       return () -> {
          R r = p_77022_.m_76488_();
          R r1 = p_77023_.m_76488_();
-         return p_77021_.m_6887_((p_77018_, p_77019_) -> {
-            p_77021_.m_6687_((long)p_77018_, (long)p_77019_);
-            return this.m_5924_(p_77021_, r, r1, p_77018_, p_77019_);
+         return p_77021_.m_6887_((p_215724_4_, p_215724_5_) -> {
+            p_77021_.m_6687_((long)p_215724_4_, (long)p_215724_5_);
+            return this.m_5924_(p_77021_, r, r1, p_215724_4_, p_215724_5_);
          }, r, r1);
       };
    }
