--- a/net/minecraft/network/chat/ClickEvent.java
+++ b/net/minecraft/network/chat/ClickEvent.java
@@ -61,8 +_,8 @@
       CHANGE_PAGE("change_page", true),
       COPY_TO_CLIPBOARD("copy_to_clipboard", true);
 
-      private static final Map<String, ClickEvent.Action> f_130634_ = Arrays.stream(values()).collect(Collectors.toMap(ClickEvent.Action::m_130649_, (p_130648_) -> {
-         return p_130648_;
+      private static final Map<String, ClickEvent.Action> f_130634_ = Arrays.stream(values()).collect(Collectors.toMap(ClickEvent.Action::m_130649_, (p_199851_0_) -> {
+         return p_199851_0_;
       }));
       private final boolean f_130635_;
       private final String f_130636_;
