--- a/net/minecraft/network/chat/Component.java
+++ b/net/minecraft/network/chat/Component.java
@@ -40,12 +_,12 @@
 
    default String m_130668_(int p_130669_) {
       StringBuilder stringbuilder = new StringBuilder();
-      this.m_5651_((p_130673_) -> {
+      this.m_5651_((p_240639_2_) -> {
          int i = p_130669_ - stringbuilder.length();
          if (i <= 0) {
             return f_130759_;
          } else {
-            stringbuilder.append(p_130673_.length() <= i ? p_130673_ : p_130673_.substring(0, i));
+            stringbuilder.append(p_240639_2_.length() <= i ? p_240639_2_ : p_240639_2_.substring(0, i));
             return Optional.empty();
          }
       });
