--- a/net/minecraft/network/chat/FormattedText.java
+++ b/net/minecraft/network/chat/FormattedText.java
@@ -85,19 +_,19 @@
 
    default String getString() {
       StringBuilder stringbuilder = new StringBuilder();
-      this.m_5651_((p_130767_) -> {
-         stringbuilder.append(p_130767_);
+      this.m_5651_((p_241754_1_) -> {
+         stringbuilder.append(p_241754_1_);
          return Optional.empty();
       });
       return stringbuilder.toString();
    }
 
-   public interface ContentConsumer<T> {
-      Optional<T> m_130809_(String p_130810_);
-   }
-
    @OnlyIn(Dist.CLIENT)
    public interface StyledContentConsumer<T> {
       Optional<T> m_7164_(Style p_130811_, String p_130812_);
+   }
+
+   public interface ContentConsumer<T> {
+      Optional<T> m_130809_(String p_130810_);
    }
 }
