--- a/net/minecraft/network/chat/KeybindComponent.java
+++ b/net/minecraft/network/chat/KeybindComponent.java
@@ -7,9 +_,9 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class KeybindComponent extends BaseComponent {
-   private static Function<String, Supplier<Component>> f_130913_ = (p_130928_) -> {
+   private static Function<String, Supplier<Component>> f_130913_ = (p_193635_0_) -> {
       return () -> {
-         return new TextComponent(p_130928_);
+         return new TextComponent(p_193635_0_);
       };
    };
    private final String f_130914_;
