--- a/net/minecraft/network/chat/NbtComponent.java
+++ b/net/minecraft/network/chat/NbtComponent.java
@@ -19,7 +_,6 @@
 import net.minecraft.nbt.Tag;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerLevel;
-import net.minecraft.world.entity.Entity;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -60,25 +_,25 @@
       return this.f_130950_;
    }
 
-   public MutableComponent m_5638_(@Nullable CommandSourceStack p_130964_, @Nullable Entity p_130965_, int p_130966_) throws CommandSyntaxException {
+   public MutableComponent m_5638_(@Nullable CommandSourceStack p_130964_, @Nullable net.minecraft.world.entity.Entity p_130965_, int p_130966_) throws CommandSyntaxException {
       if (p_130964_ != null && this.f_130952_ != null) {
-         Stream<String> stream = this.m_7353_(p_130964_).flatMap((p_130973_) -> {
+         Stream<String> stream = this.m_7353_(p_130964_).flatMap((p_218675_1_) -> {
             try {
-               return this.f_130952_.m_99638_(p_130973_).stream();
+               return this.f_130952_.m_99638_(p_218675_1_).stream();
             } catch (CommandSyntaxException commandsyntaxexception) {
                return Stream.empty();
             }
          }).map(Tag::m_7916_);
-         return (MutableComponent)(this.f_130950_ ? stream.flatMap((p_130971_) -> {
+         return (MutableComponent)(this.f_130950_ ? stream.flatMap((p_223137_3_) -> {
             try {
-               MutableComponent mutablecomponent = Component.Serializer.m_130701_(p_130971_);
+               MutableComponent mutablecomponent = Component.Serializer.m_130701_(p_223137_3_);
                return Stream.of(ComponentUtils.m_130731_(p_130964_, mutablecomponent, p_130965_, p_130966_));
             } catch (Exception exception) {
-               f_130953_.warn("Failed to parse component: " + p_130971_, (Throwable)exception);
+               f_130953_.warn("Failed to parse component: " + p_223137_3_, (Throwable)exception);
                return Stream.of();
             }
-         }).reduce((p_130975_, p_130976_) -> {
-            return p_130975_.m_130946_(", ").m_7220_(p_130976_);
+         }).reduce((p_240704_0_, p_240704_1_) -> {
+            return p_240704_0_.m_130946_(", ").m_7220_(p_240704_1_);
          }).orElse(new TextComponent("")) : new TextComponent(Joiner.on(", ").join(stream.iterator())));
       } else {
          return new TextComponent("");
@@ -188,7 +_,7 @@
 
       protected Stream<CompoundTag> m_7353_(CommandSourceStack p_131017_) throws CommandSyntaxException {
          if (this.f_131005_ != null) {
-            List<? extends Entity> list = this.f_131005_.m_121160_(p_131017_);
+            List<? extends net.minecraft.world.entity.Entity> list = this.f_131005_.m_121160_(p_131017_);
             return list.stream().map(NbtPredicate::m_57485_);
          } else {
             return Stream.empty();
