--- a/com/mojang/realmsclient/gui/screens/RealmsDownloadLatestWorldScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsDownloadLatestWorldScreen.java
@@ -69,7 +_,7 @@
 
    public void m_7856_() {
       this.f_96541_.f_91068_.m_90926_(true);
-      this.f_88615_ = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ - 42, 200, 20, CommonComponents.f_130656_, (p_88642_) -> {
+      this.f_88615_ = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ - 42, 200, 20, CommonComponents.f_130656_, (p_237834_1_) -> {
          this.f_88621_ = true;
          this.m_88656_();
       }));
@@ -81,7 +_,7 @@
          if (!this.f_88607_ && this.m_88646_(this.f_88612_.f_87718_) >= 5368709120L) {
             Component component = new TranslatableComponent("mco.download.confirmation.line1", Unit.m_86945_(5368709120L));
             Component component1 = new TranslatableComponent("mco.download.confirmation.line2");
-            this.f_96541_.m_91152_(new RealmsLongConfirmationScreen((p_88651_) -> {
+            this.f_96541_.m_91152_(new RealmsLongConfirmationScreen((p_237837_1_) -> {
                this.f_88607_ = true;
                this.f_96541_.m_91152_(this);
                this.m_88657_();
