--- a/com/mojang/realmsclient/gui/screens/RealmsResetWorldScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsResetWorldScreen.java
@@ -78,7 +_,7 @@
    }
 
    public void m_7856_() {
-      this.m_7082_(new Button(this.f_96543_ / 2 - 40, m_120774_(14) - 10, 80, 20, this.f_89321_, (p_89419_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 40, m_120774_(14) - 10, 80, 20, this.f_89321_, (p_237959_1_) -> {
          this.f_96541_.m_91152_(this.f_89315_);
       }));
       (new Thread("Realms-reset-world-fetcher") {
@@ -104,29 +_,29 @@
       }).start();
       this.f_89317_ = this.m_7787_(new RealmsLabel(this.f_89319_, this.f_96543_ / 2, 7, 16777215));
       this.f_89318_ = this.m_7787_(new RealmsLabel(this.f_89320_, this.f_96543_ / 2, 22, this.f_89322_));
-      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(1), m_120774_(0) + 10, new TranslatableComponent("mco.reset.world.generate"), f_89300_, (p_89417_) -> {
+      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(1), m_120774_(0) + 10, new TranslatableComponent("mco.reset.world.generate"), f_89300_, (p_237958_1_) -> {
          this.f_96541_.m_91152_(new RealmsResetNormalWorldScreen(this, this.f_89319_));
       }));
-      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(2), m_120774_(0) + 10, new TranslatableComponent("mco.reset.world.upload"), f_89324_, (p_89415_) -> {
+      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(2), m_120774_(0) + 10, new TranslatableComponent("mco.reset.world.upload"), f_89324_, (p_237957_1_) -> {
          Screen screen = new RealmsSelectFileToUploadScreen(this.f_89316_.f_87473_, this.f_89313_ != -1 ? this.f_89313_ : this.f_89316_.f_87486_, this, this.f_89312_);
          this.f_96541_.m_91152_(screen);
       }));
-      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(3), m_120774_(0) + 10, new TranslatableComponent("mco.reset.world.template"), f_89326_, (p_89412_) -> {
+      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(3), m_120774_(0) + 10, new TranslatableComponent("mco.reset.world.template"), f_89326_, (p_237956_1_) -> {
          RealmsSelectWorldTemplateScreen realmsselectworldtemplatescreen = new RealmsSelectWorldTemplateScreen(this, RealmsServer.WorldType.NORMAL, this.f_89303_);
          realmsselectworldtemplatescreen.m_89680_(new TranslatableComponent("mco.reset.world.template"));
          this.f_96541_.m_91152_(realmsselectworldtemplatescreen);
       }));
-      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(1), m_120774_(6) + 20, new TranslatableComponent("mco.reset.world.adventure"), f_89325_, (p_89407_) -> {
+      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(1), m_120774_(6) + 20, new TranslatableComponent("mco.reset.world.adventure"), f_89325_, (p_237955_1_) -> {
          RealmsSelectWorldTemplateScreen realmsselectworldtemplatescreen = new RealmsSelectWorldTemplateScreen(this, RealmsServer.WorldType.ADVENTUREMAP, this.f_89304_);
          realmsselectworldtemplatescreen.m_89680_(new TranslatableComponent("mco.reset.world.adventure"));
          this.f_96541_.m_91152_(realmsselectworldtemplatescreen);
       }));
-      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(2), m_120774_(6) + 20, new TranslatableComponent("mco.reset.world.experience"), f_89301_, (p_89402_) -> {
+      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(2), m_120774_(6) + 20, new TranslatableComponent("mco.reset.world.experience"), f_89301_, (p_237954_1_) -> {
          RealmsSelectWorldTemplateScreen realmsselectworldtemplatescreen = new RealmsSelectWorldTemplateScreen(this, RealmsServer.WorldType.EXPERIENCE, this.f_89305_);
          realmsselectworldtemplatescreen.m_89680_(new TranslatableComponent("mco.reset.world.experience"));
          this.f_96541_.m_91152_(realmsselectworldtemplatescreen);
       }));
-      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(3), m_120774_(6) + 20, new TranslatableComponent("mco.reset.world.inspiration"), f_89302_, (p_89381_) -> {
+      this.m_7082_(new RealmsResetWorldScreen.FrameButton(this.m_89392_(3), m_120774_(6) + 20, new TranslatableComponent("mco.reset.world.inspiration"), f_89302_, (p_237951_1_) -> {
          RealmsSelectWorldTemplateScreen realmsselectworldtemplatescreen = new RealmsSelectWorldTemplateScreen(this, RealmsServer.WorldType.INSPIRATION, this.f_89306_);
          realmsselectworldtemplatescreen.m_89680_(new TranslatableComponent("mco.reset.world.inspiration"));
          this.f_96541_.m_91152_(realmsselectworldtemplatescreen);
@@ -253,20 +_,6 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   class FrameButton extends Button {
-      private final ResourceLocation f_89436_;
-
-      public FrameButton(int p_89439_, int p_89440_, Component p_89441_, ResourceLocation p_89442_, Button.OnPress p_89443_) {
-         super(p_89439_, p_89440_, 60, 72, p_89441_, p_89443_);
-         this.f_89436_ = p_89442_;
-      }
-
-      public void m_6303_(PoseStack p_89445_, int p_89446_, int p_89447_, float p_89448_) {
-         RealmsResetWorldScreen.this.m_89354_(p_89445_, this.f_93620_, this.f_93621_, this.m_6035_(), this.f_89436_, this.m_5702_(), this.m_5953_((double)p_89446_, (double)p_89447_));
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
    static enum ResetType {
       NONE,
       GENERATE,
@@ -287,6 +_,20 @@
          this.f_89464_ = p_89468_;
          this.f_89465_ = p_89469_;
          this.f_89466_ = p_89470_;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   class FrameButton extends Button {
+      private final ResourceLocation f_89436_;
+
+      public FrameButton(int p_89439_, int p_89440_, Component p_89441_, ResourceLocation p_89442_, Button.OnPress p_89443_) {
+         super(p_89439_, p_89440_, 60, 72, p_89441_, p_89443_);
+         this.f_89436_ = p_89442_;
+      }
+
+      public void m_6303_(PoseStack p_89445_, int p_89446_, int p_89447_, float p_89448_) {
+         RealmsResetWorldScreen.this.m_89354_(p_89445_, this.f_93620_, this.f_93621_, this.m_6035_(), this.f_89436_, this.m_5702_(), this.m_5953_((double)p_89446_, (double)p_89447_));
       }
    }
 }
