--- a/com/mojang/realmsclient/gui/screens/RealmsBrokenWorldScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsBrokenWorldScreen.java
@@ -55,7 +_,7 @@
    public void m_7856_() {
       this.f_88290_ = this.f_96543_ / 2 - 150;
       this.f_88291_ = this.f_96543_ / 2 + 190;
-      this.m_7082_(new Button(this.f_88291_ - 80 + 8, m_120774_(13) - 5, 70, 20, CommonComponents.f_130660_, (p_88333_) -> {
+      this.m_7082_(new Button(this.f_88291_ - 80 + 8, m_120774_(13) - 5, 70, 20, CommonComponents.f_130660_, (p_237776_1_) -> {
          this.m_88351_();
       }));
       if (this.f_88286_ == null) {
@@ -74,7 +_,7 @@
          boolean flag = i != this.f_88286_.f_87486_ || this.f_88286_.f_87485_ == RealmsServer.WorldType.MINIGAME;
          Button button;
          if (flag) {
-            button = new Button(this.m_88301_(i), m_120774_(8), 80, 20, new TranslatableComponent("mco.brokenworld.play"), (p_88347_) -> {
+            button = new Button(this.m_88301_(i), m_120774_(8), 80, 20, new TranslatableComponent("mco.brokenworld.play"), (p_237780_2_) -> {
                if ((this.f_88286_.f_87481_.get(i)).f_87611_) {
                   RealmsResetWorldScreen realmsresetworldscreen = new RealmsResetWorldScreen(this, this.f_88286_, new TranslatableComponent("mco.configure.world.switch.slot"), new TranslatableComponent("mco.configure.world.switch.slot.subtitle"), 10526880, CommonComponents.f_130656_, this::m_88300_, () -> {
                      this.f_96541_.m_91152_(this);
@@ -89,11 +_,11 @@
 
             });
          } else {
-            button = new Button(this.m_88301_(i), m_120774_(8), 80, 20, new TranslatableComponent("mco.brokenworld.download"), (p_88339_) -> {
+            button = new Button(this.m_88301_(i), m_120774_(8), 80, 20, new TranslatableComponent("mco.brokenworld.download"), (p_237777_2_) -> {
                Component component = new TranslatableComponent("mco.configure.world.restore.download.question.line1");
                Component component1 = new TranslatableComponent("mco.configure.world.restore.download.question.line2");
-               this.f_96541_.m_91152_(new RealmsLongConfirmationScreen((p_88342_) -> {
-                  if (p_88342_) {
+               this.f_96541_.m_91152_(new RealmsLongConfirmationScreen((p_237778_2_) -> {
+                  if (p_237778_2_) {
                      this.m_88335_(i);
                   } else {
                      this.f_96541_.m_91152_(this);
@@ -109,7 +_,7 @@
          }
 
          this.m_7082_(button);
-         this.m_7082_(new Button(this.m_88301_(i), m_120774_(10), 80, 20, new TranslatableComponent("mco.brokenworld.reset"), (p_88309_) -> {
+         this.m_7082_(new Button(this.m_88301_(i), m_120774_(10), 80, 20, new TranslatableComponent("mco.brokenworld.reset"), (p_237773_2_) -> {
             RealmsResetWorldScreen realmsresetworldscreen = new RealmsResetWorldScreen(this, this.f_88286_, this::m_88300_, () -> {
                this.f_96541_.m_91152_(this);
                this.m_88300_();
@@ -211,8 +_,8 @@
 
       try {
          WorldDownload worlddownload = realmsclient.m_87209_(this.f_88286_.f_87473_, p_88336_);
-         RealmsDownloadLatestWorldScreen realmsdownloadlatestworldscreen = new RealmsDownloadLatestWorldScreen(this, worlddownload, this.f_88286_.m_87495_(p_88336_), (p_88312_) -> {
-            if (p_88312_) {
+         RealmsDownloadLatestWorldScreen realmsdownloadlatestworldscreen = new RealmsDownloadLatestWorldScreen(this, worlddownload, this.f_88286_.m_87495_(p_88336_), (p_237774_2_) -> {
+            if (p_237774_2_) {
                this.f_88292_.add(p_88336_);
                this.f_96540_.clear();
                this.m_88350_();
