--- a/com/mojang/realmsclient/gui/screens/RealmsCreateRealmScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsCreateRealmScreen.java
@@ -43,10 +_,10 @@
 
    public void m_7856_() {
       this.f_96541_.f_91068_.m_90926_(true);
-      this.f_88570_ = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 120 + 17, 97, 20, new TranslatableComponent("mco.create.world"), (p_88592_) -> {
+      this.f_88570_ = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 120 + 17, 97, 20, new TranslatableComponent("mco.create.world"), (p_237828_1_) -> {
          this.m_88595_();
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 4 + 120 + 17, 95, 20, CommonComponents.f_130656_, (p_88589_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 4 + 120 + 17, 95, 20, CommonComponents.f_130656_, (p_237827_1_) -> {
          this.f_96541_.m_91152_(this.f_88567_);
       }));
       this.f_88570_.f_93623_ = false;
