--- a/com/mojang/realmsclient/gui/screens/RealmsConfirmScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsConfirmScreen.java
@@ -24,10 +_,10 @@
    }
 
    public void m_7856_() {
-      this.m_7082_(new Button(this.f_96543_ / 2 - 105, m_120774_(9), 100, 20, CommonComponents.f_130657_, (p_88562_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 105, m_120774_(9), 100, 20, CommonComponents.f_130657_, (p_237826_1_) -> {
          this.f_88545_.accept(true);
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 5, m_120774_(9), 100, 20, CommonComponents.f_130658_, (p_88559_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 5, m_120774_(9), 100, 20, CommonComponents.f_130658_, (p_237825_1_) -> {
          this.f_88545_.accept(false);
       }));
    }
