--- a/com/mojang/realmsclient/gui/screens/RealmsInviteScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsInviteScreen.java
@@ -44,10 +_,10 @@
       this.f_88696_ = new EditBox(this.f_96541_.f_91062_, this.f_96543_ / 2 - 100, m_120774_(2), 200, 20, (EditBox)null, new TranslatableComponent("mco.configure.world.invite.profile.name"));
       this.m_7787_(this.f_88696_);
       this.m_94718_(this.f_88696_);
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, m_120774_(10), 200, 20, new TranslatableComponent("mco.configure.world.buttons.invite"), (p_88721_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, m_120774_(10), 200, 20, new TranslatableComponent("mco.configure.world.buttons.invite"), (p_237844_1_) -> {
          this.m_88724_();
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, m_120774_(12), 200, 20, CommonComponents.f_130656_, (p_88716_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, m_120774_(12), 200, 20, CommonComponents.f_130656_, (p_237843_1_) -> {
          this.f_96541_.m_91152_(this.f_88699_);
       }));
    }
