--- a/com/mojang/realmsclient/gui/screens/RealmsTermsScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsTermsScreen.java
@@ -41,10 +_,10 @@
    public void m_7856_() {
       this.f_96541_.f_91068_.m_90926_(true);
       int i = this.f_96543_ / 4 - 2;
-      this.m_7082_(new Button(this.f_96543_ / 4, m_120774_(12), i, 20, new TranslatableComponent("mco.terms.buttons.agree"), (p_90054_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 4, m_120774_(12), i, 20, new TranslatableComponent("mco.terms.buttons.agree"), (p_238078_1_) -> {
          this.m_90056_();
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 4, m_120774_(12), i, 20, new TranslatableComponent("mco.terms.buttons.disagree"), (p_90050_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 4, m_120774_(12), i, 20, new TranslatableComponent("mco.terms.buttons.disagree"), (p_238077_1_) -> {
          this.f_96541_.m_91152_(this.f_90026_);
       }));
    }
