--- a/net/minecraft/world/level/block/piston/PistonBaseBlock.java
+++ b/net/minecraft/world/level/block/piston/PistonBaseBlock.java
@@ -162,6 +_,7 @@
       }
 
       if (p_60195_ == 0) {
+         if (net.minecraftforge.event.ForgeEventFactory.onPistonMovePre(p_60193_, p_60194_, direction, true)) return false;
          if (!this.m_60181_(p_60193_, p_60194_, direction, true)) {
             return false;
          }
@@ -169,6 +_,7 @@
          p_60193_.m_7731_(p_60194_, p_60192_.m_61124_(f_60153_, Boolean.valueOf(true)), 67);
          p_60193_.m_5594_((Player)null, p_60194_, SoundEvents.f_12312_, SoundSource.BLOCKS, 0.5F, p_60193_.f_46441_.nextFloat() * 0.25F + 0.6F);
       } else if (p_60195_ == 1 || p_60195_ == 2) {
+         if (net.minecraftforge.event.ForgeEventFactory.onPistonMovePre(p_60193_, p_60194_, direction, false)) return false;
          BlockEntity blockentity1 = p_60193_.m_7702_(p_60194_.m_121945_(direction));
          if (blockentity1 instanceof PistonMovingBlockEntity) {
             ((PistonMovingBlockEntity)blockentity1).m_60401_();
@@ -208,6 +_,7 @@
          p_60193_.m_5594_((Player)null, p_60194_, SoundEvents.f_12311_, SoundSource.BLOCKS, 0.5F, p_60193_.f_46441_.nextFloat() * 0.15F + 0.6F);
       }
 
+      net.minecraftforge.event.ForgeEventFactory.onPistonMovePost(p_60193_, p_60194_, direction, (p_60195_ == 0));
       return true;
    }
 
@@ -238,7 +_,7 @@
                   return false;
                }
 
-               return !p_60205_.m_60734_().m_60588_();
+               return !p_60205_.hasTileEntity();
             }
          } else {
             return false;
@@ -277,7 +_,7 @@
          for(int k = list2.size() - 1; k >= 0; --k) {
             BlockPos blockpos2 = list2.get(k);
             BlockState blockstate1 = p_60182_.m_8055_(blockpos2);
-            BlockEntity blockentity = blockstate1.m_60734_().m_60588_() ? p_60182_.m_7702_(blockpos2) : null;
+            BlockEntity blockentity = blockstate1.hasTileEntity() ? p_60182_.m_7702_(blockpos2) : null;
             m_49892_(blockstate1, p_60182_, blockpos2, blockentity);
             p_60182_.m_7731_(blockpos2, Blocks.f_50016_.m_49966_(), 18);
             ablockstate[j++] = blockstate1;
@@ -339,6 +_,10 @@
 
    public BlockState m_6843_(BlockState p_60215_, Rotation p_60216_) {
       return p_60215_.m_61124_(f_52588_, p_60216_.m_55954_(p_60215_.m_61143_(f_52588_)));
+   }
+
+   public BlockState rotate(BlockState state, net.minecraft.world.level.LevelAccessor world, BlockPos pos, Rotation direction) {
+       return state.m_61143_(f_60153_) ? state : super.rotate(state, world, pos, direction);
    }
 
    public BlockState m_6943_(BlockState p_60212_, Mirror p_60213_) {
