--- a/net/minecraft/world/level/block/piston/PistonHeadBlock.java
+++ b/net/minecraft/world/level/block/piston/PistonHeadBlock.java
@@ -25,8 +_,6 @@
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.Shapes;
 import net.minecraft.world.phys.shapes.VoxelShape;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class PistonHeadBlock extends DirectionalBlock {
    public static final EnumProperty<PistonType> f_60235_ = BlockStateProperties.f_61396_;
@@ -53,10 +_,10 @@
    private static final VoxelShape[] f_60256_ = m_60312_(false);
 
    private static VoxelShape[] m_60312_(boolean p_60313_) {
-      return Arrays.stream(Direction.values()).map((p_60316_) -> {
-         return m_60309_(p_60316_, p_60313_);
-      }).toArray((p_60318_) -> {
-         return new VoxelShape[p_60318_];
+      return Arrays.stream(Direction.values()).map((p_242695_1_) -> {
+         return m_60309_(p_242695_1_, p_60313_);
+      }).toArray((p_242696_0_) -> {
+         return new VoxelShape[p_242696_0_];
       });
    }
 
@@ -135,7 +_,6 @@
 
    }
 
-   @OnlyIn(Dist.CLIENT)
    public ItemStack m_7397_(BlockGetter p_60261_, BlockPos p_60262_, BlockState p_60263_) {
       return new ItemStack(p_60263_.m_61143_(f_60235_) == PistonType.STICKY ? Blocks.f_50032_ : Blocks.f_50039_);
    }
