--- a/net/minecraft/world/level/storage/loot/predicates/DamageSourceCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/DamageSourceCondition.java
@@ -8,6 +_,7 @@
 import net.minecraft.advancements.critereon.DamageSourcePredicate;
 import net.minecraft.world.damagesource.DamageSource;
 import net.minecraft.world.level.storage.loot.LootContext;
+import net.minecraft.world.level.storage.loot.Serializer;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
 import net.minecraft.world.phys.Vec3;
