--- a/net/minecraft/world/level/storage/loot/predicates/LootItemKilledByPlayerCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/LootItemKilledByPlayerCondition.java
@@ -6,6 +_,7 @@
 import com.google.gson.JsonSerializationContext;
 import java.util.Set;
 import net.minecraft.world.level.storage.loot.LootContext;
+import net.minecraft.world.level.storage.loot.Serializer;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParam;
 import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
 
