--- a/net/minecraft/world/level/storage/loot/predicates/TimeCheck.java
+++ b/net/minecraft/world/level/storage/loot/predicates/TimeCheck.java
@@ -8,6 +_,7 @@
 import net.minecraft.util.GsonHelper;
 import net.minecraft.world.level.storage.loot.LootContext;
 import net.minecraft.world.level.storage.loot.RandomValueBounds;
+import net.minecraft.world.level.storage.loot.Serializer;
 
 public class TimeCheck implements LootItemCondition {
    @Nullable
