--- a/net/minecraft/client/model/BookModel.java
+++ b/net/minecraft/client/model/BookModel.java
@@ -38,8 +_,8 @@
    }
 
    public void m_102316_(PoseStack p_102317_, VertexConsumer p_102318_, int p_102319_, int p_102320_, float p_102321_, float p_102322_, float p_102323_, float p_102324_) {
-      this.f_102290_.forEach((p_102315_) -> {
-         p_102315_.m_104306_(p_102317_, p_102318_, p_102319_, p_102320_, p_102321_, p_102322_, p_102323_, p_102324_);
+      this.f_102290_.forEach((p_228248_8_) -> {
+         p_228248_8_.m_104306_(p_102317_, p_102318_, p_102319_, p_102320_, p_102321_, p_102322_, p_102323_, p_102324_);
       });
    }
 
