--- a/net/minecraft/client/model/ListModel.java
+++ b/net/minecraft/client/model/ListModel.java
@@ -21,8 +_,8 @@
    }
 
    public void m_7695_(PoseStack p_103013_, VertexConsumer p_103014_, int p_103015_, int p_103016_, float p_103017_, float p_103018_, float p_103019_, float p_103020_) {
-      this.m_6195_().forEach((p_103030_) -> {
-         p_103030_.m_104306_(p_103013_, p_103014_, p_103015_, p_103016_, p_103017_, p_103018_, p_103019_, p_103020_);
+      this.m_6195_().forEach((p_228272_8_) -> {
+         p_228272_8_.m_104306_(p_103013_, p_103014_, p_103015_, p_103016_, p_103017_, p_103018_, p_103019_, p_103020_);
       });
    }
 
