--- a/net/minecraft/client/CloudStatus.java
+++ b/net/minecraft/client/CloudStatus.java
@@ -12,8 +_,8 @@
    FAST(1, "options.clouds.fast"),
    FANCY(2, "options.clouds.fancy");
 
-   private static final CloudStatus[] f_90653_ = Arrays.stream(values()).sorted(Comparator.comparingInt(CloudStatus::m_90663_)).toArray((p_90668_) -> {
-      return new CloudStatus[p_90668_];
+   private static final CloudStatus[] f_90653_ = Arrays.stream(values()).sorted(Comparator.comparingInt(CloudStatus::m_90663_)).toArray((p_216805_0_) -> {
+      return new CloudStatus[p_216805_0_];
    });
    private final int f_90654_;
    private final String f_90655_;
