--- a/net/minecraft/client/CycleOption.java
+++ b/net/minecraft/client/CycleOption.java
@@ -25,9 +_,9 @@
    }
 
    public AbstractWidget m_7496_(Options p_90688_, int p_90689_, int p_90690_, int p_90691_) {
-      return new OptionButton(p_90689_, p_90690_, p_90691_, 20, this, this.m_90695_(p_90688_), (p_90694_) -> {
+      return new OptionButton(p_90689_, p_90690_, p_90691_, 20, this, this.m_90695_(p_90688_), (p_216721_2_) -> {
          this.m_90684_(p_90688_, 1);
-         p_90694_.m_93666_(this.m_90695_(p_90688_));
+         p_216721_2_.m_93666_(this.m_90695_(p_90688_));
       });
    }
 
