--- a/net/minecraft/client/NarratorStatus.java
+++ b/net/minecraft/client/NarratorStatus.java
@@ -15,8 +_,8 @@
    CHAT(2, "options.narrator.chat"),
    SYSTEM(3, "options.narrator.system");
 
-   private static final NarratorStatus[] f_91608_ = Arrays.stream(values()).sorted(Comparator.comparingInt(NarratorStatus::m_91618_)).toArray((p_91623_) -> {
-      return new NarratorStatus[p_91623_];
+   private static final NarratorStatus[] f_91608_ = Arrays.stream(values()).sorted(Comparator.comparingInt(NarratorStatus::m_91618_)).toArray((p_216826_0_) -> {
+      return new NarratorStatus[p_216826_0_];
    });
    private final int f_91609_;
    private final Component f_91610_;
