--- a/net/minecraft/client/Option.java
+++ b/net/minecraft/client/Option.java
@@ -23,267 +_,267 @@
 
 @OnlyIn(Dist.CLIENT)
 public abstract class Option {
-   public static final ProgressOption f_91653_ = new ProgressOption("options.biomeBlendRadius", 0.0D, 7.0D, 1.0F, (p_91713_) -> {
-      return (double)p_91713_.f_92032_;
-   }, (p_92009_, p_92010_) -> {
-      p_92009_.f_92032_ = Mth.m_14045_((int)p_92010_.doubleValue(), 0, 7);
+   public static final ProgressOption f_91653_ = new ProgressOption("options.biomeBlendRadius", 0.0D, 7.0D, 1.0F, (p_216607_0_) -> {
+      return (double)p_216607_0_.f_92032_;
+   }, (p_216660_0_, p_216660_1_) -> {
+      p_216660_0_.f_92032_ = Mth.m_14045_((int)p_216660_1_.doubleValue(), 0, 7);
       Minecraft.m_91087_().f_91060_.m_109818_();
-   }, (p_92003_, p_92004_) -> {
-      double d0 = p_92004_.m_92221_(p_92003_);
+   }, (p_216595_0_, p_216595_1_) -> {
+      double d0 = p_216595_1_.m_92221_(p_216595_0_);
       int i = (int)d0 * 2 + 1;
-      return p_92004_.m_91740_(new TranslatableComponent("options.biomeBlendRadius." + i));
-   });
-   public static final ProgressOption f_91660_ = new ProgressOption("options.chat.height.focused", 0.0D, 1.0D, 0.0F, (p_91711_) -> {
-      return p_91711_.f_92134_;
-   }, (p_91998_, p_91999_) -> {
-      p_91998_.f_92134_ = p_91999_;
-      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
-   }, (p_91992_, p_91993_) -> {
-      double d0 = p_91993_.m_6917_(p_91993_.m_92221_(p_91992_));
-      return p_91993_.m_91715_(ChatComponent.m_93811_(d0));
-   });
-   public static final ProgressOption f_91661_ = new ProgressOption("options.chat.height.unfocused", 0.0D, 1.0D, 0.0F, (p_91709_) -> {
-      return p_91709_.f_92133_;
-   }, (p_91987_, p_91988_) -> {
-      p_91987_.f_92133_ = p_91988_;
-      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
-   }, (p_91981_, p_91982_) -> {
-      double d0 = p_91982_.m_6917_(p_91982_.m_92221_(p_91981_));
-      return p_91982_.m_91715_(ChatComponent.m_93811_(d0));
-   });
-   public static final ProgressOption f_91662_ = new ProgressOption("options.chat.opacity", 0.0D, 1.0D, 0.0F, (p_91707_) -> {
-      return p_91707_.f_92120_;
-   }, (p_91976_, p_91977_) -> {
-      p_91976_.f_92120_ = p_91977_;
-      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
-   }, (p_91970_, p_91971_) -> {
-      double d0 = p_91971_.m_6917_(p_91971_.m_92221_(p_91970_));
-      return p_91971_.m_91762_(d0 * 0.9D + 0.1D);
-   });
-   public static final ProgressOption f_91663_ = new ProgressOption("options.chat.scale", 0.0D, 1.0D, 0.0F, (p_91705_) -> {
-      return p_91705_.f_92131_;
-   }, (p_91965_, p_91966_) -> {
-      p_91965_.f_92131_ = p_91966_;
-      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
-   }, (p_91959_, p_91960_) -> {
-      double d0 = p_91960_.m_6917_(p_91960_.m_92221_(p_91959_));
-      return (Component)(d0 == 0.0D ? CommonComponents.m_130663_(p_91960_.m_91714_(), false) : p_91960_.m_91762_(d0));
-   });
-   public static final ProgressOption f_91664_ = new ProgressOption("options.chat.width", 0.0D, 1.0D, 0.0F, (p_91703_) -> {
-      return p_91703_.f_92132_;
-   }, (p_91954_, p_91955_) -> {
-      p_91954_.f_92132_ = p_91955_;
-      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
-   }, (p_91948_, p_91949_) -> {
-      double d0 = p_91949_.m_6917_(p_91949_.m_92221_(p_91948_));
-      return p_91949_.m_91715_(ChatComponent.m_93798_(d0));
-   });
-   public static final ProgressOption f_91665_ = new ProgressOption("options.chat.line_spacing", 0.0D, 1.0D, 0.0F, (p_91701_) -> {
-      return p_91701_.f_92121_;
-   }, (p_91943_, p_91944_) -> {
-      p_91943_.f_92121_ = p_91944_;
-   }, (p_91937_, p_91938_) -> {
-      return p_91938_.m_91762_(p_91938_.m_6917_(p_91938_.m_92221_(p_91937_)));
-   });
-   public static final ProgressOption f_91666_ = new ProgressOption("options.chat.delay_instant", 0.0D, 6.0D, 0.1F, (p_91699_) -> {
-      return p_91699_.f_92135_;
-   }, (p_91929_, p_91930_) -> {
-      p_91929_.f_92135_ = p_91930_;
-   }, (p_91923_, p_91924_) -> {
-      double d0 = p_91924_.m_92221_(p_91923_);
+      return p_216595_1_.m_91740_(new TranslatableComponent("options.biomeBlendRadius." + i));
+   });
+   public static final ProgressOption f_91660_ = new ProgressOption("options.chat.height.focused", 0.0D, 1.0D, 0.0F, (p_216587_0_) -> {
+      return p_216587_0_.f_92134_;
+   }, (p_216600_0_, p_216600_1_) -> {
+      p_216600_0_.f_92134_ = p_216600_1_;
+      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
+   }, (p_216642_0_, p_216642_1_) -> {
+      double d0 = p_216642_1_.m_6917_(p_216642_1_.m_92221_(p_216642_0_));
+      return p_216642_1_.m_91715_(ChatComponent.m_93811_(d0));
+   });
+   public static final ProgressOption f_91661_ = new ProgressOption("options.chat.height.unfocused", 0.0D, 1.0D, 0.0F, (p_216611_0_) -> {
+      return p_216611_0_.f_92133_;
+   }, (p_216650_0_, p_216650_1_) -> {
+      p_216650_0_.f_92133_ = p_216650_1_;
+      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
+   }, (p_216604_0_, p_216604_1_) -> {
+      double d0 = p_216604_1_.m_6917_(p_216604_1_.m_92221_(p_216604_0_));
+      return p_216604_1_.m_91715_(ChatComponent.m_93811_(d0));
+   });
+   public static final ProgressOption f_91662_ = new ProgressOption("options.chat.opacity", 0.0D, 1.0D, 0.0F, (p_216649_0_) -> {
+      return p_216649_0_.f_92120_;
+   }, (p_216578_0_, p_216578_1_) -> {
+      p_216578_0_.f_92120_ = p_216578_1_;
+      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
+   }, (p_216592_0_, p_216592_1_) -> {
+      double d0 = p_216592_1_.m_6917_(p_216592_1_.m_92221_(p_216592_0_));
+      return p_216592_1_.m_91762_(d0 * 0.9D + 0.1D);
+   });
+   public static final ProgressOption f_91663_ = new ProgressOption("options.chat.scale", 0.0D, 1.0D, 0.0F, (p_216591_0_) -> {
+      return p_216591_0_.f_92131_;
+   }, (p_216624_0_, p_216624_1_) -> {
+      p_216624_0_.f_92131_ = p_216624_1_;
+      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
+   }, (p_216637_0_, p_216637_1_) -> {
+      double d0 = p_216637_1_.m_6917_(p_216637_1_.m_92221_(p_216637_0_));
+      return (Component)(d0 == 0.0D ? CommonComponents.m_130663_(p_216637_1_.m_91714_(), false) : p_216637_1_.m_91762_(d0));
+   });
+   public static final ProgressOption f_91664_ = new ProgressOption("options.chat.width", 0.0D, 1.0D, 0.0F, (p_216601_0_) -> {
+      return p_216601_0_.f_92132_;
+   }, (p_216620_0_, p_216620_1_) -> {
+      p_216620_0_.f_92132_ = p_216620_1_;
+      Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
+   }, (p_216673_0_, p_216673_1_) -> {
+      double d0 = p_216673_1_.m_6917_(p_216673_1_.m_92221_(p_216673_0_));
+      return p_216673_1_.m_91715_(ChatComponent.m_93798_(d0));
+   });
+   public static final ProgressOption f_91665_ = new ProgressOption("options.chat.line_spacing", 0.0D, 1.0D, 0.0F, (p_238287_0_) -> {
+      return p_238287_0_.f_92121_;
+   }, (p_238282_0_, p_238282_1_) -> {
+      p_238282_0_.f_92121_ = p_238282_1_;
+   }, (p_238297_0_, p_238297_1_) -> {
+      return p_238297_1_.m_91762_(p_238297_1_.m_6917_(p_238297_1_.m_92221_(p_238297_0_)));
+   });
+   public static final ProgressOption f_91666_ = new ProgressOption("options.chat.delay_instant", 0.0D, 6.0D, 0.1F, (p_238271_0_) -> {
+      return p_238271_0_.f_92135_;
+   }, (p_238273_0_, p_238273_1_) -> {
+      p_238273_0_.f_92135_ = p_238273_1_;
+   }, (p_238292_0_, p_238292_1_) -> {
+      double d0 = p_238292_1_.m_92221_(p_238292_0_);
       return d0 <= 0.0D ? new TranslatableComponent("options.chat.delay_none") : new TranslatableComponent("options.chat.delay", String.format("%.1f", d0));
    });
-   public static final ProgressOption f_91667_ = new ProgressOption("options.fov", 30.0D, 110.0D, 1.0F, (p_91697_) -> {
-      return p_91697_.f_92068_;
-   }, (p_91912_, p_91913_) -> {
-      p_91912_.f_92068_ = p_91913_;
-   }, (p_91906_, p_91907_) -> {
-      double d0 = p_91907_.m_92221_(p_91906_);
+   public static final ProgressOption f_91667_ = new ProgressOption("options.fov", 30.0D, 110.0D, 1.0F, (p_216655_0_) -> {
+      return p_216655_0_.f_92068_;
+   }, (p_216612_0_, p_216612_1_) -> {
+      p_216612_0_.f_92068_ = p_216612_1_;
+   }, (p_216590_0_, p_216590_1_) -> {
+      double d0 = p_216590_1_.m_92221_(p_216590_0_);
       if (d0 == 70.0D) {
-         return p_91907_.m_91740_(new TranslatableComponent("options.fov.min"));
+         return p_216590_1_.m_91740_(new TranslatableComponent("options.fov.min"));
       } else {
-         return d0 == p_91907_.m_92235_() ? p_91907_.m_91740_(new TranslatableComponent("options.fov.max")) : p_91907_.m_91764_((int)d0);
+         return d0 == p_216590_1_.m_92235_() ? p_216590_1_.m_91740_(new TranslatableComponent("options.fov.max")) : p_216590_1_.m_91764_((int)d0);
       }
    });
    private static final Component f_91651_ = new TranslatableComponent("options.fovEffectScale.tooltip");
-   public static final ProgressOption f_91668_ = new ProgressOption("options.fovEffectScale", 0.0D, 1.0D, 0.0F, (p_91695_) -> {
-      return Math.pow((double)p_91695_.f_92070_, 2.0D);
-   }, (p_91895_, p_91896_) -> {
-      p_91895_.f_92070_ = Mth.m_14003_(p_91896_);
-   }, (p_91889_, p_91890_) -> {
-      p_91890_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91651_, 200));
-      double d0 = p_91890_.m_6917_(p_91890_.m_92221_(p_91889_));
-      return d0 == 0.0D ? p_91890_.m_91740_(new TranslatableComponent("options.fovEffectScale.off")) : p_91890_.m_91762_(d0);
+   public static final ProgressOption f_91668_ = new ProgressOption("options.fovEffectScale", 0.0D, 1.0D, 0.0F, (p_216672_0_) -> {
+      return Math.pow((double)p_216672_0_.f_92070_, 2.0D);
+   }, (p_216608_0_, p_216608_1_) -> {
+      p_216608_0_.f_92070_ = Mth.m_14003_(p_216608_1_);
+   }, (p_216645_0_, p_216645_1_) -> {
+      p_216645_1_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91651_, 200));
+      double d0 = p_216645_1_.m_6917_(p_216645_1_.m_92221_(p_216645_0_));
+      return d0 == 0.0D ? p_216645_1_.m_91740_(new TranslatableComponent("options.fovEffectScale.off")) : p_216645_1_.m_91762_(d0);
    });
    private static final Component f_91652_ = new TranslatableComponent("options.screenEffectScale.tooltip");
-   public static final ProgressOption f_91669_ = new ProgressOption("options.screenEffectScale", 0.0D, 1.0D, 0.0F, (p_91693_) -> {
-      return (double)p_91693_.f_92069_;
-   }, (p_91878_, p_91879_) -> {
-      p_91878_.f_92069_ = p_91879_.floatValue();
-   }, (p_91872_, p_91873_) -> {
-      p_91873_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91652_, 200));
-      double d0 = p_91873_.m_6917_(p_91873_.m_92221_(p_91872_));
-      return d0 == 0.0D ? p_91873_.m_91740_(new TranslatableComponent("options.screenEffectScale.off")) : p_91873_.m_91762_(d0);
-   });
-   public static final ProgressOption f_91670_ = new ProgressOption("options.framerateLimit", 10.0D, 260.0D, 10.0F, (p_91691_) -> {
-      return (double)p_91691_.f_92113_;
-   }, (p_91861_, p_91862_) -> {
-      p_91861_.f_92113_ = (int)p_91862_.doubleValue();
-      Minecraft.m_91087_().m_91268_().m_85380_(p_91861_.f_92113_);
-   }, (p_91855_, p_91856_) -> {
-      double d0 = p_91856_.m_92221_(p_91855_);
-      return d0 == p_91856_.m_92235_() ? p_91856_.m_91740_(new TranslatableComponent("options.framerateLimit.max")) : p_91856_.m_91740_(new TranslatableComponent("options.framerate", (int)d0));
-   });
-   public static final ProgressOption f_91671_ = new ProgressOption("options.gamma", 0.0D, 1.0D, 0.0F, (p_91689_) -> {
-      return p_91689_.f_92071_;
-   }, (p_91844_, p_91845_) -> {
-      p_91844_.f_92071_ = p_91845_;
-   }, (p_91838_, p_91839_) -> {
-      double d0 = p_91839_.m_6917_(p_91839_.m_92221_(p_91838_));
+   public static final ProgressOption f_91669_ = new ProgressOption("options.screenEffectScale", 0.0D, 1.0D, 0.0F, (p_244419_0_) -> {
+      return (double)p_244419_0_.f_92069_;
+   }, (p_244415_0_, p_244415_1_) -> {
+      p_244415_0_.f_92069_ = p_244415_1_.floatValue();
+   }, (p_244416_0_, p_244416_1_) -> {
+      p_244416_1_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91652_, 200));
+      double d0 = p_244416_1_.m_6917_(p_244416_1_.m_92221_(p_244416_0_));
+      return d0 == 0.0D ? p_244416_1_.m_91740_(new TranslatableComponent("options.screenEffectScale.off")) : p_244416_1_.m_91762_(d0);
+   });
+   public static final ProgressOption f_91670_ = new ProgressOption("options.framerateLimit", 10.0D, 260.0D, 10.0F, (p_244406_0_) -> {
+      return (double)p_244406_0_.f_92113_;
+   }, (p_244411_0_, p_244411_1_) -> {
+      p_244411_0_.f_92113_ = (int)p_244411_1_.doubleValue();
+      Minecraft.m_91087_().m_91268_().m_85380_(p_244411_0_.f_92113_);
+   }, (p_244413_0_, p_244413_1_) -> {
+      double d0 = p_244413_1_.m_92221_(p_244413_0_);
+      return d0 == p_244413_1_.m_92235_() ? p_244413_1_.m_91740_(new TranslatableComponent("options.framerateLimit.max")) : p_244413_1_.m_91740_(new TranslatableComponent("options.framerate", (int)d0));
+   });
+   public static final ProgressOption f_91671_ = new ProgressOption("options.gamma", 0.0D, 1.0D, 0.0F, (p_216636_0_) -> {
+      return p_216636_0_.f_92071_;
+   }, (p_216651_0_, p_216651_1_) -> {
+      p_216651_0_.f_92071_ = p_216651_1_;
+   }, (p_216594_0_, p_216594_1_) -> {
+      double d0 = p_216594_1_.m_6917_(p_216594_1_.m_92221_(p_216594_0_));
       if (d0 == 0.0D) {
-         return p_91839_.m_91740_(new TranslatableComponent("options.gamma.min"));
+         return p_216594_1_.m_91740_(new TranslatableComponent("options.gamma.min"));
       } else {
-         return d0 == 1.0D ? p_91839_.m_91740_(new TranslatableComponent("options.gamma.max")) : p_91839_.m_91743_((int)(d0 * 100.0D));
+         return d0 == 1.0D ? p_216594_1_.m_91740_(new TranslatableComponent("options.gamma.max")) : p_216594_1_.m_91743_((int)(d0 * 100.0D));
       }
    });
-   public static final ProgressOption f_91672_ = new ProgressOption("options.mipmapLevels", 0.0D, 4.0D, 1.0F, (p_92024_) -> {
-      return (double)p_92024_.f_92027_;
-   }, (p_91827_, p_91828_) -> {
-      p_91827_.f_92027_ = (int)p_91828_.doubleValue();
-   }, (p_91821_, p_91822_) -> {
-      double d0 = p_91822_.m_92221_(p_91821_);
-      return (Component)(d0 == 0.0D ? CommonComponents.m_130663_(p_91822_.m_91714_(), false) : p_91822_.m_91764_((int)d0));
-   });
-   public static final ProgressOption f_91673_ = new LogaritmicProgressOption("options.mouseWheelSensitivity", 0.01D, 10.0D, 0.01F, (p_92022_) -> {
-      return p_92022_.f_92033_;
-   }, (p_91810_, p_91811_) -> {
-      p_91810_.f_92033_ = p_91811_;
-   }, (p_91804_, p_91805_) -> {
-      double d0 = p_91805_.m_6917_(p_91805_.m_92221_(p_91804_));
-      return p_91805_.m_91740_(new TextComponent(String.format("%.2f", p_91805_.m_6912_(d0))));
-   });
-   public static final BooleanOption f_91674_ = new BooleanOption("options.rawMouseInput", (p_92020_) -> {
-      return p_92020_.f_92034_;
-   }, (p_92006_, p_92007_) -> {
-      p_92006_.f_92034_ = p_92007_;
+   public static final ProgressOption f_91672_ = new ProgressOption("options.mipmapLevels", 0.0D, 4.0D, 1.0F, (p_216667_0_) -> {
+      return (double)p_216667_0_.f_92027_;
+   }, (p_216585_0_, p_216585_1_) -> {
+      p_216585_0_.f_92027_ = (int)p_216585_1_.doubleValue();
+   }, (p_216629_0_, p_216629_1_) -> {
+      double d0 = p_216629_1_.m_92221_(p_216629_0_);
+      return (Component)(d0 == 0.0D ? CommonComponents.m_130663_(p_216629_1_.m_91714_(), false) : p_216629_1_.m_91764_((int)d0));
+   });
+   public static final ProgressOption f_91673_ = new LogaritmicProgressOption("options.mouseWheelSensitivity", 0.01D, 10.0D, 0.01F, (p_216581_0_) -> {
+      return p_216581_0_.f_92033_;
+   }, (p_216628_0_, p_216628_1_) -> {
+      p_216628_0_.f_92033_ = p_216628_1_;
+   }, (p_216675_0_, p_216675_1_) -> {
+      double d0 = p_216675_1_.m_6917_(p_216675_1_.m_92221_(p_216675_0_));
+      return p_216675_1_.m_91740_(new TextComponent(String.format("%.2f", p_216675_1_.m_6912_(d0))));
+   });
+   public static final BooleanOption f_91674_ = new BooleanOption("options.rawMouseInput", (p_225287_0_) -> {
+      return p_225287_0_.f_92034_;
+   }, (p_225259_0_, p_225259_1_) -> {
+      p_225259_0_.f_92034_ = p_225259_1_;
       Window window = Minecraft.m_91087_().m_91268_();
       if (window != null) {
-         window.m_85424_(p_92007_);
+         window.m_85424_(p_225259_1_);
       }
 
    });
-   public static final ProgressOption f_91675_ = new ProgressOption("options.renderDistance", 2.0D, 16.0D, 1.0F, (p_92018_) -> {
-      return (double)p_92018_.f_92106_;
-   }, (p_91793_, p_91794_) -> {
-      p_91793_.f_92106_ = (int)p_91794_.doubleValue();
+   public static final ProgressOption f_91675_ = new ProgressOption("options.renderDistance", 2.0D, 16.0D, 1.0F, (p_216658_0_) -> {
+      return (double)p_216658_0_.f_92106_;
+   }, (p_216579_0_, p_216579_1_) -> {
+      p_216579_0_.f_92106_ = (int)p_216579_1_.doubleValue();
       Minecraft.m_91087_().f_91060_.m_109826_();
-   }, (p_91787_, p_91788_) -> {
-      double d0 = p_91788_.m_92221_(p_91787_);
-      return p_91788_.m_91740_(new TranslatableComponent("options.chunks", (int)d0));
-   });
-   public static final ProgressOption f_91676_ = new ProgressOption("options.entityDistanceScaling", 0.5D, 5.0D, 0.25F, (p_92016_) -> {
-      return (double)p_92016_.f_92112_;
-   }, (p_91776_, p_91777_) -> {
-      p_91776_.f_92112_ = (float)p_91777_.doubleValue();
-   }, (p_91770_, p_91771_) -> {
-      double d0 = p_91771_.m_92221_(p_91770_);
-      return p_91771_.m_91762_(d0);
-   });
-   public static final ProgressOption f_91677_ = new ProgressOption("options.sensitivity", 0.0D, 1.0D, 0.0F, (p_92014_) -> {
-      return p_92014_.f_92053_;
-   }, (p_91757_, p_91758_) -> {
-      p_91757_.f_92053_ = p_91758_;
-   }, (p_91751_, p_91752_) -> {
-      double d0 = p_91752_.m_6917_(p_91752_.m_92221_(p_91751_));
+   }, (p_216664_0_, p_216664_1_) -> {
+      double d0 = p_216664_1_.m_92221_(p_216664_0_);
+      return p_216664_1_.m_91740_(new TranslatableComponent("options.chunks", (int)d0));
+   });
+   public static final ProgressOption f_91676_ = new ProgressOption("options.entityDistanceScaling", 0.5D, 5.0D, 0.25F, (p_238324_0_) -> {
+      return (double)p_238324_0_.f_92112_;
+   }, (p_238256_0_, p_238256_1_) -> {
+      p_238256_0_.f_92112_ = (float)p_238256_1_.doubleValue();
+   }, (p_238254_0_, p_238254_1_) -> {
+      double d0 = p_238254_1_.m_92221_(p_238254_0_);
+      return p_238254_1_.m_91762_(d0);
+   });
+   public static final ProgressOption f_91677_ = new ProgressOption("options.sensitivity", 0.0D, 1.0D, 0.0F, (p_216654_0_) -> {
+      return p_216654_0_.f_92053_;
+   }, (p_216644_0_, p_216644_1_) -> {
+      p_216644_0_.f_92053_ = p_216644_1_;
+   }, (p_216641_0_, p_216641_1_) -> {
+      double d0 = p_216641_1_.m_6917_(p_216641_1_.m_92221_(p_216641_0_));
       if (d0 == 0.0D) {
-         return p_91752_.m_91740_(new TranslatableComponent("options.sensitivity.min"));
+         return p_216641_1_.m_91740_(new TranslatableComponent("options.sensitivity.min"));
       } else {
-         return d0 == 1.0D ? p_91752_.m_91740_(new TranslatableComponent("options.sensitivity.max")) : p_91752_.m_91762_(2.0D * d0);
+         return d0 == 1.0D ? p_216641_1_.m_91740_(new TranslatableComponent("options.sensitivity.max")) : p_216641_1_.m_91762_(2.0D * d0);
       }
    });
-   public static final ProgressOption f_91678_ = new ProgressOption("options.accessibility.text_background_opacity", 0.0D, 1.0D, 0.0F, (p_92012_) -> {
-      return p_92012_.f_92122_;
-   }, (p_91733_, p_91734_) -> {
-      p_91733_.f_92122_ = p_91734_;
+   public static final ProgressOption f_91678_ = new ProgressOption("options.accessibility.text_background_opacity", 0.0D, 1.0D, 0.0F, (p_216597_0_) -> {
+      return p_216597_0_.f_92122_;
+   }, (p_216593_0_, p_216593_1_) -> {
+      p_216593_0_.f_92122_ = p_216593_1_;
       Minecraft.m_91087_().f_91065_.m_93076_().m_93769_();
-   }, (p_91727_, p_91728_) -> {
-      return p_91728_.m_91762_(p_91728_.m_6917_(p_91728_.m_92221_(p_91727_)));
+   }, (p_216626_0_, p_216626_1_) -> {
+      return p_216626_1_.m_91762_(p_216626_1_.m_6917_(p_216626_1_.m_92221_(p_216626_0_)));
    });
-   public static final CycleOption f_91679_ = new CycleOption("options.ao", (p_91932_, p_91933_) -> {
-      p_91932_.f_92116_ = AmbientOcclusionStatus.m_90487_(p_91932_.f_92116_.m_90486_() + p_91933_);
+   public static final CycleOption f_91679_ = new CycleOption("options.ao", (p_216653_0_, p_216653_1_) -> {
+      p_216653_0_.f_92116_ = AmbientOcclusionStatus.m_90487_(p_216653_0_.f_92116_.m_90486_() + p_216653_1_);
       Minecraft.m_91087_().f_91060_.m_109818_();
-   }, (p_91920_, p_91921_) -> {
-      return p_91921_.m_91740_(new TranslatableComponent(p_91920_.f_92116_.m_90489_()));
-   });
-   public static final CycleOption f_91680_ = new CycleOption("options.attackIndicator", (p_91915_, p_91916_) -> {
-      p_91915_.f_92029_ = AttackIndicatorStatus.m_90509_(p_91915_.f_92029_.m_90508_() + p_91916_);
-   }, (p_91903_, p_91904_) -> {
-      return p_91904_.m_91740_(new TranslatableComponent(p_91903_.f_92029_.m_90511_()));
-   });
-   public static final CycleOption f_91681_ = new CycleOption("options.chat.visibility", (p_91898_, p_91899_) -> {
-      p_91898_.f_92119_ = ChatVisiblity.m_35966_((p_91898_.f_92119_.m_35965_() + p_91899_) % 3);
-   }, (p_91886_, p_91887_) -> {
-      return p_91887_.m_91740_(new TranslatableComponent(p_91886_.f_92119_.m_35968_()));
+   }, (p_216630_0_, p_216630_1_) -> {
+      return p_216630_1_.m_91740_(new TranslatableComponent(p_216630_0_.f_92116_.m_90489_()));
+   });
+   public static final CycleOption f_91680_ = new CycleOption("options.attackIndicator", (p_216615_0_, p_216615_1_) -> {
+      p_216615_0_.f_92029_ = AttackIndicatorStatus.m_90509_(p_216615_0_.f_92029_.m_90508_() + p_216615_1_);
+   }, (p_216609_0_, p_216609_1_) -> {
+      return p_216609_1_.m_91740_(new TranslatableComponent(p_216609_0_.f_92029_.m_90511_()));
+   });
+   public static final CycleOption f_91681_ = new CycleOption("options.chat.visibility", (p_216640_0_, p_216640_1_) -> {
+      p_216640_0_.f_92119_ = ChatVisiblity.m_35966_((p_216640_0_.f_92119_.m_35965_() + p_216640_1_) % 3);
+   }, (p_216598_0_, p_216598_1_) -> {
+      return p_216598_1_.m_91740_(new TranslatableComponent(p_216598_0_.f_92119_.m_35968_()));
    });
    private static final Component f_91654_ = new TranslatableComponent("options.graphics.fast.tooltip");
    private static final Component f_91655_ = new TranslatableComponent("options.graphics.fabulous.tooltip", (new TranslatableComponent("options.graphics.fabulous")).m_130940_(ChatFormatting.ITALIC));
    private static final Component f_91656_ = new TranslatableComponent("options.graphics.fancy.tooltip");
-   public static final CycleOption f_91682_ = new CycleOption("options.graphics", (p_91881_, p_91882_) -> {
+   public static final CycleOption f_91682_ = new CycleOption("options.graphics", (p_216577_0_, p_216577_1_) -> {
       Minecraft minecraft = Minecraft.m_91087_();
       GpuWarnlistManager gpuwarnlistmanager = minecraft.m_91105_();
-      if (p_91881_.f_92115_ == GraphicsStatus.FANCY && gpuwarnlistmanager.m_109240_()) {
+      if (p_216577_0_.f_92115_ == GraphicsStatus.FANCY && gpuwarnlistmanager.m_109240_()) {
          gpuwarnlistmanager.m_109247_();
       } else {
-         p_91881_.f_92115_ = p_91881_.f_92115_.m_90779_();
-         if (p_91881_.f_92115_ == GraphicsStatus.FABULOUS && (!GlStateManager.m_84119_() || gpuwarnlistmanager.m_109251_())) {
-            p_91881_.f_92115_ = GraphicsStatus.FAST;
+         p_216577_0_.f_92115_ = p_216577_0_.f_92115_.m_90779_();
+         if (p_216577_0_.f_92115_ == GraphicsStatus.FABULOUS && (!GlStateManager.m_84119_() || gpuwarnlistmanager.m_109251_())) {
+            p_216577_0_.f_92115_ = GraphicsStatus.FAST;
          }
 
          minecraft.f_91060_.m_109818_();
       }
-   }, (p_91869_, p_91870_) -> {
-      switch(p_91869_.f_92115_) {
+   }, (p_216633_0_, p_216633_1_) -> {
+      switch(p_216633_0_.f_92115_) {
       case FAST:
-         p_91870_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91654_, 200));
+         p_216633_1_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91654_, 200));
          break;
       case FANCY:
-         p_91870_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91656_, 200));
+         p_216633_1_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91656_, 200));
          break;
       case FABULOUS:
-         p_91870_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91655_, 200));
+         p_216633_1_.m_91738_(Minecraft.m_91087_().f_91062_.m_92923_(f_91655_, 200));
       }
 
-      MutableComponent mutablecomponent = new TranslatableComponent(p_91869_.f_92115_.m_90776_());
-      return p_91869_.f_92115_ == GraphicsStatus.FABULOUS ? p_91870_.m_91740_(mutablecomponent.m_130940_(ChatFormatting.ITALIC)) : p_91870_.m_91740_(mutablecomponent);
-   });
-   public static final CycleOption f_91683_ = new CycleOption("options.guiScale", (p_91864_, p_91865_) -> {
-      p_91864_.f_92072_ = Integer.remainderUnsigned(p_91864_.f_92072_ + p_91865_, Minecraft.m_91087_().m_91268_().m_85385_(0, Minecraft.m_91087_().m_91390_()) + 1);
-   }, (p_91852_, p_91853_) -> {
-      return p_91852_.f_92072_ == 0 ? p_91853_.m_91740_(new TranslatableComponent("options.guiScale.auto")) : p_91853_.m_91764_(p_91852_.f_92072_);
-   });
-   public static final CycleOption f_91684_ = new CycleOption("options.mainHand", (p_91847_, p_91848_) -> {
-      p_91847_.f_92127_ = p_91847_.f_92127_.m_20828_();
-   }, (p_91835_, p_91836_) -> {
-      return p_91836_.m_91740_(p_91835_.f_92127_.m_20829_());
-   });
-   public static final CycleOption f_91627_ = new CycleOption("options.narrator", (p_91830_, p_91831_) -> {
+      MutableComponent mutablecomponent = new TranslatableComponent(p_216633_0_.f_92115_.m_90776_());
+      return p_216633_0_.f_92115_ == GraphicsStatus.FABULOUS ? p_216633_1_.m_91740_(mutablecomponent.m_130940_(ChatFormatting.ITALIC)) : p_216633_1_.m_91740_(mutablecomponent);
+   });
+   public static final CycleOption f_91683_ = new CycleOption("options.guiScale", (p_216674_0_, p_216674_1_) -> {
+      p_216674_0_.f_92072_ = Integer.remainderUnsigned(p_216674_0_.f_92072_ + p_216674_1_, Minecraft.m_91087_().m_91268_().m_85385_(0, Minecraft.m_91087_().m_91390_()) + 1);
+   }, (p_216668_0_, p_216668_1_) -> {
+      return p_216668_0_.f_92072_ == 0 ? p_216668_1_.m_91740_(new TranslatableComponent("options.guiScale.auto")) : p_216668_1_.m_91764_(p_216668_0_.f_92072_);
+   });
+   public static final CycleOption f_91684_ = new CycleOption("options.mainHand", (p_216584_0_, p_216584_1_) -> {
+      p_216584_0_.f_92127_ = p_216584_0_.f_92127_.m_20828_();
+   }, (p_216596_0_, p_216596_1_) -> {
+      return p_216596_1_.m_91740_(p_216596_0_.f_92127_.m_20829_());
+   });
+   public static final CycleOption f_91627_ = new CycleOption("options.narrator", (p_216648_0_, p_216648_1_) -> {
       if (NarratorChatListener.f_93311_.m_93316_()) {
-         p_91830_.f_92074_ = NarratorStatus.m_91619_(p_91830_.f_92074_.m_91618_() + p_91831_);
+         p_216648_0_.f_92074_ = NarratorStatus.m_91619_(p_216648_0_.f_92074_.m_91618_() + p_216648_1_);
       } else {
-         p_91830_.f_92074_ = NarratorStatus.OFF;
+         p_216648_0_.f_92074_ = NarratorStatus.OFF;
       }
 
-      NarratorChatListener.f_93311_.m_93317_(p_91830_.f_92074_);
-   }, (p_91818_, p_91819_) -> {
-      return NarratorChatListener.f_93311_.m_93316_() ? p_91819_.m_91740_(p_91818_.f_92074_.m_91621_()) : p_91819_.m_91740_(new TranslatableComponent("options.narrator.notavailable"));
-   });
-   public static final CycleOption f_91628_ = new CycleOption("options.particles", (p_91813_, p_91814_) -> {
-      p_91813_.f_92073_ = ParticleStatus.m_92196_(p_91813_.f_92073_.m_92198_() + p_91814_);
-   }, (p_91801_, p_91802_) -> {
-      return p_91802_.m_91740_(new TranslatableComponent(p_91801_.f_92073_.m_92195_()));
-   });
-   public static final CycleOption f_91629_ = new CycleOption("options.renderClouds", (p_91796_, p_91797_) -> {
-      p_91796_.f_92114_ = CloudStatus.m_90664_(p_91796_.f_92114_.m_90663_() + p_91797_);
+      NarratorChatListener.f_93311_.m_93317_(p_216648_0_.f_92074_);
+   }, (p_216632_0_, p_216632_1_) -> {
+      return NarratorChatListener.f_93311_.m_93316_() ? p_216632_1_.m_91740_(p_216632_0_.f_92074_.m_91621_()) : p_216632_1_.m_91740_(new TranslatableComponent("options.narrator.notavailable"));
+   });
+   public static final CycleOption f_91628_ = new CycleOption("options.particles", (p_216622_0_, p_216622_1_) -> {
+      p_216622_0_.f_92073_ = ParticleStatus.m_92196_(p_216622_0_.f_92073_.m_92198_() + p_216622_1_);
+   }, (p_216616_0_, p_216616_1_) -> {
+      return p_216616_1_.m_91740_(new TranslatableComponent(p_216616_0_.f_92073_.m_92195_()));
+   });
+   public static final CycleOption f_91629_ = new CycleOption("options.renderClouds", (p_216605_0_, p_216605_1_) -> {
+      p_216605_0_.f_92114_ = CloudStatus.m_90664_(p_216605_0_.f_92114_.m_90663_() + p_216605_1_);
       if (Minecraft.m_91085_()) {
          RenderTarget rendertarget = Minecraft.m_91087_().f_91060_.m_109832_();
          if (rendertarget != null) {
@@ -291,132 +_,132 @@
          }
       }
 
-   }, (p_91784_, p_91785_) -> {
-      return p_91785_.m_91740_(new TranslatableComponent(p_91784_.f_92114_.m_90666_()));
+   }, (p_216602_0_, p_216602_1_) -> {
+      return p_216602_1_.m_91740_(new TranslatableComponent(p_216602_0_.f_92114_.m_90666_()));
    });
-   public static final CycleOption f_91630_ = new CycleOption("options.accessibility.text_background", (p_91779_, p_91780_) -> {
-      p_91779_.f_92050_ = !p_91779_.f_92050_;
-   }, (p_91767_, p_91768_) -> {
-      return p_91768_.m_91740_(new TranslatableComponent(p_91767_.f_92050_ ? "options.accessibility.text_background.chat" : "options.accessibility.text_background.everywhere"));
+   public static final CycleOption f_91630_ = new CycleOption("options.accessibility.text_background", (p_216665_0_, p_216665_1_) -> {
+      p_216665_0_.f_92050_ = !p_216665_0_.f_92050_;
+   }, (p_216639_0_, p_216639_1_) -> {
+      return p_216639_1_.m_91740_(new TranslatableComponent(p_216639_0_.f_92050_ ? "options.accessibility.text_background.chat" : "options.accessibility.text_background.everywhere"));
    });
    private static final Component f_91657_ = new TranslatableComponent("options.hideMatchedNames.tooltip");
-   public static final BooleanOption f_91631_ = new BooleanOption("options.autoJump", (p_92001_) -> {
-      return p_92001_.f_92036_;
-   }, (p_91995_, p_91996_) -> {
-      p_91995_.f_92036_ = p_91996_;
-   });
-   public static final BooleanOption f_91632_ = new BooleanOption("options.autoSuggestCommands", (p_91990_) -> {
-      return p_91990_.f_92037_;
-   }, (p_91984_, p_91985_) -> {
-      p_91984_.f_92037_ = p_91985_;
-   });
-   public static final BooleanOption f_91633_ = new BooleanOption("options.hideMatchedNames", f_91657_, (p_91979_) -> {
-      return p_91979_.f_92084_;
-   }, (p_91973_, p_91974_) -> {
-      p_91973_.f_92084_ = p_91974_;
-   });
-   public static final BooleanOption f_91634_ = new BooleanOption("options.chat.color", (p_91968_) -> {
-      return p_91968_.f_92038_;
-   }, (p_91962_, p_91963_) -> {
-      p_91962_.f_92038_ = p_91963_;
-   });
-   public static final BooleanOption f_91635_ = new BooleanOption("options.chat.links", (p_91957_) -> {
-      return p_91957_.f_92039_;
-   }, (p_91951_, p_91952_) -> {
-      p_91951_.f_92039_ = p_91952_;
-   });
-   public static final BooleanOption f_91636_ = new BooleanOption("options.chat.links.prompt", (p_91946_) -> {
-      return p_91946_.f_92040_;
-   }, (p_91940_, p_91941_) -> {
-      p_91940_.f_92040_ = p_91941_;
-   });
-   public static final BooleanOption f_91637_ = new BooleanOption("options.discrete_mouse_scroll", (p_91935_) -> {
-      return p_91935_.f_92045_;
-   }, (p_91926_, p_91927_) -> {
-      p_91926_.f_92045_ = p_91927_;
-   });
-   public static final BooleanOption f_91638_ = new BooleanOption("options.vsync", (p_91918_) -> {
-      return p_91918_.f_92041_;
-   }, (p_91909_, p_91910_) -> {
-      p_91909_.f_92041_ = p_91910_;
+   public static final BooleanOption f_91631_ = new BooleanOption("options.autoJump", (p_216619_0_) -> {
+      return p_216619_0_.f_92036_;
+   }, (p_216621_0_, p_216621_1_) -> {
+      p_216621_0_.f_92036_ = p_216621_1_;
+   });
+   public static final BooleanOption f_91632_ = new BooleanOption("options.autoSuggestCommands", (p_216643_0_) -> {
+      return p_216643_0_.f_92037_;
+   }, (p_216656_0_, p_216656_1_) -> {
+      p_216656_0_.f_92037_ = p_216656_1_;
+   });
+   public static final BooleanOption f_91633_ = new BooleanOption("options.hideMatchedNames", f_91657_, (p_244790_0_) -> {
+      return p_244790_0_.f_92084_;
+   }, (p_244791_0_, p_244791_1_) -> {
+      p_244791_0_.f_92084_ = p_244791_1_;
+   });
+   public static final BooleanOption f_91634_ = new BooleanOption("options.chat.color", (p_216669_0_) -> {
+      return p_216669_0_.f_92038_;
+   }, (p_216659_0_, p_216659_1_) -> {
+      p_216659_0_.f_92038_ = p_216659_1_;
+   });
+   public static final BooleanOption f_91635_ = new BooleanOption("options.chat.links", (p_216583_0_) -> {
+      return p_216583_0_.f_92039_;
+   }, (p_216670_0_, p_216670_1_) -> {
+      p_216670_0_.f_92039_ = p_216670_1_;
+   });
+   public static final BooleanOption f_91636_ = new BooleanOption("options.chat.links.prompt", (p_216610_0_) -> {
+      return p_216610_0_.f_92040_;
+   }, (p_216652_0_, p_216652_1_) -> {
+      p_216652_0_.f_92040_ = p_216652_1_;
+   });
+   public static final BooleanOption f_91637_ = new BooleanOption("options.discrete_mouse_scroll", (p_216634_0_) -> {
+      return p_216634_0_.f_92045_;
+   }, (p_216625_0_, p_216625_1_) -> {
+      p_216625_0_.f_92045_ = p_216625_1_;
+   });
+   public static final BooleanOption f_91638_ = new BooleanOption("options.vsync", (p_216661_0_) -> {
+      return p_216661_0_.f_92041_;
+   }, (p_216635_0_, p_216635_1_) -> {
+      p_216635_0_.f_92041_ = p_216635_1_;
       if (Minecraft.m_91087_().m_91268_() != null) {
-         Minecraft.m_91087_().m_91268_().m_85409_(p_91909_.f_92041_);
+         Minecraft.m_91087_().m_91268_().m_85409_(p_216635_0_.f_92041_);
       }
 
    });
-   public static final BooleanOption f_91639_ = new BooleanOption("options.entityShadows", (p_91901_) -> {
-      return p_91901_.f_92042_;
-   }, (p_91892_, p_91893_) -> {
-      p_91892_.f_92042_ = p_91893_;
+   public static final BooleanOption f_91639_ = new BooleanOption("options.entityShadows", (p_216576_0_) -> {
+      return p_216576_0_.f_92042_;
+   }, (p_216588_0_, p_216588_1_) -> {
+      p_216588_0_.f_92042_ = p_216588_1_;
    });
-   public static final BooleanOption f_91640_ = new BooleanOption("options.forceUnicodeFont", (p_91884_) -> {
-      return p_91884_.f_92043_;
-   }, (p_91875_, p_91876_) -> {
-      p_91875_.f_92043_ = p_91876_;
+   public static final BooleanOption f_91640_ = new BooleanOption("options.forceUnicodeFont", (p_216657_0_) -> {
+      return p_216657_0_.f_92043_;
+   }, (p_216631_0_, p_216631_1_) -> {
+      p_216631_0_.f_92043_ = p_216631_1_;
       Minecraft minecraft = Minecraft.m_91087_();
       if (minecraft.m_91268_() != null) {
-         minecraft.m_91336_(p_91876_);
+         minecraft.m_91336_(p_216631_1_);
       }
 
    });
-   public static final BooleanOption f_91641_ = new BooleanOption("options.invertMouse", (p_91867_) -> {
-      return p_91867_.f_92044_;
-   }, (p_91858_, p_91859_) -> {
-      p_91858_.f_92044_ = p_91859_;
-   });
-   public static final BooleanOption f_91642_ = new BooleanOption("options.realmsNotifications", (p_91850_) -> {
-      return p_91850_.f_92046_;
-   }, (p_91841_, p_91842_) -> {
-      p_91841_.f_92046_ = p_91842_;
-   });
-   public static final BooleanOption f_91643_ = new BooleanOption("options.reducedDebugInfo", (p_91833_) -> {
-      return p_91833_.f_92047_;
-   }, (p_91824_, p_91825_) -> {
-      p_91824_.f_92047_ = p_91825_;
-   });
-   public static final BooleanOption f_91644_ = new BooleanOption("options.showSubtitles", (p_91816_) -> {
-      return p_91816_.f_92049_;
-   }, (p_91807_, p_91808_) -> {
-      p_91807_.f_92049_ = p_91808_;
-   });
-   public static final BooleanOption f_91645_ = new BooleanOption("options.snooper", (p_91799_) -> {
-      if (p_91799_.f_92048_) {
+   public static final BooleanOption f_91641_ = new BooleanOption("options.invertMouse", (p_216627_0_) -> {
+      return p_216627_0_.f_92044_;
+   }, (p_216603_0_, p_216603_1_) -> {
+      p_216603_0_.f_92044_ = p_216603_1_;
+   });
+   public static final BooleanOption f_91642_ = new BooleanOption("options.realmsNotifications", (p_216606_0_) -> {
+      return p_216606_0_.f_92046_;
+   }, (p_216618_0_, p_216618_1_) -> {
+      p_216618_0_.f_92046_ = p_216618_1_;
+   });
+   public static final BooleanOption f_91643_ = new BooleanOption("options.reducedDebugInfo", (p_216582_0_) -> {
+      return p_216582_0_.f_92047_;
+   }, (p_216613_0_, p_216613_1_) -> {
+      p_216613_0_.f_92047_ = p_216613_1_;
+   });
+   public static final BooleanOption f_91644_ = new BooleanOption("options.showSubtitles", (p_216663_0_) -> {
+      return p_216663_0_.f_92049_;
+   }, (p_216662_0_, p_216662_1_) -> {
+      p_216662_0_.f_92049_ = p_216662_1_;
+   });
+   public static final BooleanOption f_91645_ = new BooleanOption("options.snooper", (p_216638_0_) -> {
+      if (p_216638_0_.f_92048_) {
       }
 
       return false;
-   }, (p_91790_, p_91791_) -> {
-      p_91790_.f_92048_ = p_91791_;
-   });
-   public static final CycleOption f_91646_ = new CycleOption("key.sneak", (p_91760_, p_91761_) -> {
-      p_91760_.f_92081_ = !p_91760_.f_92081_;
-   }, (p_91748_, p_91749_) -> {
-      return p_91749_.m_91740_(new TranslatableComponent(p_91748_.f_92081_ ? "options.key.toggle" : "options.key.hold"));
-   });
-   public static final CycleOption f_91647_ = new CycleOption("key.sprint", (p_91736_, p_91737_) -> {
-      p_91736_.f_92082_ = !p_91736_.f_92082_;
-   }, (p_91724_, p_91725_) -> {
-      return p_91725_.m_91740_(new TranslatableComponent(p_91724_.f_92082_ ? "options.key.toggle" : "options.key.hold"));
-   });
-   public static final BooleanOption f_91648_ = new BooleanOption("options.touchscreen", (p_91782_) -> {
-      return p_91782_.f_92051_;
-   }, (p_91773_, p_91774_) -> {
-      p_91773_.f_92051_ = p_91774_;
-   });
-   public static final BooleanOption f_91649_ = new BooleanOption("options.fullscreen", (p_91746_) -> {
-      return p_91746_.f_92052_;
-   }, (p_91754_, p_91755_) -> {
-      p_91754_.f_92052_ = p_91755_;
+   }, (p_216676_0_, p_216676_1_) -> {
+      p_216676_0_.f_92048_ = p_216676_1_;
+   });
+   public static final CycleOption f_91646_ = new CycleOption("key.sneak", (p_228043_0_, p_228043_1_) -> {
+      p_228043_0_.f_92081_ = !p_228043_0_.f_92081_;
+   }, (p_228041_0_, p_228041_1_) -> {
+      return p_228041_1_.m_91740_(new TranslatableComponent(p_228041_0_.f_92081_ ? "options.key.toggle" : "options.key.hold"));
+   });
+   public static final CycleOption f_91647_ = new CycleOption("key.sprint", (p_228039_0_, p_228039_1_) -> {
+      p_228039_0_.f_92082_ = !p_228039_0_.f_92082_;
+   }, (p_228037_0_, p_228037_1_) -> {
+      return p_228037_1_.m_91740_(new TranslatableComponent(p_228037_0_.f_92082_ ? "options.key.toggle" : "options.key.hold"));
+   });
+   public static final BooleanOption f_91648_ = new BooleanOption("options.touchscreen", (p_216647_0_) -> {
+      return p_216647_0_.f_92051_;
+   }, (p_216580_0_, p_216580_1_) -> {
+      p_216580_0_.f_92051_ = p_216580_1_;
+   });
+   public static final BooleanOption f_91649_ = new BooleanOption("options.fullscreen", (p_228040_0_) -> {
+      return p_228040_0_.f_92052_;
+   }, (p_228042_0_, p_228042_1_) -> {
+      p_228042_0_.f_92052_ = p_228042_1_;
       Minecraft minecraft = Minecraft.m_91087_();
-      if (minecraft.m_91268_() != null && minecraft.m_91268_().m_85440_() != p_91754_.f_92052_) {
+      if (minecraft.m_91268_() != null && minecraft.m_91268_().m_85440_() != p_228042_0_.f_92052_) {
          minecraft.m_91268_().m_85438_();
-         p_91754_.f_92052_ = minecraft.m_91268_().m_85440_();
+         p_228042_0_.f_92052_ = minecraft.m_91268_().m_85440_();
       }
 
    });
-   public static final BooleanOption f_91650_ = new BooleanOption("options.viewBobbing", (p_91718_) -> {
-      return p_91718_.f_92080_;
-   }, (p_91730_, p_91731_) -> {
-      p_91730_.f_92080_ = p_91731_;
+   public static final BooleanOption f_91650_ = new BooleanOption("options.viewBobbing", (p_228036_0_) -> {
+      return p_228036_0_.f_92080_;
+   }, (p_228038_0_, p_228038_1_) -> {
+      p_228038_0_.f_92080_ = p_228038_1_;
    });
    private final Component f_91658_;
    private Optional<List<FormattedCharSequence>> f_91659_ = Optional.empty();
