--- a/net/minecraft/commands/arguments/ColorArgument.java
+++ b/net/minecraft/commands/arguments/ColorArgument.java
@@ -17,8 +_,8 @@
 
 public class ColorArgument implements ArgumentType<ChatFormatting> {
    private static final Collection<String> f_85460_ = Arrays.asList("red", "green");
-   public static final DynamicCommandExceptionType f_85459_ = new DynamicCommandExceptionType((p_85470_) -> {
-      return new TranslatableComponent("argument.color.invalid", p_85470_);
+   public static final DynamicCommandExceptionType f_85459_ = new DynamicCommandExceptionType((p_208659_0_) -> {
+      return new TranslatableComponent("argument.color.invalid", p_208659_0_);
    });
 
    private ColorArgument() {
