--- a/net/minecraft/commands/arguments/ComponentArgument.java
+++ b/net/minecraft/commands/arguments/ComponentArgument.java
@@ -14,8 +_,8 @@
 
 public class ComponentArgument implements ArgumentType<Component> {
    private static final Collection<String> f_87111_ = Arrays.asList("\"hello world\"", "\"\"", "\"{\"text\":\"hello world\"}", "[\"\"]");
-   public static final DynamicCommandExceptionType f_87110_ = new DynamicCommandExceptionType((p_87121_) -> {
-      return new TranslatableComponent("argument.component.invalid", p_87121_);
+   public static final DynamicCommandExceptionType f_87110_ = new DynamicCommandExceptionType((p_208660_0_) -> {
+      return new TranslatableComponent("argument.component.invalid", p_208660_0_);
    });
 
    private ComponentArgument() {
