--- a/net/minecraft/commands/arguments/DimensionArgument.java
+++ b/net/minecraft/commands/arguments/DimensionArgument.java
@@ -21,11 +_,11 @@
 import net.minecraft.world.level.Level;
 
 public class DimensionArgument implements ArgumentType<ResourceLocation> {
-   private static final Collection<String> f_88801_ = Stream.of(Level.f_46428_, Level.f_46429_).map((p_88814_) -> {
-      return p_88814_.m_135782_().toString();
+   private static final Collection<String> f_88801_ = Stream.of(Level.f_46428_, Level.f_46429_).map((p_212593_0_) -> {
+      return p_212593_0_.m_135782_().toString();
    }).collect(Collectors.toList());
-   private static final DynamicCommandExceptionType f_88802_ = new DynamicCommandExceptionType((p_88812_) -> {
-      return new TranslatableComponent("argument.dimension.invalid", p_88812_);
+   private static final DynamicCommandExceptionType f_88802_ = new DynamicCommandExceptionType((p_212594_0_) -> {
+      return new TranslatableComponent("argument.dimension.invalid", p_212594_0_);
    });
 
    public ResourceLocation parse(StringReader p_88807_) throws CommandSyntaxException {
