--- a/net/minecraft/commands/arguments/EntityAnchorArgument.java
+++ b/net/minecraft/commands/arguments/EntityAnchorArgument.java
@@ -23,8 +_,8 @@
 
 public class EntityAnchorArgument implements ArgumentType<EntityAnchorArgument.Anchor> {
    private static final Collection<String> f_90346_ = Arrays.asList("eyes", "feet");
-   private static final DynamicCommandExceptionType f_90347_ = new DynamicCommandExceptionType((p_90357_) -> {
-      return new TranslatableComponent("argument.anchor.invalid", p_90357_);
+   private static final DynamicCommandExceptionType f_90347_ = new DynamicCommandExceptionType((p_208661_0_) -> {
+      return new TranslatableComponent("argument.anchor.invalid", p_208661_0_);
    });
 
    public static EntityAnchorArgument.Anchor m_90353_(CommandContext<CommandSourceStack> p_90354_, String p_90355_) {
@@ -56,16 +_,16 @@
    }
 
    public static enum Anchor {
-      FEET("feet", (p_90389_, p_90390_) -> {
-         return p_90389_;
+      FEET("feet", (p_201019_0_, p_201019_1_) -> {
+         return p_201019_0_;
       }),
-      EYES("eyes", (p_90382_, p_90383_) -> {
-         return new Vec3(p_90382_.f_82479_, p_90382_.f_82480_ + (double)p_90383_.m_20192_(), p_90382_.f_82481_);
+      EYES("eyes", (p_201018_0_, p_201018_1_) -> {
+         return new Vec3(p_201018_0_.f_82479_, p_201018_0_.f_82480_ + (double)p_201018_1_.m_20192_(), p_201018_0_.f_82481_);
       });
 
-      private static final Map<String, EntityAnchorArgument.Anchor> f_90366_ = Util.m_137469_(Maps.newHashMap(), (p_90387_) -> {
+      private static final Map<String, EntityAnchorArgument.Anchor> f_90366_ = Util.m_137469_(Maps.newHashMap(), (p_209384_0_) -> {
          for(EntityAnchorArgument.Anchor entityanchorargument$anchor : values()) {
-            p_90387_.put(entityanchorargument$anchor.f_90367_, entityanchorargument$anchor);
+            p_209384_0_.put(entityanchorargument$anchor.f_90367_, entityanchorargument$anchor);
          }
 
       });
