--- a/net/minecraft/commands/arguments/EntitySummonArgument.java
+++ b/net/minecraft/commands/arguments/EntitySummonArgument.java
@@ -15,8 +_,8 @@
 
 public class EntitySummonArgument implements ArgumentType<ResourceLocation> {
    private static final Collection<String> f_93332_ = Arrays.asList("minecraft:pig", "cow");
-   public static final DynamicCommandExceptionType f_93331_ = new DynamicCommandExceptionType((p_93342_) -> {
-      return new TranslatableComponent("entity.notFound", p_93342_);
+   public static final DynamicCommandExceptionType f_93331_ = new DynamicCommandExceptionType((p_211367_0_) -> {
+      return new TranslatableComponent("entity.notFound", p_211367_0_);
    });
 
    public static EntitySummonArgument m_93335_() {
