--- a/net/minecraft/commands/arguments/GameProfileArgument.java
+++ b/net/minecraft/commands/arguments/GameProfileArgument.java
@@ -50,8 +_,8 @@
          }
 
          String s = p_94586_.getString().substring(i, p_94586_.getCursor());
-         return (p_94595_) -> {
-            GameProfile gameprofile = p_94595_.m_81377_().m_129927_().m_10996_(s);
+         return (p_197107_1_) -> {
+            GameProfile gameprofile = p_197107_1_.m_81377_().m_129927_().m_10996_(s);
             if (gameprofile == null) {
                throw f_94580_.create();
             } else {
@@ -72,8 +_,8 @@
          } catch (CommandSyntaxException commandsyntaxexception) {
          }
 
-         return entityselectorparser.m_121249_(p_94599_, (p_94589_) -> {
-            SharedSuggestionProvider.m_82970_(((SharedSuggestionProvider)p_94598_.getSource()).m_5982_(), p_94589_);
+         return entityselectorparser.m_121249_(p_94599_, (p_201943_1_) -> {
+            SharedSuggestionProvider.m_82970_(((SharedSuggestionProvider)p_94598_.getSource()).m_5982_(), p_201943_1_);
          });
       } else {
          return Suggestions.empty();
