--- a/net/minecraft/commands/arguments/NbtPathArgument.java
+++ b/net/minecraft/commands/arguments/NbtPathArgument.java
@@ -30,8 +_,8 @@
 public class NbtPathArgument implements ArgumentType<NbtPathArgument.NbtPath> {
    private static final Collection<String> f_99484_ = Arrays.asList("foo", "foo.bar", "foo[0]", "[0]", "[]", "{foo=bar}");
    public static final SimpleCommandExceptionType f_99482_ = new SimpleCommandExceptionType(new TranslatableComponent("arguments.nbtpath.node.invalid"));
-   public static final DynamicCommandExceptionType f_99483_ = new DynamicCommandExceptionType((p_99502_) -> {
-      return new TranslatableComponent("arguments.nbtpath.nothing_found", p_99502_);
+   public static final DynamicCommandExceptionType f_99483_ = new DynamicCommandExceptionType((p_208665_0_) -> {
+      return new TranslatableComponent("arguments.nbtpath.nothing_found", p_208665_0_);
    });
 
    public static NbtPathArgument m_99487_() {
@@ -131,11 +_,101 @@
    }
 
    private static Predicate<Tag> m_99510_(CompoundTag p_99511_) {
-      return (p_99507_) -> {
-         return NbtUtils.m_129235_(p_99511_, p_99507_, true);
+      return (p_218081_1_) -> {
+         return NbtUtils.m_129235_(p_99511_, p_218081_1_, true);
       };
    }
 
+   static class IndexedElementNode implements NbtPathArgument.Node {
+      private final int f_99547_;
+
+      public IndexedElementNode(int p_99549_) {
+         this.f_99547_ = p_99549_;
+      }
+
+      public void m_7273_(Tag p_99554_, List<Tag> p_99555_) {
+         if (p_99554_ instanceof CollectionTag) {
+            CollectionTag<?> collectiontag = (CollectionTag)p_99554_;
+            int i = collectiontag.size();
+            int j = this.f_99547_ < 0 ? i + this.f_99547_ : this.f_99547_;
+            if (0 <= j && j < i) {
+               p_99555_.add(collectiontag.get(j));
+            }
+         }
+
+      }
+
+      public void m_7876_(Tag p_99560_, Supplier<Tag> p_99561_, List<Tag> p_99562_) {
+         this.m_7273_(p_99560_, p_99562_);
+      }
+
+      public Tag m_7510_() {
+         return new ListTag();
+      }
+
+      public int m_5571_(Tag p_99557_, Supplier<Tag> p_99558_) {
+         if (p_99557_ instanceof CollectionTag) {
+            CollectionTag<?> collectiontag = (CollectionTag)p_99557_;
+            int i = collectiontag.size();
+            int j = this.f_99547_ < 0 ? i + this.f_99547_ : this.f_99547_;
+            if (0 <= j && j < i) {
+               Tag tag = collectiontag.get(j);
+               Tag tag1 = p_99558_.get();
+               if (!tag1.equals(tag) && collectiontag.m_7615_(j, tag1)) {
+                  return 1;
+               }
+            }
+         }
+
+         return 0;
+      }
+
+      public int m_6015_(Tag p_99552_) {
+         if (p_99552_ instanceof CollectionTag) {
+            CollectionTag<?> collectiontag = (CollectionTag)p_99552_;
+            int i = collectiontag.size();
+            int j = this.f_99547_ < 0 ? i + this.f_99547_ : this.f_99547_;
+            if (0 <= j && j < i) {
+               collectiontag.remove(j);
+               return 1;
+            }
+         }
+
+         return 0;
+      }
+   }
+
+   static class MatchRootObjectNode implements NbtPathArgument.Node {
+      private final Predicate<Tag> f_99603_;
+
+      public MatchRootObjectNode(CompoundTag p_99605_) {
+         this.f_99603_ = NbtPathArgument.m_99510_(p_99605_);
+      }
+
+      public void m_7273_(Tag p_99610_, List<Tag> p_99611_) {
+         if (p_99610_ instanceof CompoundTag && this.f_99603_.test(p_99610_)) {
+            p_99611_.add(p_99610_);
+         }
+
+      }
+
+      public void m_7876_(Tag p_99616_, Supplier<Tag> p_99617_, List<Tag> p_99618_) {
+         this.m_7273_(p_99616_, p_99618_);
+      }
+
+      public Tag m_7510_() {
+         return new CompoundTag();
+      }
+
+      public int m_5571_(Tag p_99613_, Supplier<Tag> p_99614_) {
+         return 0;
+      }
+
+      public int m_6015_(Tag p_99608_) {
+         return 0;
+      }
+   }
+
    static class AllElementsNode implements NbtPathArgument.Node {
       public static final NbtPathArgument.AllElementsNode f_99515_ = new NbtPathArgument.AllElementsNode();
 
@@ -212,105 +_,86 @@
       }
    }
 
-   static class CompoundChildNode implements NbtPathArgument.Node {
-      private final String f_99531_;
-
-      public CompoundChildNode(String p_99533_) {
-         this.f_99531_ = p_99533_;
-      }
-
-      public void m_7273_(Tag p_99538_, List<Tag> p_99539_) {
-         if (p_99538_ instanceof CompoundTag) {
-            Tag tag = ((CompoundTag)p_99538_).m_128423_(this.f_99531_);
-            if (tag != null) {
-               p_99539_.add(tag);
-            }
-         }
-
-      }
-
-      public void m_7876_(Tag p_99544_, Supplier<Tag> p_99545_, List<Tag> p_99546_) {
-         if (p_99544_ instanceof CompoundTag) {
-            CompoundTag compoundtag = (CompoundTag)p_99544_;
-            Tag tag;
-            if (compoundtag.m_128441_(this.f_99531_)) {
-               tag = compoundtag.m_128423_(this.f_99531_);
-            } else {
-               tag = p_99545_.get();
-               compoundtag.m_128365_(this.f_99531_, tag);
-            }
-
-            p_99546_.add(tag);
-         }
-
-      }
-
-      public Tag m_7510_() {
-         return new CompoundTag();
-      }
-
-      public int m_5571_(Tag p_99541_, Supplier<Tag> p_99542_) {
-         if (p_99541_ instanceof CompoundTag) {
-            CompoundTag compoundtag = (CompoundTag)p_99541_;
-            Tag tag = p_99542_.get();
-            Tag tag1 = compoundtag.m_128365_(this.f_99531_, tag);
-            if (!tag.equals(tag1)) {
-               return 1;
-            }
-         }
-
-         return 0;
-      }
-
-      public int m_6015_(Tag p_99536_) {
-         if (p_99536_ instanceof CompoundTag) {
-            CompoundTag compoundtag = (CompoundTag)p_99536_;
-            if (compoundtag.m_128441_(this.f_99531_)) {
-               compoundtag.m_128473_(this.f_99531_);
-               return 1;
-            }
-         }
-
-         return 0;
+   interface Node {
+      void m_7273_(Tag p_99666_, List<Tag> p_99667_);
+
+      void m_7876_(Tag p_99670_, Supplier<Tag> p_99671_, List<Tag> p_99672_);
+
+      Tag m_7510_();
+
+      int m_5571_(Tag p_99668_, Supplier<Tag> p_99669_);
+
+      int m_6015_(Tag p_99665_);
+
+      default List<Tag> m_99653_(List<Tag> p_99654_) {
+         return this.m_99655_(p_99654_, this::m_7273_);
+      }
+
+      default List<Tag> m_99658_(List<Tag> p_99659_, Supplier<Tag> p_99660_) {
+         return this.m_99655_(p_99659_, (p_218055_2_, p_218055_3_) -> {
+            this.m_7876_(p_218055_2_, p_99660_, p_218055_3_);
+         });
+      }
+
+      default List<Tag> m_99655_(List<Tag> p_99656_, BiConsumer<Tag, List<Tag>> p_99657_) {
+         List<Tag> list = Lists.newArrayList();
+
+         for(Tag tag : p_99656_) {
+            p_99657_.accept(tag, list);
+         }
+
+         return list;
       }
    }
 
-   static class IndexedElementNode implements NbtPathArgument.Node {
-      private final int f_99547_;
-
-      public IndexedElementNode(int p_99549_) {
-         this.f_99547_ = p_99549_;
-      }
-
-      public void m_7273_(Tag p_99554_, List<Tag> p_99555_) {
-         if (p_99554_ instanceof CollectionTag) {
-            CollectionTag<?> collectiontag = (CollectionTag)p_99554_;
-            int i = collectiontag.size();
-            int j = this.f_99547_ < 0 ? i + this.f_99547_ : this.f_99547_;
-            if (0 <= j && j < i) {
-               p_99555_.add(collectiontag.get(j));
-            }
-         }
-
-      }
-
-      public void m_7876_(Tag p_99560_, Supplier<Tag> p_99561_, List<Tag> p_99562_) {
-         this.m_7273_(p_99560_, p_99562_);
+   static class MatchObjectNode implements NbtPathArgument.Node {
+      private final String f_99584_;
+      private final CompoundTag f_99585_;
+      private final Predicate<Tag> f_99586_;
+
+      public MatchObjectNode(String p_99588_, CompoundTag p_99589_) {
+         this.f_99584_ = p_99588_;
+         this.f_99585_ = p_99589_;
+         this.f_99586_ = NbtPathArgument.m_99510_(p_99589_);
+      }
+
+      public void m_7273_(Tag p_99594_, List<Tag> p_99595_) {
+         if (p_99594_ instanceof CompoundTag) {
+            Tag tag = ((CompoundTag)p_99594_).m_128423_(this.f_99584_);
+            if (this.f_99586_.test(tag)) {
+               p_99595_.add(tag);
+            }
+         }
+
+      }
+
+      public void m_7876_(Tag p_99600_, Supplier<Tag> p_99601_, List<Tag> p_99602_) {
+         if (p_99600_ instanceof CompoundTag) {
+            CompoundTag compoundtag = (CompoundTag)p_99600_;
+            Tag tag = compoundtag.m_128423_(this.f_99584_);
+            if (tag == null) {
+               Tag compoundnbt1 = this.f_99585_.m_6426_();
+               compoundtag.m_128365_(this.f_99584_, compoundnbt1);
+               p_99602_.add(compoundnbt1);
+            } else if (this.f_99586_.test(tag)) {
+               p_99602_.add(tag);
+            }
+         }
+
       }
 
       public Tag m_7510_() {
-         return new ListTag();
+         return new CompoundTag();
       }
 
-      public int m_5571_(Tag p_99557_, Supplier<Tag> p_99558_) {
-         if (p_99557_ instanceof CollectionTag) {
-            CollectionTag<?> collectiontag = (CollectionTag)p_99557_;
-            int i = collectiontag.size();
-            int j = this.f_99547_ < 0 ? i + this.f_99547_ : this.f_99547_;
-            if (0 <= j && j < i) {
-               Tag tag = collectiontag.get(j);
-               Tag tag1 = p_99558_.get();
-               if (!tag1.equals(tag) && collectiontag.m_7615_(j, tag1)) {
+      public int m_5571_(Tag p_99597_, Supplier<Tag> p_99598_) {
+         if (p_99597_ instanceof CompoundTag) {
+            CompoundTag compoundtag = (CompoundTag)p_99597_;
+            Tag tag = compoundtag.m_128423_(this.f_99584_);
+            if (this.f_99586_.test(tag)) {
+               Tag tag1 = p_99598_.get();
+               if (!tag1.equals(tag)) {
+                  compoundtag.m_128365_(this.f_99584_, tag1);
                   return 1;
                }
             }
@@ -319,13 +_,12 @@
          return 0;
       }
 
-      public int m_6015_(Tag p_99552_) {
-         if (p_99552_ instanceof CollectionTag) {
-            CollectionTag<?> collectiontag = (CollectionTag)p_99552_;
-            int i = collectiontag.size();
-            int j = this.f_99547_ < 0 ? i + this.f_99547_ : this.f_99547_;
-            if (0 <= j && j < i) {
-               collectiontag.remove(j);
+      public int m_6015_(Tag p_99592_) {
+         if (p_99592_ instanceof CompoundTag) {
+            CompoundTag compoundtag = (CompoundTag)p_99592_;
+            Tag tag = compoundtag.m_128423_(this.f_99584_);
+            if (this.f_99586_.test(tag)) {
+               compoundtag.m_128473_(this.f_99584_);
                return 1;
             }
          }
@@ -355,8 +_,8 @@
          MutableBoolean mutableboolean = new MutableBoolean();
          if (p_99581_ instanceof ListTag) {
             ListTag listtag = (ListTag)p_99581_;
-            listtag.stream().filter(this.f_99564_).forEach((p_99571_) -> {
-               p_99583_.add(p_99571_);
+            listtag.stream().filter(this.f_99564_).forEach((p_218060_2_) -> {
+               p_99583_.add(p_218060_2_);
                mutableboolean.setTrue();
             });
             if (mutableboolean.isFalse()) {
@@ -413,107 +_,6 @@
       }
    }
 
-   static class MatchObjectNode implements NbtPathArgument.Node {
-      private final String f_99584_;
-      private final CompoundTag f_99585_;
-      private final Predicate<Tag> f_99586_;
-
-      public MatchObjectNode(String p_99588_, CompoundTag p_99589_) {
-         this.f_99584_ = p_99588_;
-         this.f_99585_ = p_99589_;
-         this.f_99586_ = NbtPathArgument.m_99510_(p_99589_);
-      }
-
-      public void m_7273_(Tag p_99594_, List<Tag> p_99595_) {
-         if (p_99594_ instanceof CompoundTag) {
-            Tag tag = ((CompoundTag)p_99594_).m_128423_(this.f_99584_);
-            if (this.f_99586_.test(tag)) {
-               p_99595_.add(tag);
-            }
-         }
-
-      }
-
-      public void m_7876_(Tag p_99600_, Supplier<Tag> p_99601_, List<Tag> p_99602_) {
-         if (p_99600_ instanceof CompoundTag) {
-            CompoundTag compoundtag = (CompoundTag)p_99600_;
-            Tag tag = compoundtag.m_128423_(this.f_99584_);
-            if (tag == null) {
-               Tag compoundtag1 = this.f_99585_.m_6426_();
-               compoundtag.m_128365_(this.f_99584_, compoundtag1);
-               p_99602_.add(compoundtag1);
-            } else if (this.f_99586_.test(tag)) {
-               p_99602_.add(tag);
-            }
-         }
-
-      }
-
-      public Tag m_7510_() {
-         return new CompoundTag();
-      }
-
-      public int m_5571_(Tag p_99597_, Supplier<Tag> p_99598_) {
-         if (p_99597_ instanceof CompoundTag) {
-            CompoundTag compoundtag = (CompoundTag)p_99597_;
-            Tag tag = compoundtag.m_128423_(this.f_99584_);
-            if (this.f_99586_.test(tag)) {
-               Tag tag1 = p_99598_.get();
-               if (!tag1.equals(tag)) {
-                  compoundtag.m_128365_(this.f_99584_, tag1);
-                  return 1;
-               }
-            }
-         }
-
-         return 0;
-      }
-
-      public int m_6015_(Tag p_99592_) {
-         if (p_99592_ instanceof CompoundTag) {
-            CompoundTag compoundtag = (CompoundTag)p_99592_;
-            Tag tag = compoundtag.m_128423_(this.f_99584_);
-            if (this.f_99586_.test(tag)) {
-               compoundtag.m_128473_(this.f_99584_);
-               return 1;
-            }
-         }
-
-         return 0;
-      }
-   }
-
-   static class MatchRootObjectNode implements NbtPathArgument.Node {
-      private final Predicate<Tag> f_99603_;
-
-      public MatchRootObjectNode(CompoundTag p_99605_) {
-         this.f_99603_ = NbtPathArgument.m_99510_(p_99605_);
-      }
-
-      public void m_7273_(Tag p_99610_, List<Tag> p_99611_) {
-         if (p_99610_ instanceof CompoundTag && this.f_99603_.test(p_99610_)) {
-            p_99611_.add(p_99610_);
-         }
-
-      }
-
-      public void m_7876_(Tag p_99616_, Supplier<Tag> p_99617_, List<Tag> p_99618_) {
-         this.m_7273_(p_99616_, p_99618_);
-      }
-
-      public Tag m_7510_() {
-         return new CompoundTag();
-      }
-
-      public int m_5571_(Tag p_99613_, Supplier<Tag> p_99614_) {
-         return 0;
-      }
-
-      public int m_6015_(Tag p_99608_) {
-         return 0;
-      }
-   }
-
    public static class NbtPath {
       private final String f_99619_;
       private final Object2IntMap<NbtPathArgument.Node> f_99620_;
@@ -573,16 +_,16 @@
       }
 
       private static int m_99635_(List<Tag> p_99636_, Function<Tag, Integer> p_99637_) {
-         return p_99636_.stream().map(p_99637_).reduce(0, (p_99633_, p_99634_) -> {
-            return p_99633_ + p_99634_;
+         return p_99636_.stream().map(p_99637_).reduce(0, (p_218074_0_, p_218074_1_) -> {
+            return p_218074_0_ + p_218074_1_;
          });
       }
 
       public int m_99645_(Tag p_99646_, Supplier<Tag> p_99647_) throws CommandSyntaxException {
          List<Tag> list = this.m_99650_(p_99646_);
          NbtPathArgument.Node nbtpathargument$node = this.f_99621_[this.f_99621_.length - 1];
-         return m_99635_(list, (p_99631_) -> {
-            return nbtpathargument$node.m_5571_(p_99631_, p_99647_);
+         return m_99635_(list, (p_218077_2_) -> {
+            return nbtpathargument$node.m_5571_(p_218077_2_, p_99647_);
          });
       }
 
@@ -607,35 +_,66 @@
       }
    }
 
-   interface Node {
-      void m_7273_(Tag p_99666_, List<Tag> p_99667_);
-
-      void m_7876_(Tag p_99670_, Supplier<Tag> p_99671_, List<Tag> p_99672_);
-
-      Tag m_7510_();
-
-      int m_5571_(Tag p_99668_, Supplier<Tag> p_99669_);
-
-      int m_6015_(Tag p_99665_);
-
-      default List<Tag> m_99653_(List<Tag> p_99654_) {
-         return this.m_99655_(p_99654_, this::m_7273_);
-      }
-
-      default List<Tag> m_99658_(List<Tag> p_99659_, Supplier<Tag> p_99660_) {
-         return this.m_99655_(p_99659_, (p_99663_, p_99664_) -> {
-            this.m_7876_(p_99663_, p_99660_, p_99664_);
-         });
-      }
-
-      default List<Tag> m_99655_(List<Tag> p_99656_, BiConsumer<Tag, List<Tag>> p_99657_) {
-         List<Tag> list = Lists.newArrayList();
-
-         for(Tag tag : p_99656_) {
-            p_99657_.accept(tag, list);
-         }
-
-         return list;
+   static class CompoundChildNode implements NbtPathArgument.Node {
+      private final String f_99531_;
+
+      public CompoundChildNode(String p_99533_) {
+         this.f_99531_ = p_99533_;
+      }
+
+      public void m_7273_(Tag p_99538_, List<Tag> p_99539_) {
+         if (p_99538_ instanceof CompoundTag) {
+            Tag tag = ((CompoundTag)p_99538_).m_128423_(this.f_99531_);
+            if (tag != null) {
+               p_99539_.add(tag);
+            }
+         }
+
+      }
+
+      public void m_7876_(Tag p_99544_, Supplier<Tag> p_99545_, List<Tag> p_99546_) {
+         if (p_99544_ instanceof CompoundTag) {
+            CompoundTag compoundtag = (CompoundTag)p_99544_;
+            Tag tag;
+            if (compoundtag.m_128441_(this.f_99531_)) {
+               tag = compoundtag.m_128423_(this.f_99531_);
+            } else {
+               tag = p_99545_.get();
+               compoundtag.m_128365_(this.f_99531_, tag);
+            }
+
+            p_99546_.add(tag);
+         }
+
+      }
+
+      public Tag m_7510_() {
+         return new CompoundTag();
+      }
+
+      public int m_5571_(Tag p_99541_, Supplier<Tag> p_99542_) {
+         if (p_99541_ instanceof CompoundTag) {
+            CompoundTag compoundtag = (CompoundTag)p_99541_;
+            Tag tag = p_99542_.get();
+            Tag tag1 = compoundtag.m_128365_(this.f_99531_, tag);
+            if (!tag.equals(tag1)) {
+               return 1;
+            }
+         }
+
+         return 0;
+      }
+
+      public int m_6015_(Tag p_99536_) {
+         if (p_99536_ instanceof CompoundTag) {
+            CompoundTag compoundtag = (CompoundTag)p_99536_;
+            if (compoundtag.m_128441_(this.f_99531_)) {
+               compoundtag.m_128473_(this.f_99531_);
+               return 1;
+            }
+         }
+
+         return 0;
       }
    }
 }
