--- a/net/minecraft/commands/arguments/ObjectiveArgument.java
+++ b/net/minecraft/commands/arguments/ObjectiveArgument.java
@@ -18,14 +_,14 @@
 
 public class ObjectiveArgument implements ArgumentType<String> {
    private static final Collection<String> f_101952_ = Arrays.asList("foo", "*", "012");
-   private static final DynamicCommandExceptionType f_101953_ = new DynamicCommandExceptionType((p_101971_) -> {
-      return new TranslatableComponent("arguments.objective.notFound", p_101971_);
-   });
-   private static final DynamicCommandExceptionType f_101954_ = new DynamicCommandExceptionType((p_101969_) -> {
-      return new TranslatableComponent("arguments.objective.readonly", p_101969_);
-   });
-   public static final DynamicCommandExceptionType f_101951_ = new DynamicCommandExceptionType((p_101964_) -> {
-      return new TranslatableComponent("commands.scoreboard.objectives.add.longName", p_101964_);
+   private static final DynamicCommandExceptionType f_101953_ = new DynamicCommandExceptionType((p_208671_0_) -> {
+      return new TranslatableComponent("arguments.objective.notFound", p_208671_0_);
+   });
+   private static final DynamicCommandExceptionType f_101954_ = new DynamicCommandExceptionType((p_208669_0_) -> {
+      return new TranslatableComponent("arguments.objective.readonly", p_208669_0_);
+   });
+   public static final DynamicCommandExceptionType f_101951_ = new DynamicCommandExceptionType((p_208670_0_) -> {
+      return new TranslatableComponent("commands.scoreboard.objectives.add.longName", p_208670_0_);
    });
 
    public static ObjectiveArgument m_101957_() {
