--- a/net/minecraft/commands/arguments/ScoreboardSlotArgument.java
+++ b/net/minecraft/commands/arguments/ScoreboardSlotArgument.java
@@ -17,8 +_,8 @@
 
 public class ScoreboardSlotArgument implements ArgumentType<Integer> {
    private static final Collection<String> f_109193_ = Arrays.asList("sidebar", "foo.bar");
-   public static final DynamicCommandExceptionType f_109192_ = new DynamicCommandExceptionType((p_109203_) -> {
-      return new TranslatableComponent("argument.scoreboardDisplaySlot.invalid", p_109203_);
+   public static final DynamicCommandExceptionType f_109192_ = new DynamicCommandExceptionType((p_208678_0_) -> {
+      return new TranslatableComponent("argument.scoreboardDisplaySlot.invalid", p_208678_0_);
    });
 
    private ScoreboardSlotArgument() {
