--- a/net/minecraft/commands/arguments/TimeArgument.java
+++ b/net/minecraft/commands/arguments/TimeArgument.java
@@ -19,8 +_,8 @@
 public class TimeArgument implements ArgumentType<Integer> {
    private static final Collection<String> f_113031_ = Arrays.asList("0d", "0s", "0t", "0");
    private static final SimpleCommandExceptionType f_113032_ = new SimpleCommandExceptionType(new TranslatableComponent("argument.time.invalid_unit"));
-   private static final DynamicCommandExceptionType f_113033_ = new DynamicCommandExceptionType((p_113041_) -> {
-      return new TranslatableComponent("argument.time.invalid_tick_count", p_113041_);
+   private static final DynamicCommandExceptionType f_113033_ = new DynamicCommandExceptionType((p_218092_0_) -> {
+      return new TranslatableComponent("argument.time.invalid_tick_count", p_218092_0_);
    });
    private static final Object2IntMap<String> f_113034_ = new Object2IntOpenHashMap<>();
 
