--- a/net/minecraft/world/level/block/state/pattern/BlockPattern.java
+++ b/net/minecraft/world/level/block/state/pattern/BlockPattern.java
@@ -2,7 +_,6 @@
 
 import com.google.common.base.MoreObjects;
 import com.google.common.cache.CacheBuilder;
-import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
 import java.util.function.Predicate;
 import javax.annotation.Nullable;
@@ -97,7 +_,7 @@
       }
    }
 
-   static class BlockCacheLoader extends CacheLoader<BlockPos, BlockInWorld> {
+   static class BlockCacheLoader extends com.google.common.cache.CacheLoader<BlockPos, BlockInWorld> {
       private final LevelReader f_61204_;
       private final boolean f_61205_;
 
