--- a/net/minecraft/client/resources/model/BlockModelRotation.java
+++ b/net/minecraft/client/resources/model/BlockModelRotation.java
@@ -30,10 +_,10 @@
    X270_Y180(270, 180),
    X270_Y270(270, 270);
 
-   private static final Map<Integer, BlockModelRotation> f_119142_ = Arrays.stream(values()).collect(Collectors.toMap((p_119163_) -> {
-      return p_119163_.f_119145_;
-   }, (p_119157_) -> {
-      return p_119157_;
+   private static final Map<Integer, BlockModelRotation> f_119142_ = Arrays.stream(values()).collect(Collectors.toMap((p_229306_0_) -> {
+      return p_229306_0_.f_119145_;
+   }, (p_229305_0_) -> {
+      return p_229305_0_;
    }));
    private final Transformation f_119143_;
    private final OctahedralGroup f_119144_;
