--- a/net/minecraft/world/level/levelgen/feature/configurations/DeltaFeatureConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/DeltaFeatureConfiguration.java
@@ -6,16 +_,16 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class DeltaFeatureConfiguration implements FeatureConfiguration {
-   public static final Codec<DeltaFeatureConfiguration> f_67593_ = RecordCodecBuilder.create((p_67607_) -> {
-      return p_67607_.group(BlockState.f_61039_.fieldOf("contents").forGetter((p_67616_) -> {
-         return p_67616_.f_67594_;
-      }), BlockState.f_61039_.fieldOf("rim").forGetter((p_67613_) -> {
-         return p_67613_.f_67595_;
-      }), UniformInt.m_14432_(0, 8, 8).fieldOf("size").forGetter((p_67610_) -> {
-         return p_67610_.f_67596_;
-      }), UniformInt.m_14432_(0, 8, 8).fieldOf("rim_size").forGetter((p_67605_) -> {
-         return p_67605_.f_67597_;
-      })).apply(p_67607_, DeltaFeatureConfiguration::new);
+   public static final Codec<DeltaFeatureConfiguration> f_67593_ = RecordCodecBuilder.create((p_242803_0_) -> {
+      return p_242803_0_.group(BlockState.f_61039_.fieldOf("contents").forGetter((p_236506_0_) -> {
+         return p_236506_0_.f_67594_;
+      }), BlockState.f_61039_.fieldOf("rim").forGetter((p_236505_0_) -> {
+         return p_236505_0_.f_67595_;
+      }), UniformInt.m_14432_(0, 8, 8).fieldOf("size").forGetter((p_242805_0_) -> {
+         return p_242805_0_.f_67596_;
+      }), UniformInt.m_14432_(0, 8, 8).fieldOf("rim_size").forGetter((p_242802_0_) -> {
+         return p_242802_0_.f_67597_;
+      })).apply(p_242803_0_, DeltaFeatureConfiguration::new);
    });
    private final BlockState f_67594_;
    private final BlockState f_67595_;
