--- a/net/minecraft/world/level/levelgen/feature/configurations/EndGatewayConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/EndGatewayConfiguration.java
@@ -6,12 +_,12 @@
 import net.minecraft.core.BlockPos;
 
 public class EndGatewayConfiguration implements FeatureConfiguration {
-   public static final Codec<EndGatewayConfiguration> f_67639_ = RecordCodecBuilder.create((p_67649_) -> {
-      return p_67649_.group(BlockPos.f_121852_.optionalFieldOf("exit").forGetter((p_67655_) -> {
-         return p_67655_.f_67640_;
-      }), Codec.BOOL.fieldOf("exact").forGetter((p_67647_) -> {
-         return p_67647_.f_67641_;
-      })).apply(p_67649_, EndGatewayConfiguration::new);
+   public static final Codec<EndGatewayConfiguration> f_67639_ = RecordCodecBuilder.create((p_236524_0_) -> {
+      return p_236524_0_.group(BlockPos.f_121852_.optionalFieldOf("exit").forGetter((p_236525_0_) -> {
+         return p_236525_0_.f_67640_;
+      }), Codec.BOOL.fieldOf("exact").forGetter((p_236523_0_) -> {
+         return p_236523_0_.f_67641_;
+      })).apply(p_236524_0_, EndGatewayConfiguration::new);
    });
    private final Optional<BlockPos> f_67640_;
    private final boolean f_67641_;
