--- a/net/minecraft/world/level/levelgen/feature/configurations/HugeMushroomFeatureConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/HugeMushroomFeatureConfiguration.java
@@ -5,14 +_,14 @@
 import net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider;
 
 public class HugeMushroomFeatureConfiguration implements FeatureConfiguration {
-   public static final Codec<HugeMushroomFeatureConfiguration> f_67739_ = RecordCodecBuilder.create((p_67751_) -> {
-      return p_67751_.group(BlockStateProvider.f_68747_.fieldOf("cap_provider").forGetter((p_67755_) -> {
-         return p_67755_.f_67740_;
-      }), BlockStateProvider.f_68747_.fieldOf("stem_provider").forGetter((p_67753_) -> {
-         return p_67753_.f_67741_;
-      }), Codec.INT.fieldOf("foliage_radius").orElse(2).forGetter((p_67749_) -> {
-         return p_67749_.f_67742_;
-      })).apply(p_67751_, HugeMushroomFeatureConfiguration::new);
+   public static final Codec<HugeMushroomFeatureConfiguration> f_67739_ = RecordCodecBuilder.create((p_236530_0_) -> {
+      return p_236530_0_.group(BlockStateProvider.f_68747_.fieldOf("cap_provider").forGetter((p_236532_0_) -> {
+         return p_236532_0_.f_67740_;
+      }), BlockStateProvider.f_68747_.fieldOf("stem_provider").forGetter((p_236531_0_) -> {
+         return p_236531_0_.f_67741_;
+      }), Codec.INT.fieldOf("foliage_radius").orElse(2).forGetter((p_236529_0_) -> {
+         return p_236529_0_.f_67742_;
+      })).apply(p_236530_0_, HugeMushroomFeatureConfiguration::new);
    });
    public final BlockStateProvider f_67740_;
    public final BlockStateProvider f_67741_;
