--- a/net/minecraft/world/level/levelgen/feature/configurations/OceanRuinConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/OceanRuinConfiguration.java
@@ -5,14 +_,14 @@
 import net.minecraft.world.level.levelgen.structure.OceanRuinFeature;
 
 public class OceanRuinConfiguration implements FeatureConfiguration {
-   public static final Codec<OceanRuinConfiguration> f_67820_ = RecordCodecBuilder.create((p_67832_) -> {
-      return p_67832_.group(OceanRuinFeature.Type.f_72501_.fieldOf("biome_temp").forGetter((p_67836_) -> {
-         return p_67836_.f_67821_;
-      }), Codec.floatRange(0.0F, 1.0F).fieldOf("large_probability").forGetter((p_67834_) -> {
-         return p_67834_.f_67822_;
-      }), Codec.floatRange(0.0F, 1.0F).fieldOf("cluster_probability").forGetter((p_67830_) -> {
-         return p_67830_.f_67823_;
-      })).apply(p_67832_, OceanRuinConfiguration::new);
+   public static final Codec<OceanRuinConfiguration> f_67820_ = RecordCodecBuilder.create((p_236563_0_) -> {
+      return p_236563_0_.group(OceanRuinFeature.Type.f_72501_.fieldOf("biome_temp").forGetter((p_236565_0_) -> {
+         return p_236565_0_.f_67821_;
+      }), Codec.floatRange(0.0F, 1.0F).fieldOf("large_probability").forGetter((p_236564_0_) -> {
+         return p_236564_0_.f_67822_;
+      }), Codec.floatRange(0.0F, 1.0F).fieldOf("cluster_probability").forGetter((p_236562_0_) -> {
+         return p_236562_0_.f_67823_;
+      })).apply(p_236563_0_, OceanRuinConfiguration::new);
    });
    public final OceanRuinFeature.Type f_67821_;
    public final float f_67822_;
