--- a/net/minecraft/world/level/levelgen/feature/configurations/ProbabilityFeatureConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/ProbabilityFeatureConfiguration.java
@@ -5,10 +_,10 @@
 import net.minecraft.world.level.levelgen.carver.CarverConfiguration;
 
 public class ProbabilityFeatureConfiguration implements CarverConfiguration, FeatureConfiguration {
-   public static final Codec<ProbabilityFeatureConfiguration> f_67858_ = RecordCodecBuilder.create((p_67866_) -> {
-      return p_67866_.group(Codec.floatRange(0.0F, 1.0F).fieldOf("probability").forGetter((p_67864_) -> {
-         return p_67864_.f_67859_;
-      })).apply(p_67866_, ProbabilityFeatureConfiguration::new);
+   public static final Codec<ProbabilityFeatureConfiguration> f_67858_ = RecordCodecBuilder.create((p_236578_0_) -> {
+      return p_236578_0_.group(Codec.floatRange(0.0F, 1.0F).fieldOf("probability").forGetter((p_236577_0_) -> {
+         return p_236577_0_.f_67859_;
+      })).apply(p_236578_0_, ProbabilityFeatureConfiguration::new);
    });
    public final float f_67859_;
 
