--- a/net/minecraft/world/level/levelgen/feature/configurations/RandomBooleanFeatureConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/RandomBooleanFeatureConfiguration.java
@@ -7,12 +_,12 @@
 import net.minecraft.world.level.levelgen.feature.ConfiguredFeature;
 
 public class RandomBooleanFeatureConfiguration implements FeatureConfiguration {
-   public static final Codec<RandomBooleanFeatureConfiguration> f_67867_ = RecordCodecBuilder.create((p_67877_) -> {
-      return p_67877_.group(ConfiguredFeature.f_65374_.fieldOf("feature_true").forGetter((p_67880_) -> {
-         return p_67880_.f_67868_;
-      }), ConfiguredFeature.f_65374_.fieldOf("feature_false").forGetter((p_67875_) -> {
-         return p_67875_.f_67869_;
-      })).apply(p_67877_, RandomBooleanFeatureConfiguration::new);
+   public static final Codec<RandomBooleanFeatureConfiguration> f_67867_ = RecordCodecBuilder.create((p_236581_0_) -> {
+      return p_236581_0_.group(ConfiguredFeature.f_65374_.fieldOf("feature_true").forGetter((p_236582_0_) -> {
+         return p_236582_0_.f_67868_;
+      }), ConfiguredFeature.f_65374_.fieldOf("feature_false").forGetter((p_236580_0_) -> {
+         return p_236580_0_.f_67869_;
+      })).apply(p_236581_0_, RandomBooleanFeatureConfiguration::new);
    });
    public final Supplier<ConfiguredFeature<?, ?>> f_67868_;
    public final Supplier<ConfiguredFeature<?, ?>> f_67869_;
