--- a/net/minecraft/world/level/levelgen/feature/configurations/RangeDecoratorConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/RangeDecoratorConfiguration.java
@@ -4,14 +_,14 @@
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 
 public class RangeDecoratorConfiguration implements DecoratorConfiguration {
-   public static final Codec<RangeDecoratorConfiguration> f_68006_ = RecordCodecBuilder.create((p_68018_) -> {
-      return p_68018_.group(Codec.INT.fieldOf("bottom_offset").orElse(0).forGetter((p_68022_) -> {
-         return p_68022_.f_68007_;
-      }), Codec.INT.fieldOf("top_offset").orElse(0).forGetter((p_68020_) -> {
-         return p_68020_.f_68008_;
-      }), Codec.INT.fieldOf("maximum").orElse(0).forGetter((p_68016_) -> {
-         return p_68016_.f_68009_;
-      })).apply(p_68018_, RangeDecoratorConfiguration::new);
+   public static final Codec<RangeDecoratorConfiguration> f_68006_ = RecordCodecBuilder.create((p_236986_0_) -> {
+      return p_236986_0_.group(Codec.INT.fieldOf("bottom_offset").orElse(0).forGetter((p_236988_0_) -> {
+         return p_236988_0_.f_68007_;
+      }), Codec.INT.fieldOf("top_offset").orElse(0).forGetter((p_236987_0_) -> {
+         return p_236987_0_.f_68008_;
+      }), Codec.INT.fieldOf("maximum").orElse(0).forGetter((p_242816_0_) -> {
+         return p_242816_0_.f_68009_;
+      })).apply(p_236986_0_, RangeDecoratorConfiguration::new);
    });
    public final int f_68007_;
    public final int f_68008_;
