--- a/net/minecraft/world/level/levelgen/feature/configurations/ReplaceBlockConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/ReplaceBlockConfiguration.java
@@ -5,12 +_,12 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class ReplaceBlockConfiguration implements FeatureConfiguration {
-   public static final Codec<ReplaceBlockConfiguration> f_68023_ = RecordCodecBuilder.create((p_68033_) -> {
-      return p_68033_.group(BlockState.f_61039_.fieldOf("target").forGetter((p_68035_) -> {
-         return p_68035_.f_68024_;
-      }), BlockState.f_61039_.fieldOf("state").forGetter((p_68031_) -> {
-         return p_68031_.f_68025_;
-      })).apply(p_68033_, ReplaceBlockConfiguration::new);
+   public static final Codec<ReplaceBlockConfiguration> f_68023_ = RecordCodecBuilder.create((p_236606_0_) -> {
+      return p_236606_0_.group(BlockState.f_61039_.fieldOf("target").forGetter((p_236607_0_) -> {
+         return p_236607_0_.f_68024_;
+      }), BlockState.f_61039_.fieldOf("state").forGetter((p_236605_0_) -> {
+         return p_236605_0_.f_68025_;
+      })).apply(p_236606_0_, ReplaceBlockConfiguration::new);
    });
    public final BlockState f_68024_;
    public final BlockState f_68025_;
