--- a/net/minecraft/world/level/biome/BiomeSource.java
+++ b/net/minecraft/world/level/biome/BiomeSource.java
@@ -116,9 +_,9 @@
    }
 
    public boolean m_47917_(StructureFeature<?> p_47918_) {
-      return this.f_47889_.computeIfAbsent(p_47918_, (p_47924_) -> {
-         return this.f_47891_.stream().anyMatch((p_47921_) -> {
-            return p_47921_.m_47536_().m_47808_(p_47924_);
+      return this.f_47889_.computeIfAbsent(p_47918_, (p_226839_1_) -> {
+         return this.f_47891_.stream().anyMatch((p_226838_1_) -> {
+            return p_226838_1_.m_47536_().m_47808_(p_226839_1_);
          });
       });
    }
