--- a/net/minecraft/world/level/biome/TheEndBiomeSource.java
+++ b/net/minecraft/world/level/biome/TheEndBiomeSource.java
@@ -12,12 +_,12 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class TheEndBiomeSource extends BiomeSource {
-   public static final Codec<TheEndBiomeSource> f_48617_ = RecordCodecBuilder.create((p_48644_) -> {
-      return p_48644_.group(RegistryLookupCodec.m_135622_(Registry.f_122885_).forGetter((p_48656_) -> {
-         return p_48656_.f_48619_;
-      }), Codec.LONG.fieldOf("seed").stable().forGetter((p_48642_) -> {
-         return p_48642_.f_48620_;
-      })).apply(p_48644_, p_48644_.stable(TheEndBiomeSource::new));
+   public static final Codec<TheEndBiomeSource> f_48617_ = RecordCodecBuilder.create((p_242647_0_) -> {
+      return p_242647_0_.group(RegistryLookupCodec.m_135622_(Registry.f_122885_).forGetter((p_242648_0_) -> {
+         return p_242648_0_.f_48619_;
+      }), Codec.LONG.fieldOf("seed").stable().forGetter((p_242646_0_) -> {
+         return p_242646_0_.f_48620_;
+      })).apply(p_242647_0_, p_242647_0_.stable(TheEndBiomeSource::new));
    });
    private final SimplexNoise f_48618_;
    private final Registry<Biome> f_48619_;
