--- a/net/minecraft/world/entity/monster/AbstractSkeleton.java
+++ b/net/minecraft/world/entity/monster/AbstractSkeleton.java
@@ -85,7 +_,7 @@
       this.m_5496_(this.m_7878_(), 0.15F, 1.0F);
    }
 
-   abstract SoundEvent m_7878_();
+   protected abstract SoundEvent m_7878_();
 
    public MobType m_6336_() {
       return MobType.f_21641_;
@@ -172,6 +_,8 @@
    public void m_6504_(LivingEntity p_32141_, float p_32142_) {
       ItemStack itemstack = this.m_6298_(this.m_21120_(ProjectileUtil.m_37297_(this, Items.f_42411_)));
       AbstractArrow abstractarrow = this.m_7932_(itemstack, p_32142_);
+      if (this.m_21205_().m_41720_() instanceof net.minecraft.world.item.BowItem)
+         abstractarrow = ((net.minecraft.world.item.BowItem)this.m_21205_().m_41720_()).customArrow(abstractarrow);
       double d0 = p_32141_.m_20185_() - this.m_20185_();
       double d1 = p_32141_.m_20227_(0.3333333333333333D) - abstractarrow.m_20186_();
       double d2 = p_32141_.m_20189_() - this.m_20189_();
