--- a/net/minecraft/world/entity/monster/Ghast.java
+++ b/net/minecraft/world/entity/monster/Ghast.java
@@ -48,8 +_,8 @@
       this.f_21345_.m_25352_(5, new Ghast.RandomFloatAroundGoal(this));
       this.f_21345_.m_25352_(7, new Ghast.GhastLookGoal(this));
       this.f_21345_.m_25352_(7, new Ghast.GhastShootFireballGoal(this));
-      this.f_21346_.m_25352_(1, new NearestAttackableTargetGoal<>(this, Player.class, 10, true, false, (p_32755_) -> {
-         return Math.abs(p_32755_.m_20186_() - this.m_20186_()) <= 4.0D;
+      this.f_21346_.m_25352_(1, new NearestAttackableTargetGoal<>(this, Player.class, 10, true, false, (p_213812_1_) -> {
+         return Math.abs(p_213812_1_.m_20186_() - this.m_20186_()) <= 4.0D;
       }));
    }
 
@@ -135,6 +_,60 @@
       return 2.6F;
    }
 
+   static class GhastShootFireballGoal extends Goal {
+      private final Ghast f_32774_;
+      public int f_32773_;
+
+      public GhastShootFireballGoal(Ghast p_32776_) {
+         this.f_32774_ = p_32776_;
+      }
+
+      public boolean m_8036_() {
+         return this.f_32774_.m_5448_() != null;
+      }
+
+      public void m_8056_() {
+         this.f_32773_ = 0;
+      }
+
+      public void m_8041_() {
+         this.f_32774_.m_32758_(false);
+      }
+
+      public void m_8037_() {
+         LivingEntity livingentity = this.f_32774_.m_5448_();
+         double d0 = 64.0D;
+         if (livingentity.m_20280_(this.f_32774_) < 4096.0D && this.f_32774_.m_7331_(livingentity)) {
+            Level level = this.f_32774_.f_19853_;
+            ++this.f_32773_;
+            if (this.f_32773_ == 10 && !this.f_32774_.m_20067_()) {
+               level.m_5898_((Player)null, 1015, this.f_32774_.m_20183_(), 0);
+            }
+
+            if (this.f_32773_ == 20) {
+               double d1 = 4.0D;
+               Vec3 vec3 = this.f_32774_.m_20252_(1.0F);
+               double d2 = livingentity.m_20185_() - (this.f_32774_.m_20185_() + vec3.f_82479_ * 4.0D);
+               double d3 = livingentity.m_20227_(0.5D) - (0.5D + this.f_32774_.m_20227_(0.5D));
+               double d4 = livingentity.m_20189_() - (this.f_32774_.m_20189_() + vec3.f_82481_ * 4.0D);
+               if (!this.f_32774_.m_20067_()) {
+                  level.m_5898_((Player)null, 1016, this.f_32774_.m_20183_(), 0);
+               }
+
+               LargeFireball largefireball = new LargeFireball(level, this.f_32774_, d2, d3, d4);
+               largefireball.f_37197_ = this.f_32774_.m_32751_();
+               largefireball.m_6034_(this.f_32774_.m_20185_() + vec3.f_82479_ * 4.0D, this.f_32774_.m_20227_(0.5D) + 0.5D, largefireball.m_20189_() + vec3.f_82481_ * 4.0D);
+               level.m_7967_(largefireball);
+               this.f_32773_ = -40;
+            }
+         } else if (this.f_32773_ > 0) {
+            --this.f_32773_;
+         }
+
+         this.f_32774_.m_32758_(this.f_32773_ > 10);
+      }
+   }
+
    static class GhastLookGoal extends Goal {
       private final Ghast f_32760_;
 
@@ -203,60 +_,6 @@
          }
 
          return true;
-      }
-   }
-
-   static class GhastShootFireballGoal extends Goal {
-      private final Ghast f_32774_;
-      public int f_32773_;
-
-      public GhastShootFireballGoal(Ghast p_32776_) {
-         this.f_32774_ = p_32776_;
-      }
-
-      public boolean m_8036_() {
-         return this.f_32774_.m_5448_() != null;
-      }
-
-      public void m_8056_() {
-         this.f_32773_ = 0;
-      }
-
-      public void m_8041_() {
-         this.f_32774_.m_32758_(false);
-      }
-
-      public void m_8037_() {
-         LivingEntity livingentity = this.f_32774_.m_5448_();
-         double d0 = 64.0D;
-         if (livingentity.m_20280_(this.f_32774_) < 4096.0D && this.f_32774_.m_7331_(livingentity)) {
-            Level level = this.f_32774_.f_19853_;
-            ++this.f_32773_;
-            if (this.f_32773_ == 10 && !this.f_32774_.m_20067_()) {
-               level.m_5898_((Player)null, 1015, this.f_32774_.m_20183_(), 0);
-            }
-
-            if (this.f_32773_ == 20) {
-               double d1 = 4.0D;
-               Vec3 vec3 = this.f_32774_.m_20252_(1.0F);
-               double d2 = livingentity.m_20185_() - (this.f_32774_.m_20185_() + vec3.f_82479_ * 4.0D);
-               double d3 = livingentity.m_20227_(0.5D) - (0.5D + this.f_32774_.m_20227_(0.5D));
-               double d4 = livingentity.m_20189_() - (this.f_32774_.m_20189_() + vec3.f_82481_ * 4.0D);
-               if (!this.f_32774_.m_20067_()) {
-                  level.m_5898_((Player)null, 1016, this.f_32774_.m_20183_(), 0);
-               }
-
-               LargeFireball largefireball = new LargeFireball(level, this.f_32774_, d2, d3, d4);
-               largefireball.f_37197_ = this.f_32774_.m_32751_();
-               largefireball.m_6034_(this.f_32774_.m_20185_() + vec3.f_82479_ * 4.0D, this.f_32774_.m_20227_(0.5D) + 0.5D, largefireball.m_20189_() + vec3.f_82481_ * 4.0D);
-               level.m_7967_(largefireball);
-               this.f_32773_ = -40;
-            }
-         } else if (this.f_32773_ > 0) {
-            --this.f_32773_;
-         }
-
-         this.f_32774_.m_32758_(this.f_32773_ > 10);
       }
    }
 
