--- a/net/minecraft/world/entity/monster/Guardian.java
+++ b/net/minecraft/world/entity/monster/Guardian.java
@@ -390,18 +_,6 @@
       }
    }
 
-   static class GuardianAttackSelector implements Predicate<LivingEntity> {
-      private final Guardian f_32877_;
-
-      public GuardianAttackSelector(Guardian p_32879_) {
-         this.f_32877_ = p_32879_;
-      }
-
-      public boolean test(@Nullable LivingEntity p_32881_) {
-         return (p_32881_ instanceof Player || p_32881_ instanceof Squid) && p_32881_.m_20280_(this.f_32877_) > 9.0D;
-      }
-   }
-
    static class GuardianMoveControl extends MoveControl {
       private final Guardian f_32884_;
 
@@ -447,6 +_,18 @@
             this.f_32884_.m_7910_(0.0F);
             this.f_32884_.m_32861_(false);
          }
+      }
+   }
+
+   static class GuardianAttackSelector implements Predicate<LivingEntity> {
+      private final Guardian f_32877_;
+
+      public GuardianAttackSelector(Guardian p_32879_) {
+         this.f_32877_ = p_32879_;
+      }
+
+      public boolean test(@Nullable LivingEntity p_32881_) {
+         return (p_32881_ instanceof Player || p_32881_ instanceof Squid) && p_32881_.m_20280_(this.f_32877_) > 9.0D;
       }
    }
 }
