--- a/net/minecraft/world/entity/monster/PatrollingMonster.java
+++ b/net/minecraft/world/entity/monster/PatrollingMonster.java
@@ -185,8 +_,8 @@
       }
 
       private List<PatrollingMonster> m_33093_() {
-         return this.f_33079_.f_19853_.m_6443_(PatrollingMonster.class, this.f_33079_.m_20191_().m_82400_(16.0D), (p_33089_) -> {
-            return p_33089_.m_7492_() && !p_33089_.m_7306_(this.f_33079_);
+         return this.f_33079_.f_19853_.m_6443_(PatrollingMonster.class, this.f_33079_.m_20191_().m_82400_(16.0D), (p_226543_1_) -> {
+            return p_226543_1_.m_7492_() && !p_226543_1_.m_7306_(this.f_33079_);
          });
       }
 
