--- a/net/minecraft/world/entity/monster/Phantom.java
+++ b/net/minecraft/world/entity/monster/Phantom.java
@@ -234,49 +_,6 @@
       }
    }
 
-   class PhantomAttackStrategyGoal extends Goal {
-      private int f_33202_;
-
-      private PhantomAttackStrategyGoal() {
-      }
-
-      public boolean m_8036_() {
-         LivingEntity livingentity = Phantom.this.m_5448_();
-         return livingentity != null ? Phantom.this.m_21040_(Phantom.this.m_5448_(), TargetingConditions.f_26872_) : false;
-      }
-
-      public void m_8056_() {
-         this.f_33202_ = 10;
-         Phantom.this.f_33096_ = Phantom.AttackPhase.CIRCLE;
-         this.m_33212_();
-      }
-
-      public void m_8041_() {
-         Phantom.this.f_33098_ = Phantom.this.f_19853_.m_5452_(Heightmap.Types.MOTION_BLOCKING, Phantom.this.f_33098_).m_6630_(10 + Phantom.this.f_19796_.nextInt(20));
-      }
-
-      public void m_8037_() {
-         if (Phantom.this.f_33096_ == Phantom.AttackPhase.CIRCLE) {
-            --this.f_33202_;
-            if (this.f_33202_ <= 0) {
-               Phantom.this.f_33096_ = Phantom.AttackPhase.SWOOP;
-               this.m_33212_();
-               this.f_33202_ = (8 + Phantom.this.f_19796_.nextInt(4)) * 20;
-               Phantom.this.m_5496_(SoundEvents.f_12232_, 10.0F, 0.95F + Phantom.this.f_19796_.nextFloat() * 0.1F);
-            }
-         }
-
-      }
-
-      private void m_33212_() {
-         Phantom.this.f_33098_ = Phantom.this.m_5448_().m_20183_().m_6630_(20 + Phantom.this.f_19796_.nextInt(20));
-         if (Phantom.this.f_33098_.m_123342_() < Phantom.this.f_19853_.m_5736_()) {
-            Phantom.this.f_33098_ = new BlockPos(Phantom.this.f_33098_.m_123341_(), Phantom.this.f_19853_.m_5736_() + 1, Phantom.this.f_33098_.m_123343_());
-         }
-
-      }
-   }
-
    class PhantomBodyRotationControl extends BodyRotationControl {
       public PhantomBodyRotationControl(Mob p_33216_) {
          super(p_33216_);
@@ -288,70 +_,6 @@
       }
    }
 
-   class PhantomCircleAroundAnchorGoal extends Phantom.PhantomMoveTargetGoal {
-      private float f_33219_;
-      private float f_33220_;
-      private float f_33221_;
-      private float f_33222_;
-
-      private PhantomCircleAroundAnchorGoal() {
-      }
-
-      public boolean m_8036_() {
-         return Phantom.this.m_5448_() == null || Phantom.this.f_33096_ == Phantom.AttackPhase.CIRCLE;
-      }
-
-      public void m_8056_() {
-         this.f_33220_ = 5.0F + Phantom.this.f_19796_.nextFloat() * 10.0F;
-         this.f_33221_ = -4.0F + Phantom.this.f_19796_.nextFloat() * 9.0F;
-         this.f_33222_ = Phantom.this.f_19796_.nextBoolean() ? 1.0F : -1.0F;
-         this.m_33231_();
-      }
-
-      public void m_8037_() {
-         if (Phantom.this.f_19796_.nextInt(350) == 0) {
-            this.f_33221_ = -4.0F + Phantom.this.f_19796_.nextFloat() * 9.0F;
-         }
-
-         if (Phantom.this.f_19796_.nextInt(250) == 0) {
-            ++this.f_33220_;
-            if (this.f_33220_ > 15.0F) {
-               this.f_33220_ = 5.0F;
-               this.f_33222_ = -this.f_33222_;
-            }
-         }
-
-         if (Phantom.this.f_19796_.nextInt(450) == 0) {
-            this.f_33219_ = Phantom.this.f_19796_.nextFloat() * 2.0F * (float)Math.PI;
-            this.m_33231_();
-         }
-
-         if (this.m_33246_()) {
-            this.m_33231_();
-         }
-
-         if (Phantom.this.f_33097_.f_82480_ < Phantom.this.m_20186_() && !Phantom.this.f_19853_.m_46859_(Phantom.this.m_20183_().m_6625_(1))) {
-            this.f_33221_ = Math.max(1.0F, this.f_33221_);
-            this.m_33231_();
-         }
-
-         if (Phantom.this.f_33097_.f_82480_ > Phantom.this.m_20186_() && !Phantom.this.f_19853_.m_46859_(Phantom.this.m_20183_().m_6630_(1))) {
-            this.f_33221_ = Math.min(-1.0F, this.f_33221_);
-            this.m_33231_();
-         }
-
-      }
-
-      private void m_33231_() {
-         if (BlockPos.f_121853_.equals(Phantom.this.f_33098_)) {
-            Phantom.this.f_33098_ = Phantom.this.m_20183_();
-         }
-
-         this.f_33219_ += this.f_33222_ * 15.0F * ((float)Math.PI / 180F);
-         Phantom.this.f_33097_ = Vec3.m_82528_(Phantom.this.f_33098_).m_82520_((double)(this.f_33220_ * Mth.m_14089_(this.f_33219_)), (double)(-4.0F + this.f_33221_), (double)(this.f_33220_ * Mth.m_14031_(this.f_33219_)));
-      }
-   }
-
    class PhantomLookControl extends LookControl {
       public PhantomLookControl(Mob p_33235_) {
          super(p_33235_);
@@ -361,6 +_,16 @@
       }
    }
 
+   abstract class PhantomMoveTargetGoal extends Goal {
+      public PhantomMoveTargetGoal() {
+         this.m_7021_(EnumSet.of(Goal.Flag.MOVE));
+      }
+
+      protected boolean m_33246_() {
+         return Phantom.this.f_33097_.m_82531_(Phantom.this.m_20185_(), Phantom.this.m_20186_(), Phantom.this.m_20189_()) < 4.0D;
+      }
+   }
+
    class PhantomMoveControl extends MoveControl {
       private float f_33238_ = 0.1F;
 
@@ -406,13 +_,110 @@
       }
    }
 
-   abstract class PhantomMoveTargetGoal extends Goal {
-      public PhantomMoveTargetGoal() {
-         this.m_7021_(EnumSet.of(Goal.Flag.MOVE));
-      }
-
-      protected boolean m_33246_() {
-         return Phantom.this.f_33097_.m_82531_(Phantom.this.m_20185_(), Phantom.this.m_20186_(), Phantom.this.m_20189_()) < 4.0D;
+   class PhantomCircleAroundAnchorGoal extends Phantom.PhantomMoveTargetGoal {
+      private float f_33219_;
+      private float f_33220_;
+      private float f_33221_;
+      private float f_33222_;
+
+      private PhantomCircleAroundAnchorGoal() {
+      }
+
+      public boolean m_8036_() {
+         return Phantom.this.m_5448_() == null || Phantom.this.f_33096_ == Phantom.AttackPhase.CIRCLE;
+      }
+
+      public void m_8056_() {
+         this.f_33220_ = 5.0F + Phantom.this.f_19796_.nextFloat() * 10.0F;
+         this.f_33221_ = -4.0F + Phantom.this.f_19796_.nextFloat() * 9.0F;
+         this.f_33222_ = Phantom.this.f_19796_.nextBoolean() ? 1.0F : -1.0F;
+         this.m_33231_();
+      }
+
+      public void m_8037_() {
+         if (Phantom.this.f_19796_.nextInt(350) == 0) {
+            this.f_33221_ = -4.0F + Phantom.this.f_19796_.nextFloat() * 9.0F;
+         }
+
+         if (Phantom.this.f_19796_.nextInt(250) == 0) {
+            ++this.f_33220_;
+            if (this.f_33220_ > 15.0F) {
+               this.f_33220_ = 5.0F;
+               this.f_33222_ = -this.f_33222_;
+            }
+         }
+
+         if (Phantom.this.f_19796_.nextInt(450) == 0) {
+            this.f_33219_ = Phantom.this.f_19796_.nextFloat() * 2.0F * (float)Math.PI;
+            this.m_33231_();
+         }
+
+         if (this.m_33246_()) {
+            this.m_33231_();
+         }
+
+         if (Phantom.this.f_33097_.f_82480_ < Phantom.this.m_20186_() && !Phantom.this.f_19853_.m_46859_(Phantom.this.m_20183_().m_6625_(1))) {
+            this.f_33221_ = Math.max(1.0F, this.f_33221_);
+            this.m_33231_();
+         }
+
+         if (Phantom.this.f_33097_.f_82480_ > Phantom.this.m_20186_() && !Phantom.this.f_19853_.m_46859_(Phantom.this.m_20183_().m_6630_(1))) {
+            this.f_33221_ = Math.min(-1.0F, this.f_33221_);
+            this.m_33231_();
+         }
+
+      }
+
+      private void m_33231_() {
+         if (BlockPos.f_121853_.equals(Phantom.this.f_33098_)) {
+            Phantom.this.f_33098_ = Phantom.this.m_20183_();
+         }
+
+         this.f_33219_ += this.f_33222_ * 15.0F * ((float)Math.PI / 180F);
+         Phantom.this.f_33097_ = Vec3.m_82528_(Phantom.this.f_33098_).m_82520_((double)(this.f_33220_ * Mth.m_14089_(this.f_33219_)), (double)(-4.0F + this.f_33221_), (double)(this.f_33220_ * Mth.m_14031_(this.f_33219_)));
+      }
+   }
+
+   class PhantomAttackStrategyGoal extends Goal {
+      private int f_33202_;
+
+      private PhantomAttackStrategyGoal() {
+      }
+
+      public boolean m_8036_() {
+         LivingEntity livingentity = Phantom.this.m_5448_();
+         return livingentity != null ? Phantom.this.m_21040_(Phantom.this.m_5448_(), TargetingConditions.f_26872_) : false;
+      }
+
+      public void m_8056_() {
+         this.f_33202_ = 10;
+         Phantom.this.f_33096_ = Phantom.AttackPhase.CIRCLE;
+         this.m_33212_();
+      }
+
+      public void m_8041_() {
+         Phantom.this.f_33098_ = Phantom.this.f_19853_.m_5452_(Heightmap.Types.MOTION_BLOCKING, Phantom.this.f_33098_).m_6630_(10 + Phantom.this.f_19796_.nextInt(20));
+      }
+
+      public void m_8037_() {
+         if (Phantom.this.f_33096_ == Phantom.AttackPhase.CIRCLE) {
+            --this.f_33202_;
+            if (this.f_33202_ <= 0) {
+               Phantom.this.f_33096_ = Phantom.AttackPhase.SWOOP;
+               this.m_33212_();
+               this.f_33202_ = (8 + Phantom.this.f_19796_.nextInt(4)) * 20;
+               Phantom.this.m_5496_(SoundEvents.f_12232_, 10.0F, 0.95F + Phantom.this.f_19796_.nextFloat() * 0.1F);
+            }
+         }
+
+      }
+
+      private void m_33212_() {
+         Phantom.this.f_33098_ = Phantom.this.m_5448_().m_20183_().m_6630_(20 + Phantom.this.f_19796_.nextInt(20));
+         if (Phantom.this.f_33098_.m_123342_() < Phantom.this.f_19853_.m_5736_()) {
+            Phantom.this.f_33098_ = new BlockPos(Phantom.this.f_33098_.m_123341_(), Phantom.this.f_19853_.m_5736_() + 1, Phantom.this.f_33098_.m_123343_());
+         }
+
       }
    }
 
