--- a/net/minecraft/world/entity/monster/SpellcasterIllager.java
+++ b/net/minecraft/world/entity/monster/SpellcasterIllager.java
@@ -96,6 +_,33 @@
 
    protected abstract SoundEvent m_7894_();
 
+   public class SpellcasterCastingSpellGoal extends Goal {
+      public SpellcasterCastingSpellGoal() {
+         this.m_7021_(EnumSet.of(Goal.Flag.MOVE, Goal.Flag.LOOK));
+      }
+
+      public boolean m_8036_() {
+         return SpellcasterIllager.this.m_33738_() > 0;
+      }
+
+      public void m_8056_() {
+         super.m_8056_();
+         SpellcasterIllager.this.f_21344_.m_26573_();
+      }
+
+      public void m_8041_() {
+         super.m_8041_();
+         SpellcasterIllager.this.m_33727_(SpellcasterIllager.IllagerSpell.NONE);
+      }
+
+      public void m_8037_() {
+         if (SpellcasterIllager.this.m_5448_() != null) {
+            SpellcasterIllager.this.m_21563_().m_24960_(SpellcasterIllager.this.m_5448_(), (float)SpellcasterIllager.this.m_8085_(), (float)SpellcasterIllager.this.m_8132_());
+         }
+
+      }
+   }
+
    public static enum IllagerSpell {
       NONE(0, 0.0D, 0.0D, 0.0D),
       SUMMON_VEX(1, 0.7D, 0.7D, 0.8D),
@@ -120,33 +_,6 @@
          }
 
          return NONE;
-      }
-   }
-
-   public class SpellcasterCastingSpellGoal extends Goal {
-      public SpellcasterCastingSpellGoal() {
-         this.m_7021_(EnumSet.of(Goal.Flag.MOVE, Goal.Flag.LOOK));
-      }
-
-      public boolean m_8036_() {
-         return SpellcasterIllager.this.m_33738_() > 0;
-      }
-
-      public void m_8056_() {
-         super.m_8056_();
-         SpellcasterIllager.this.f_21344_.m_26573_();
-      }
-
-      public void m_8041_() {
-         super.m_8041_();
-         SpellcasterIllager.this.m_33727_(SpellcasterIllager.IllagerSpell.NONE);
-      }
-
-      public void m_8037_() {
-         if (SpellcasterIllager.this.m_5448_() != null) {
-            SpellcasterIllager.this.m_21563_().m_24960_(SpellcasterIllager.this.m_5448_(), (float)SpellcasterIllager.this.m_8085_(), (float)SpellcasterIllager.this.m_8132_());
-         }
-
       }
    }
 
