--- a/net/minecraft/world/entity/monster/Strider.java
+++ b/net/minecraft/world/entity/monster/Strider.java
@@ -418,6 +_,25 @@
       return new AgableMob.AgableMobGroupData(0.0F);
    }
 
+   static class StriderPathNavigation extends GroundPathNavigation {
+      StriderPathNavigation(Strider p_33969_, Level p_33970_) {
+         super(p_33969_, p_33970_);
+      }
+
+      protected PathFinder m_5532_(int p_33972_) {
+         this.f_26508_ = new WalkNodeEvaluator();
+         return new PathFinder(this.f_26508_, p_33972_);
+      }
+
+      protected boolean m_7367_(BlockPathTypes p_33974_) {
+         return p_33974_ != BlockPathTypes.LAVA && p_33974_ != BlockPathTypes.DAMAGE_FIRE && p_33974_ != BlockPathTypes.DANGER_FIRE ? super.m_7367_(p_33974_) : true;
+      }
+
+      public boolean m_6342_(BlockPos p_33976_) {
+         return this.f_26495_.m_8055_(p_33976_).m_60713_(Blocks.f_49991_) || super.m_6342_(p_33976_);
+      }
+   }
+
    static class StriderGoToLavaGoal extends MoveToBlockGoal {
       private final Strider f_33953_;
 
@@ -444,25 +_,6 @@
 
       protected boolean m_6465_(LevelReader p_33963_, BlockPos p_33964_) {
          return p_33963_.m_8055_(p_33964_).m_60713_(Blocks.f_49991_) && p_33963_.m_8055_(p_33964_.m_7494_()).m_60647_(p_33963_, p_33964_, PathComputationType.LAND);
-      }
-   }
-
-   static class StriderPathNavigation extends GroundPathNavigation {
-      StriderPathNavigation(Strider p_33969_, Level p_33970_) {
-         super(p_33969_, p_33970_);
-      }
-
-      protected PathFinder m_5532_(int p_33972_) {
-         this.f_26508_ = new WalkNodeEvaluator();
-         return new PathFinder(this.f_26508_, p_33972_);
-      }
-
-      protected boolean m_7367_(BlockPathTypes p_33974_) {
-         return p_33974_ != BlockPathTypes.LAVA && p_33974_ != BlockPathTypes.DAMAGE_FIRE && p_33974_ != BlockPathTypes.DANGER_FIRE ? super.m_7367_(p_33974_) : true;
-      }
-
-      public boolean m_6342_(BlockPos p_33976_) {
-         return this.f_26495_.m_8055_(p_33976_).m_60713_(Blocks.f_49991_) || super.m_6342_(p_33976_);
       }
    }
 }
