--- a/net/minecraft/network/protocol/status/ClientboundStatusResponsePacket.java
+++ b/net/minecraft/network/protocol/status/ClientboundStatusResponsePacket.java
@@ -13,7 +_,7 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class ClientboundStatusResponsePacket implements Packet<ClientStatusPacketListener> {
-   private static final Gson f_134885_ = (new GsonBuilder()).registerTypeAdapter(ServerStatus.Version.class, new ServerStatus.Version.Serializer()).registerTypeAdapter(ServerStatus.Players.class, new ServerStatus.Players.Serializer()).registerTypeAdapter(ServerStatus.class, new ServerStatus.Serializer()).registerTypeHierarchyAdapter(Component.class, new Component.Serializer()).registerTypeHierarchyAdapter(Style.class, new Style.Serializer()).registerTypeAdapterFactory(new LowerCaseEnumTypeAdapterFactory()).create();
+   public static final Gson f_134885_ = (new GsonBuilder()).registerTypeAdapter(ServerStatus.Version.class, new ServerStatus.Version.Serializer()).registerTypeAdapter(ServerStatus.Players.class, new ServerStatus.Players.Serializer()).registerTypeAdapter(ServerStatus.class, new ServerStatus.Serializer()).registerTypeHierarchyAdapter(Component.class, new Component.Serializer()).registerTypeHierarchyAdapter(Style.class, new Style.Serializer()).registerTypeAdapterFactory(new LowerCaseEnumTypeAdapterFactory()).create();
    private ServerStatus f_134886_;
 
    public ClientboundStatusResponsePacket() {
@@ -28,7 +_,7 @@
    }
 
    public void m_5779_(FriendlyByteBuf p_134899_) throws IOException {
-      p_134899_.m_130070_(f_134885_.toJson(this.f_134886_));
+      p_134899_.m_130070_(this.f_134886_.getJson()); //Forge: Let the response cache the JSON
    }
 
    public void m_5797_(ClientStatusPacketListener p_134896_) {
