--- a/net/minecraft/client/gui/screens/AccessibilityOptionsScreen.java
+++ b/net/minecraft/client/gui/screens/AccessibilityOptionsScreen.java
@@ -18,16 +_,16 @@
    }
 
    protected void m_7853_() {
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ - 27, 150, 20, new TranslatableComponent("options.accessibility.link"), (p_95509_) -> {
-         this.f_96541_.m_91152_(new ConfirmLinkScreen((p_95512_) -> {
-            if (p_95512_) {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ - 27, 150, 20, new TranslatableComponent("options.accessibility.link"), (p_244738_1_) -> {
+         this.f_96541_.m_91152_(new ConfirmLinkScreen((p_244739_1_) -> {
+            if (p_244739_1_) {
                Util.m_137581_().m_137646_("https://aka.ms/MinecraftJavaAccessibility");
             }
 
             this.f_96541_.m_91152_(this);
          }, "https://aka.ms/MinecraftJavaAccessibility", true));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ - 27, 150, 20, CommonComponents.f_130655_, (p_95507_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ - 27, 150, 20, CommonComponents.f_130655_, (p_244737_1_) -> {
          this.f_96541_.m_91152_(this.f_96281_);
       }));
    }
