--- a/net/minecraft/client/gui/screens/AlertScreen.java
+++ b/net/minecraft/client/gui/screens/AlertScreen.java
@@ -30,7 +_,7 @@
 
    protected void m_7856_() {
       super.m_7856_();
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 6 + 168, 200, 20, this.f_95514_, (p_95533_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 6 + 168, 200, 20, this.f_95514_, (p_212983_1_) -> {
          this.f_95515_.run();
       }));
       this.f_95516_ = MultiLineLabel.m_94341_(this.f_96547_, this.f_95513_, this.f_96543_ - 50);
