--- a/net/minecraft/client/gui/screens/BackupConfirmScreen.java
+++ b/net/minecraft/client/gui/screens/BackupConfirmScreen.java
@@ -33,13 +_,13 @@
       super.m_7856_();
       this.f_95540_ = MultiLineLabel.m_94341_(this.f_96547_, this.f_95538_, this.f_96543_ - 50);
       int i = (this.f_95540_.m_5770_() + 1) * 9;
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155, 100 + i, 150, 20, new TranslatableComponent("selectWorld.backupJoinConfirmButton"), (p_95564_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155, 100 + i, 150, 20, new TranslatableComponent("selectWorld.backupJoinConfirmButton"), (p_212993_1_) -> {
          this.f_95536_.m_95565_(true, this.f_95541_.m_93840_());
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155 + 160, 100 + i, 150, 20, new TranslatableComponent("selectWorld.backupJoinSkipButton"), (p_95562_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155 + 160, 100 + i, 150, 20, new TranslatableComponent("selectWorld.backupJoinSkipButton"), (p_212992_1_) -> {
          this.f_95536_.m_95565_(false, this.f_95541_.m_93840_());
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155 + 80, 124 + i, 150, 20, CommonComponents.f_130656_, (p_95558_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155 + 80, 124 + i, 150, 20, CommonComponents.f_130656_, (p_212991_1_) -> {
          this.f_96541_.m_91152_(this.f_95537_);
       }));
       this.f_95541_ = new Checkbox(this.f_96543_ / 2 - 155 + 80, 76 + i, 150, 20, new TranslatableComponent("selectWorld.backupEraseCache"), false);
