--- a/net/minecraft/client/gui/screens/ConfirmLinkScreen.java
+++ b/net/minecraft/client/gui/screens/ConfirmLinkScreen.java
@@ -31,14 +_,14 @@
       super.m_7856_();
       this.f_96545_.clear();
       this.f_96540_.clear();
-      this.m_7082_(new Button(this.f_96543_ / 2 - 50 - 105, this.f_96544_ / 6 + 96, 100, 20, this.f_95647_, (p_95645_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 50 - 105, this.f_96544_ / 6 + 96, 100, 20, this.f_95647_, (p_213006_1_) -> {
          this.f_95649_.accept(true);
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 50, this.f_96544_ / 6 + 96, 100, 20, this.f_95627_, (p_95643_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 50, this.f_96544_ / 6 + 96, 100, 20, this.f_95627_, (p_213005_1_) -> {
          this.m_95646_();
          this.f_95649_.accept(false);
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 50 + 105, this.f_96544_ / 6 + 96, 100, 20, this.f_95648_, (p_95640_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 50 + 105, this.f_96544_ / 6 + 96, 100, 20, this.f_95648_, (p_213004_1_) -> {
          this.f_95649_.accept(false);
       }));
    }
