--- a/net/minecraft/client/gui/screens/DatapackLoadFailureScreen.java
+++ b/net/minecraft/client/gui/screens/DatapackLoadFailureScreen.java
@@ -20,10 +_,10 @@
    protected void m_7856_() {
       super.m_7856_();
       this.f_95891_ = MultiLineLabel.m_94341_(this.f_96547_, this.m_96636_(), this.f_96543_ - 50);
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 96, 150, 20, new TranslatableComponent("datapackFailure.safeMode"), (p_95905_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 96, 150, 20, new TranslatableComponent("datapackFailure.safeMode"), (p_238622_1_) -> {
          this.f_95892_.run();
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155 + 160, this.f_96544_ / 6 + 96, 150, 20, new TranslatableComponent("gui.toTitle"), (p_95901_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155 + 160, this.f_96544_ / 6 + 96, 150, 20, new TranslatableComponent("gui.toTitle"), (p_238621_1_) -> {
          this.f_96541_.m_91152_((Screen)null);
       }));
    }
