--- a/net/minecraft/client/gui/screens/DeathScreen.java
+++ b/net/minecraft/client/gui/screens/DeathScreen.java
@@ -29,12 +_,13 @@
 
    protected void m_7856_() {
       this.f_95906_ = 0;
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 72, 200, 20, this.f_95908_ ? new TranslatableComponent("deathScreen.spectate") : new TranslatableComponent("deathScreen.respawn"), (p_95930_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 72, 200, 20, this.f_95908_ ? new TranslatableComponent("deathScreen.spectate") : new TranslatableComponent("deathScreen.respawn"), (p_213021_1_) -> {
          this.f_96541_.f_91074_.m_7583_();
          this.f_96541_.m_91152_((Screen)null);
       }));
-      Button button = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 96, 200, 20, new TranslatableComponent("deathScreen.titleScreen"), (p_95925_) -> {
+      Button button = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 96, 200, 20, new TranslatableComponent("deathScreen.titleScreen"), (p_213020_1_) -> {
          if (this.f_95908_) {
+            m_95931_(true);
             this.m_95934_();
          } else {
             ConfirmScreen confirmscreen = new ConfirmScreen(this::m_95931_, new TranslatableComponent("deathScreen.quit.confirm"), TextComponent.f_131282_, new TranslatableComponent("deathScreen.titleScreen"), new TranslatableComponent("deathScreen.respawn"));
