--- a/net/minecraft/client/gui/screens/DemoIntroScreen.java
+++ b/net/minecraft/client/gui/screens/DemoIntroScreen.java
@@ -23,11 +_,11 @@
 
    protected void m_7856_() {
       int i = -16;
-      this.m_7082_(new Button(this.f_96543_ / 2 - 116, this.f_96544_ / 2 + 62 + -16, 114, 20, new TranslatableComponent("demo.help.buy"), (p_95951_) -> {
-         p_95951_.f_93623_ = false;
+      this.m_7082_(new Button(this.f_96543_ / 2 - 116, this.f_96544_ / 2 + 62 + -16, 114, 20, new TranslatableComponent("demo.help.buy"), (p_213019_0_) -> {
+         p_213019_0_.f_93623_ = false;
          Util.m_137581_().m_137646_("http://www.minecraft.net/store?source=demo");
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 2, this.f_96544_ / 2 + 62 + -16, 114, 20, new TranslatableComponent("demo.help.later"), (p_95948_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 2, this.f_96544_ / 2 + 62 + -16, 114, 20, new TranslatableComponent("demo.help.later"), (p_213018_1_) -> {
          this.f_96541_.m_91152_((Screen)null);
          this.f_96541_.f_91067_.m_91601_();
       }));
