--- a/net/minecraft/client/gui/screens/DirectJoinServerScreen.java
+++ b/net/minecraft/client/gui/screens/DirectJoinServerScreen.java
@@ -43,17 +_,17 @@
 
    protected void m_7856_() {
       this.f_96541_.f_91068_.m_90926_(true);
-      this.f_95953_ = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 96 + 12, 200, 20, new TranslatableComponent("selectServer.select"), (p_95981_) -> {
+      this.f_95953_ = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 96 + 12, 200, 20, new TranslatableComponent("selectServer.select"), (p_213026_1_) -> {
          this.m_95986_();
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 120 + 12, 200, 20, CommonComponents.f_130656_, (p_95977_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 120 + 12, 200, 20, CommonComponents.f_130656_, (p_213025_1_) -> {
          this.f_95956_.accept(false);
       }));
       this.f_95955_ = new EditBox(this.f_96547_, this.f_96543_ / 2 - 100, 116, 200, 20, new TranslatableComponent("addServer.enterIp"));
       this.f_95955_.m_94199_(128);
       this.f_95955_.m_94178_(true);
       this.f_95955_.m_94144_(this.f_96541_.f_91066_.f_92066_);
-      this.f_95955_.m_94151_((p_95983_) -> {
+      this.f_95955_.m_94151_((p_213024_1_) -> {
          this.m_95987_();
       });
       this.f_96540_.add(this.f_95955_);
