--- a/net/minecraft/client/gui/screens/LevelLoadingScreen.java
+++ b/net/minecraft/client/gui/screens/LevelLoadingScreen.java
@@ -16,21 +_,21 @@
 public class LevelLoadingScreen extends Screen {
    private final StoringChunkProgressListener f_96138_;
    private long f_96139_ = -1L;
-   private static final Object2IntMap<ChunkStatus> f_96140_ = Util.m_137469_(new Object2IntOpenHashMap<>(), (p_96157_) -> {
-      p_96157_.defaultReturnValue(0);
-      p_96157_.put(ChunkStatus.f_62314_, 5526612);
-      p_96157_.put(ChunkStatus.f_62315_, 10066329);
-      p_96157_.put(ChunkStatus.f_62316_, 6250897);
-      p_96157_.put(ChunkStatus.f_62317_, 8434258);
-      p_96157_.put(ChunkStatus.f_62318_, 13750737);
-      p_96157_.put(ChunkStatus.f_62319_, 7497737);
-      p_96157_.put(ChunkStatus.f_62320_, 7169628);
-      p_96157_.put(ChunkStatus.f_62321_, 3159410);
-      p_96157_.put(ChunkStatus.f_62322_, 2213376);
-      p_96157_.put(ChunkStatus.f_62323_, 13421772);
-      p_96157_.put(ChunkStatus.f_62324_, 15884384);
-      p_96157_.put(ChunkStatus.f_62325_, 15658734);
-      p_96157_.put(ChunkStatus.f_62326_, 16777215);
+   private static final Object2IntMap<ChunkStatus> f_96140_ = Util.m_137469_(new Object2IntOpenHashMap<>(), (p_213039_0_) -> {
+      p_213039_0_.defaultReturnValue(0);
+      p_213039_0_.put(ChunkStatus.f_62314_, 5526612);
+      p_213039_0_.put(ChunkStatus.f_62315_, 10066329);
+      p_213039_0_.put(ChunkStatus.f_62316_, 6250897);
+      p_213039_0_.put(ChunkStatus.f_62317_, 8434258);
+      p_213039_0_.put(ChunkStatus.f_62318_, 13750737);
+      p_213039_0_.put(ChunkStatus.f_62319_, 7497737);
+      p_213039_0_.put(ChunkStatus.f_62320_, 7169628);
+      p_213039_0_.put(ChunkStatus.f_62321_, 3159410);
+      p_213039_0_.put(ChunkStatus.f_62322_, 2213376);
+      p_213039_0_.put(ChunkStatus.f_62323_, 13421772);
+      p_213039_0_.put(ChunkStatus.f_62324_, 15884384);
+      p_213039_0_.put(ChunkStatus.f_62325_, 15658734);
+      p_213039_0_.put(ChunkStatus.f_62326_, 16777215);
    });
 
    public LevelLoadingScreen(StoringChunkProgressListener p_96143_) {
