--- a/net/minecraft/client/gui/screens/MouseSettingsScreen.java
+++ b/net/minecraft/client/gui/screens/MouseSettingsScreen.java
@@ -25,15 +_,15 @@
    protected void m_7856_() {
       this.f_96218_ = new OptionsList(this.f_96541_, this.f_96543_, this.f_96544_, 32, this.f_96544_ - 32, 25);
       if (InputConstants.m_84826_()) {
-         this.f_96218_.m_94476_(Stream.concat(Arrays.stream(f_96219_), Stream.of(Option.f_91674_)).toArray((p_96225_) -> {
-            return new Option[p_96225_];
+         this.f_96218_.m_94476_(Stream.concat(Arrays.stream(f_96219_), Stream.of(Option.f_91674_)).toArray((p_223702_0_) -> {
+            return new Option[p_223702_0_];
          }));
       } else {
          this.f_96218_.m_94476_(f_96219_);
       }
 
       this.f_96540_.add(this.f_96218_);
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ - 27, 200, 20, CommonComponents.f_130655_, (p_96232_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ - 27, 200, 20, CommonComponents.f_130655_, (p_223703_1_) -> {
          this.f_96282_.m_92169_();
          this.f_96541_.m_91152_(this.f_96281_);
       }));
