--- a/net/minecraft/client/gui/screens/OptionsScreen.java
+++ b/net/minecraft/client/gui/screens/OptionsScreen.java
@@ -48,14 +_,14 @@
 
       if (this.f_96541_.f_91073_ != null) {
          this.f_96239_ = this.f_96541_.f_91073_.m_46791_();
-         this.f_96237_ = this.m_7082_(new Button(this.f_96543_ / 2 - 155 + i % 2 * 160, this.f_96544_ / 6 - 12 + 24 * (i >> 1), 150, 20, this.m_96246_(this.f_96239_), (p_96280_) -> {
+         this.f_96237_ = this.m_7082_(new Button(this.f_96543_ / 2 - 155 + i % 2 * 160, this.f_96544_ / 6 - 12 + 24 * (i >> 1), 150, 20, this.m_96246_(this.f_96239_), (p_213051_1_) -> {
             this.f_96239_ = Difficulty.m_19029_(this.f_96239_.m_19028_() + 1);
             this.f_96541_.m_91403_().m_104955_(new ServerboundChangeDifficultyPacket(this.f_96239_));
             this.f_96237_.m_93666_(this.m_96246_(this.f_96239_));
          }));
          if (this.f_96541_.m_91091_() && !this.f_96541_.f_91073_.m_6106_().m_5466_()) {
             this.f_96237_.m_93674_(this.f_96237_.m_5711_() - 20);
-            this.f_96238_ = this.m_7082_(new LockIconButton(this.f_96237_.f_93620_ + this.f_96237_.m_5711_(), this.f_96237_.f_93621_, (p_96278_) -> {
+            this.f_96238_ = this.m_7082_(new LockIconButton(this.f_96237_.f_93620_ + this.f_96237_.m_5711_(), this.f_96237_.f_93621_, (p_213054_1_) -> {
                this.f_96541_.m_91152_(new ConfirmScreen(this::m_96260_, new TranslatableComponent("difficulty.lock.title"), new TranslatableComponent("difficulty.lock.question", new TranslatableComponent("options.difficulty." + this.f_96541_.f_91073_.m_6106_().m_5472_().m_19036_()))));
             }));
             this.f_96238_.m_94309_(this.f_96541_.f_91073_.m_6106_().m_5474_());
@@ -65,38 +_,38 @@
             this.f_96237_.f_93623_ = false;
          }
       } else {
-         this.m_7082_(new OptionButton(this.f_96543_ / 2 - 155 + i % 2 * 160, this.f_96544_ / 6 - 12 + 24 * (i >> 1), 150, 20, Option.f_91642_, Option.f_91642_.m_90547_(this.f_96236_), (p_96276_) -> {
+         this.m_7082_(new OptionButton(this.f_96543_ / 2 - 155 + i % 2 * 160, this.f_96544_ / 6 - 12 + 24 * (i >> 1), 150, 20, Option.f_91642_, Option.f_91642_.m_90547_(this.f_96236_), (p_213057_1_) -> {
             Option.f_91642_.m_90529_(this.f_96236_);
             this.f_96236_.m_92169_();
-            p_96276_.m_93666_(Option.f_91642_.m_90547_(this.f_96236_));
+            p_213057_1_.m_93666_(Option.f_91642_.m_90547_(this.f_96236_));
          }));
       }
 
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 48 - 6, 150, 20, new TranslatableComponent("options.skinCustomisation"), (p_96274_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 48 - 6, 150, 20, new TranslatableComponent("options.skinCustomisation"), (p_213055_1_) -> {
          this.f_96541_.m_91152_(new SkinCustomizationScreen(this, this.f_96236_));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 6 + 48 - 6, 150, 20, new TranslatableComponent("options.sounds"), (p_96272_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 6 + 48 - 6, 150, 20, new TranslatableComponent("options.sounds"), (p_213061_1_) -> {
          this.f_96541_.m_91152_(new SoundOptionsScreen(this, this.f_96236_));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 72 - 6, 150, 20, new TranslatableComponent("options.video"), (p_96270_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 72 - 6, 150, 20, new TranslatableComponent("options.video"), (p_213059_1_) -> {
          this.f_96541_.m_91152_(new VideoSettingsScreen(this, this.f_96236_));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 6 + 72 - 6, 150, 20, new TranslatableComponent("options.controls"), (p_96268_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 6 + 72 - 6, 150, 20, new TranslatableComponent("options.controls"), (p_213052_1_) -> {
          this.f_96541_.m_91152_(new ControlsScreen(this, this.f_96236_));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 96 - 6, 150, 20, new TranslatableComponent("options.language"), (p_96266_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 96 - 6, 150, 20, new TranslatableComponent("options.language"), (p_213053_1_) -> {
          this.f_96541_.m_91152_(new LanguageSelectScreen(this, this.f_96236_, this.f_96541_.m_91102_()));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 6 + 96 - 6, 150, 20, new TranslatableComponent("options.chat.title"), (p_96263_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 6 + 96 - 6, 150, 20, new TranslatableComponent("options.chat.title"), (p_213049_1_) -> {
          this.f_96541_.m_91152_(new ChatOptionsScreen(this, this.f_96236_));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 120 - 6, 150, 20, new TranslatableComponent("options.resourcepack"), (p_96259_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ / 6 + 120 - 6, 150, 20, new TranslatableComponent("options.resourcepack"), (p_213060_1_) -> {
          this.f_96541_.m_91152_(new PackSelectionScreen(this, this.f_96541_.m_91099_(), this::m_96244_, this.f_96541_.m_91101_(), new TranslatableComponent("resourcePack.title")));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 6 + 120 - 6, 150, 20, new TranslatableComponent("options.accessibility.title"), (p_96257_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ / 6 + 120 - 6, 150, 20, new TranslatableComponent("options.accessibility.title"), (p_213058_1_) -> {
          this.f_96541_.m_91152_(new AccessibilityOptionsScreen(this, this.f_96236_));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 6 + 168, 200, 20, CommonComponents.f_130655_, (p_96254_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 6 + 168, 200, 20, CommonComponents.f_130655_, (p_213056_1_) -> {
          this.f_96541_.m_91152_(this.f_96235_);
       }));
    }
@@ -147,4 +_,12 @@
       m_93215_(p_96249_, this.f_96547_, this.f_96539_, this.f_96543_ / 2, 15, 16777215);
       super.m_6305_(p_96249_, p_96250_, p_96251_, p_96252_);
    }
+
+    @Override
+    public void m_7379_() {
+        // We need to consider 2 potential parent screens here:
+        // 1. From the main menu, in which case display the main menu
+        // 2. From the pause menu, in which case exit back to game
+        this.f_96541_.m_91152_(this.f_96235_ instanceof PauseScreen ? null : this.f_96235_);
+    }
 }
