--- a/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
+++ b/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
@@ -175,12 +_,12 @@
       this.f_96540_.add(this.f_96375_);
       this.f_96373_ = new PresetFlatWorldScreen.PresetsList();
       this.f_96540_.add(this.f_96373_);
-      this.f_96374_ = this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ - 28, 150, 20, new TranslatableComponent("createWorld.customize.presets.select"), (p_96405_) -> {
+      this.f_96374_ = this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ - 28, 150, 20, new TranslatableComponent("createWorld.customize.presets.select"), (p_243298_2_) -> {
          FlatLevelGeneratorSettings flatlevelgeneratorsettings = m_96406_(registry, this.f_96375_.m_94155_(), this.f_96376_);
          this.f_96370_.m_95825_(flatlevelgeneratorsettings);
          this.f_96541_.m_91152_(this.f_96370_);
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ - 28, 150, 20, CommonComponents.f_130656_, (p_96394_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 5, this.f_96544_ - 28, 150, 20, CommonComponents.f_130656_, (p_243294_1_) -> {
          this.f_96541_.m_91152_(this.f_96370_);
       }));
       this.m_96449_(this.f_96373_.m_93511_() != null);
@@ -227,7 +_,7 @@
    }
 
    private static void m_96424_(Component p_96425_, ItemLike p_96426_, ResourceKey<Biome> p_96427_, List<StructureFeature<?>> p_96428_, boolean p_96429_, boolean p_96430_, boolean p_96431_, FlatLayerInfo... p_96432_) {
-      f_96369_.add(new PresetFlatWorldScreen.PresetInfo(p_96426_.m_5456_(), p_96425_, (p_96423_) -> {
+      f_96369_.add(new PresetFlatWorldScreen.PresetInfo(p_96426_.m_5456_(), p_96425_, (p_243301_6_) -> {
          Map<StructureFeature<?>, StructureFeatureConfiguration> map = Maps.newHashMap();
 
          for(StructureFeature<?> structurefeature : p_96428_) {
@@ -235,7 +_,7 @@
          }
 
          StructureSettings structuresettings = new StructureSettings(p_96429_ ? Optional.of(StructureSettings.f_64581_) : Optional.empty(), map);
-         FlatLevelGeneratorSettings flatlevelgeneratorsettings = new FlatLevelGeneratorSettings(structuresettings, p_96423_);
+         FlatLevelGeneratorSettings flatlevelgeneratorsettings = new FlatLevelGeneratorSettings(structuresettings, p_243301_6_);
          if (p_96430_) {
             flatlevelgeneratorsettings.m_70369_();
          }
@@ -249,7 +_,7 @@
          }
 
          flatlevelgeneratorsettings.m_70383_(() -> {
-            return p_96423_.m_123013_(p_96427_);
+            return p_243301_6_.m_123013_(p_96427_);
          });
          flatlevelgeneratorsettings.m_70403_();
          return flatlevelgeneratorsettings.m_70370_(structuresettings);
