--- a/net/minecraft/client/gui/screens/SoundOptionsScreen.java
+++ b/net/minecraft/client/gui/screens/SoundOptionsScreen.java
@@ -33,12 +_,12 @@
       int j = this.f_96543_ / 2 - 75;
       int k = this.f_96544_ / 6 - 12;
       ++i;
-      this.m_7082_(new OptionButton(j, k + 24 * (i >> 1), 150, 20, Option.f_91644_, Option.f_91644_.m_90547_(this.f_96282_), (p_96713_) -> {
+      this.m_7082_(new OptionButton(j, k + 24 * (i >> 1), 150, 20, Option.f_91644_, Option.f_91644_.m_90547_(this.f_96282_), (p_213105_1_) -> {
          Option.f_91644_.m_90529_(this.f_96541_.f_91066_);
-         p_96713_.m_93666_(Option.f_91644_.m_90547_(this.f_96541_.f_91066_));
+         p_213105_1_.m_93666_(Option.f_91644_.m_90547_(this.f_96541_.f_91066_));
          this.f_96541_.f_91066_.m_92169_();
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 6 + 168, 200, 20, CommonComponents.f_130655_, (p_96710_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 6 + 168, 200, 20, CommonComponents.f_130655_, (p_213104_1_) -> {
          this.f_96541_.m_91152_(this.f_96281_);
       }));
    }
