--- a/net/minecraft/client/gui/screens/VideoSettingsScreen.java
+++ b/net/minecraft/client/gui/screens/VideoSettingsScreen.java
@@ -52,7 +_,7 @@
       this.f_96801_.m_94471_(Option.f_91653_);
       this.f_96801_.m_94476_(f_96800_);
       this.f_96540_.add(this.f_96801_);
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ - 27, 200, 20, CommonComponents.f_130655_, (p_96827_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ - 27, 200, 20, CommonComponents.f_130655_, (p_213106_1_) -> {
          this.f_96541_.f_91066_.m_92169_();
          this.f_96541_.m_91268_().m_85437_();
          this.f_96541_.m_91152_(this.f_96281_);
@@ -95,12 +_,12 @@
                list.add((new TranslatableComponent("options.graphics.warning.version", s2)).m_130940_(ChatFormatting.GRAY));
             }
 
-            this.f_96541_.m_91152_(new PopupScreen(f_96796_, list, ImmutableList.of(new PopupScreen.ButtonOption(f_96797_, (p_96821_) -> {
+            this.f_96541_.m_91152_(new PopupScreen(f_96796_, list, ImmutableList.of(new PopupScreen.ButtonOption(f_96797_, (p_241606_1_) -> {
                this.f_96282_.f_92115_ = GraphicsStatus.FABULOUS;
                Minecraft.m_91087_().f_91060_.m_109818_();
                this.f_96802_.m_109248_();
                this.f_96541_.m_91152_(this);
-            }), new PopupScreen.ButtonOption(f_96798_, (p_96818_) -> {
+            }), new PopupScreen.ButtonOption(f_96798_, (p_241605_1_) -> {
                this.f_96802_.m_109249_();
                this.f_96541_.m_91152_(this);
             }))));
