--- a/net/minecraft/client/gui/screens/WinScreen.java
+++ b/net/minecraft/client/gui/screens/WinScreen.java
@@ -187,9 +_,9 @@
       RenderSystem.m_69439_(1.0F, 1.0F, 1.0F, 1.0F);
       RenderSystem.m_69477_();
       RenderSystem.m_69478_();
-      this.m_93101_(j, k, (p_96890_, p_96891_) -> {
-         this.m_93228_(p_96884_, p_96890_ + 0, p_96891_, 0, 0, 155, 44);
-         this.m_93228_(p_96884_, p_96890_ + 155, p_96891_, 0, 45, 155, 44);
+      this.m_93101_(j, k, (p_238665_2_, p_238665_3_) -> {
+         this.m_93228_(p_96884_, p_238665_2_ + 0, p_238665_3_, 0, 0, 155, 44);
+         this.m_93228_(p_96884_, p_238665_2_ + 155, p_238665_3_, 0, 45, 155, 44);
       });
       RenderSystem.m_69461_();
       this.f_96541_.m_91097_().m_118493_(f_96865_);
