--- a/net/minecraft/world/level/levelgen/Heightmap.java
+++ b/net/minecraft/world/level/levelgen/Heightmap.java
@@ -21,11 +_,11 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class Heightmap {
-   private static final Predicate<BlockState> f_64230_ = (p_64263_) -> {
-      return !p_64263_.m_60795_();
+   private static final Predicate<BlockState> f_64230_ = (p_222688_0_) -> {
+      return !p_222688_0_.m_60795_();
    };
-   private static final Predicate<BlockState> f_64231_ = (p_64255_) -> {
-      return p_64255_.m_60767_().m_76334_();
+   private static final Predicate<BlockState> f_64231_ = (p_222689_0_) -> {
+      return p_222689_0_.m_60767_().m_76334_();
    };
    private final BitStorage f_64232_ = new BitStorage(9, 256);
    private final Predicate<BlockState> f_64233_;
@@ -131,20 +_,20 @@
       WORLD_SURFACE("WORLD_SURFACE", Heightmap.Usage.CLIENT, Heightmap.f_64230_),
       OCEAN_FLOOR_WG("OCEAN_FLOOR_WG", Heightmap.Usage.WORLDGEN, Heightmap.f_64231_),
       OCEAN_FLOOR("OCEAN_FLOOR", Heightmap.Usage.LIVE_WORLD, Heightmap.f_64231_),
-      MOTION_BLOCKING("MOTION_BLOCKING", Heightmap.Usage.CLIENT, (p_64296_) -> {
-         return p_64296_.m_60767_().m_76334_() || !p_64296_.m_60819_().m_76178_();
+      MOTION_BLOCKING("MOTION_BLOCKING", Heightmap.Usage.CLIENT, (p_222680_0_) -> {
+         return p_222680_0_.m_60767_().m_76334_() || !p_222680_0_.m_60819_().m_76178_();
       }),
-      MOTION_BLOCKING_NO_LEAVES("MOTION_BLOCKING_NO_LEAVES", Heightmap.Usage.LIVE_WORLD, (p_64289_) -> {
-         return (p_64289_.m_60767_().m_76334_() || !p_64289_.m_60819_().m_76178_()) && !(p_64289_.m_60734_() instanceof LeavesBlock);
+      MOTION_BLOCKING_NO_LEAVES("MOTION_BLOCKING_NO_LEAVES", Heightmap.Usage.LIVE_WORLD, (p_222682_0_) -> {
+         return (p_222682_0_.m_60767_().m_76334_() || !p_222682_0_.m_60819_().m_76178_()) && !(p_222682_0_.m_60734_() instanceof LeavesBlock);
       });
 
       public static final Codec<Heightmap.Types> f_64274_ = StringRepresentable.m_14350_(Heightmap.Types::values, Heightmap.Types::m_64290_);
       private final String f_64275_;
       private final Heightmap.Usage f_64276_;
       private final Predicate<BlockState> f_64277_;
-      private static final Map<String, Heightmap.Types> f_64278_ = Util.m_137469_(Maps.newHashMap(), (p_64293_) -> {
+      private static final Map<String, Heightmap.Types> f_64278_ = Util.m_137469_(Maps.newHashMap(), (p_222679_0_) -> {
          for(Heightmap.Types heightmap$types : values()) {
-            p_64293_.put(heightmap$types.f_64275_, heightmap$types);
+            p_222679_0_.put(heightmap$types.f_64275_, heightmap$types);
          }
 
       });
