--- a/net/minecraft/world/level/levelgen/NoiseGeneratorSettings.java
+++ b/net/minecraft/world/level/levelgen/NoiseGeneratorSettings.java
@@ -18,8 +_,8 @@
 import net.minecraft.world.level.levelgen.feature.configurations.StructureFeatureConfiguration;
 
 public final class NoiseGeneratorSettings {
-   public static final Codec<NoiseGeneratorSettings> f_64430_ = RecordCodecBuilder.create((p_64475_) -> {
-      return p_64475_.group(StructureSettings.f_64579_.fieldOf("structures").forGetter(NoiseGeneratorSettings::m_64457_), NoiseSettings.f_64507_.fieldOf("noise").forGetter(NoiseGeneratorSettings::m_64481_), BlockState.f_61039_.fieldOf("default_block").forGetter(NoiseGeneratorSettings::m_64482_), BlockState.f_61039_.fieldOf("default_fluid").forGetter(NoiseGeneratorSettings::m_64483_), Codec.intRange(-20, 276).fieldOf("bedrock_roof_position").forGetter(NoiseGeneratorSettings::m_64484_), Codec.intRange(-20, 276).fieldOf("bedrock_floor_position").forGetter(NoiseGeneratorSettings::m_64485_), Codec.intRange(0, 255).fieldOf("sea_level").forGetter(NoiseGeneratorSettings::m_64486_), Codec.BOOL.fieldOf("disable_mob_generation").forGetter(NoiseGeneratorSettings::m_64487_)).apply(p_64475_, NoiseGeneratorSettings::new);
+   public static final Codec<NoiseGeneratorSettings> f_64430_ = RecordCodecBuilder.create((p_236112_0_) -> {
+      return p_236112_0_.group(StructureSettings.f_64579_.fieldOf("structures").forGetter(NoiseGeneratorSettings::m_64457_), NoiseSettings.f_64507_.fieldOf("noise").forGetter(NoiseGeneratorSettings::m_64481_), BlockState.f_61039_.fieldOf("default_block").forGetter(NoiseGeneratorSettings::m_64482_), BlockState.f_61039_.fieldOf("default_fluid").forGetter(NoiseGeneratorSettings::m_64483_), Codec.intRange(-20, 276).fieldOf("bedrock_roof_position").forGetter(NoiseGeneratorSettings::m_64484_), Codec.intRange(-20, 276).fieldOf("bedrock_floor_position").forGetter(NoiseGeneratorSettings::m_64485_), Codec.intRange(0, 255).fieldOf("sea_level").forGetter(NoiseGeneratorSettings::m_64486_), Codec.BOOL.fieldOf("disable_mob_generation").forGetter(NoiseGeneratorSettings::m_64487_)).apply(p_236112_0_, NoiseGeneratorSettings::new);
    });
    public static final Codec<Supplier<NoiseGeneratorSettings>> f_64431_ = RegistryFileCodec.m_135589_(Registry.f_122878_, f_64430_);
    private final StructureSettings f_64438_;
