--- a/net/minecraft/world/level/levelgen/NoiseBasedChunkGenerator.java
+++ b/net/minecraft/world/level/levelgen/NoiseBasedChunkGenerator.java
@@ -530,8 +_,8 @@
                         objectlistiterator1.back(objectlist1.size());
                         BlockState blockstate = this.m_64346_(d18, i2);
                         if (blockstate != f_64321_) {
-                           if (blockstate.m_60791_() != 0) {
-                              blockpos$mutableblockpos.m_122178_(i3, i2, l3);
+                           blockpos$mutableblockpos.m_122178_(i3, i2, l3);
+                           if (blockstate.getLightValue(protochunk, blockpos$mutableblockpos) != 0) {
                               protochunk.m_63277_(blockpos$mutableblockpos);
                            }
 
@@ -587,6 +_,9 @@
    }
 
    public List<MobSpawnSettings.SpawnerData> m_7694_(Biome p_64395_, StructureFeatureManager p_64396_, MobCategory p_64397_, BlockPos p_64398_) {
+      List<MobSpawnSettings.SpawnerData> spawns = net.minecraftforge.common.world.StructureSpawnManager.getStructureSpawns(p_64396_, p_64397_, p_64398_);
+      if (spawns != null) return spawns;
+      if (false) {//Forge: We handle these hardcoded cases above in StructureSpawnManager#getStructureSpawns, but allow for insideOnly to be changed and allow for creatures to be spawned in ones other than just the witch hut
       if (p_64396_.m_47285_(p_64398_, true, StructureFeature.f_67021_).m_73603_()) {
          if (p_64397_ == MobCategory.MONSTER) {
             return StructureFeature.f_67021_.m_7963_();
@@ -609,6 +_,7 @@
          if (p_64396_.m_47285_(p_64398_, true, StructureFeature.f_67025_).m_73603_()) {
             return StructureFeature.f_67025_.m_7963_();
          }
+      }
       }
 
       return super.m_7694_(p_64395_, p_64396_, p_64397_, p_64398_);
