--- a/net/minecraft/world/entity/animal/horse/Llama.java
+++ b/net/minecraft/world/entity/animal/horse/Llama.java
@@ -33,7 +_,6 @@
 import net.minecraft.world.entity.ai.goal.RangedAttackGoal;
 import net.minecraft.world.entity.ai.goal.RunAroundLikeCrazyGoal;
 import net.minecraft.world.entity.ai.goal.WaterAvoidingRandomStrollGoal;
-import net.minecraft.world.entity.ai.goal.target.HurtByTargetGoal;
 import net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal;
 import net.minecraft.world.entity.animal.Animal;
 import net.minecraft.world.entity.animal.Wolf;
@@ -445,8 +_,8 @@
 
    static class LlamaAttackWolfGoal extends NearestAttackableTargetGoal<Wolf> {
       public LlamaAttackWolfGoal(Llama p_30843_) {
-         super(p_30843_, Wolf.class, 16, false, true, (p_30845_) -> {
-            return !((Wolf)p_30845_).m_21824_();
+         super(p_30843_, Wolf.class, 16, false, true, (p_220789_0_) -> {
+            return !((Wolf)p_220789_0_).m_21824_();
          });
       }
 
@@ -455,16 +_,7 @@
       }
    }
 
-   static class LlamaGroupData extends AgableMob.AgableMobGroupData {
-      public final int f_30847_;
-
-      private LlamaGroupData(int p_30849_) {
-         super(true);
-         this.f_30847_ = p_30849_;
-      }
-   }
-
-   static class LlamaHurtByTargetGoal extends HurtByTargetGoal {
+   static class LlamaHurtByTargetGoal extends net.minecraft.world.entity.ai.goal.target.HurtByTargetGoal {
       public LlamaHurtByTargetGoal(Llama p_30854_) {
          super(p_30854_);
       }
@@ -479,6 +_,15 @@
          }
 
          return super.m_8045_();
+      }
+   }
+
+   static class LlamaGroupData extends AgableMob.AgableMobGroupData {
+      public final int f_30847_;
+
+      private LlamaGroupData(int p_30849_) {
+         super(true);
+         this.f_30847_ = p_30849_;
       }
    }
 }
