--- a/net/minecraft/data/loot/GiftLoot.java
+++ b/net/minecraft/data/loot/GiftLoot.java
@@ -24,32 +_,32 @@
       p_124402_.accept(BuiltInLootTables.f_78728_, LootTable.m_79147_().m_79161_(LootPool.m_79043_().m_79074_(ConstantIntValue.m_78775_(1)).m_79076_(LootItem.m_79579_(Items.f_42451_)).m_79076_(LootItem.m_79579_(Items.f_42534_))));
       p_124402_.accept(BuiltInLootTables.f_78729_, LootTable.m_79147_().m_79161_(LootPool.m_79043_().m_79074_(ConstantIntValue.m_78775_(1)).m_79076_(LootItem.m_79579_(Items.f_42406_)).m_79076_(LootItem.m_79579_(Items.f_42687_)).m_79076_(LootItem.m_79579_(Items.f_42572_))));
       p_124402_.accept(BuiltInLootTables.f_78730_, LootTable.m_79147_().m_79161_(LootPool.m_79043_().m_79074_(ConstantIntValue.m_78775_(1)).m_79076_(LootItem.m_79579_(Items.f_42526_)).m_79076_(LootItem.m_79579_(Items.f_42527_))));
-      p_124402_.accept(BuiltInLootTables.f_78731_, LootTable.m_79147_().m_79161_(LootPool.m_79043_().m_79074_(ConstantIntValue.m_78775_(1)).m_79076_(LootItem.m_79579_(Items.f_42412_).m_79707_(26)).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124430_) -> {
-         p_124430_.m_128359_("Potion", "minecraft:swiftness");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124428_) -> {
-         p_124428_.m_128359_("Potion", "minecraft:slowness");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124426_) -> {
-         p_124426_.m_128359_("Potion", "minecraft:strength");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124424_) -> {
-         p_124424_.m_128359_("Potion", "minecraft:healing");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124422_) -> {
-         p_124422_.m_128359_("Potion", "minecraft:harming");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124420_) -> {
-         p_124420_.m_128359_("Potion", "minecraft:leaping");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124418_) -> {
-         p_124418_.m_128359_("Potion", "minecraft:regeneration");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124416_) -> {
-         p_124416_.m_128359_("Potion", "minecraft:fire_resistance");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124414_) -> {
-         p_124414_.m_128359_("Potion", "minecraft:water_breathing");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124412_) -> {
-         p_124412_.m_128359_("Potion", "minecraft:invisibility");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124410_) -> {
-         p_124410_.m_128359_("Potion", "minecraft:night_vision");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124408_) -> {
-         p_124408_.m_128359_("Potion", "minecraft:weakness");
-      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_124404_) -> {
-         p_124404_.m_128359_("Potion", "minecraft:poison");
+      p_124402_.accept(BuiltInLootTables.f_78731_, LootTable.m_79147_().m_79161_(LootPool.m_79043_().m_79074_(ConstantIntValue.m_78775_(1)).m_79076_(LootItem.m_79579_(Items.f_42412_).m_79707_(26)).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218596_0_) -> {
+         p_218596_0_.m_128359_("Potion", "minecraft:swiftness");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218597_0_) -> {
+         p_218597_0_.m_128359_("Potion", "minecraft:slowness");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218608_0_) -> {
+         p_218608_0_.m_128359_("Potion", "minecraft:strength");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218606_0_) -> {
+         p_218606_0_.m_128359_("Potion", "minecraft:healing");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218603_0_) -> {
+         p_218603_0_.m_128359_("Potion", "minecraft:harming");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218607_0_) -> {
+         p_218607_0_.m_128359_("Potion", "minecraft:leaping");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218601_0_) -> {
+         p_218601_0_.m_128359_("Potion", "minecraft:regeneration");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218600_0_) -> {
+         p_218600_0_.m_128359_("Potion", "minecraft:fire_resistance");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218599_0_) -> {
+         p_218599_0_.m_128359_("Potion", "minecraft:water_breathing");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218598_0_) -> {
+         p_218598_0_.m_128359_("Potion", "minecraft:invisibility");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218605_0_) -> {
+         p_218605_0_.m_128359_("Potion", "minecraft:night_vision");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218602_0_) -> {
+         p_218602_0_.m_128359_("Potion", "minecraft:weakness");
+      })))).m_79076_(LootItem.m_79579_(Items.f_42738_).m_5577_(SetItemCountFunction.m_81008_(RandomValueBounds.m_79291_(0.0F, 1.0F))).m_5577_(SetNbtFunction.m_81187_(Util.m_137469_(new CompoundTag(), (p_218604_0_) -> {
+         p_218604_0_.m_128359_("Potion", "minecraft:poison");
       }))))));
       p_124402_.accept(BuiltInLootTables.f_78732_, LootTable.m_79147_().m_79161_(LootPool.m_79043_().m_79074_(ConstantIntValue.m_78775_(1)).m_79076_(LootItem.m_79579_(Items.f_42454_))));
       p_124402_.accept(BuiltInLootTables.f_78733_, LootTable.m_79147_().m_79161_(LootPool.m_79043_().m_79074_(ConstantIntValue.m_78775_(1)).m_79076_(LootItem.m_79579_(Items.f_42517_))));
