--- a/net/minecraft/server/packs/resources/FallbackResourceManager.java
+++ b/net/minecraft/server/packs/resources/FallbackResourceManager.java
@@ -24,7 +_,7 @@
 
 public class FallbackResourceManager implements ResourceManager {
    private static final Logger f_10600_ = LogManager.getLogger();
-   protected final List<PackResources> f_10599_ = Lists.newArrayList();
+   public final List<PackResources> f_10599_ = Lists.newArrayList();
    private final PackType f_10601_;
    private final String f_10602_;
 
@@ -37,7 +_,6 @@
       this.f_10599_.add(p_10609_);
    }
 
-   @OnlyIn(Dist.CLIENT)
    public Set<String> m_7187_() {
       return ImmutableSet.of(this.f_10602_);
    }
@@ -66,7 +_,6 @@
       throw new FileNotFoundException(p_10614_.toString());
    }
 
-   @OnlyIn(Dist.CLIENT)
    public boolean m_7165_(ResourceLocation p_10620_) {
       if (!this.m_10628_(p_10620_)) {
          return false;
