--- a/net/minecraft/server/packs/resources/SimplePreparableReloadListener.java
+++ b/net/minecraft/server/packs/resources/SimplePreparableReloadListener.java
@@ -8,8 +_,8 @@
    public final CompletableFuture<Void> m_5540_(PreparableReloadListener.PreparationBarrier p_10780_, ResourceManager p_10781_, ProfilerFiller p_10782_, ProfilerFiller p_10783_, Executor p_10784_, Executor p_10785_) {
       return CompletableFuture.supplyAsync(() -> {
          return this.m_5944_(p_10781_, p_10782_);
-      }, p_10784_).thenCompose(p_10780_::m_6769_).thenAcceptAsync((p_10792_) -> {
-         this.m_5787_(p_10792_, p_10781_, p_10783_);
+      }, p_10784_).thenCompose(p_10780_::m_6769_).thenAcceptAsync((p_215269_3_) -> {
+         this.m_5787_(p_215269_3_, p_10781_, p_10783_);
       }, p_10785_);
    }
 
