--- a/net/minecraft/server/packs/resources/SimpleReloadInstance.java
+++ b/net/minecraft/server/packs/resources/SimpleReloadInstance.java
@@ -25,8 +_,8 @@
    private final AtomicInteger f_10806_ = new AtomicInteger();
 
    public static SimpleReloadInstance<Void> m_10815_(ResourceManager p_10816_, List<PreparableReloadListener> p_10817_, Executor p_10818_, Executor p_10819_, CompletableFuture<Unit> p_10820_) {
-      return new SimpleReloadInstance<>(p_10818_, p_10819_, p_10816_, p_10817_, (p_10829_, p_10830_, p_10831_, p_10832_, p_10833_) -> {
-         return p_10831_.m_5540_(p_10829_, p_10830_, InactiveProfiler.f_18554_, InactiveProfiler.f_18554_, p_10818_, p_10833_);
+      return new SimpleReloadInstance<>(p_10818_, p_10819_, p_10816_, p_10817_, (p_219561_1_, p_219561_2_, p_219561_3_, p_219561_4_, p_219561_5_) -> {
+         return p_219561_3_.m_5540_(p_219561_1_, p_219561_2_, InactiveProfiler.f_18554_, InactiveProfiler.f_18554_, p_10818_, p_219561_5_);
       }, p_10820_);
    }
 
@@ -50,20 +_,20 @@
                   }
 
                });
-               return SimpleReloadInstance.this.f_10799_.thenCombine(completablefuture1, (p_10861_, p_10862_) -> {
+               return SimpleReloadInstance.this.f_10799_.thenCombine(completablefuture1, (p_216874_1_, p_216874_2_) -> {
                   return p_10858_;
                });
             }
-         }, p_10810_, preparablereloadlistener, (p_10842_) -> {
+         }, p_10810_, preparablereloadlistener, (p_219564_2_) -> {
             this.f_10805_.incrementAndGet();
             p_10808_.execute(() -> {
-               p_10842_.run();
+               p_219564_2_.run();
                this.f_10806_.incrementAndGet();
             });
-         }, (p_10836_) -> {
+         }, (p_219560_2_) -> {
             ++this.f_10803_;
             p_10809_.execute(() -> {
-               p_10836_.run();
+               p_219560_2_.run();
                ++this.f_10804_;
             });
          });
@@ -75,7 +_,7 @@
    }
 
    public CompletableFuture<Unit> m_7237_() {
-      return this.f_10800_.thenApply((p_10826_) -> {
+      return this.f_10800_.thenApply((p_219558_0_) -> {
          return Unit.INSTANCE;
       });
    }
