--- a/net/minecraft/client/renderer/blockentity/BlockEntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/blockentity/BlockEntityRenderDispatcher.java
@@ -135,4 +_,9 @@
    public Font m_112256_() {
       return this.f_112253_;
    }
+
+   //Internal, Do not call Use ClientRegistry.
+   public synchronized <T extends BlockEntity> void setSpecialRendererInternal(BlockEntityType<T> tileEntityType, BlockEntityRenderer<? super T> specialRenderer) {
+      this.f_112251_.put(tileEntityType, specialRenderer);
+   }
 }
