--- a/net/minecraft/client/renderer/blockentity/BrightnessCombiner.java
+++ b/net/minecraft/client/renderer/blockentity/BrightnessCombiner.java
@@ -11,7 +_,7 @@
 @OnlyIn(Dist.CLIENT)
 public class BrightnessCombiner<S extends BlockEntity> implements DoubleBlockCombiner.Combiner<S, Int2IntFunction> {
    public Int2IntFunction m_6959_(S p_112320_, S p_112321_) {
-      return (p_112325_) -> {
+      return (p_228860_2_) -> {
          int i = LevelRenderer.m_109541_(p_112320_.m_7642_(), p_112320_.m_58899_());
          int j = LevelRenderer.m_109541_(p_112321_.m_7642_(), p_112321_.m_58899_());
          int k = LightTexture.m_109883_(i);
@@ -23,14 +_,14 @@
    }
 
    public Int2IntFunction m_7693_(S p_112318_) {
-      return (p_112333_) -> {
-         return p_112333_;
+      return (p_228861_0_) -> {
+         return p_228861_0_;
       };
    }
 
    public Int2IntFunction m_6502_() {
-      return (p_112316_) -> {
-         return p_112316_;
+      return (p_228859_0_) -> {
+         return p_228859_0_;
       };
    }
 }
