--- a/net/minecraft/client/renderer/blockentity/SkullBlockRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/SkullBlockRenderer.java
@@ -30,24 +_,24 @@
 
 @OnlyIn(Dist.CLIENT)
 public class SkullBlockRenderer extends BlockEntityRenderer<SkullBlockEntity> {
-   private static final Map<SkullBlock.Type, SkullModel> f_112518_ = Util.m_137469_(Maps.newHashMap(), (p_112552_) -> {
+   private static final Map<SkullBlock.Type, SkullModel> f_112518_ = Util.m_137469_(Maps.newHashMap(), (p_209262_0_) -> {
       SkullModel skullmodel = new SkullModel(0, 0, 64, 32);
       SkullModel skullmodel1 = new HumanoidHeadModel();
       DragonHeadModel dragonheadmodel = new DragonHeadModel(0.0F);
-      p_112552_.put(SkullBlock.Types.SKELETON, skullmodel);
-      p_112552_.put(SkullBlock.Types.WITHER_SKELETON, skullmodel);
-      p_112552_.put(SkullBlock.Types.PLAYER, skullmodel1);
-      p_112552_.put(SkullBlock.Types.ZOMBIE, skullmodel1);
-      p_112552_.put(SkullBlock.Types.CREEPER, skullmodel);
-      p_112552_.put(SkullBlock.Types.DRAGON, dragonheadmodel);
+      p_209262_0_.put(SkullBlock.Types.SKELETON, skullmodel);
+      p_209262_0_.put(SkullBlock.Types.WITHER_SKELETON, skullmodel);
+      p_209262_0_.put(SkullBlock.Types.PLAYER, skullmodel1);
+      p_209262_0_.put(SkullBlock.Types.ZOMBIE, skullmodel1);
+      p_209262_0_.put(SkullBlock.Types.CREEPER, skullmodel);
+      p_209262_0_.put(SkullBlock.Types.DRAGON, dragonheadmodel);
    });
-   private static final Map<SkullBlock.Type, ResourceLocation> f_112519_ = Util.m_137469_(Maps.newHashMap(), (p_112550_) -> {
-      p_112550_.put(SkullBlock.Types.SKELETON, new ResourceLocation("textures/entity/skeleton/skeleton.png"));
-      p_112550_.put(SkullBlock.Types.WITHER_SKELETON, new ResourceLocation("textures/entity/skeleton/wither_skeleton.png"));
-      p_112550_.put(SkullBlock.Types.ZOMBIE, new ResourceLocation("textures/entity/zombie/zombie.png"));
-      p_112550_.put(SkullBlock.Types.CREEPER, new ResourceLocation("textures/entity/creeper/creeper.png"));
-      p_112550_.put(SkullBlock.Types.DRAGON, new ResourceLocation("textures/entity/enderdragon/dragon.png"));
-      p_112550_.put(SkullBlock.Types.PLAYER, DefaultPlayerSkin.m_118626_());
+   private static final Map<SkullBlock.Type, ResourceLocation> f_112519_ = Util.m_137469_(Maps.newHashMap(), (p_209263_0_) -> {
+      p_209263_0_.put(SkullBlock.Types.SKELETON, new ResourceLocation("textures/entity/skeleton/skeleton.png"));
+      p_209263_0_.put(SkullBlock.Types.WITHER_SKELETON, new ResourceLocation("textures/entity/skeleton/wither_skeleton.png"));
+      p_209263_0_.put(SkullBlock.Types.ZOMBIE, new ResourceLocation("textures/entity/zombie/zombie.png"));
+      p_209263_0_.put(SkullBlock.Types.CREEPER, new ResourceLocation("textures/entity/creeper/creeper.png"));
+      p_209263_0_.put(SkullBlock.Types.DRAGON, new ResourceLocation("textures/entity/enderdragon/dragon.png"));
+      p_209263_0_.put(SkullBlock.Types.PLAYER, DefaultPlayerSkin.m_118626_());
    });
 
    public SkullBlockRenderer(BlockEntityRenderDispatcher p_112522_) {
