--- a/net/minecraft/client/renderer/blockentity/TheEndPortalRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/TheEndPortalRenderer.java
@@ -20,8 +_,8 @@
    public static final ResourceLocation f_112626_ = new ResourceLocation("textures/environment/end_sky.png");
    public static final ResourceLocation f_112627_ = new ResourceLocation("textures/entity/end_portal.png");
    private static final Random f_112628_ = new Random(31100L);
-   private static final List<RenderType> f_112629_ = IntStream.range(0, 16).mapToObj((p_112635_) -> {
-      return RenderType.m_110410_(p_112635_ + 1);
+   private static final List<RenderType> f_112629_ = IntStream.range(0, 16).mapToObj((p_228882_0_) -> {
+      return RenderType.m_110410_(p_228882_0_ + 1);
    }).collect(ImmutableList.toImmutableList());
 
    public TheEndPortalRenderer(BlockEntityRenderDispatcher p_112632_) {
