--- a/net/minecraft/world/entity/player/ChatVisiblity.java
+++ b/net/minecraft/world/entity/player/ChatVisiblity.java
@@ -11,8 +_,8 @@
    SYSTEM(1, "options.chat.visibility.system"),
    HIDDEN(2, "options.chat.visibility.hidden");
 
-   private static final ChatVisiblity[] f_35955_ = Arrays.stream(values()).sorted(Comparator.comparingInt(ChatVisiblity::m_35965_)).toArray((p_35970_) -> {
-      return new ChatVisiblity[p_35970_];
+   private static final ChatVisiblity[] f_35955_ = Arrays.stream(values()).sorted(Comparator.comparingInt(ChatVisiblity::m_35965_)).toArray((p_221253_0_) -> {
+      return new ChatVisiblity[p_221253_0_];
    });
    private final int f_35956_;
    private final String f_35957_;
