--- a/net/minecraft/world/level/storage/loot/parameters/LootContextParamSets.java
+++ b/net/minecraft/world/level/storage/loot/parameters/LootContextParamSets.java
@@ -8,40 +_,42 @@
 
 public class LootContextParamSets {
    private static final BiMap<ResourceLocation, LootContextParamSet> f_81422_ = HashBiMap.create();
-   public static final LootContextParamSet f_81410_ = m_81428_("empty", (p_81454_) -> {
-   });
-   public static final LootContextParamSet f_81411_ = m_81428_("chest", (p_81452_) -> {
-      p_81452_.m_81406_(LootContextParams.f_81460_).m_81408_(LootContextParams.f_81455_);
-   });
-   public static final LootContextParamSet f_81412_ = m_81428_("command", (p_81450_) -> {
-      p_81450_.m_81406_(LootContextParams.f_81460_).m_81408_(LootContextParams.f_81455_);
-   });
-   public static final LootContextParamSet f_81413_ = m_81428_("selector", (p_81448_) -> {
-      p_81448_.m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81455_);
-   });
-   public static final LootContextParamSet f_81414_ = m_81428_("fishing", (p_81446_) -> {
-      p_81446_.m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81463_).m_81408_(LootContextParams.f_81455_);
-   });
-   public static final LootContextParamSet f_81415_ = m_81428_("entity", (p_81444_) -> {
-      p_81444_.m_81406_(LootContextParams.f_81455_).m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81457_).m_81408_(LootContextParams.f_81458_).m_81408_(LootContextParams.f_81459_).m_81408_(LootContextParams.f_81456_);
-   });
-   public static final LootContextParamSet f_81416_ = m_81428_("gift", (p_81442_) -> {
-      p_81442_.m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81455_);
-   });
-   public static final LootContextParamSet f_81417_ = m_81428_("barter", (p_81440_) -> {
-      p_81440_.m_81406_(LootContextParams.f_81455_);
-   });
-   public static final LootContextParamSet f_81418_ = m_81428_("advancement_reward", (p_81438_) -> {
-      p_81438_.m_81406_(LootContextParams.f_81455_).m_81406_(LootContextParams.f_81460_);
-   });
-   public static final LootContextParamSet f_81419_ = m_81428_("advancement_entity", (p_81436_) -> {
-      p_81436_.m_81406_(LootContextParams.f_81455_).m_81406_(LootContextParams.f_81460_);
-   });
-   public static final LootContextParamSet f_81420_ = m_81428_("generic", (p_81434_) -> {
-      p_81434_.m_81406_(LootContextParams.f_81455_).m_81406_(LootContextParams.f_81456_).m_81406_(LootContextParams.f_81457_).m_81406_(LootContextParams.f_81458_).m_81406_(LootContextParams.f_81459_).m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81461_).m_81406_(LootContextParams.f_81462_).m_81406_(LootContextParams.f_81463_).m_81406_(LootContextParams.f_81464_);
-   });
-   public static final LootContextParamSet f_81421_ = m_81428_("block", (p_81425_) -> {
-      p_81425_.m_81406_(LootContextParams.f_81461_).m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81463_).m_81408_(LootContextParams.f_81455_).m_81408_(LootContextParams.f_81462_).m_81408_(LootContextParams.f_81464_);
+   public static final LootContextParamSet f_81410_ = m_81428_("empty", (p_216249_0_) -> {
+   });
+   public static final LootContextParamSet f_81411_ = m_81428_("chest", (p_216259_0_) -> {
+      p_216259_0_.m_81406_(LootContextParams.f_81460_).m_81408_(LootContextParams.f_81455_);
+      p_216259_0_.m_81408_(LootContextParams.f_81458_); //Forge: Chest Minecarts can have killers.
+   });
+   public static final LootContextParamSet f_81412_ = m_81428_("command", (p_216250_0_) -> {
+      p_216250_0_.m_81406_(LootContextParams.f_81460_).m_81408_(LootContextParams.f_81455_);
+   });
+   public static final LootContextParamSet f_81413_ = m_81428_("selector", (p_216254_0_) -> {
+      p_216254_0_.m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81455_);
+   });
+   public static final LootContextParamSet f_81414_ = m_81428_("fishing", (p_216258_0_) -> {
+      p_216258_0_.m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81463_).m_81408_(LootContextParams.f_81455_);
+      p_216258_0_.m_81408_(LootContextParams.f_81458_).m_81408_(LootContextParams.f_81455_); //Forge: Allow fisher, and bobber
+   });
+   public static final LootContextParamSet f_81415_ = m_81428_("entity", (p_216251_0_) -> {
+      p_216251_0_.m_81406_(LootContextParams.f_81455_).m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81457_).m_81408_(LootContextParams.f_81458_).m_81408_(LootContextParams.f_81459_).m_81408_(LootContextParams.f_81456_);
+   });
+   public static final LootContextParamSet f_81416_ = m_81428_("gift", (p_216255_0_) -> {
+      p_216255_0_.m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81455_);
+   });
+   public static final LootContextParamSet f_81417_ = m_81428_("barter", (p_216252_0_) -> {
+      p_216252_0_.m_81406_(LootContextParams.f_81455_);
+   });
+   public static final LootContextParamSet f_81418_ = m_81428_("advancement_reward", (p_227560_0_) -> {
+      p_227560_0_.m_81406_(LootContextParams.f_81455_).m_81406_(LootContextParams.f_81460_);
+   });
+   public static final LootContextParamSet f_81419_ = m_81428_("advancement_entity", (p_227559_0_) -> {
+      p_227559_0_.m_81406_(LootContextParams.f_81455_).m_81406_(LootContextParams.f_81460_);
+   });
+   public static final LootContextParamSet f_81420_ = m_81428_("generic", (p_237456_0_) -> {
+      p_237456_0_.m_81406_(LootContextParams.f_81455_).m_81406_(LootContextParams.f_81456_).m_81406_(LootContextParams.f_81457_).m_81406_(LootContextParams.f_81458_).m_81406_(LootContextParams.f_81459_).m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81461_).m_81406_(LootContextParams.f_81462_).m_81406_(LootContextParams.f_81463_).m_81406_(LootContextParams.f_81464_);
+   });
+   public static final LootContextParamSet f_81421_ = m_81428_("block", (p_237455_0_) -> {
+      p_237455_0_.m_81406_(LootContextParams.f_81461_).m_81406_(LootContextParams.f_81460_).m_81406_(LootContextParams.f_81463_).m_81408_(LootContextParams.f_81455_).m_81408_(LootContextParams.f_81462_).m_81408_(LootContextParams.f_81464_);
    });
 
    private static LootContextParamSet m_81428_(String p_81429_, Consumer<LootContextParamSet.Builder> p_81430_) {
