--- a/net/minecraft/server/players/OldUsersConverter.java
+++ b/net/minecraft/server/players/OldUsersConverter.java
@@ -48,10 +_,10 @@
    }
 
    private static void m_11086_(MinecraftServer p_11087_, Collection<String> p_11088_, ProfileLookupCallback p_11089_) {
-      String[] astring = p_11088_.stream().filter((p_11077_) -> {
-         return !StringUtil.m_14408_(p_11077_);
-      }).toArray((p_11070_) -> {
-         return new String[p_11070_];
+      String[] astring = p_11088_.stream().filter((p_201150_0_) -> {
+         return !StringUtil.m_14408_(p_201150_0_);
+      }).toArray((p_201149_0_) -> {
+         return new String[p_201149_0_];
       });
       if (p_11087_.m_129797_()) {
          p_11087_.m_129926_().findProfilesByNames(astring, Agent.MINECRAFT, p_11089_);
