--- a/net/minecraft/commands/synchronization/SuggestionProviders.java
+++ b/net/minecraft/commands/synchronization/SuggestionProviders.java
@@ -19,21 +_,21 @@
 public class SuggestionProviders {
    private static final Map<ResourceLocation, SuggestionProvider<SharedSuggestionProvider>> f_121646_ = Maps.newHashMap();
    private static final ResourceLocation f_121647_ = new ResourceLocation("ask_server");
-   public static final SuggestionProvider<SharedSuggestionProvider> f_121641_ = m_121658_(f_121647_, (p_121673_, p_121674_) -> {
-      return p_121673_.getSource().m_5497_(p_121673_, p_121674_);
-   });
-   public static final SuggestionProvider<CommandSourceStack> f_121642_ = m_121658_(new ResourceLocation("all_recipes"), (p_121670_, p_121671_) -> {
-      return SharedSuggestionProvider.m_82957_(p_121670_.getSource().m_6860_(), p_121671_);
-   });
-   public static final SuggestionProvider<CommandSourceStack> f_121643_ = m_121658_(new ResourceLocation("available_sounds"), (p_121667_, p_121668_) -> {
-      return SharedSuggestionProvider.m_82926_(p_121667_.getSource().m_5984_(), p_121668_);
-   });
-   public static final SuggestionProvider<CommandSourceStack> f_121644_ = m_121658_(new ResourceLocation("available_biomes"), (p_121662_, p_121663_) -> {
-      return SharedSuggestionProvider.m_82926_(p_121662_.getSource().m_5894_().m_123090_(Registry.f_122885_).m_6566_(), p_121663_);
-   });
-   public static final SuggestionProvider<CommandSourceStack> f_121645_ = m_121658_(new ResourceLocation("summonable_entities"), (p_121652_, p_121653_) -> {
-      return SharedSuggestionProvider.m_82960_(Registry.f_122826_.m_123024_().filter(EntityType::m_20654_), p_121653_, EntityType::m_20613_, (p_121650_) -> {
-         return new TranslatableComponent(Util.m_137492_("entity", EntityType.m_20613_(p_121650_)));
+   public static final SuggestionProvider<SharedSuggestionProvider> f_121641_ = m_121658_(f_121647_, (p_197500_0_, p_197500_1_) -> {
+      return p_197500_0_.getSource().m_5497_(p_197500_0_, p_197500_1_);
+   });
+   public static final SuggestionProvider<CommandSourceStack> f_121642_ = m_121658_(new ResourceLocation("all_recipes"), (p_197501_0_, p_197501_1_) -> {
+      return SharedSuggestionProvider.m_82957_(p_197501_0_.getSource().m_6860_(), p_197501_1_);
+   });
+   public static final SuggestionProvider<CommandSourceStack> f_121643_ = m_121658_(new ResourceLocation("available_sounds"), (p_197495_0_, p_197495_1_) -> {
+      return SharedSuggestionProvider.m_82926_(p_197495_0_.getSource().m_5984_(), p_197495_1_);
+   });
+   public static final SuggestionProvider<CommandSourceStack> f_121644_ = m_121658_(new ResourceLocation("available_biomes"), (p_239577_0_, p_239577_1_) -> {
+      return SharedSuggestionProvider.m_82926_(p_239577_0_.getSource().m_5894_().m_123090_(Registry.f_122885_).m_6566_(), p_239577_1_);
+   });
+   public static final SuggestionProvider<CommandSourceStack> f_121645_ = m_121658_(new ResourceLocation("summonable_entities"), (p_201210_0_, p_201210_1_) -> {
+      return SharedSuggestionProvider.m_82960_(Registry.f_122826_.m_123024_().filter(EntityType::m_20654_), p_201210_1_, EntityType::m_20613_, (p_201209_0_) -> {
+         return new TranslatableComponent(Util.m_137492_("entity", EntityType.m_20613_(p_201209_0_)));
       });
    });
 
