--- a/net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
+++ b/net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
@@ -21,6 +_,12 @@
       return new GsonAdapterFactory.Builder<>(p_78802_, p_78803_, p_78804_, p_78805_);
    }
 
+   public interface DefaultSerializer<T> {
+      JsonElement m_78826_(T p_78827_, JsonSerializationContext p_78828_);
+
+      T m_78823_(JsonElement p_78824_, JsonDeserializationContext p_78825_);
+   }
+
    public static class Builder<E, T extends SerializerType<E>> {
       private final Registry<T> f_78806_;
       private final String f_78807_;
@@ -41,12 +_,6 @@
       }
    }
 
-   public interface DefaultSerializer<T> {
-      JsonElement m_78826_(T p_78827_, JsonSerializationContext p_78828_);
-
-      T m_78823_(JsonElement p_78824_, JsonDeserializationContext p_78825_);
-   }
-
    static class JsonAdapter<E, T extends SerializerType<E>> implements JsonDeserializer<E>, JsonSerializer<E> {
       private final Registry<T> f_78829_;
       private final String f_78830_;
@@ -83,7 +_,7 @@
       public JsonElement serialize(E p_78852_, Type p_78853_, JsonSerializationContext p_78854_) {
          T t = this.f_78832_.apply(p_78852_);
          if (this.f_78833_ != null && this.f_78833_.getFirst() == t) {
-            return ((GsonAdapterFactory.DefaultSerializer)this.f_78833_.getSecond()).m_78826_(p_78852_, p_78854_);
+            return ((DefaultSerializer<E>)this.f_78833_.getSecond()).m_78826_(p_78852_, p_78854_);
          } else if (t == null) {
             throw new JsonSyntaxException("Unknown type: " + p_78852_);
          } else {
