--- a/net/minecraft/world/phys/shapes/BooleanOp.java
+++ b/net/minecraft/world/phys/shapes/BooleanOp.java
@@ -1,52 +_,52 @@
 package net.minecraft.world.phys.shapes;
 
 public interface BooleanOp {
-   BooleanOp f_82681_ = (p_82747_, p_82748_) -> {
+   BooleanOp f_82681_ = (p_223272_0_, p_223272_1_) -> {
       return false;
    };
-   BooleanOp f_82682_ = (p_82744_, p_82745_) -> {
-      return !p_82744_ && !p_82745_;
-   };
-   BooleanOp f_82683_ = (p_82741_, p_82742_) -> {
-      return p_82742_ && !p_82741_;
-   };
-   BooleanOp f_82684_ = (p_82738_, p_82739_) -> {
-      return !p_82738_;
-   };
-   BooleanOp f_82685_ = (p_82735_, p_82736_) -> {
-      return p_82735_ && !p_82736_;
-   };
-   BooleanOp f_82686_ = (p_82732_, p_82733_) -> {
-      return !p_82733_;
-   };
-   BooleanOp f_82687_ = (p_82729_, p_82730_) -> {
-      return p_82729_ != p_82730_;
-   };
-   BooleanOp f_82688_ = (p_82726_, p_82727_) -> {
-      return !p_82726_ || !p_82727_;
-   };
-   BooleanOp f_82689_ = (p_82723_, p_82724_) -> {
-      return p_82723_ && p_82724_;
-   };
-   BooleanOp f_82690_ = (p_82720_, p_82721_) -> {
-      return p_82720_ == p_82721_;
-   };
-   BooleanOp f_82691_ = (p_82717_, p_82718_) -> {
-      return p_82718_;
-   };
-   BooleanOp f_82692_ = (p_82714_, p_82715_) -> {
-      return !p_82714_ || p_82715_;
-   };
-   BooleanOp f_82693_ = (p_82711_, p_82712_) -> {
-      return p_82711_;
-   };
-   BooleanOp f_82694_ = (p_82708_, p_82709_) -> {
-      return p_82708_ || !p_82709_;
-   };
-   BooleanOp f_82695_ = (p_82705_, p_82706_) -> {
-      return p_82705_ || p_82706_;
-   };
-   BooleanOp f_82696_ = (p_82699_, p_82700_) -> {
+   BooleanOp f_82682_ = (p_223271_0_, p_223271_1_) -> {
+      return !p_223271_0_ && !p_223271_1_;
+   };
+   BooleanOp f_82683_ = (p_223270_0_, p_223270_1_) -> {
+      return p_223270_1_ && !p_223270_0_;
+   };
+   BooleanOp f_82684_ = (p_223269_0_, p_223269_1_) -> {
+      return !p_223269_0_;
+   };
+   BooleanOp f_82685_ = (p_223268_0_, p_223268_1_) -> {
+      return p_223268_0_ && !p_223268_1_;
+   };
+   BooleanOp f_82686_ = (p_223267_0_, p_223267_1_) -> {
+      return !p_223267_1_;
+   };
+   BooleanOp f_82687_ = (p_223266_0_, p_223266_1_) -> {
+      return p_223266_0_ != p_223266_1_;
+   };
+   BooleanOp f_82688_ = (p_223265_0_, p_223265_1_) -> {
+      return !p_223265_0_ || !p_223265_1_;
+   };
+   BooleanOp f_82689_ = (p_223264_0_, p_223264_1_) -> {
+      return p_223264_0_ && p_223264_1_;
+   };
+   BooleanOp f_82690_ = (p_223263_0_, p_223263_1_) -> {
+      return p_223263_0_ == p_223263_1_;
+   };
+   BooleanOp f_82691_ = (p_223262_0_, p_223262_1_) -> {
+      return p_223262_1_;
+   };
+   BooleanOp f_82692_ = (p_223261_0_, p_223261_1_) -> {
+      return !p_223261_0_ || p_223261_1_;
+   };
+   BooleanOp f_82693_ = (p_223260_0_, p_223260_1_) -> {
+      return p_223260_0_;
+   };
+   BooleanOp f_82694_ = (p_223259_0_, p_223259_1_) -> {
+      return p_223259_0_ || !p_223259_1_;
+   };
+   BooleanOp f_82695_ = (p_223258_0_, p_223258_1_) -> {
+      return p_223258_0_ || p_223258_1_;
+   };
+   BooleanOp f_82696_ = (p_223257_0_, p_223257_1_) -> {
       return true;
    };
 
