--- a/net/minecraft/world/entity/animal/Fox.java
+++ b/net/minecraft/world/entity/animal/Fox.java
@@ -51,15 +_,12 @@
 import net.minecraft.world.entity.ai.goal.AvoidEntityGoal;
 import net.minecraft.world.entity.ai.goal.BreedGoal;
 import net.minecraft.world.entity.ai.goal.FleeSunGoal;
-import net.minecraft.world.entity.ai.goal.FloatGoal;
 import net.minecraft.world.entity.ai.goal.FollowParentGoal;
 import net.minecraft.world.entity.ai.goal.Goal;
-import net.minecraft.world.entity.ai.goal.JumpGoal;
 import net.minecraft.world.entity.ai.goal.LeapAtTargetGoal;
 import net.minecraft.world.entity.ai.goal.LookAtPlayerGoal;
 import net.minecraft.world.entity.ai.goal.MeleeAttackGoal;
 import net.minecraft.world.entity.ai.goal.MoveToBlockGoal;
-import net.minecraft.world.entity.ai.goal.PanicGoal;
 import net.minecraft.world.entity.ai.goal.StrollThroughVillageGoal;
 import net.minecraft.world.entity.ai.goal.WaterAvoidingRandomStrollGoal;
 import net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal;
@@ -665,79 +_,6 @@
       return new Vec3(0.0D, (double)(0.55F * this.m_20192_()), (double)(this.m_20205_() * 0.4F));
    }
 
-   class DefendTrustedTargetGoal extends NearestAttackableTargetGoal<LivingEntity> {
-      @Nullable
-      private LivingEntity f_28629_;
-      private LivingEntity f_28630_;
-      private int f_28631_;
-
-      public DefendTrustedTargetGoal(Class<LivingEntity> p_28634_, boolean p_28635_, boolean p_28636_, @Nullable Predicate<LivingEntity> p_28637_) {
-         super(Fox.this, p_28634_, 10, p_28635_, p_28636_, p_28637_);
-      }
-
-      public boolean m_8036_() {
-         if (this.f_26049_ > 0 && this.f_26135_.m_21187_().nextInt(this.f_26049_) != 0) {
-            return false;
-         } else {
-            for(UUID uuid : Fox.this.m_28566_()) {
-               if (uuid != null && Fox.this.f_19853_ instanceof ServerLevel) {
-                  Entity entity = ((ServerLevel)Fox.this.f_19853_).m_8791_(uuid);
-                  if (entity instanceof LivingEntity) {
-                     LivingEntity livingentity = (LivingEntity)entity;
-                     this.f_28630_ = livingentity;
-                     this.f_28629_ = livingentity.m_21188_();
-                     int i = livingentity.m_21213_();
-                     return i != this.f_28631_ && this.m_26150_(this.f_28629_, this.f_26051_);
-                  }
-               }
-            }
-
-            return false;
-         }
-      }
-
-      public void m_8056_() {
-         this.m_26070_(this.f_28629_);
-         this.f_26050_ = this.f_28629_;
-         if (this.f_28630_ != null) {
-            this.f_28631_ = this.f_28630_.m_21213_();
-         }
-
-         Fox.this.m_5496_(SoundEvents.f_11943_, 1.0F, 1.0F);
-         Fox.this.m_28622_(true);
-         Fox.this.m_28568_();
-         super.m_8056_();
-      }
-   }
-
-   class FaceplantGoal extends Goal {
-      int f_28640_;
-
-      public FaceplantGoal() {
-         this.m_7021_(EnumSet.of(Goal.Flag.LOOK, Goal.Flag.JUMP, Goal.Flag.MOVE));
-      }
-
-      public boolean m_8036_() {
-         return Fox.this.m_28556_();
-      }
-
-      public boolean m_8045_() {
-         return this.m_8036_() && this.f_28640_ > 0;
-      }
-
-      public void m_8056_() {
-         this.f_28640_ = 40;
-      }
-
-      public void m_8041_() {
-         Fox.this.m_28618_(false);
-      }
-
-      public void m_8037_() {
-         --this.f_28640_;
-      }
-   }
-
    public class FoxAlertableEntitiesSelector implements Predicate<LivingEntity> {
       public boolean test(LivingEntity p_28653_) {
          if (p_28653_ instanceof Fox) {
@@ -776,52 +_,28 @@
       }
    }
 
-   class FoxBreedGoal extends BreedGoal {
-      public FoxBreedGoal(double p_28668_) {
-         super(Fox.this, p_28668_);
+   class FoxMeleeAttackGoal extends MeleeAttackGoal {
+      public FoxMeleeAttackGoal(double p_28720_, boolean p_28721_) {
+         super(Fox.this, p_28720_, p_28721_);
+      }
+
+      protected void m_6739_(LivingEntity p_28724_, double p_28725_) {
+         double d0 = this.m_6639_(p_28724_);
+         if (p_28725_ <= d0 && this.m_25564_()) {
+            this.m_25563_();
+            this.f_25540_.m_7327_(p_28724_);
+            Fox.this.m_5496_(SoundEvents.f_11945_, 1.0F, 1.0F);
+         }
+
       }
 
       public void m_8056_() {
-         ((Fox)this.f_25113_).m_28569_();
-         ((Fox)this.f_25115_).m_28569_();
+         Fox.this.m_28616_(false);
          super.m_8056_();
       }
 
-      protected void m_8026_() {
-         ServerLevel serverlevel = (ServerLevel)this.f_25114_;
-         Fox fox = (Fox)this.f_25113_.m_7550_(serverlevel, this.f_25115_);
-         if (fox != null) {
-            ServerPlayer serverplayer = this.f_25113_.m_27592_();
-            ServerPlayer serverplayer1 = this.f_25115_.m_27592_();
-            ServerPlayer serverplayer2 = serverplayer;
-            if (serverplayer != null) {
-               fox.m_28515_(serverplayer.m_20148_());
-            } else {
-               serverplayer2 = serverplayer1;
-            }
-
-            if (serverplayer1 != null && serverplayer != serverplayer1) {
-               fox.m_28515_(serverplayer1.m_20148_());
-            }
-
-            if (serverplayer2 != null) {
-               serverplayer2.m_36220_(Stats.f_12937_);
-               CriteriaTriggers.f_10581_.m_18640_(serverplayer2, this.f_25113_, this.f_25115_, fox);
-            }
-
-            this.f_25113_.m_19665_(6000);
-            this.f_25115_.m_19665_(6000);
-            this.f_25113_.m_27594_();
-            this.f_25115_.m_27594_();
-            fox.m_19665_(-24000);
-            fox.m_7678_(this.f_25113_.m_20185_(), this.f_25113_.m_20186_(), this.f_25113_.m_20189_(), 0.0F, 0.0F);
-            serverlevel.m_47205_(fox);
-            this.f_25114_.m_7605_(this.f_25113_, (byte)18);
-            if (this.f_25114_.m_46469_().m_46207_(GameRules.f_46135_)) {
-               this.f_25114_.m_7967_(new ExperienceOrb(this.f_25114_, this.f_25113_.m_20185_(), this.f_25113_.m_20186_(), this.f_25113_.m_20189_(), this.f_25113_.m_21187_().nextInt(7) + 1));
-            }
-
-         }
+      public boolean m_8036_() {
+         return !Fox.this.m_28555_() && !Fox.this.m_5803_() && !Fox.this.m_6047_() && !Fox.this.m_28556_() && super.m_8036_();
       }
    }
 
@@ -860,7 +_,7 @@
       }
 
       protected void m_28686_() {
-         if (Fox.this.f_19853_.m_46469_().m_46207_(GameRules.f_46132_)) {
+         if (net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(Fox.this.f_19853_, Fox.this)) {
             BlockState blockstate = Fox.this.f_19853_.m_8055_(this.f_25602_);
             if (blockstate.m_60713_(Blocks.f_50685_)) {
                int i = blockstate.m_61143_(SweetBerryBushBlock.f_57244_);
@@ -893,18 +_,73 @@
       }
    }
 
-   class FoxFloatGoal extends FloatGoal {
-      public FoxFloatGoal() {
-         super(Fox.this);
-      }
-
-      public void m_8056_() {
-         super.m_8056_();
+   class FoxSearchForItemsGoal extends Goal {
+      public FoxSearchForItemsGoal() {
+         this.m_7021_(EnumSet.of(Goal.Flag.MOVE));
+      }
+
+      public boolean m_8036_() {
+         if (!Fox.this.m_6844_(EquipmentSlot.MAINHAND).m_41619_()) {
+            return false;
+         } else if (Fox.this.m_5448_() == null && Fox.this.m_21188_() == null) {
+            if (!Fox.this.m_28570_()) {
+               return false;
+            } else if (Fox.this.m_21187_().nextInt(10) != 0) {
+               return false;
+            } else {
+               List<ItemEntity> list = Fox.this.f_19853_.m_6443_(ItemEntity.class, Fox.this.m_20191_().m_82377_(8.0D, 8.0D, 8.0D), Fox.f_28441_);
+               return !list.isEmpty() && Fox.this.m_6844_(EquipmentSlot.MAINHAND).m_41619_();
+            }
+         } else {
+            return false;
+         }
+      }
+
+      public void m_8037_() {
+         List<ItemEntity> list = Fox.this.f_19853_.m_6443_(ItemEntity.class, Fox.this.m_20191_().m_82377_(8.0D, 8.0D, 8.0D), Fox.f_28441_);
+         ItemStack itemstack = Fox.this.m_6844_(EquipmentSlot.MAINHAND);
+         if (itemstack.m_41619_() && !list.isEmpty()) {
+            Fox.this.m_21573_().m_5624_(list.get(0), (double)1.2F);
+         }
+
+      }
+
+      public void m_8056_() {
+         List<ItemEntity> list = Fox.this.f_19853_.m_6443_(ItemEntity.class, Fox.this.m_20191_().m_82377_(8.0D, 8.0D, 8.0D), Fox.f_28441_);
+         if (!list.isEmpty()) {
+            Fox.this.m_21573_().m_5624_(list.get(0), (double)1.2F);
+         }
+
+      }
+   }
+
+   class SeekShelterGoal extends FleeSunGoal {
+      private int f_28774_ = 100;
+
+      public SeekShelterGoal(double p_28777_) {
+         super(Fox.this, p_28777_);
+      }
+
+      public boolean m_8036_() {
+         if (!Fox.this.m_5803_() && this.f_25214_.m_5448_() == null) {
+            if (Fox.this.f_19853_.m_46470_()) {
+               return true;
+            } else if (this.f_28774_ > 0) {
+               --this.f_28774_;
+               return false;
+            } else {
+               this.f_28774_ = 100;
+               BlockPos blockpos = this.f_25214_.m_20183_();
+               return Fox.this.f_19853_.m_46461_() && Fox.this.f_19853_.m_45527_(blockpos) && !((ServerLevel)Fox.this.f_19853_).m_8802_(blockpos) && this.m_25226_();
+            }
+         } else {
+            return false;
+         }
+      }
+
+      public void m_8056_() {
          Fox.this.m_28569_();
-      }
-
-      public boolean m_8036_() {
-         return Fox.this.m_20069_() && Fox.this.m_20120_(FluidTags.f_13131_) > 0.25D || Fox.this.m_20077_();
+         super.m_8056_();
       }
    }
 
@@ -930,6 +_,53 @@
       }
    }
 
+   class StalkPreyGoal extends Goal {
+      public StalkPreyGoal() {
+         this.m_7021_(EnumSet.of(Goal.Flag.MOVE, Goal.Flag.LOOK));
+      }
+
+      public boolean m_8036_() {
+         if (Fox.this.m_5803_()) {
+            return false;
+         } else {
+            LivingEntity livingentity = Fox.this.m_5448_();
+            return livingentity != null && livingentity.m_6084_() && Fox.f_28443_.test(livingentity) && Fox.this.m_20280_(livingentity) > 36.0D && !Fox.this.m_6047_() && !Fox.this.m_28559_() && !Fox.this.f_20899_;
+         }
+      }
+
+      public void m_8056_() {
+         Fox.this.m_28610_(false);
+         Fox.this.m_28618_(false);
+      }
+
+      public void m_8041_() {
+         LivingEntity livingentity = Fox.this.m_5448_();
+         if (livingentity != null && Fox.m_28471_(Fox.this, livingentity)) {
+            Fox.this.m_28616_(true);
+            Fox.this.m_28614_(true);
+            Fox.this.m_21573_().m_26573_();
+            Fox.this.m_21563_().m_24960_(livingentity, (float)Fox.this.m_8085_(), (float)Fox.this.m_8132_());
+         } else {
+            Fox.this.m_28616_(false);
+            Fox.this.m_28614_(false);
+         }
+
+      }
+
+      public void m_8037_() {
+         LivingEntity livingentity = Fox.this.m_5448_();
+         Fox.this.m_21563_().m_24960_(livingentity, (float)Fox.this.m_8085_(), (float)Fox.this.m_8132_());
+         if (Fox.this.m_20280_(livingentity) <= 36.0D) {
+            Fox.this.m_28616_(true);
+            Fox.this.m_28614_(true);
+            Fox.this.m_21573_().m_26573_();
+         } else {
+            Fox.this.m_21573_().m_5624_(livingentity, 1.5D);
+         }
+
+      }
+   }
+
    public static class FoxGroupData extends AgableMob.AgableMobGroupData {
       public final Fox.Type f_28701_;
 
@@ -939,17 +_,31 @@
       }
    }
 
-   class FoxLookAtPlayerGoal extends LookAtPlayerGoal {
-      public FoxLookAtPlayerGoal(Mob p_28707_, Class<? extends LivingEntity> p_28708_, float p_28709_) {
-         super(p_28707_, p_28708_, p_28709_);
+   class FaceplantGoal extends Goal {
+      int f_28640_;
+
+      public FaceplantGoal() {
+         this.m_7021_(EnumSet.of(Goal.Flag.LOOK, Goal.Flag.JUMP, Goal.Flag.MOVE));
       }
 
       public boolean m_8036_() {
-         return super.m_8036_() && !Fox.this.m_28556_() && !Fox.this.m_28559_();
+         return Fox.this.m_28556_();
       }
 
       public boolean m_8045_() {
-         return super.m_8045_() && !Fox.this.m_28556_() && !Fox.this.m_28559_();
+         return this.m_8036_() && this.f_28640_ > 0;
+      }
+
+      public void m_8056_() {
+         this.f_28640_ = 40;
+      }
+
+      public void m_8041_() {
+         Fox.this.m_28618_(false);
+      }
+
+      public void m_8037_() {
+         --this.f_28640_;
       }
    }
 
@@ -970,28 +_,63 @@
       }
    }
 
-   class FoxMeleeAttackGoal extends MeleeAttackGoal {
-      public FoxMeleeAttackGoal(double p_28720_, boolean p_28721_) {
-         super(Fox.this, p_28720_, p_28721_);
-      }
-
-      protected void m_6739_(LivingEntity p_28724_, double p_28725_) {
-         double d0 = this.m_6639_(p_28724_);
-         if (p_28725_ <= d0 && this.m_25564_()) {
-            this.m_25563_();
-            this.f_25540_.m_7327_(p_28724_);
-            Fox.this.m_5496_(SoundEvents.f_11945_, 1.0F, 1.0F);
-         }
-
+   class FoxBreedGoal extends BreedGoal {
+      public FoxBreedGoal(double p_28668_) {
+         super(Fox.this, p_28668_);
       }
 
       public void m_8056_() {
-         Fox.this.m_28616_(false);
+         ((Fox)this.f_25113_).m_28569_();
+         ((Fox)this.f_25115_).m_28569_();
          super.m_8056_();
       }
 
-      public boolean m_8036_() {
-         return !Fox.this.m_28555_() && !Fox.this.m_5803_() && !Fox.this.m_6047_() && !Fox.this.m_28556_() && super.m_8036_();
+      protected void m_8026_() {
+         ServerLevel serverlevel = (ServerLevel)this.f_25114_;
+         Fox fox = (Fox)this.f_25113_.m_7550_(serverlevel, this.f_25115_);
+         final net.minecraftforge.event.entity.living.BabyEntitySpawnEvent event = new net.minecraftforge.event.entity.living.BabyEntitySpawnEvent(f_25113_, f_25115_, fox);
+         final boolean cancelled = net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event);
+         fox = (Fox) event.getChild();
+         if (cancelled) {
+            //Reset the "inLove" state for the animals
+            this.f_25113_.m_19665_(6000);
+            this.f_25115_.m_19665_(6000);
+            this.f_25113_.m_27594_();
+            this.f_25115_.m_27594_();
+            return;
+         }
+         if (fox != null) {
+            ServerPlayer serverplayer = this.f_25113_.m_27592_();
+            ServerPlayer serverplayer1 = this.f_25115_.m_27592_();
+            ServerPlayer serverplayer2 = serverplayer;
+            if (serverplayer != null) {
+               fox.m_28515_(serverplayer.m_20148_());
+            } else {
+               serverplayer2 = serverplayer1;
+            }
+
+            if (serverplayer1 != null && serverplayer != serverplayer1) {
+               fox.m_28515_(serverplayer1.m_20148_());
+            }
+
+            if (serverplayer2 != null) {
+               serverplayer2.m_36220_(Stats.f_12937_);
+               CriteriaTriggers.f_10581_.m_18640_(serverplayer2, this.f_25113_, this.f_25115_, fox);
+            }
+
+            this.f_25113_.m_19665_(6000);
+            this.f_25115_.m_19665_(6000);
+            this.f_25113_.m_27594_();
+            this.f_25115_.m_27594_();
+            fox.m_19665_(-24000);
+            fox.m_7678_(this.f_25113_.m_20185_(), this.f_25113_.m_20186_(), this.f_25113_.m_20189_(), 0.0F, 0.0F);
+            serverlevel.m_47205_(fox);
+            this.f_25114_.m_7605_(this.f_25113_, (byte)18);
+            if (this.f_25114_.m_46469_().m_46207_(GameRules.f_46135_)) {
+               this.f_25114_.m_7967_(new ExperienceOrb(this.f_25114_, this.f_25113_.m_20185_(), this.f_25113_.m_20186_(), this.f_25113_.m_20189_(), this.f_25113_.m_21187_().nextInt(7) + 1));
+            }
+
+         }
       }
    }
 
@@ -1008,7 +_,7 @@
       }
    }
 
-   class FoxPanicGoal extends PanicGoal {
+   class FoxPanicGoal extends net.minecraft.world.entity.ai.goal.PanicGoal {
       public FoxPanicGoal(double p_28734_) {
          super(Fox.this, p_28734_);
       }
@@ -1018,7 +_,7 @@
       }
    }
 
-   public class FoxPounceGoal extends JumpGoal {
+   public class FoxPounceGoal extends net.minecraft.world.entity.ai.goal.JumpGoal {
       public boolean m_8036_() {
          if (!Fox.this.m_28558_()) {
             return false;
@@ -1104,66 +_,48 @@
       }
    }
 
-   class FoxSearchForItemsGoal extends Goal {
-      public FoxSearchForItemsGoal() {
-         this.m_7021_(EnumSet.of(Goal.Flag.MOVE));
+   class DefendTrustedTargetGoal extends NearestAttackableTargetGoal<LivingEntity> {
+      @Nullable
+      private LivingEntity f_28629_;
+      private LivingEntity f_28630_;
+      private int f_28631_;
+
+      public DefendTrustedTargetGoal(Class<LivingEntity> p_28634_, boolean p_28635_, boolean p_28636_, @Nullable Predicate<LivingEntity> p_28637_) {
+         super(Fox.this, p_28634_, 10, p_28635_, p_28636_, p_28637_);
       }
 
       public boolean m_8036_() {
-         if (!Fox.this.m_6844_(EquipmentSlot.MAINHAND).m_41619_()) {
+         if (this.f_26049_ > 0 && this.f_26135_.m_21187_().nextInt(this.f_26049_) != 0) {
             return false;
-         } else if (Fox.this.m_5448_() == null && Fox.this.m_21188_() == null) {
-            if (!Fox.this.m_28570_()) {
-               return false;
-            } else if (Fox.this.m_21187_().nextInt(10) != 0) {
-               return false;
-            } else {
-               List<ItemEntity> list = Fox.this.f_19853_.m_6443_(ItemEntity.class, Fox.this.m_20191_().m_82377_(8.0D, 8.0D, 8.0D), Fox.f_28441_);
-               return !list.isEmpty() && Fox.this.m_6844_(EquipmentSlot.MAINHAND).m_41619_();
-            }
          } else {
+            for(UUID uuid : Fox.this.m_28566_()) {
+               if (uuid != null && Fox.this.f_19853_ instanceof ServerLevel) {
+                  Entity entity = ((ServerLevel)Fox.this.f_19853_).m_8791_(uuid);
+                  if (entity instanceof LivingEntity) {
+                     LivingEntity livingentity = (LivingEntity)entity;
+                     this.f_28630_ = livingentity;
+                     this.f_28629_ = livingentity.m_21188_();
+                     int i = livingentity.m_21213_();
+                     return i != this.f_28631_ && this.m_26150_(this.f_28629_, this.f_26051_);
+                  }
+               }
+            }
+
             return false;
          }
       }
 
-      public void m_8037_() {
-         List<ItemEntity> list = Fox.this.f_19853_.m_6443_(ItemEntity.class, Fox.this.m_20191_().m_82377_(8.0D, 8.0D, 8.0D), Fox.f_28441_);
-         ItemStack itemstack = Fox.this.m_6844_(EquipmentSlot.MAINHAND);
-         if (itemstack.m_41619_() && !list.isEmpty()) {
-            Fox.this.m_21573_().m_5624_(list.get(0), (double)1.2F);
-         }
-
-      }
-
-      public void m_8056_() {
-         List<ItemEntity> list = Fox.this.f_19853_.m_6443_(ItemEntity.class, Fox.this.m_20191_().m_82377_(8.0D, 8.0D, 8.0D), Fox.f_28441_);
-         if (!list.isEmpty()) {
-            Fox.this.m_21573_().m_5624_(list.get(0), (double)1.2F);
-         }
-
-      }
-   }
-
-   class FoxStrollThroughVillageGoal extends StrollThroughVillageGoal {
-      public FoxStrollThroughVillageGoal(int p_28754_, int p_28755_) {
-         super(Fox.this, p_28755_);
-      }
-
-      public void m_8056_() {
-         Fox.this.m_28569_();
+      public void m_8056_() {
+         this.m_26070_(this.f_28629_);
+         this.f_26050_ = this.f_28629_;
+         if (this.f_28630_ != null) {
+            this.f_28631_ = this.f_28630_.m_21213_();
+         }
+
+         Fox.this.m_5496_(SoundEvents.f_11943_, 1.0F, 1.0F);
+         Fox.this.m_28622_(true);
+         Fox.this.m_28568_();
          super.m_8056_();
-      }
-
-      public boolean m_8036_() {
-         return super.m_8036_() && this.m_28759_();
-      }
-
-      public boolean m_8045_() {
-         return super.m_8045_() && this.m_28759_();
-      }
-
-      private boolean m_28759_() {
-         return !Fox.this.m_5803_() && !Fox.this.m_28555_() && !Fox.this.m_28567_() && Fox.this.m_5448_() == null;
       }
    }
 
@@ -1214,36 +_,6 @@
       }
    }
 
-   class SeekShelterGoal extends FleeSunGoal {
-      private int f_28774_ = 100;
-
-      public SeekShelterGoal(double p_28777_) {
-         super(Fox.this, p_28777_);
-      }
-
-      public boolean m_8036_() {
-         if (!Fox.this.m_5803_() && this.f_25214_.m_5448_() == null) {
-            if (Fox.this.f_19853_.m_46470_()) {
-               return true;
-            } else if (this.f_28774_ > 0) {
-               --this.f_28774_;
-               return false;
-            } else {
-               this.f_28774_ = 100;
-               BlockPos blockpos = this.f_25214_.m_20183_();
-               return Fox.this.f_19853_.m_46461_() && Fox.this.f_19853_.m_45527_(blockpos) && !((ServerLevel)Fox.this.f_19853_).m_8802_(blockpos) && this.m_25226_();
-            }
-         } else {
-            return false;
-         }
-      }
-
-      public void m_8056_() {
-         Fox.this.m_28569_();
-         super.m_8056_();
-      }
-   }
-
    class SleepGoal extends Fox.FoxBehaviorGoal {
       private int f_28781_ = Fox.this.f_19796_.nextInt(140);
 
@@ -1288,50 +_,41 @@
       }
    }
 
-   class StalkPreyGoal extends Goal {
-      public StalkPreyGoal() {
-         this.m_7021_(EnumSet.of(Goal.Flag.MOVE, Goal.Flag.LOOK));
-      }
-
-      public boolean m_8036_() {
-         if (Fox.this.m_5803_()) {
-            return false;
-         } else {
-            LivingEntity livingentity = Fox.this.m_5448_();
-            return livingentity != null && livingentity.m_6084_() && Fox.f_28443_.test(livingentity) && Fox.this.m_20280_(livingentity) > 36.0D && !Fox.this.m_6047_() && !Fox.this.m_28559_() && !Fox.this.f_20899_;
-         }
-      }
-
-      public void m_8056_() {
-         Fox.this.m_28610_(false);
-         Fox.this.m_28618_(false);
-      }
-
-      public void m_8041_() {
-         LivingEntity livingentity = Fox.this.m_5448_();
-         if (livingentity != null && Fox.m_28471_(Fox.this, livingentity)) {
-            Fox.this.m_28616_(true);
-            Fox.this.m_28614_(true);
-            Fox.this.m_21573_().m_26573_();
-            Fox.this.m_21563_().m_24960_(livingentity, (float)Fox.this.m_8085_(), (float)Fox.this.m_8132_());
-         } else {
-            Fox.this.m_28616_(false);
-            Fox.this.m_28614_(false);
-         }
-
-      }
-
-      public void m_8037_() {
-         LivingEntity livingentity = Fox.this.m_5448_();
-         Fox.this.m_21563_().m_24960_(livingentity, (float)Fox.this.m_8085_(), (float)Fox.this.m_8132_());
-         if (Fox.this.m_20280_(livingentity) <= 36.0D) {
-            Fox.this.m_28616_(true);
-            Fox.this.m_28614_(true);
-            Fox.this.m_21573_().m_26573_();
-         } else {
-            Fox.this.m_21573_().m_5624_(livingentity, 1.5D);
-         }
-
+   class FoxStrollThroughVillageGoal extends StrollThroughVillageGoal {
+      public FoxStrollThroughVillageGoal(int p_28754_, int p_28755_) {
+         super(Fox.this, p_28755_);
+      }
+
+      public void m_8056_() {
+         Fox.this.m_28569_();
+         super.m_8056_();
+      }
+
+      public boolean m_8036_() {
+         return super.m_8036_() && this.m_28759_();
+      }
+
+      public boolean m_8045_() {
+         return super.m_8045_() && this.m_28759_();
+      }
+
+      private boolean m_28759_() {
+         return !Fox.this.m_5803_() && !Fox.this.m_28555_() && !Fox.this.m_28567_() && Fox.this.m_5448_() == null;
+      }
+   }
+
+   class FoxFloatGoal extends net.minecraft.world.entity.ai.goal.FloatGoal {
+      public FoxFloatGoal() {
+         super(Fox.this);
+      }
+
+      public void m_8056_() {
+         super.m_8056_();
+         Fox.this.m_28569_();
+      }
+
+      public boolean m_8036_() {
+         return Fox.this.m_20069_() && Fox.this.m_20120_(FluidTags.f_13131_) > 0.25D || Fox.this.m_20077_();
       }
    }
 
@@ -1377,6 +_,20 @@
 
       public static Fox.Type m_28818_(Optional<ResourceKey<Biome>> p_28819_) {
          return p_28819_.isPresent() && SNOW.f_28802_.contains(p_28819_.get()) ? SNOW : RED;
+      }
+   }
+
+   class FoxLookAtPlayerGoal extends LookAtPlayerGoal {
+      public FoxLookAtPlayerGoal(Mob p_28707_, Class<? extends LivingEntity> p_28708_, float p_28709_) {
+         super(p_28707_, p_28708_, p_28709_);
+      }
+
+      public boolean m_8036_() {
+         return super.m_8036_() && !Fox.this.m_28556_() && !Fox.this.m_28559_();
+      }
+
+      public boolean m_8045_() {
+         return super.m_8045_() && !Fox.this.m_28556_() && !Fox.this.m_28559_();
       }
    }
 }
