--- a/net/minecraft/world/entity/animal/Panda.java
+++ b/net/minecraft/world/entity/animal/Panda.java
@@ -41,7 +_,6 @@
 import net.minecraft.world.entity.ai.goal.Goal;
 import net.minecraft.world.entity.ai.goal.LookAtPlayerGoal;
 import net.minecraft.world.entity.ai.goal.MeleeAttackGoal;
-import net.minecraft.world.entity.ai.goal.PanicGoal;
 import net.minecraft.world.entity.ai.goal.RandomLookAroundGoal;
 import net.minecraft.world.entity.ai.goal.TemptGoal;
 import net.minecraft.world.entity.ai.goal.WaterAvoidingRandomStrollGoal;
@@ -82,9 +_,9 @@
    private float f_29068_;
    private float f_29069_;
    private Panda.PandaLookAtPlayerGoal f_29070_;
-   private static final Predicate<ItemEntity> f_29071_ = (p_29133_) -> {
-      Item item = p_29133_.m_32055_().m_41720_();
-      return (item == Blocks.f_50571_.m_5456_() || item == Blocks.f_50145_.m_5456_()) && p_29133_.m_6084_() && !p_29133_.m_32063_();
+   private static final Predicate<ItemEntity> f_29071_ = (p_213575_0_) -> {
+      Item item = p_213575_0_.m_32055_().m_41720_();
+      return (item == Blocks.f_50571_.m_5456_() || item == Blocks.f_50145_.m_5456_()) && p_213575_0_.m_6084_() && !p_213575_0_.m_32063_();
    };
 
    public Panda(EntityType<? extends Panda> p_29086_, Level p_29087_) {
@@ -647,6 +_,60 @@
       return !this.m_29151_() && !this.m_29165_() && !this.m_29152_() && !this.m_29156_() && !this.m_29150_();
    }
 
+   static class PandaAttackGoal extends MeleeAttackGoal {
+      private final Panda f_29267_;
+
+      public PandaAttackGoal(Panda p_29269_, double p_29270_, boolean p_29271_) {
+         super(p_29269_, p_29270_, p_29271_);
+         this.f_29267_ = p_29269_;
+      }
+
+      public boolean m_8036_() {
+         return this.f_29267_.m_29167_() && super.m_8036_();
+      }
+   }
+
+   static class PandaAvoidGoal<T extends LivingEntity> extends AvoidEntityGoal<T> {
+      private final Panda f_29273_;
+
+      public PandaAvoidGoal(Panda p_29275_, Class<T> p_29276_, float p_29277_, double p_29278_, double p_29279_) {
+         super(p_29275_, p_29276_, p_29277_, p_29278_, p_29279_, EntitySelector.f_20408_::test);
+         this.f_29273_ = p_29275_;
+      }
+
+      public boolean m_8036_() {
+         return this.f_29273_.m_29162_() && this.f_29273_.m_29167_() && super.m_8036_();
+      }
+   }
+
+   static class PandaSneezeGoal extends Goal {
+      private final Panda f_29342_;
+
+      public PandaSneezeGoal(Panda p_29344_) {
+         this.f_29342_ = p_29344_;
+      }
+
+      public boolean m_8036_() {
+         if (this.f_29342_.m_6162_() && this.f_29342_.m_29167_()) {
+            if (this.f_29342_.m_29164_() && this.f_29342_.f_19796_.nextInt(500) == 1) {
+               return true;
+            } else {
+               return this.f_29342_.f_19796_.nextInt(6000) == 1;
+            }
+         } else {
+            return false;
+         }
+      }
+
+      public boolean m_8045_() {
+         return false;
+      }
+
+      public void m_8056_() {
+         this.f_29342_.m_29220_(true);
+      }
+   }
+
    public static enum Gene {
       NORMAL(0, "normal", false),
       LAZY(1, "lazy", false),
@@ -656,8 +_,8 @@
       WEAK(5, "weak", true),
       AGGRESSIVE(6, "aggressive", false);
 
-      private static final Panda.Gene[] f_29235_ = Arrays.stream(values()).sorted(Comparator.comparingInt(Panda.Gene::m_29247_)).toArray((p_29259_) -> {
-         return new Panda.Gene[p_29259_];
+      private static final Panda.Gene[] f_29235_ = Arrays.stream(values()).sorted(Comparator.comparingInt(Panda.Gene::m_29247_)).toArray((p_221102_0_) -> {
+         return new Panda.Gene[p_221102_0_];
       });
       private final int f_29236_;
       private final String f_29237_;
@@ -725,29 +_,34 @@
       }
    }
 
-   static class PandaAttackGoal extends MeleeAttackGoal {
-      private final Panda f_29267_;
-
-      public PandaAttackGoal(Panda p_29269_, double p_29270_, boolean p_29271_) {
-         super(p_29269_, p_29270_, p_29271_);
-         this.f_29267_ = p_29269_;
-      }
-
-      public boolean m_8036_() {
-         return this.f_29267_.m_29167_() && super.m_8036_();
-      }
-   }
-
-   static class PandaAvoidGoal<T extends LivingEntity> extends AvoidEntityGoal<T> {
-      private final Panda f_29273_;
-
-      public PandaAvoidGoal(Panda p_29275_, Class<T> p_29276_, float p_29277_, double p_29278_, double p_29279_) {
-         super(p_29275_, p_29276_, p_29277_, p_29278_, p_29279_, EntitySelector.f_20408_::test);
-         this.f_29273_ = p_29275_;
-      }
-
-      public boolean m_8036_() {
-         return this.f_29273_.m_29162_() && this.f_29273_.m_29167_() && super.m_8036_();
+   static class PandaLieOnBackGoal extends Goal {
+      private final Panda f_29298_;
+      private int f_29299_;
+
+      public PandaLieOnBackGoal(Panda p_29301_) {
+         this.f_29298_ = p_29301_;
+      }
+
+      public boolean m_8036_() {
+         return this.f_29299_ < this.f_29298_.f_19797_ && this.f_29298_.m_29161_() && this.f_29298_.m_29167_() && this.f_29298_.f_19796_.nextInt(400) == 1;
+      }
+
+      public boolean m_8045_() {
+         if (!this.f_29298_.m_20069_() && (this.f_29298_.m_29161_() || this.f_29298_.f_19796_.nextInt(600) != 1)) {
+            return this.f_29298_.f_19796_.nextInt(2000) != 1;
+         } else {
+            return false;
+         }
+      }
+
+      public void m_8056_() {
+         this.f_29298_.m_29212_(true);
+         this.f_29299_ = 0;
+      }
+
+      public void m_8041_() {
+         this.f_29298_.m_29212_(false);
+         this.f_29299_ = this.f_29298_.f_19797_ + 200;
       }
    }
 
@@ -802,102 +_,6 @@
       }
    }
 
-   static class PandaHurtByTargetGoal extends HurtByTargetGoal {
-      private final Panda f_29290_;
-
-      public PandaHurtByTargetGoal(Panda p_29292_, Class<?>... p_29293_) {
-         super(p_29292_, p_29293_);
-         this.f_29290_ = p_29292_;
-      }
-
-      public boolean m_8045_() {
-         if (!this.f_29290_.f_29080_ && !this.f_29290_.f_29081_) {
-            return super.m_8045_();
-         } else {
-            this.f_29290_.m_6710_((LivingEntity)null);
-            return false;
-         }
-      }
-
-      protected void m_5766_(Mob p_29295_, LivingEntity p_29296_) {
-         if (p_29295_ instanceof Panda && ((Panda)p_29295_).m_5912_()) {
-            p_29295_.m_6710_(p_29296_);
-         }
-
-      }
-   }
-
-   static class PandaLieOnBackGoal extends Goal {
-      private final Panda f_29298_;
-      private int f_29299_;
-
-      public PandaLieOnBackGoal(Panda p_29301_) {
-         this.f_29298_ = p_29301_;
-      }
-
-      public boolean m_8036_() {
-         return this.f_29299_ < this.f_29298_.f_19797_ && this.f_29298_.m_29161_() && this.f_29298_.m_29167_() && this.f_29298_.f_19796_.nextInt(400) == 1;
-      }
-
-      public boolean m_8045_() {
-         if (!this.f_29298_.m_20069_() && (this.f_29298_.m_29161_() || this.f_29298_.f_19796_.nextInt(600) != 1)) {
-            return this.f_29298_.f_19796_.nextInt(2000) != 1;
-         } else {
-            return false;
-         }
-      }
-
-      public void m_8056_() {
-         this.f_29298_.m_29212_(true);
-         this.f_29299_ = 0;
-      }
-
-      public void m_8041_() {
-         this.f_29298_.m_29212_(false);
-         this.f_29299_ = this.f_29298_.f_19797_ + 200;
-      }
-   }
-
-   static class PandaLookAtPlayerGoal extends LookAtPlayerGoal {
-      private final Panda f_29306_;
-
-      public PandaLookAtPlayerGoal(Panda p_29308_, Class<? extends LivingEntity> p_29309_, float p_29310_) {
-         super(p_29308_, p_29309_, p_29310_);
-         this.f_29306_ = p_29308_;
-      }
-
-      public void m_29312_(LivingEntity p_29313_) {
-         this.f_25513_ = p_29313_;
-      }
-
-      public boolean m_8045_() {
-         return this.f_25513_ != null && super.m_8045_();
-      }
-
-      public boolean m_8036_() {
-         if (this.f_25512_.m_21187_().nextFloat() >= this.f_25515_) {
-            return false;
-         } else {
-            if (this.f_25513_ == null) {
-               if (this.f_25516_ == Player.class) {
-                  this.f_25513_ = this.f_25512_.f_19853_.m_45949_(this.f_25517_, this.f_25512_, this.f_25512_.m_20185_(), this.f_25512_.m_20188_(), this.f_25512_.m_20189_());
-               } else {
-                  this.f_25513_ = this.f_25512_.f_19853_.m_45989_(this.f_25516_, this.f_25517_, this.f_25512_, this.f_25512_.m_20185_(), this.f_25512_.m_20188_(), this.f_25512_.m_20189_(), this.f_25512_.m_20191_().m_82377_((double)this.f_25514_, 3.0D, (double)this.f_25514_));
-               }
-            }
-
-            return this.f_29306_.m_29167_() && this.f_25513_ != null;
-         }
-      }
-
-      public void m_8037_() {
-         if (this.f_25513_ != null) {
-            super.m_8037_();
-         }
-
-      }
-   }
-
    static class PandaMoveControl extends MoveControl {
       private final Panda f_29316_;
 
@@ -913,7 +_,7 @@
       }
    }
 
-   static class PandaPanicGoal extends PanicGoal {
+   static class PandaPanicGoal extends net.minecraft.world.entity.ai.goal.PanicGoal {
       private final Panda f_29320_;
 
       public PandaPanicGoal(Panda p_29322_, double p_29323_) {
@@ -947,6 +_,31 @@
       }
    }
 
+   static class PandaHurtByTargetGoal extends HurtByTargetGoal {
+      private final Panda f_29290_;
+
+      public PandaHurtByTargetGoal(Panda p_29292_, Class<?>... p_29293_) {
+         super(p_29292_, p_29293_);
+         this.f_29290_ = p_29292_;
+      }
+
+      public boolean m_8045_() {
+         if (!this.f_29290_.f_29080_ && !this.f_29290_.f_29081_) {
+            return super.m_8045_();
+         } else {
+            this.f_29290_.m_6710_((LivingEntity)null);
+            return false;
+         }
+      }
+
+      protected void m_5766_(Mob p_29295_, LivingEntity p_29296_) {
+         if (p_29295_ instanceof Panda && ((Panda)p_29295_).m_5912_()) {
+            p_29295_.m_6710_(p_29296_);
+         }
+
+      }
+   }
+
    static class PandaRollGoal extends Goal {
       private final Panda f_29326_;
 
@@ -1054,31 +_,43 @@
       }
    }
 
-   static class PandaSneezeGoal extends Goal {
-      private final Panda f_29342_;
-
-      public PandaSneezeGoal(Panda p_29344_) {
-         this.f_29342_ = p_29344_;
+   static class PandaLookAtPlayerGoal extends LookAtPlayerGoal {
+      private final Panda f_29306_;
+
+      public PandaLookAtPlayerGoal(Panda p_29308_, Class<? extends LivingEntity> p_29309_, float p_29310_) {
+         super(p_29308_, p_29309_, p_29310_);
+         this.f_29306_ = p_29308_;
+      }
+
+      public void m_29312_(LivingEntity p_29313_) {
+         this.f_25513_ = p_29313_;
+      }
+
+      public boolean m_8045_() {
+         return this.f_25513_ != null && super.m_8045_();
       }
 
       public boolean m_8036_() {
-         if (this.f_29342_.m_6162_() && this.f_29342_.m_29167_()) {
-            if (this.f_29342_.m_29164_() && this.f_29342_.f_19796_.nextInt(500) == 1) {
-               return true;
-            } else {
-               return this.f_29342_.f_19796_.nextInt(6000) == 1;
-            }
-         } else {
+         if (this.f_25512_.m_21187_().nextFloat() >= this.f_25515_) {
             return false;
-         }
-      }
-
-      public boolean m_8045_() {
-         return false;
-      }
-
-      public void m_8056_() {
-         this.f_29342_.m_29220_(true);
+         } else {
+            if (this.f_25513_ == null) {
+               if (this.f_25516_ == Player.class) {
+                  this.f_25513_ = this.f_25512_.f_19853_.m_45949_(this.f_25517_, this.f_25512_, this.f_25512_.m_20185_(), this.f_25512_.m_20188_(), this.f_25512_.m_20189_());
+               } else {
+                  this.f_25513_ = this.f_25512_.f_19853_.m_45989_(this.f_25516_, this.f_25517_, this.f_25512_, this.f_25512_.m_20185_(), this.f_25512_.m_20188_(), this.f_25512_.m_20189_(), this.f_25512_.m_20191_().m_82377_((double)this.f_25514_, 3.0D, (double)this.f_25514_));
+               }
+            }
+
+            return this.f_29306_.m_29167_() && this.f_25513_ != null;
+         }
+      }
+
+      public void m_8037_() {
+         if (this.f_25513_ != null) {
+            super.m_8037_();
+         }
+
       }
    }
 }
