--- a/net/minecraft/world/entity/animal/Parrot.java
+++ b/net/minecraft/world/entity/animal/Parrot.java
@@ -72,40 +_,40 @@
    };
    private static final Item f_29356_ = Items.f_42572_;
    private static final Set<Item> f_29357_ = Sets.newHashSet(Items.f_42404_, Items.f_42578_, Items.f_42577_, Items.f_42733_);
-   private static final Map<EntityType<?>, SoundEvent> f_29358_ = Util.m_137469_(Maps.newHashMap(), (p_29398_) -> {
-      p_29398_.put(EntityType.f_20551_, SoundEvents.f_12246_);
-      p_29398_.put(EntityType.f_20554_, SoundEvents.f_12268_);
-      p_29398_.put(EntityType.f_20558_, SoundEvents.f_12247_);
-      p_29398_.put(EntityType.f_20562_, SoundEvents.f_12248_);
-      p_29398_.put(EntityType.f_20563_, SoundEvents.f_12249_);
-      p_29398_.put(EntityType.f_20565_, SoundEvents.f_12250_);
-      p_29398_.put(EntityType.f_20567_, SoundEvents.f_12251_);
-      p_29398_.put(EntityType.f_20568_, SoundEvents.f_12252_);
-      p_29398_.put(EntityType.f_20453_, SoundEvents.f_12253_);
-      p_29398_.put(EntityType.f_20455_, SoundEvents.f_12254_);
-      p_29398_.put(EntityType.f_20456_, SoundEvents.f_12255_);
-      p_29398_.put(EntityType.f_20458_, SoundEvents.f_12256_);
-      p_29398_.put(EntityType.f_20459_, SoundEvents.f_12257_);
-      p_29398_.put(EntityType.f_20468_, SoundEvents.f_12258_);
-      p_29398_.put(EntityType.f_20509_, SoundEvents.f_12259_);
-      p_29398_.put(EntityType.f_20511_, SoundEvents.f_12260_);
-      p_29398_.put(EntityType.f_20512_, SoundEvents.f_12261_);
-      p_29398_.put(EntityType.f_20513_, SoundEvents.f_12262_);
-      p_29398_.put(EntityType.f_20518_, SoundEvents.f_12263_);
-      p_29398_.put(EntityType.f_20521_, SoundEvents.f_12264_);
-      p_29398_.put(EntityType.f_20523_, SoundEvents.f_12265_);
-      p_29398_.put(EntityType.f_20524_, SoundEvents.f_12266_);
-      p_29398_.put(EntityType.f_20526_, SoundEvents.f_12267_);
-      p_29398_.put(EntityType.f_20479_, SoundEvents.f_12268_);
-      p_29398_.put(EntityType.f_20481_, SoundEvents.f_12269_);
-      p_29398_.put(EntityType.f_20491_, SoundEvents.f_12270_);
-      p_29398_.put(EntityType.f_20493_, SoundEvents.f_12271_);
-      p_29398_.put(EntityType.f_20495_, SoundEvents.f_12220_);
-      p_29398_.put(EntityType.f_20496_, SoundEvents.f_12221_);
-      p_29398_.put(EntityType.f_20497_, SoundEvents.f_12222_);
-      p_29398_.put(EntityType.f_20500_, SoundEvents.f_12223_);
-      p_29398_.put(EntityType.f_20501_, SoundEvents.f_12224_);
-      p_29398_.put(EntityType.f_20530_, SoundEvents.f_12225_);
+   private static final Map<EntityType<?>, SoundEvent> f_29358_ = Util.m_137469_(Maps.newHashMap(), (p_200609_0_) -> {
+      p_200609_0_.put(EntityType.f_20551_, SoundEvents.f_12246_);
+      p_200609_0_.put(EntityType.f_20554_, SoundEvents.f_12268_);
+      p_200609_0_.put(EntityType.f_20558_, SoundEvents.f_12247_);
+      p_200609_0_.put(EntityType.f_20562_, SoundEvents.f_12248_);
+      p_200609_0_.put(EntityType.f_20563_, SoundEvents.f_12249_);
+      p_200609_0_.put(EntityType.f_20565_, SoundEvents.f_12250_);
+      p_200609_0_.put(EntityType.f_20567_, SoundEvents.f_12251_);
+      p_200609_0_.put(EntityType.f_20568_, SoundEvents.f_12252_);
+      p_200609_0_.put(EntityType.f_20453_, SoundEvents.f_12253_);
+      p_200609_0_.put(EntityType.f_20455_, SoundEvents.f_12254_);
+      p_200609_0_.put(EntityType.f_20456_, SoundEvents.f_12255_);
+      p_200609_0_.put(EntityType.f_20458_, SoundEvents.f_12256_);
+      p_200609_0_.put(EntityType.f_20459_, SoundEvents.f_12257_);
+      p_200609_0_.put(EntityType.f_20468_, SoundEvents.f_12258_);
+      p_200609_0_.put(EntityType.f_20509_, SoundEvents.f_12259_);
+      p_200609_0_.put(EntityType.f_20511_, SoundEvents.f_12260_);
+      p_200609_0_.put(EntityType.f_20512_, SoundEvents.f_12261_);
+      p_200609_0_.put(EntityType.f_20513_, SoundEvents.f_12262_);
+      p_200609_0_.put(EntityType.f_20518_, SoundEvents.f_12263_);
+      p_200609_0_.put(EntityType.f_20521_, SoundEvents.f_12264_);
+      p_200609_0_.put(EntityType.f_20523_, SoundEvents.f_12265_);
+      p_200609_0_.put(EntityType.f_20524_, SoundEvents.f_12266_);
+      p_200609_0_.put(EntityType.f_20526_, SoundEvents.f_12267_);
+      p_200609_0_.put(EntityType.f_20479_, SoundEvents.f_12268_);
+      p_200609_0_.put(EntityType.f_20481_, SoundEvents.f_12269_);
+      p_200609_0_.put(EntityType.f_20491_, SoundEvents.f_12270_);
+      p_200609_0_.put(EntityType.f_20493_, SoundEvents.f_12271_);
+      p_200609_0_.put(EntityType.f_20495_, SoundEvents.f_12220_);
+      p_200609_0_.put(EntityType.f_20496_, SoundEvents.f_12221_);
+      p_200609_0_.put(EntityType.f_20497_, SoundEvents.f_12222_);
+      p_200609_0_.put(EntityType.f_20500_, SoundEvents.f_12223_);
+      p_200609_0_.put(EntityType.f_20501_, SoundEvents.f_12224_);
+      p_200609_0_.put(EntityType.f_20530_, SoundEvents.f_12225_);
    });
    public float f_29350_;
    public float f_29351_;
@@ -237,7 +_,7 @@
          }
 
          if (!this.f_19853_.f_46443_) {
-            if (this.f_19796_.nextInt(10) == 0) {
+            if (this.f_19796_.nextInt(10) == 0 && !net.minecraftforge.event.ForgeEventFactory.onAnimalTame(this, p_29414_)) {
                this.m_21828_(p_29414_);
                this.f_19853_.m_7605_(this, (byte)7);
             } else {
