--- a/net/minecraft/world/entity/animal/Pufferfish.java
+++ b/net/minecraft/world/entity/animal/Pufferfish.java
@@ -29,11 +_,11 @@
    private static final EntityDataAccessor<Integer> f_29596_ = SynchedEntityData.m_135353_(Pufferfish.class, EntityDataSerializers.f_135028_);
    private int f_29598_;
    private int f_29599_;
-   private static final Predicate<LivingEntity> f_29597_ = (p_29634_) -> {
-      if (p_29634_ == null) {
+   private static final Predicate<LivingEntity> f_29597_ = (p_210139_0_) -> {
+      if (p_210139_0_ == null) {
          return false;
-      } else if (!(p_29634_ instanceof Player) || !p_29634_.m_5833_() && !((Player)p_29634_).m_7500_()) {
-         return p_29634_.m_6336_() != MobType.f_21644_;
+      } else if (!(p_210139_0_ instanceof Player) || !p_210139_0_.m_5833_() && !((Player)p_210139_0_).m_7500_()) {
+         return p_210139_0_.m_6336_() != MobType.f_21644_;
       } else {
          return false;
       }
