--- a/net/minecraft/world/entity/animal/Rabbit.java
+++ b/net/minecraft/world/entity/animal/Rabbit.java
@@ -28,13 +_,11 @@
 import net.minecraft.world.entity.ai.attributes.Attributes;
 import net.minecraft.world.entity.ai.control.JumpControl;
 import net.minecraft.world.entity.ai.control.MoveControl;
-import net.minecraft.world.entity.ai.goal.AvoidEntityGoal;
 import net.minecraft.world.entity.ai.goal.BreedGoal;
 import net.minecraft.world.entity.ai.goal.FloatGoal;
 import net.minecraft.world.entity.ai.goal.LookAtPlayerGoal;
 import net.minecraft.world.entity.ai.goal.MeleeAttackGoal;
 import net.minecraft.world.entity.ai.goal.MoveToBlockGoal;
-import net.minecraft.world.entity.ai.goal.PanicGoal;
 import net.minecraft.world.entity.ai.goal.TemptGoal;
 import net.minecraft.world.entity.ai.goal.WaterAvoidingRandomStrollGoal;
 import net.minecraft.world.entity.ai.goal.target.HurtByTargetGoal;
@@ -45,7 +_,6 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.item.crafting.Ingredient;
-import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.LevelAccessor;
 import net.minecraft.world.level.LevelReader;
@@ -382,17 +_,7 @@
       return new Vec3(0.0D, (double)(0.6F * this.m_20192_()), (double)(this.m_20205_() * 0.4F));
    }
 
-   static class EvilRabbitAttackGoal extends MeleeAttackGoal {
-      public EvilRabbitAttackGoal(Rabbit p_29738_) {
-         super(p_29738_, 1.4D, true);
-      }
-
-      protected double m_6639_(LivingEntity p_29740_) {
-         return (double)(4.0F + p_29740_.m_20205_());
-      }
-   }
-
-   static class RabbitAvoidEntityGoal<T extends LivingEntity> extends AvoidEntityGoal<T> {
+   static class RabbitAvoidEntityGoal<T extends LivingEntity> extends net.minecraft.world.entity.ai.goal.AvoidEntityGoal<T> {
       private final Rabbit f_29741_;
 
       public RabbitAvoidEntityGoal(Rabbit p_29743_, Class<T> p_29744_, float p_29745_, double p_29746_, double p_29747_) {
@@ -405,12 +_,13 @@
       }
    }
 
-   public static class RabbitGroupData extends AgableMob.AgableMobGroupData {
-      public final int f_29749_;
+   static class EvilRabbitAttackGoal extends MeleeAttackGoal {
+      public EvilRabbitAttackGoal(Rabbit p_29738_) {
+         super(p_29738_, 1.4D, true);
+      }
 
-      public RabbitGroupData(int p_29751_) {
-         super(1.0F);
-         this.f_29749_ = p_29751_;
+      protected double m_6639_(LivingEntity p_29740_) {
+         return (double)(4.0F + p_29740_.m_20205_());
       }
    }
 
@@ -476,7 +_,7 @@
       }
    }
 
-   static class RabbitPanicGoal extends PanicGoal {
+   static class RabbitPanicGoal extends net.minecraft.world.entity.ai.goal.PanicGoal {
       private final Rabbit f_29773_;
 
       public RabbitPanicGoal(Rabbit p_29775_, double p_29776_) {
@@ -490,6 +_,15 @@
       }
    }
 
+   public static class RabbitGroupData extends AgableMob.AgableMobGroupData {
+      public final int f_29749_;
+
+      public RabbitGroupData(int p_29751_) {
+         super(1.0F);
+         this.f_29749_ = p_29751_;
+      }
+   }
+
    static class RaidGardenGoal extends MoveToBlockGoal {
       private final Rabbit f_29778_;
       private boolean f_29779_;
@@ -502,7 +_,7 @@
 
       public boolean m_8036_() {
          if (this.f_25600_ <= 0) {
-            if (!this.f_29778_.f_19853_.m_46469_().m_46207_(GameRules.f_46132_)) {
+            if (!net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(this.f_29778_.f_19853_, this.f_29778_)) {
                return false;
             }
 
