--- a/net/minecraft/world/entity/animal/Wolf.java
+++ b/net/minecraft/world/entity/animal/Wolf.java
@@ -135,6 +_,7 @@
          this.m_30397_(DyeColor.m_41053_(p_30402_.m_128451_("CollarColor")));
       }
 
+      if(!f_19853_.f_46443_) //FORGE: allow this entity to be read from nbt on client. (Fixes MC-189565)
       this.m_21663_((ServerLevel)this.f_19853_, p_30402_);
    }
 
@@ -349,7 +_,7 @@
                itemstack.m_41774_(1);
             }
 
-            if (this.f_19796_.nextInt(3) == 0) {
+            if (this.f_19796_.nextInt(3) == 0 && !net.minecraftforge.event.ForgeEventFactory.onAnimalTame(this, p_30412_)) {
                this.m_21828_(p_30412_);
                this.f_21344_.m_26573_();
                this.m_6710_((LivingEntity)null);
@@ -491,7 +_,7 @@
       return new Vec3(0.0D, (double)(0.6F * this.m_20192_()), (double)(this.m_20205_() * 0.4F));
    }
 
-   class WolfAvoidEntityGoal<T extends LivingEntity> extends AvoidEntityGoal<T> {
+   class WolfAvoidEntityGoal<T extends LivingEntity> extends net.minecraft.world.entity.ai.goal.AvoidEntityGoal<T> {
       private final Wolf f_30451_;
 
       public WolfAvoidEntityGoal(Wolf p_30454_, Class<T> p_30455_, float p_30456_, double p_30457_, double p_30458_) {
