--- a/net/minecraft/commands/CommandFunction.java
+++ b/net/minecraft/commands/CommandFunction.java
@@ -94,8 +_,8 @@
 
       @Nullable
       public ResourceLocation m_77999_() {
-         return this.f_77993_.map((p_78001_) -> {
-            return p_78001_.f_77977_;
+         return this.f_77993_.map((p_218040_0_) -> {
+            return p_218040_0_.f_77977_;
          }).orElse(this.f_77991_);
       }
    }
@@ -116,10 +_,6 @@
       }
    }
 
-   public interface Entry {
-      void m_7184_(ServerFunctionManager p_78013_, CommandSourceStack p_78014_, ArrayDeque<ServerFunctionManager.QueuedCommand> p_78015_, int p_78016_) throws CommandSyntaxException;
-   }
-
    public static class FunctionEntry implements CommandFunction.Entry {
       private final CommandFunction.CacheableFunction f_78017_;
 
@@ -128,8 +_,8 @@
       }
 
       public void m_7184_(ServerFunctionManager p_78027_, CommandSourceStack p_78028_, ArrayDeque<ServerFunctionManager.QueuedCommand> p_78029_, int p_78030_) {
-         this.f_78017_.m_78002_(p_78027_).ifPresent((p_78025_) -> {
-            CommandFunction.Entry[] acommandfunction$entry = p_78025_.m_77989_();
+         this.f_78017_.m_78002_(p_78027_).ifPresent((p_218041_4_) -> {
+            CommandFunction.Entry[] acommandfunction$entry = p_218041_4_.m_77989_();
             int i = p_78030_ - p_78029_.size();
             int j = Math.min(acommandfunction$entry.length, i);
 
@@ -143,5 +_,9 @@
       public String toString() {
          return "function " + this.f_78017_.m_77999_();
       }
+   }
+
+   public interface Entry {
+      void m_7184_(ServerFunctionManager p_78013_, CommandSourceStack p_78014_, ArrayDeque<ServerFunctionManager.QueuedCommand> p_78015_, int p_78016_) throws CommandSyntaxException;
    }
 }
