--- a/net/minecraft/commands/SharedSuggestionProvider.java
+++ b/net/minecraft/commands/SharedSuggestionProvider.java
@@ -82,28 +_,28 @@
 
    static CompletableFuture<Suggestions> m_82929_(Iterable<ResourceLocation> p_82930_, SuggestionsBuilder p_82931_, String p_82932_) {
       String s = p_82931_.getRemaining().toLowerCase(Locale.ROOT);
-      m_82938_(p_82930_, s, p_82932_, (p_82985_) -> {
-         return p_82985_;
-      }, (p_82917_) -> {
-         p_82931_.suggest(p_82932_ + p_82917_);
+      m_82938_(p_82930_, s, p_82932_, (p_210519_0_) -> {
+         return p_210519_0_;
+      }, (p_210518_2_) -> {
+         p_82931_.suggest(p_82932_ + p_210518_2_);
       });
       return p_82931_.buildFuture();
    }
 
    static CompletableFuture<Suggestions> m_82926_(Iterable<ResourceLocation> p_82927_, SuggestionsBuilder p_82928_) {
       String s = p_82928_.getRemaining().toLowerCase(Locale.ROOT);
-      m_82944_(p_82927_, s, (p_82966_) -> {
-         return p_82966_;
-      }, (p_82925_) -> {
-         p_82928_.suggest(p_82925_.toString());
+      m_82944_(p_82927_, s, (p_210517_0_) -> {
+         return p_210517_0_;
+      }, (p_210513_1_) -> {
+         p_82928_.suggest(p_210513_1_.toString());
       });
       return p_82928_.buildFuture();
    }
 
    static <T> CompletableFuture<Suggestions> m_82933_(Iterable<T> p_82934_, SuggestionsBuilder p_82935_, Function<T, ResourceLocation> p_82936_, Function<T, Message> p_82937_) {
       String s = p_82935_.getRemaining().toLowerCase(Locale.ROOT);
-      m_82944_(p_82934_, s, p_82936_, (p_82922_) -> {
-         p_82935_.suggest(p_82936_.apply(p_82922_).toString(), p_82937_.apply(p_82922_));
+      m_82944_(p_82934_, s, p_82936_, (p_210515_3_) -> {
+         p_82935_.suggest(p_82936_.apply(p_210515_3_).toString(), p_82937_.apply(p_210515_3_));
       });
       return p_82935_.buildFuture();
    }
@@ -189,8 +_,8 @@
 
    static CompletableFuture<Suggestions> m_82981_(Stream<String> p_82982_, SuggestionsBuilder p_82983_) {
       String s = p_82983_.getRemaining().toLowerCase(Locale.ROOT);
-      p_82982_.filter((p_82975_) -> {
-         return m_82949_(s, p_82975_.toLowerCase(Locale.ROOT));
+      p_82982_.filter((p_197007_1_) -> {
+         return m_82949_(s, p_197007_1_.toLowerCase(Locale.ROOT));
       }).forEach(p_82983_::suggest);
       return p_82983_.buildFuture();
    }
