--- a/net/minecraft/world/level/levelgen/structure/LegacyStructureDataHandler.java
+++ b/net/minecraft/world/level/levelgen/structure/LegacyStructureDataHandler.java
@@ -21,24 +_,24 @@
 import net.minecraft.world.level.storage.DimensionDataStorage;
 
 public class LegacyStructureDataHandler {
-   private static final Map<String, String> f_71299_ = Util.m_137469_(Maps.newHashMap(), (p_71337_) -> {
-      p_71337_.put("Village", "Village");
-      p_71337_.put("Mineshaft", "Mineshaft");
-      p_71337_.put("Mansion", "Mansion");
-      p_71337_.put("Igloo", "Temple");
-      p_71337_.put("Desert_Pyramid", "Temple");
-      p_71337_.put("Jungle_Pyramid", "Temple");
-      p_71337_.put("Swamp_Hut", "Temple");
-      p_71337_.put("Stronghold", "Stronghold");
-      p_71337_.put("Monument", "Monument");
-      p_71337_.put("Fortress", "Fortress");
-      p_71337_.put("EndCity", "EndCity");
+   private static final Map<String, String> f_71299_ = Util.m_137469_(Maps.newHashMap(), (p_208213_0_) -> {
+      p_208213_0_.put("Village", "Village");
+      p_208213_0_.put("Mineshaft", "Mineshaft");
+      p_208213_0_.put("Mansion", "Mansion");
+      p_208213_0_.put("Igloo", "Temple");
+      p_208213_0_.put("Desert_Pyramid", "Temple");
+      p_208213_0_.put("Jungle_Pyramid", "Temple");
+      p_208213_0_.put("Swamp_Hut", "Temple");
+      p_208213_0_.put("Stronghold", "Stronghold");
+      p_208213_0_.put("Monument", "Monument");
+      p_208213_0_.put("Fortress", "Fortress");
+      p_208213_0_.put("EndCity", "EndCity");
    });
-   private static final Map<String, String> f_71300_ = Util.m_137469_(Maps.newHashMap(), (p_71325_) -> {
-      p_71325_.put("Iglu", "Igloo");
-      p_71325_.put("TeDP", "Desert_Pyramid");
-      p_71325_.put("TeJP", "Jungle_Pyramid");
-      p_71325_.put("TeSH", "Swamp_Hut");
+   private static final Map<String, String> f_71300_ = Util.m_137469_(Maps.newHashMap(), (p_208215_0_) -> {
+      p_208215_0_.put("Iglu", "Igloo");
+      p_208215_0_.put("TeDP", "Desert_Pyramid");
+      p_208215_0_.put("TeJP", "Jungle_Pyramid");
+      p_208215_0_.put("TeSH", "Swamp_Hut");
    });
    private final boolean f_71301_;
    private final Map<String, Long2ObjectMap<CompoundTag>> f_71302_ = Maps.newHashMap();
@@ -176,7 +_,7 @@
                }
 
                String s6 = compoundtag1.m_128461_("id");
-               this.f_71302_.computeIfAbsent(s6, (p_71335_) -> {
+               this.f_71302_.computeIfAbsent(s6, (p_208208_0_) -> {
                   return new Long2ObjectOpenHashMap();
                }).put(i, compoundtag1);
             }
