--- a/net/minecraft/world/level/levelgen/structure/OceanMonumentPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/OceanMonumentPieces.java
@@ -24,97 +_,504 @@
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureManager;
 
 public class OceanMonumentPieces {
-   static class FitDoubleXRoom implements OceanMonumentPieces.MonumentRoomFitter {
-      private FitDoubleXRoom() {
-      }
-
-      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72097_) {
-         return p_72097_.f_72445_[Direction.EAST.m_122411_()] && !p_72097_.f_72444_[Direction.EAST.m_122411_()].f_72446_;
-      }
-
-      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72099_, OceanMonumentPieces.RoomDefinition p_72100_, Random p_72101_) {
-         p_72100_.f_72446_ = true;
-         p_72100_.f_72444_[Direction.EAST.m_122411_()].f_72446_ = true;
-         return new OceanMonumentPieces.OceanMonumentDoubleXRoom(p_72099_, p_72100_);
-      }
-   }
-
-   static class FitDoubleXYRoom implements OceanMonumentPieces.MonumentRoomFitter {
-      private FitDoubleXYRoom() {
-      }
-
-      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72106_) {
-         if (p_72106_.f_72445_[Direction.EAST.m_122411_()] && !p_72106_.f_72444_[Direction.EAST.m_122411_()].f_72446_ && p_72106_.f_72445_[Direction.UP.m_122411_()] && !p_72106_.f_72444_[Direction.UP.m_122411_()].f_72446_) {
-            OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = p_72106_.f_72444_[Direction.EAST.m_122411_()];
-            return oceanmonumentpieces$roomdefinition.f_72445_[Direction.UP.m_122411_()] && !oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()].f_72446_;
-         } else {
-            return false;
-         }
-      }
-
-      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72108_, OceanMonumentPieces.RoomDefinition p_72109_, Random p_72110_) {
-         p_72109_.f_72446_ = true;
-         p_72109_.f_72444_[Direction.EAST.m_122411_()].f_72446_ = true;
-         p_72109_.f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
-         p_72109_.f_72444_[Direction.EAST.m_122411_()].f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
-         return new OceanMonumentPieces.OceanMonumentDoubleXYRoom(p_72108_, p_72109_);
-      }
-   }
-
-   static class FitDoubleYRoom implements OceanMonumentPieces.MonumentRoomFitter {
-      private FitDoubleYRoom() {
-      }
-
-      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72115_) {
-         return p_72115_.f_72445_[Direction.UP.m_122411_()] && !p_72115_.f_72444_[Direction.UP.m_122411_()].f_72446_;
-      }
-
-      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72117_, OceanMonumentPieces.RoomDefinition p_72118_, Random p_72119_) {
-         p_72118_.f_72446_ = true;
-         p_72118_.f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
-         return new OceanMonumentPieces.OceanMonumentDoubleYRoom(p_72117_, p_72118_);
-      }
-   }
-
-   static class FitDoubleYZRoom implements OceanMonumentPieces.MonumentRoomFitter {
-      private FitDoubleYZRoom() {
-      }
-
-      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72124_) {
-         if (p_72124_.f_72445_[Direction.NORTH.m_122411_()] && !p_72124_.f_72444_[Direction.NORTH.m_122411_()].f_72446_ && p_72124_.f_72445_[Direction.UP.m_122411_()] && !p_72124_.f_72444_[Direction.UP.m_122411_()].f_72446_) {
-            OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = p_72124_.f_72444_[Direction.NORTH.m_122411_()];
-            return oceanmonumentpieces$roomdefinition.f_72445_[Direction.UP.m_122411_()] && !oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()].f_72446_;
-         } else {
-            return false;
-         }
-      }
-
-      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72126_, OceanMonumentPieces.RoomDefinition p_72127_, Random p_72128_) {
-         p_72127_.f_72446_ = true;
-         p_72127_.f_72444_[Direction.NORTH.m_122411_()].f_72446_ = true;
-         p_72127_.f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
-         p_72127_.f_72444_[Direction.NORTH.m_122411_()].f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
-         return new OceanMonumentPieces.OceanMonumentDoubleYZRoom(p_72126_, p_72127_);
-      }
-   }
-
-   static class FitDoubleZRoom implements OceanMonumentPieces.MonumentRoomFitter {
-      private FitDoubleZRoom() {
-      }
-
-      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72133_) {
-         return p_72133_.f_72445_[Direction.NORTH.m_122411_()] && !p_72133_.f_72444_[Direction.NORTH.m_122411_()].f_72446_;
-      }
-
-      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72135_, OceanMonumentPieces.RoomDefinition p_72136_, Random p_72137_) {
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = p_72136_;
-         if (!p_72136_.f_72445_[Direction.NORTH.m_122411_()] || p_72136_.f_72444_[Direction.NORTH.m_122411_()].f_72446_) {
-            oceanmonumentpieces$roomdefinition = p_72136_.f_72444_[Direction.SOUTH.m_122411_()];
-         }
-
-         oceanmonumentpieces$roomdefinition.f_72446_ = true;
-         oceanmonumentpieces$roomdefinition.f_72444_[Direction.NORTH.m_122411_()].f_72446_ = true;
-         return new OceanMonumentPieces.OceanMonumentDoubleZRoom(p_72135_, oceanmonumentpieces$roomdefinition);
+   public static class OceanMonumentDoubleXRoom extends OceanMonumentPieces.OceanMonumentPiece {
+      public OceanMonumentDoubleXRoom(Direction p_72229_, OceanMonumentPieces.RoomDefinition p_72230_) {
+         super(StructurePieceType.f_67120_, 1, p_72229_, p_72230_, 2, 1, 1);
+      }
+
+      public OceanMonumentDoubleXRoom(StructureManager p_72226_, CompoundTag p_72227_) {
+         super(StructurePieceType.f_67120_, p_72227_);
+      }
+
+      public boolean m_7832_(WorldGenLevel p_72232_, StructureFeatureManager p_72233_, ChunkGenerator p_72234_, Random p_72235_, BoundingBox p_72236_, ChunkPos p_72237_, BlockPos p_72238_) {
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.EAST.m_122411_()];
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
+         if (this.f_72334_.f_72443_ / 25 > 0) {
+            this.m_72379_(p_72232_, p_72236_, 8, 0, oceanmonumentpieces$roomdefinition.f_72445_[Direction.DOWN.m_122411_()]);
+            this.m_72379_(p_72232_, p_72236_, 0, 0, oceanmonumentpieces$roomdefinition1.f_72445_[Direction.DOWN.m_122411_()]);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72232_, p_72236_, 1, 4, 1, 7, 4, 6, f_72323_);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72232_, p_72236_, 8, 4, 1, 14, 4, 6, f_72323_);
+         }
+
+         this.m_73441_(p_72232_, p_72236_, 0, 3, 0, 0, 3, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 15, 3, 0, 15, 3, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 1, 3, 0, 15, 3, 0, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 1, 3, 7, 14, 3, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 0, 2, 0, 0, 2, 7, f_72323_, f_72323_, false);
+         this.m_73441_(p_72232_, p_72236_, 15, 2, 0, 15, 2, 7, f_72323_, f_72323_, false);
+         this.m_73441_(p_72232_, p_72236_, 1, 2, 0, 15, 2, 0, f_72323_, f_72323_, false);
+         this.m_73441_(p_72232_, p_72236_, 1, 2, 7, 14, 2, 7, f_72323_, f_72323_, false);
+         this.m_73441_(p_72232_, p_72236_, 0, 1, 0, 0, 1, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 15, 1, 0, 15, 1, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 1, 1, 0, 15, 1, 0, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 1, 1, 7, 14, 1, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 5, 1, 0, 10, 1, 4, f_72324_, f_72324_, false);
+         this.m_73441_(p_72232_, p_72236_, 6, 2, 0, 9, 2, 3, f_72323_, f_72323_, false);
+         this.m_73441_(p_72232_, p_72236_, 5, 3, 0, 10, 3, 4, f_72324_, f_72324_, false);
+         this.m_73434_(p_72232_, f_72327_, 6, 2, 3, p_72236_);
+         this.m_73434_(p_72232_, f_72327_, 9, 2, 3, p_72236_);
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72232_, p_72236_, 3, 1, 0, 4, 2, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72232_, p_72236_, 3, 1, 7, 4, 2, 7);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72232_, p_72236_, 0, 1, 3, 0, 2, 4);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72232_, p_72236_, 11, 1, 0, 12, 2, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72232_, p_72236_, 11, 1, 7, 12, 2, 7);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72232_, p_72236_, 15, 1, 3, 15, 2, 4);
+         }
+
+         return true;
+      }
+   }
+
+   public static class OceanMonumentDoubleXYRoom extends OceanMonumentPieces.OceanMonumentPiece {
+      public OceanMonumentDoubleXYRoom(Direction p_72243_, OceanMonumentPieces.RoomDefinition p_72244_) {
+         super(StructurePieceType.f_67121_, 1, p_72243_, p_72244_, 2, 2, 1);
+      }
+
+      public OceanMonumentDoubleXYRoom(StructureManager p_72240_, CompoundTag p_72241_) {
+         super(StructurePieceType.f_67121_, p_72241_);
+      }
+
+      public boolean m_7832_(WorldGenLevel p_72246_, StructureFeatureManager p_72247_, ChunkGenerator p_72248_, Random p_72249_, BoundingBox p_72250_, ChunkPos p_72251_, BlockPos p_72252_) {
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.EAST.m_122411_()];
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition2 = oceanmonumentpieces$roomdefinition1.f_72444_[Direction.UP.m_122411_()];
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition3 = oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()];
+         if (this.f_72334_.f_72443_ / 25 > 0) {
+            this.m_72379_(p_72246_, p_72250_, 8, 0, oceanmonumentpieces$roomdefinition.f_72445_[Direction.DOWN.m_122411_()]);
+            this.m_72379_(p_72246_, p_72250_, 0, 0, oceanmonumentpieces$roomdefinition1.f_72445_[Direction.DOWN.m_122411_()]);
+         }
+
+         if (oceanmonumentpieces$roomdefinition2.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72246_, p_72250_, 1, 8, 1, 7, 8, 6, f_72323_);
+         }
+
+         if (oceanmonumentpieces$roomdefinition3.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72246_, p_72250_, 8, 8, 1, 14, 8, 6, f_72323_);
+         }
+
+         for(int i = 1; i <= 7; ++i) {
+            BlockState blockstate = f_72324_;
+            if (i == 2 || i == 6) {
+               blockstate = f_72323_;
+            }
+
+            this.m_73441_(p_72246_, p_72250_, 0, i, 0, 0, i, 7, blockstate, blockstate, false);
+            this.m_73441_(p_72246_, p_72250_, 15, i, 0, 15, i, 7, blockstate, blockstate, false);
+            this.m_73441_(p_72246_, p_72250_, 1, i, 0, 15, i, 0, blockstate, blockstate, false);
+            this.m_73441_(p_72246_, p_72250_, 1, i, 7, 14, i, 7, blockstate, blockstate, false);
+         }
+
+         this.m_73441_(p_72246_, p_72250_, 2, 1, 3, 2, 7, 4, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 3, 1, 2, 4, 7, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 3, 1, 5, 4, 7, 5, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 13, 1, 3, 13, 7, 4, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 11, 1, 2, 12, 7, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 11, 1, 5, 12, 7, 5, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 5, 1, 3, 5, 3, 4, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 10, 1, 3, 10, 3, 4, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 5, 7, 2, 10, 7, 5, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 5, 5, 2, 5, 7, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 10, 5, 2, 10, 7, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 5, 5, 5, 5, 7, 5, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 10, 5, 5, 10, 7, 5, f_72324_, f_72324_, false);
+         this.m_73434_(p_72246_, f_72324_, 6, 6, 2, p_72250_);
+         this.m_73434_(p_72246_, f_72324_, 9, 6, 2, p_72250_);
+         this.m_73434_(p_72246_, f_72324_, 6, 6, 5, p_72250_);
+         this.m_73434_(p_72246_, f_72324_, 9, 6, 5, p_72250_);
+         this.m_73441_(p_72246_, p_72250_, 5, 4, 3, 6, 4, 4, f_72324_, f_72324_, false);
+         this.m_73441_(p_72246_, p_72250_, 9, 4, 3, 10, 4, 4, f_72324_, f_72324_, false);
+         this.m_73434_(p_72246_, f_72327_, 5, 4, 2, p_72250_);
+         this.m_73434_(p_72246_, f_72327_, 5, 4, 5, p_72250_);
+         this.m_73434_(p_72246_, f_72327_, 10, 4, 2, p_72250_);
+         this.m_73434_(p_72246_, f_72327_, 10, 4, 5, p_72250_);
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 3, 1, 0, 4, 2, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 3, 1, 7, 4, 2, 7);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 0, 1, 3, 0, 2, 4);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 11, 1, 0, 12, 2, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 11, 1, 7, 12, 2, 7);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 15, 1, 3, 15, 2, 4);
+         }
+
+         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 3, 5, 0, 4, 6, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 3, 5, 7, 4, 6, 7);
+         }
+
+         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 0, 5, 3, 0, 6, 4);
+         }
+
+         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 11, 5, 0, 12, 6, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 11, 5, 7, 12, 6, 7);
+         }
+
+         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72246_, p_72250_, 15, 5, 3, 15, 6, 4);
+         }
+
+         return true;
+      }
+   }
+
+   public static class OceanMonumentDoubleYRoom extends OceanMonumentPieces.OceanMonumentPiece {
+      public OceanMonumentDoubleYRoom(Direction p_72257_, OceanMonumentPieces.RoomDefinition p_72258_) {
+         super(StructurePieceType.f_67122_, 1, p_72257_, p_72258_, 1, 2, 1);
+      }
+
+      public OceanMonumentDoubleYRoom(StructureManager p_72254_, CompoundTag p_72255_) {
+         super(StructurePieceType.f_67122_, p_72255_);
+      }
+
+      public boolean m_7832_(WorldGenLevel p_72260_, StructureFeatureManager p_72261_, ChunkGenerator p_72262_, Random p_72263_, BoundingBox p_72264_, ChunkPos p_72265_, BlockPos p_72266_) {
+         if (this.f_72334_.f_72443_ / 25 > 0) {
+            this.m_72379_(p_72260_, p_72264_, 0, 0, this.f_72334_.f_72445_[Direction.DOWN.m_122411_()]);
+         }
+
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.UP.m_122411_()];
+         if (oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72260_, p_72264_, 1, 8, 1, 6, 8, 6, f_72323_);
+         }
+
+         this.m_73441_(p_72260_, p_72264_, 0, 4, 0, 0, 4, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 7, 4, 0, 7, 4, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 1, 4, 0, 6, 4, 0, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 1, 4, 7, 6, 4, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 2, 4, 1, 2, 4, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 1, 4, 2, 1, 4, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 5, 4, 1, 5, 4, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 6, 4, 2, 6, 4, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 2, 4, 5, 2, 4, 6, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 1, 4, 5, 1, 4, 5, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 5, 4, 5, 5, 4, 6, f_72324_, f_72324_, false);
+         this.m_73441_(p_72260_, p_72264_, 6, 4, 5, 6, 4, 5, f_72324_, f_72324_, false);
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
+
+         for(int i = 1; i <= 5; i += 4) {
+            int j = 0;
+            if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
+               this.m_73441_(p_72260_, p_72264_, 2, i, j, 2, i + 2, j, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, 5, i, j, 5, i + 2, j, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, 3, i + 2, j, 4, i + 2, j, f_72324_, f_72324_, false);
+            } else {
+               this.m_73441_(p_72260_, p_72264_, 0, i, j, 7, i + 2, j, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, 0, i + 1, j, 7, i + 1, j, f_72323_, f_72323_, false);
+            }
+
+            j = 7;
+            if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.NORTH.m_122411_()]) {
+               this.m_73441_(p_72260_, p_72264_, 2, i, j, 2, i + 2, j, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, 5, i, j, 5, i + 2, j, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, 3, i + 2, j, 4, i + 2, j, f_72324_, f_72324_, false);
+            } else {
+               this.m_73441_(p_72260_, p_72264_, 0, i, j, 7, i + 2, j, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, 0, i + 1, j, 7, i + 1, j, f_72323_, f_72323_, false);
+            }
+
+            int k = 0;
+            if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
+               this.m_73441_(p_72260_, p_72264_, k, i, 2, k, i + 2, 2, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, k, i, 5, k, i + 2, 5, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, k, i + 2, 3, k, i + 2, 4, f_72324_, f_72324_, false);
+            } else {
+               this.m_73441_(p_72260_, p_72264_, k, i, 0, k, i + 2, 7, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, k, i + 1, 0, k, i + 1, 7, f_72323_, f_72323_, false);
+            }
+
+            k = 7;
+            if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.EAST.m_122411_()]) {
+               this.m_73441_(p_72260_, p_72264_, k, i, 2, k, i + 2, 2, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, k, i, 5, k, i + 2, 5, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, k, i + 2, 3, k, i + 2, 4, f_72324_, f_72324_, false);
+            } else {
+               this.m_73441_(p_72260_, p_72264_, k, i, 0, k, i + 2, 7, f_72324_, f_72324_, false);
+               this.m_73441_(p_72260_, p_72264_, k, i + 1, 0, k, i + 1, 7, f_72323_, f_72323_, false);
+            }
+
+            oceanmonumentpieces$roomdefinition1 = oceanmonumentpieces$roomdefinition;
+         }
+
+         return true;
+      }
+   }
+
+   public static class OceanMonumentDoubleYZRoom extends OceanMonumentPieces.OceanMonumentPiece {
+      public OceanMonumentDoubleYZRoom(Direction p_72271_, OceanMonumentPieces.RoomDefinition p_72272_) {
+         super(StructurePieceType.f_67123_, 1, p_72271_, p_72272_, 1, 2, 2);
+      }
+
+      public OceanMonumentDoubleYZRoom(StructureManager p_72268_, CompoundTag p_72269_) {
+         super(StructurePieceType.f_67123_, p_72269_);
+      }
+
+      public boolean m_7832_(WorldGenLevel p_72274_, StructureFeatureManager p_72275_, ChunkGenerator p_72276_, Random p_72277_, BoundingBox p_72278_, ChunkPos p_72279_, BlockPos p_72280_) {
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.NORTH.m_122411_()];
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition2 = oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()];
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition3 = oceanmonumentpieces$roomdefinition1.f_72444_[Direction.UP.m_122411_()];
+         if (this.f_72334_.f_72443_ / 25 > 0) {
+            this.m_72379_(p_72274_, p_72278_, 0, 8, oceanmonumentpieces$roomdefinition.f_72445_[Direction.DOWN.m_122411_()]);
+            this.m_72379_(p_72274_, p_72278_, 0, 0, oceanmonumentpieces$roomdefinition1.f_72445_[Direction.DOWN.m_122411_()]);
+         }
+
+         if (oceanmonumentpieces$roomdefinition3.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72274_, p_72278_, 1, 8, 1, 6, 8, 7, f_72323_);
+         }
+
+         if (oceanmonumentpieces$roomdefinition2.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72274_, p_72278_, 1, 8, 8, 6, 8, 14, f_72323_);
+         }
+
+         for(int i = 1; i <= 7; ++i) {
+            BlockState blockstate = f_72324_;
+            if (i == 2 || i == 6) {
+               blockstate = f_72323_;
+            }
+
+            this.m_73441_(p_72274_, p_72278_, 0, i, 0, 0, i, 15, blockstate, blockstate, false);
+            this.m_73441_(p_72274_, p_72278_, 7, i, 0, 7, i, 15, blockstate, blockstate, false);
+            this.m_73441_(p_72274_, p_72278_, 1, i, 0, 6, i, 0, blockstate, blockstate, false);
+            this.m_73441_(p_72274_, p_72278_, 1, i, 15, 6, i, 15, blockstate, blockstate, false);
+         }
+
+         for(int j = 1; j <= 7; ++j) {
+            BlockState blockstate1 = f_72325_;
+            if (j == 2 || j == 6) {
+               blockstate1 = f_72327_;
+            }
+
+            this.m_73441_(p_72274_, p_72278_, 3, j, 7, 4, j, 8, blockstate1, blockstate1, false);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 3, 1, 0, 4, 2, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 7, 1, 3, 7, 2, 4);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 0, 1, 3, 0, 2, 4);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 3, 1, 15, 4, 2, 15);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 0, 1, 11, 0, 2, 12);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 7, 1, 11, 7, 2, 12);
+         }
+
+         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 3, 5, 0, 4, 6, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 7, 5, 3, 7, 6, 4);
+            this.m_73441_(p_72274_, p_72278_, 5, 4, 2, 6, 4, 5, f_72324_, f_72324_, false);
+            this.m_73441_(p_72274_, p_72278_, 6, 1, 2, 6, 3, 2, f_72324_, f_72324_, false);
+            this.m_73441_(p_72274_, p_72278_, 6, 1, 5, 6, 3, 5, f_72324_, f_72324_, false);
+         }
+
+         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 0, 5, 3, 0, 6, 4);
+            this.m_73441_(p_72274_, p_72278_, 1, 4, 2, 2, 4, 5, f_72324_, f_72324_, false);
+            this.m_73441_(p_72274_, p_72278_, 1, 1, 2, 1, 3, 2, f_72324_, f_72324_, false);
+            this.m_73441_(p_72274_, p_72278_, 1, 1, 5, 1, 3, 5, f_72324_, f_72324_, false);
+         }
+
+         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 3, 5, 15, 4, 6, 15);
+         }
+
+         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 0, 5, 11, 0, 6, 12);
+            this.m_73441_(p_72274_, p_72278_, 1, 4, 10, 2, 4, 13, f_72324_, f_72324_, false);
+            this.m_73441_(p_72274_, p_72278_, 1, 1, 10, 1, 3, 10, f_72324_, f_72324_, false);
+            this.m_73441_(p_72274_, p_72278_, 1, 1, 13, 1, 3, 13, f_72324_, f_72324_, false);
+         }
+
+         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72274_, p_72278_, 7, 5, 11, 7, 6, 12);
+            this.m_73441_(p_72274_, p_72278_, 5, 4, 10, 6, 4, 13, f_72324_, f_72324_, false);
+            this.m_73441_(p_72274_, p_72278_, 6, 1, 10, 6, 3, 10, f_72324_, f_72324_, false);
+            this.m_73441_(p_72274_, p_72278_, 6, 1, 13, 6, 3, 13, f_72324_, f_72324_, false);
+         }
+
+         return true;
+      }
+   }
+
+   public static class OceanMonumentDoubleZRoom extends OceanMonumentPieces.OceanMonumentPiece {
+      public OceanMonumentDoubleZRoom(Direction p_72285_, OceanMonumentPieces.RoomDefinition p_72286_) {
+         super(StructurePieceType.f_67124_, 1, p_72285_, p_72286_, 1, 1, 2);
+      }
+
+      public OceanMonumentDoubleZRoom(StructureManager p_72282_, CompoundTag p_72283_) {
+         super(StructurePieceType.f_67124_, p_72283_);
+      }
+
+      public boolean m_7832_(WorldGenLevel p_72288_, StructureFeatureManager p_72289_, ChunkGenerator p_72290_, Random p_72291_, BoundingBox p_72292_, ChunkPos p_72293_, BlockPos p_72294_) {
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.NORTH.m_122411_()];
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
+         if (this.f_72334_.f_72443_ / 25 > 0) {
+            this.m_72379_(p_72288_, p_72292_, 0, 8, oceanmonumentpieces$roomdefinition.f_72445_[Direction.DOWN.m_122411_()]);
+            this.m_72379_(p_72288_, p_72292_, 0, 0, oceanmonumentpieces$roomdefinition1.f_72445_[Direction.DOWN.m_122411_()]);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72288_, p_72292_, 1, 4, 1, 6, 4, 7, f_72323_);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()] == null) {
+            this.m_72369_(p_72288_, p_72292_, 1, 4, 8, 6, 4, 14, f_72323_);
+         }
+
+         this.m_73441_(p_72288_, p_72292_, 0, 3, 0, 0, 3, 15, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 7, 3, 0, 7, 3, 15, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 3, 0, 7, 3, 0, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 3, 15, 6, 3, 15, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 0, 2, 0, 0, 2, 15, f_72323_, f_72323_, false);
+         this.m_73441_(p_72288_, p_72292_, 7, 2, 0, 7, 2, 15, f_72323_, f_72323_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 2, 0, 7, 2, 0, f_72323_, f_72323_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 2, 15, 6, 2, 15, f_72323_, f_72323_, false);
+         this.m_73441_(p_72288_, p_72292_, 0, 1, 0, 0, 1, 15, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 7, 1, 0, 7, 1, 15, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 1, 0, 7, 1, 0, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 1, 15, 6, 1, 15, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 1, 1, 1, 1, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 6, 1, 1, 6, 1, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 3, 1, 1, 3, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 6, 3, 1, 6, 3, 2, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 1, 13, 1, 1, 14, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 6, 1, 13, 6, 1, 14, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 1, 3, 13, 1, 3, 14, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 6, 3, 13, 6, 3, 14, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 2, 1, 6, 2, 3, 6, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 5, 1, 6, 5, 3, 6, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 2, 1, 9, 2, 3, 9, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 5, 1, 9, 5, 3, 9, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 3, 2, 6, 4, 2, 6, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 3, 2, 9, 4, 2, 9, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 2, 2, 7, 2, 2, 8, f_72324_, f_72324_, false);
+         this.m_73441_(p_72288_, p_72292_, 5, 2, 7, 5, 2, 8, f_72324_, f_72324_, false);
+         this.m_73434_(p_72288_, f_72327_, 2, 2, 5, p_72292_);
+         this.m_73434_(p_72288_, f_72327_, 5, 2, 5, p_72292_);
+         this.m_73434_(p_72288_, f_72327_, 2, 2, 10, p_72292_);
+         this.m_73434_(p_72288_, f_72327_, 5, 2, 10, p_72292_);
+         this.m_73434_(p_72288_, f_72324_, 2, 3, 5, p_72292_);
+         this.m_73434_(p_72288_, f_72324_, 5, 3, 5, p_72292_);
+         this.m_73434_(p_72288_, f_72324_, 2, 3, 10, p_72292_);
+         this.m_73434_(p_72288_, f_72324_, 5, 3, 10, p_72292_);
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
+            this.m_72360_(p_72288_, p_72292_, 3, 1, 0, 4, 2, 0);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72288_, p_72292_, 7, 1, 3, 7, 2, 4);
+         }
+
+         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72288_, p_72292_, 0, 1, 3, 0, 2, 4);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72288_, p_72292_, 3, 1, 15, 4, 2, 15);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72288_, p_72292_, 0, 1, 11, 0, 2, 12);
+         }
+
+         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72288_, p_72292_, 7, 1, 11, 7, 2, 12);
+         }
+
+         return true;
+      }
+   }
+
+   public static class OceanMonumentEntryRoom extends OceanMonumentPieces.OceanMonumentPiece {
+      public OceanMonumentEntryRoom(Direction p_72299_, OceanMonumentPieces.RoomDefinition p_72300_) {
+         super(StructurePieceType.f_67125_, 1, p_72299_, p_72300_, 1, 1, 1);
+      }
+
+      public OceanMonumentEntryRoom(StructureManager p_72296_, CompoundTag p_72297_) {
+         super(StructurePieceType.f_67125_, p_72297_);
+      }
+
+      public boolean m_7832_(WorldGenLevel p_72302_, StructureFeatureManager p_72303_, ChunkGenerator p_72304_, Random p_72305_, BoundingBox p_72306_, ChunkPos p_72307_, BlockPos p_72308_) {
+         this.m_73441_(p_72302_, p_72306_, 0, 3, 0, 2, 3, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72302_, p_72306_, 5, 3, 0, 7, 3, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72302_, p_72306_, 0, 2, 0, 1, 2, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72302_, p_72306_, 6, 2, 0, 7, 2, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72302_, p_72306_, 0, 1, 0, 0, 1, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72302_, p_72306_, 7, 1, 0, 7, 1, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72302_, p_72306_, 0, 1, 7, 7, 3, 7, f_72324_, f_72324_, false);
+         this.m_73441_(p_72302_, p_72306_, 1, 1, 0, 2, 3, 0, f_72324_, f_72324_, false);
+         this.m_73441_(p_72302_, p_72306_, 5, 1, 0, 6, 3, 0, f_72324_, f_72324_, false);
+         if (this.f_72334_.f_72445_[Direction.NORTH.m_122411_()]) {
+            this.m_72360_(p_72302_, p_72306_, 3, 1, 7, 4, 2, 7);
+         }
+
+         if (this.f_72334_.f_72445_[Direction.WEST.m_122411_()]) {
+            this.m_72360_(p_72302_, p_72306_, 0, 1, 3, 1, 2, 4);
+         }
+
+         if (this.f_72334_.f_72445_[Direction.EAST.m_122411_()]) {
+            this.m_72360_(p_72302_, p_72306_, 6, 1, 3, 7, 2, 4);
+         }
+
+         return true;
       }
    }
 
@@ -146,6 +_,12 @@
       }
    }
 
+   interface MonumentRoomFitter {
+      boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72207_);
+
+      OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72208_, OceanMonumentPieces.RoomDefinition p_72209_, Random p_72210_);
+   }
+
    public static class MonumentBuilding extends OceanMonumentPieces.OceanMonumentPiece {
       private OceanMonumentPieces.RoomDefinition f_72156_;
       private OceanMonumentPieces.RoomDefinition f_72157_;
@@ -697,12 +_,6 @@
       }
    }
 
-   interface MonumentRoomFitter {
-      boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72207_);
-
-      OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72208_, OceanMonumentPieces.RoomDefinition p_72209_, Random p_72210_);
-   }
-
    public static class OceanMonumentCoreRoom extends OceanMonumentPieces.OceanMonumentPiece {
       public OceanMonumentCoreRoom(Direction p_72215_, OceanMonumentPieces.RoomDefinition p_72216_) {
          super(StructurePieceType.f_67119_, 1, p_72215_, p_72216_, 2, 2, 2);
@@ -781,507 +_,6 @@
       }
    }
 
-   public static class OceanMonumentDoubleXRoom extends OceanMonumentPieces.OceanMonumentPiece {
-      public OceanMonumentDoubleXRoom(Direction p_72229_, OceanMonumentPieces.RoomDefinition p_72230_) {
-         super(StructurePieceType.f_67120_, 1, p_72229_, p_72230_, 2, 1, 1);
-      }
-
-      public OceanMonumentDoubleXRoom(StructureManager p_72226_, CompoundTag p_72227_) {
-         super(StructurePieceType.f_67120_, p_72227_);
-      }
-
-      public boolean m_7832_(WorldGenLevel p_72232_, StructureFeatureManager p_72233_, ChunkGenerator p_72234_, Random p_72235_, BoundingBox p_72236_, ChunkPos p_72237_, BlockPos p_72238_) {
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.EAST.m_122411_()];
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
-         if (this.f_72334_.f_72443_ / 25 > 0) {
-            this.m_72379_(p_72232_, p_72236_, 8, 0, oceanmonumentpieces$roomdefinition.f_72445_[Direction.DOWN.m_122411_()]);
-            this.m_72379_(p_72232_, p_72236_, 0, 0, oceanmonumentpieces$roomdefinition1.f_72445_[Direction.DOWN.m_122411_()]);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72232_, p_72236_, 1, 4, 1, 7, 4, 6, f_72323_);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72232_, p_72236_, 8, 4, 1, 14, 4, 6, f_72323_);
-         }
-
-         this.m_73441_(p_72232_, p_72236_, 0, 3, 0, 0, 3, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 15, 3, 0, 15, 3, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 1, 3, 0, 15, 3, 0, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 1, 3, 7, 14, 3, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 0, 2, 0, 0, 2, 7, f_72323_, f_72323_, false);
-         this.m_73441_(p_72232_, p_72236_, 15, 2, 0, 15, 2, 7, f_72323_, f_72323_, false);
-         this.m_73441_(p_72232_, p_72236_, 1, 2, 0, 15, 2, 0, f_72323_, f_72323_, false);
-         this.m_73441_(p_72232_, p_72236_, 1, 2, 7, 14, 2, 7, f_72323_, f_72323_, false);
-         this.m_73441_(p_72232_, p_72236_, 0, 1, 0, 0, 1, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 15, 1, 0, 15, 1, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 1, 1, 0, 15, 1, 0, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 1, 1, 7, 14, 1, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 5, 1, 0, 10, 1, 4, f_72324_, f_72324_, false);
-         this.m_73441_(p_72232_, p_72236_, 6, 2, 0, 9, 2, 3, f_72323_, f_72323_, false);
-         this.m_73441_(p_72232_, p_72236_, 5, 3, 0, 10, 3, 4, f_72324_, f_72324_, false);
-         this.m_73434_(p_72232_, f_72327_, 6, 2, 3, p_72236_);
-         this.m_73434_(p_72232_, f_72327_, 9, 2, 3, p_72236_);
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72232_, p_72236_, 3, 1, 0, 4, 2, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72232_, p_72236_, 3, 1, 7, 4, 2, 7);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72232_, p_72236_, 0, 1, 3, 0, 2, 4);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72232_, p_72236_, 11, 1, 0, 12, 2, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72232_, p_72236_, 11, 1, 7, 12, 2, 7);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72232_, p_72236_, 15, 1, 3, 15, 2, 4);
-         }
-
-         return true;
-      }
-   }
-
-   public static class OceanMonumentDoubleXYRoom extends OceanMonumentPieces.OceanMonumentPiece {
-      public OceanMonumentDoubleXYRoom(Direction p_72243_, OceanMonumentPieces.RoomDefinition p_72244_) {
-         super(StructurePieceType.f_67121_, 1, p_72243_, p_72244_, 2, 2, 1);
-      }
-
-      public OceanMonumentDoubleXYRoom(StructureManager p_72240_, CompoundTag p_72241_) {
-         super(StructurePieceType.f_67121_, p_72241_);
-      }
-
-      public boolean m_7832_(WorldGenLevel p_72246_, StructureFeatureManager p_72247_, ChunkGenerator p_72248_, Random p_72249_, BoundingBox p_72250_, ChunkPos p_72251_, BlockPos p_72252_) {
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.EAST.m_122411_()];
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition2 = oceanmonumentpieces$roomdefinition1.f_72444_[Direction.UP.m_122411_()];
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition3 = oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()];
-         if (this.f_72334_.f_72443_ / 25 > 0) {
-            this.m_72379_(p_72246_, p_72250_, 8, 0, oceanmonumentpieces$roomdefinition.f_72445_[Direction.DOWN.m_122411_()]);
-            this.m_72379_(p_72246_, p_72250_, 0, 0, oceanmonumentpieces$roomdefinition1.f_72445_[Direction.DOWN.m_122411_()]);
-         }
-
-         if (oceanmonumentpieces$roomdefinition2.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72246_, p_72250_, 1, 8, 1, 7, 8, 6, f_72323_);
-         }
-
-         if (oceanmonumentpieces$roomdefinition3.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72246_, p_72250_, 8, 8, 1, 14, 8, 6, f_72323_);
-         }
-
-         for(int i = 1; i <= 7; ++i) {
-            BlockState blockstate = f_72324_;
-            if (i == 2 || i == 6) {
-               blockstate = f_72323_;
-            }
-
-            this.m_73441_(p_72246_, p_72250_, 0, i, 0, 0, i, 7, blockstate, blockstate, false);
-            this.m_73441_(p_72246_, p_72250_, 15, i, 0, 15, i, 7, blockstate, blockstate, false);
-            this.m_73441_(p_72246_, p_72250_, 1, i, 0, 15, i, 0, blockstate, blockstate, false);
-            this.m_73441_(p_72246_, p_72250_, 1, i, 7, 14, i, 7, blockstate, blockstate, false);
-         }
-
-         this.m_73441_(p_72246_, p_72250_, 2, 1, 3, 2, 7, 4, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 3, 1, 2, 4, 7, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 3, 1, 5, 4, 7, 5, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 13, 1, 3, 13, 7, 4, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 11, 1, 2, 12, 7, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 11, 1, 5, 12, 7, 5, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 5, 1, 3, 5, 3, 4, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 10, 1, 3, 10, 3, 4, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 5, 7, 2, 10, 7, 5, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 5, 5, 2, 5, 7, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 10, 5, 2, 10, 7, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 5, 5, 5, 5, 7, 5, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 10, 5, 5, 10, 7, 5, f_72324_, f_72324_, false);
-         this.m_73434_(p_72246_, f_72324_, 6, 6, 2, p_72250_);
-         this.m_73434_(p_72246_, f_72324_, 9, 6, 2, p_72250_);
-         this.m_73434_(p_72246_, f_72324_, 6, 6, 5, p_72250_);
-         this.m_73434_(p_72246_, f_72324_, 9, 6, 5, p_72250_);
-         this.m_73441_(p_72246_, p_72250_, 5, 4, 3, 6, 4, 4, f_72324_, f_72324_, false);
-         this.m_73441_(p_72246_, p_72250_, 9, 4, 3, 10, 4, 4, f_72324_, f_72324_, false);
-         this.m_73434_(p_72246_, f_72327_, 5, 4, 2, p_72250_);
-         this.m_73434_(p_72246_, f_72327_, 5, 4, 5, p_72250_);
-         this.m_73434_(p_72246_, f_72327_, 10, 4, 2, p_72250_);
-         this.m_73434_(p_72246_, f_72327_, 10, 4, 5, p_72250_);
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 3, 1, 0, 4, 2, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 3, 1, 7, 4, 2, 7);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 0, 1, 3, 0, 2, 4);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 11, 1, 0, 12, 2, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 11, 1, 7, 12, 2, 7);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 15, 1, 3, 15, 2, 4);
-         }
-
-         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 3, 5, 0, 4, 6, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 3, 5, 7, 4, 6, 7);
-         }
-
-         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 0, 5, 3, 0, 6, 4);
-         }
-
-         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 11, 5, 0, 12, 6, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 11, 5, 7, 12, 6, 7);
-         }
-
-         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72246_, p_72250_, 15, 5, 3, 15, 6, 4);
-         }
-
-         return true;
-      }
-   }
-
-   public static class OceanMonumentDoubleYRoom extends OceanMonumentPieces.OceanMonumentPiece {
-      public OceanMonumentDoubleYRoom(Direction p_72257_, OceanMonumentPieces.RoomDefinition p_72258_) {
-         super(StructurePieceType.f_67122_, 1, p_72257_, p_72258_, 1, 2, 1);
-      }
-
-      public OceanMonumentDoubleYRoom(StructureManager p_72254_, CompoundTag p_72255_) {
-         super(StructurePieceType.f_67122_, p_72255_);
-      }
-
-      public boolean m_7832_(WorldGenLevel p_72260_, StructureFeatureManager p_72261_, ChunkGenerator p_72262_, Random p_72263_, BoundingBox p_72264_, ChunkPos p_72265_, BlockPos p_72266_) {
-         if (this.f_72334_.f_72443_ / 25 > 0) {
-            this.m_72379_(p_72260_, p_72264_, 0, 0, this.f_72334_.f_72445_[Direction.DOWN.m_122411_()]);
-         }
-
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.UP.m_122411_()];
-         if (oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72260_, p_72264_, 1, 8, 1, 6, 8, 6, f_72323_);
-         }
-
-         this.m_73441_(p_72260_, p_72264_, 0, 4, 0, 0, 4, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 7, 4, 0, 7, 4, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 1, 4, 0, 6, 4, 0, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 1, 4, 7, 6, 4, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 2, 4, 1, 2, 4, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 1, 4, 2, 1, 4, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 5, 4, 1, 5, 4, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 6, 4, 2, 6, 4, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 2, 4, 5, 2, 4, 6, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 1, 4, 5, 1, 4, 5, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 5, 4, 5, 5, 4, 6, f_72324_, f_72324_, false);
-         this.m_73441_(p_72260_, p_72264_, 6, 4, 5, 6, 4, 5, f_72324_, f_72324_, false);
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
-
-         for(int i = 1; i <= 5; i += 4) {
-            int j = 0;
-            if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
-               this.m_73441_(p_72260_, p_72264_, 2, i, j, 2, i + 2, j, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, 5, i, j, 5, i + 2, j, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, 3, i + 2, j, 4, i + 2, j, f_72324_, f_72324_, false);
-            } else {
-               this.m_73441_(p_72260_, p_72264_, 0, i, j, 7, i + 2, j, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, 0, i + 1, j, 7, i + 1, j, f_72323_, f_72323_, false);
-            }
-
-            j = 7;
-            if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.NORTH.m_122411_()]) {
-               this.m_73441_(p_72260_, p_72264_, 2, i, j, 2, i + 2, j, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, 5, i, j, 5, i + 2, j, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, 3, i + 2, j, 4, i + 2, j, f_72324_, f_72324_, false);
-            } else {
-               this.m_73441_(p_72260_, p_72264_, 0, i, j, 7, i + 2, j, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, 0, i + 1, j, 7, i + 1, j, f_72323_, f_72323_, false);
-            }
-
-            int k = 0;
-            if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
-               this.m_73441_(p_72260_, p_72264_, k, i, 2, k, i + 2, 2, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, k, i, 5, k, i + 2, 5, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, k, i + 2, 3, k, i + 2, 4, f_72324_, f_72324_, false);
-            } else {
-               this.m_73441_(p_72260_, p_72264_, k, i, 0, k, i + 2, 7, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, k, i + 1, 0, k, i + 1, 7, f_72323_, f_72323_, false);
-            }
-
-            k = 7;
-            if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.EAST.m_122411_()]) {
-               this.m_73441_(p_72260_, p_72264_, k, i, 2, k, i + 2, 2, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, k, i, 5, k, i + 2, 5, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, k, i + 2, 3, k, i + 2, 4, f_72324_, f_72324_, false);
-            } else {
-               this.m_73441_(p_72260_, p_72264_, k, i, 0, k, i + 2, 7, f_72324_, f_72324_, false);
-               this.m_73441_(p_72260_, p_72264_, k, i + 1, 0, k, i + 1, 7, f_72323_, f_72323_, false);
-            }
-
-            oceanmonumentpieces$roomdefinition1 = oceanmonumentpieces$roomdefinition;
-         }
-
-         return true;
-      }
-   }
-
-   public static class OceanMonumentDoubleYZRoom extends OceanMonumentPieces.OceanMonumentPiece {
-      public OceanMonumentDoubleYZRoom(Direction p_72271_, OceanMonumentPieces.RoomDefinition p_72272_) {
-         super(StructurePieceType.f_67123_, 1, p_72271_, p_72272_, 1, 2, 2);
-      }
-
-      public OceanMonumentDoubleYZRoom(StructureManager p_72268_, CompoundTag p_72269_) {
-         super(StructurePieceType.f_67123_, p_72269_);
-      }
-
-      public boolean m_7832_(WorldGenLevel p_72274_, StructureFeatureManager p_72275_, ChunkGenerator p_72276_, Random p_72277_, BoundingBox p_72278_, ChunkPos p_72279_, BlockPos p_72280_) {
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.NORTH.m_122411_()];
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition2 = oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()];
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition3 = oceanmonumentpieces$roomdefinition1.f_72444_[Direction.UP.m_122411_()];
-         if (this.f_72334_.f_72443_ / 25 > 0) {
-            this.m_72379_(p_72274_, p_72278_, 0, 8, oceanmonumentpieces$roomdefinition.f_72445_[Direction.DOWN.m_122411_()]);
-            this.m_72379_(p_72274_, p_72278_, 0, 0, oceanmonumentpieces$roomdefinition1.f_72445_[Direction.DOWN.m_122411_()]);
-         }
-
-         if (oceanmonumentpieces$roomdefinition3.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72274_, p_72278_, 1, 8, 1, 6, 8, 7, f_72323_);
-         }
-
-         if (oceanmonumentpieces$roomdefinition2.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72274_, p_72278_, 1, 8, 8, 6, 8, 14, f_72323_);
-         }
-
-         for(int i = 1; i <= 7; ++i) {
-            BlockState blockstate = f_72324_;
-            if (i == 2 || i == 6) {
-               blockstate = f_72323_;
-            }
-
-            this.m_73441_(p_72274_, p_72278_, 0, i, 0, 0, i, 15, blockstate, blockstate, false);
-            this.m_73441_(p_72274_, p_72278_, 7, i, 0, 7, i, 15, blockstate, blockstate, false);
-            this.m_73441_(p_72274_, p_72278_, 1, i, 0, 6, i, 0, blockstate, blockstate, false);
-            this.m_73441_(p_72274_, p_72278_, 1, i, 15, 6, i, 15, blockstate, blockstate, false);
-         }
-
-         for(int j = 1; j <= 7; ++j) {
-            BlockState blockstate1 = f_72325_;
-            if (j == 2 || j == 6) {
-               blockstate1 = f_72327_;
-            }
-
-            this.m_73441_(p_72274_, p_72278_, 3, j, 7, 4, j, 8, blockstate1, blockstate1, false);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 3, 1, 0, 4, 2, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 7, 1, 3, 7, 2, 4);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 0, 1, 3, 0, 2, 4);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 3, 1, 15, 4, 2, 15);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 0, 1, 11, 0, 2, 12);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 7, 1, 11, 7, 2, 12);
-         }
-
-         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 3, 5, 0, 4, 6, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 7, 5, 3, 7, 6, 4);
-            this.m_73441_(p_72274_, p_72278_, 5, 4, 2, 6, 4, 5, f_72324_, f_72324_, false);
-            this.m_73441_(p_72274_, p_72278_, 6, 1, 2, 6, 3, 2, f_72324_, f_72324_, false);
-            this.m_73441_(p_72274_, p_72278_, 6, 1, 5, 6, 3, 5, f_72324_, f_72324_, false);
-         }
-
-         if (oceanmonumentpieces$roomdefinition3.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 0, 5, 3, 0, 6, 4);
-            this.m_73441_(p_72274_, p_72278_, 1, 4, 2, 2, 4, 5, f_72324_, f_72324_, false);
-            this.m_73441_(p_72274_, p_72278_, 1, 1, 2, 1, 3, 2, f_72324_, f_72324_, false);
-            this.m_73441_(p_72274_, p_72278_, 1, 1, 5, 1, 3, 5, f_72324_, f_72324_, false);
-         }
-
-         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 3, 5, 15, 4, 6, 15);
-         }
-
-         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 0, 5, 11, 0, 6, 12);
-            this.m_73441_(p_72274_, p_72278_, 1, 4, 10, 2, 4, 13, f_72324_, f_72324_, false);
-            this.m_73441_(p_72274_, p_72278_, 1, 1, 10, 1, 3, 10, f_72324_, f_72324_, false);
-            this.m_73441_(p_72274_, p_72278_, 1, 1, 13, 1, 3, 13, f_72324_, f_72324_, false);
-         }
-
-         if (oceanmonumentpieces$roomdefinition2.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72274_, p_72278_, 7, 5, 11, 7, 6, 12);
-            this.m_73441_(p_72274_, p_72278_, 5, 4, 10, 6, 4, 13, f_72324_, f_72324_, false);
-            this.m_73441_(p_72274_, p_72278_, 6, 1, 10, 6, 3, 10, f_72324_, f_72324_, false);
-            this.m_73441_(p_72274_, p_72278_, 6, 1, 13, 6, 3, 13, f_72324_, f_72324_, false);
-         }
-
-         return true;
-      }
-   }
-
-   public static class OceanMonumentDoubleZRoom extends OceanMonumentPieces.OceanMonumentPiece {
-      public OceanMonumentDoubleZRoom(Direction p_72285_, OceanMonumentPieces.RoomDefinition p_72286_) {
-         super(StructurePieceType.f_67124_, 1, p_72285_, p_72286_, 1, 1, 2);
-      }
-
-      public OceanMonumentDoubleZRoom(StructureManager p_72282_, CompoundTag p_72283_) {
-         super(StructurePieceType.f_67124_, p_72283_);
-      }
-
-      public boolean m_7832_(WorldGenLevel p_72288_, StructureFeatureManager p_72289_, ChunkGenerator p_72290_, Random p_72291_, BoundingBox p_72292_, ChunkPos p_72293_, BlockPos p_72294_) {
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = this.f_72334_.f_72444_[Direction.NORTH.m_122411_()];
-         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition1 = this.f_72334_;
-         if (this.f_72334_.f_72443_ / 25 > 0) {
-            this.m_72379_(p_72288_, p_72292_, 0, 8, oceanmonumentpieces$roomdefinition.f_72445_[Direction.DOWN.m_122411_()]);
-            this.m_72379_(p_72288_, p_72292_, 0, 0, oceanmonumentpieces$roomdefinition1.f_72445_[Direction.DOWN.m_122411_()]);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72288_, p_72292_, 1, 4, 1, 6, 4, 7, f_72323_);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()] == null) {
-            this.m_72369_(p_72288_, p_72292_, 1, 4, 8, 6, 4, 14, f_72323_);
-         }
-
-         this.m_73441_(p_72288_, p_72292_, 0, 3, 0, 0, 3, 15, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 7, 3, 0, 7, 3, 15, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 3, 0, 7, 3, 0, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 3, 15, 6, 3, 15, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 0, 2, 0, 0, 2, 15, f_72323_, f_72323_, false);
-         this.m_73441_(p_72288_, p_72292_, 7, 2, 0, 7, 2, 15, f_72323_, f_72323_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 2, 0, 7, 2, 0, f_72323_, f_72323_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 2, 15, 6, 2, 15, f_72323_, f_72323_, false);
-         this.m_73441_(p_72288_, p_72292_, 0, 1, 0, 0, 1, 15, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 7, 1, 0, 7, 1, 15, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 1, 0, 7, 1, 0, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 1, 15, 6, 1, 15, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 1, 1, 1, 1, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 6, 1, 1, 6, 1, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 3, 1, 1, 3, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 6, 3, 1, 6, 3, 2, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 1, 13, 1, 1, 14, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 6, 1, 13, 6, 1, 14, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 1, 3, 13, 1, 3, 14, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 6, 3, 13, 6, 3, 14, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 2, 1, 6, 2, 3, 6, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 5, 1, 6, 5, 3, 6, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 2, 1, 9, 2, 3, 9, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 5, 1, 9, 5, 3, 9, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 3, 2, 6, 4, 2, 6, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 3, 2, 9, 4, 2, 9, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 2, 2, 7, 2, 2, 8, f_72324_, f_72324_, false);
-         this.m_73441_(p_72288_, p_72292_, 5, 2, 7, 5, 2, 8, f_72324_, f_72324_, false);
-         this.m_73434_(p_72288_, f_72327_, 2, 2, 5, p_72292_);
-         this.m_73434_(p_72288_, f_72327_, 5, 2, 5, p_72292_);
-         this.m_73434_(p_72288_, f_72327_, 2, 2, 10, p_72292_);
-         this.m_73434_(p_72288_, f_72327_, 5, 2, 10, p_72292_);
-         this.m_73434_(p_72288_, f_72324_, 2, 3, 5, p_72292_);
-         this.m_73434_(p_72288_, f_72324_, 5, 3, 5, p_72292_);
-         this.m_73434_(p_72288_, f_72324_, 2, 3, 10, p_72292_);
-         this.m_73434_(p_72288_, f_72324_, 5, 3, 10, p_72292_);
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.SOUTH.m_122411_()]) {
-            this.m_72360_(p_72288_, p_72292_, 3, 1, 0, 4, 2, 0);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72288_, p_72292_, 7, 1, 3, 7, 2, 4);
-         }
-
-         if (oceanmonumentpieces$roomdefinition1.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72288_, p_72292_, 0, 1, 3, 0, 2, 4);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72288_, p_72292_, 3, 1, 15, 4, 2, 15);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72288_, p_72292_, 0, 1, 11, 0, 2, 12);
-         }
-
-         if (oceanmonumentpieces$roomdefinition.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72288_, p_72292_, 7, 1, 11, 7, 2, 12);
-         }
-
-         return true;
-      }
-   }
-
-   public static class OceanMonumentEntryRoom extends OceanMonumentPieces.OceanMonumentPiece {
-      public OceanMonumentEntryRoom(Direction p_72299_, OceanMonumentPieces.RoomDefinition p_72300_) {
-         super(StructurePieceType.f_67125_, 1, p_72299_, p_72300_, 1, 1, 1);
-      }
-
-      public OceanMonumentEntryRoom(StructureManager p_72296_, CompoundTag p_72297_) {
-         super(StructurePieceType.f_67125_, p_72297_);
-      }
-
-      public boolean m_7832_(WorldGenLevel p_72302_, StructureFeatureManager p_72303_, ChunkGenerator p_72304_, Random p_72305_, BoundingBox p_72306_, ChunkPos p_72307_, BlockPos p_72308_) {
-         this.m_73441_(p_72302_, p_72306_, 0, 3, 0, 2, 3, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72302_, p_72306_, 5, 3, 0, 7, 3, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72302_, p_72306_, 0, 2, 0, 1, 2, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72302_, p_72306_, 6, 2, 0, 7, 2, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72302_, p_72306_, 0, 1, 0, 0, 1, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72302_, p_72306_, 7, 1, 0, 7, 1, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72302_, p_72306_, 0, 1, 7, 7, 3, 7, f_72324_, f_72324_, false);
-         this.m_73441_(p_72302_, p_72306_, 1, 1, 0, 2, 3, 0, f_72324_, f_72324_, false);
-         this.m_73441_(p_72302_, p_72306_, 5, 1, 0, 6, 3, 0, f_72324_, f_72324_, false);
-         if (this.f_72334_.f_72445_[Direction.NORTH.m_122411_()]) {
-            this.m_72360_(p_72302_, p_72306_, 3, 1, 7, 4, 2, 7);
-         }
-
-         if (this.f_72334_.f_72445_[Direction.WEST.m_122411_()]) {
-            this.m_72360_(p_72302_, p_72306_, 0, 1, 3, 1, 2, 4);
-         }
-
-         if (this.f_72334_.f_72445_[Direction.EAST.m_122411_()]) {
-            this.m_72360_(p_72302_, p_72306_, 6, 1, 3, 7, 2, 4);
-         }
-
-         return true;
-      }
-   }
-
    public static class OceanMonumentPenthouse extends OceanMonumentPieces.OceanMonumentPiece {
       public OceanMonumentPenthouse(Direction p_72313_, BoundingBox p_72314_) {
          super(StructurePieceType.f_67126_, p_72313_, p_72314_);
@@ -1474,6 +_,63 @@
       }
    }
 
+   static class RoomDefinition {
+      private final int f_72443_;
+      private final OceanMonumentPieces.RoomDefinition[] f_72444_ = new OceanMonumentPieces.RoomDefinition[6];
+      private final boolean[] f_72445_ = new boolean[6];
+      private boolean f_72446_;
+      private boolean f_72447_;
+      private int f_72448_;
+
+      public RoomDefinition(int p_72450_) {
+         this.f_72443_ = p_72450_;
+      }
+
+      public void m_72459_(Direction p_72460_, OceanMonumentPieces.RoomDefinition p_72461_) {
+         this.f_72444_[p_72460_.m_122411_()] = p_72461_;
+         p_72461_.f_72444_[p_72460_.m_122424_().m_122411_()] = this;
+      }
+
+      public void m_72451_() {
+         for(int i = 0; i < 6; ++i) {
+            this.f_72445_[i] = this.f_72444_[i] != null;
+         }
+
+      }
+
+      public boolean m_72452_(int p_72453_) {
+         if (this.f_72447_) {
+            return true;
+         } else {
+            this.f_72448_ = p_72453_;
+
+            for(int i = 0; i < 6; ++i) {
+               if (this.f_72444_[i] != null && this.f_72445_[i] && this.f_72444_[i].f_72448_ != p_72453_ && this.f_72444_[i].m_72452_(p_72453_)) {
+                  return true;
+               }
+            }
+
+            return false;
+         }
+      }
+
+      public boolean m_72462_() {
+         return this.f_72443_ >= 75;
+      }
+
+      public int m_72468_() {
+         int i = 0;
+
+         for(int j = 0; j < 6; ++j) {
+            if (this.f_72445_[j]) {
+               ++i;
+            }
+         }
+
+         return i;
+      }
+   }
+
    public static class OceanMonumentSimpleRoom extends OceanMonumentPieces.OceanMonumentPiece {
       private int f_72397_;
 
@@ -1797,60 +_,97 @@
       }
    }
 
-   static class RoomDefinition {
-      private final int f_72443_;
-      private final OceanMonumentPieces.RoomDefinition[] f_72444_ = new OceanMonumentPieces.RoomDefinition[6];
-      private final boolean[] f_72445_ = new boolean[6];
-      private boolean f_72446_;
-      private boolean f_72447_;
-      private int f_72448_;
-
-      public RoomDefinition(int p_72450_) {
-         this.f_72443_ = p_72450_;
-      }
-
-      public void m_72459_(Direction p_72460_, OceanMonumentPieces.RoomDefinition p_72461_) {
-         this.f_72444_[p_72460_.m_122411_()] = p_72461_;
-         p_72461_.f_72444_[p_72460_.m_122424_().m_122411_()] = this;
-      }
-
-      public void m_72451_() {
-         for(int i = 0; i < 6; ++i) {
-            this.f_72445_[i] = this.f_72444_[i] != null;
-         }
-
-      }
-
-      public boolean m_72452_(int p_72453_) {
-         if (this.f_72447_) {
-            return true;
-         } else {
-            this.f_72448_ = p_72453_;
-
-            for(int i = 0; i < 6; ++i) {
-               if (this.f_72444_[i] != null && this.f_72445_[i] && this.f_72444_[i].f_72448_ != p_72453_ && this.f_72444_[i].m_72452_(p_72453_)) {
-                  return true;
-               }
-            }
-
-            return false;
-         }
-      }
-
-      public boolean m_72462_() {
-         return this.f_72443_ >= 75;
-      }
-
-      public int m_72468_() {
-         int i = 0;
-
-         for(int j = 0; j < 6; ++j) {
-            if (this.f_72445_[j]) {
-               ++i;
-            }
-         }
-
-         return i;
+   static class FitDoubleXRoom implements OceanMonumentPieces.MonumentRoomFitter {
+      private FitDoubleXRoom() {
+      }
+
+      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72097_) {
+         return p_72097_.f_72445_[Direction.EAST.m_122411_()] && !p_72097_.f_72444_[Direction.EAST.m_122411_()].f_72446_;
+      }
+
+      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72099_, OceanMonumentPieces.RoomDefinition p_72100_, Random p_72101_) {
+         p_72100_.f_72446_ = true;
+         p_72100_.f_72444_[Direction.EAST.m_122411_()].f_72446_ = true;
+         return new OceanMonumentPieces.OceanMonumentDoubleXRoom(p_72099_, p_72100_);
+      }
+   }
+
+   static class FitDoubleXYRoom implements OceanMonumentPieces.MonumentRoomFitter {
+      private FitDoubleXYRoom() {
+      }
+
+      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72106_) {
+         if (p_72106_.f_72445_[Direction.EAST.m_122411_()] && !p_72106_.f_72444_[Direction.EAST.m_122411_()].f_72446_ && p_72106_.f_72445_[Direction.UP.m_122411_()] && !p_72106_.f_72444_[Direction.UP.m_122411_()].f_72446_) {
+            OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = p_72106_.f_72444_[Direction.EAST.m_122411_()];
+            return oceanmonumentpieces$roomdefinition.f_72445_[Direction.UP.m_122411_()] && !oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()].f_72446_;
+         } else {
+            return false;
+         }
+      }
+
+      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72108_, OceanMonumentPieces.RoomDefinition p_72109_, Random p_72110_) {
+         p_72109_.f_72446_ = true;
+         p_72109_.f_72444_[Direction.EAST.m_122411_()].f_72446_ = true;
+         p_72109_.f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
+         p_72109_.f_72444_[Direction.EAST.m_122411_()].f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
+         return new OceanMonumentPieces.OceanMonumentDoubleXYRoom(p_72108_, p_72109_);
+      }
+   }
+
+   static class FitDoubleYRoom implements OceanMonumentPieces.MonumentRoomFitter {
+      private FitDoubleYRoom() {
+      }
+
+      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72115_) {
+         return p_72115_.f_72445_[Direction.UP.m_122411_()] && !p_72115_.f_72444_[Direction.UP.m_122411_()].f_72446_;
+      }
+
+      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72117_, OceanMonumentPieces.RoomDefinition p_72118_, Random p_72119_) {
+         p_72118_.f_72446_ = true;
+         p_72118_.f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
+         return new OceanMonumentPieces.OceanMonumentDoubleYRoom(p_72117_, p_72118_);
+      }
+   }
+
+   static class FitDoubleYZRoom implements OceanMonumentPieces.MonumentRoomFitter {
+      private FitDoubleYZRoom() {
+      }
+
+      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72124_) {
+         if (p_72124_.f_72445_[Direction.NORTH.m_122411_()] && !p_72124_.f_72444_[Direction.NORTH.m_122411_()].f_72446_ && p_72124_.f_72445_[Direction.UP.m_122411_()] && !p_72124_.f_72444_[Direction.UP.m_122411_()].f_72446_) {
+            OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = p_72124_.f_72444_[Direction.NORTH.m_122411_()];
+            return oceanmonumentpieces$roomdefinition.f_72445_[Direction.UP.m_122411_()] && !oceanmonumentpieces$roomdefinition.f_72444_[Direction.UP.m_122411_()].f_72446_;
+         } else {
+            return false;
+         }
+      }
+
+      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72126_, OceanMonumentPieces.RoomDefinition p_72127_, Random p_72128_) {
+         p_72127_.f_72446_ = true;
+         p_72127_.f_72444_[Direction.NORTH.m_122411_()].f_72446_ = true;
+         p_72127_.f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
+         p_72127_.f_72444_[Direction.NORTH.m_122411_()].f_72444_[Direction.UP.m_122411_()].f_72446_ = true;
+         return new OceanMonumentPieces.OceanMonumentDoubleYZRoom(p_72126_, p_72127_);
+      }
+   }
+
+   static class FitDoubleZRoom implements OceanMonumentPieces.MonumentRoomFitter {
+      private FitDoubleZRoom() {
+      }
+
+      public boolean m_7725_(OceanMonumentPieces.RoomDefinition p_72133_) {
+         return p_72133_.f_72445_[Direction.NORTH.m_122411_()] && !p_72133_.f_72444_[Direction.NORTH.m_122411_()].f_72446_;
+      }
+
+      public OceanMonumentPieces.OceanMonumentPiece m_7924_(Direction p_72135_, OceanMonumentPieces.RoomDefinition p_72136_, Random p_72137_) {
+         OceanMonumentPieces.RoomDefinition oceanmonumentpieces$roomdefinition = p_72136_;
+         if (!p_72136_.f_72445_[Direction.NORTH.m_122411_()] || p_72136_.f_72444_[Direction.NORTH.m_122411_()].f_72446_) {
+            oceanmonumentpieces$roomdefinition = p_72136_.f_72444_[Direction.SOUTH.m_122411_()];
+         }
+
+         oceanmonumentpieces$roomdefinition.f_72446_ = true;
+         oceanmonumentpieces$roomdefinition.f_72444_[Direction.NORTH.m_122411_()].f_72446_ = true;
+         return new OceanMonumentPieces.OceanMonumentDoubleZRoom(p_72135_, oceanmonumentpieces$roomdefinition);
       }
    }
 }
