--- a/net/minecraft/world/level/levelgen/structure/RuinedPortalPiece.java
+++ b/net/minecraft/world/level/levelgen/structure/RuinedPortalPiece.java
@@ -82,8 +_,8 @@
       p_72715_.m_128359_("Rotation", this.f_72652_.name());
       p_72715_.m_128359_("Mirror", this.f_72653_.name());
       p_72715_.m_128359_("VerticalPlacement", this.f_72654_.m_72779_());
-      RuinedPortalPiece.Properties.f_72735_.encodeStart(NbtOps.f_128958_, this.f_72655_).resultOrPartial(f_72650_::error).ifPresent((p_72718_) -> {
-         p_72715_.m_128365_("Properties", p_72718_);
+      RuinedPortalPiece.Properties.f_72735_.encodeStart(NbtOps.f_128958_, this.f_72655_).resultOrPartial(f_72650_::error).ifPresent((p_237018_1_) -> {
+         p_72715_.m_128365_("Properties", p_237018_1_);
       });
    }
 
@@ -121,13 +_,13 @@
          this.m_72719_(p_72681_, p_72678_);
          this.m_72703_(p_72681_, p_72678_);
          if (this.f_72655_.f_72740_ || this.f_72655_.f_72739_) {
-            BlockPos.m_121919_(this.m_73547_()).forEach((p_72713_) -> {
+            BlockPos.m_121919_(this.m_73547_()).forEach((p_237017_3_) -> {
                if (this.f_72655_.f_72740_) {
-                  this.m_72706_(p_72681_, p_72678_, p_72713_);
+                  this.m_72706_(p_72681_, p_72678_, p_237017_3_);
                }
 
                if (this.f_72655_.f_72739_) {
-                  this.m_72722_(p_72681_, p_72678_, p_72713_);
+                  this.m_72722_(p_72681_, p_72678_, p_237017_3_);
                }
 
             });
@@ -254,42 +_,6 @@
       return new ProcessorRule(new BlockMatchTest(p_72690_), AlwaysTrueTest.f_73954_, p_72691_.m_49966_());
    }
 
-   public static class Properties {
-      public static final Codec<RuinedPortalPiece.Properties> f_72735_ = RecordCodecBuilder.create((p_72752_) -> {
-         return p_72752_.group(Codec.BOOL.fieldOf("cold").forGetter((p_72764_) -> {
-            return p_72764_.f_72736_;
-         }), Codec.FLOAT.fieldOf("mossiness").forGetter((p_72762_) -> {
-            return p_72762_.f_72737_;
-         }), Codec.BOOL.fieldOf("air_pocket").forGetter((p_72760_) -> {
-            return p_72760_.f_72738_;
-         }), Codec.BOOL.fieldOf("overgrown").forGetter((p_72758_) -> {
-            return p_72758_.f_72739_;
-         }), Codec.BOOL.fieldOf("vines").forGetter((p_72756_) -> {
-            return p_72756_.f_72740_;
-         }), Codec.BOOL.fieldOf("replace_with_blackstone").forGetter((p_72754_) -> {
-            return p_72754_.f_72741_;
-         })).apply(p_72752_, RuinedPortalPiece.Properties::new);
-      });
-      public boolean f_72736_;
-      public float f_72737_ = 0.2F;
-      public boolean f_72738_;
-      public boolean f_72739_;
-      public boolean f_72740_;
-      public boolean f_72741_;
-
-      public Properties() {
-      }
-
-      public <T> Properties(boolean p_72745_, float p_72746_, boolean p_72747_, boolean p_72748_, boolean p_72749_, boolean p_72750_) {
-         this.f_72736_ = p_72745_;
-         this.f_72737_ = p_72746_;
-         this.f_72738_ = p_72747_;
-         this.f_72739_ = p_72748_;
-         this.f_72740_ = p_72749_;
-         this.f_72741_ = p_72750_;
-      }
-   }
-
    public static enum VerticalPlacement {
       ON_LAND_SURFACE("on_land_surface"),
       PARTLY_BURIED("partly_buried"),
@@ -298,8 +_,8 @@
       UNDERGROUND("underground"),
       IN_NETHER("in_nether");
 
-      private static final Map<String, RuinedPortalPiece.VerticalPlacement> f_72771_ = Arrays.stream(values()).collect(Collectors.toMap(RuinedPortalPiece.VerticalPlacement::m_72779_, (p_72781_) -> {
-         return p_72781_;
+      private static final Map<String, RuinedPortalPiece.VerticalPlacement> f_72771_ = Arrays.stream(values()).collect(Collectors.toMap(RuinedPortalPiece.VerticalPlacement::m_72779_, (p_237041_0_) -> {
+         return p_237041_0_;
       }));
       private final String f_72772_;
 
@@ -313,6 +_,42 @@
 
       public static RuinedPortalPiece.VerticalPlacement m_72782_(String p_72783_) {
          return f_72771_.get(p_72783_);
+      }
+   }
+
+   public static class Properties {
+      public static final Codec<RuinedPortalPiece.Properties> f_72735_ = RecordCodecBuilder.create((p_237031_0_) -> {
+         return p_237031_0_.group(Codec.BOOL.fieldOf("cold").forGetter((p_237037_0_) -> {
+            return p_237037_0_.f_72736_;
+         }), Codec.FLOAT.fieldOf("mossiness").forGetter((p_237036_0_) -> {
+            return p_237036_0_.f_72737_;
+         }), Codec.BOOL.fieldOf("air_pocket").forGetter((p_237035_0_) -> {
+            return p_237035_0_.f_72738_;
+         }), Codec.BOOL.fieldOf("overgrown").forGetter((p_237034_0_) -> {
+            return p_237034_0_.f_72739_;
+         }), Codec.BOOL.fieldOf("vines").forGetter((p_237033_0_) -> {
+            return p_237033_0_.f_72740_;
+         }), Codec.BOOL.fieldOf("replace_with_blackstone").forGetter((p_237032_0_) -> {
+            return p_237032_0_.f_72741_;
+         })).apply(p_237031_0_, RuinedPortalPiece.Properties::new);
+      });
+      public boolean f_72736_;
+      public float f_72737_ = 0.2F;
+      public boolean f_72738_;
+      public boolean f_72739_;
+      public boolean f_72740_;
+      public boolean f_72741_;
+
+      public Properties() {
+      }
+
+      public <T> Properties(boolean p_72745_, float p_72746_, boolean p_72747_, boolean p_72748_, boolean p_72749_, boolean p_72750_) {
+         this.f_72736_ = p_72745_;
+         this.f_72737_ = p_72746_;
+         this.f_72738_ = p_72747_;
+         this.f_72739_ = p_72748_;
+         this.f_72740_ = p_72749_;
+         this.f_72741_ = p_72750_;
       }
    }
 }
