--- a/net/minecraft/stats/ServerStatsCounter.java
+++ b/net/minecraft/stats/ServerStatsCounter.java
@@ -95,13 +_,13 @@
 
             for(String s : compoundtag1.m_128431_()) {
                if (compoundtag1.m_128425_(s, 10)) {
-                  Util.m_137521_(Registry.f_122867_.m_6612_(new ResourceLocation(s)), (p_12844_) -> {
+                  Util.m_137521_(Registry.f_122867_.m_6612_(new ResourceLocation(s)), (p_219731_3_) -> {
                      CompoundTag compoundtag2 = compoundtag1.m_128469_(s);
 
                      for(String s1 : compoundtag2.m_128431_()) {
                         if (compoundtag2.m_128425_(s1, 99)) {
-                           Util.m_137521_(this.m_12823_(p_12844_, s1), (p_12840_) -> {
-                              this.f_13013_.put(p_12840_, compoundtag2.m_128451_(s1));
+                           Util.m_137521_(this.m_12823_(p_219731_3_, s1), (p_219730_3_) -> {
+                              this.f_13013_.put(p_219730_3_, compoundtag2.m_128451_(s1));
                            }, () -> {
                               f_12809_.warn("Invalid statistic in {}: Don't know what {} is", this.f_12811_, s1);
                            });
@@ -149,7 +_,7 @@
 
       for(it.unimi.dsi.fastutil.objects.Object2IntMap.Entry<Stat<?>> entry : this.f_13013_.object2IntEntrySet()) {
          Stat<?> stat = entry.getKey();
-         map.computeIfAbsent(stat.m_12859_(), (p_12822_) -> {
+         map.computeIfAbsent(stat.m_12859_(), (p_199064_0_) -> {
             return new JsonObject();
          }).addProperty(m_12846_(stat).toString(), entry.getIntValue());
       }
