--- a/net/minecraft/stats/StatFormatter.java
+++ b/net/minecraft/stats/StatFormatter.java
@@ -9,24 +_,24 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public interface StatFormatter {
-   DecimalFormat f_12872_ = Util.m_137469_(new DecimalFormat("########0.00"), (p_12881_) -> {
-      p_12881_.setDecimalFormatSymbols(DecimalFormatSymbols.getInstance(Locale.ROOT));
+   DecimalFormat f_12872_ = Util.m_137469_(new DecimalFormat("########0.00"), (p_223254_0_) -> {
+      p_223254_0_.setDecimalFormatSymbols(DecimalFormatSymbols.getInstance(Locale.ROOT));
    });
    StatFormatter f_12873_ = NumberFormat.getIntegerInstance(Locale.US)::format;
-   StatFormatter f_12874_ = (p_12885_) -> {
-      return f_12872_.format((double)p_12885_ * 0.1D);
+   StatFormatter f_12874_ = (p_223256_0_) -> {
+      return f_12872_.format((double)p_223256_0_ * 0.1D);
    };
-   StatFormatter f_12875_ = (p_12883_) -> {
-      double d0 = (double)p_12883_ / 100.0D;
+   StatFormatter f_12875_ = (p_223255_0_) -> {
+      double d0 = (double)p_223255_0_ / 100.0D;
       double d1 = d0 / 1000.0D;
       if (d1 > 0.5D) {
          return f_12872_.format(d1) + " km";
       } else {
-         return d0 > 0.5D ? f_12872_.format(d0) + " m" : p_12883_ + " cm";
+         return d0 > 0.5D ? f_12872_.format(d0) + " m" : p_223255_0_ + " cm";
       }
    };
-   StatFormatter f_12876_ = (p_12879_) -> {
-      double d0 = (double)p_12879_ / 20.0D;
+   StatFormatter f_12876_ = (p_223253_0_) -> {
+      double d0 = (double)p_223253_0_ / 20.0D;
       double d1 = d0 / 60.0D;
       double d2 = d1 / 60.0D;
       double d3 = d2 / 24.0D;
