--- a/com/mojang/blaze3d/vertex/VertexFormatElement.java
+++ b/com/mojang/blaze3d/vertex/VertexFormatElement.java
@@ -88,6 +_,12 @@
       this.f_86031_.m_86098_(this.f_86032_);
    }
 
+    //Forge Start
+    public int getElementCount() {
+       return f_86033_;
+    }
+    //Forge End
+
    @OnlyIn(Dist.CLIENT)
    public static enum Type {
       FLOAT(4, "Float", 5126),
@@ -123,41 +_,41 @@
 
    @OnlyIn(Dist.CLIENT)
    public static enum Usage {
-      POSITION("Position", (p_86154_, p_86155_, p_86156_, p_86157_, p_86158_) -> {
-         GlStateManager.m_84340_(p_86154_, p_86155_, p_86156_, p_86157_);
+      POSITION("Position", (p_227914_0_, p_227914_1_, p_227914_2_, p_227914_3_, p_227914_5_) -> {
+         GlStateManager.m_84340_(p_227914_0_, p_227914_1_, p_227914_2_, p_227914_3_);
          GlStateManager.m_84559_(32884);
-      }, (p_86146_) -> {
+      }, (p_227912_0_) -> {
          GlStateManager.m_84562_(32884);
       }),
-      NORMAL("Normal", (p_86148_, p_86149_, p_86150_, p_86151_, p_86152_) -> {
-         GlStateManager.m_84245_(p_86149_, p_86150_, p_86151_);
+      NORMAL("Normal", (p_227913_0_, p_227913_1_, p_227913_2_, p_227913_3_, p_227913_5_) -> {
+         GlStateManager.m_84245_(p_227913_1_, p_227913_2_, p_227913_3_);
          GlStateManager.m_84559_(32885);
-      }, (p_86138_) -> {
+      }, (p_227910_0_) -> {
          GlStateManager.m_84562_(32885);
       }),
-      COLOR("Vertex Color", (p_86140_, p_86141_, p_86142_, p_86143_, p_86144_) -> {
-         GlStateManager.m_84393_(p_86140_, p_86141_, p_86142_, p_86143_);
+      COLOR("Vertex Color", (p_227911_0_, p_227911_1_, p_227911_2_, p_227911_3_, p_227911_5_) -> {
+         GlStateManager.m_84393_(p_227911_0_, p_227911_1_, p_227911_2_, p_227911_3_);
          GlStateManager.m_84559_(32886);
-      }, (p_86130_) -> {
+      }, (p_227908_0_) -> {
          GlStateManager.m_84562_(32886);
          GlStateManager.m_84117_();
       }),
-      UV("UV", (p_86132_, p_86133_, p_86134_, p_86135_, p_86136_) -> {
-         GlStateManager.m_84520_('\u84c0' + p_86136_);
-         GlStateManager.m_84233_(p_86132_, p_86133_, p_86134_, p_86135_);
+      UV("UV", (p_227909_0_, p_227909_1_, p_227909_2_, p_227909_3_, p_227909_5_) -> {
+         GlStateManager.m_84520_('\u84c0' + p_227909_5_);
+         GlStateManager.m_84233_(p_227909_0_, p_227909_1_, p_227909_2_, p_227909_3_);
          GlStateManager.m_84559_(32888);
          GlStateManager.m_84520_(33984);
-      }, (p_86122_) -> {
-         GlStateManager.m_84520_('\u84c0' + p_86122_);
+      }, (p_227906_0_) -> {
+         GlStateManager.m_84520_('\u84c0' + p_227906_0_);
          GlStateManager.m_84562_(32888);
          GlStateManager.m_84520_(33984);
       }),
-      PADDING("Padding", (p_86124_, p_86125_, p_86126_, p_86127_, p_86128_) -> {
-      }, (p_86114_) -> {
+      PADDING("Padding", (p_227907_0_, p_227907_1_, p_227907_2_, p_227907_3_, p_227907_5_) -> {
+      }, (p_227904_0_) -> {
       }),
-      GENERIC("Generic", (p_86116_, p_86117_, p_86118_, p_86119_, p_86120_) -> {
-         GlStateManager.m_84565_(p_86120_);
-         GlStateManager.m_84238_(p_86120_, p_86116_, p_86117_, false, p_86118_, p_86119_);
+      GENERIC("Generic", (p_227905_0_, p_227905_1_, p_227905_2_, p_227905_3_, p_227905_5_) -> {
+         GlStateManager.m_84565_(p_227905_5_);
+         GlStateManager.m_84238_(p_227905_5_, p_227905_0_, p_227905_1_, false, p_227905_2_, p_227905_3_);
       }, GlStateManager::m_84086_);
 
       private final String f_86086_;
