--- a/net/minecraft/core/GlobalPos.java
+++ b/net/minecraft/core/GlobalPos.java
@@ -7,8 +_,8 @@
 import net.minecraft.world.level.Level;
 
 public final class GlobalPos {
-   public static final Codec<GlobalPos> f_122633_ = RecordCodecBuilder.create((p_122642_) -> {
-      return p_122642_.group(Level.f_46427_.fieldOf("dimension").forGetter(GlobalPos::m_122640_), BlockPos.f_121852_.fieldOf("pos").forGetter(GlobalPos::m_122646_)).apply(p_122642_, GlobalPos::m_122643_);
+   public static final Codec<GlobalPos> f_122633_ = RecordCodecBuilder.create((p_239647_0_) -> {
+      return p_239647_0_.group(Level.f_46427_.fieldOf("dimension").forGetter(GlobalPos::m_122640_), BlockPos.f_121852_.fieldOf("pos").forGetter(GlobalPos::m_122646_)).apply(p_239647_0_, GlobalPos::m_122643_);
    });
    private final ResourceKey<Level> f_122634_;
    private final BlockPos f_122635_;
