--- a/net/minecraft/core/MappedRegistry.java
+++ b/net/minecraft/core/MappedRegistry.java
@@ -166,7 +_,7 @@
       return Collections.unmodifiableSet(this.f_122674_.keySet());
    }
 
-   public Set<Entry<ResourceKey<T>, T>> m_6579_() {
+   public Set<Map.Entry<ResourceKey<T>, T>> m_6579_() {
       return Collections.unmodifiableMap(this.f_122675_).entrySet();
    }
 
@@ -214,7 +_,8 @@
    }
 
    public static <T> Codec<MappedRegistry<T>> m_122756_(ResourceKey<? extends Registry<T>> p_122757_, Lifecycle p_122758_, Codec<T> p_122759_) {
-      return Codec.unboundedMap(ResourceLocation.f_135803_.xmap(ResourceKey.m_135797_(p_122757_), ResourceKey::m_135782_), p_122759_).xmap((p_122726_) -> {
+      // FORGE: Fix MC-197860
+      return new net.minecraftforge.common.LenientUnboundedMapCodec<>(ResourceLocation.f_135803_.xmap(ResourceKey.m_135797_(p_122757_), ResourceKey::m_135782_), p_122759_).xmap((p_122726_) -> {
          MappedRegistry<T> mappedregistry = new MappedRegistry<>(p_122757_, p_122758_);
          p_122726_.forEach((p_122703_, p_122704_) -> {
             mappedregistry.m_7135_(p_122703_, p_122704_, p_122758_);
