--- a/net/minecraft/core/SerializableUUID.java
+++ b/net/minecraft/core/SerializableUUID.java
@@ -6,10 +_,10 @@
 import net.minecraft.Util;
 
 public final class SerializableUUID {
-   public static final Codec<UUID> f_123272_ = Codec.INT_STREAM.comapFlatMap((p_123280_) -> {
-      return Util.m_137539_(p_123280_, 4).map(SerializableUUID::m_123281_);
-   }, (p_123284_) -> {
-      return Arrays.stream(m_123277_(p_123284_));
+   public static final Codec<UUID> f_123272_ = Codec.INT_STREAM.comapFlatMap((p_239778_0_) -> {
+      return Util.m_137539_(p_239778_0_, 4).map(SerializableUUID::m_123281_);
+   }, (p_239780_0_) -> {
+      return Arrays.stream(m_123277_(p_239780_0_));
    });
 
    public static UUID m_123281_(int[] p_123282_) {
