--- a/net/minecraft/world/level/levelgen/surfacebuilders/ConfiguredSurfaceBuilder.java
+++ b/net/minecraft/world/level/levelgen/surfacebuilders/ConfiguredSurfaceBuilder.java
@@ -10,8 +_,8 @@
 import net.minecraft.world.level.chunk.ChunkAccess;
 
 public class ConfiguredSurfaceBuilder<SC extends SurfaceBuilderConfiguration> {
-   public static final Codec<ConfiguredSurfaceBuilder<?>> f_74762_ = Registry.f_122835_.dispatch((p_74774_) -> {
-      return p_74774_.f_74764_;
+   public static final Codec<ConfiguredSurfaceBuilder<?>> f_74762_ = Registry.f_122835_.dispatch((p_237169_0_) -> {
+      return p_237169_0_.f_74764_;
    }, SurfaceBuilder::m_75240_);
    public static final Codec<Supplier<ConfiguredSurfaceBuilder<?>>> f_74763_ = RegistryFileCodec.m_135589_(Registry.f_122879_, f_74762_);
    public final SurfaceBuilder<SC> f_74764_;
