--- a/net/minecraft/data/structures/NbtToSnbt.java
+++ b/net/minecraft/data/structures/NbtToSnbt.java
@@ -26,10 +_,10 @@
       Path path = this.f_126422_.m_123916_();
 
       for(Path path1 : this.f_126422_.m_123913_()) {
-         Files.walk(path1).filter((p_126430_) -> {
-            return p_126430_.toString().endsWith(".nbt");
-         }).forEach((p_126441_) -> {
-            m_126431_(p_126441_, this.m_126435_(path1, p_126441_), path);
+         Files.walk(path1).filter((p_200416_0_) -> {
+            return p_200416_0_.toString().endsWith(".nbt");
+         }).forEach((p_200415_3_) -> {
+            m_126431_(p_200415_3_, this.m_126435_(path1, p_200415_3_), path);
          });
       }
 
