--- a/net/minecraft/data/structures/SnbtToNbt.java
+++ b/net/minecraft/data/structures/SnbtToNbt.java
@@ -57,17 +_,17 @@
       List<CompletableFuture<SnbtToNbt.TaskResult>> list = Lists.newArrayList();
 
       for(Path path1 : this.f_126444_.m_123913_()) {
-         Files.walk(path1).filter((p_126464_) -> {
-            return p_126464_.toString().endsWith(".snbt");
-         }).forEach((p_126474_) -> {
+         Files.walk(path1).filter((p_200422_0_) -> {
+            return p_200422_0_.toString().endsWith(".snbt");
+         }).forEach((p_229447_3_) -> {
             list.add(CompletableFuture.supplyAsync(() -> {
-               return this.m_126465_(p_126474_, this.m_126468_(path1, p_126474_));
+               return this.m_126465_(p_229447_3_, this.m_126468_(path1, p_229447_3_));
             }, Util.m_137578_()));
          });
       }
 
-      Util.m_137567_(list).join().stream().filter(Objects::nonNull).forEach((p_126455_) -> {
-         this.m_126456_(p_126451_, p_126455_, path);
+      Util.m_137567_(list).join().stream().filter(Objects::nonNull).forEach((p_229445_3_) -> {
+         this.m_126456_(p_126451_, p_229445_3_, path);
       });
    }
 
