--- a/net/minecraft/world/entity/monster/piglin/PiglinBruteAi.java
+++ b/net/minecraft/world/entity/monster/piglin/PiglinBruteAi.java
@@ -54,8 +_,8 @@
    }
 
    private static void m_35124_(PiglinBrute p_35125_, Brain<PiglinBrute> p_35126_) {
-      p_35126_.m_21895_(Activity.f_37988_, 10, ImmutableList.of(new StopAttackingIfTargetInvalid<>((p_35118_) -> {
-         return !m_35088_(p_35125_, p_35118_);
+      p_35126_.m_21895_(Activity.f_37988_, 10, ImmutableList.of(new StopAttackingIfTargetInvalid<>((p_242361_1_) -> {
+         return !m_35088_(p_35125_, p_242361_1_);
       }), new SetWalkTargetFromAttackTargetIfTargetOutOfReach(1.0F), new MeleeAttack(20)), MemoryModuleType.f_26372_);
    }
 
@@ -80,8 +_,8 @@
    }
 
    private static boolean m_35088_(AbstractPiglin p_35089_, LivingEntity p_35090_) {
-      return m_35086_(p_35089_).filter((p_35085_) -> {
-         return p_35085_ == p_35090_;
+      return m_35086_(p_35089_).filter((p_242348_1_) -> {
+         return p_242348_1_ == p_35090_;
       }).isPresent();
    }
 
@@ -100,8 +_,8 @@
    }
 
    private static Optional<? extends LivingEntity> m_35091_(AbstractPiglin p_35092_, MemoryModuleType<? extends LivingEntity> p_35093_) {
-      return p_35092_.m_6274_().m_21952_(p_35093_).filter((p_35108_) -> {
-         return p_35108_.m_19950_(p_35092_, 12.0D);
+      return p_35092_.m_6274_().m_21952_(p_35093_).filter((p_242357_1_) -> {
+         return p_242357_1_.m_19950_(p_35092_, 12.0D);
       });
    }
 
@@ -119,8 +_,8 @@
    }
 
    private static void m_35122_(PiglinBrute p_35123_) {
-      p_35123_.m_6274_().m_21968_().ifPresent((p_35104_) -> {
-         if (p_35104_ == Activity.f_37988_) {
+      p_35123_.m_6274_().m_21968_().ifPresent((p_242355_1_) -> {
+         if (p_242355_1_ == Activity.f_37988_) {
             p_35123_.m_35076_();
          }
 
