--- a/net/minecraft/world/entity/ai/goal/AvoidEntityGoal.java
+++ b/net/minecraft/world/entity/ai/goal/AvoidEntityGoal.java
@@ -25,7 +_,7 @@
    private final TargetingConditions f_25025_;
 
    public AvoidEntityGoal(PathfinderMob p_25027_, Class<T> p_25028_, float p_25029_, double p_25030_, double p_25031_) {
-      this(p_25027_, p_25028_, (p_25052_) -> {
+      this(p_25027_, p_25028_, (p_200828_0_) -> {
          return true;
       }, p_25029_, p_25030_, p_25031_, EntitySelector.f_20406_::test);
    }
@@ -44,7 +_,7 @@
    }
 
    public AvoidEntityGoal(PathfinderMob p_25033_, Class<T> p_25034_, float p_25035_, double p_25036_, double p_25037_, Predicate<LivingEntity> p_25038_) {
-      this(p_25033_, p_25034_, (p_25049_) -> {
+      this(p_25033_, p_25034_, (p_203782_0_) -> {
          return true;
       }, p_25035_, p_25036_, p_25037_, p_25038_);
    }
