--- a/net/minecraft/world/entity/ai/goal/FollowMobGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FollowMobGoal.java
@@ -23,8 +_,8 @@
 
    public FollowMobGoal(Mob p_25271_, double p_25272_, float p_25273_, float p_25274_) {
       this.f_25261_ = p_25271_;
-      this.f_25262_ = (p_25278_) -> {
-         return p_25278_ != null && p_25271_.getClass() != p_25278_.getClass();
+      this.f_25262_ = (p_210291_1_) -> {
+         return p_210291_1_ != null && p_25271_.getClass() != p_210291_1_.getClass();
       };
       this.f_25264_ = p_25272_;
       this.f_25265_ = p_25271_.m_21573_();
