--- a/net/minecraft/world/entity/ai/goal/GoalSelector.java
+++ b/net/minecraft/world/entity/ai/goal/GoalSelector.java
@@ -37,43 +_,43 @@
    }
 
    public void m_25363_(Goal p_25364_) {
-      this.f_25345_.stream().filter((p_25378_) -> {
-         return p_25378_.m_26015_() == p_25364_;
+      this.f_25345_.stream().filter((p_220882_1_) -> {
+         return p_220882_1_.m_26015_() == p_25364_;
       }).filter(WrappedGoal::m_7620_).forEach(WrappedGoal::m_8041_);
-      this.f_25345_.removeIf((p_25367_) -> {
-         return p_25367_.m_26015_() == p_25364_;
+      this.f_25345_.removeIf((p_220884_1_) -> {
+         return p_220884_1_.m_26015_() == p_25364_;
       });
    }
 
    public void m_25373_() {
       ProfilerFiller profilerfiller = this.f_25346_.get();
       profilerfiller.m_6180_("goalCleanup");
-      this.m_25386_().filter((p_25390_) -> {
-         return !p_25390_.m_7620_() || p_25390_.m_7684_().stream().anyMatch(this.f_25347_::contains) || !p_25390_.m_8045_();
+      this.m_25386_().filter((p_220881_1_) -> {
+         return !p_220881_1_.m_7620_() || p_220881_1_.m_7684_().stream().anyMatch(this.f_25347_::contains) || !p_220881_1_.m_8045_();
       }).forEach(Goal::m_8041_);
-      this.f_25344_.forEach((p_25358_, p_25359_) -> {
-         if (!p_25359_.m_7620_()) {
-            this.f_25344_.remove(p_25358_);
+      this.f_25344_.forEach((p_220885_1_, p_220885_2_) -> {
+         if (!p_220885_2_.m_7620_()) {
+            this.f_25344_.remove(p_220885_1_);
          }
 
       });
       profilerfiller.m_7238_();
       profilerfiller.m_6180_("goalUpdate");
-      this.f_25345_.stream().filter((p_25388_) -> {
-         return !p_25388_.m_7620_();
-      }).filter((p_25385_) -> {
-         return p_25385_.m_7684_().stream().noneMatch(this.f_25347_::contains);
-      }).filter((p_25380_) -> {
-         return p_25380_.m_7684_().stream().allMatch((p_25383_) -> {
-            return this.f_25344_.getOrDefault(p_25383_, f_25343_).m_26002_(p_25380_);
+      this.f_25345_.stream().filter((p_220883_0_) -> {
+         return !p_220883_0_.m_7620_();
+      }).filter((p_220879_1_) -> {
+         return p_220879_1_.m_7684_().stream().noneMatch(this.f_25347_::contains);
+      }).filter((p_220889_1_) -> {
+         return p_220889_1_.m_7684_().stream().allMatch((p_220887_2_) -> {
+            return this.f_25344_.getOrDefault(p_220887_2_, f_25343_).m_26002_(p_220889_1_);
          });
-      }).filter(WrappedGoal::m_8036_).forEach((p_25369_) -> {
-         p_25369_.m_7684_().forEach((p_25372_) -> {
-            WrappedGoal wrappedgoal = this.f_25344_.getOrDefault(p_25372_, f_25343_);
+      }).filter(WrappedGoal::m_8036_).forEach((p_220877_1_) -> {
+         p_220877_1_.m_7684_().forEach((p_220876_2_) -> {
+            WrappedGoal wrappedgoal = this.f_25344_.getOrDefault(p_220876_2_, f_25343_);
             wrappedgoal.m_8041_();
-            this.f_25344_.put(p_25372_, p_25369_);
+            this.f_25344_.put(p_220876_2_, p_220877_1_);
          });
-         p_25369_.m_8056_();
+         p_220877_1_.m_8056_();
       });
       profilerfiller.m_7238_();
       profilerfiller.m_6180_("goalTick");
