--- a/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
+++ b/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
@@ -21,8 +_,8 @@
 
    public boolean m_8036_() {
       if (!this.f_25497_.m_21523_() && !this.f_25497_.m_30811_()) {
-         List<Entity> list = this.f_25497_.f_19853_.m_6249_(this.f_25497_, this.f_25497_.m_20191_().m_82377_(9.0D, 4.0D, 9.0D), (p_25505_) -> {
-            EntityType<?> entitytype = p_25505_.m_6095_();
+         List<Entity> list = this.f_25497_.f_19853_.m_6249_(this.f_25497_, this.f_25497_.m_20191_().m_82377_(9.0D, 4.0D, 9.0D), (p_220719_0_) -> {
+            EntityType<?> entitytype = p_220719_0_.m_6095_();
             return entitytype == EntityType.f_20466_ || entitytype == EntityType.f_20488_;
          });
          Llama llama = null;
