--- a/net/minecraft/world/entity/ai/goal/PathfindToRaidGoal.java
+++ b/net/minecraft/world/entity/ai/goal/PathfindToRaidGoal.java
@@ -48,8 +_,8 @@
    private void m_25708_(Raid p_25709_) {
       if (p_25709_.m_37782_()) {
          Set<Raider> set = Sets.newHashSet();
-         List<Raider> list = this.f_25704_.f_19853_.m_6443_(Raider.class, this.f_25704_.m_20191_().m_82400_(16.0D), (p_25712_) -> {
-            return !p_25712_.m_37886_() && Raids.m_37965_(p_25712_, p_25709_);
+         List<Raider> list = this.f_25704_.f_19853_.m_6443_(Raider.class, this.f_25704_.m_20191_().m_82400_(16.0D), (p_220742_1_) -> {
+            return !p_220742_1_.m_37886_() && Raids.m_37965_(p_220742_1_, p_25709_);
          });
          set.addAll(list);
 
