--- a/net/minecraft/world/entity/ai/goal/StrollThroughVillageGoal.java
+++ b/net/minecraft/world/entity/ai/goal/StrollThroughVillageGoal.java
@@ -37,8 +_,8 @@
          if (!serverlevel.m_8736_(blockpos, 6)) {
             return false;
          } else {
-            Vec3 vec3 = RandomPos.m_26939_(this.f_25903_, 15, 7, (p_25912_) -> {
-               return (double)(-serverlevel.m_8828_(SectionPos.m_123199_(p_25912_)));
+            Vec3 vec3 = RandomPos.m_26939_(this.f_25903_, 15, 7, (p_220755_1_) -> {
+               return (double)(-serverlevel.m_8828_(SectionPos.m_123199_(p_220755_1_)));
             });
             this.f_25905_ = vec3 == null ? null : new BlockPos(vec3);
             return this.f_25905_ != null;
