--- a/net/minecraft/nbt/TagParser.java
+++ b/net/minecraft/nbt/TagParser.java
@@ -15,14 +_,14 @@
    public static final SimpleCommandExceptionType f_129334_ = new SimpleCommandExceptionType(new TranslatableComponent("argument.nbt.trailing"));
    public static final SimpleCommandExceptionType f_129335_ = new SimpleCommandExceptionType(new TranslatableComponent("argument.nbt.expected.key"));
    public static final SimpleCommandExceptionType f_129336_ = new SimpleCommandExceptionType(new TranslatableComponent("argument.nbt.expected.value"));
-   public static final Dynamic2CommandExceptionType f_129337_ = new Dynamic2CommandExceptionType((p_129366_, p_129367_) -> {
-      return new TranslatableComponent("argument.nbt.list.mixed", p_129366_, p_129367_);
-   });
-   public static final Dynamic2CommandExceptionType f_129338_ = new Dynamic2CommandExceptionType((p_129357_, p_129358_) -> {
-      return new TranslatableComponent("argument.nbt.array.mixed", p_129357_, p_129358_);
-   });
-   public static final DynamicCommandExceptionType f_129339_ = new DynamicCommandExceptionType((p_129355_) -> {
-      return new TranslatableComponent("argument.nbt.array.invalid", p_129355_);
+   public static final Dynamic2CommandExceptionType f_129337_ = new Dynamic2CommandExceptionType((p_208775_0_, p_208775_1_) -> {
+      return new TranslatableComponent("argument.nbt.list.mixed", p_208775_0_, p_208775_1_);
+   });
+   public static final Dynamic2CommandExceptionType f_129338_ = new Dynamic2CommandExceptionType((p_208774_0_, p_208774_1_) -> {
+      return new TranslatableComponent("argument.nbt.array.mixed", p_208774_0_, p_208774_1_);
+   });
+   public static final DynamicCommandExceptionType f_129339_ = new DynamicCommandExceptionType((p_208773_0_) -> {
+      return new TranslatableComponent("argument.nbt.array.invalid", p_208773_0_);
    });
    private static final Pattern f_129340_ = Pattern.compile("[-+]?(?:[0-9]+[.]|[0-9]*[.][0-9]+)(?:e[-+]?[0-9]+)?", 2);
    private static final Pattern f_129341_ = Pattern.compile("[-+]?(?:[0-9]+[.]?|[0-9]*[.][0-9]+)(?:e[-+]?[0-9]+)?d", 2);
