--- a/net/minecraft/server/packs/repository/PackRepository.java
+++ b/net/minecraft/server/packs/repository/PackRepository.java
@@ -23,7 +_,7 @@
 
    public PackRepository(Pack.PackConstructor p_10502_, RepositorySource... p_10503_) {
       this.f_10500_ = p_10502_;
-      this.f_10497_ = ImmutableSet.copyOf(p_10503_);
+      this.f_10497_ = new java.util.HashSet<>(java.util.Arrays.asList(p_10503_));
    }
 
    public PackRepository(RepositorySource... p_10505_) {
@@ -88,6 +_,10 @@
    @Nullable
    public Pack m_10507_(String p_10508_) {
       return this.f_10498_.get(p_10508_);
+   }
+
+   public void addPackFinder(RepositorySource packFinder) {
+      this.f_10497_.add(packFinder);
    }
 
    public void close() {
