--- a/net/minecraft/server/packs/repository/Pack.java
+++ b/net/minecraft/server/packs/repository/Pack.java
@@ -31,6 +_,7 @@
    private final Pack.Position f_10406_;
    private final boolean f_10407_;
    private final boolean f_10408_;
+   private final boolean hidden; // Forge: Allow packs to be hidden from the UI entirely
    private final PackSource f_10409_;
 
    @Nullable
@@ -54,7 +_,12 @@
       return null;
    }
 
+   @Deprecated
    public Pack(String p_10420_, boolean p_10421_, Supplier<PackResources> p_10422_, Component p_10423_, Component p_10424_, PackCompatibility p_10425_, Pack.Position p_10426_, boolean p_10427_, PackSource p_10428_) {
+       this(p_10420_, p_10421_, p_10422_, p_10423_, p_10424_, p_10425_, p_10426_, p_10427_, p_10428_, false);
+   }
+
+   public Pack(String p_10420_, boolean p_10421_, Supplier<PackResources> p_10422_, Component p_10423_, Component p_10424_, PackCompatibility p_10425_, Pack.Position p_10426_, boolean p_10427_, PackSource p_10428_, boolean hidden) {
       this.f_10401_ = p_10420_;
       this.f_10402_ = p_10422_;
       this.f_10403_ = p_10423_;
@@ -64,10 +_,16 @@
       this.f_10406_ = p_10426_;
       this.f_10408_ = p_10427_;
       this.f_10409_ = p_10428_;
+      this.hidden = hidden;
    }
 
+   @Deprecated
    public Pack(String p_10412_, boolean p_10413_, Supplier<PackResources> p_10414_, PackResources p_10415_, PackMetadataSection p_10416_, Pack.Position p_10417_, PackSource p_10418_) {
-      this(p_10412_, p_10413_, p_10414_, new TextComponent(p_10415_.m_8017_()), p_10416_.m_10373_(), PackCompatibility.m_10490_(p_10416_.m_10374_()), p_10417_, false, p_10418_);
+      this(p_10412_, p_10413_, p_10414_, p_10415_, p_10416_, p_10417_, p_10418_, false);
+   }
+
+   public Pack(String p_10412_, boolean p_10413_, Supplier<PackResources> p_10414_, PackResources p_10415_, PackMetadataSection p_10416_, Pack.Position p_10417_, PackSource p_10418_, boolean hidden) {
+      this(p_10412_, p_10413_, p_10414_, new TextComponent(p_10415_.m_8017_()), p_10416_.m_10373_(), PackCompatibility.m_10490_(p_10416_.m_10374_()), p_10417_, false, p_10418_, hidden);
    }
 
    @OnlyIn(Dist.CLIENT)
@@ -114,6 +_,8 @@
    public PackSource m_10453_() {
       return this.f_10409_;
    }
+
+   public boolean isHidden() { return hidden; }
 
    public boolean equals(Object p_10448_) {
       if (this == p_10448_) {
