--- a/net/minecraft/world/item/BottleItem.java
+++ b/net/minecraft/world/item/BottleItem.java
@@ -24,8 +_,8 @@
    }
 
    public InteractionResultHolder<ItemStack> m_7203_(Level p_40656_, Player p_40657_, InteractionHand p_40658_) {
-      List<AreaEffectCloud> list = p_40656_.m_6443_(AreaEffectCloud.class, p_40657_.m_20191_().m_82400_(2.0D), (p_40650_) -> {
-         return p_40650_ != null && p_40650_.m_6084_() && p_40650_.m_19749_() instanceof EnderDragon;
+      List<AreaEffectCloud> list = p_40656_.m_6443_(AreaEffectCloud.class, p_40657_.m_20191_().m_82400_(2.0D), (p_210311_0_) -> {
+         return p_210311_0_ != null && p_210311_0_.m_6084_() && p_210311_0_.m_19749_() instanceof EnderDragon;
       });
       ItemStack itemstack = p_40657_.m_21120_(p_40658_);
       if (!list.isEmpty()) {
