--- a/net/minecraft/world/item/FlintAndSteelItem.java
+++ b/net/minecraft/world/item/FlintAndSteelItem.java
@@ -28,8 +_,8 @@
          level.m_5594_(player, blockpos, SoundEvents.f_11942_, SoundSource.BLOCKS, 1.0F, f_41376_.nextFloat() * 0.4F + 0.8F);
          level.m_7731_(blockpos, blockstate.m_61124_(BlockStateProperties.f_61443_, Boolean.valueOf(true)), 11);
          if (player != null) {
-            p_41297_.m_43722_().m_41622_(1, player, (p_41303_) -> {
-               p_41303_.m_21190_(p_41297_.m_43724_());
+            p_41297_.m_43722_().m_41622_(1, player, (p_219999_1_) -> {
+               p_219999_1_.m_21190_(p_41297_.m_43724_());
             });
          }
 
@@ -43,8 +_,8 @@
             ItemStack itemstack = p_41297_.m_43722_();
             if (player instanceof ServerPlayer) {
                CriteriaTriggers.f_10591_.m_59469_((ServerPlayer)player, blockpos1, itemstack);
-               itemstack.m_41622_(1, player, (p_41300_) -> {
-                  p_41300_.m_21190_(p_41297_.m_43724_());
+               itemstack.m_41622_(1, player, (p_219998_1_) -> {
+                  p_219998_1_.m_21190_(p_41297_.m_43724_());
                });
             }
 
