--- a/net/minecraft/world/item/ProjectileWeaponItem.java
+++ b/net/minecraft/world/item/ProjectileWeaponItem.java
@@ -6,11 +_,11 @@
 import net.minecraft.world.entity.LivingEntity;
 
 public abstract class ProjectileWeaponItem extends Item {
-   public static final Predicate<ItemStack> f_43005_ = (p_43017_) -> {
-      return p_43017_.m_41720_().m_41384_(ItemTags.f_13161_);
+   public static final Predicate<ItemStack> f_43005_ = (p_220002_0_) -> {
+      return p_220002_0_.m_41720_().m_41384_(ItemTags.f_13161_);
    };
-   public static final Predicate<ItemStack> f_43006_ = f_43005_.or((p_43015_) -> {
-      return p_43015_.m_41720_() == Items.f_42688_;
+   public static final Predicate<ItemStack> f_43006_ = f_43005_.or((p_220003_0_) -> {
+      return p_220003_0_.m_41720_() == Items.f_42688_;
    });
 
    public ProjectileWeaponItem(Item.Properties p_43009_) {
