--- a/net/minecraft/world/item/SwordItem.java
+++ b/net/minecraft/world/item/SwordItem.java
@@ -47,16 +_,16 @@
    }
 
    public boolean m_7579_(ItemStack p_43278_, LivingEntity p_43279_, LivingEntity p_43280_) {
-      p_43278_.m_41622_(1, p_43280_, (p_43296_) -> {
-         p_43296_.m_21166_(EquipmentSlot.MAINHAND);
+      p_43278_.m_41622_(1, p_43280_, (p_220045_0_) -> {
+         p_220045_0_.m_21166_(EquipmentSlot.MAINHAND);
       });
       return true;
    }
 
    public boolean m_6813_(ItemStack p_43282_, Level p_43283_, BlockState p_43284_, BlockPos p_43285_, LivingEntity p_43286_) {
       if (p_43284_.m_60800_(p_43283_, p_43285_) != 0.0F) {
-         p_43282_.m_41622_(2, p_43286_, (p_43276_) -> {
-            p_43276_.m_21166_(EquipmentSlot.MAINHAND);
+         p_43282_.m_41622_(2, p_43286_, (p_220044_0_) -> {
+            p_220044_0_.m_21166_(EquipmentSlot.MAINHAND);
          });
       }
 
