--- a/net/minecraft/world/item/TridentItem.java
+++ b/net/minecraft/world/item/TridentItem.java
@@ -56,8 +_,8 @@
             int j = EnchantmentHelper.m_44932_(p_43394_);
             if (j <= 0 || player.m_20070_()) {
                if (!p_43395_.f_46443_) {
-                  p_43394_.m_41622_(1, player, (p_43388_) -> {
-                     p_43388_.m_21190_(p_43396_.m_7655_());
+                  p_43394_.m_41622_(1, player, (p_220047_1_) -> {
+                     p_220047_1_.m_21190_(p_43396_.m_7655_());
                   });
                   if (j == 0) {
                      ThrownTrident throwntrident = new ThrownTrident(p_43395_, player, p_43394_);
@@ -123,16 +_,16 @@
    }
 
    public boolean m_7579_(ItemStack p_43390_, LivingEntity p_43391_, LivingEntity p_43392_) {
-      p_43390_.m_41622_(1, p_43392_, (p_43414_) -> {
-         p_43414_.m_21166_(EquipmentSlot.MAINHAND);
+      p_43390_.m_41622_(1, p_43392_, (p_220048_0_) -> {
+         p_220048_0_.m_21166_(EquipmentSlot.MAINHAND);
       });
       return true;
    }
 
    public boolean m_6813_(ItemStack p_43399_, Level p_43400_, BlockState p_43401_, BlockPos p_43402_, LivingEntity p_43403_) {
       if ((double)p_43401_.m_60800_(p_43400_, p_43402_) != 0.0D) {
-         p_43399_.m_41622_(2, p_43403_, (p_43385_) -> {
-            p_43385_.m_21166_(EquipmentSlot.MAINHAND);
+         p_43399_.m_41622_(2, p_43403_, (p_220046_0_) -> {
+            p_220046_0_.m_21166_(EquipmentSlot.MAINHAND);
          });
       }
 
