--- a/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -213,7 +_,7 @@
       if (this.m_31502_() > 0) {
          int j1 = this.m_31502_() - 1;
          if (j1 <= 0) {
-            Explosion.BlockInteraction explosion$blockinteraction = this.f_19853_.m_46469_().m_46207_(GameRules.f_46132_) ? Explosion.BlockInteraction.DESTROY : Explosion.BlockInteraction.NONE;
+            Explosion.BlockInteraction explosion$blockinteraction = net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(this.f_19853_, this) ? Explosion.BlockInteraction.DESTROY : Explosion.BlockInteraction.NONE;
             this.f_19853_.m_46518_(this, this.m_20185_(), this.m_20188_(), this.m_20189_(), 7.0F, false, explosion$blockinteraction);
             if (!this.m_20067_()) {
                this.f_19853_.m_6798_(1023, this.m_20183_(), 0);
@@ -290,7 +_,7 @@
 
          if (this.f_31429_ > 0) {
             --this.f_31429_;
-            if (this.f_31429_ == 0 && this.f_19853_.m_46469_().m_46207_(GameRules.f_46132_)) {
+            if (this.f_31429_ == 0 && net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(this.f_19853_, this)) {
                int i1 = Mth.m_14107_(this.m_20186_());
                int l1 = Mth.m_14107_(this.m_20185_());
                int i2 = Mth.m_14107_(this.m_20189_());
@@ -304,7 +_,7 @@
                         int l = i2 + l2;
                         BlockPos blockpos = new BlockPos(i3, k, l);
                         BlockState blockstate = this.f_19853_.m_8055_(blockpos);
-                        if (m_31491_(blockstate)) {
+                        if (blockstate.canEntityDestroy(this.f_19853_, blockpos, this) && net.minecraftforge.event.ForgeEventFactory.onEntityDestroyBlock(this, blockpos, blockstate)) {
                            flag = this.f_19853_.m_46953_(blockpos, true, this) || flag;
                         }
                      }
@@ -325,6 +_,7 @@
       }
    }
 
+   @Deprecated //Forge: DO NOT USE use BlockState.canEntityDestroy
    public static boolean m_31491_(BlockState p_31492_) {
       return !p_31492_.m_60795_() && !BlockTags.f_13070_.m_8110_(p_31492_.m_60734_());
    }
