--- a/net/minecraft/server/rcon/thread/QueryThreadGs4.java
+++ b/net/minecraft/server/rcon/thread/QueryThreadGs4.java
@@ -202,8 +_,8 @@
          long i = Util.m_137550_();
          if (i >= this.f_11525_ + 30000L) {
             this.f_11525_ = i;
-            this.f_11535_.values().removeIf((p_11546_) -> {
-               return p_11546_.m_11575_(i);
+            this.f_11535_.values().removeIf((p_232650_2_) -> {
+               return p_232650_2_.m_11575_(i);
             });
          }
       }
