--- a/net/minecraft/commands/arguments/blocks/BlockStateParser.java
+++ b/net/minecraft/commands/arguments/blocks/BlockStateParser.java
@@ -169,7 +_,7 @@
 
    private boolean m_116767_(TagCollection<Block> p_116768_) {
       if (this.f_116755_ != null) {
-         return this.f_116755_.m_60734_().m_60588_();
+         return this.f_116755_.hasTileEntity();
       } else {
          if (this.f_116757_ != null) {
             Tag<Block> tag = p_116768_.m_13404_(this.f_116757_);
@@ -209,7 +_,7 @@
    private static <T extends Comparable<T>> SuggestionsBuilder m_116786_(SuggestionsBuilder p_116787_, Property<T> p_116788_) {
       for(T t : p_116788_.m_6908_()) {
          if (t instanceof Integer) {
-            p_116787_.suggest((Integer)t);
+            p_116787_.suggest((Integer) t);
          } else {
             p_116787_.suggest(p_116788_.m_6940_(t));
          }
@@ -283,7 +_,7 @@
             p_116848_.suggest(String.valueOf('['));
          }
 
-         if (this.f_116755_.m_60734_().m_60588_()) {
+         if (this.f_116755_.hasTileEntity()) {
             p_116848_.suggest(String.valueOf('{'));
          }
       }
