--- a/net/minecraft/commands/arguments/blocks/BlockStateParser.java
+++ b/net/minecraft/commands/arguments/blocks/BlockStateParser.java
@@ -169,7 +_,7 @@
 
    private boolean m_116767_(TagCollection<Block> p_116768_) {
       if (this.f_116755_ != null) {
-         return this.f_116755_.m_60734_().m_60588_();
+         return this.f_116755_.hasTileEntity();
       } else {
          if (this.f_116757_ != null) {
             Tag<Block> tag = p_116768_.m_13404_(this.f_116757_);
@@ -283,7 +_,7 @@
             p_116848_.suggest(String.valueOf('['));
          }
 
-         if (this.f_116755_.m_60734_().m_60588_()) {
+         if (this.f_116755_.hasTileEntity()) {
             p_116848_.suggest(String.valueOf('{'));
          }
       }
