--- a/com/mojang/realmsclient/RealmsMainScreen.java
+++ b/com/mojang/realmsclient/RealmsMainScreen.java
@@ -236,38 +_,38 @@
    }
 
    public void m_86570_() {
-      this.f_86289_ = this.m_7082_(new Button(this.f_96543_ / 2 - 202, this.f_96544_ - 32, 90, 20, new TranslatableComponent("mco.selectServer.leave"), (p_86679_) -> {
+      this.f_86289_ = this.m_7082_(new Button(this.f_96543_ / 2 - 202, this.f_96544_ - 32, 90, 20, new TranslatableComponent("mco.selectServer.leave"), (p_237624_1_) -> {
          this.m_86669_(this.m_86404_(this.f_86284_));
       }));
-      this.f_86288_ = this.m_7082_(new Button(this.f_96543_ / 2 - 190, this.f_96544_ - 32, 90, 20, new TranslatableComponent("mco.selectServer.configure"), (p_86672_) -> {
+      this.f_86288_ = this.m_7082_(new Button(this.f_96543_ / 2 - 190, this.f_96544_ - 32, 90, 20, new TranslatableComponent("mco.selectServer.configure"), (p_237637_1_) -> {
          this.m_86656_(this.m_86404_(this.f_86284_));
       }));
-      this.f_86285_ = this.m_7082_(new Button(this.f_96543_ / 2 - 93, this.f_96544_ - 32, 90, 20, new TranslatableComponent("mco.selectServer.play"), (p_86659_) -> {
+      this.f_86285_ = this.m_7082_(new Button(this.f_96543_ / 2 - 93, this.f_96544_ - 32, 90, 20, new TranslatableComponent("mco.selectServer.play"), (p_237635_1_) -> {
          RealmsServer realmsserver1 = this.m_86404_(this.f_86284_);
          if (realmsserver1 != null) {
             this.m_86515_(realmsserver1, this);
          }
       }));
-      this.f_86286_ = this.m_7082_(new Button(this.f_96543_ / 2 + 4, this.f_96544_ - 32, 90, 20, CommonComponents.f_130660_, (p_86647_) -> {
+      this.f_86286_ = this.m_7082_(new Button(this.f_96543_ / 2 + 4, this.f_96544_ - 32, 90, 20, CommonComponents.f_130660_, (p_237632_1_) -> {
          if (!this.f_86296_) {
             this.f_96541_.m_91152_(this.f_86282_);
          }
 
       }));
-      this.f_86287_ = this.m_7082_(new Button(this.f_96543_ / 2 + 100, this.f_96544_ - 32, 90, 20, new TranslatableComponent("mco.selectServer.expiredRenew"), (p_86622_) -> {
+      this.f_86287_ = this.m_7082_(new Button(this.f_96543_ / 2 + 100, this.f_96544_ - 32, 90, 20, new TranslatableComponent("mco.selectServer.expiredRenew"), (p_237629_1_) -> {
          this.m_86333_();
       }));
       this.f_86269_ = this.m_7082_(new RealmsMainScreen.PendingInvitesButton());
       this.f_86270_ = this.m_7082_(new RealmsMainScreen.NewsButton());
       this.f_86268_ = this.m_7082_(new RealmsMainScreen.ShowPopupButton());
       this.f_86273_ = this.m_7082_(new RealmsMainScreen.CloseButton());
-      this.f_86271_ = this.m_7082_(new Button(this.f_96543_ / 2 + 52, this.m_86366_() + 137 - 20, 98, 20, new TranslatableComponent("mco.selectServer.trial"), (p_86597_) -> {
+      this.f_86271_ = this.m_7082_(new Button(this.f_96543_ / 2 + 52, this.m_86366_() + 137 - 20, 98, 20, new TranslatableComponent("mco.selectServer.trial"), (p_237618_1_) -> {
          if (this.f_86297_ && !this.f_86298_) {
             Util.m_137581_().m_137646_("https://aka.ms/startjavarealmstrial");
             this.f_96541_.m_91152_(this.f_86282_);
          }
       }));
-      this.f_86272_ = this.m_7082_(new Button(this.f_96543_ / 2 + 52, this.m_86366_() + 160 - 20, 98, 20, new TranslatableComponent("mco.selectServer.buy"), (p_86565_) -> {
+      this.f_86272_ = this.m_7082_(new Button(this.f_96543_ / 2 + 52, this.m_86366_() + 160 - 20, 98, 20, new TranslatableComponent("mco.selectServer.buy"), (p_237612_0_) -> {
          Util.m_137581_().m_137646_("https://aka.ms/BuyJavaRealms");
       }));
       RealmsServer realmsserver = this.m_86404_(this.f_86284_);
@@ -649,8 +_,8 @@
    private void m_86676_(RealmsServer p_86677_) {
       f_86255_.m_87818_(p_86677_);
       this.f_86291_.remove(p_86677_);
-      this.f_86283_.m_6702_().removeIf((p_86447_) -> {
-         return p_86447_ instanceof RealmsMainScreen.ServerEntry && ((RealmsMainScreen.ServerEntry)p_86447_).f_86853_.f_87473_ == this.f_86284_;
+      this.f_86283_.m_6702_().removeIf((p_243041_1_) -> {
+         return p_243041_1_ instanceof RealmsMainScreen.ServerEntry && ((RealmsMainScreen.ServerEntry)p_243041_1_).f_86853_.f_87473_ == this.f_86284_;
       });
       this.f_86283_.m_6987_((RealmsMainScreen.Entry)null);
       this.m_86513_((RealmsServer)null);
@@ -682,8 +_,8 @@
    }
 
    public boolean m_5534_(char p_86388_, int p_86389_) {
-      this.f_86263_.forEach((p_86392_) -> {
-         p_86392_.m_86228_(p_86388_);
+      this.f_86263_.forEach((p_237578_1_) -> {
+         p_237578_1_.m_86228_(p_86388_);
       });
       return true;
    }
@@ -1091,11 +_,11 @@
    }
 
    public static void m_86406_(ResourceManager p_86407_) {
-      Collection<ResourceLocation> collection = p_86407_.m_6540_("textures/gui/images", (p_86567_) -> {
-         return p_86567_.endsWith(".png");
+      Collection<ResourceLocation> collection = p_86407_.m_6540_("textures/gui/images", (p_227934_0_) -> {
+         return p_227934_0_.endsWith(".png");
       });
-      f_86254_ = collection.stream().filter((p_86523_) -> {
-         return p_86523_.m_135827_().equals("realms");
+      f_86254_ = collection.stream().filter((p_227931_0_) -> {
+         return p_227931_0_.m_135827_().equals("realms");
       }).collect(ImmutableList.toImmutableList());
    }
 
@@ -1131,26 +_,32 @@
    }
 
    @OnlyIn(Dist.CLIENT)
+   class ShowPopupButton extends Button {
+      public ShowPopupButton() {
+         super(RealmsMainScreen.this.f_96543_ - 37, 6, 20, 20, new TranslatableComponent("mco.selectServer.info"), null);
+      }
+      @Override
+      public void m_5691_() {
+            RealmsMainScreen.this.f_86295_ = !RealmsMainScreen.this.f_86295_;
+      }
+
+      public void m_6303_(PoseStack p_86899_, int p_86900_, int p_86901_, float p_86902_) {
+         RealmsMainScreen.this.m_86417_(p_86899_, p_86900_, p_86901_, this.f_93620_, this.f_93621_, this.m_5702_());
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
    abstract class Entry extends ObjectSelectionList.Entry<RealmsMainScreen.Entry> {
       private Entry() {
       }
    }
 
    @OnlyIn(Dist.CLIENT)
-   static enum HoveredElement {
-      NONE,
-      EXPIRED,
-      LEAVE,
-      CONFIGURE;
-   }
-
-   @OnlyIn(Dist.CLIENT)
    class NewsButton extends Button {
       public NewsButton() {
          super(RealmsMainScreen.this.f_96543_ - 62, 6, 20, 20, TextComponent.f_131282_, null);
          this.m_93666_(new TranslatableComponent("mco.news"));
       }
-
       @Override
       public void m_5691_() {
             if (RealmsMainScreen.this.f_86259_ != null) {
@@ -1191,155 +_,6 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   class RealmSelectionList extends RealmsObjectSelectionList<RealmsMainScreen.Entry> {
-      private boolean f_86823_;
-
-      public RealmSelectionList() {
-         super(RealmsMainScreen.this.f_96543_, RealmsMainScreen.this.f_96544_, 32, RealmsMainScreen.this.f_96544_ - 40, 36);
-      }
-
-      public void m_7178_() {
-         super.m_7178_();
-         this.f_86823_ = false;
-      }
-
-      public int m_86843_(RealmsMainScreen.Entry p_86844_) {
-         this.f_86823_ = true;
-         return this.m_7085_(p_86844_);
-      }
-
-      public boolean m_5694_() {
-         return RealmsMainScreen.this.m_7222_() == this;
-      }
-
-      public boolean m_7933_(int p_86840_, int p_86841_, int p_86842_) {
-         if (p_86840_ != 257 && p_86840_ != 32 && p_86840_ != 335) {
-            return super.m_7933_(p_86840_, p_86841_, p_86842_);
-         } else {
-            ObjectSelectionList.Entry objectselectionlist$entry = this.m_93511_();
-            return objectselectionlist$entry == null ? super.m_7933_(p_86840_, p_86841_, p_86842_) : objectselectionlist$entry.m_6375_(0.0D, 0.0D, 0);
-         }
-      }
-
-      public boolean m_6375_(double p_86828_, double p_86829_, int p_86830_) {
-         if (p_86830_ == 0 && p_86828_ < (double)this.m_5756_() && p_86829_ >= (double)this.f_93390_ && p_86829_ <= (double)this.f_93391_) {
-            int i = RealmsMainScreen.this.f_86283_.m_5747_();
-            int j = this.m_5756_();
-            int k = (int)Math.floor(p_86829_ - (double)this.f_93390_) - this.f_93395_ + (int)this.m_93517_() - 4;
-            int l = k / this.f_93387_;
-            if (p_86828_ >= (double)i && p_86828_ <= (double)j && l >= 0 && k >= 0 && l < this.m_5773_()) {
-               this.m_7980_(k, l, p_86828_, p_86829_, this.f_93388_);
-               RealmsMainScreen.this.f_86264_ = RealmsMainScreen.this.f_86264_ + 7;
-               this.m_7109_(l);
-            }
-
-            return true;
-         } else {
-            return super.m_6375_(p_86828_, p_86829_, p_86830_);
-         }
-      }
-
-      public void m_7109_(int p_86832_) {
-         this.m_120767_(p_86832_);
-         if (p_86832_ != -1) {
-            RealmsServer realmsserver;
-            if (this.f_86823_) {
-               if (p_86832_ == 0) {
-                  realmsserver = null;
-               } else {
-                  if (p_86832_ - 1 >= RealmsMainScreen.this.f_86291_.size()) {
-                     RealmsMainScreen.this.f_86284_ = -1L;
-                     return;
-                  }
-
-                  realmsserver = RealmsMainScreen.this.f_86291_.get(p_86832_ - 1);
-               }
-            } else {
-               if (p_86832_ >= RealmsMainScreen.this.f_86291_.size()) {
-                  RealmsMainScreen.this.f_86284_ = -1L;
-                  return;
-               }
-
-               realmsserver = RealmsMainScreen.this.f_86291_.get(p_86832_);
-            }
-
-            RealmsMainScreen.this.m_86513_(realmsserver);
-            if (realmsserver == null) {
-               RealmsMainScreen.this.f_86284_ = -1L;
-            } else if (realmsserver.f_87477_ == RealmsServer.State.UNINITIALIZED) {
-               RealmsMainScreen.this.f_86284_ = -1L;
-            } else {
-               RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
-               if (RealmsMainScreen.this.f_86264_ >= 10 && RealmsMainScreen.this.f_86285_.f_93623_) {
-                  RealmsMainScreen.this.m_86515_(RealmsMainScreen.this.m_86404_(RealmsMainScreen.this.f_86284_), RealmsMainScreen.this);
-               }
-
-            }
-         }
-      }
-
-      public void m_6987_(@Nullable RealmsMainScreen.Entry p_86849_) {
-         super.m_6987_(p_86849_);
-         int i = this.m_6702_().indexOf(p_86849_);
-         if (this.f_86823_ && i == 0) {
-            NarrationHelper.m_120672_(I18n.m_118938_("mco.trial.message.line1"), I18n.m_118938_("mco.trial.message.line2"));
-         } else if (!this.f_86823_ || i > 0) {
-            RealmsServer realmsserver = RealmsMainScreen.this.f_86291_.get(i - (this.f_86823_ ? 1 : 0));
-            RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
-            RealmsMainScreen.this.m_86513_(realmsserver);
-            if (realmsserver.f_87477_ == RealmsServer.State.UNINITIALIZED) {
-               NarrationHelper.m_120670_(I18n.m_118938_("mco.selectServer.uninitialized") + I18n.m_118938_("mco.gui.button"));
-            } else {
-               NarrationHelper.m_120670_(I18n.m_118938_("narrator.select", realmsserver.f_87475_));
-            }
-         }
-
-      }
-
-      public void m_7980_(int p_86834_, int p_86835_, double p_86836_, double p_86837_, int p_86838_) {
-         if (this.f_86823_) {
-            if (p_86835_ == 0) {
-               RealmsMainScreen.this.f_86295_ = true;
-               return;
-            }
-
-            --p_86835_;
-         }
-
-         if (p_86835_ < RealmsMainScreen.this.f_86291_.size()) {
-            RealmsServer realmsserver = RealmsMainScreen.this.f_86291_.get(p_86835_);
-            if (realmsserver != null) {
-               if (realmsserver.f_87477_ == RealmsServer.State.UNINITIALIZED) {
-                  RealmsMainScreen.this.f_86284_ = -1L;
-                  Minecraft.m_91087_().m_91152_(new RealmsCreateRealmScreen(realmsserver, RealmsMainScreen.this));
-               } else {
-                  RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
-               }
-
-               if (RealmsMainScreen.this.f_86267_ == RealmsMainScreen.HoveredElement.CONFIGURE) {
-                  RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
-                  RealmsMainScreen.this.m_86656_(realmsserver);
-               } else if (RealmsMainScreen.this.f_86267_ == RealmsMainScreen.HoveredElement.LEAVE) {
-                  RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
-                  RealmsMainScreen.this.m_86669_(realmsserver);
-               } else if (RealmsMainScreen.this.f_86267_ == RealmsMainScreen.HoveredElement.EXPIRED) {
-                  RealmsMainScreen.this.m_86333_();
-               }
-
-            }
-         }
-      }
-
-      public int m_5775_() {
-         return this.m_5773_() * 36;
-      }
-
-      public int m_5759_() {
-         return 300;
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
    class ServerEntry extends RealmsMainScreen.Entry {
       private final RealmsServer f_86853_;
 
@@ -1463,19 +_,160 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   class ShowPopupButton extends Button {
-      public ShowPopupButton() {
-         super(RealmsMainScreen.this.f_96543_ - 37, 6, 20, 20, new TranslatableComponent("mco.selectServer.info"), null);
-      }
-
-      @Override
-      public void m_5691_() {
-            RealmsMainScreen.this.f_86295_ = !RealmsMainScreen.this.f_86295_;
-      }
-
-      public void m_6303_(PoseStack p_86899_, int p_86900_, int p_86901_, float p_86902_) {
-         RealmsMainScreen.this.m_86417_(p_86899_, p_86900_, p_86901_, this.f_93620_, this.f_93621_, this.m_5702_());
-      }
+   class RealmSelectionList extends RealmsObjectSelectionList<RealmsMainScreen.Entry> {
+      private boolean f_86823_;
+
+      public RealmSelectionList() {
+         super(RealmsMainScreen.this.f_96543_, RealmsMainScreen.this.f_96544_, 32, RealmsMainScreen.this.f_96544_ - 40, 36);
+      }
+
+      public void m_7178_() {
+         super.m_7178_();
+         this.f_86823_ = false;
+      }
+
+      public int m_86843_(RealmsMainScreen.Entry p_86844_) {
+         this.f_86823_ = true;
+         return this.m_7085_(p_86844_);
+      }
+
+      public boolean m_5694_() {
+         return RealmsMainScreen.this.m_7222_() == this;
+      }
+
+      public boolean m_7933_(int p_86840_, int p_86841_, int p_86842_) {
+         if (p_86840_ != 257 && p_86840_ != 32 && p_86840_ != 335) {
+            return super.m_7933_(p_86840_, p_86841_, p_86842_);
+         } else {
+            ObjectSelectionList.Entry objectselectionlist$entry = this.m_93511_();
+            return objectselectionlist$entry == null ? super.m_7933_(p_86840_, p_86841_, p_86842_) : objectselectionlist$entry.m_6375_(0.0D, 0.0D, 0);
+         }
+      }
+
+      public boolean m_6375_(double p_86828_, double p_86829_, int p_86830_) {
+         if (p_86830_ == 0 && p_86828_ < (double)this.m_5756_() && p_86829_ >= (double)this.f_93390_ && p_86829_ <= (double)this.f_93391_) {
+            int i = RealmsMainScreen.this.f_86283_.m_5747_();
+            int j = this.m_5756_();
+            int k = (int)Math.floor(p_86829_ - (double)this.f_93390_) - this.f_93395_ + (int)this.m_93517_() - 4;
+            int l = k / this.f_93387_;
+            if (p_86828_ >= (double)i && p_86828_ <= (double)j && l >= 0 && k >= 0 && l < this.m_5773_()) {
+               this.m_7980_(k, l, p_86828_, p_86829_, this.f_93388_);
+               RealmsMainScreen.this.f_86264_ = RealmsMainScreen.this.f_86264_ + 7;
+               this.m_7109_(l);
+            }
+
+            return true;
+         } else {
+            return super.m_6375_(p_86828_, p_86829_, p_86830_);
+         }
+      }
+
+      public void m_7109_(int p_86832_) {
+         this.m_120767_(p_86832_);
+         if (p_86832_ != -1) {
+            RealmsServer realmsserver;
+            if (this.f_86823_) {
+               if (p_86832_ == 0) {
+                  realmsserver = null;
+               } else {
+                  if (p_86832_ - 1 >= RealmsMainScreen.this.f_86291_.size()) {
+                     RealmsMainScreen.this.f_86284_ = -1L;
+                     return;
+                  }
+
+                  realmsserver = RealmsMainScreen.this.f_86291_.get(p_86832_ - 1);
+               }
+            } else {
+               if (p_86832_ >= RealmsMainScreen.this.f_86291_.size()) {
+                  RealmsMainScreen.this.f_86284_ = -1L;
+                  return;
+               }
+
+               realmsserver = RealmsMainScreen.this.f_86291_.get(p_86832_);
+            }
+
+            RealmsMainScreen.this.m_86513_(realmsserver);
+            if (realmsserver == null) {
+               RealmsMainScreen.this.f_86284_ = -1L;
+            } else if (realmsserver.f_87477_ == RealmsServer.State.UNINITIALIZED) {
+               RealmsMainScreen.this.f_86284_ = -1L;
+            } else {
+               RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
+               if (RealmsMainScreen.this.f_86264_ >= 10 && RealmsMainScreen.this.f_86285_.f_93623_) {
+                  RealmsMainScreen.this.m_86515_(RealmsMainScreen.this.m_86404_(RealmsMainScreen.this.f_86284_), RealmsMainScreen.this);
+               }
+
+            }
+         }
+      }
+
+      public void m_6987_(@Nullable RealmsMainScreen.Entry p_86849_) {
+         super.m_6987_(p_86849_);
+         int i = this.m_6702_().indexOf(p_86849_);
+         if (this.f_86823_ && i == 0) {
+            NarrationHelper.m_120672_(I18n.m_118938_("mco.trial.message.line1"), I18n.m_118938_("mco.trial.message.line2"));
+         } else if (!this.f_86823_ || i > 0) {
+            RealmsServer realmsserver = RealmsMainScreen.this.f_86291_.get(i - (this.f_86823_ ? 1 : 0));
+            RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
+            RealmsMainScreen.this.m_86513_(realmsserver);
+            if (realmsserver.f_87477_ == RealmsServer.State.UNINITIALIZED) {
+               NarrationHelper.m_120670_(I18n.m_118938_("mco.selectServer.uninitialized") + I18n.m_118938_("mco.gui.button"));
+            } else {
+               NarrationHelper.m_120670_(I18n.m_118938_("narrator.select", realmsserver.f_87475_));
+            }
+         }
+
+      }
+
+      public void m_7980_(int p_86834_, int p_86835_, double p_86836_, double p_86837_, int p_86838_) {
+         if (this.f_86823_) {
+            if (p_86835_ == 0) {
+               RealmsMainScreen.this.f_86295_ = true;
+               return;
+            }
+
+            --p_86835_;
+         }
+
+         if (p_86835_ < RealmsMainScreen.this.f_86291_.size()) {
+            RealmsServer realmsserver = RealmsMainScreen.this.f_86291_.get(p_86835_);
+            if (realmsserver != null) {
+               if (realmsserver.f_87477_ == RealmsServer.State.UNINITIALIZED) {
+                  RealmsMainScreen.this.f_86284_ = -1L;
+                  Minecraft.m_91087_().m_91152_(new RealmsCreateRealmScreen(realmsserver, RealmsMainScreen.this));
+               } else {
+                  RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
+               }
+
+               if (RealmsMainScreen.this.f_86267_ == RealmsMainScreen.HoveredElement.CONFIGURE) {
+                  RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
+                  RealmsMainScreen.this.m_86656_(realmsserver);
+               } else if (RealmsMainScreen.this.f_86267_ == RealmsMainScreen.HoveredElement.LEAVE) {
+                  RealmsMainScreen.this.f_86284_ = realmsserver.f_87473_;
+                  RealmsMainScreen.this.m_86669_(realmsserver);
+               } else if (RealmsMainScreen.this.f_86267_ == RealmsMainScreen.HoveredElement.EXPIRED) {
+                  RealmsMainScreen.this.m_86333_();
+               }
+
+            }
+         }
+      }
+
+      public int m_5775_() {
+         return this.m_5773_() * 36;
+      }
+
+      public int m_5759_() {
+         return 300;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   static enum HoveredElement {
+      NONE,
+      EXPIRED,
+      LEAVE,
+      CONFIGURE;
    }
 
    @OnlyIn(Dist.CLIENT)
