--- a/net/minecraft/world/level/block/AnvilBlock.java
+++ b/net/minecraft/world/level/block/AnvilBlock.java
@@ -63,8 +_,8 @@
 
    @Nullable
    public MenuProvider m_7246_(BlockState p_48821_, Level p_48822_, BlockPos p_48823_) {
-      return new SimpleMenuProvider((p_48785_, p_48786_, p_48787_) -> {
-         return new AnvilMenu(p_48785_, p_48786_, ContainerLevelAccess.m_39289_(p_48822_, p_48823_));
+      return new SimpleMenuProvider((p_220272_2_, p_220272_3_, p_220272_4_) -> {
+         return new AnvilMenu(p_220272_2_, p_220272_3_, ContainerLevelAccess.m_39289_(p_48822_, p_48823_));
       }, f_48774_);
    }
 
