--- a/net/minecraft/world/level/block/BambooSaplingBlock.java
+++ b/net/minecraft/world/level/block/BambooSaplingBlock.java
@@ -19,8 +_,6 @@
 import net.minecraft.world.phys.Vec3;
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.VoxelShape;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class BambooSaplingBlock extends Block implements BonemealableBlock {
    protected static final VoxelShape f_48954_ = Block.m_49796_(4.0D, 0.0D, 4.0D, 12.0D, 12.0D, 12.0D);
@@ -61,7 +_,6 @@
       }
    }
 
-   @OnlyIn(Dist.CLIENT)
    public ItemStack m_7397_(BlockGetter p_48964_, BlockPos p_48965_, BlockState p_48966_) {
       return new ItemStack(Items.f_41911_);
    }
