--- a/net/minecraft/world/level/block/BaseCoralWallFanBlock.java
+++ b/net/minecraft/world/level/block/BaseCoralWallFanBlock.java
@@ -22,7 +_,7 @@
    public static final DirectionProperty f_49192_ = HorizontalDirectionalBlock.f_54117_;
    private static final Map<Direction, VoxelShape> f_49193_ = Maps.newEnumMap(ImmutableMap.of(Direction.NORTH, Block.m_49796_(0.0D, 4.0D, 5.0D, 16.0D, 12.0D, 16.0D), Direction.SOUTH, Block.m_49796_(0.0D, 4.0D, 0.0D, 16.0D, 12.0D, 11.0D), Direction.WEST, Block.m_49796_(5.0D, 4.0D, 0.0D, 16.0D, 12.0D, 16.0D), Direction.EAST, Block.m_49796_(0.0D, 4.0D, 0.0D, 11.0D, 12.0D, 16.0D)));
 
-   protected BaseCoralWallFanBlock(BlockBehaviour.Properties p_49196_) {
+   public BaseCoralWallFanBlock(BlockBehaviour.Properties p_49196_) {
       super(p_49196_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_49192_, Direction.NORTH).m_61124_(f_49158_, Boolean.valueOf(true)));
    }
