--- a/net/minecraft/world/level/block/BaseCoralPlantTypeBlock.java
+++ b/net/minecraft/world/level/block/BaseCoralPlantTypeBlock.java
@@ -22,7 +_,7 @@
    public static final BooleanProperty f_49158_ = BlockStateProperties.f_61362_;
    private static final VoxelShape f_49157_ = Block.m_49796_(2.0D, 0.0D, 2.0D, 14.0D, 4.0D, 14.0D);
 
-   protected BaseCoralPlantTypeBlock(BlockBehaviour.Properties p_49161_) {
+   public BaseCoralPlantTypeBlock(BlockBehaviour.Properties p_49161_) {
       super(p_49161_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_49158_, Boolean.valueOf(true)));
    }
