--- a/net/minecraft/world/level/block/BaseRailBlock.java
+++ b/net/minecraft/world/level/block/BaseRailBlock.java
@@ -15,7 +_,7 @@
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.VoxelShape;
 
-public abstract class BaseRailBlock extends Block {
+public abstract class BaseRailBlock extends Block implements net.minecraftforge.common.extensions.IAbstractRailBlock {
    protected static final VoxelShape f_49355_ = Block.m_49796_(0.0D, 0.0D, 0.0D, 16.0D, 2.0D, 16.0D);
    protected static final VoxelShape f_49356_ = Block.m_49796_(0.0D, 0.0D, 0.0D, 16.0D, 8.0D, 16.0D);
    private final boolean f_49357_;
@@ -39,6 +_,7 @@
 
    public VoxelShape m_5940_(BlockState p_49403_, BlockGetter p_49404_, BlockPos p_49405_, CollisionContext p_49406_) {
       RailShape railshape = p_49403_.m_60713_(this) ? p_49403_.m_61143_(this.m_7978_()) : null;
+      RailShape railShape2 = p_49403_.m_60713_(this) ? getRailDirection(p_49403_, p_49404_, p_49405_, null) : null;
       return railshape != null && railshape.m_61745_() ? f_49356_ : f_49355_;
    }
 
@@ -63,7 +_,7 @@
 
    public void m_6861_(BlockState p_49377_, Level p_49378_, BlockPos p_49379_, Block p_49380_, BlockPos p_49381_, boolean p_49382_) {
       if (!p_49378_.f_46443_ && p_49378_.m_8055_(p_49379_).m_60713_(this)) {
-         RailShape railshape = p_49377_.m_61143_(this.m_7978_());
+         RailShape railshape = getRailDirection(p_49377_, p_49378_, p_49379_, null);
          if (m_49398_(p_49379_, p_49378_, railshape)) {
             m_49950_(p_49377_, p_49378_, p_49379_);
             p_49378_.m_7471_(p_49379_, p_49382_);
@@ -112,7 +_,7 @@
    public void m_6810_(BlockState p_49384_, Level p_49385_, BlockPos p_49386_, BlockState p_49387_, boolean p_49388_) {
       if (!p_49388_) {
          super.m_6810_(p_49384_, p_49385_, p_49386_, p_49387_, p_49388_);
-         if (p_49384_.m_61143_(this.m_7978_()).m_61745_()) {
+         if (getRailDirection(p_49384_, p_49385_, p_49386_, null).m_61745_()) {
             p_49385_.m_46672_(p_49386_.m_7494_(), this);
          }
 
@@ -131,5 +_,20 @@
       return blockstate.m_61124_(this.m_7978_(), flag ? RailShape.EAST_WEST : RailShape.NORTH_SOUTH);
    }
 
+   @Deprecated //Forge: Use getRailDirection(IBlockAccess, BlockPos, IBlockState, EntityMinecart) for enhanced ability
    public abstract Property<RailShape> m_7978_();
+
+   /* ======================================== FORGE START =====================================*/
+
+   @Override
+   public boolean isFlexibleRail(BlockState state, BlockGetter world, BlockPos pos)
+   {
+      return  !this.f_49357_;
+   }
+
+   @Override
+   public RailShape getRailDirection(BlockState state, BlockGetter world, BlockPos pos, @javax.annotation.Nullable net.minecraft.world.entity.vehicle.AbstractMinecart cart) {
+      return state.m_61143_(m_7978_());
+   }
+   /* ========================================= FORGE END ======================================*/
 }
