--- a/net/minecraft/world/level/block/BeehiveBlock.java
+++ b/net/minecraft/world/level/block/BeehiveBlock.java
@@ -87,6 +_,7 @@
       List<Bee> list = p_49650_.m_45976_(Bee.class, (new AABB(p_49651_)).m_82377_(8.0D, 6.0D, 8.0D));
       if (!list.isEmpty()) {
          List<Player> list1 = p_49650_.m_45976_(Player.class, (new AABB(p_49651_)).m_82377_(8.0D, 6.0D, 8.0D));
+         if (list1.isEmpty()) return; //Forge: Prevent Error when no players are around.
          int i = list1.size();
 
          for(Bee bee : list) {
