--- a/net/minecraft/world/level/block/BlastFurnaceBlock.java
+++ b/net/minecraft/world/level/block/BlastFurnaceBlock.java
@@ -19,7 +_,7 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class BlastFurnaceBlock extends AbstractFurnaceBlock {
-   protected BlastFurnaceBlock(BlockBehaviour.Properties p_49773_) {
+   public BlastFurnaceBlock(BlockBehaviour.Properties p_49773_) {
       super(p_49773_);
    }
 
