--- a/net/minecraft/world/level/block/CartographyTableBlock.java
+++ b/net/minecraft/world/level/block/CartographyTableBlock.java
@@ -20,7 +_,7 @@
 public class CartographyTableBlock extends Block {
    private static final Component f_51346_ = new TranslatableComponent("container.cartography_table");
 
-   protected CartographyTableBlock(BlockBehaviour.Properties p_51349_) {
+   public CartographyTableBlock(BlockBehaviour.Properties p_51349_) {
       super(p_51349_);
    }
 
@@ -36,8 +_,8 @@
 
    @Nullable
    public MenuProvider m_7246_(BlockState p_51364_, Level p_51365_, BlockPos p_51366_) {
-      return new SimpleMenuProvider((p_51353_, p_51354_, p_51355_) -> {
-         return new CartographyTableMenu(p_51353_, p_51354_, ContainerLevelAccess.m_39289_(p_51365_, p_51366_));
+      return new SimpleMenuProvider((p_220267_2_, p_220267_3_, p_220267_4_) -> {
+         return new CartographyTableMenu(p_220267_2_, p_220267_3_, ContainerLevelAccess.m_39289_(p_51365_, p_51366_));
       }, f_51346_);
    }
 }
