--- a/net/minecraft/world/level/block/CarvedPumpkinBlock.java
+++ b/net/minecraft/world/level/block/CarvedPumpkinBlock.java
@@ -34,11 +_,11 @@
    private BlockPattern f_51370_;
    @Nullable
    private BlockPattern f_51371_;
-   private static final Predicate<BlockState> f_51372_ = (p_51396_) -> {
-      return p_51396_ != null && (p_51396_.m_60713_(Blocks.f_50143_) || p_51396_.m_60713_(Blocks.f_50144_));
+   private static final Predicate<BlockState> f_51372_ = (p_210301_0_) -> {
+      return p_210301_0_ != null && (p_210301_0_.m_60713_(Blocks.f_50143_) || p_210301_0_.m_60713_(Blocks.f_50144_));
    };
 
-   protected CarvedPumpkinBlock(BlockBehaviour.Properties p_51375_) {
+   public CarvedPumpkinBlock(BlockBehaviour.Properties p_51375_) {
       super(p_51375_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_51367_, Direction.NORTH));
    }
