--- a/net/minecraft/world/level/block/ChestBlock.java
+++ b/net/minecraft/world/level/block/ChestBlock.java
@@ -109,7 +_,7 @@
       }
    };
 
-   protected ChestBlock(BlockBehaviour.Properties p_51490_, Supplier<BlockEntityType<? extends ChestBlockEntity>> p_51491_) {
+   public ChestBlock(BlockBehaviour.Properties p_51490_, Supplier<BlockEntityType<? extends ChestBlockEntity>> p_51491_) {
       super(p_51490_, p_51491_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_51478_, Direction.NORTH).m_61124_(f_51479_, ChestType.SINGLE).m_61124_(f_51480_, Boolean.valueOf(false)));
    }
@@ -251,7 +_,7 @@
    public DoubleBlockCombiner.NeighborCombineResult<? extends ChestBlockEntity> m_5641_(BlockState p_51544_, Level p_51545_, BlockPos p_51546_, boolean p_51547_) {
       BiPredicate<LevelAccessor, BlockPos> bipredicate;
       if (p_51547_) {
-         bipredicate = (p_51578_, p_51579_) -> {
+         bipredicate = (p_226918_0_, p_226918_1_) -> {
             return false;
          };
       } else {
@@ -270,8 +_,8 @@
    public static DoubleBlockCombiner.Combiner<ChestBlockEntity, Float2FloatFunction> m_51517_(final LidBlockEntity p_51518_) {
       return new DoubleBlockCombiner.Combiner<ChestBlockEntity, Float2FloatFunction>() {
          public Float2FloatFunction m_6959_(ChestBlockEntity p_51633_, ChestBlockEntity p_51634_) {
-            return (p_51638_) -> {
-               return Math.max(p_51633_.m_6683_(p_51638_), p_51634_.m_6683_(p_51638_));
+            return (p_226921_2_) -> {
+               return Math.max(p_51633_.m_6683_(p_226921_2_), p_51634_.m_6683_(p_226921_2_));
             };
          }
 
