--- a/net/minecraft/world/level/block/CoralWallFanBlock.java
+++ b/net/minecraft/world/level/block/CoralWallFanBlock.java
@@ -13,7 +_,7 @@
 public class CoralWallFanBlock extends BaseCoralWallFanBlock {
    private final Block f_52200_;
 
-   protected CoralWallFanBlock(Block p_52202_, BlockBehaviour.Properties p_52203_) {
+   public CoralWallFanBlock(Block p_52202_, BlockBehaviour.Properties p_52203_) {
       super(p_52203_);
       this.f_52200_ = p_52202_;
    }
