--- a/net/minecraft/world/level/block/CoralPlantBlock.java
+++ b/net/minecraft/world/level/block/CoralPlantBlock.java
@@ -17,7 +_,7 @@
    private final Block f_52172_;
    protected static final VoxelShape f_52171_ = Block.m_49796_(2.0D, 0.0D, 2.0D, 14.0D, 15.0D, 14.0D);
 
-   protected CoralPlantBlock(Block p_52175_, BlockBehaviour.Properties p_52176_) {
+   public CoralPlantBlock(Block p_52175_, BlockBehaviour.Properties p_52176_) {
       super(p_52176_);
       this.f_52172_ = p_52175_;
    }
