--- a/net/minecraft/world/level/block/DispenserBlock.java
+++ b/net/minecraft/world/level/block/DispenserBlock.java
@@ -40,15 +_,15 @@
 public class DispenserBlock extends BaseEntityBlock {
    public static final DirectionProperty f_52659_ = DirectionalBlock.f_52588_;
    public static final BooleanProperty f_52660_ = BlockStateProperties.f_61360_;
-   private static final Map<Item, DispenseItemBehavior> f_52661_ = Util.m_137469_(new Object2ObjectOpenHashMap<>(), (p_52723_) -> {
-      p_52723_.defaultReturnValue(new DefaultDispenseItemBehavior());
+   private static final Map<Item, DispenseItemBehavior> f_52661_ = Util.m_137469_(new Object2ObjectOpenHashMap<>(), (p_212564_0_) -> {
+      p_212564_0_.defaultReturnValue(new DefaultDispenseItemBehavior());
    });
 
    public static void m_52672_(ItemLike p_52673_, DispenseItemBehavior p_52674_) {
       f_52661_.put(p_52673_.m_5456_(), p_52674_);
    }
 
-   protected DispenserBlock(BlockBehaviour.Properties p_52664_) {
+   public DispenserBlock(BlockBehaviour.Properties p_52664_) {
       super(p_52664_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_52659_, Direction.NORTH).m_61124_(f_52660_, Boolean.valueOf(false)));
    }
