--- a/net/minecraft/world/level/block/EnderChestBlock.java
+++ b/net/minecraft/world/level/block/EnderChestBlock.java
@@ -43,7 +_,7 @@
    protected static final VoxelShape f_53117_ = Block.m_49796_(1.0D, 0.0D, 1.0D, 15.0D, 14.0D, 15.0D);
    private static final Component f_53118_ = new TranslatableComponent("container.enderchest");
 
-   protected EnderChestBlock(BlockBehaviour.Properties p_53121_) {
+   public EnderChestBlock(BlockBehaviour.Properties p_53121_) {
       super(p_53121_, () -> {
          return BlockEntityType.f_58920_;
       });
@@ -80,8 +_,8 @@
          } else {
             EnderChestBlockEntity enderchestblockentity = (EnderChestBlockEntity)blockentity;
             playerenderchestcontainer.m_40105_(enderchestblockentity);
-            p_53140_.m_5893_(new SimpleMenuProvider((p_53124_, p_53125_, p_53126_) -> {
-               return ChestMenu.m_39237_(p_53124_, p_53125_, playerenderchestcontainer);
+            p_53140_.m_5893_(new SimpleMenuProvider((p_226928_1_, p_226928_2_, p_226928_3_) -> {
+               return ChestMenu.m_39237_(p_226928_1_, p_226928_2_, playerenderchestcontainer);
             }, f_53118_));
             p_53140_.m_36220_(Stats.f_12963_);
             PiglinAi.m_34873_(p_53140_, true);
