--- a/net/minecraft/world/level/block/EndGatewayBlock.java
+++ b/net/minecraft/world/level/block/EndGatewayBlock.java
@@ -15,7 +_,7 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class EndGatewayBlock extends BaseEntityBlock {
-   protected EndGatewayBlock(BlockBehaviour.Properties p_52999_) {
+   public EndGatewayBlock(BlockBehaviour.Properties p_52999_) {
       super(p_52999_);
    }
 
@@ -51,7 +_,6 @@
       }
    }
 
-   @OnlyIn(Dist.CLIENT)
    public ItemStack m_7397_(BlockGetter p_53003_, BlockPos p_53004_, BlockState p_53005_) {
       return ItemStack.f_41583_;
    }
