--- a/net/minecraft/world/level/block/FungusBlock.java
+++ b/net/minecraft/world/level/block/FungusBlock.java
@@ -18,7 +_,7 @@
    protected static final VoxelShape f_53596_ = Block.m_49796_(4.0D, 0.0D, 4.0D, 12.0D, 9.0D, 12.0D);
    private final Supplier<ConfiguredFeature<HugeFungusConfiguration, ?>> f_53597_;
 
-   protected FungusBlock(BlockBehaviour.Properties p_53600_, Supplier<ConfiguredFeature<HugeFungusConfiguration, ?>> p_53601_) {
+   public FungusBlock(BlockBehaviour.Properties p_53600_, Supplier<ConfiguredFeature<HugeFungusConfiguration, ?>> p_53601_) {
       super(p_53600_);
       this.f_53597_ = p_53601_;
    }
