--- a/net/minecraft/world/level/block/GrindstoneBlock.java
+++ b/net/minecraft/world/level/block/GrindstoneBlock.java
@@ -92,7 +_,7 @@
    public static final VoxelShape f_53784_ = Shapes.m_83110_(f_53783_, Block.m_49796_(2.0D, 0.0D, 4.0D, 14.0D, 12.0D, 12.0D));
    private static final Component f_53768_ = new TranslatableComponent("container.grindstone_title");
 
-   protected GrindstoneBlock(BlockBehaviour.Properties p_53808_) {
+   public GrindstoneBlock(BlockBehaviour.Properties p_53808_) {
       super(p_53808_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_54117_, Direction.NORTH).m_61124_(f_53179_, AttachFace.WALL));
    }
@@ -156,8 +_,8 @@
    }
 
    public MenuProvider m_7246_(BlockState p_53847_, Level p_53848_, BlockPos p_53849_) {
-      return new SimpleMenuProvider((p_53812_, p_53813_, p_53814_) -> {
-         return new GrindstoneMenu(p_53812_, p_53813_, ContainerLevelAccess.m_39289_(p_53848_, p_53849_));
+      return new SimpleMenuProvider((p_220187_2_, p_220187_3_, p_220187_4_) -> {
+         return new GrindstoneMenu(p_220187_2_, p_220187_3_, ContainerLevelAccess.m_39289_(p_53848_, p_53849_));
       }, f_53768_);
    }
 
