--- a/net/minecraft/world/level/block/HalfTransparentBlock.java
+++ b/net/minecraft/world/level/block/HalfTransparentBlock.java
@@ -7,7 +_,7 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class HalfTransparentBlock extends Block {
-   protected HalfTransparentBlock(BlockBehaviour.Properties p_53970_) {
+   public HalfTransparentBlock(BlockBehaviour.Properties p_53970_) {
       super(p_53970_);
    }
 
