--- a/net/minecraft/world/level/block/IronBarsBlock.java
+++ b/net/minecraft/world/level/block/IronBarsBlock.java
@@ -18,7 +_,7 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class IronBarsBlock extends CrossCollisionBlock {
-   protected IronBarsBlock(BlockBehaviour.Properties p_54198_) {
+   public IronBarsBlock(BlockBehaviour.Properties p_54198_) {
       super(1.0F, 1.0F, 16.0F, 16.0F, 16.0F, p_54198_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_52309_, Boolean.valueOf(false)).m_61124_(f_52310_, Boolean.valueOf(false)).m_61124_(f_52311_, Boolean.valueOf(false)).m_61124_(f_52312_, Boolean.valueOf(false)).m_61124_(f_52313_, Boolean.valueOf(false)));
    }
