--- a/net/minecraft/world/level/block/KelpBlock.java
+++ b/net/minecraft/world/level/block/KelpBlock.java
@@ -18,7 +_,7 @@
 public class KelpBlock extends GrowingPlantHeadBlock implements LiquidBlockContainer {
    protected static final VoxelShape f_54297_ = Block.m_49796_(0.0D, 0.0D, 0.0D, 16.0D, 9.0D, 16.0D);
 
-   protected KelpBlock(BlockBehaviour.Properties p_54300_) {
+   public KelpBlock(BlockBehaviour.Properties p_54300_) {
       super(p_54300_, Direction.UP, f_54297_, true, 0.14D);
    }
 
