--- a/net/minecraft/world/level/block/KelpPlantBlock.java
+++ b/net/minecraft/world/level/block/KelpPlantBlock.java
@@ -12,7 +_,7 @@
 import net.minecraft.world.phys.shapes.Shapes;
 
 public class KelpPlantBlock extends GrowingPlantBodyBlock implements LiquidBlockContainer {
-   protected KelpPlantBlock(BlockBehaviour.Properties p_54323_) {
+   public KelpPlantBlock(BlockBehaviour.Properties p_54323_) {
       super(p_54323_, Direction.UP, Shapes.m_83144_(), true);
    }
 
