--- a/net/minecraft/world/level/block/LecternBlock.java
+++ b/net/minecraft/world/level/block/LecternBlock.java
@@ -47,7 +_,7 @@
    public static final VoxelShape f_54475_ = Shapes.m_83124_(Block.m_49796_(15.0D, 10.0D, 0.0D, 10.666667D, 14.0D, 16.0D), Block.m_49796_(10.666667D, 12.0D, 0.0D, 6.333333D, 16.0D, 16.0D), Block.m_49796_(6.333333D, 14.0D, 0.0D, 2.0D, 18.0D, 16.0D), f_54470_);
    public static final VoxelShape f_54476_ = Shapes.m_83124_(Block.m_49796_(0.0D, 10.0D, 15.0D, 16.0D, 14.0D, 10.666667D), Block.m_49796_(0.0D, 12.0D, 10.666667D, 16.0D, 16.0D, 6.333333D), Block.m_49796_(0.0D, 14.0D, 6.333333D, 16.0D, 18.0D, 2.0D), f_54470_);
 
-   protected LecternBlock(BlockBehaviour.Properties p_54479_) {
+   public LecternBlock(BlockBehaviour.Properties p_54479_) {
       super(p_54479_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_54465_, Direction.NORTH).m_61124_(f_54466_, Boolean.valueOf(false)).m_61124_(f_54467_, Boolean.valueOf(false)));
    }
