--- a/net/minecraft/world/level/block/PlayerWallHeadBlock.java
+++ b/net/minecraft/world/level/block/PlayerWallHeadBlock.java
@@ -11,7 +_,7 @@
 import net.minecraft.world.level.storage.loot.LootContext;
 
 public class PlayerWallHeadBlock extends WallSkullBlock {
-   protected PlayerWallHeadBlock(BlockBehaviour.Properties p_55185_) {
+   public PlayerWallHeadBlock(BlockBehaviour.Properties p_55185_) {
       super(SkullBlock.Types.PLAYER, p_55185_);
    }
 
