--- a/net/minecraft/world/level/block/RedstoneWallTorchBlock.java
+++ b/net/minecraft/world/level/block/RedstoneWallTorchBlock.java
@@ -23,7 +_,7 @@
    public static final DirectionProperty f_55740_ = HorizontalDirectionalBlock.f_54117_;
    public static final BooleanProperty f_55741_ = RedstoneTorchBlock.f_55674_;
 
-   protected RedstoneWallTorchBlock(BlockBehaviour.Properties p_55744_) {
+   public RedstoneWallTorchBlock(BlockBehaviour.Properties p_55744_) {
       super(p_55744_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_55740_, Direction.NORTH).m_61124_(f_55741_, Boolean.valueOf(true)));
    }
