--- a/net/minecraft/world/level/block/RepeaterBlock.java
+++ b/net/minecraft/world/level/block/RepeaterBlock.java
@@ -25,7 +_,7 @@
    public static final BooleanProperty f_55797_ = BlockStateProperties.f_61444_;
    public static final IntegerProperty f_55798_ = BlockStateProperties.f_61413_;
 
-   protected RepeaterBlock(BlockBehaviour.Properties p_55801_) {
+   public RepeaterBlock(BlockBehaviour.Properties p_55801_) {
       super(p_55801_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_54117_, Direction.NORTH).m_61124_(f_55798_, Integer.valueOf(1)).m_61124_(f_55797_, Boolean.valueOf(false)).m_61124_(f_52496_, Boolean.valueOf(false)));
    }
