--- a/net/minecraft/world/level/block/RespawnAnchorBlock.java
+++ b/net/minecraft/world/level/block/RespawnAnchorBlock.java
@@ -109,8 +_,8 @@
 
    private void m_55890_(BlockState p_55891_, Level p_55892_, final BlockPos p_55893_) {
       p_55892_.m_7471_(p_55893_, false);
-      boolean flag = Direction.Plane.HORIZONTAL.m_122557_().map(p_55893_::m_121945_).anyMatch((p_55854_) -> {
-         return m_55887_(p_55854_, p_55892_);
+      boolean flag = Direction.Plane.HORIZONTAL.m_122557_().map(p_55893_::m_121945_).anyMatch((p_235563_1_) -> {
+         return m_55887_(p_235563_1_, p_55892_);
       });
       final boolean flag1 = flag || p_55892_.m_6425_(p_55893_.m_7494_()).m_76153_(FluidTags.f_13131_);
       ExplosionDamageCalculator explosiondamagecalculator = new ExplosionDamageCalculator() {
