--- a/net/minecraft/world/level/block/RootsBlock.java
+++ b/net/minecraft/world/level/block/RootsBlock.java
@@ -11,7 +_,7 @@
 public class RootsBlock extends BushBlock {
    protected static final VoxelShape f_55909_ = Block.m_49796_(2.0D, 0.0D, 2.0D, 14.0D, 13.0D, 14.0D);
 
-   protected RootsBlock(BlockBehaviour.Properties p_55912_) {
+   public RootsBlock(BlockBehaviour.Properties p_55912_) {
       super(p_55912_);
    }
 
