--- a/net/minecraft/world/level/block/SeaPickleBlock.java
+++ b/net/minecraft/world/level/block/SeaPickleBlock.java
@@ -31,7 +_,7 @@
    protected static final VoxelShape f_56078_ = Block.m_49796_(2.0D, 0.0D, 2.0D, 14.0D, 6.0D, 14.0D);
    protected static final VoxelShape f_56079_ = Block.m_49796_(2.0D, 0.0D, 2.0D, 14.0D, 7.0D, 14.0D);
 
-   protected SeaPickleBlock(BlockBehaviour.Properties p_56082_) {
+   public SeaPickleBlock(BlockBehaviour.Properties p_56082_) {
       super(p_56082_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_56074_, Integer.valueOf(1)).m_61124_(f_56075_, Boolean.valueOf(true)));
    }
