--- a/net/minecraft/world/level/block/SkullBlock.java
+++ b/net/minecraft/world/level/block/SkullBlock.java
@@ -17,7 +_,7 @@
    public static final IntegerProperty f_56314_ = BlockStateProperties.f_61390_;
    protected static final VoxelShape f_56315_ = Block.m_49796_(4.0D, 0.0D, 4.0D, 12.0D, 8.0D, 12.0D);
 
-   protected SkullBlock(SkullBlock.Type p_56318_, BlockBehaviour.Properties p_56319_) {
+   public SkullBlock(SkullBlock.Type p_56318_, BlockBehaviour.Properties p_56319_) {
       super(p_56318_, p_56319_);
       this.m_49959_(this.f_49792_.m_61090_().m_61124_(f_56314_, Integer.valueOf(0)));
    }
