--- a/net/minecraft/world/level/block/SpawnerBlock.java
+++ b/net/minecraft/world/level/block/SpawnerBlock.java
@@ -8,11 +_,9 @@
 import net.minecraft.world.level.block.entity.SpawnerBlockEntity;
 import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class SpawnerBlock extends BaseEntityBlock {
-   protected SpawnerBlock(BlockBehaviour.Properties p_56781_) {
+   public SpawnerBlock(BlockBehaviour.Properties p_56781_) {
       super(p_56781_);
    }
 
@@ -22,15 +_,17 @@
 
    public void m_8101_(BlockState p_56789_, ServerLevel p_56790_, BlockPos p_56791_, ItemStack p_56792_) {
       super.m_8101_(p_56789_, p_56790_, p_56791_, p_56792_);
-      int i = 15 + p_56790_.f_46441_.nextInt(15) + p_56790_.f_46441_.nextInt(15);
-      this.m_49805_(p_56790_, p_56791_, i);
+   }
+
+   @Override
+   public int getExpDrop(BlockState state, net.minecraft.world.level.LevelReader world, BlockPos pos, int fortune, int silktouch) {
+      return 15 + RANDOM.nextInt(15) + RANDOM.nextInt(15);
    }
 
    public RenderShape m_7514_(BlockState p_56794_) {
       return RenderShape.MODEL;
    }
 
-   @OnlyIn(Dist.CLIENT)
    public ItemStack m_7397_(BlockGetter p_56785_, BlockPos p_56786_, BlockState p_56787_) {
       return ItemStack.f_41583_;
    }
