--- a/net/minecraft/world/level/block/SpongeBlock.java
+++ b/net/minecraft/world/level/block/SpongeBlock.java
@@ -66,7 +_,7 @@
                      queue.add(new Tuple<>(blockpos1, j + 1));
                   }
                } else if (material == Material.f_76301_ || material == Material.f_76304_) {
-                  BlockEntity blockentity = blockstate.m_60734_().m_60588_() ? p_56808_.m_7702_(blockpos1) : null;
+                  BlockEntity blockentity = blockstate.hasTileEntity() ? p_56808_.m_7702_(blockpos1) : null;
                   m_49892_(blockstate, p_56808_, blockpos1, blockentity);
                   p_56808_.m_7731_(blockpos1, Blocks.f_50016_.m_49966_(), 3);
                   ++i;
