--- a/net/minecraft/world/level/block/StoneButtonBlock.java
+++ b/net/minecraft/world/level/block/StoneButtonBlock.java
@@ -5,7 +_,7 @@
 import net.minecraft.world.level.block.state.BlockBehaviour;
 
 public class StoneButtonBlock extends ButtonBlock {
-   protected StoneButtonBlock(BlockBehaviour.Properties p_57060_) {
+   public StoneButtonBlock(BlockBehaviour.Properties p_57060_) {
       super(false, p_57060_);
    }
 
