--- a/net/minecraft/world/level/block/StructureBlock.java
+++ b/net/minecraft/world/level/block/StructureBlock.java
@@ -24,7 +_,7 @@
 public class StructureBlock extends BaseEntityBlock {
    public static final EnumProperty<StructureMode> f_57110_ = BlockStateProperties.f_61399_;
 
-   protected StructureBlock(BlockBehaviour.Properties p_57113_) {
+   public StructureBlock(BlockBehaviour.Properties p_57113_) {
       super(p_57113_);
    }
 
