--- a/net/minecraft/world/level/block/TallSeagrass.java
+++ b/net/minecraft/world/level/block/TallSeagrass.java
@@ -18,8 +_,6 @@
 import net.minecraft.world.level.material.Fluids;
 import net.minecraft.world.phys.shapes.CollisionContext;
 import net.minecraft.world.phys.shapes.VoxelShape;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class TallSeagrass extends DoublePlantBlock implements LiquidBlockContainer {
    public static final EnumProperty<DoubleBlockHalf> f_57340_ = DoublePlantBlock.f_52858_;
@@ -37,7 +_,6 @@
       return p_57371_.m_60783_(p_57372_, p_57373_, Direction.UP) && !p_57371_.m_60713_(Blocks.f_50450_);
    }
 
-   @OnlyIn(Dist.CLIENT)
    public ItemStack m_7397_(BlockGetter p_57348_, BlockPos p_57349_, BlockState p_57350_) {
       return new ItemStack(Blocks.f_50037_);
    }
