--- a/net/minecraft/world/level/block/TorchBlock.java
+++ b/net/minecraft/world/level/block/TorchBlock.java
@@ -20,7 +_,7 @@
    protected static final VoxelShape f_57487_ = Block.m_49796_(6.0D, 0.0D, 6.0D, 10.0D, 10.0D, 10.0D);
    protected final ParticleOptions f_57488_;
 
-   protected TorchBlock(BlockBehaviour.Properties p_57491_, ParticleOptions p_57492_) {
+   public TorchBlock(BlockBehaviour.Properties p_57491_, ParticleOptions p_57492_) {
       super(p_57491_);
       this.f_57488_ = p_57492_;
    }
