--- a/net/minecraft/world/level/block/WitherSkullBlock.java
+++ b/net/minecraft/world/level/block/WitherSkullBlock.java
@@ -30,7 +_,7 @@
    @Nullable
    private static BlockPattern f_58252_;
 
-   protected WitherSkullBlock(BlockBehaviour.Properties p_58254_) {
+   public WitherSkullBlock(BlockBehaviour.Properties p_58254_) {
       super(SkullBlock.Types.WITHER_SKELETON, p_58254_);
    }
 
@@ -92,8 +_,8 @@
 
    private static BlockPattern m_58273_() {
       if (f_58251_ == null) {
-         f_58251_ = BlockPatternBuilder.m_61243_().m_61247_("^^^", "###", "~#~").m_61244_('#', (p_58272_) -> {
-            return p_58272_.m_61168_().m_60620_(BlockTags.f_13071_);
+         f_58251_ = BlockPatternBuilder.m_61243_().m_61247_("^^^", "###", "~#~").m_61244_('#', (p_235639_0_) -> {
+            return p_235639_0_.m_61168_().m_60620_(BlockTags.f_13071_);
          }).m_61244_('^', BlockInWorld.m_61169_(BlockStatePredicate.m_61287_(Blocks.f_50312_).or(BlockStatePredicate.m_61287_(Blocks.f_50313_)))).m_61244_('~', BlockInWorld.m_61169_(BlockMaterialPredicate.m_61262_(Material.f_76296_))).m_61249_();
       }
 
@@ -102,8 +_,8 @@
 
    private static BlockPattern m_58274_() {
       if (f_58252_ == null) {
-         f_58252_ = BlockPatternBuilder.m_61243_().m_61247_("   ", "###", "~#~").m_61244_('#', (p_58266_) -> {
-            return p_58266_.m_61168_().m_60620_(BlockTags.f_13071_);
+         f_58252_ = BlockPatternBuilder.m_61243_().m_61247_("   ", "###", "~#~").m_61244_('#', (p_235638_0_) -> {
+            return p_235638_0_.m_61168_().m_60620_(BlockTags.f_13071_);
          }).m_61244_('~', BlockInWorld.m_61169_(BlockMaterialPredicate.m_61262_(Material.f_76296_))).m_61249_();
       }
 
