--- a/net/minecraft/world/level/block/WoodButtonBlock.java
+++ b/net/minecraft/world/level/block/WoodButtonBlock.java
@@ -5,7 +_,7 @@
 import net.minecraft.world.level.block.state.BlockBehaviour;
 
 public class WoodButtonBlock extends ButtonBlock {
-   protected WoodButtonBlock(BlockBehaviour.Properties p_58284_) {
+   public WoodButtonBlock(BlockBehaviour.Properties p_58284_) {
       super(true, p_58284_);
    }
 
