--- a/net/minecraft/world/level/block/WoolCarpetBlock.java
+++ b/net/minecraft/world/level/block/WoolCarpetBlock.java
@@ -15,7 +_,7 @@
    protected static final VoxelShape f_58287_ = Block.m_49796_(0.0D, 0.0D, 0.0D, 16.0D, 1.0D, 16.0D);
    private final DyeColor f_58288_;
 
-   protected WoolCarpetBlock(DyeColor p_58291_, BlockBehaviour.Properties p_58292_) {
+   public WoolCarpetBlock(DyeColor p_58291_, BlockBehaviour.Properties p_58292_) {
       super(p_58292_);
       this.f_58288_ = p_58291_;
    }
