--- a/mcp/MethodsReturnNonnullByDefault.java
+++ b/mcp/MethodsReturnNonnullByDefault.java
@@ -1,10 +_,9 @@
 package mcp;
-import javax.annotation.Nonnull;
 import java.lang.annotation.Documented;
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
-
+import javax.annotation.Nonnull;
 import javax.annotation.meta.TypeQualifierDefault;
 
 /**
