--- a/net/minecraft/world/level/levelgen/feature/trunkplacers/ForkingTrunkPlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/trunkplacers/ForkingTrunkPlacer.java
@@ -14,8 +_,8 @@
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 
 public class ForkingTrunkPlacer extends TrunkPlacer {
-   public static final Codec<ForkingTrunkPlacer> f_70145_ = RecordCodecBuilder.create((p_70161_) -> {
-      return m_70305_(p_70161_).apply(p_70161_, ForkingTrunkPlacer::new);
+   public static final Codec<ForkingTrunkPlacer> f_70145_ = RecordCodecBuilder.create((p_236897_0_) -> {
+      return m_70305_(p_236897_0_).apply(p_236897_0_, ForkingTrunkPlacer::new);
    });
 
    public ForkingTrunkPlacer(int p_70148_, int p_70149_, int p_70150_) {
