--- a/net/minecraft/world/level/levelgen/feature/trunkplacers/MegaJungleTrunkPlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/trunkplacers/MegaJungleTrunkPlacer.java
@@ -14,8 +_,8 @@
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 
 public class MegaJungleTrunkPlacer extends GiantTrunkPlacer {
-   public static final Codec<MegaJungleTrunkPlacer> f_70190_ = RecordCodecBuilder.create((p_70206_) -> {
-      return m_70305_(p_70206_).apply(p_70206_, MegaJungleTrunkPlacer::new);
+   public static final Codec<MegaJungleTrunkPlacer> f_70190_ = RecordCodecBuilder.create((p_236902_0_) -> {
+      return m_70305_(p_236902_0_).apply(p_236902_0_, MegaJungleTrunkPlacer::new);
    });
 
    public MegaJungleTrunkPlacer(int p_70193_, int p_70194_, int p_70195_) {
