--- a/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
@@ -189,7 +_,10 @@
       this.f_114364_ = new PlayerRenderer(this);
       this.f_114363_.put("default", this.f_114364_);
       this.f_114363_.put("slim", new PlayerRenderer(this, true));
+   }
 
+   //FORGE: Validate the existence of a renderer for each entity type after modded entity renderers have been added
+   public void validateRendererExistence() {
       for(EntityType<?> entitytype : Registry.f_122826_) {
          if (entitytype != EntityType.f_20532_ && !this.f_114362_.containsKey(entitytype)) {
             throw new IllegalStateException("No renderer registered for " + Registry.f_122826_.m_7981_(entitytype));
@@ -281,12 +_,12 @@
    private void m_114441_(PoseStack p_114442_, VertexConsumer p_114443_, Entity p_114444_, float p_114445_) {
       float f = p_114444_.m_20205_() / 2.0F;
       this.m_114446_(p_114442_, p_114443_, p_114444_, 1.0F, 1.0F, 1.0F);
-      if (p_114444_ instanceof EnderDragon) {
+      if (p_114444_.isMultipartEntity()) {
          double d0 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19790_, p_114444_.m_20185_());
          double d1 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19791_, p_114444_.m_20186_());
          double d2 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19792_, p_114444_.m_20189_());
 
-         for(EnderDragonPart enderdragonpart : ((EnderDragon)p_114444_).m_31156_()) {
+         for(net.minecraftforge.entity.PartEntity<?> enderdragonpart : p_114444_.getParts()) {
             p_114442_.m_85836_();
             double d3 = d0 + Mth.m_14139_((double)p_114445_, enderdragonpart.f_19790_, enderdragonpart.m_20185_());
             double d4 = d1 + Mth.m_14139_((double)p_114445_, enderdragonpart.f_19791_, enderdragonpart.m_20186_());
@@ -450,5 +_,9 @@
 
    public Font m_114475_() {
       return this.f_114365_;
+   }
+
+   public Map<String, PlayerRenderer> getSkinMap() {
+      return (Map<String, PlayerRenderer>) java.util.Collections.unmodifiableMap(f_114363_);
    }
 }
