--- a/net/minecraft/client/renderer/entity/FireworkEntityRenderer.java
+++ b/net/minecraft/client/renderer/entity/FireworkEntityRenderer.java
@@ -13,9 +_,9 @@
 
 @OnlyIn(Dist.CLIENT)
 public class FireworkEntityRenderer extends EntityRenderer<FireworkRocketEntity> {
-   private final ItemRenderer f_114640_;
+   private final net.minecraft.client.renderer.entity.ItemRenderer f_114640_;
 
-   public FireworkEntityRenderer(EntityRenderDispatcher p_114642_, ItemRenderer p_114643_) {
+   public FireworkEntityRenderer(EntityRenderDispatcher p_114642_, net.minecraft.client.renderer.entity.ItemRenderer p_114643_) {
       super(p_114642_);
       this.f_114640_ = p_114643_;
    }
