--- a/net/minecraft/client/renderer/entity/HorseRenderer.java
+++ b/net/minecraft/client/renderer/entity/HorseRenderer.java
@@ -14,14 +_,14 @@
 
 @OnlyIn(Dist.CLIENT)
 public final class HorseRenderer extends AbstractHorseRenderer<Horse, HorseModel<Horse>> {
-   private static final Map<Variant, ResourceLocation> f_114865_ = Util.m_137469_(Maps.newEnumMap(Variant.class), (p_114874_) -> {
-      p_114874_.put(Variant.WHITE, new ResourceLocation("textures/entity/horse/horse_white.png"));
-      p_114874_.put(Variant.CREAMY, new ResourceLocation("textures/entity/horse/horse_creamy.png"));
-      p_114874_.put(Variant.CHESTNUT, new ResourceLocation("textures/entity/horse/horse_chestnut.png"));
-      p_114874_.put(Variant.BROWN, new ResourceLocation("textures/entity/horse/horse_brown.png"));
-      p_114874_.put(Variant.BLACK, new ResourceLocation("textures/entity/horse/horse_black.png"));
-      p_114874_.put(Variant.GRAY, new ResourceLocation("textures/entity/horse/horse_gray.png"));
-      p_114874_.put(Variant.DARKBROWN, new ResourceLocation("textures/entity/horse/horse_darkbrown.png"));
+   private static final Map<Variant, ResourceLocation> f_114865_ = Util.m_137469_(Maps.newEnumMap(Variant.class), (p_239384_0_) -> {
+      p_239384_0_.put(Variant.WHITE, new ResourceLocation("textures/entity/horse/horse_white.png"));
+      p_239384_0_.put(Variant.CREAMY, new ResourceLocation("textures/entity/horse/horse_creamy.png"));
+      p_239384_0_.put(Variant.CHESTNUT, new ResourceLocation("textures/entity/horse/horse_chestnut.png"));
+      p_239384_0_.put(Variant.BROWN, new ResourceLocation("textures/entity/horse/horse_brown.png"));
+      p_239384_0_.put(Variant.BLACK, new ResourceLocation("textures/entity/horse/horse_black.png"));
+      p_239384_0_.put(Variant.GRAY, new ResourceLocation("textures/entity/horse/horse_gray.png"));
+      p_239384_0_.put(Variant.DARKBROWN, new ResourceLocation("textures/entity/horse/horse_darkbrown.png"));
    });
 
    public HorseRenderer(EntityRenderDispatcher p_114868_) {
