--- a/net/minecraft/client/renderer/entity/PandaRenderer.java
+++ b/net/minecraft/client/renderer/entity/PandaRenderer.java
@@ -15,14 +_,14 @@
 
 @OnlyIn(Dist.CLIENT)
 public class PandaRenderer extends MobRenderer<Panda, PandaModel<Panda>> {
-   private static final Map<Panda.Gene, ResourceLocation> f_115620_ = Util.m_137469_(Maps.newEnumMap(Panda.Gene.class), (p_115647_) -> {
-      p_115647_.put(Panda.Gene.NORMAL, new ResourceLocation("textures/entity/panda/panda.png"));
-      p_115647_.put(Panda.Gene.LAZY, new ResourceLocation("textures/entity/panda/lazy_panda.png"));
-      p_115647_.put(Panda.Gene.WORRIED, new ResourceLocation("textures/entity/panda/worried_panda.png"));
-      p_115647_.put(Panda.Gene.PLAYFUL, new ResourceLocation("textures/entity/panda/playful_panda.png"));
-      p_115647_.put(Panda.Gene.BROWN, new ResourceLocation("textures/entity/panda/brown_panda.png"));
-      p_115647_.put(Panda.Gene.WEAK, new ResourceLocation("textures/entity/panda/weak_panda.png"));
-      p_115647_.put(Panda.Gene.AGGRESSIVE, new ResourceLocation("textures/entity/panda/aggressive_panda.png"));
+   private static final Map<Panda.Gene, ResourceLocation> f_115620_ = Util.m_137469_(Maps.newEnumMap(Panda.Gene.class), (p_217776_0_) -> {
+      p_217776_0_.put(Panda.Gene.NORMAL, new ResourceLocation("textures/entity/panda/panda.png"));
+      p_217776_0_.put(Panda.Gene.LAZY, new ResourceLocation("textures/entity/panda/lazy_panda.png"));
+      p_217776_0_.put(Panda.Gene.WORRIED, new ResourceLocation("textures/entity/panda/worried_panda.png"));
+      p_217776_0_.put(Panda.Gene.PLAYFUL, new ResourceLocation("textures/entity/panda/playful_panda.png"));
+      p_217776_0_.put(Panda.Gene.BROWN, new ResourceLocation("textures/entity/panda/brown_panda.png"));
+      p_217776_0_.put(Panda.Gene.WEAK, new ResourceLocation("textures/entity/panda/weak_panda.png"));
+      p_217776_0_.put(Panda.Gene.AGGRESSIVE, new ResourceLocation("textures/entity/panda/aggressive_panda.png"));
    });
 
    public PandaRenderer(EntityRenderDispatcher p_115623_) {
