--- a/net/minecraft/client/renderer/entity/ShulkerRenderer.java
+++ b/net/minecraft/client/renderer/entity/ShulkerRenderer.java
@@ -16,10 +_,10 @@
 @OnlyIn(Dist.CLIENT)
 public class ShulkerRenderer extends MobRenderer<Shulker, ShulkerModel<Shulker>> {
    public static final ResourceLocation f_115871_ = new ResourceLocation("textures/" + Sheets.f_110741_.m_119203_().m_135815_() + ".png");
-   public static final ResourceLocation[] f_115872_ = Sheets.f_110742_.stream().map((p_115919_) -> {
-      return new ResourceLocation("textures/" + p_115919_.m_119203_().m_135815_() + ".png");
-   }).toArray((p_115877_) -> {
-      return new ResourceLocation[p_115877_];
+   public static final ResourceLocation[] f_115872_ = Sheets.f_110742_.stream().map((p_229125_0_) -> {
+      return new ResourceLocation("textures/" + p_229125_0_.m_119203_().m_135815_() + ".png");
+   }).toArray((p_229124_0_) -> {
+      return new ResourceLocation[p_229124_0_];
    });
 
    public ShulkerRenderer(EntityRenderDispatcher p_115875_) {
