--- a/net/minecraft/client/renderer/entity/ThrownItemRenderer.java
+++ b/net/minecraft/client/renderer/entity/ThrownItemRenderer.java
@@ -15,18 +_,18 @@
 
 @OnlyIn(Dist.CLIENT)
 public class ThrownItemRenderer<T extends Entity & ItemSupplier> extends EntityRenderer<T> {
-   private final ItemRenderer f_116071_;
+   private final net.minecraft.client.renderer.entity.ItemRenderer f_116071_;
    private final float f_116072_;
    private final boolean f_116073_;
 
-   public ThrownItemRenderer(EntityRenderDispatcher p_116078_, ItemRenderer p_116079_, float p_116080_, boolean p_116081_) {
+   public ThrownItemRenderer(EntityRenderDispatcher p_116078_, net.minecraft.client.renderer.entity.ItemRenderer p_116079_, float p_116080_, boolean p_116081_) {
       super(p_116078_);
       this.f_116071_ = p_116079_;
       this.f_116072_ = p_116080_;
       this.f_116073_ = p_116081_;
    }
 
-   public ThrownItemRenderer(EntityRenderDispatcher p_116075_, ItemRenderer p_116076_) {
+   public ThrownItemRenderer(EntityRenderDispatcher p_116075_, net.minecraft.client.renderer.entity.ItemRenderer p_116076_) {
       this(p_116075_, p_116076_, 1.0F, false);
    }
 
