--- a/net/minecraft/world/entity/projectile/Fireball.java
+++ b/net/minecraft/world/entity/projectile/Fireball.java
@@ -34,8 +_,8 @@
 
    public void m_37010_(ItemStack p_37011_) {
       if (p_37011_.m_41720_() != Items.f_42613_ || p_37011_.m_41782_()) {
-         this.m_20088_().m_135381_(f_36987_, Util.m_137469_(p_37011_.m_41777_(), (p_37015_) -> {
-            p_37015_.m_41764_(1);
+         this.m_20088_().m_135381_(f_36987_, Util.m_137469_(p_37011_.m_41777_(), (p_213897_0_) -> {
+            p_213897_0_.m_41764_(1);
          }));
       }
 
