--- a/net/minecraft/world/entity/projectile/Projectile.java
+++ b/net/minecraft/world/entity/projectile/Projectile.java
@@ -72,8 +_,8 @@
    private boolean m_37276_() {
       Entity entity = this.m_37282_();
       if (entity != null) {
-         for(Entity entity1 : this.f_19853_.m_6249_(this, this.m_20191_().m_82369_(this.m_20184_()).m_82400_(1.0D), (p_37272_) -> {
-            return !p_37272_.m_5833_() && p_37272_.m_6087_();
+         for(Entity entity1 : this.f_19853_.m_6249_(this, this.m_20191_().m_82369_(this.m_20184_()).m_82400_(1.0D), (p_234613_0_) -> {
+            return !p_234613_0_.m_5833_() && p_234613_0_.m_6087_();
          })) {
             if (entity1.m_20201_() == entity.m_20201_()) {
                return false;
