--- a/net/minecraft/world/level/levelgen/placement/nether/GlowstoneDecorator.java
+++ b/net/minecraft/world/level/levelgen/placement/nether/GlowstoneDecorator.java
@@ -14,7 +_,7 @@
    }
 
    public Stream<BlockPos> m_6193_(Random p_70924_, CountConfiguration p_70925_, BlockPos p_70926_) {
-      return IntStream.range(0, p_70924_.nextInt(p_70924_.nextInt(p_70925_.m_67575_().m_14447_(p_70924_)) + 1)).mapToObj((p_70934_) -> {
+      return IntStream.range(0, p_70924_.nextInt(p_70924_.nextInt(p_70925_.m_67575_().m_14447_(p_70924_)) + 1)).mapToObj((p_242916_2_) -> {
          int i = p_70924_.nextInt(16) + p_70926_.m_123341_();
          int j = p_70924_.nextInt(16) + p_70926_.m_123343_();
          int k = p_70924_.nextInt(120) + 4;
