--- a/net/minecraft/network/protocol/game/ClientboundUpdateAttributesPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundUpdateAttributesPacket.java
@@ -12,8 +_,6 @@
 import net.minecraft.world.entity.ai.attributes.Attribute;
 import net.minecraft.world.entity.ai.attributes.AttributeInstance;
 import net.minecraft.world.entity.ai.attributes.AttributeModifier;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class ClientboundUpdateAttributesPacket implements Packet<ClientGamePacketListener> {
    private int f_133576_;
@@ -74,12 +_,10 @@
       p_133587_.m_7710_(this);
    }
 
-   @OnlyIn(Dist.CLIENT)
    public int m_133588_() {
       return this.f_133576_;
    }
 
-   @OnlyIn(Dist.CLIENT)
    public List<ClientboundUpdateAttributesPacket.AttributeSnapshot> m_133591_() {
       return this.f_133577_;
    }
