--- a/net/minecraft/network/protocol/game/ClientboundMoveEntityPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundMoveEntityPacket.java
@@ -97,34 +_,31 @@
       return this.f_132505_;
    }
 
-   public static class Pos extends ClientboundMoveEntityPacket {
-      public Pos() {
-         this.f_132507_ = true;
-      }
-
-      public Pos(int p_132539_, short p_132540_, short p_132541_, short p_132542_, boolean p_132543_) {
-         super(p_132539_);
-         this.f_132500_ = p_132540_;
-         this.f_132501_ = p_132541_;
-         this.f_132502_ = p_132542_;
-         this.f_132505_ = p_132543_;
-         this.f_132507_ = true;
-      }
-
-      public void m_5778_(FriendlyByteBuf p_132545_) throws IOException {
-         super.m_5778_(p_132545_);
-         this.f_132500_ = p_132545_.readShort();
-         this.f_132501_ = p_132545_.readShort();
-         this.f_132502_ = p_132545_.readShort();
-         this.f_132505_ = p_132545_.readBoolean();
-      }
-
-      public void m_5779_(FriendlyByteBuf p_132549_) throws IOException {
-         super.m_5779_(p_132549_);
-         p_132549_.writeShort(this.f_132500_);
-         p_132549_.writeShort(this.f_132501_);
-         p_132549_.writeShort(this.f_132502_);
-         p_132549_.writeBoolean(this.f_132505_);
+   public static class Rot extends ClientboundMoveEntityPacket {
+      public Rot() {
+         this.f_132506_ = true;
+      }
+
+      public Rot(int p_132567_, byte p_132568_, byte p_132569_, boolean p_132570_) {
+         super(p_132567_);
+         this.f_132503_ = p_132568_;
+         this.f_132504_ = p_132569_;
+         this.f_132506_ = true;
+         this.f_132505_ = p_132570_;
+      }
+
+      public void m_5778_(FriendlyByteBuf p_132572_) throws IOException {
+         super.m_5778_(p_132572_);
+         this.f_132503_ = p_132572_.readByte();
+         this.f_132504_ = p_132572_.readByte();
+         this.f_132505_ = p_132572_.readBoolean();
+      }
+
+      public void m_5779_(FriendlyByteBuf p_132576_) throws IOException {
+         super.m_5779_(p_132576_);
+         p_132576_.writeByte(this.f_132503_);
+         p_132576_.writeByte(this.f_132504_);
+         p_132576_.writeBoolean(this.f_132505_);
       }
    }
 
@@ -167,31 +_,34 @@
       }
    }
 
-   public static class Rot extends ClientboundMoveEntityPacket {
-      public Rot() {
-         this.f_132506_ = true;
-      }
-
-      public Rot(int p_132567_, byte p_132568_, byte p_132569_, boolean p_132570_) {
-         super(p_132567_);
-         this.f_132503_ = p_132568_;
-         this.f_132504_ = p_132569_;
-         this.f_132506_ = true;
-         this.f_132505_ = p_132570_;
-      }
-
-      public void m_5778_(FriendlyByteBuf p_132572_) throws IOException {
-         super.m_5778_(p_132572_);
-         this.f_132503_ = p_132572_.readByte();
-         this.f_132504_ = p_132572_.readByte();
-         this.f_132505_ = p_132572_.readBoolean();
-      }
-
-      public void m_5779_(FriendlyByteBuf p_132576_) throws IOException {
-         super.m_5779_(p_132576_);
-         p_132576_.writeByte(this.f_132503_);
-         p_132576_.writeByte(this.f_132504_);
-         p_132576_.writeBoolean(this.f_132505_);
+   public static class Pos extends ClientboundMoveEntityPacket {
+      public Pos() {
+         this.f_132507_ = true;
+      }
+
+      public Pos(int p_132539_, short p_132540_, short p_132541_, short p_132542_, boolean p_132543_) {
+         super(p_132539_);
+         this.f_132500_ = p_132540_;
+         this.f_132501_ = p_132541_;
+         this.f_132502_ = p_132542_;
+         this.f_132505_ = p_132543_;
+         this.f_132507_ = true;
+      }
+
+      public void m_5778_(FriendlyByteBuf p_132545_) throws IOException {
+         super.m_5778_(p_132545_);
+         this.f_132500_ = p_132545_.readShort();
+         this.f_132501_ = p_132545_.readShort();
+         this.f_132502_ = p_132545_.readShort();
+         this.f_132505_ = p_132545_.readBoolean();
+      }
+
+      public void m_5779_(FriendlyByteBuf p_132549_) throws IOException {
+         super.m_5779_(p_132549_);
+         p_132549_.writeShort(this.f_132500_);
+         p_132549_.writeShort(this.f_132501_);
+         p_132549_.writeShort(this.f_132502_);
+         p_132549_.writeBoolean(this.f_132505_);
       }
    }
 }
