--- a/net/minecraft/client/gui/screens/social/SocialInteractionsPlayerList.java
+++ b/net/minecraft/client/gui/screens/social/SocialInteractionsPlayerList.java
@@ -49,8 +_,8 @@
       }
 
       this.m_100725_();
-      this.f_100694_.sort((p_100712_, p_100713_) -> {
-         return p_100712_.m_100600_().compareToIgnoreCase(p_100713_.m_100600_());
+      this.f_100694_.sort((p_244655_0_, p_244655_1_) -> {
+         return p_244655_0_.m_100600_().compareToIgnoreCase(p_244655_1_.m_100600_());
       });
       this.m_5988_(this.f_100694_);
       this.m_93410_(p_100721_);
@@ -58,8 +_,8 @@
 
    private void m_100725_() {
       if (this.f_100695_ != null) {
-         this.f_100694_.removeIf((p_100710_) -> {
-            return !p_100710_.m_100600_().toLowerCase(Locale.ROOT).contains(this.f_100695_);
+         this.f_100694_.removeIf((p_244654_1_) -> {
+            return !p_244654_1_.m_100600_().toLowerCase(Locale.ROOT).contains(this.f_100695_);
          });
          this.m_5988_(this.f_100694_);
       }
