--- a/net/minecraft/client/gui/screens/social/SocialInteractionsScreen.java
+++ b/net/minecraft/client/gui/screens/social/SocialInteractionsScreen.java
@@ -97,18 +_,18 @@
       int l = this.f_96547_.m_92852_(f_100747_) + 40;
       int i1 = 64 + 16 * this.m_100800_();
       int j1 = (this.f_96543_ - l) / 2;
-      this.f_100728_ = this.m_7082_(new Button(j, 45, i, 20, f_100737_, (p_100796_) -> {
+      this.f_100728_ = this.m_7082_(new Button(j, 45, i, 20, f_100737_, (p_244686_1_) -> {
          this.m_100771_(SocialInteractionsScreen.Page.ALL);
       }));
-      this.f_100729_ = this.m_7082_(new Button((j + k - i) / 2 + 1, 45, i, 20, f_100738_, (p_100791_) -> {
+      this.f_100729_ = this.m_7082_(new Button((j + k - i) / 2 + 1, 45, i, 20, f_100738_, (p_244681_1_) -> {
          this.m_100771_(SocialInteractionsScreen.Page.HIDDEN);
       }));
-      this.f_100730_ = this.m_7082_(new Button(k - i + 1, 45, i, 20, f_100739_, (p_100785_) -> {
+      this.f_100730_ = this.m_7082_(new Button(k - i + 1, 45, i, 20, f_100739_, (p_244769_1_) -> {
          this.m_100771_(SocialInteractionsScreen.Page.BLOCKED);
       }));
-      this.f_100731_ = this.m_7082_(new Button(j1, i1, l, 20, f_100747_, (p_100770_) -> {
-         this.f_96541_.m_91152_(new ConfirmLinkScreen((p_100793_) -> {
-            if (p_100793_) {
+      this.f_100731_ = this.m_7082_(new Button(j1, i1, l, 20, f_100747_, (p_244767_1_) -> {
+         this.f_96541_.m_91152_(new ConfirmLinkScreen((p_244771_1_) -> {
+            if (p_244771_1_) {
                Util.m_137581_().m_137646_("https://aka.ms/javablocking");
             }
 
