--- a/net/minecraft/client/renderer/block/model/BlockFaceUV.java
+++ b/net/minecraft/client/renderer/block/model/BlockFaceUV.java
@@ -57,9 +_,6 @@
 
    @OnlyIn(Dist.CLIENT)
    public static class Deserializer implements JsonDeserializer<BlockFaceUV> {
-      protected Deserializer() {
-      }
-
       public BlockFaceUV deserialize(JsonElement p_111404_, Type p_111405_, JsonDeserializationContext p_111406_) throws JsonParseException {
          JsonObject jsonobject = p_111404_.getAsJsonObject();
          float[] afloat = this.m_111409_(jsonobject);
