--- a/net/minecraft/client/renderer/block/model/ItemOverride.java
+++ b/net/minecraft/client/renderer/block/model/ItemOverride.java
@@ -50,9 +_,6 @@
 
    @OnlyIn(Dist.CLIENT)
    public static class Deserializer implements JsonDeserializer<ItemOverride> {
-      protected Deserializer() {
-      }
-
       public ItemOverride deserialize(JsonElement p_111725_, Type p_111726_, JsonDeserializationContext p_111727_) throws JsonParseException {
          JsonObject jsonobject = p_111725_.getAsJsonObject();
          ResourceLocation resourcelocation = new ResourceLocation(GsonHelper.m_13906_(jsonobject, "model"));
