--- a/net/minecraft/server/level/ChunkTaskPriorityQueueSorter.java
+++ b/net/minecraft/server/level/ChunkTaskPriorityQueueSorter.java
@@ -30,18 +_,18 @@
    private final ProcessorMailbox<StrictQueue.IntRunnable> f_140552_;
 
    public ChunkTaskPriorityQueueSorter(List<ProcessorHandle<?>> p_140555_, Executor p_140556_, int p_140557_) {
-      this.f_140550_ = p_140555_.stream().collect(Collectors.toMap(Function.identity(), (p_140561_) -> {
-         return new ChunkTaskPriorityQueue<>(p_140561_.m_7326_() + "_queue", p_140557_);
+      this.f_140550_ = p_140555_.stream().collect(Collectors.toMap(Function.identity(), (p_219084_1_) -> {
+         return new ChunkTaskPriorityQueue<>(p_219084_1_.m_7326_() + "_queue", p_140557_);
       }));
       this.f_140551_ = Sets.newHashSet(p_140555_);
       this.f_140552_ = new ProcessorMailbox<>(new StrictQueue.FixedPriorityQueue(4), p_140556_, "sorter");
    }
 
    public static ChunkTaskPriorityQueueSorter.Message<Runnable> m_140624_(Runnable p_140625_, long p_140626_, IntSupplier p_140627_) {
-      return new ChunkTaskPriorityQueueSorter.Message<>((p_140634_) -> {
+      return new ChunkTaskPriorityQueueSorter.Message<>((p_219072_1_) -> {
          return () -> {
             p_140625_.run();
-            p_140634_.m_6937_(Unit.INSTANCE);
+            p_219072_1_.m_6937_(Unit.INSTANCE);
          };
       }, p_140626_, p_140627_);
    }
@@ -55,21 +_,21 @@
    }
 
    public <T> ProcessorHandle<ChunkTaskPriorityQueueSorter.Message<T>> m_140604_(ProcessorHandle<T> p_140605_, boolean p_140606_) {
-      return this.f_140552_.<ProcessorHandle<ChunkTaskPriorityQueueSorter.Message<T>>>m_18720_((p_140610_) -> {
+      return this.f_140552_.<ProcessorHandle<Message<T>>>m_18720_((p_219086_3_) -> {
          return new StrictQueue.IntRunnable(0, () -> {
             this.m_140652_(p_140605_);
-            p_140610_.m_6937_(ProcessorHandle.m_18714_("chunk priority sorter around " + p_140605_.m_7326_(), (p_140614_) -> {
-               this.m_140589_(p_140605_, p_140614_.f_140664_, p_140614_.f_140665_, p_140614_.f_140666_, p_140606_);
+            p_219086_3_.m_6937_(ProcessorHandle.m_18714_("chunk priority sorter around " + p_140605_.m_7326_(), (p_219071_3_) -> {
+               this.m_140589_(p_140605_, p_219071_3_.f_140664_, p_219071_3_.f_140665_, p_219071_3_.f_140666_, p_140606_);
             }));
          });
       }).join();
    }
 
    public ProcessorHandle<ChunkTaskPriorityQueueSorter.Release> m_140567_(ProcessorHandle<Runnable> p_140568_) {
-      return this.f_140552_.<ProcessorHandle<ChunkTaskPriorityQueueSorter.Release>>m_18720_((p_140581_) -> {
+      return this.f_140552_.<ProcessorHandle<Release>>m_18720_((p_219080_2_) -> {
          return new StrictQueue.IntRunnable(0, () -> {
-            p_140581_.m_6937_(ProcessorHandle.m_18714_("chunk priority sorter around " + p_140568_.m_7326_(), (p_140603_) -> {
-               this.m_140569_(p_140568_, p_140603_.f_140683_, p_140603_.f_140682_, p_140603_.f_140684_);
+            p_219080_2_.m_6937_(ProcessorHandle.m_18714_("chunk priority sorter around " + p_140568_.m_7326_(), (p_219075_2_) -> {
+               this.m_140569_(p_140568_, p_219075_2_.f_140683_, p_219075_2_.f_140682_, p_219075_2_.f_140684_);
             }));
          });
       }).join();
@@ -78,8 +_,8 @@
    public void m_6250_(ChunkPos p_140616_, IntSupplier p_140617_, int p_140618_, IntConsumer p_140619_) {
       this.f_140552_.m_6937_(new StrictQueue.IntRunnable(0, () -> {
          int i = p_140617_.getAsInt();
-         this.f_140550_.values().forEach((p_140566_) -> {
-            p_140566_.m_140521_(i, p_140616_, p_140618_);
+         this.f_140550_.values().forEach((p_219076_3_) -> {
+            p_219076_3_.m_140521_(i, p_140616_, p_140618_);
          });
          p_140619_.accept(p_140618_);
       }));
@@ -119,12 +_,12 @@
          if (stream == null) {
             this.f_140551_.add(p_140647_);
          } else {
-            Util.m_137567_(stream.map((p_140588_) -> {
-               return p_140588_.map(p_140647_::m_18720_, (p_140623_) -> {
-                  p_140623_.run();
+            Util.m_137567_(stream.map((p_219092_1_) -> {
+               return p_219092_1_.map(p_140647_::m_18720_, (p_219077_0_) -> {
+                  p_219077_0_.run();
                   return CompletableFuture.completedFuture(Unit.INSTANCE);
                });
-            }).collect(Collectors.toList())).thenAccept((p_140651_) -> {
+            }).collect(Collectors.toList())).thenAccept((p_219088_3_) -> {
                this.m_140645_(p_140646_, p_140647_);
             });
          }
@@ -137,15 +_,15 @@
       if (chunktaskpriorityqueue == null) {
          throw (IllegalArgumentException)Util.m_137570_(new IllegalArgumentException("No queue for: " + p_140653_));
       } else {
-         return (ChunkTaskPriorityQueue<Function<ProcessorHandle<Unit>, T>>)chunktaskpriorityqueue;
+         return (ChunkTaskPriorityQueue<Function<ProcessorHandle<Unit>, T>>) chunktaskpriorityqueue;
       }
    }
 
    @VisibleForTesting
    public String m_140558_() {
-      return (String)this.f_140550_.entrySet().stream().map((p_140636_) -> {
-         return p_140636_.getKey().m_7326_() + "=[" + (String)p_140636_.getValue().m_140539_().stream().map((p_140621_) -> {
-            return p_140621_ + ":" + new ChunkPos(p_140621_);
+      return (String)this.f_140550_.entrySet().stream().map((p_225397_0_) -> {
+         return p_225397_0_.getKey().m_7326_() + "=[" + (String)p_225397_0_.getValue().m_140539_().stream().map((p_225398_0_) -> {
+            return p_225398_0_ + ":" + new ChunkPos(p_225398_0_);
          }).collect(Collectors.joining(",")) + "]";
       }).collect(Collectors.joining(",")) + ", s=" + this.f_140551_.size();
    }
