--- a/net/minecraft/server/level/ThreadedLevelLightEngine.java
+++ b/net/minecraft/server/level/ThreadedLevelLightEngine.java
@@ -147,8 +_,8 @@
 
          super.m_6460_(chunkpos, true);
          if (!p_9355_) {
-            p_9354_.m_6267_().forEach((p_9352_) -> {
-               super.m_8116_(p_9352_, p_9354_.m_7146_(p_9352_));
+            p_9354_.m_6267_().forEach((p_215579_2_) -> {
+               super.m_8116_(p_215579_2_, p_9354_.m_7146_(p_215579_2_));
             });
          }
 
@@ -160,8 +_,8 @@
          p_9354_.m_8094_(true);
          super.m_6462_(chunkpos, false);
          return p_9354_;
-      }, (p_9334_) -> {
-         this.m_9312_(chunkpos.f_45578_, chunkpos.f_45579_, ThreadedLevelLightEngine.TaskType.POST_UPDATE, p_9334_);
+      }, (p_215597_2_) -> {
+         this.m_9312_(chunkpos.f_45578_, chunkpos.f_45579_, ThreadedLevelLightEngine.TaskType.POST_UPDATE, p_215597_2_);
       });
    }
 
