--- a/net/minecraft/server/level/TicketType.java
+++ b/net/minecraft/server/level/TicketType.java
@@ -10,10 +_,10 @@
    private final String f_9450_;
    private final Comparator<T> f_9451_;
    private final long f_9452_;
-   public static final TicketType<Unit> f_9442_ = m_9462_("start", (p_9471_, p_9472_) -> {
+   public static final TicketType<Unit> f_9442_ = m_9462_("start", (p_219486_0_, p_219486_1_) -> {
       return 0;
    });
-   public static final TicketType<Unit> f_9443_ = m_9462_("dragon", (p_9460_, p_9461_) -> {
+   public static final TicketType<Unit> f_9443_ = m_9462_("dragon", (p_219485_0_, p_219485_1_) -> {
       return 0;
    });
    public static final TicketType<ChunkPos> f_9444_ = m_9462_("player", Comparator.comparingLong(ChunkPos::m_45588_));
