--- a/net/minecraft/world/item/crafting/AbstractCookingRecipe.java
+++ b/net/minecraft/world/item/crafting/AbstractCookingRecipe.java
@@ -5,8 +_,6 @@
 import net.minecraft.world.Container;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public abstract class AbstractCookingRecipe implements Recipe<Container> {
    protected final RecipeType<?> f_43726_;
@@ -35,7 +_,6 @@
       return this.f_43730_.m_41777_();
    }
 
-   @OnlyIn(Dist.CLIENT)
    public boolean m_8004_(int p_43743_, int p_43744_) {
       return true;
    }
@@ -54,7 +_,6 @@
       return this.f_43730_;
    }
 
-   @OnlyIn(Dist.CLIENT)
    public String m_6076_() {
       return this.f_43728_;
    }
