--- a/net/minecraft/world/item/crafting/FireworkStarFadeRecipe.java
+++ b/net/minecraft/world/item/crafting/FireworkStarFadeRecipe.java
@@ -9,8 +_,6 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.level.Level;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class FireworkStarFadeRecipe extends CustomRecipe {
    private static final Ingredient f_43858_ = Ingredient.m_43929_(Items.f_42689_);
@@ -68,7 +_,6 @@
       }
    }
 
-   @OnlyIn(Dist.CLIENT)
    public boolean m_8004_(int p_43863_, int p_43864_) {
       return p_43863_ * p_43864_ >= 2;
    }
