--- a/net/minecraft/world/item/crafting/FireworkStarRecipe.java
+++ b/net/minecraft/world/item/crafting/FireworkStarRecipe.java
@@ -14,23 +_,21 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.level.Level;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class FireworkStarRecipe extends CustomRecipe {
    private static final Ingredient f_43876_ = Ingredient.m_43929_(Items.f_42613_, Items.f_42402_, Items.f_42587_, Items.f_42678_, Items.f_42679_, Items.f_42682_, Items.f_42680_, Items.f_42683_, Items.f_42681_);
    private static final Ingredient f_43877_ = Ingredient.m_43929_(Items.f_42415_);
    private static final Ingredient f_43878_ = Ingredient.m_43929_(Items.f_42525_);
-   private static final Map<Item, FireworkRocketItem.Shape> f_43879_ = Util.m_137469_(Maps.newHashMap(), (p_43898_) -> {
-      p_43898_.put(Items.f_42613_, FireworkRocketItem.Shape.LARGE_BALL);
-      p_43898_.put(Items.f_42402_, FireworkRocketItem.Shape.BURST);
-      p_43898_.put(Items.f_42587_, FireworkRocketItem.Shape.STAR);
-      p_43898_.put(Items.f_42678_, FireworkRocketItem.Shape.CREEPER);
-      p_43898_.put(Items.f_42679_, FireworkRocketItem.Shape.CREEPER);
-      p_43898_.put(Items.f_42682_, FireworkRocketItem.Shape.CREEPER);
-      p_43898_.put(Items.f_42680_, FireworkRocketItem.Shape.CREEPER);
-      p_43898_.put(Items.f_42683_, FireworkRocketItem.Shape.CREEPER);
-      p_43898_.put(Items.f_42681_, FireworkRocketItem.Shape.CREEPER);
+   private static final Map<Item, FireworkRocketItem.Shape> f_43879_ = Util.m_137469_(Maps.newHashMap(), (p_209352_0_) -> {
+      p_209352_0_.put(Items.f_42613_, FireworkRocketItem.Shape.LARGE_BALL);
+      p_209352_0_.put(Items.f_42402_, FireworkRocketItem.Shape.BURST);
+      p_209352_0_.put(Items.f_42587_, FireworkRocketItem.Shape.STAR);
+      p_209352_0_.put(Items.f_42678_, FireworkRocketItem.Shape.CREEPER);
+      p_209352_0_.put(Items.f_42679_, FireworkRocketItem.Shape.CREEPER);
+      p_209352_0_.put(Items.f_42682_, FireworkRocketItem.Shape.CREEPER);
+      p_209352_0_.put(Items.f_42680_, FireworkRocketItem.Shape.CREEPER);
+      p_209352_0_.put(Items.f_42683_, FireworkRocketItem.Shape.CREEPER);
+      p_209352_0_.put(Items.f_42681_, FireworkRocketItem.Shape.CREEPER);
    });
    private static final Ingredient f_43880_ = Ingredient.m_43929_(Items.f_42403_);
 
@@ -111,7 +_,6 @@
       return itemstack;
    }
 
-   @OnlyIn(Dist.CLIENT)
    public boolean m_8004_(int p_43885_, int p_43886_) {
       return p_43885_ * p_43886_ >= 2;
    }
