--- a/net/minecraft/world/item/crafting/ShieldDecorationRecipe.java
+++ b/net/minecraft/world/item/crafting/ShieldDecorationRecipe.java
@@ -7,8 +_,6 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
 import net.minecraft.world.level.Level;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class ShieldDecorationRecipe extends CustomRecipe {
    public ShieldDecorationRecipe(ResourceLocation p_44296_) {
@@ -75,7 +_,6 @@
       }
    }
 
-   @OnlyIn(Dist.CLIENT)
    public boolean m_8004_(int p_44298_, int p_44299_) {
       return p_44298_ * p_44299_ >= 2;
    }
