--- a/net/minecraft/world/item/crafting/StonecutterRecipe.java
+++ b/net/minecraft/world/item/crafting/StonecutterRecipe.java
@@ -5,8 +_,6 @@
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Blocks;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class StonecutterRecipe extends SingleItemRecipe {
    public StonecutterRecipe(ResourceLocation p_44478_, String p_44479_, Ingredient p_44480_, ItemStack p_44481_) {
@@ -17,7 +_,6 @@
       return this.f_44409_.test(p_44483_.m_8020_(0));
    }
 
-   @OnlyIn(Dist.CLIENT)
    public ItemStack m_8042_() {
       return new ItemStack(Blocks.f_50679_);
    }
