--- a/net/minecraft/world/item/crafting/TippedArrowRecipe.java
+++ b/net/minecraft/world/item/crafting/TippedArrowRecipe.java
@@ -7,8 +_,6 @@
 import net.minecraft.world.item.Items;
 import net.minecraft.world.item.alchemy.PotionUtils;
 import net.minecraft.world.level.Level;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class TippedArrowRecipe extends CustomRecipe {
    public TippedArrowRecipe(ResourceLocation p_44503_) {
@@ -53,7 +_,6 @@
       }
    }
 
-   @OnlyIn(Dist.CLIENT)
    public boolean m_8004_(int p_44505_, int p_44506_) {
       return p_44505_ >= 2 && p_44506_ >= 2;
    }
