--- a/net/minecraft/world/entity/schedule/ScheduleBuilder.java
+++ b/net/minecraft/world/entity/schedule/ScheduleBuilder.java
@@ -19,12 +_,12 @@
 
    public Schedule m_38039_() {
       this.f_38036_.stream().map(ScheduleBuilder.ActivityTransition::m_38054_).collect(Collectors.toSet()).forEach(this.f_38035_::m_38024_);
-      this.f_38036_.forEach((p_38044_) -> {
-         Activity activity = p_38044_.m_38054_();
-         this.f_38035_.m_38033_(activity).forEach((p_38047_) -> {
-            p_38047_.m_38060_(p_38044_.m_38053_(), 0.0F);
+      this.f_38036_.forEach((p_221405_1_) -> {
+         Activity activity = p_221405_1_.m_38054_();
+         this.f_38035_.m_38033_(activity).forEach((p_221403_1_) -> {
+            p_221403_1_.m_38060_(p_221405_1_.m_38053_(), 0.0F);
          });
-         this.f_38035_.m_38031_(activity).m_38060_(p_38044_.m_38053_(), 1.0F);
+         this.f_38035_.m_38031_(activity).m_38060_(p_221405_1_.m_38053_(), 1.0F);
       });
       return this.f_38035_;
    }
