--- a/net/minecraft/world/entity/schedule/Timeline.java
+++ b/net/minecraft/world/entity/schedule/Timeline.java
@@ -17,8 +_,8 @@
 
    private void m_38066_() {
       Int2ObjectSortedMap<Keyframe> int2objectsortedmap = new Int2ObjectAVLTreeMap<>();
-      this.f_38055_.forEach((p_38065_) -> {
-         Keyframe keyframe = int2objectsortedmap.put(p_38065_.m_38010_(), p_38065_);
+      this.f_38055_.forEach((p_221393_1_) -> {
+         Keyframe keyframe = int2objectsortedmap.put(p_221393_1_.m_38010_(), p_221393_1_);
       });
       this.f_38055_.clear();
       this.f_38055_.addAll(int2objectsortedmap.values());
