--- a/net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen.java
+++ b/net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen.java
@@ -53,10 +_,10 @@
       super.m_7856_();
       this.f_101045_ = new EditGameRulesScreen.RuleList(this.f_101049_);
       this.f_96540_.add(this.f_101045_);
-      this.m_7082_(new Button(this.f_96543_ / 2 - 155 + 160, this.f_96544_ - 29, 150, 20, CommonComponents.f_130656_, (p_101073_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 155 + 160, this.f_96544_ - 29, 150, 20, CommonComponents.f_130656_, (p_238976_1_) -> {
          this.f_101044_.accept(Optional.empty());
       }));
-      this.f_101047_ = this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ - 29, 150, 20, CommonComponents.f_130655_, (p_101059_) -> {
+      this.f_101047_ = this.m_7082_(new Button(this.f_96543_ / 2 - 155, this.f_96544_ - 29, 150, 20, CommonComponents.f_130655_, (p_238971_1_) -> {
          this.f_101044_.accept(Optional.of(this.f_101049_));
       }));
    }
@@ -104,10 +_,10 @@
 
       public BooleanRuleEntry(final Component p_101101_, List<FormattedCharSequence> p_101102_, final String p_101103_, final GameRules.BooleanValue p_101104_) {
          super(p_101102_, p_101101_);
-         this.f_101098_ = new Button(10, 5, 44, 20, CommonComponents.m_130666_(p_101104_.m_46223_()), (p_101107_) -> {
+         this.f_101098_ = new Button(10, 5, 44, 20, CommonComponents.m_130666_(p_101104_.m_46223_()), (p_238988_1_) -> {
             boolean flag = !p_101104_.m_46223_();
             p_101104_.m_46246_(flag, (MinecraftServer)null);
-            p_101107_.m_93666_(CommonComponents.m_130666_(p_101104_.m_46223_()));
+            p_238988_1_.m_93666_(CommonComponents.m_130666_(p_101104_.m_46223_()));
          }) {
             protected MutableComponent m_5646_() {
                return CommonComponents.m_130663_(p_101101_, p_101104_.m_46223_()).m_130946_("\n").m_130946_(p_101103_);
@@ -125,84 +_,6 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   public class CategoryRuleEntry extends EditGameRulesScreen.RuleEntry {
-      private final Component f_101138_;
-
-      public CategoryRuleEntry(Component p_101141_) {
-         super((List<FormattedCharSequence>)null);
-         this.f_101138_ = p_101141_;
-      }
-
-      public void m_6311_(PoseStack p_101143_, int p_101144_, int p_101145_, int p_101146_, int p_101147_, int p_101148_, int p_101149_, int p_101150_, boolean p_101151_, float p_101152_) {
-         GuiComponent.m_93215_(p_101143_, EditGameRulesScreen.this.f_96541_.f_91062_, this.f_101138_, p_101146_ + p_101147_ / 2, p_101145_ + 5, 16777215);
-      }
-
-      public List<? extends GuiEventListener> m_6702_() {
-         return ImmutableList.of();
-      }
-   }
-
-   @FunctionalInterface
-   @OnlyIn(Dist.CLIENT)
-   interface EntryFactory<T extends GameRules.Value<T>> {
-      EditGameRulesScreen.RuleEntry m_101154_(Component p_101155_, List<FormattedCharSequence> p_101156_, String p_101157_, T p_101158_);
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   public abstract class GameRuleEntry extends EditGameRulesScreen.RuleEntry {
-      private final List<FormattedCharSequence> f_101159_;
-      protected final List<GuiEventListener> f_101160_ = Lists.newArrayList();
-
-      public GameRuleEntry(@Nullable List<FormattedCharSequence> p_101164_, Component p_101165_) {
-         super(p_101164_);
-         this.f_101159_ = EditGameRulesScreen.this.f_96541_.f_91062_.m_92923_(p_101165_, 175);
-      }
-
-      public List<? extends GuiEventListener> m_6702_() {
-         return this.f_101160_;
-      }
-
-      protected void m_101166_(PoseStack p_101167_, int p_101168_, int p_101169_) {
-         if (this.f_101159_.size() == 1) {
-            EditGameRulesScreen.this.f_96541_.f_91062_.m_92877_(p_101167_, this.f_101159_.get(0), (float)p_101169_, (float)(p_101168_ + 5), 16777215);
-         } else if (this.f_101159_.size() >= 2) {
-            EditGameRulesScreen.this.f_96541_.f_91062_.m_92877_(p_101167_, this.f_101159_.get(0), (float)p_101169_, (float)p_101168_, 16777215);
-            EditGameRulesScreen.this.f_96541_.f_91062_.m_92877_(p_101167_, this.f_101159_.get(1), (float)p_101169_, (float)(p_101168_ + 10), 16777215);
-         }
-
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   public class IntegerRuleEntry extends EditGameRulesScreen.GameRuleEntry {
-      private final EditBox f_101172_;
-
-      public IntegerRuleEntry(Component p_101175_, List<FormattedCharSequence> p_101176_, String p_101177_, GameRules.IntegerValue p_101178_) {
-         super(p_101176_, p_101175_);
-         this.f_101172_ = new EditBox(EditGameRulesScreen.this.f_96541_.f_91062_, 10, 5, 42, 20, p_101175_.m_6881_().m_130946_("\n").m_130946_(p_101177_).m_130946_("\n"));
-         this.f_101172_.m_94144_(Integer.toString(p_101178_.m_46288_()));
-         this.f_101172_.m_94151_((p_101181_) -> {
-            if (p_101178_.m_46314_(p_101181_)) {
-               this.f_101172_.m_94202_(14737632);
-               EditGameRulesScreen.this.m_101074_(this);
-            } else {
-               this.f_101172_.m_94202_(16711680);
-               EditGameRulesScreen.this.m_101060_(this);
-            }
-
-         });
-         this.f_101160_.add(this.f_101172_);
-      }
-
-      public void m_6311_(PoseStack p_101183_, int p_101184_, int p_101185_, int p_101186_, int p_101187_, int p_101188_, int p_101189_, int p_101190_, boolean p_101191_, float p_101192_) {
-         this.m_101166_(p_101183_, p_101185_, p_101186_);
-         this.f_101172_.f_93620_ = p_101186_ + p_101187_ - 44;
-         this.f_101172_.f_93621_ = p_101185_;
-         this.f_101172_.m_6305_(p_101183_, p_101189_, p_101190_, p_101192_);
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
    public abstract class RuleEntry extends ContainerObjectSelectionList.Entry<EditGameRulesScreen.RuleEntry> {
       @Nullable
       private final List<FormattedCharSequence> f_101193_;
@@ -219,14 +_,14 @@
          final Map<GameRules.Category, Map<GameRules.Key<?>, EditGameRulesScreen.RuleEntry>> map = Maps.newHashMap();
          GameRules.m_46164_(new GameRules.GameRuleTypeVisitor() {
             public void m_6891_(GameRules.Key<GameRules.BooleanValue> p_101238_, GameRules.Type<GameRules.BooleanValue> p_101239_) {
-               this.m_101224_(p_101238_, (p_101228_, p_101229_, p_101230_, p_101231_) -> {
-                  return EditGameRulesScreen.this.new BooleanRuleEntry(p_101228_, p_101229_, p_101230_, p_101231_);
+               this.m_101224_(p_101238_, (p_239012_1_, p_239012_2_, p_239012_3_, p_239012_4_) -> {
+                  return EditGameRulesScreen.this.new BooleanRuleEntry(p_239012_1_, p_239012_2_, p_239012_3_, p_239012_4_);
                });
             }
 
             public void m_6894_(GameRules.Key<GameRules.IntegerValue> p_101241_, GameRules.Type<GameRules.IntegerValue> p_101242_) {
-               this.m_101224_(p_101241_, (p_101233_, p_101234_, p_101235_, p_101236_) -> {
-                  return EditGameRulesScreen.this.new IntegerRuleEntry(p_101233_, p_101234_, p_101235_, p_101236_);
+               this.m_101224_(p_101241_, (p_239013_1_, p_239013_2_, p_239013_3_, p_239013_4_) -> {
+                  return EditGameRulesScreen.this.new IntegerRuleEntry(p_239013_1_, p_239013_2_, p_239013_3_, p_239013_4_);
                });
             }
 
@@ -250,15 +_,15 @@
                   s2 = component2.getString();
                }
 
-               map.computeIfAbsent(p_101225_.m_46332_(), (p_101223_) -> {
+               map.computeIfAbsent(p_101225_.m_46332_(), (p_239010_0_) -> {
                   return Maps.newHashMap();
                }).put(p_101225_, p_101226_.m_101154_(component, list, s2, t));
             }
          });
-         map.entrySet().stream().sorted(Map.Entry.comparingByKey()).forEach((p_101210_) -> {
-            this.m_7085_(EditGameRulesScreen.this.new CategoryRuleEntry((new TranslatableComponent(p_101210_.getKey().m_46274_())).m_130944_(new ChatFormatting[]{ChatFormatting.BOLD, ChatFormatting.YELLOW})));
-            p_101210_.getValue().entrySet().stream().sorted(Map.Entry.comparingByKey(Comparator.comparing(GameRules.Key::m_46328_))).forEach((p_101212_) -> {
-               this.m_7085_(p_101212_.getValue());
+         map.entrySet().stream().sorted(Map.Entry.comparingByKey()).forEach((p_239004_1_) -> {
+            this.m_7085_(EditGameRulesScreen.this.new CategoryRuleEntry((new TranslatableComponent(p_239004_1_.getKey().m_46274_())).m_130944_(new ChatFormatting[]{ChatFormatting.BOLD, ChatFormatting.YELLOW})));
+            p_239004_1_.getValue().entrySet().stream().sorted(Map.Entry.comparingByKey(Comparator.comparing(GameRules.Key::m_46328_))).forEach((p_239005_1_) -> {
+               this.m_7085_(p_239005_1_.getValue());
             });
          });
       }
@@ -270,6 +_,84 @@
             if (editgamerulesscreen$ruleentry != null) {
                EditGameRulesScreen.this.m_101081_(editgamerulesscreen$ruleentry.f_101193_);
             }
+         }
+
+      }
+   }
+
+   @FunctionalInterface
+   @OnlyIn(Dist.CLIENT)
+   interface EntryFactory<T extends GameRules.Value<T>> {
+      EditGameRulesScreen.RuleEntry m_101154_(Component p_101155_, List<FormattedCharSequence> p_101156_, String p_101157_, T p_101158_);
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public class IntegerRuleEntry extends EditGameRulesScreen.GameRuleEntry {
+      private final EditBox f_101172_;
+
+      public IntegerRuleEntry(Component p_101175_, List<FormattedCharSequence> p_101176_, String p_101177_, GameRules.IntegerValue p_101178_) {
+         super(p_101176_, p_101175_);
+         this.f_101172_ = new EditBox(EditGameRulesScreen.this.f_96541_.f_91062_, 10, 5, 42, 20, p_101175_.m_6881_().m_130946_("\n").m_130946_(p_101177_).m_130946_("\n"));
+         this.f_101172_.m_94144_(Integer.toString(p_101178_.m_46288_()));
+         this.f_101172_.m_94151_((p_238999_2_) -> {
+            if (p_101178_.m_46314_(p_238999_2_)) {
+               this.f_101172_.m_94202_(14737632);
+               EditGameRulesScreen.this.m_101074_(this);
+            } else {
+               this.f_101172_.m_94202_(16711680);
+               EditGameRulesScreen.this.m_101060_(this);
+            }
+
+         });
+         this.f_101160_.add(this.f_101172_);
+      }
+
+      public void m_6311_(PoseStack p_101183_, int p_101184_, int p_101185_, int p_101186_, int p_101187_, int p_101188_, int p_101189_, int p_101190_, boolean p_101191_, float p_101192_) {
+         this.m_101166_(p_101183_, p_101185_, p_101186_);
+         this.f_101172_.f_93620_ = p_101186_ + p_101187_ - 44;
+         this.f_101172_.f_93621_ = p_101185_;
+         this.f_101172_.m_6305_(p_101183_, p_101189_, p_101190_, p_101192_);
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public class CategoryRuleEntry extends EditGameRulesScreen.RuleEntry {
+      private final Component f_101138_;
+
+      public CategoryRuleEntry(Component p_101141_) {
+         super((List<FormattedCharSequence>)null);
+         this.f_101138_ = p_101141_;
+      }
+
+      public void m_6311_(PoseStack p_101143_, int p_101144_, int p_101145_, int p_101146_, int p_101147_, int p_101148_, int p_101149_, int p_101150_, boolean p_101151_, float p_101152_) {
+         GuiComponent.m_93215_(p_101143_, EditGameRulesScreen.this.f_96541_.f_91062_, this.f_101138_, p_101146_ + p_101147_ / 2, p_101145_ + 5, 16777215);
+      }
+
+      public List<? extends GuiEventListener> m_6702_() {
+         return ImmutableList.of();
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public abstract class GameRuleEntry extends EditGameRulesScreen.RuleEntry {
+      private final List<FormattedCharSequence> f_101159_;
+      protected final List<GuiEventListener> f_101160_ = Lists.newArrayList();
+
+      public GameRuleEntry(@Nullable List<FormattedCharSequence> p_101164_, Component p_101165_) {
+         super(p_101164_);
+         this.f_101159_ = EditGameRulesScreen.this.f_96541_.f_91062_.m_92923_(p_101165_, 175);
+      }
+
+      public List<? extends GuiEventListener> m_6702_() {
+         return this.f_101160_;
+      }
+
+      protected void m_101166_(PoseStack p_101167_, int p_101168_, int p_101169_) {
+         if (this.f_101159_.size() == 1) {
+            EditGameRulesScreen.this.f_96541_.f_91062_.m_92877_(p_101167_, this.f_101159_.get(0), (float)p_101169_, (float)(p_101168_ + 5), 16777215);
+         } else if (this.f_101159_.size() >= 2) {
+            EditGameRulesScreen.this.f_96541_.f_91062_.m_92877_(p_101167_, this.f_101159_.get(0), (float)p_101169_, (float)p_101168_, 16777215);
+            EditGameRulesScreen.this.f_96541_.f_91062_.m_92877_(p_101167_, this.f_101159_.get(1), (float)p_101169_, (float)(p_101168_ + 10), 16777215);
          }
 
       }
