--- a/net/minecraft/client/gui/screens/worldselection/EditWorldScreen.java
+++ b/net/minecraft/client/gui/screens/worldselection/EditWorldScreen.java
@@ -63,18 +_,18 @@
 
    protected void m_7856_() {
       this.f_96541_.f_91068_.m_90926_(true);
-      Button button = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 0 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.resetIcon"), (p_101297_) -> {
+      Button button = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 0 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.resetIcon"), (p_214309_1_) -> {
          FileUtils.deleteQuietly(this.f_101249_.m_78310_());
-         p_101297_.f_93623_ = false;
+         p_214309_1_.f_93623_ = false;
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 24 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.openFolder"), (p_101294_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 24 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.openFolder"), (p_214303_1_) -> {
          Util.m_137581_().m_137644_(this.f_101249_.m_78283_(LevelResource.f_78182_).toFile());
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 48 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.backup"), (p_101292_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 48 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.backup"), (p_214304_1_) -> {
          boolean flag = m_101258_(this.f_101249_);
          this.f_101247_.accept(!flag);
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 72 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.backupFolder"), (p_101290_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 72 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.backupFolder"), (p_214302_1_) -> {
          LevelStorageSource levelstoragesource = this.f_96541_.m_91392_();
          Path path = levelstoragesource.m_78262_();
 
@@ -86,27 +_,27 @@
 
          Util.m_137581_().m_137644_(path.toFile());
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 96 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.optimize"), (p_101287_) -> {
-         this.f_96541_.m_91152_(new BackupConfirmScreen(this, (p_101275_, p_101276_) -> {
-            if (p_101275_) {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 96 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.optimize"), (p_214310_1_) -> {
+         this.f_96541_.m_91152_(new BackupConfirmScreen(this, (p_214305_1_, p_214305_2_) -> {
+            if (p_214305_1_) {
                m_101258_(this.f_101249_);
             }
 
-            this.f_96541_.m_91152_(OptimizeWorldScreen.m_101315_(this.f_96541_, this.f_101247_, this.f_96541_.m_91295_(), this.f_101249_, p_101276_));
+            this.f_96541_.m_91152_(OptimizeWorldScreen.m_101315_(this.f_96541_, this.f_101247_, this.f_96541_.m_91295_(), this.f_101249_, p_214305_2_));
          }, new TranslatableComponent("optimizeWorld.confirm.title"), new TranslatableComponent("optimizeWorld.confirm.description"), true));
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 120 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.export_worldgen_settings"), (p_101284_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 120 + 5, 200, 20, new TranslatableComponent("selectWorld.edit.export_worldgen_settings"), (p_239023_1_) -> {
          RegistryAccess.RegistryHolder registryaccess$registryholder = RegistryAccess.m_123086_();
 
          DataResult<String> dataresult;
          try (Minecraft.ServerStem minecraft$serverstem = this.f_96541_.m_91190_(registryaccess$registryholder, Minecraft::m_91133_, Minecraft::m_91135_, false, this.f_101249_)) {
             DynamicOps<JsonElement> dynamicops = RegistryWriteOps.m_135767_(JsonOps.INSTANCE, registryaccess$registryholder);
             DataResult<JsonElement> dataresult1 = WorldGenSettings.f_64600_.encodeStart(dynamicops, minecraft$serverstem.m_91434_().m_5961_());
-            dataresult = dataresult1.flatMap((p_101255_) -> {
+            dataresult = dataresult1.flatMap((p_239017_1_) -> {
                Path path = this.f_101249_.m_78283_(LevelResource.f_78182_).resolve("worldgen_settings_export.json");
 
                try (JsonWriter jsonwriter = f_101244_.newJsonWriter(Files.newBufferedWriter(path, StandardCharsets.UTF_8))) {
-                  f_101244_.toJson(p_101255_, jsonwriter);
+                  f_101244_.toJson(p_239017_1_, jsonwriter);
                } catch (JsonIOException | IOException ioexception) {
                   return DataResult.error("Error writing file: " + ioexception.getMessage());
                }
@@ -119,15 +_,15 @@
 
          Component component = new TextComponent(dataresult.get().map(Function.identity(), PartialResult::message));
          Component component1 = new TranslatableComponent(dataresult.result().isPresent() ? "selectWorld.edit.export_worldgen_settings.success" : "selectWorld.edit.export_worldgen_settings.failure");
-         dataresult.error().ifPresent((p_101257_) -> {
-            f_101243_.error("Error exporting world settings: {}", (Object)p_101257_);
+         dataresult.error().ifPresent((p_239018_0_) -> {
+            f_101243_.error("Error exporting world settings: {}", (Object)p_239018_0_);
          });
          this.f_96541_.m_91300_().m_94922_(SystemToast.m_94847_(this.f_96541_, SystemToast.SystemToastIds.WORLD_GEN_SETTINGS_TRANSFER, component1, component));
       }));
-      this.f_101246_ = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 144 + 5, 98, 20, new TranslatableComponent("selectWorld.edit.save"), (p_101280_) -> {
+      this.f_101246_ = this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 144 + 5, 98, 20, new TranslatableComponent("selectWorld.edit.save"), (p_214308_1_) -> {
          this.m_101295_();
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 + 2, this.f_96544_ / 4 + 144 + 5, 98, 20, CommonComponents.f_130656_, (p_101273_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 + 2, this.f_96544_ / 4 + 144 + 5, 98, 20, CommonComponents.f_130656_, (p_214306_1_) -> {
          this.f_101247_.accept(false);
       }));
       button.f_93623_ = this.f_101249_.m_78310_().isFile();
@@ -135,8 +_,8 @@
       String s = levelsummary == null ? "" : levelsummary.m_78361_();
       this.f_101248_ = new EditBox(this.f_96547_, this.f_96543_ / 2 - 100, 38, 200, 20, new TranslatableComponent("selectWorld.enterName"));
       this.f_101248_.m_94144_(s);
-      this.f_101248_.m_94151_((p_101282_) -> {
-         this.f_101246_.f_93623_ = !p_101282_.trim().isEmpty();
+      this.f_101248_.m_94151_((p_214301_1_) -> {
+         this.f_101246_.f_93623_ = !p_214301_1_.trim().isEmpty();
       });
       this.f_96540_.add(this.f_101248_);
       this.m_94718_(this.f_101248_);
