--- a/net/minecraft/client/gui/screens/worldselection/OptimizeWorldScreen.java
+++ b/net/minecraft/client/gui/screens/worldselection/OptimizeWorldScreen.java
@@ -29,11 +_,11 @@
 @OnlyIn(Dist.CLIENT)
 public class OptimizeWorldScreen extends Screen {
    private static final Logger f_101298_ = LogManager.getLogger();
-   private static final Object2IntMap<ResourceKey<Level>> f_101299_ = Util.m_137469_(new Object2IntOpenCustomHashMap<>(Util.m_137583_()), (p_101324_) -> {
-      p_101324_.put(Level.f_46428_, -13408734);
-      p_101324_.put(Level.f_46429_, -10075085);
-      p_101324_.put(Level.f_46430_, -8943531);
-      p_101324_.defaultReturnValue(-2236963);
+   private static final Object2IntMap<ResourceKey<Level>> f_101299_ = Util.m_137469_(new Object2IntOpenCustomHashMap<>(Util.m_137583_()), (p_212346_0_) -> {
+      p_212346_0_.put(Level.f_46428_, -13408734);
+      p_212346_0_.put(Level.f_46429_, -10075085);
+      p_212346_0_.put(Level.f_46430_, -8943531);
+      p_212346_0_.defaultReturnValue(-2236963);
    });
    private final BooleanConsumer f_101300_;
    private final WorldUpgrader f_101301_;
@@ -61,7 +_,7 @@
 
    protected void m_7856_() {
       super.m_7856_();
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 150, 200, 20, CommonComponents.f_130656_, (p_101322_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 150, 200, 20, CommonComponents.f_130656_, (p_214331_1_) -> {
          this.f_101301_.m_18820_();
          this.f_101300_.accept(false);
       }));
