--- a/net/minecraft/world/level/block/state/StateHolder.java
+++ b/net/minecraft/world/level/block/state/StateHolder.java
@@ -145,10 +_,10 @@
    }
 
    protected static <O, S extends StateHolder<O, S>> Codec<S> m_61127_(Codec<O> p_61128_, Function<O, S> p_61129_) {
-      return p_61128_.dispatch("Name", (p_61121_) -> {
-         return p_61121_.f_61112_;
-      }, (p_61137_) -> {
-         S s = p_61129_.apply(p_61137_);
+      return p_61128_.dispatch("Name", (p_235895_0_) -> {
+         return p_235895_0_.f_61112_;
+      }, (p_235900_1_) -> {
+         S s = p_61129_.apply(p_235900_1_);
          return s.m_61148_().isEmpty() ? Codec.unit(s) : s.f_61113_.fieldOf("Properties").codec();
       });
    }
