--- a/net/minecraft/world/level/chunk/storage/SectionStorage.java
+++ b/net/minecraft/world/level/chunk/storage/SectionStorage.java
@@ -127,13 +_,13 @@
 
          for(int l = 0; l < 16; ++l) {
             long i1 = SectionPos.m_123196_(p_63802_, l).m_123252_();
-            Optional<R> optional = optionaldynamic.get(Integer.toString(l)).result().flatMap((p_63791_) -> {
+            Optional<R> optional = optionaldynamic.get(Integer.toString(l)).result().flatMap((p_235989_3_) -> {
                return this.f_63776_.apply(() -> {
                   this.m_5838_(i1);
-               }).parse(p_63791_).resultOrPartial(f_63772_::error);
+               }).parse(p_235989_3_).resultOrPartial(f_63772_::error);
             });
             this.f_63774_.put(i1, optional);
-            optional.ifPresent((p_63795_) -> {
+            optional.ifPresent((p_235990_4_) -> {
                this.m_5839_(i1);
                if (flag) {
                   this.m_5838_(i1);
@@ -168,8 +_,8 @@
                this.m_5838_(j);
             }).encodeStart(p_63800_, optional.get());
             String s = Integer.toString(i);
-            dataresult.resultOrPartial(f_63772_::error).ifPresent((p_63811_) -> {
-               map.put(p_63800_.createString(s), p_63811_);
+            dataresult.resultOrPartial(f_63772_::error).ifPresent((p_235994_3_) -> {
+               map.put(p_63800_.createString(s), p_235994_3_);
             });
          }
       }
