--- a/net/minecraft/world/level/levelgen/feature/structures/FeaturePoolElement.java
+++ b/net/minecraft/world/level/levelgen/feature/structures/FeaturePoolElement.java
@@ -24,10 +_,10 @@
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate;
 
 public class FeaturePoolElement extends StructurePoolElement {
-   public static final Codec<FeaturePoolElement> f_68882_ = RecordCodecBuilder.create((p_68893_) -> {
-      return p_68893_.group(ConfiguredFeature.f_65374_.fieldOf("feature").forGetter((p_68891_) -> {
-         return p_68891_.f_68883_;
-      }), m_69229_()).apply(p_68893_, FeaturePoolElement::new);
+   public static final Codec<FeaturePoolElement> f_68882_ = RecordCodecBuilder.create((p_236817_0_) -> {
+      return p_236817_0_.group(ConfiguredFeature.f_65374_.fieldOf("feature").forGetter((p_236818_0_) -> {
+         return p_236818_0_.f_68883_;
+      }), m_69229_()).apply(p_236817_0_, FeaturePoolElement::new);
    });
    private final Supplier<ConfiguredFeature<?, ?>> f_68883_;
    private final CompoundTag f_68884_;
