--- a/net/minecraft/world/entity/ai/behavior/AnimalMakeLove.java
+++ b/net/minecraft/world/entity/ai/behavior/AnimalMakeLove.java
@@ -72,10 +_,10 @@
    }
 
    private Optional<? extends Animal> m_22431_(Animal p_22432_) {
-      return p_22432_.m_6274_().m_21952_(MemoryModuleType.f_26365_).get().stream().filter((p_22420_) -> {
-         return p_22420_.m_6095_() == this.f_22387_;
-      }).map((p_22408_) -> {
-         return (Animal)p_22408_;
+      return p_22432_.m_6274_().m_21952_(MemoryModuleType.f_26365_).get().stream().filter((p_233847_1_) -> {
+         return p_233847_1_.m_6095_() == this.f_22387_;
+      }).map((p_233845_0_) -> {
+         return (Animal)p_233845_0_;
       }).filter(p_22432_::m_7848_).findFirst();
    }
 }
