--- a/net/minecraft/world/entity/ai/behavior/BehaviorUtils.java
+++ b/net/minecraft/world/entity/ai/behavior/BehaviorUtils.java
@@ -29,20 +_,20 @@
    }
 
    public static boolean m_22636_(Brain<?> p_22637_, LivingEntity p_22638_) {
-      return p_22637_.m_21952_(MemoryModuleType.f_26365_).filter((p_22624_) -> {
-         return p_22624_.contains(p_22638_);
+      return p_22637_.m_21952_(MemoryModuleType.f_26365_).filter((p_220614_1_) -> {
+         return p_220614_1_.contains(p_22638_);
       }).isPresent();
    }
 
    public static boolean m_22639_(Brain<?> p_22640_, MemoryModuleType<? extends LivingEntity> p_22641_, EntityType<?> p_22642_) {
-      return m_22643_(p_22640_, p_22641_, (p_22587_) -> {
-         return p_22587_.m_6095_() == p_22642_;
+      return m_22643_(p_22640_, p_22641_, (p_220622_1_) -> {
+         return p_220622_1_.m_6095_() == p_22642_;
       });
    }
 
    private static boolean m_22643_(Brain<?> p_22644_, MemoryModuleType<? extends LivingEntity> p_22645_, Predicate<LivingEntity> p_22646_) {
-      return p_22644_.m_21952_(p_22645_).filter(p_22646_).filter(LivingEntity::m_6084_).filter((p_22666_) -> {
-         return m_22636_(p_22644_, p_22666_);
+      return p_22644_.m_21952_(p_22645_).filter(p_22646_).filter(LivingEntity::m_6084_).filter((p_220615_1_) -> {
+         return m_22636_(p_22644_, p_220615_1_);
       }).isPresent();
    }
 
@@ -86,8 +_,8 @@
 
    public static SectionPos m_22581_(ServerLevel p_22582_, SectionPos p_22583_, int p_22584_) {
       int i = p_22582_.m_8828_(p_22583_);
-      return SectionPos.m_123201_(p_22583_, p_22584_).filter((p_22580_) -> {
-         return p_22582_.m_8828_(p_22580_) < i;
+      return SectionPos.m_123201_(p_22583_, p_22584_).filter((p_220620_2_) -> {
+         return p_22582_.m_8828_(p_220620_2_) < i;
       }).min(Comparator.comparingInt(p_22582_::m_8828_)).orElse(p_22583_);
    }
 
@@ -135,17 +_,17 @@
 
    public static Optional<LivingEntity> m_22610_(LivingEntity p_22611_, MemoryModuleType<UUID> p_22612_) {
       Optional<UUID> optional = p_22611_.m_6274_().m_21952_(p_22612_);
-      return optional.map((p_22631_) -> {
-         return (LivingEntity)((ServerLevel)p_22611_.f_19853_).m_8791_(p_22631_);
+      return optional.map((p_233868_1_) -> {
+         return (LivingEntity)((ServerLevel)p_22611_.f_19853_).m_8791_(p_233868_1_);
       });
    }
 
    public static Stream<Villager> m_22650_(Villager p_22651_, Predicate<Villager> p_22652_) {
-      return p_22651_.m_6274_().m_21952_(MemoryModuleType.f_26364_).map((p_22656_) -> {
-         return p_22656_.stream().filter((p_22649_) -> {
-            return p_22649_ instanceof Villager && p_22649_ != p_22651_;
-         }).map((p_22589_) -> {
-            return (Villager)p_22589_;
+      return p_22651_.m_6274_().m_21952_(MemoryModuleType.f_26364_).map((p_233873_2_) -> {
+         return p_233873_2_.stream().filter((p_233871_1_) -> {
+            return p_233871_1_ instanceof Villager && p_233871_1_ != p_22651_;
+         }).map((p_233859_0_) -> {
+            return (Villager)p_233859_0_;
          }).filter(LivingEntity::m_6084_).filter(p_22652_);
       }).orElseGet(Stream::empty);
    }
