--- a/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
+++ b/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
@@ -26,20 +_,20 @@
 import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
 
 public class GiveGiftToHero extends Behavior<Villager> {
-   private static final Map<VillagerProfession, ResourceLocation> f_22986_ = Util.m_137469_(Maps.newHashMap(), (p_23020_) -> {
-      p_23020_.put(VillagerProfession.f_35586_, BuiltInLootTables.f_78725_);
-      p_23020_.put(VillagerProfession.f_35587_, BuiltInLootTables.f_78726_);
-      p_23020_.put(VillagerProfession.f_35588_, BuiltInLootTables.f_78727_);
-      p_23020_.put(VillagerProfession.f_35589_, BuiltInLootTables.f_78728_);
-      p_23020_.put(VillagerProfession.f_35590_, BuiltInLootTables.f_78729_);
-      p_23020_.put(VillagerProfession.f_35591_, BuiltInLootTables.f_78730_);
-      p_23020_.put(VillagerProfession.f_35592_, BuiltInLootTables.f_78731_);
-      p_23020_.put(VillagerProfession.f_35593_, BuiltInLootTables.f_78732_);
-      p_23020_.put(VillagerProfession.f_35594_, BuiltInLootTables.f_78733_);
-      p_23020_.put(VillagerProfession.f_35595_, BuiltInLootTables.f_78734_);
-      p_23020_.put(VillagerProfession.f_35597_, BuiltInLootTables.f_78735_);
-      p_23020_.put(VillagerProfession.f_35598_, BuiltInLootTables.f_78736_);
-      p_23020_.put(VillagerProfession.f_35599_, BuiltInLootTables.f_78737_);
+   private static final Map<VillagerProfession, ResourceLocation> f_22986_ = Util.m_137469_(Maps.newHashMap(), (p_220395_0_) -> {
+      p_220395_0_.put(VillagerProfession.f_35586_, BuiltInLootTables.f_78725_);
+      p_220395_0_.put(VillagerProfession.f_35587_, BuiltInLootTables.f_78726_);
+      p_220395_0_.put(VillagerProfession.f_35588_, BuiltInLootTables.f_78727_);
+      p_220395_0_.put(VillagerProfession.f_35589_, BuiltInLootTables.f_78728_);
+      p_220395_0_.put(VillagerProfession.f_35590_, BuiltInLootTables.f_78729_);
+      p_220395_0_.put(VillagerProfession.f_35591_, BuiltInLootTables.f_78730_);
+      p_220395_0_.put(VillagerProfession.f_35592_, BuiltInLootTables.f_78731_);
+      p_220395_0_.put(VillagerProfession.f_35593_, BuiltInLootTables.f_78732_);
+      p_220395_0_.put(VillagerProfession.f_35594_, BuiltInLootTables.f_78733_);
+      p_220395_0_.put(VillagerProfession.f_35595_, BuiltInLootTables.f_78734_);
+      p_220395_0_.put(VillagerProfession.f_35597_, BuiltInLootTables.f_78735_);
+      p_220395_0_.put(VillagerProfession.f_35598_, BuiltInLootTables.f_78736_);
+      p_220395_0_.put(VillagerProfession.f_35599_, BuiltInLootTables.f_78737_);
    });
    private int f_22987_ = 600;
    private boolean f_22988_;
