--- a/net/minecraft/world/entity/ai/behavior/GoToClosestVillage.java
+++ b/net/minecraft/world/entity/ai/behavior/GoToClosestVillage.java
@@ -32,8 +_,8 @@
       Vec3 vec3 = null;
 
       for(int j = 0; j < 5; ++j) {
-         Vec3 vec31 = RandomPos.m_26939_(p_23091_, 15, 7, (p_23095_) -> {
-            return (double)(-p_23090_.m_8828_(SectionPos.m_123199_(p_23095_)));
+         Vec3 vec31 = RandomPos.m_26939_(p_23091_, 15, 7, (p_225444_1_) -> {
+            return (double)(-p_23090_.m_8828_(SectionPos.m_123199_(p_225444_1_)));
          });
          if (vec31 != null) {
             int k = poimanager.m_27098_(SectionPos.m_123199_(new BlockPos(vec31)));
