--- a/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
+++ b/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
@@ -28,15 +_,15 @@
       BlockPos blockpos = p_23222_.m_20183_();
       List<BlockPos> list = BlockPos.m_121990_(blockpos.m_7918_(-1, -1, -1), blockpos.m_7918_(1, 1, 1)).map(BlockPos::m_7949_).collect(Collectors.toList());
       Collections.shuffle(list);
-      Optional<BlockPos> optional = list.stream().filter((p_23230_) -> {
-         return !p_23221_.m_45527_(p_23230_);
-      }).filter((p_23237_) -> {
-         return p_23221_.m_46575_(p_23237_, p_23222_);
-      }).filter((p_23227_) -> {
+      Optional<BlockPos> optional = list.stream().filter((p_220428_1_) -> {
+         return !p_23221_.m_45527_(p_220428_1_);
+      }).filter((p_220427_2_) -> {
+         return p_23221_.m_46575_(p_220427_2_, p_23222_);
+      }).filter((p_220429_2_) -> {
          return p_23221_.m_45786_(p_23222_);
       }).findFirst();
-      optional.ifPresent((p_23233_) -> {
-         p_23222_.m_6274_().m_21879_(MemoryModuleType.f_26370_, new WalkTarget(p_23233_, this.f_23207_, 0));
+      optional.ifPresent((p_220430_2_) -> {
+         p_23222_.m_6274_().m_21879_(MemoryModuleType.f_26370_, new WalkTarget(p_220430_2_, this.f_23207_, 0));
       });
    }
 }
