--- a/net/minecraft/world/entity/ai/behavior/InteractWith.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWith.java
@@ -32,9 +_,9 @@
    }
 
    public static <T extends LivingEntity> InteractWith<LivingEntity, T> m_23260_(EntityType<? extends T> p_23261_, int p_23262_, MemoryModuleType<T> p_23263_, float p_23264_, int p_23265_) {
-      return new InteractWith<>(p_23261_, p_23262_, (p_23287_) -> {
+      return new InteractWith<>(p_23261_, p_23262_, (p_220441_0_) -> {
          return true;
-      }, (p_23285_) -> {
+      }, (p_220442_0_) -> {
          return true;
       }, p_23263_, p_23264_, p_23265_);
    }
@@ -54,17 +_,17 @@
 
    protected void m_6735_(ServerLevel p_23257_, E p_23258_, long p_23259_) {
       Brain<?> brain = p_23258_.m_6274_();
-      brain.m_21952_(MemoryModuleType.f_26365_).ifPresent((p_23274_) -> {
-         p_23274_.stream().filter((p_23283_) -> {
-            return this.f_23240_.equals(p_23283_.m_6095_());
-         }).map((p_23281_) -> {
-            return (T)p_23281_;
-         }).filter((p_23270_) -> {
-            return p_23270_.m_20280_(p_23258_) <= (double)this.f_23241_;
-         }).filter(this.f_23242_).findFirst().ifPresent((p_23277_) -> {
-            brain.m_21879_(this.f_23244_, (T)p_23277_);
-            brain.m_21879_(MemoryModuleType.f_26371_, new EntityTracker(p_23277_, true));
-            brain.m_21879_(MemoryModuleType.f_26370_, new WalkTarget(new EntityTracker(p_23277_, false), this.f_23239_, this.f_23238_));
+      brain.m_21952_(MemoryModuleType.f_26365_).ifPresent((p_220437_3_) -> {
+         p_220437_3_.stream().filter((p_220440_1_) -> {
+            return this.f_23240_.equals(p_220440_1_.m_6095_());
+         }).map((p_220439_0_) -> {
+            return (T)p_220439_0_;
+         }).filter((p_220443_2_) -> {
+            return p_220443_2_.m_20280_(p_23258_) <= (double)this.f_23241_;
+         }).filter(this.f_23242_).findFirst().ifPresent((p_220438_2_) -> {
+            brain.m_21879_(this.f_23244_, (T)p_220438_2_);
+            brain.m_21879_(MemoryModuleType.f_26371_, new EntityTracker(p_220438_2_, true));
+            brain.m_21879_(MemoryModuleType.f_26370_, new WalkTarget(new EntityTracker(p_220438_2_, false), this.f_23239_, this.f_23238_));
          });
       });
    }
