--- a/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
@@ -109,12 +_,12 @@
 
    private static boolean m_23303_(ServerLevel p_23304_, LivingEntity p_23305_, BlockPos p_23306_) {
       Brain<?> brain = p_23305_.m_6274_();
-      return !brain.m_21874_(MemoryModuleType.f_26364_) ? false : brain.m_21952_(MemoryModuleType.f_26364_).get().stream().filter((p_23317_) -> {
-         return p_23317_.m_6095_() == p_23305_.m_6095_();
-      }).filter((p_23320_) -> {
-         return p_23306_.m_123306_(p_23320_.m_20182_(), 2.0D);
-      }).anyMatch((p_23314_) -> {
-         return m_23321_(p_23304_, p_23314_, p_23306_);
+      return !brain.m_21874_(MemoryModuleType.f_26364_) ? false : brain.m_21952_(MemoryModuleType.f_26364_).get().stream().filter((p_242298_1_) -> {
+         return p_242298_1_.m_6095_() == p_23305_.m_6095_();
+      }).filter((p_242299_1_) -> {
+         return p_23306_.m_123306_(p_242299_1_.m_20182_(), 2.0D);
+      }).anyMatch((p_242297_2_) -> {
+         return m_23321_(p_23304_, p_242297_2_, p_23306_);
       });
    }
 
