--- a/net/minecraft/world/entity/ai/behavior/JumpOnBed.java
+++ b/net/minecraft/world/entity/ai/behavior/JumpOnBed.java
@@ -30,12 +_,12 @@
 
    protected void m_6735_(ServerLevel p_23349_, Mob p_23350_, long p_23351_) {
       super.m_6735_(p_23349_, p_23350_, p_23351_);
-      this.m_23359_(p_23350_).ifPresent((p_23355_) -> {
-         this.f_23330_ = p_23355_;
+      this.m_23359_(p_23350_).ifPresent((p_220461_3_) -> {
+         this.f_23330_ = p_220461_3_;
          this.f_23331_ = 100;
          this.f_23332_ = 3 + p_23349_.f_46441_.nextInt(4);
          this.f_23333_ = 0;
-         this.m_23361_(p_23350_, p_23355_);
+         this.m_23361_(p_23350_, p_220461_3_);
       });
    }
 
