--- a/net/minecraft/world/entity/ai/behavior/LocateHidingPlace.java
+++ b/net/minecraft/world/entity/ai/behavior/LocateHidingPlace.java
@@ -27,9 +_,9 @@
    }
 
    protected boolean m_6114_(ServerLevel p_23412_, LivingEntity p_23413_) {
-      Optional<BlockPos> optional = p_23412_.m_8904_().m_27186_((p_23423_) -> {
-         return p_23423_ == PoiType.f_27346_;
-      }, (p_23425_) -> {
+      Optional<BlockPos> optional = p_23412_.m_8904_().m_27186_((p_220454_0_) -> {
+         return p_220454_0_ == PoiType.f_27346_;
+      }, (p_220456_0_) -> {
          return true;
       }, p_23413_.m_20183_(), this.f_23405_ + 1, PoiManager.Occupancy.ANY);
       if (optional.isPresent() && optional.get().m_123306_(p_23413_.m_20182_(), (double)this.f_23405_)) {
@@ -45,9 +_,9 @@
       Brain<?> brain = p_23416_.m_6274_();
       Optional<BlockPos> optional = this.f_23406_;
       if (!optional.isPresent()) {
-         optional = p_23415_.m_8904_().m_27126_((p_23419_) -> {
-            return p_23419_ == PoiType.f_27346_;
-         }, (p_23421_) -> {
+         optional = p_23415_.m_8904_().m_27126_((p_220453_0_) -> {
+            return p_220453_0_ == PoiType.f_27346_;
+         }, (p_220455_0_) -> {
             return true;
          }, PoiManager.Occupancy.ANY, p_23416_.m_20183_(), this.f_23404_, p_23416_.m_21187_());
          if (!optional.isPresent()) {
