--- a/net/minecraft/world/entity/ai/behavior/MeleeAttack.java
+++ b/net/minecraft/world/entity/ai/behavior/MeleeAttack.java
@@ -23,8 +_,8 @@
    }
 
    private boolean m_23527_(Mob p_23528_) {
-      return p_23528_.m_21093_((p_23531_) -> {
-         return p_23531_ instanceof ProjectileWeaponItem && p_23528_.m_5886_((ProjectileWeaponItem)p_23531_);
+      return p_23528_.m_21093_((p_233922_1_) -> {
+         return p_233922_1_ instanceof ProjectileWeaponItem && p_23528_.m_5886_((ProjectileWeaponItem)p_233922_1_);
       });
    }
 
