--- a/net/minecraft/world/entity/ai/behavior/MoveToSkySeeingSpot.java
+++ b/net/minecraft/world/entity/ai/behavior/MoveToSkySeeingSpot.java
@@ -24,8 +_,8 @@
    protected void m_6735_(ServerLevel p_23555_, LivingEntity p_23556_, long p_23557_) {
       Optional<Vec3> optional = Optional.ofNullable(this.m_23564_(p_23555_, p_23556_));
       if (optional.isPresent()) {
-         p_23556_.m_6274_().m_21886_(MemoryModuleType.f_26370_, optional.map((p_23563_) -> {
-            return new WalkTarget(p_23563_, this.f_23548_, 0);
+         p_23556_.m_6274_().m_21886_(MemoryModuleType.f_26370_, optional.map((p_220492_1_) -> {
+            return new WalkTarget(p_220492_1_, this.f_23548_, 0);
          }));
       }
 
