--- a/net/minecraft/world/entity/ai/behavior/RandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/RandomStroll.java
@@ -28,8 +_,8 @@
 
    protected void m_6735_(ServerLevel p_23754_, PathfinderMob p_23755_, long p_23756_) {
       Optional<Vec3> optional = Optional.ofNullable(RandomPos.m_26959_(p_23755_, this.f_23741_, this.f_23742_));
-      p_23755_.m_6274_().m_21886_(MemoryModuleType.f_26370_, optional.map((p_23758_) -> {
-         return new WalkTarget(p_23758_, this.f_23740_, 0);
+      p_23755_.m_6274_().m_21886_(MemoryModuleType.f_26370_, optional.map((p_233939_1_) -> {
+         return new WalkTarget(p_233939_1_, this.f_23740_, 0);
       }));
    }
 }
