--- a/net/minecraft/world/entity/ai/behavior/SetEntityLookTarget.java
+++ b/net/minecraft/world/entity/ai/behavior/SetEntityLookTarget.java
@@ -15,19 +_,19 @@
    private final float f_23890_;
 
    public SetEntityLookTarget(MobCategory p_23897_, float p_23898_) {
-      this((p_23923_) -> {
-         return p_23897_.equals(p_23923_.m_6095_().m_20674_());
+      this((p_220514_1_) -> {
+         return p_23897_.equals(p_220514_1_.m_6095_().m_20674_());
       }, p_23898_);
    }
 
    public SetEntityLookTarget(EntityType<?> p_23894_, float p_23895_) {
-      this((p_23911_) -> {
-         return p_23894_.equals(p_23911_.m_6095_());
+      this((p_220518_1_) -> {
+         return p_23894_.equals(p_220518_1_.m_6095_());
       }, p_23895_);
    }
 
    public SetEntityLookTarget(float p_23892_) {
-      this((p_23913_) -> {
+      this((p_233953_0_) -> {
          return true;
       }, p_23892_);
    }
@@ -44,11 +_,11 @@
 
    protected void m_6735_(ServerLevel p_23906_, LivingEntity p_23907_, long p_23908_) {
       Brain<?> brain = p_23907_.m_6274_();
-      brain.m_21952_(MemoryModuleType.f_26365_).ifPresent((p_23920_) -> {
-         p_23920_.stream().filter(this.f_23889_).filter((p_23916_) -> {
-            return p_23916_.m_20280_(p_23907_) <= (double)this.f_23890_;
-         }).findFirst().ifPresent((p_23926_) -> {
-            brain.m_21879_(MemoryModuleType.f_26371_, new EntityTracker(p_23926_, true));
+      brain.m_21952_(MemoryModuleType.f_26365_).ifPresent((p_220515_3_) -> {
+         p_220515_3_.stream().filter(this.f_23889_).filter((p_220517_2_) -> {
+            return p_220517_2_.m_20280_(p_23907_) <= (double)this.f_23890_;
+         }).findFirst().ifPresent((p_220516_1_) -> {
+            brain.m_21879_(MemoryModuleType.f_26371_, new EntityTracker(p_220516_1_, true));
          });
       });
    }
