--- a/net/minecraft/world/entity/ai/behavior/SetLookAndInteract.java
+++ b/net/minecraft/world/entity/ai/behavior/SetLookAndInteract.java
@@ -25,9 +_,9 @@
    }
 
    public SetLookAndInteract(EntityType<?> p_23942_, int p_23943_) {
-      this(p_23942_, p_23943_, (p_23973_) -> {
+      this(p_23942_, p_23943_, (p_220528_0_) -> {
          return true;
-      }, (p_23971_) -> {
+      }, (p_220531_0_) -> {
          return true;
       });
    }
@@ -39,12 +_,12 @@
    public void m_6735_(ServerLevel p_23953_, LivingEntity p_23954_, long p_23955_) {
       super.m_6735_(p_23953_, p_23954_, p_23955_);
       Brain<?> brain = p_23954_.m_6274_();
-      brain.m_21952_(MemoryModuleType.f_26365_).ifPresent((p_23964_) -> {
-         p_23964_.stream().filter((p_23960_) -> {
-            return p_23960_.m_20280_(p_23954_) <= (double)this.f_23938_;
-         }).filter(this::m_23956_).findFirst().ifPresent((p_23967_) -> {
-            brain.m_21879_(MemoryModuleType.f_26374_, p_23967_);
-            brain.m_21879_(MemoryModuleType.f_26371_, new EntityTracker(p_23967_, true));
+      brain.m_21952_(MemoryModuleType.f_26365_).ifPresent((p_220526_3_) -> {
+         p_220526_3_.stream().filter((p_220529_2_) -> {
+            return p_220529_2_.m_20280_(p_23954_) <= (double)this.f_23938_;
+         }).filter(this::m_23956_).findFirst().ifPresent((p_220527_1_) -> {
+            brain.m_21879_(MemoryModuleType.f_26374_, p_220527_1_);
+            brain.m_21879_(MemoryModuleType.f_26371_, new EntityTracker(p_220527_1_, true));
          });
       });
    }
