--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
@@ -38,14 +_,14 @@
 
    protected void m_6735_(ServerLevel p_24059_, Villager p_24060_, long p_24061_) {
       Brain<?> brain = p_24060_.m_6274_();
-      brain.m_21952_(this.f_24040_).ifPresent((p_24067_) -> {
-         if (!this.m_24072_(p_24059_, p_24067_) && !this.m_24055_(p_24059_, p_24060_)) {
-            if (this.m_24078_(p_24060_, p_24067_)) {
+      brain.m_21952_(this.f_24040_).ifPresent((p_220545_6_) -> {
+         if (!this.m_24072_(p_24059_, p_220545_6_) && !this.m_24055_(p_24059_, p_24060_)) {
+            if (this.m_24078_(p_24060_, p_220545_6_)) {
                Vec3 vec3 = null;
                int i = 0;
 
                for(int j = 1000; i < 1000 && (vec3 == null || this.m_24078_(p_24060_, GlobalPos.m_122643_(p_24059_.m_46472_(), new BlockPos(vec3)))); ++i) {
-                  vec3 = RandomPos.m_26970_(p_24060_, 15, 7, Vec3.m_82539_(p_24067_.m_122646_()));
+                  vec3 = RandomPos.m_26970_(p_24060_, 15, 7, Vec3.m_82539_(p_220545_6_.m_122646_()));
                }
 
                if (i == 1000) {
@@ -54,8 +_,8 @@
                }
 
                brain.m_21879_(MemoryModuleType.f_26370_, new WalkTarget(vec3, this.f_24041_, this.f_24042_));
-            } else if (!this.m_24068_(p_24059_, p_24060_, p_24067_)) {
-               brain.m_21879_(MemoryModuleType.f_26370_, new WalkTarget(p_24067_.m_122646_(), this.f_24041_, this.f_24042_));
+            } else if (!this.m_24068_(p_24059_, p_24060_, p_220545_6_)) {
+               brain.m_21879_(MemoryModuleType.f_26370_, new WalkTarget(p_220545_6_.m_122646_(), this.f_24041_, this.f_24042_));
             }
          } else {
             this.m_24075_(p_24060_, p_24061_);
