--- a/net/minecraft/world/entity/ai/behavior/StopBeingAngryIfTargetDead.java
+++ b/net/minecraft/world/entity/ai/behavior/StopBeingAngryIfTargetDead.java
@@ -14,8 +_,8 @@
    }
 
    protected void m_6735_(ServerLevel p_24263_, E p_24264_, long p_24265_) {
-      BehaviorUtils.m_22610_(p_24264_, MemoryModuleType.f_26334_).ifPresent((p_24269_) -> {
-         if (p_24269_.m_21224_() && (p_24269_.m_6095_() != EntityType.f_20532_ || p_24263_.m_46469_().m_46207_(GameRules.f_46126_))) {
+      BehaviorUtils.m_22610_(p_24264_, MemoryModuleType.f_26334_).ifPresent((p_233988_2_) -> {
+         if (p_233988_2_.m_21224_() && (p_233988_2_.m_6095_() != EntityType.f_20532_ || p_24263_.m_46469_().m_46207_(GameRules.f_46126_))) {
             p_24264_.m_6274_().m_21936_(MemoryModuleType.f_26334_);
          }
 
