--- a/net/minecraft/world/entity/ai/behavior/StrollToPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollToPoi.java
@@ -34,8 +_,8 @@
       if (p_24350_ > this.f_24331_) {
          Brain<?> brain = p_24349_.m_6274_();
          Optional<GlobalPos> optional = brain.m_21952_(this.f_24327_);
-         optional.ifPresent((p_24353_) -> {
-            brain.m_21879_(MemoryModuleType.f_26370_, new WalkTarget(p_24353_.m_122646_(), this.f_24330_, this.f_24328_));
+         optional.ifPresent((p_220580_2_) -> {
+            brain.m_21879_(MemoryModuleType.f_26370_, new WalkTarget(p_220580_2_.m_122646_(), this.f_24330_, this.f_24328_));
          });
          this.f_24331_ = p_24350_ + 80L;
       }
