--- a/net/minecraft/world/entity/ai/behavior/StrollAroundPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollAroundPoi.java
@@ -32,8 +_,8 @@
    protected void m_6735_(ServerLevel p_24322_, PathfinderMob p_24323_, long p_24324_) {
       if (p_24324_ > this.f_24304_) {
          Optional<Vec3> optional = Optional.ofNullable(RandomPos.m_26959_(p_24323_, 8, 6));
-         p_24323_.m_6274_().m_21886_(MemoryModuleType.f_26370_, optional.map((p_24326_) -> {
-            return new WalkTarget(p_24326_, this.f_24306_, 1);
+         p_24323_.m_6274_().m_21886_(MemoryModuleType.f_26370_, optional.map((p_220564_1_) -> {
+            return new WalkTarget(p_220564_1_, this.f_24306_, 1);
          }));
          this.f_24304_ = p_24324_ + 180L;
       }
