--- a/net/minecraft/world/entity/ai/behavior/TradeWithVillager.java
+++ b/net/minecraft/world/entity/ai/behavior/TradeWithVillager.java
@@ -64,8 +_,8 @@
    private static Set<Item> m_24422_(Villager p_24423_, Villager p_24424_) {
       ImmutableSet<Item> immutableset = p_24424_.m_7141_().m_35571_().m_35623_();
       ImmutableSet<Item> immutableset1 = p_24423_.m_7141_().m_35571_().m_35623_();
-      return immutableset.stream().filter((p_24431_) -> {
-         return !immutableset1.contains(p_24431_);
+      return immutableset.stream().filter((p_220587_1_) -> {
+         return !immutableset1.contains(p_220587_1_);
       }).collect(Collectors.toSet());
    }
 
