--- a/net/minecraft/world/entity/ai/behavior/UseBonemeal.java
+++ b/net/minecraft/world/entity/ai/behavior/UseBonemeal.java
@@ -80,8 +_,8 @@
    }
 
    private void m_24480_(Villager p_24481_) {
-      this.f_24464_.ifPresent((p_24484_) -> {
-         BlockPosTracker blockpostracker = new BlockPosTracker(p_24484_);
+      this.f_24464_.ifPresent((p_233995_1_) -> {
+         BlockPosTracker blockpostracker = new BlockPosTracker(p_233995_1_);
          p_24481_.m_6274_().m_21879_(MemoryModuleType.f_26371_, blockpostracker);
          p_24481_.m_6274_().m_21879_(MemoryModuleType.f_26370_, new WalkTarget(blockpostracker, 0.5F, 1));
       });
