--- a/net/minecraft/world/entity/ai/behavior/VillagerMakeLove.java
+++ b/net/minecraft/world/entity/ai/behavior/VillagerMakeLove.java
@@ -78,8 +_,8 @@
 
    private boolean m_24639_(Villager p_24640_) {
       Brain<Villager> brain = p_24640_.m_6274_();
-      Optional<AgableMob> optional = brain.m_21952_(MemoryModuleType.f_26375_).filter((p_24638_) -> {
-         return p_24638_.m_6095_() == EntityType.f_20492_;
+      Optional<AgableMob> optional = brain.m_21952_(MemoryModuleType.f_26375_).filter((p_233999_0_) -> {
+         return p_233999_0_.m_6095_() == EntityType.f_20492_;
       });
       if (!optional.isPresent()) {
          return false;
@@ -89,8 +_,8 @@
    }
 
    private Optional<BlockPos> m_24648_(ServerLevel p_24649_, Villager p_24650_) {
-      return p_24649_.m_8904_().m_27133_(PoiType.f_27346_.m_27392_(), (p_24661_) -> {
-         return this.m_24641_(p_24650_, p_24661_);
+      return p_24649_.m_8904_().m_27133_(PoiType.f_27346_.m_27392_(), (p_220481_2_) -> {
+         return this.m_24641_(p_24650_, p_220481_2_);
       }, p_24650_.m_20183_(), 48);
    }
 
