--- a/net/minecraft/world/entity/ai/behavior/VillagerCalmDown.java
+++ b/net/minecraft/world/entity/ai/behavior/VillagerCalmDown.java
@@ -21,8 +_,8 @@
    }
 
    private static boolean m_24577_(Villager p_24578_) {
-      return p_24578_.m_6274_().m_21952_(MemoryModuleType.f_26382_).filter((p_24581_) -> {
-         return p_24581_.m_20280_(p_24578_) <= 36.0D;
+      return p_24578_.m_6274_().m_21952_(MemoryModuleType.f_26382_).filter((p_223523_1_) -> {
+         return p_223523_1_.m_20280_(p_24578_) <= 36.0D;
       }).isPresent();
    }
 }
