--- a/net/minecraft/world/entity/ai/behavior/WeightedList.java
+++ b/net/minecraft/world/entity/ai/behavior/WeightedList.java
@@ -24,13 +_,13 @@
    }
 
    public static <U> Codec<WeightedList<U>> m_24723_(Codec<U> p_24724_) {
-      return WeightedList.WeightedEntry.<U>m_24758_(p_24724_).listOf().xmap(WeightedList::new, (p_24722_) -> {
-         return p_24722_.f_24715_;
+      return WeightedList.WeightedEntry.<U>m_24758_(p_24724_).listOf().xmap(WeightedList::new, (p_234001_0_) -> {
+         return p_234001_0_.f_24715_;
       });
    }
 
    public WeightedList<U> m_24727_(U p_24728_, int p_24729_) {
-      this.f_24715_.add(new WeightedList.WeightedEntry<>(p_24728_, p_24729_));
+      this.f_24715_.add(new WeightedList.WeightedEntry(p_24728_, p_24729_));
       return this;
    }
 
@@ -39,11 +_,11 @@
    }
 
    public WeightedList<U> m_24730_(Random p_24731_) {
-      this.f_24715_.forEach((p_24734_) -> {
-         p_24734_.m_24751_(p_24731_.nextFloat());
+      this.f_24715_.forEach((p_234004_1_) -> {
+         p_234004_1_.m_24751_(p_24731_.nextFloat());
       });
-      this.f_24715_.sort(Comparator.comparingDouble((p_24726_) -> {
-         return p_24726_.m_24762_();
+      this.f_24715_.sort(Comparator.comparingDouble((p_234003_0_) -> {
+         return p_234003_0_.m_24762_();
       }));
       return this;
    }
@@ -94,10 +_,10 @@
          return new Codec<WeightedList.WeightedEntry<E>>() {
             public <T> DataResult<Pair<WeightedList.WeightedEntry<E>, T>> decode(DynamicOps<T> p_24780_, T p_24781_) {
                Dynamic<T> dynamic = new Dynamic<>(p_24780_, p_24781_);
-               return dynamic.get("data").flatMap(p_24759_::parse).map((p_24775_) -> {
-                  return new WeightedList.WeightedEntry(p_24775_, dynamic.get("weight").asInt(1));
-               }).map((p_24778_) -> {
-                  return Pair.of(p_24778_, p_24780_.empty());
+               return dynamic.get("data").flatMap(p_24759_::parse).map((p_234012_1_) -> {
+                  return new WeightedList.WeightedEntry(p_234012_1_, dynamic.get("weight").asInt(1));
+               }).map((p_234013_1_) -> {
+                  return Pair.of(p_234013_1_, p_24780_.empty());
                });
             }
 
