--- a/net/minecraft/world/entity/ai/behavior/WorkAtPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/WorkAtPoi.java
@@ -31,8 +_,8 @@
    protected void m_6735_(ServerLevel p_24816_, Villager p_24817_, long p_24818_) {
       Brain<Villager> brain = p_24817_.m_6274_();
       brain.m_21879_(MemoryModuleType.f_26330_, p_24818_);
-      brain.m_21952_(MemoryModuleType.f_26360_).ifPresent((p_24821_) -> {
-         brain.m_21879_(MemoryModuleType.f_26371_, new BlockPosTracker(p_24821_.m_122646_()));
+      brain.m_21952_(MemoryModuleType.f_26360_).ifPresent((p_225460_1_) -> {
+         brain.m_21879_(MemoryModuleType.f_26371_, new BlockPosTracker(p_225460_1_.m_122646_()));
       });
       p_24817_.m_35512_();
       this.m_5628_(p_24816_, p_24817_);
