--- a/net/minecraft/world/entity/ai/behavior/YieldJobSite.java
+++ b/net/minecraft/world/entity/ai/behavior/YieldJobSite.java
@@ -33,10 +_,10 @@
       BlockPos blockpos = p_24848_.m_6274_().m_21952_(MemoryModuleType.f_26361_).get().m_122646_();
       Optional<PoiType> optional = p_24847_.m_8904_().m_27177_(blockpos);
       if (optional.isPresent()) {
-         BehaviorUtils.m_22650_(p_24848_, (p_24874_) -> {
-            return this.m_24861_(optional.get(), p_24874_, blockpos);
-         }).findFirst().ifPresent((p_24860_) -> {
-            this.m_24850_(p_24847_, p_24848_, p_24860_, blockpos, p_24860_.m_6274_().m_21952_(MemoryModuleType.f_26360_).isPresent());
+         BehaviorUtils.m_22650_(p_24848_, (p_234021_3_) -> {
+            return this.m_24861_(optional.get(), p_234021_3_, blockpos);
+         }).findFirst().ifPresent((p_234023_4_) -> {
+            this.m_24850_(p_24847_, p_24848_, p_234023_4_, blockpos, p_234023_4_.m_6274_().m_21952_(MemoryModuleType.f_26360_).isPresent());
          });
       }
    }
