--- a/net/minecraft/client/renderer/chunk/VisGraph.java
+++ b/net/minecraft/client/renderer/chunk/VisGraph.java
@@ -18,7 +_,7 @@
    private static final int f_112951_ = (int)Math.pow(16.0D, 2.0D);
    private static final Direction[] f_112952_ = Direction.values();
    private final BitSet f_112953_ = new BitSet(4096);
-   private static final int[] f_112954_ = Util.m_137469_(new int[1352], (p_112974_) -> {
+   private static final int[] f_112954_ = Util.m_137469_(new int[1352], (p_209264_0_) -> {
       int i = 0;
       int j = 15;
       int k = 0;
@@ -27,7 +_,7 @@
          for(int i1 = 0; i1 < 16; ++i1) {
             for(int j1 = 0; j1 < 16; ++j1) {
                if (l == 0 || l == 15 || i1 == 0 || i1 == 15 || j1 == 0 || j1 == 15) {
-                  p_112974_[k++] = m_112961_(l, i1, j1);
+                  p_209264_0_[k++] = m_112961_(l, i1, j1);
                }
             }
          }
