--- a/net/minecraft/client/resources/PackResourcesAdapterV4.java
+++ b/net/minecraft/client/resources/PackResourcesAdapterV4.java
@@ -25,24 +_,24 @@
 
 @OnlyIn(Dist.CLIENT)
 public class PackResourcesAdapterV4 implements PackResources {
-   private static final Map<String, Pair<ChestType, ResourceLocation>> f_118737_ = Util.m_137469_(Maps.newHashMap(), (p_118784_) -> {
-      p_118784_.put("textures/entity/chest/normal_left.png", new Pair<>(ChestType.LEFT, new ResourceLocation("textures/entity/chest/normal_double.png")));
-      p_118784_.put("textures/entity/chest/normal_right.png", new Pair<>(ChestType.RIGHT, new ResourceLocation("textures/entity/chest/normal_double.png")));
-      p_118784_.put("textures/entity/chest/normal.png", new Pair<>(ChestType.SINGLE, new ResourceLocation("textures/entity/chest/normal.png")));
-      p_118784_.put("textures/entity/chest/trapped_left.png", new Pair<>(ChestType.LEFT, new ResourceLocation("textures/entity/chest/trapped_double.png")));
-      p_118784_.put("textures/entity/chest/trapped_right.png", new Pair<>(ChestType.RIGHT, new ResourceLocation("textures/entity/chest/trapped_double.png")));
-      p_118784_.put("textures/entity/chest/trapped.png", new Pair<>(ChestType.SINGLE, new ResourceLocation("textures/entity/chest/trapped.png")));
-      p_118784_.put("textures/entity/chest/christmas_left.png", new Pair<>(ChestType.LEFT, new ResourceLocation("textures/entity/chest/christmas_double.png")));
-      p_118784_.put("textures/entity/chest/christmas_right.png", new Pair<>(ChestType.RIGHT, new ResourceLocation("textures/entity/chest/christmas_double.png")));
-      p_118784_.put("textures/entity/chest/christmas.png", new Pair<>(ChestType.SINGLE, new ResourceLocation("textures/entity/chest/christmas.png")));
-      p_118784_.put("textures/entity/chest/ender.png", new Pair<>(ChestType.SINGLE, new ResourceLocation("textures/entity/chest/ender.png")));
+   private static final Map<String, Pair<ChestType, ResourceLocation>> f_118737_ = Util.m_137469_(Maps.newHashMap(), (p_229288_0_) -> {
+      p_229288_0_.put("textures/entity/chest/normal_left.png", new Pair<>(ChestType.LEFT, new ResourceLocation("textures/entity/chest/normal_double.png")));
+      p_229288_0_.put("textures/entity/chest/normal_right.png", new Pair<>(ChestType.RIGHT, new ResourceLocation("textures/entity/chest/normal_double.png")));
+      p_229288_0_.put("textures/entity/chest/normal.png", new Pair<>(ChestType.SINGLE, new ResourceLocation("textures/entity/chest/normal.png")));
+      p_229288_0_.put("textures/entity/chest/trapped_left.png", new Pair<>(ChestType.LEFT, new ResourceLocation("textures/entity/chest/trapped_double.png")));
+      p_229288_0_.put("textures/entity/chest/trapped_right.png", new Pair<>(ChestType.RIGHT, new ResourceLocation("textures/entity/chest/trapped_double.png")));
+      p_229288_0_.put("textures/entity/chest/trapped.png", new Pair<>(ChestType.SINGLE, new ResourceLocation("textures/entity/chest/trapped.png")));
+      p_229288_0_.put("textures/entity/chest/christmas_left.png", new Pair<>(ChestType.LEFT, new ResourceLocation("textures/entity/chest/christmas_double.png")));
+      p_229288_0_.put("textures/entity/chest/christmas_right.png", new Pair<>(ChestType.RIGHT, new ResourceLocation("textures/entity/chest/christmas_double.png")));
+      p_229288_0_.put("textures/entity/chest/christmas.png", new Pair<>(ChestType.SINGLE, new ResourceLocation("textures/entity/chest/christmas.png")));
+      p_229288_0_.put("textures/entity/chest/ender.png", new Pair<>(ChestType.SINGLE, new ResourceLocation("textures/entity/chest/ender.png")));
    });
    private static final List<String> f_118738_ = Lists.newArrayList("base", "border", "bricks", "circle", "creeper", "cross", "curly_border", "diagonal_left", "diagonal_right", "diagonal_up_left", "diagonal_up_right", "flower", "globe", "gradient", "gradient_up", "half_horizontal", "half_horizontal_bottom", "half_vertical", "half_vertical_right", "mojang", "rhombus", "skull", "small_stripes", "square_bottom_left", "square_bottom_right", "square_top_left", "square_top_right", "straight_cross", "stripe_bottom", "stripe_center", "stripe_downleft", "stripe_downright", "stripe_left", "stripe_middle", "stripe_right", "stripe_top", "triangle_bottom", "triangle_top", "triangles_bottom", "triangles_top");
-   private static final Set<String> f_118739_ = f_118738_.stream().map((p_118795_) -> {
-      return "textures/entity/shield/" + p_118795_ + ".png";
+   private static final Set<String> f_118739_ = f_118738_.stream().map((p_229291_0_) -> {
+      return "textures/entity/shield/" + p_229291_0_ + ".png";
    }).collect(Collectors.toSet());
-   private static final Set<String> f_118740_ = f_118738_.stream().map((p_118782_) -> {
-      return "textures/entity/banner/" + p_118782_ + ".png";
+   private static final Set<String> f_118740_ = f_118738_.stream().map((p_229287_0_) -> {
+      return "textures/entity/banner/" + p_229287_0_ + ".png";
    }).collect(Collectors.toSet());
    public static final ResourceLocation f_118734_ = new ResourceLocation("textures/entity/shield_base.png");
    public static final ResourceLocation f_118735_ = new ResourceLocation("textures/entity/banner_base.png");
