--- a/net/minecraft/client/gui/screens/inventory/AbstractFurnaceScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/AbstractFurnaceScreen.java
@@ -33,11 +_,11 @@
       this.f_97821_ = this.f_96543_ < 379;
       this.f_97819_.m_100309_(this.f_96543_, this.f_96544_, this.f_96541_, this.f_97821_, this.f_97732_);
       this.f_97735_ = this.f_97819_.m_100347_(this.f_97821_, this.f_96543_, this.f_97726_);
-      this.m_7082_(new ImageButton(this.f_97735_ + 20, this.f_96544_ / 2 - 49, 20, 18, 0, 0, 19, f_97820_, (p_97863_) -> {
+      this.m_7082_(new ImageButton(this.f_97735_ + 20, this.f_96544_ / 2 - 49, 20, 18, 0, 0, 19, f_97820_, (p_214087_1_) -> {
          this.f_97819_.m_100345_(this.f_97821_);
          this.f_97819_.m_100384_();
          this.f_97735_ = this.f_97819_.m_100347_(this.f_97821_, this.f_96543_, this.f_97726_);
-         ((ImageButton)p_97863_).m_94278_(this.f_97735_ + 20, this.f_96544_ / 2 - 49);
+         ((ImageButton)p_214087_1_).m_94278_(this.f_97735_ + 20, this.f_96544_ / 2 - 49);
       }));
       this.f_97728_ = (this.f_97726_ - this.f_96547_.m_92852_(this.f_96539_)) / 2;
    }
