--- a/net/minecraft/client/gui/screens/inventory/BeaconScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/BeaconScreen.java
@@ -151,6 +_,42 @@
    }
 
    @OnlyIn(Dist.CLIENT)
+   abstract static class BeaconScreenButton extends AbstractButton {
+      private boolean f_98020_;
+
+      protected BeaconScreenButton(int p_98022_, int p_98023_) {
+         super(p_98022_, p_98023_, 22, 22, TextComponent.f_131282_);
+      }
+
+      public void m_6303_(PoseStack p_98027_, int p_98028_, int p_98029_, float p_98030_) {
+         Minecraft.m_91087_().m_91097_().m_118493_(BeaconScreen.f_97903_);
+         RenderSystem.m_69439_(1.0F, 1.0F, 1.0F, 1.0F);
+         int i = 219;
+         int j = 0;
+         if (!this.f_93623_) {
+            j += this.f_93618_ * 2;
+         } else if (this.f_98020_) {
+            j += this.f_93618_ * 1;
+         } else if (this.m_5702_()) {
+            j += this.f_93618_ * 3;
+         }
+
+         this.m_93228_(p_98027_, this.f_93620_, this.f_93621_, j, 219, this.f_93618_, this.f_93619_);
+         this.m_6805_(p_98027_);
+      }
+
+      protected abstract void m_6805_(PoseStack p_98025_);
+
+      public boolean m_98024_() {
+         return this.f_98020_;
+      }
+
+      public void m_98031_(boolean p_98032_) {
+         this.f_98020_ = p_98032_;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
    class BeaconCancelButton extends BeaconScreen.BeaconSpriteScreenButton {
       public BeaconCancelButton(int p_97982_, int p_97983_) {
          super(p_97982_, p_97983_, 112, 220);
@@ -229,42 +_,6 @@
       protected void m_6805_(PoseStack p_98014_) {
          Minecraft.m_91087_().m_91097_().m_118493_(this.f_98001_.m_118414_().m_118330_());
          m_93200_(p_98014_, this.f_93620_ + 2, this.f_93621_ + 2, this.m_93252_(), 18, 18, this.f_98001_);
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   abstract static class BeaconScreenButton extends AbstractButton {
-      private boolean f_98020_;
-
-      protected BeaconScreenButton(int p_98022_, int p_98023_) {
-         super(p_98022_, p_98023_, 22, 22, TextComponent.f_131282_);
-      }
-
-      public void m_6303_(PoseStack p_98027_, int p_98028_, int p_98029_, float p_98030_) {
-         Minecraft.m_91087_().m_91097_().m_118493_(BeaconScreen.f_97903_);
-         RenderSystem.m_69439_(1.0F, 1.0F, 1.0F, 1.0F);
-         int i = 219;
-         int j = 0;
-         if (!this.f_93623_) {
-            j += this.f_93618_ * 2;
-         } else if (this.f_98020_) {
-            j += this.f_93618_ * 1;
-         } else if (this.m_5702_()) {
-            j += this.f_93618_ * 3;
-         }
-
-         this.m_93228_(p_98027_, this.f_93620_, this.f_93621_, j, 219, this.f_93618_, this.f_93619_);
-         this.m_6805_(p_98027_);
-      }
-
-      protected abstract void m_6805_(PoseStack p_98025_);
-
-      public boolean m_98024_() {
-         return this.f_98020_;
-      }
-
-      public void m_98031_(boolean p_98032_) {
-         this.f_98020_ = p_98032_;
       }
    }
 
