--- a/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
@@ -94,7 +_,7 @@
    }
 
    protected void m_7829_() {
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, 196, 200, 20, CommonComponents.f_130655_, (p_98299_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, 196, 200, 20, CommonComponents.f_130655_, (p_214161_1_) -> {
          this.f_96541_.m_91152_((Screen)null);
       }));
    }
@@ -102,10 +_,10 @@
    protected void m_98301_() {
       int i = (this.f_96543_ - 192) / 2;
       int j = 2;
-      this.f_98258_ = this.m_7082_(new PageButton(i + 116, 159, true, (p_98297_) -> {
+      this.f_98258_ = this.m_7082_(new PageButton(i + 116, 159, true, (p_214159_1_) -> {
          this.m_7815_();
       }, this.f_98260_));
-      this.f_98259_ = this.m_7082_(new PageButton(i + 43, 159, false, (p_98287_) -> {
+      this.f_98259_ = this.m_7082_(new PageButton(i + 43, 159, false, (p_214158_1_) -> {
          this.m_7811_();
       }, this.f_98260_));
       this.m_98302_();
