--- a/net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
@@ -34,15 +_,15 @@
 
    protected void m_7856_() {
       super.m_7856_();
-      this.f_98375_ = this.m_7082_(new Button(this.f_96543_ / 2 - 50 - 100 - 4, 165, 100, 20, new TranslatableComponent("advMode.mode.sequence"), (p_98395_) -> {
+      this.f_98375_ = this.m_7082_(new Button(this.f_96543_ / 2 - 50 - 100 - 4, 165, 100, 20, new TranslatableComponent("advMode.mode.sequence"), (p_214191_1_) -> {
          this.m_98400_();
          this.m_98399_();
       }));
-      this.f_98376_ = this.m_7082_(new Button(this.f_96543_ / 2 - 50, 165, 100, 20, new TranslatableComponent("advMode.mode.unconditional"), (p_98393_) -> {
+      this.f_98376_ = this.m_7082_(new Button(this.f_96543_ / 2 - 50, 165, 100, 20, new TranslatableComponent("advMode.mode.unconditional"), (p_214190_1_) -> {
          this.f_98379_ = !this.f_98379_;
          this.m_98401_();
       }));
-      this.f_98377_ = this.m_7082_(new Button(this.f_96543_ / 2 + 50 + 4, 165, 100, 20, new TranslatableComponent("advMode.mode.redstoneTriggered"), (p_98390_) -> {
+      this.f_98377_ = this.m_7082_(new Button(this.f_96543_ / 2 + 50 + 4, 165, 100, 20, new TranslatableComponent("advMode.mode.redstoneTriggered"), (p_214189_1_) -> {
          this.f_98380_ = !this.f_98380_;
          this.m_98402_();
       }));
