--- a/net/minecraft/client/gui/screens/inventory/CraftingScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/CraftingScreen.java
@@ -32,11 +_,11 @@
       this.f_97735_ = this.f_98444_.m_100347_(this.f_98445_, this.f_96543_, this.f_97726_);
       this.f_96540_.add(this.f_98444_);
       this.m_94718_(this.f_98444_);
-      this.m_7082_(new ImageButton(this.f_97735_ + 5, this.f_96544_ / 2 - 49, 20, 18, 0, 0, 19, f_98443_, (p_98484_) -> {
+      this.m_7082_(new ImageButton(this.f_97735_ + 5, this.f_96544_ / 2 - 49, 20, 18, 0, 0, 19, f_98443_, (p_214076_1_) -> {
          this.f_98444_.m_100345_(this.f_98445_);
          this.f_98444_.m_100384_();
          this.f_97735_ = this.f_98444_.m_100347_(this.f_98445_, this.f_96543_, this.f_97726_);
-         ((ImageButton)p_98484_).m_94278_(this.f_97735_ + 5, this.f_96544_ / 2 - 49);
+         ((ImageButton)p_214076_1_).m_94278_(this.f_97735_ + 5, this.f_96544_ / 2 - 49);
       }));
       this.f_97728_ = 29;
    }
