--- a/net/minecraft/client/gui/screens/inventory/SignEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/SignEditScreen.java
@@ -41,25 +_,25 @@
 
    public SignEditScreen(SignBlockEntity p_99260_) {
       super(new TranslatableComponent("sign.edit"));
-      this.f_99258_ = IntStream.range(0, 4).mapToObj(p_99260_::m_59727_).map(Component::getString).toArray((p_99265_) -> {
-         return new String[p_99265_];
+      this.f_99258_ = IntStream.range(0, 4).mapToObj(p_99260_::m_59727_).map(Component::getString).toArray((p_243354_0_) -> {
+         return new String[p_243354_0_];
       });
       this.f_99254_ = p_99260_;
    }
 
    protected void m_7856_() {
       this.f_96541_.f_91068_.m_90926_(true);
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 120, 200, 20, CommonComponents.f_130655_, (p_99276_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ / 4 + 120, 200, 20, CommonComponents.f_130655_, (p_238847_1_) -> {
          this.m_99285_();
       }));
       this.f_99254_.m_59746_(false);
       this.f_99257_ = new TextFieldHelper(() -> {
          return this.f_99258_[this.f_99256_];
-      }, (p_99282_) -> {
-         this.f_99258_[this.f_99256_] = p_99282_;
-         this.f_99254_.m_59732_(this.f_99256_, new TextComponent(p_99282_));
-      }, TextFieldHelper.m_95153_(this.f_96541_), TextFieldHelper.m_95182_(this.f_96541_), (p_99280_) -> {
-         return this.f_96541_.f_91062_.m_92895_(p_99280_) <= 90;
+      }, (p_238850_1_) -> {
+         this.f_99258_[this.f_99256_] = p_238850_1_;
+         this.f_99254_.m_59732_(this.f_99256_, new TextComponent(p_238850_1_));
+      }, TextFieldHelper.m_95153_(this.f_96541_), TextFieldHelper.m_95182_(this.f_96541_), (p_238848_1_) -> {
+         return this.f_96541_.f_91062_.m_92895_(p_238848_1_) <= 90;
       });
    }
 
