--- a/net/minecraft/client/gui/screens/recipebook/OverlayRecipeComponent.java
+++ b/net/minecraft/client/gui/screens/recipebook/OverlayRecipeComponent.java
@@ -187,6 +_,18 @@
    }
 
    @OnlyIn(Dist.CLIENT)
+   class OverlaySmeltingRecipeButton extends OverlayRecipeComponent.OverlayRecipeButton {
+      public OverlaySmeltingRecipeButton(int p_100262_, int p_100263_, Recipe<?> p_100264_, boolean p_100265_) {
+         super(p_100262_, p_100263_, p_100264_, p_100265_);
+      }
+
+      protected void m_6222_(Recipe<?> p_100267_) {
+         ItemStack[] aitemstack = p_100267_.m_7527_().get(0).m_43908_();
+         this.f_100226_.add(new OverlayRecipeComponent.OverlayRecipeButton.Pos(10, 10, aitemstack));
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
    class OverlayRecipeButton extends AbstractWidget implements PlaceRecipe<Ingredient> {
       private final Recipe<?> f_100228_;
       private final boolean f_100229_;
@@ -252,18 +_,6 @@
             this.f_100252_ = p_100257_;
             this.f_100250_ = p_100258_;
          }
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   class OverlaySmeltingRecipeButton extends OverlayRecipeComponent.OverlayRecipeButton {
-      public OverlaySmeltingRecipeButton(int p_100262_, int p_100263_, Recipe<?> p_100264_, boolean p_100265_) {
-         super(p_100262_, p_100263_, p_100264_, p_100265_);
-      }
-
-      protected void m_6222_(Recipe<?> p_100267_) {
-         ItemStack[] aitemstack = p_100267_.m_7527_().get(0).m_43908_();
-         this.f_100226_.add(new OverlayRecipeComponent.OverlayRecipeButton.Pos(10, 10, aitemstack));
       }
    }
 }
