--- a/net/minecraft/util/datafix/fixes/AttributesRename.java
+++ b/net/minecraft/util/datafix/fixes/AttributesRename.java
@@ -22,37 +_,37 @@
    protected TypeRewriteRule makeRule() {
       Type<?> type = this.getInputSchema().getType(References.f_16782_);
       OpticFinder<?> opticfinder = type.findField("tag");
-      return TypeRewriteRule.seq(this.fixTypeEverywhereTyped("Rename ItemStack Attributes", type, (p_14674_) -> {
-         return p_14674_.updateTyped(opticfinder, AttributesRename::m_14675_);
+      return TypeRewriteRule.seq(this.fixTypeEverywhereTyped("Rename ItemStack Attributes", type, (p_233071_1_) -> {
+         return p_233071_1_.updateTyped(opticfinder, AttributesRename::m_14675_);
       }), this.fixTypeEverywhereTyped("Rename Entity Attributes", this.getInputSchema().getType(References.f_16786_), AttributesRename::m_14683_), this.fixTypeEverywhereTyped("Rename Player Attributes", this.getInputSchema().getType(References.f_16772_), AttributesRename::m_14683_));
    }
 
    private static Dynamic<?> m_14677_(Dynamic<?> p_14678_) {
-      return DataFixUtils.orElse(p_14678_.asString().result().map((p_14680_) -> {
-         return f_14668_.getOrDefault(p_14680_, p_14680_);
+      return DataFixUtils.orElse(p_14678_.asString().result().map((p_233074_0_) -> {
+         return f_14668_.getOrDefault(p_233074_0_, p_233074_0_);
       }).map(p_14678_::createString), p_14678_);
    }
 
    private static Typed<?> m_14675_(Typed<?> p_14676_) {
-      return p_14676_.update(DSL.remainderFinder(), (p_14694_) -> {
-         return p_14694_.update("AttributeModifiers", (p_14696_) -> {
-            return DataFixUtils.orElse(p_14696_.asStreamOpt().result().map((p_14688_) -> {
-               return p_14688_.map((p_14698_) -> {
-                  return p_14698_.update("AttributeName", AttributesRename::m_14677_);
+      return p_14676_.update(DSL.remainderFinder(), (p_233081_0_) -> {
+         return p_233081_0_.update("AttributeModifiers", (p_233082_0_) -> {
+            return DataFixUtils.orElse(p_233082_0_.asStreamOpt().result().map((p_233078_0_) -> {
+               return p_233078_0_.map((p_233083_0_) -> {
+                  return p_233083_0_.update("AttributeName", AttributesRename::m_14677_);
                });
-            }).map(p_14696_::createList), p_14696_);
+            }).map(p_233082_0_::createList), p_233082_0_);
          });
       });
    }
 
    private static Typed<?> m_14683_(Typed<?> p_14684_) {
-      return p_14684_.update(DSL.remainderFinder(), (p_14686_) -> {
-         return p_14686_.update("Attributes", (p_14690_) -> {
-            return DataFixUtils.orElse(p_14690_.asStreamOpt().result().map((p_14682_) -> {
-               return p_14682_.map((p_14692_) -> {
-                  return p_14692_.update("Name", AttributesRename::m_14677_);
+      return p_14684_.update(DSL.remainderFinder(), (p_233077_0_) -> {
+         return p_233077_0_.update("Attributes", (p_233079_0_) -> {
+            return DataFixUtils.orElse(p_233079_0_.asStreamOpt().result().map((p_233075_0_) -> {
+               return p_233075_0_.map((p_233080_0_) -> {
+                  return p_233080_0_.update("Name", AttributesRename::m_14677_);
                });
-            }).map(p_14690_::createList), p_14690_);
+            }).map(p_233079_0_::createList), p_233079_0_);
          });
       });
    }
