--- a/net/minecraft/util/datafix/fixes/BedBlockEntityInjecter.java
+++ b/net/minecraft/util/datafix/fixes/BedBlockEntityInjecter.java
@@ -38,12 +_,12 @@
       OpticFinder<?> opticfinder = DSL.fieldFinder("Level", p_14710_);
       OpticFinder<List<TE>> opticfinder1 = DSL.fieldFinder("TileEntities", p_14711_);
       int i = 416;
-      return TypeRewriteRule.seq(this.fixTypeEverywhere("InjectBedBlockEntityType", this.getInputSchema().findChoiceType(References.f_16781_), this.getOutputSchema().findChoiceType(References.f_16781_), (p_14717_) -> {
-         return (p_14713_) -> {
-            return (com.mojang.datafixers.util.Pair)p_14713_;
+      return TypeRewriteRule.seq(this.fixTypeEverywhere("InjectBedBlockEntityType", this.getInputSchema().findChoiceType(References.f_16781_), this.getOutputSchema().findChoiceType(References.f_16781_), (p_233085_0_) -> {
+         return (p_209696_0_) -> {
+            return (com.mojang.datafixers.util.Pair)p_209696_0_;
          };
-      }), this.fixTypeEverywhereTyped("BedBlockEntityInjecter", this.getOutputSchema().getType(References.f_16773_), (p_14708_) -> {
-         Typed<?> typed = p_14708_.getTyped(opticfinder);
+      }), this.fixTypeEverywhereTyped("BedBlockEntityInjecter", this.getOutputSchema().getType(References.f_16773_), (p_207434_3_) -> {
+         Typed<?> typed = p_207434_3_.getTyped(opticfinder);
          Dynamic<?> dynamic = typed.get(DSL.remainderFinder());
          int j = dynamic.get("xPos").asInt(0);
          int k = dynamic.get("zPos").asInt(0);
@@ -53,8 +_,8 @@
          for(int l = 0; l < list1.size(); ++l) {
             Dynamic<?> dynamic1 = list1.get(l);
             int i1 = dynamic1.get("Y").asInt(0);
-            Stream<Integer> stream = dynamic1.get("Blocks").asStream().map((p_14715_) -> {
-               return p_14715_.asInt(0);
+            Stream<Integer> stream = dynamic1.get("Blocks").asStream().map((p_233084_0_) -> {
+               return p_233084_0_.asInt(0);
             });
             int j1 = 0;
 
@@ -78,7 +_,7 @@
             }
          }
 
-         return !list.isEmpty() ? p_14708_.set(opticfinder, typed.set(opticfinder1, list)) : p_14708_;
+         return !list.isEmpty() ? p_207434_3_.set(opticfinder, typed.set(opticfinder1, list)) : p_207434_3_;
       }));
    }
 }
