--- a/net/minecraft/util/datafix/fixes/BitStorageAlignFix.java
+++ b/net/minecraft/util/datafix/fixes/BitStorageAlignFix.java
@@ -29,19 +_,19 @@
       OpticFinder<?> opticfinder2 = DSL.typeFinder(type2);
       Type<Pair<String, Dynamic<?>>> type3 = DSL.named(References.f_16783_.typeName(), DSL.remainderType());
       OpticFinder<List<Pair<String, Dynamic<?>>>> opticfinder3 = DSL.fieldFinder("Palette", DSL.list(type3));
-      return this.fixTypeEverywhereTyped("BitStorageAlignFix", type, this.getOutputSchema().getType(References.f_16773_), (p_14749_) -> {
-         return p_14749_.updateTyped(opticfinder, (p_14787_) -> {
-            return this.m_14762_(m_14750_(opticfinder1, opticfinder2, opticfinder3, p_14787_));
+      return this.fixTypeEverywhereTyped("BitStorageAlignFix", type, this.getOutputSchema().getType(References.f_16773_), (p_233088_5_) -> {
+         return p_233088_5_.updateTyped(opticfinder, (p_233099_4_) -> {
+            return this.m_14762_(m_14750_(opticfinder1, opticfinder2, opticfinder3, p_233099_4_));
          });
       });
    }
 
    private Typed<?> m_14762_(Typed<?> p_14763_) {
-      return p_14763_.update(DSL.remainderFinder(), (p_14765_) -> {
-         return p_14765_.update("Heightmaps", (p_14775_) -> {
-            return p_14775_.updateMapValues((p_14772_) -> {
-               return p_14772_.mapSecond((p_14790_) -> {
-                  return m_14776_(p_14765_, p_14790_, 256, 9);
+      return p_14763_.update(DSL.remainderFinder(), (p_233093_0_) -> {
+         return p_233093_0_.update("Heightmaps", (p_233096_1_) -> {
+            return p_233096_1_.updateMapValues((p_233095_1_) -> {
+               return p_233095_1_.mapSecond((p_233100_1_) -> {
+                  return m_14776_(p_233093_0_, p_233100_1_, 256, 9);
                });
             });
          });
@@ -49,16 +_,16 @@
    }
 
    private static Typed<?> m_14750_(OpticFinder<?> p_14751_, OpticFinder<?> p_14752_, OpticFinder<List<Pair<String, Dynamic<?>>>> p_14753_, Typed<?> p_14754_) {
-      return p_14754_.updateTyped(p_14751_, (p_14758_) -> {
-         return p_14758_.updateTyped(p_14752_, (p_14761_) -> {
-            int i = p_14761_.getOptional(p_14753_).map((p_14782_) -> {
-               return Math.max(4, DataFixUtils.ceillog2(p_14782_.size()));
+      return p_14754_.updateTyped(p_14751_, (p_233090_2_) -> {
+         return p_233090_2_.updateTyped(p_14752_, (p_233091_1_) -> {
+            int i = p_233091_1_.getOptional(p_14753_).map((p_233098_0_) -> {
+               return Math.max(4, DataFixUtils.ceillog2(p_233098_0_.size()));
             }).orElse(0);
-            return i != 0 && !Mth.m_14152_(i) ? p_14761_.update(DSL.remainderFinder(), (p_14743_) -> {
-               return p_14743_.update("BlockStates", (p_14769_) -> {
-                  return m_14776_(p_14743_, p_14769_, 4096, i);
+            return i != 0 && !Mth.m_14152_(i) ? p_233091_1_.update(DSL.remainderFinder(), (p_233087_1_) -> {
+               return p_233087_1_.update("BlockStates", (p_233094_2_) -> {
+                  return m_14776_(p_233087_1_, p_233094_2_, 4096, i);
                });
-            }) : p_14761_;
+            }) : p_233091_1_;
          });
       });
    }
