--- a/net/minecraft/util/datafix/fixes/BlockNameFlatteningFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockNameFlatteningFix.java
@@ -21,11 +_,11 @@
       Type<Pair<String, Either<Integer, String>>> type2 = DSL.named(References.f_16787_.typeName(), DSL.or(DSL.intType(), NamespacedSchema.m_17310_()));
       Type<Pair<String, String>> type3 = DSL.named(References.f_16787_.typeName(), NamespacedSchema.m_17310_());
       if (Objects.equals(type, type2) && Objects.equals(type1, type3)) {
-         return this.fixTypeEverywhere("BlockNameFlatteningFix", type2, type3, (p_14904_) -> {
-            return (p_14902_) -> {
-               return p_14902_.mapSecond((p_14900_) -> {
-                  return p_14900_.map(BlockStateData::m_14940_, (p_14906_) -> {
-                     return BlockStateData.m_14950_(NamespacedSchema.m_17311_(p_14906_));
+         return this.fixTypeEverywhere("BlockNameFlatteningFix", type2, type3, (p_209702_0_) -> {
+            return (p_206303_0_) -> {
+               return p_206303_0_.mapSecond((p_206304_0_) -> {
+                  return p_206304_0_.map(BlockStateData::m_14940_, (p_206305_0_) -> {
+                     return BlockStateData.m_14950_(NamespacedSchema.m_17311_(p_206305_0_));
                   });
                });
             };
