--- a/net/minecraft/util/datafix/fixes/BlockEntityCustomNameToComponentFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntityCustomNameToComponentFix.java
@@ -16,10 +_,10 @@
 
    public TypeRewriteRule makeRule() {
       OpticFinder<String> opticfinder = DSL.fieldFinder("id", NamespacedSchema.m_17310_());
-      return this.fixTypeEverywhereTyped("BlockEntityCustomNameToComponentFix", this.getInputSchema().getType(References.f_16781_), (p_14821_) -> {
-         return p_14821_.update(DSL.remainderFinder(), (p_14825_) -> {
-            Optional<String> optional = p_14821_.getOptional(opticfinder);
-            return optional.isPresent() && Objects.equals(optional.get(), "minecraft:command_block") ? p_14825_ : EntityCustomNameToComponentFix.m_15407_(p_14825_);
+      return this.fixTypeEverywhereTyped("BlockEntityCustomNameToComponentFix", this.getInputSchema().getType(References.f_16781_), (p_207790_1_) -> {
+         return p_207790_1_.update(DSL.remainderFinder(), (p_207789_2_) -> {
+            Optional<String> optional = p_207790_1_.getOptional(opticfinder);
+            return optional.isPresent() && Objects.equals(optional.get(), "minecraft:command_block") ? p_207789_2_ : EntityCustomNameToComponentFix.m_15407_(p_207789_2_);
          });
       });
    }
