--- a/net/minecraft/util/datafix/fixes/BlockStateData.java
+++ b/net/minecraft/util/datafix/fixes/BlockStateData.java
@@ -13,11 +_,11 @@
    private static final Logger f_14933_ = LogManager.getLogger();
    private static final Dynamic<?>[] f_14934_ = new Dynamic[4096];
    private static final Dynamic<?>[] f_14935_ = new Dynamic[256];
-   private static final Object2IntMap<Dynamic<?>> f_14936_ = DataFixUtils.make(new Object2IntOpenHashMap<>(), (p_14955_) -> {
-      p_14955_.defaultReturnValue(-1);
+   private static final Object2IntMap<Dynamic<?>> f_14936_ = DataFixUtils.make(new Object2IntOpenHashMap<>(), (p_209420_0_) -> {
+      p_209420_0_.defaultReturnValue(-1);
    });
-   private static final Object2IntMap<String> f_14937_ = DataFixUtils.make(new Object2IntOpenHashMap<>(), (p_14949_) -> {
-      p_14949_.defaultReturnValue(-1);
+   private static final Object2IntMap<String> f_14937_ = DataFixUtils.make(new Object2IntOpenHashMap<>(), (p_209419_0_) -> {
+      p_209419_0_.defaultReturnValue(-1);
    });
 
    private static void m_14942_(int p_14943_, String p_14944_, String... p_14945_) {
