--- a/net/minecraft/util/datafix/fixes/BlockEntitySignTextStrictJsonFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntitySignTextStrictJsonFix.java
@@ -88,11 +_,11 @@
    }
 
    protected Typed<?> m_7504_(Typed<?> p_14867_) {
-      return p_14867_.update(DSL.remainderFinder(), (p_14869_) -> {
-         p_14869_ = this.m_14870_(p_14869_, "Text1");
-         p_14869_ = this.m_14870_(p_14869_, "Text2");
-         p_14869_ = this.m_14870_(p_14869_, "Text3");
-         return this.m_14870_(p_14869_, "Text4");
+      return p_14867_.update(DSL.remainderFinder(), (p_206380_1_) -> {
+         p_206380_1_ = this.m_14870_(p_206380_1_, "Text1");
+         p_206380_1_ = this.m_14870_(p_206380_1_, "Text2");
+         p_206380_1_ = this.m_14870_(p_206380_1_, "Text3");
+         return this.m_14870_(p_206380_1_, "Text4");
       });
    }
 }
