--- a/net/minecraft/util/datafix/fixes/BlockEntityBannerColorFix.java
+++ b/net/minecraft/util/datafix/fixes/BlockEntityBannerColorFix.java
@@ -12,17 +_,17 @@
    }
 
    public Dynamic<?> m_14797_(Dynamic<?> p_14798_) {
-      p_14798_ = p_14798_.update("Base", (p_14808_) -> {
-         return p_14808_.createInt(15 - p_14808_.asInt(0));
+      p_14798_ = p_14798_.update("Base", (p_209642_0_) -> {
+         return p_209642_0_.createInt(15 - p_209642_0_.asInt(0));
       });
-      return p_14798_.update("Patterns", (p_14802_) -> {
-         return DataFixUtils.orElse(p_14802_.asStreamOpt().map((p_14800_) -> {
-            return p_14800_.map((p_14804_) -> {
-               return p_14804_.update("Color", (p_14806_) -> {
-                  return p_14806_.createInt(15 - p_14806_.asInt(0));
+      return p_14798_.update("Patterns", (p_209641_0_) -> {
+         return DataFixUtils.orElse(p_209641_0_.asStreamOpt().map((p_209639_0_) -> {
+            return p_209639_0_.map((p_210561_0_) -> {
+               return p_210561_0_.update("Color", (p_209644_0_) -> {
+                  return p_209644_0_.createInt(15 - p_209644_0_.asInt(0));
                });
             });
-         }).map(p_14802_::createList).result(), p_14802_);
+         }).map(p_209641_0_::createList).result(), p_209641_0_);
       });
    }
 
