--- a/net/minecraft/util/datafix/fixes/ChunkPalettedStorageFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkPalettedStorageFix.java
@@ -43,84 +_,84 @@
    private static final Dynamic<?> f_15045_ = BlockStateData.m_14956_("{Name:'minecraft:large_fern',Properties:{half:'upper'}}");
    private static final Dynamic<?> f_15046_ = BlockStateData.m_14956_("{Name:'minecraft:rose_bush',Properties:{half:'upper'}}");
    private static final Dynamic<?> f_15047_ = BlockStateData.m_14956_("{Name:'minecraft:peony',Properties:{half:'upper'}}");
-   private static final Map<String, Dynamic<?>> f_15048_ = DataFixUtils.make(Maps.newHashMap(), (p_15111_) -> {
-      p_15111_.put("minecraft:air0", BlockStateData.m_14956_("{Name:'minecraft:flower_pot'}"));
-      p_15111_.put("minecraft:red_flower0", BlockStateData.m_14956_("{Name:'minecraft:potted_poppy'}"));
-      p_15111_.put("minecraft:red_flower1", BlockStateData.m_14956_("{Name:'minecraft:potted_blue_orchid'}"));
-      p_15111_.put("minecraft:red_flower2", BlockStateData.m_14956_("{Name:'minecraft:potted_allium'}"));
-      p_15111_.put("minecraft:red_flower3", BlockStateData.m_14956_("{Name:'minecraft:potted_azure_bluet'}"));
-      p_15111_.put("minecraft:red_flower4", BlockStateData.m_14956_("{Name:'minecraft:potted_red_tulip'}"));
-      p_15111_.put("minecraft:red_flower5", BlockStateData.m_14956_("{Name:'minecraft:potted_orange_tulip'}"));
-      p_15111_.put("minecraft:red_flower6", BlockStateData.m_14956_("{Name:'minecraft:potted_white_tulip'}"));
-      p_15111_.put("minecraft:red_flower7", BlockStateData.m_14956_("{Name:'minecraft:potted_pink_tulip'}"));
-      p_15111_.put("minecraft:red_flower8", BlockStateData.m_14956_("{Name:'minecraft:potted_oxeye_daisy'}"));
-      p_15111_.put("minecraft:yellow_flower0", BlockStateData.m_14956_("{Name:'minecraft:potted_dandelion'}"));
-      p_15111_.put("minecraft:sapling0", BlockStateData.m_14956_("{Name:'minecraft:potted_oak_sapling'}"));
-      p_15111_.put("minecraft:sapling1", BlockStateData.m_14956_("{Name:'minecraft:potted_spruce_sapling'}"));
-      p_15111_.put("minecraft:sapling2", BlockStateData.m_14956_("{Name:'minecraft:potted_birch_sapling'}"));
-      p_15111_.put("minecraft:sapling3", BlockStateData.m_14956_("{Name:'minecraft:potted_jungle_sapling'}"));
-      p_15111_.put("minecraft:sapling4", BlockStateData.m_14956_("{Name:'minecraft:potted_acacia_sapling'}"));
-      p_15111_.put("minecraft:sapling5", BlockStateData.m_14956_("{Name:'minecraft:potted_dark_oak_sapling'}"));
-      p_15111_.put("minecraft:red_mushroom0", BlockStateData.m_14956_("{Name:'minecraft:potted_red_mushroom'}"));
-      p_15111_.put("minecraft:brown_mushroom0", BlockStateData.m_14956_("{Name:'minecraft:potted_brown_mushroom'}"));
-      p_15111_.put("minecraft:deadbush0", BlockStateData.m_14956_("{Name:'minecraft:potted_dead_bush'}"));
-      p_15111_.put("minecraft:tallgrass2", BlockStateData.m_14956_("{Name:'minecraft:potted_fern'}"));
-      p_15111_.put("minecraft:cactus0", BlockStateData.m_14952_(2240));
-   });
-   private static final Map<String, Dynamic<?>> f_15049_ = DataFixUtils.make(Maps.newHashMap(), (p_15108_) -> {
-      m_15077_(p_15108_, 0, "skeleton", "skull");
-      m_15077_(p_15108_, 1, "wither_skeleton", "skull");
-      m_15077_(p_15108_, 2, "zombie", "head");
-      m_15077_(p_15108_, 3, "player", "head");
-      m_15077_(p_15108_, 4, "creeper", "head");
-      m_15077_(p_15108_, 5, "dragon", "head");
-   });
-   private static final Map<String, Dynamic<?>> f_15050_ = DataFixUtils.make(Maps.newHashMap(), (p_15105_) -> {
-      m_15082_(p_15105_, "oak_door", 1024);
-      m_15082_(p_15105_, "iron_door", 1136);
-      m_15082_(p_15105_, "spruce_door", 3088);
-      m_15082_(p_15105_, "birch_door", 3104);
-      m_15082_(p_15105_, "jungle_door", 3120);
-      m_15082_(p_15105_, "acacia_door", 3136);
-      m_15082_(p_15105_, "dark_oak_door", 3152);
-   });
-   private static final Map<String, Dynamic<?>> f_15051_ = DataFixUtils.make(Maps.newHashMap(), (p_15102_) -> {
+   private static final Map<String, Dynamic<?>> f_15048_ = DataFixUtils.make(Maps.newHashMap(), (p_209306_0_) -> {
+      p_209306_0_.put("minecraft:air0", BlockStateData.m_14956_("{Name:'minecraft:flower_pot'}"));
+      p_209306_0_.put("minecraft:red_flower0", BlockStateData.m_14956_("{Name:'minecraft:potted_poppy'}"));
+      p_209306_0_.put("minecraft:red_flower1", BlockStateData.m_14956_("{Name:'minecraft:potted_blue_orchid'}"));
+      p_209306_0_.put("minecraft:red_flower2", BlockStateData.m_14956_("{Name:'minecraft:potted_allium'}"));
+      p_209306_0_.put("minecraft:red_flower3", BlockStateData.m_14956_("{Name:'minecraft:potted_azure_bluet'}"));
+      p_209306_0_.put("minecraft:red_flower4", BlockStateData.m_14956_("{Name:'minecraft:potted_red_tulip'}"));
+      p_209306_0_.put("minecraft:red_flower5", BlockStateData.m_14956_("{Name:'minecraft:potted_orange_tulip'}"));
+      p_209306_0_.put("minecraft:red_flower6", BlockStateData.m_14956_("{Name:'minecraft:potted_white_tulip'}"));
+      p_209306_0_.put("minecraft:red_flower7", BlockStateData.m_14956_("{Name:'minecraft:potted_pink_tulip'}"));
+      p_209306_0_.put("minecraft:red_flower8", BlockStateData.m_14956_("{Name:'minecraft:potted_oxeye_daisy'}"));
+      p_209306_0_.put("minecraft:yellow_flower0", BlockStateData.m_14956_("{Name:'minecraft:potted_dandelion'}"));
+      p_209306_0_.put("minecraft:sapling0", BlockStateData.m_14956_("{Name:'minecraft:potted_oak_sapling'}"));
+      p_209306_0_.put("minecraft:sapling1", BlockStateData.m_14956_("{Name:'minecraft:potted_spruce_sapling'}"));
+      p_209306_0_.put("minecraft:sapling2", BlockStateData.m_14956_("{Name:'minecraft:potted_birch_sapling'}"));
+      p_209306_0_.put("minecraft:sapling3", BlockStateData.m_14956_("{Name:'minecraft:potted_jungle_sapling'}"));
+      p_209306_0_.put("minecraft:sapling4", BlockStateData.m_14956_("{Name:'minecraft:potted_acacia_sapling'}"));
+      p_209306_0_.put("minecraft:sapling5", BlockStateData.m_14956_("{Name:'minecraft:potted_dark_oak_sapling'}"));
+      p_209306_0_.put("minecraft:red_mushroom0", BlockStateData.m_14956_("{Name:'minecraft:potted_red_mushroom'}"));
+      p_209306_0_.put("minecraft:brown_mushroom0", BlockStateData.m_14956_("{Name:'minecraft:potted_brown_mushroom'}"));
+      p_209306_0_.put("minecraft:deadbush0", BlockStateData.m_14956_("{Name:'minecraft:potted_dead_bush'}"));
+      p_209306_0_.put("minecraft:tallgrass2", BlockStateData.m_14956_("{Name:'minecraft:potted_fern'}"));
+      p_209306_0_.put("minecraft:cactus0", BlockStateData.m_14952_(2240));
+   });
+   private static final Map<String, Dynamic<?>> f_15049_ = DataFixUtils.make(Maps.newHashMap(), (p_209308_0_) -> {
+      m_15077_(p_209308_0_, 0, "skeleton", "skull");
+      m_15077_(p_209308_0_, 1, "wither_skeleton", "skull");
+      m_15077_(p_209308_0_, 2, "zombie", "head");
+      m_15077_(p_209308_0_, 3, "player", "head");
+      m_15077_(p_209308_0_, 4, "creeper", "head");
+      m_15077_(p_209308_0_, 5, "dragon", "head");
+   });
+   private static final Map<String, Dynamic<?>> f_15050_ = DataFixUtils.make(Maps.newHashMap(), (p_209298_0_) -> {
+      m_15082_(p_209298_0_, "oak_door", 1024);
+      m_15082_(p_209298_0_, "iron_door", 1136);
+      m_15082_(p_209298_0_, "spruce_door", 3088);
+      m_15082_(p_209298_0_, "birch_door", 3104);
+      m_15082_(p_209298_0_, "jungle_door", 3120);
+      m_15082_(p_209298_0_, "acacia_door", 3136);
+      m_15082_(p_209298_0_, "dark_oak_door", 3152);
+   });
+   private static final Map<String, Dynamic<?>> f_15051_ = DataFixUtils.make(Maps.newHashMap(), (p_209302_0_) -> {
       for(int i = 0; i < 26; ++i) {
-         p_15102_.put("true" + i, BlockStateData.m_14956_("{Name:'minecraft:note_block',Properties:{powered:'true',note:'" + i + "'}}"));
-         p_15102_.put("false" + i, BlockStateData.m_14956_("{Name:'minecraft:note_block',Properties:{powered:'false',note:'" + i + "'}}"));
+         p_209302_0_.put("true" + i, BlockStateData.m_14956_("{Name:'minecraft:note_block',Properties:{powered:'true',note:'" + i + "'}}"));
+         p_209302_0_.put("false" + i, BlockStateData.m_14956_("{Name:'minecraft:note_block',Properties:{powered:'false',note:'" + i + "'}}"));
       }
 
    });
-   private static final Int2ObjectMap<String> f_15052_ = DataFixUtils.make(new Int2ObjectOpenHashMap<>(), (p_15070_) -> {
-      p_15070_.put(0, "white");
-      p_15070_.put(1, "orange");
-      p_15070_.put(2, "magenta");
-      p_15070_.put(3, "light_blue");
-      p_15070_.put(4, "yellow");
-      p_15070_.put(5, "lime");
-      p_15070_.put(6, "pink");
-      p_15070_.put(7, "gray");
-      p_15070_.put(8, "light_gray");
-      p_15070_.put(9, "cyan");
-      p_15070_.put(10, "purple");
-      p_15070_.put(11, "blue");
-      p_15070_.put(12, "brown");
-      p_15070_.put(13, "green");
-      p_15070_.put(14, "red");
-      p_15070_.put(15, "black");
+   private static final Int2ObjectMap<String> f_15052_ = DataFixUtils.make(new Int2ObjectOpenHashMap<>(), (p_209296_0_) -> {
+      p_209296_0_.put(0, "white");
+      p_209296_0_.put(1, "orange");
+      p_209296_0_.put(2, "magenta");
+      p_209296_0_.put(3, "light_blue");
+      p_209296_0_.put(4, "yellow");
+      p_209296_0_.put(5, "lime");
+      p_209296_0_.put(6, "pink");
+      p_209296_0_.put(7, "gray");
+      p_209296_0_.put(8, "light_gray");
+      p_209296_0_.put(9, "cyan");
+      p_209296_0_.put(10, "purple");
+      p_209296_0_.put(11, "blue");
+      p_209296_0_.put(12, "brown");
+      p_209296_0_.put(13, "green");
+      p_209296_0_.put(14, "red");
+      p_209296_0_.put(15, "black");
    });
-   private static final Map<String, Dynamic<?>> f_15053_ = DataFixUtils.make(Maps.newHashMap(), (p_15095_) -> {
+   private static final Map<String, Dynamic<?>> f_15053_ = DataFixUtils.make(Maps.newHashMap(), (p_209304_0_) -> {
       for(Entry<String> entry : f_15052_.int2ObjectEntrySet()) {
          if (!Objects.equals(entry.getValue(), "red")) {
-            m_15073_(p_15095_, entry.getIntKey(), entry.getValue());
+            m_15073_(p_209304_0_, entry.getIntKey(), entry.getValue());
          }
       }
 
    });
-   private static final Map<String, Dynamic<?>> f_15054_ = DataFixUtils.make(Maps.newHashMap(), (p_15072_) -> {
+   private static final Map<String, Dynamic<?>> f_15054_ = DataFixUtils.make(Maps.newHashMap(), (p_209299_0_) -> {
       for(Entry<String> entry : f_15052_.int2ObjectEntrySet()) {
          if (!Objects.equals(entry.getValue(), "white")) {
-            m_15096_(p_15072_, 15 - entry.getIntKey(), entry.getValue());
+            m_15096_(p_209299_0_, 15 - entry.getIntKey(), entry.getValue());
          }
       }
 
@@ -354,34 +_,6 @@
       f_15036_.set(199);
    }
 
-   static class DataLayer {
-      private final byte[] f_15129_;
-
-      public DataLayer() {
-         this.f_15129_ = new byte[2048];
-      }
-
-      public DataLayer(byte[] p_15132_) {
-         this.f_15129_ = p_15132_;
-         if (p_15132_.length != 2048) {
-            throw new IllegalArgumentException("ChunkNibbleArrays should be 2048 bytes not: " + p_15132_.length);
-         }
-      }
-
-      public int m_15135_(int p_15136_, int p_15137_, int p_15138_) {
-         int i = this.m_15139_(p_15137_ << 8 | p_15138_ << 4 | p_15136_);
-         return this.m_15133_(p_15137_ << 8 | p_15138_ << 4 | p_15136_) ? this.f_15129_[i] & 15 : this.f_15129_[i] >> 4 & 15;
-      }
-
-      private boolean m_15133_(int p_15134_) {
-         return (p_15134_ & 1) == 0;
-      }
-
-      private int m_15139_(int p_15140_) {
-         return p_15140_ >> 1;
-      }
-   }
-
    public static enum Direction {
       DOWN(ChunkPalettedStorageFix.Direction.AxisDirection.NEGATIVE, ChunkPalettedStorageFix.Direction.Axis.Y),
       UP(ChunkPalettedStorageFix.Direction.AxisDirection.POSITIVE, ChunkPalettedStorageFix.Direction.Axis.Y),
@@ -428,6 +_,34 @@
       }
    }
 
+   static class DataLayer {
+      private final byte[] f_15129_;
+
+      public DataLayer() {
+         this.f_15129_ = new byte[2048];
+      }
+
+      public DataLayer(byte[] p_15132_) {
+         this.f_15129_ = p_15132_;
+         if (p_15132_.length != 2048) {
+            throw new IllegalArgumentException("ChunkNibbleArrays should be 2048 bytes not: " + p_15132_.length);
+         }
+      }
+
+      public int m_15135_(int p_15136_, int p_15137_, int p_15138_) {
+         int i = this.m_15139_(p_15137_ << 8 | p_15138_ << 4 | p_15136_);
+         return this.m_15133_(p_15137_ << 8 | p_15138_ << 4 | p_15136_) ? this.f_15129_[i] & 15 : this.f_15129_[i] >> 4 & 15;
+      }
+
+      private boolean m_15133_(int p_15134_) {
+         return (p_15134_ & 1) == 0;
+      }
+
+      private int m_15139_(int p_15140_) {
+         return p_15140_ >> 1;
+      }
+   }
+
    static class Section {
       private final CrudeIncrementalIntIdentityHashBiMap<Dynamic<?>> f_15186_ = new CrudeIncrementalIntIdentityHashBiMap<>(32);
       private final List<Dynamic<?>> f_15187_;
@@ -468,11 +_,11 @@
             return p_15210_;
          } else {
             ByteBuffer bytebuffer = this.f_15188_.get("Blocks").asByteBufferOpt().result().get();
-            ChunkPalettedStorageFix.DataLayer chunkpalettedstoragefix$datalayer = this.f_15188_.get("Data").asByteBufferOpt().map((p_15214_) -> {
-               return new ChunkPalettedStorageFix.DataLayer(DataFixUtils.toArray(p_15214_));
+            ChunkPalettedStorageFix.DataLayer chunkpalettedstoragefix$datalayer = this.f_15188_.get("Data").asByteBufferOpt().map((p_210055_0_) -> {
+               return new ChunkPalettedStorageFix.DataLayer(DataFixUtils.toArray(p_210055_0_));
             }).result().orElseGet(ChunkPalettedStorageFix.DataLayer::new);
-            ChunkPalettedStorageFix.DataLayer chunkpalettedstoragefix$datalayer1 = this.f_15188_.get("Add").asByteBufferOpt().map((p_15208_) -> {
-               return new ChunkPalettedStorageFix.DataLayer(DataFixUtils.toArray(p_15208_));
+            ChunkPalettedStorageFix.DataLayer chunkpalettedstoragefix$datalayer1 = this.f_15188_.get("Add").asByteBufferOpt().map((p_210052_0_) -> {
+               return new ChunkPalettedStorageFix.DataLayer(DataFixUtils.toArray(p_210052_0_));
             }).result().orElseGet(ChunkPalettedStorageFix.DataLayer::new);
             this.f_15192_.add(ChunkPalettedStorageFix.f_15055_);
             ChunkPalettedStorageFix.m_15061_(this.f_15186_, ChunkPalettedStorageFix.f_15055_);
@@ -546,22 +_,22 @@
          this.f_15217_ = p_15222_;
          this.f_15218_ = p_15222_.get("xPos").asInt(0) << 4;
          this.f_15219_ = p_15222_.get("zPos").asInt(0) << 4;
-         p_15222_.get("TileEntities").asStreamOpt().result().ifPresent((p_15241_) -> {
-            p_15241_.forEach((p_15239_) -> {
-               int l3 = p_15239_.get("x").asInt(0) - this.f_15218_ & 15;
-               int i4 = p_15239_.get("y").asInt(0);
-               int j4 = p_15239_.get("z").asInt(0) - this.f_15219_ & 15;
+         p_15222_.get("TileEntities").asStreamOpt().result().ifPresent((p_210061_1_) -> {
+            p_210061_1_.forEach((p_233150_1_) -> {
+               int l3 = p_233150_1_.get("x").asInt(0) - this.f_15218_ & 15;
+               int i4 = p_233150_1_.get("y").asInt(0);
+               int j4 = p_233150_1_.get("z").asInt(0) - this.f_15219_ & 15;
                int k4 = i4 << 8 | j4 << 4 | l3;
-               if (this.f_15220_.put(k4, p_15239_) != null) {
+               if (this.f_15220_.put(k4, p_233150_1_) != null) {
                   ChunkPalettedStorageFix.f_15035_.warn("In chunk: {}x{} found a duplicate block entity at position: [{}, {}, {}]", this.f_15218_, this.f_15219_, l3, i4, j4);
                }
 
             });
          });
          boolean flag = p_15222_.get("convertedFromAlphaFormat").asBoolean(false);
-         p_15222_.get("Sections").asStreamOpt().result().ifPresent((p_15235_) -> {
-            p_15235_.forEach((p_15233_) -> {
-               ChunkPalettedStorageFix.Section chunkpalettedstoragefix$section1 = new ChunkPalettedStorageFix.Section(p_15233_);
+         p_15222_.get("Sections").asStreamOpt().result().ifPresent((p_210062_1_) -> {
+            p_210062_1_.forEach((p_210065_1_) -> {
+               ChunkPalettedStorageFix.Section chunkpalettedstoragefix$section1 = new ChunkPalettedStorageFix.Section(p_210065_1_);
                this.f_15215_ = chunkpalettedstoragefix$section1.m_15209_(this.f_15215_);
                this.f_15216_[chunkpalettedstoragefix$section1.f_15185_] = chunkpalettedstoragefix$section1;
             });
