--- a/net/minecraft/util/datafix/fixes/ChunkStructuresTemplateRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkStructuresTemplateRenameFix.java
@@ -20,10 +_,10 @@
    public TypeRewriteRule makeRule() {
       Type<?> type = this.getInputSchema().getType(References.f_16790_);
       OpticFinder<?> opticfinder = type.findField("Children");
-      return this.fixTypeEverywhereTyped("ChunkStructuresTemplateRenameFix", type, (p_15273_) -> {
-         return p_15273_.updateTyped(opticfinder, (p_15276_) -> {
-            return p_15276_.update(DSL.remainderFinder(), (p_15279_) -> {
-               return this.m_15280_(p_15273_.get(DSL.remainderFinder()), p_15279_);
+      return this.fixTypeEverywhereTyped("ChunkStructuresTemplateRenameFix", type, (p_207911_2_) -> {
+         return p_207911_2_.updateTyped(opticfinder, (p_207908_2_) -> {
+            return p_207908_2_.update(DSL.remainderFinder(), (p_207909_2_) -> {
+               return this.m_15280_(p_207911_2_.get(DSL.remainderFinder()), p_207909_2_);
             });
          });
       });
