--- a/net/minecraft/util/datafix/fixes/ColorlessShulkerEntityFix.java
+++ b/net/minecraft/util/datafix/fixes/ColorlessShulkerEntityFix.java
@@ -10,8 +_,8 @@
    }
 
    protected Typed<?> m_7504_(Typed<?> p_15318_) {
-      return p_15318_.update(DSL.remainderFinder(), (p_15320_) -> {
-         return p_15320_.get("Color").asInt(0) == 10 ? p_15320_.set("Color", p_15320_.createByte((byte)16)) : p_15320_;
+      return p_15318_.update(DSL.remainderFinder(), (p_207421_0_) -> {
+         return p_207421_0_.get("Color").asInt(0) == 10 ? p_207421_0_.set("Color", p_207421_0_.createByte((byte)16)) : p_207421_0_;
       });
    }
 }
