--- a/net/minecraft/util/datafix/fixes/EntityUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityUUIDFix.java
@@ -22,52 +_,52 @@
    }
 
    protected TypeRewriteRule makeRule() {
-      return this.fixTypeEverywhereTyped("EntityUUIDFixes", this.getInputSchema().getType(this.f_14569_), (p_15725_) -> {
-         p_15725_ = p_15725_.update(DSL.remainderFinder(), EntityUUIDFix::m_15734_);
+      return this.fixTypeEverywhereTyped("EntityUUIDFixes", this.getInputSchema().getType(this.f_14569_), (p_233210_1_) -> {
+         p_233210_1_ = p_233210_1_.update(DSL.remainderFinder(), EntityUUIDFix::m_15734_);
 
          for(String s : f_15715_) {
-            p_15725_ = this.m_14574_(p_15725_, s, EntityUUIDFix::m_15761_);
+            p_233210_1_ = this.m_14574_(p_233210_1_, s, EntityUUIDFix::m_15761_);
          }
 
          for(String s1 : f_15716_) {
-            p_15725_ = this.m_14574_(p_15725_, s1, EntityUUIDFix::m_15761_);
+            p_233210_1_ = this.m_14574_(p_233210_1_, s1, EntityUUIDFix::m_15761_);
          }
 
          for(String s2 : f_15717_) {
-            p_15725_ = this.m_14574_(p_15725_, s2, EntityUUIDFix::m_15763_);
+            p_233210_1_ = this.m_14574_(p_233210_1_, s2, EntityUUIDFix::m_15763_);
          }
 
          for(String s3 : f_15718_) {
-            p_15725_ = this.m_14574_(p_15725_, s3, EntityUUIDFix::m_15766_);
+            p_233210_1_ = this.m_14574_(p_233210_1_, s3, EntityUUIDFix::m_15766_);
          }
 
          for(String s4 : f_15719_) {
-            p_15725_ = this.m_14574_(p_15725_, s4, EntityUUIDFix::m_15729_);
+            p_233210_1_ = this.m_14574_(p_233210_1_, s4, EntityUUIDFix::m_15729_);
          }
 
          for(String s5 : f_15720_) {
-            p_15725_ = this.m_14574_(p_15725_, s5, EntityUUIDFix::m_15768_);
+            p_233210_1_ = this.m_14574_(p_233210_1_, s5, EntityUUIDFix::m_15768_);
          }
 
-         p_15725_ = this.m_14574_(p_15725_, "minecraft:bee", EntityUUIDFix::m_15759_);
-         p_15725_ = this.m_14574_(p_15725_, "minecraft:zombified_piglin", EntityUUIDFix::m_15759_);
-         p_15725_ = this.m_14574_(p_15725_, "minecraft:fox", EntityUUIDFix::m_15757_);
-         p_15725_ = this.m_14574_(p_15725_, "minecraft:item", EntityUUIDFix::m_15755_);
-         p_15725_ = this.m_14574_(p_15725_, "minecraft:shulker_bullet", EntityUUIDFix::m_15753_);
-         p_15725_ = this.m_14574_(p_15725_, "minecraft:area_effect_cloud", EntityUUIDFix::m_15751_);
-         p_15725_ = this.m_14574_(p_15725_, "minecraft:zombie_villager", EntityUUIDFix::m_15749_);
-         p_15725_ = this.m_14574_(p_15725_, "minecraft:evoker_fangs", EntityUUIDFix::m_15744_);
-         return this.m_14574_(p_15725_, "minecraft:piglin", EntityUUIDFix::m_15739_);
+         p_233210_1_ = this.m_14574_(p_233210_1_, "minecraft:bee", EntityUUIDFix::m_15759_);
+         p_233210_1_ = this.m_14574_(p_233210_1_, "minecraft:zombified_piglin", EntityUUIDFix::m_15759_);
+         p_233210_1_ = this.m_14574_(p_233210_1_, "minecraft:fox", EntityUUIDFix::m_15757_);
+         p_233210_1_ = this.m_14574_(p_233210_1_, "minecraft:item", EntityUUIDFix::m_15755_);
+         p_233210_1_ = this.m_14574_(p_233210_1_, "minecraft:shulker_bullet", EntityUUIDFix::m_15753_);
+         p_233210_1_ = this.m_14574_(p_233210_1_, "minecraft:area_effect_cloud", EntityUUIDFix::m_15751_);
+         p_233210_1_ = this.m_14574_(p_233210_1_, "minecraft:zombie_villager", EntityUUIDFix::m_15749_);
+         p_233210_1_ = this.m_14574_(p_233210_1_, "minecraft:evoker_fangs", EntityUUIDFix::m_15744_);
+         return this.m_14574_(p_233210_1_, "minecraft:piglin", EntityUUIDFix::m_15739_);
       });
    }
 
    private static Dynamic<?> m_15739_(Dynamic<?> p_15740_) {
-      return p_15740_.update("Brain", (p_15781_) -> {
-         return p_15781_.update("memories", (p_15783_) -> {
-            return p_15783_.update("minecraft:angry_at", (p_15785_) -> {
-               return m_14590_(p_15785_, "value", "value").orElseGet(() -> {
+      return p_15740_.update("Brain", (p_233235_0_) -> {
+         return p_233235_0_.update("memories", (p_233236_0_) -> {
+            return p_233236_0_.update("minecraft:angry_at", (p_233237_0_) -> {
+               return m_14590_(p_233237_0_, "value", "value").orElseGet(() -> {
                   f_14568_.warn("angry_at has no value.");
-                  return p_15785_;
+                  return p_233237_0_;
                });
             });
          });
@@ -97,16 +_,16 @@
    }
 
    private static Dynamic<?> m_15757_(Dynamic<?> p_15758_) {
-      Optional<Dynamic<?>> optional = p_15758_.get("TrustedUUIDs").result().map((p_15748_) -> {
-         return p_15758_.createList(p_15748_.asStream().map((p_15777_) -> {
-            return m_14578_(p_15777_).orElseGet(() -> {
+      Optional<Dynamic<?>> optional = p_15758_.get("TrustedUUIDs").result().map((p_233219_1_) -> {
+         return p_15758_.createList(p_233219_1_.asStream().map((p_233233_0_) -> {
+            return m_14578_(p_233233_0_).orElseGet(() -> {
                f_14568_.warn("Trusted contained invalid data.");
-               return p_15777_;
+               return p_233233_0_;
             });
          }));
       });
-      return DataFixUtils.orElse(optional.map((p_15743_) -> {
-         return p_15758_.remove("TrustedUUIDs").set("Trusted", p_15743_);
+      return DataFixUtils.orElse(optional.map((p_233217_1_) -> {
+         return p_15758_.remove("TrustedUUIDs").set("Trusted", p_233217_1_);
       }), p_15758_);
    }
 
@@ -125,17 +_,17 @@
    }
 
    private static Dynamic<?> m_15766_(Dynamic<?> p_15767_) {
-      return m_15729_(p_15767_).update("Leash", (p_15775_) -> {
-         return m_14617_(p_15775_, "UUID", "UUID").orElse(p_15775_);
+      return m_15729_(p_15767_).update("Leash", (p_233232_0_) -> {
+         return m_14617_(p_233232_0_, "UUID", "UUID").orElse(p_233232_0_);
       });
    }
 
    public static Dynamic<?> m_15729_(Dynamic<?> p_15730_) {
-      return p_15730_.update("Attributes", (p_15733_) -> {
-         return p_15730_.createList(p_15733_.asStream().map((p_15771_) -> {
-            return p_15771_.update("Modifiers", (p_15738_) -> {
-               return p_15771_.createList(p_15738_.asStream().map((p_15773_) -> {
-                  return m_14617_(p_15773_, "UUID", "UUID").orElse(p_15773_);
+      return p_15730_.update("Attributes", (p_233213_1_) -> {
+         return p_15730_.createList(p_233213_1_.asStream().map((p_233230_0_) -> {
+            return p_233230_0_.update("Modifiers", (p_233215_1_) -> {
+               return p_233230_0_.createList(p_233215_1_.asStream().map((p_233231_0_) -> {
+                  return m_14617_(p_233231_0_, "UUID", "UUID").orElse(p_233231_0_);
                }));
             });
          }));
@@ -143,8 +_,8 @@
    }
 
    private static Dynamic<?> m_15768_(Dynamic<?> p_15769_) {
-      return DataFixUtils.orElse(p_15769_.get("OwnerUUID").result().map((p_15728_) -> {
-         return p_15769_.remove("OwnerUUID").set("Owner", p_15728_);
+      return DataFixUtils.orElse(p_15769_.get("OwnerUUID").result().map((p_233211_1_) -> {
+         return p_15769_.remove("OwnerUUID").set("Owner", p_233211_1_);
       }), p_15769_);
    }
 
