--- a/net/minecraft/util/datafix/fixes/EntityWolfColorFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityWolfColorFix.java
@@ -11,8 +_,8 @@
    }
 
    public Dynamic<?> m_15793_(Dynamic<?> p_15794_) {
-      return p_15794_.update("CollarColor", (p_15796_) -> {
-         return p_15796_.createByte((byte)(15 - p_15796_.asInt(0)));
+      return p_15794_.update("CollarColor", (p_209654_0_) -> {
+         return p_209654_0_.createByte((byte)(15 - p_209654_0_.asInt(0)));
       });
    }
 
