--- a/net/minecraft/util/datafix/fixes/EntityProjectileOwnerFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityProjectileOwnerFix.java
@@ -72,8 +_,8 @@
    private Typed<?> m_15564_(Typed<?> p_15565_, String p_15566_, Function<Dynamic<?>, Dynamic<?>> p_15567_) {
       Type<?> type = this.getInputSchema().getChoiceType(References.f_16786_, p_15566_);
       Type<?> type1 = this.getOutputSchema().getChoiceType(References.f_16786_, p_15566_);
-      return p_15565_.updateTyped(DSL.namedChoice(p_15566_, type), type1, (p_15576_) -> {
-         return p_15576_.update(DSL.remainderFinder(), p_15567_);
+      return p_15565_.updateTyped(DSL.namedChoice(p_15566_, type), type1, (p_233187_1_) -> {
+         return p_233187_1_.update(DSL.remainderFinder(), p_15567_);
       });
    }
 }
