--- a/net/minecraft/util/datafix/fixes/FurnaceRecipeFix.java
+++ b/net/minecraft/util/datafix/fixes/FurnaceRecipeFix.java
@@ -34,13 +_,13 @@
       Type<?> type3 = this.getOutputSchema().getChoiceType(References.f_16781_, "minecraft:smoker");
       Type<?> type4 = this.getInputSchema().getType(References.f_16781_);
       Type<?> type5 = this.getOutputSchema().getType(References.f_16781_);
-      return this.fixTypeEverywhereTyped("FurnaceRecipesFix", type4, type5, (p_15848_) -> {
-         return p_15848_.updateTyped(opticfinder, type1, (p_15875_) -> {
-            return this.m_15851_(p_15850_, type, p_15875_);
-         }).updateTyped(opticfinder1, type2, (p_15871_) -> {
-            return this.m_15851_(p_15850_, type, p_15871_);
-         }).updateTyped(opticfinder2, type3, (p_15867_) -> {
-            return this.m_15851_(p_15850_, type, p_15867_);
+      return this.fixTypeEverywhereTyped("FurnaceRecipesFix", type4, type5, (p_233247_9_) -> {
+         return p_233247_9_.updateTyped(opticfinder, type1, (p_233254_3_) -> {
+            return this.m_15851_(p_15850_, type, p_233254_3_);
+         }).updateTyped(opticfinder1, type2, (p_233253_3_) -> {
+            return this.m_15851_(p_15850_, type, p_233253_3_);
+         }).updateTyped(opticfinder2, type3, (p_233252_3_) -> {
+            return this.m_15851_(p_15850_, type, p_233252_3_);
          });
       });
    }
@@ -57,10 +_,10 @@
          Optional<? extends Dynamic<?>> optional = dynamic.get(s).result();
          int k = dynamic.get(s1).asInt(0);
          if (k > 0) {
-            optional.ifPresent((p_15859_) -> {
-               Optional<? extends Pair<R, ? extends Dynamic<?>>> optional1 = p_15852_.read(p_15859_).result();
-               optional1.ifPresent((p_15863_) -> {
-                  list.add(Pair.of(p_15863_.getFirst(), k));
+            optional.ifPresent((p_233250_3_) -> {
+               Optional<? extends Pair<R, ? extends Dynamic<?>>> optional1 = p_15852_.read(p_233250_3_).result();
+               optional1.ifPresent((p_233251_2_) -> {
+                  list.add(Pair.of(p_233251_2_.getFirst(), k));
                });
             });
          }
