--- a/net/minecraft/util/datafix/fixes/HeightmapRenamingFix.java
+++ b/net/minecraft/util/datafix/fixes/HeightmapRenamingFix.java
@@ -17,9 +_,9 @@
    protected TypeRewriteRule makeRule() {
       Type<?> type = this.getInputSchema().getType(References.f_16773_);
       OpticFinder<?> opticfinder = type.findField("Level");
-      return this.fixTypeEverywhereTyped("HeightmapRenamingFix", type, (p_15895_) -> {
-         return p_15895_.updateTyped(opticfinder, (p_15897_) -> {
-            return p_15897_.update(DSL.remainderFinder(), this::m_15898_);
+      return this.fixTypeEverywhereTyped("HeightmapRenamingFix", type, (p_207306_2_) -> {
+         return p_207306_2_.updateTyped(opticfinder, (p_207307_1_) -> {
+            return p_207307_1_.update(DSL.remainderFinder(), this::m_15898_);
          });
       });
    }
