--- a/net/minecraft/util/datafix/fixes/IglooMetadataRemovalFix.java
+++ b/net/minecraft/util/datafix/fixes/IglooMetadataRemovalFix.java
@@ -18,16 +_,16 @@
    }
 
    private static <T> Dynamic<T> m_15904_(Dynamic<T> p_15905_) {
-      boolean flag = p_15905_.get("Children").asStreamOpt().map((p_15911_) -> {
-         return p_15911_.allMatch(IglooMetadataRemovalFix::m_15912_);
+      boolean flag = p_15905_.get("Children").asStreamOpt().map((p_211928_0_) -> {
+         return p_211928_0_.allMatch(IglooMetadataRemovalFix::m_15912_);
       }).result().orElse(false);
       return flag ? p_15905_.set("id", p_15905_.createString("Igloo")).remove("Children") : p_15905_.update("Children", IglooMetadataRemovalFix::m_15908_);
    }
 
    private static <T> Dynamic<T> m_15908_(Dynamic<T> p_15909_) {
-      return p_15909_.asStreamOpt().map((p_15907_) -> {
-         return p_15907_.filter((p_15915_) -> {
-            return !m_15912_(p_15915_);
+      return p_15909_.asStreamOpt().map((p_211925_0_) -> {
+         return p_211925_0_.filter((p_211927_0_) -> {
+            return !m_15912_(p_211927_0_);
          });
       }).map(p_15909_::createList).result().orElse(p_15909_);
    }
