--- a/net/minecraft/util/datafix/fixes/ItemStackUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemStackUUIDFix.java
@@ -15,34 +_,34 @@
 
    public TypeRewriteRule makeRule() {
       OpticFinder<Pair<String, String>> opticfinder = DSL.fieldFinder("id", DSL.named(References.f_16788_.typeName(), NamespacedSchema.m_17310_()));
-      return this.fixTypeEverywhereTyped("ItemStackUUIDFix", this.getInputSchema().getType(this.f_14569_), (p_16132_) -> {
-         OpticFinder<?> opticfinder1 = p_16132_.getType().findField("tag");
-         return p_16132_.updateTyped(opticfinder1, (p_16136_) -> {
-            return p_16136_.update(DSL.remainderFinder(), (p_16140_) -> {
-               p_16140_ = this.m_16146_(p_16140_);
-               if (p_16132_.getOptional(opticfinder).map((p_16142_) -> {
-                  return "minecraft:player_head".equals(p_16142_.getSecond());
+      return this.fixTypeEverywhereTyped("ItemStackUUIDFix", this.getInputSchema().getType(this.f_14569_), (p_233277_2_) -> {
+         OpticFinder<?> opticfinder1 = p_233277_2_.getType().findField("tag");
+         return p_233277_2_.updateTyped(opticfinder1, (p_233278_3_) -> {
+            return p_233278_3_.update(DSL.remainderFinder(), (p_233279_3_) -> {
+               p_233279_3_ = this.m_16146_(p_233279_3_);
+               if (p_233277_2_.getOptional(opticfinder).map((p_233280_0_) -> {
+                  return "minecraft:player_head".equals(p_233280_0_.getSecond());
                }).orElse(false)) {
-                  p_16140_ = this.m_16148_(p_16140_);
+                  p_233279_3_ = this.m_16148_(p_233279_3_);
                }
 
-               return p_16140_;
+               return p_233279_3_;
             });
          });
       });
    }
 
    private Dynamic<?> m_16146_(Dynamic<?> p_16147_) {
-      return p_16147_.update("AttributeModifiers", (p_16145_) -> {
-         return p_16147_.createList(p_16145_.asStream().map((p_16153_) -> {
-            return m_14617_(p_16153_, "UUID", "UUID").orElse(p_16153_);
+      return p_16147_.update("AttributeModifiers", (p_233281_1_) -> {
+         return p_16147_.createList(p_233281_1_.asStream().map((p_233285_0_) -> {
+            return m_14617_(p_233285_0_, "UUID", "UUID").orElse(p_233285_0_);
          }));
       });
    }
 
    private Dynamic<?> m_16148_(Dynamic<?> p_16149_) {
-      return p_16149_.update("SkullOwner", (p_16151_) -> {
-         return m_14590_(p_16151_, "Id", "Id").orElse(p_16151_);
+      return p_16149_.update("SkullOwner", (p_233284_0_) -> {
+         return m_14590_(p_233284_0_, "Id", "Id").orElse(p_233284_0_);
       });
    }
 }
