--- a/net/minecraft/util/datafix/fixes/LevelFlatGeneratorInfoFix.java
+++ b/net/minecraft/util/datafix/fixes/LevelFlatGeneratorInfoFix.java
@@ -26,14 +_,14 @@
    }
 
    public TypeRewriteRule makeRule() {
-      return this.fixTypeEverywhereTyped("LevelFlatGeneratorInfoFix", this.getInputSchema().getType(References.f_16771_), (p_16351_) -> {
-         return p_16351_.update(DSL.remainderFinder(), this::m_16352_);
+      return this.fixTypeEverywhereTyped("LevelFlatGeneratorInfoFix", this.getInputSchema().getType(References.f_16771_), (p_207414_1_) -> {
+         return p_207414_1_.update(DSL.remainderFinder(), this::m_16352_);
       });
    }
 
    private Dynamic<?> m_16352_(Dynamic<?> p_16353_) {
-      return p_16353_.get("generatorName").asString("").equalsIgnoreCase("flat") ? p_16353_.update("generatorOptions", (p_16357_) -> {
-         return DataFixUtils.orElse(p_16357_.asString().map(this::m_16354_).map(s -> p_16357_.createString(s)).result(), p_16357_);
+      return p_16353_.get("generatorName").asString("").equalsIgnoreCase("flat") ? p_16353_.update("generatorOptions", (p_209634_1_) -> {
+         return DataFixUtils.orElse(p_209634_1_.asString().map(this::m_16354_).map(s -> p_209634_1_.createString(s)).result(), p_209634_1_);
       }) : p_16353_;
    }
 
@@ -57,8 +_,8 @@
          if (i >= 0 && i <= 3) {
             StringBuilder stringbuilder = new StringBuilder();
             Splitter splitter = i < 3 ? f_16339_ : f_16340_;
-            stringbuilder.append(StreamSupport.stream(f_16338_.split(s1).spliterator(), false).map((p_16349_) -> {
-               List<String> list = splitter.splitToList(p_16349_);
+            stringbuilder.append(StreamSupport.stream(f_16338_.split(s1).spliterator(), false).map((p_206368_2_) -> {
+               List<String> list = splitter.splitToList(p_206368_2_);
                int j;
                String s2;
                if (list.size() == 2) {
