--- a/net/minecraft/util/datafix/fixes/LevelUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/LevelUUIDFix.java
@@ -11,12 +_,12 @@
    }
 
    protected TypeRewriteRule makeRule() {
-      return this.fixTypeEverywhereTyped("LevelUUIDFix", this.getInputSchema().getType(this.f_14569_), (p_16362_) -> {
-         return p_16362_.updateTyped(DSL.remainderFinder(), (p_16369_) -> {
-            return p_16369_.update(DSL.remainderFinder(), (p_16393_) -> {
-               p_16393_ = this.m_16376_(p_16393_);
-               p_16393_ = this.m_16374_(p_16393_);
-               return this.m_16372_(p_16393_);
+      return this.fixTypeEverywhereTyped("LevelUUIDFix", this.getInputSchema().getType(this.f_14569_), (p_233308_1_) -> {
+         return p_233308_1_.updateTyped(DSL.remainderFinder(), (p_233311_1_) -> {
+            return p_233311_1_.update(DSL.remainderFinder(), (p_233323_1_) -> {
+               p_233323_1_ = this.m_16376_(p_233323_1_);
+               p_233323_1_ = this.m_16374_(p_233323_1_);
+               return this.m_16372_(p_233323_1_);
             });
          });
       });
@@ -27,11 +_,11 @@
    }
 
    private Dynamic<?> m_16374_(Dynamic<?> p_16375_) {
-      return p_16375_.update("DimensionData", (p_16387_) -> {
-         return p_16387_.updateMapValues((p_16371_) -> {
-            return p_16371_.mapSecond((p_16389_) -> {
-               return p_16389_.update("DragonFight", (p_16391_) -> {
-                  return m_14617_(p_16391_, "DragonUUID", "Dragon").orElse(p_16391_);
+      return p_16375_.update("DimensionData", (p_233320_0_) -> {
+         return p_233320_0_.updateMapValues((p_233312_0_) -> {
+            return p_233312_0_.mapSecond((p_233321_0_) -> {
+               return p_233321_0_.update("DragonFight", (p_233322_0_) -> {
+                  return m_14617_(p_233322_0_, "DragonUUID", "Dragon").orElse(p_233322_0_);
                });
             });
          });
@@ -39,14 +_,14 @@
    }
 
    private Dynamic<?> m_16376_(Dynamic<?> p_16377_) {
-      return p_16377_.update("CustomBossEvents", (p_16379_) -> {
-         return p_16379_.updateMapValues((p_16364_) -> {
-            return p_16364_.mapSecond((p_16381_) -> {
-               return p_16381_.update("Players", (p_16367_) -> {
-                  return p_16381_.createList(p_16367_.asStream().map((p_16383_) -> {
-                     return m_14578_(p_16383_).orElseGet(() -> {
+      return p_16377_.update("CustomBossEvents", (p_233316_0_) -> {
+         return p_233316_0_.updateMapValues((p_233309_0_) -> {
+            return p_233309_0_.mapSecond((p_233317_0_) -> {
+               return p_233317_0_.update("Players", (p_233310_1_) -> {
+                  return p_233317_0_.createList(p_233310_1_.asStream().map((p_233318_0_) -> {
+                     return m_14578_(p_233318_0_).orElseGet(() -> {
                         f_14568_.warn("CustomBossEvents contains invalid UUIDs.");
-                        return p_16383_;
+                        return p_233318_0_;
                      });
                   }));
                });
