--- a/net/minecraft/util/datafix/fixes/MobSpawnerEntityIdentifiersFix.java
+++ b/net/minecraft/util/datafix/fixes/MobSpawnerEntityIdentifiersFix.java
@@ -32,13 +_,13 @@
 
          Optional<? extends Stream<? extends Dynamic<?>>> optional1 = p_16457_.get("SpawnPotentials").asStreamOpt().result();
          if (optional1.isPresent()) {
-            p_16457_ = p_16457_.set("SpawnPotentials", p_16457_.createList(optional1.get().map((p_16459_) -> {
-               Optional<String> optional2 = p_16459_.get("Type").asString().result();
+            p_16457_ = p_16457_.set("SpawnPotentials", p_16457_.createList(optional1.get().map((p_209657_0_) -> {
+               Optional<String> optional2 = p_209657_0_.get("Type").asString().result();
                if (optional2.isPresent()) {
-                  Dynamic<?> dynamic1 = DataFixUtils.orElse(p_16459_.get("Properties").result(), p_16459_.emptyMap()).set("id", p_16459_.createString(optional2.get()));
-                  return p_16459_.set("Entity", dynamic1).remove("Type").remove("Properties");
+                  Dynamic<?> dynamic1 = DataFixUtils.orElse(p_209657_0_.get("Properties").result(), p_209657_0_.emptyMap()).set("id", p_209657_0_.createString(optional2.get()));
+                  return p_209657_0_.set("Entity", dynamic1).remove("Type").remove("Properties");
                } else {
-                  return p_16459_;
+                  return p_209657_0_;
                }
             })));
          }
@@ -49,11 +_,11 @@
 
    public TypeRewriteRule makeRule() {
       Type<?> type = this.getOutputSchema().getType(References.f_16789_);
-      return this.fixTypeEverywhereTyped("MobSpawnerEntityIdentifiersFix", this.getInputSchema().getType(References.f_16789_), type, (p_16455_) -> {
-         Dynamic<?> dynamic = p_16455_.get(DSL.remainderFinder());
+      return this.fixTypeEverywhereTyped("MobSpawnerEntityIdentifiersFix", this.getInputSchema().getType(References.f_16789_), type, (p_206369_2_) -> {
+         Dynamic<?> dynamic = p_206369_2_.get(DSL.remainderFinder());
          dynamic = dynamic.set("id", dynamic.createString("MobSpawner"));
          DataResult<? extends Pair<? extends Typed<?>, ?>> dataresult = type.readTyped(this.m_16456_(dynamic));
-         return !dataresult.result().isPresent() ? p_16455_ : dataresult.result().get().getFirst();
+         return !dataresult.result().isPresent() ? p_206369_2_ : dataresult.result().get().getFirst();
       });
    }
 }
