--- a/net/minecraft/util/datafix/fixes/NewVillageFix.java
+++ b/net/minecraft/util/datafix/fixes/NewVillageFix.java
@@ -34,33 +_,33 @@
       OpticFinder<?> opticfinder1 = opticfinder.type().findField("Structures");
       OpticFinder<?> opticfinder2 = opticfinder1.type().findField("Starts");
       OpticFinder<List<Pair<String, SF>>> opticfinder3 = p_16499_.finder();
-      return TypeRewriteRule.seq(this.fixTypeEverywhereTyped("NewVillageFix", type, (p_16483_) -> {
-         return p_16483_.updateTyped(opticfinder, (p_16488_) -> {
-            return p_16488_.updateTyped(opticfinder1, (p_16492_) -> {
-               return p_16492_.updateTyped(opticfinder2, (p_16495_) -> {
-                  return p_16495_.update(opticfinder3, (p_16510_) -> {
-                     return p_16510_.stream().filter((p_16512_) -> {
-                        return !Objects.equals(p_16512_.getFirst(), "Village");
-                     }).map((p_16501_) -> {
-                        return p_16501_.mapFirst((p_16508_) -> {
-                           return p_16508_.equals("New_Village") ? "Village" : p_16508_;
+      return TypeRewriteRule.seq(this.fixTypeEverywhereTyped("NewVillageFix", type, (p_219841_4_) -> {
+         return p_219841_4_.updateTyped(opticfinder, (p_219849_3_) -> {
+            return p_219849_3_.updateTyped(opticfinder1, (p_219842_2_) -> {
+               return p_219842_2_.updateTyped(opticfinder2, (p_219850_1_) -> {
+                  return p_219850_1_.update(opticfinder3, (p_219851_0_) -> {
+                     return p_219851_0_.stream().filter((p_219854_0_) -> {
+                        return !Objects.equals(p_219854_0_.getFirst(), "Village");
+                     }).map((p_219852_0_) -> {
+                        return p_219852_0_.mapFirst((p_219847_0_) -> {
+                           return p_219847_0_.equals("New_Village") ? "Village" : p_219847_0_;
                         });
                      }).collect(Collectors.toList());
                   });
-               }).update(DSL.remainderFinder(), (p_16516_) -> {
-                  return p_16516_.update("References", (p_16518_) -> {
-                     Optional<? extends Dynamic<?>> optional = p_16518_.get("New_Village").result();
-                     return DataFixUtils.orElse(optional.map((p_16506_) -> {
-                        return p_16518_.remove("New_Village").set("Village", p_16506_);
-                     }), p_16518_).remove("Village");
+               }).update(DSL.remainderFinder(), (p_219843_0_) -> {
+                  return p_219843_0_.update("References", (p_219844_0_) -> {
+                     Optional<? extends Dynamic<?>> optional = p_219844_0_.get("New_Village").result();
+                     return DataFixUtils.orElse(optional.map((p_219846_1_) -> {
+                        return p_219844_0_.remove("New_Village").set("Village", p_219846_1_);
+                     }), p_219844_0_).remove("Village");
                   });
                });
             });
          });
-      }), this.fixTypeEverywhereTyped("NewVillageStartFix", type1, (p_16497_) -> {
-         return p_16497_.update(DSL.remainderFinder(), (p_16503_) -> {
-            return p_16503_.update("id", (p_16514_) -> {
-               return Objects.equals(NamespacedSchema.m_17311_(p_16514_.asString("")), "minecraft:new_village") ? p_16514_.createString("minecraft:village") : p_16514_;
+      }), this.fixTypeEverywhereTyped("NewVillageStartFix", type1, (p_219853_0_) -> {
+         return p_219853_0_.update(DSL.remainderFinder(), (p_219840_0_) -> {
+            return p_219840_0_.update("id", (p_219845_0_) -> {
+               return Objects.equals(NamespacedSchema.m_17311_(p_219845_0_.asString("")), "minecraft:new_village") ? p_219845_0_.createString("minecraft:village") : p_219845_0_;
             });
          });
       }));
