--- a/net/minecraft/util/datafix/fixes/OminousBannerRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/OminousBannerRenameFix.java
@@ -39,18 +_,18 @@
       Type<?> type = this.getInputSchema().getType(References.f_16782_);
       OpticFinder<Pair<String, String>> opticfinder = DSL.fieldFinder("id", DSL.named(References.f_16788_.typeName(), NamespacedSchema.m_17310_()));
       OpticFinder<?> opticfinder1 = type.findField("tag");
-      return this.fixTypeEverywhereTyped("OminousBannerRenameFix", type, (p_16602_) -> {
-         Optional<Pair<String, String>> optional = p_16602_.getOptional(opticfinder);
+      return this.fixTypeEverywhereTyped("OminousBannerRenameFix", type, (p_219819_3_) -> {
+         Optional<Pair<String, String>> optional = p_219819_3_.getOptional(opticfinder);
          if (optional.isPresent() && Objects.equals(optional.get().getSecond(), "minecraft:white_banner")) {
-            Optional<? extends Typed<?>> optional1 = p_16602_.getOptionalTyped(opticfinder1);
+            Optional<? extends Typed<?>> optional1 = p_219819_3_.getOptionalTyped(opticfinder1);
             if (optional1.isPresent()) {
                Typed<?> typed = optional1.get();
                Dynamic<?> dynamic = typed.get(DSL.remainderFinder());
-               return p_16602_.set(opticfinder1, typed.set(DSL.remainderFinder(), this.m_16603_(dynamic)));
+               return p_219819_3_.set(opticfinder1, typed.set(DSL.remainderFinder(), this.m_16603_(dynamic)));
             }
          }
 
-         return p_16602_;
+         return p_219819_3_;
       });
    }
 }
