--- a/net/minecraft/util/datafix/fixes/OptionsRenameFieldFix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsRenameFieldFix.java
@@ -19,11 +_,11 @@
    }
 
    public TypeRewriteRule makeRule() {
-      return this.fixTypeEverywhereTyped(this.f_16666_, this.getInputSchema().getType(References.f_16775_), (p_16676_) -> {
-         return p_16676_.update(DSL.remainderFinder(), (p_16678_) -> {
-            return DataFixUtils.orElse(p_16678_.get(this.f_16667_).result().map((p_16681_) -> {
-               return p_16678_.set(this.f_16668_, p_16681_).remove(this.f_16667_);
-            }), p_16678_);
+      return this.fixTypeEverywhereTyped(this.f_16666_, this.getInputSchema().getType(References.f_16775_), (p_241318_1_) -> {
+         return p_241318_1_.update(DSL.remainderFinder(), (p_241319_1_) -> {
+            return DataFixUtils.orElse(p_241319_1_.get(this.f_16667_).result().map((p_241320_2_) -> {
+               return p_241319_1_.set(this.f_16668_, p_241320_2_).remove(this.f_16667_);
+            }), p_241319_1_);
          });
       });
    }
