--- a/net/minecraft/util/datafix/fixes/OptionsLowerCaseLanguageFix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsLowerCaseLanguageFix.java
@@ -13,10 +_,10 @@
    }
 
    public TypeRewriteRule makeRule() {
-      return this.fixTypeEverywhereTyped("OptionsLowerCaseLanguageFix", this.getInputSchema().getType(References.f_16775_), (p_16662_) -> {
-         return p_16662_.update(DSL.remainderFinder(), (p_16664_) -> {
-            Optional<String> optional = p_16664_.get("lang").asString().result();
-            return optional.isPresent() ? p_16664_.set("lang", p_16664_.createString(optional.get().toLowerCase(Locale.ROOT))) : p_16664_;
+      return this.fixTypeEverywhereTyped("OptionsLowerCaseLanguageFix", this.getInputSchema().getType(References.f_16775_), (p_206281_0_) -> {
+         return p_206281_0_.update(DSL.remainderFinder(), (p_207428_0_) -> {
+            Optional<String> optional = p_207428_0_.get("lang").asString().result();
+            return optional.isPresent() ? p_207428_0_.set("lang", p_207428_0_.createString(optional.get().toLowerCase(Locale.ROOT))) : p_207428_0_;
          });
       });
    }
