--- a/net/minecraft/util/datafix/fixes/OptionsAddTextBackgroundFix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsAddTextBackgroundFix.java
@@ -12,11 +_,11 @@
    }
 
    public TypeRewriteRule makeRule() {
-      return this.fixTypeEverywhereTyped("OptionsAddTextBackgroundFix", this.getInputSchema().getType(References.f_16775_), (p_16610_) -> {
-         return p_16610_.update(DSL.remainderFinder(), (p_16612_) -> {
-            return DataFixUtils.orElse(p_16612_.get("chatOpacity").asString().map((p_16615_) -> {
-               return p_16612_.set("textBackgroundOpacity", p_16612_.createDouble(this.m_16616_(p_16615_)));
-            }).result(), p_16612_);
+      return this.fixTypeEverywhereTyped("OptionsAddTextBackgroundFix", this.getInputSchema().getType(References.f_16775_), (p_219858_1_) -> {
+         return p_219858_1_.update(DSL.remainderFinder(), (p_219855_1_) -> {
+            return DataFixUtils.orElse(p_219855_1_.get("chatOpacity").asString().map((p_219857_2_) -> {
+               return p_219855_1_.set("textBackgroundOpacity", p_219855_1_.createDouble(this.m_16616_(p_219857_2_)));
+            }).result(), p_219855_1_);
          });
       });
    }
