--- a/net/minecraft/util/datafix/fixes/PlayerUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/PlayerUUIDFix.java
@@ -11,14 +_,14 @@
    }
 
    protected TypeRewriteRule makeRule() {
-      return this.fixTypeEverywhereTyped("PlayerUUIDFix", this.getInputSchema().getType(this.f_14569_), (p_16686_) -> {
-         OpticFinder<?> opticfinder = p_16686_.getType().findField("RootVehicle");
-         return p_16686_.updateTyped(opticfinder, opticfinder.type(), (p_16688_) -> {
-            return p_16688_.update(DSL.remainderFinder(), (p_16692_) -> {
-               return m_14617_(p_16692_, "Attach", "Attach").orElse(p_16692_);
+      return this.fixTypeEverywhereTyped("PlayerUUIDFix", this.getInputSchema().getType(this.f_14569_), (p_233353_0_) -> {
+         OpticFinder<?> opticfinder = p_233353_0_.getType().findField("RootVehicle");
+         return p_233353_0_.updateTyped(opticfinder, opticfinder.type(), (p_233354_0_) -> {
+            return p_233354_0_.update(DSL.remainderFinder(), (p_233356_0_) -> {
+               return m_14617_(p_233356_0_, "Attach", "Attach").orElse(p_233356_0_);
             });
-         }).update(DSL.remainderFinder(), (p_16690_) -> {
-            return EntityUUIDFix.m_15734_(EntityUUIDFix.m_15729_(p_16690_));
+         }).update(DSL.remainderFinder(), (p_233355_0_) -> {
+            return EntityUUIDFix.m_15734_(EntityUUIDFix.m_15729_(p_233355_0_));
          });
       });
    }
