--- a/net/minecraft/util/datafix/fixes/PoiTypeRename.java
+++ b/net/minecraft/util/datafix/fixes/PoiTypeRename.java
@@ -21,20 +_,20 @@
       if (!Objects.equals(type, this.getInputSchema().getType(References.f_16780_))) {
          throw new IllegalStateException("Poi type is not what was expected.");
       } else {
-         return this.fixTypeEverywhere("POI rename", type, (p_16705_) -> {
-            return (p_16708_) -> {
-               return p_16708_.mapSecond(this::m_16699_);
+         return this.fixTypeEverywhere("POI rename", type, (p_226203_1_) -> {
+            return (p_226206_1_) -> {
+               return p_226206_1_.mapSecond(this::m_16699_);
             };
          });
       }
    }
 
    private <T> Dynamic<T> m_16699_(Dynamic<T> p_16700_) {
-      return p_16700_.update("Sections", (p_16716_) -> {
-         return p_16716_.updateMapValues((p_16698_) -> {
-            return p_16698_.mapSecond((p_16718_) -> {
-               return p_16718_.update("Records", (p_16720_) -> {
-                  return DataFixUtils.orElse(this.m_16709_(p_16720_), p_16720_);
+      return p_16700_.update("Sections", (p_226209_1_) -> {
+         return p_226209_1_.updateMapValues((p_226204_1_) -> {
+            return p_226204_1_.mapSecond((p_226210_1_) -> {
+               return p_226210_1_.update("Records", (p_226211_1_) -> {
+                  return DataFixUtils.orElse(this.m_16709_(p_226211_1_), p_226211_1_);
                });
             });
          });
@@ -42,10 +_,10 @@
    }
 
    private <T> Optional<Dynamic<T>> m_16709_(Dynamic<T> p_16710_) {
-      return p_16710_.asStreamOpt().map((p_16703_) -> {
-         return p_16710_.createList(p_16703_.map((p_16712_) -> {
-            return p_16712_.update("type", (p_16714_) -> {
-               return DataFixUtils.orElse(p_16714_.asString().map(this::m_7323_).map(p_16714_::createString).result(), p_16714_);
+      return p_16710_.asStreamOpt().map((p_226202_2_) -> {
+         return p_16710_.createList(p_226202_2_.map((p_226207_1_) -> {
+            return p_226207_1_.update("type", (p_226208_1_) -> {
+               return DataFixUtils.orElse(p_226208_1_.asString().map(this::m_7323_).map(p_226208_1_::createString).result(), p_226208_1_);
             });
          }));
       }).result();
