--- a/net/minecraft/util/datafix/fixes/RedstoneWireConnectionsFix.java
+++ b/net/minecraft/util/datafix/fixes/RedstoneWireConnectionsFix.java
@@ -13,32 +_,32 @@
 
    protected TypeRewriteRule makeRule() {
       Schema schema = this.getInputSchema();
-      return this.fixTypeEverywhereTyped("RedstoneConnectionsFix", schema.getType(References.f_16783_), (p_16751_) -> {
-         return p_16751_.update(DSL.remainderFinder(), this::m_16752_);
+      return this.fixTypeEverywhereTyped("RedstoneConnectionsFix", schema.getType(References.f_16783_), (p_233367_1_) -> {
+         return p_233367_1_.update(DSL.remainderFinder(), this::m_16752_);
       });
    }
 
    private <T> Dynamic<T> m_16752_(Dynamic<T> p_16753_) {
       boolean flag = p_16753_.get("Name").asString().result().filter("minecraft:redstone_wire"::equals).isPresent();
-      return !flag ? p_16753_ : p_16753_.update("Properties", (p_16760_) -> {
-         String s = p_16760_.get("east").asString("none");
-         String s1 = p_16760_.get("west").asString("none");
-         String s2 = p_16760_.get("north").asString("none");
-         String s3 = p_16760_.get("south").asString("none");
+      return !flag ? p_16753_ : p_16753_.update("Properties", (p_233371_0_) -> {
+         String s = p_233371_0_.get("east").asString("none");
+         String s1 = p_233371_0_.get("west").asString("none");
+         String s2 = p_233371_0_.get("north").asString("none");
+         String s3 = p_233371_0_.get("south").asString("none");
          boolean flag1 = m_16754_(s) || m_16754_(s1);
          boolean flag2 = m_16754_(s2) || m_16754_(s3);
          String s4 = !m_16754_(s) && !flag2 ? "side" : s;
          String s5 = !m_16754_(s1) && !flag2 ? "side" : s1;
          String s6 = !m_16754_(s2) && !flag1 ? "side" : s2;
          String s7 = !m_16754_(s3) && !flag1 ? "side" : s3;
-         return p_16760_.update("east", (p_16769_) -> {
-            return p_16769_.createString(s4);
-         }).update("west", (p_16766_) -> {
-            return p_16766_.createString(s5);
-         }).update("north", (p_16763_) -> {
-            return p_16763_.createString(s6);
-         }).update("south", (p_16758_) -> {
-            return p_16758_.createString(s7);
+         return p_233371_0_.update("east", (p_233374_1_) -> {
+            return p_233374_1_.createString(s4);
+         }).update("west", (p_233373_1_) -> {
+            return p_233373_1_.createString(s5);
+         }).update("north", (p_233372_1_) -> {
+            return p_233372_1_.createString(s6);
+         }).update("south", (p_233370_1_) -> {
+            return p_233370_1_.createString(s7);
          });
       });
    }
