--- a/net/minecraft/util/datafix/fixes/RemoveGolemGossipFix.java
+++ b/net/minecraft/util/datafix/fixes/RemoveGolemGossipFix.java
@@ -15,9 +_,9 @@
    }
 
    private static Dynamic<?> m_16827_(Dynamic<?> p_16828_) {
-      return p_16828_.update("Gossips", (p_16831_) -> {
-         return p_16828_.createList(p_16831_.asStream().filter((p_16833_) -> {
-            return !p_16833_.get("Type").asString("").equals("golem");
+      return p_16828_.update("Gossips", (p_242267_1_) -> {
+         return p_16828_.createList(p_242267_1_.asStream().filter((p_242268_0_) -> {
+            return !p_242268_0_.get("Type").asString("").equals("golem");
          }));
       });
    }
