--- a/net/minecraft/util/datafix/fixes/RenameBiomesFix.java
+++ b/net/minecraft/util/datafix/fixes/RenameBiomesFix.java
@@ -25,10 +_,10 @@
       if (!Objects.equals(type, this.getInputSchema().getType(References.f_16794_))) {
          throw new IllegalStateException("Biome type is not what was expected.");
       } else {
-         return this.fixTypeEverywhere(this.f_16834_, type, (p_16844_) -> {
-            return (p_16842_) -> {
-               return p_16842_.mapSecond((p_16846_) -> {
-                  return this.f_16835_.getOrDefault(p_16846_, p_16846_);
+         return this.fixTypeEverywhere(this.f_16834_, type, (p_233380_1_) -> {
+            return (p_233379_1_) -> {
+               return p_233379_1_.mapSecond((p_233381_1_) -> {
+                  return this.f_16835_.getOrDefault(p_233381_1_, p_233381_1_);
                });
             };
          });
