--- a/net/minecraft/util/datafix/fixes/ReorganizePoi.java
+++ b/net/minecraft/util/datafix/fixes/ReorganizePoi.java
@@ -23,9 +_,9 @@
       if (!Objects.equals(type, this.getInputSchema().getType(References.f_16780_))) {
          throw new IllegalStateException("Poi type is not what was expected.");
       } else {
-         return this.fixTypeEverywhere("POI reorganization", type, (p_16860_) -> {
-            return (p_16856_) -> {
-               return p_16856_.mapSecond(ReorganizePoi::m_16857_);
+         return this.fixTypeEverywhere("POI reorganization", type, (p_219871_0_) -> {
+            return (p_219872_0_) -> {
+               return p_219872_0_.mapSecond(ReorganizePoi::m_16857_);
             };
          });
       }
