--- a/net/minecraft/util/datafix/fixes/SavedDataVillageCropFix.java
+++ b/net/minecraft/util/datafix/fixes/SavedDataVillageCropFix.java
@@ -24,12 +_,12 @@
    }
 
    private static Stream<? extends Dynamic<?>> m_16889_(Stream<? extends Dynamic<?>> p_16890_) {
-      return p_16890_.map((Dynamic<?> p_16898_) -> {
-         String s = p_16898_.get("id").asString("");
+      return p_16890_.map((Dynamic<?> p_210587_0_) -> {
+         String s = p_210587_0_.get("id").asString("");
          if ("ViF".equals(s)) {
-            return m_16893_(p_16898_);
+            return m_16893_(p_210587_0_);
          } else {
-            return "ViDF".equals(s) ? m_16895_(p_16898_) : p_16898_;
+            return "ViDF".equals(s) ? m_16895_(p_210587_0_) : p_210587_0_;
          }
       });
    }
