--- a/net/minecraft/util/datafix/fixes/SavedDataUUIDFix.java
+++ b/net/minecraft/util/datafix/fixes/SavedDataUUIDFix.java
@@ -10,16 +_,16 @@
    }
 
    protected TypeRewriteRule makeRule() {
-      return this.fixTypeEverywhereTyped("SavedDataUUIDFix", this.getInputSchema().getType(this.f_14569_), (p_16865_) -> {
-         return p_16865_.updateTyped(p_16865_.getType().findField("data"), (p_16867_) -> {
-            return p_16867_.update(DSL.remainderFinder(), (p_16869_) -> {
-               return p_16869_.update("Raids", (p_16871_) -> {
-                  return p_16871_.createList(p_16871_.asStream().map((p_16873_) -> {
-                     return p_16873_.update("HeroesOfTheVillage", (p_16875_) -> {
-                        return p_16875_.createList(p_16875_.asStream().map((p_16877_) -> {
-                           return m_14621_(p_16877_, "UUIDMost", "UUIDLeast").orElseGet(() -> {
+      return this.fixTypeEverywhereTyped("SavedDataUUIDFix", this.getInputSchema().getType(this.f_14569_), (p_233386_0_) -> {
+         return p_233386_0_.updateTyped(p_233386_0_.getType().findField("data"), (p_233387_0_) -> {
+            return p_233387_0_.update(DSL.remainderFinder(), (p_233388_0_) -> {
+               return p_233388_0_.update("Raids", (p_233389_0_) -> {
+                  return p_233389_0_.createList(p_233389_0_.asStream().map((p_233390_0_) -> {
+                     return p_233390_0_.update("HeroesOfTheVillage", (p_233391_0_) -> {
+                        return p_233391_0_.createList(p_233391_0_.asStream().map((p_233392_0_) -> {
+                           return m_14621_(p_233392_0_, "UUIDMost", "UUIDLeast").orElseGet(() -> {
                               f_14568_.warn("HeroesOfTheVillage contained invalid UUIDs.");
-                              return p_16877_;
+                              return p_233392_0_;
                            });
                         }));
                      });
