--- a/net/minecraft/util/datafix/fixes/StructureReferenceCountFix.java
+++ b/net/minecraft/util/datafix/fixes/StructureReferenceCountFix.java
@@ -14,15 +_,15 @@
 
    protected TypeRewriteRule makeRule() {
       Type<?> type = this.getInputSchema().getType(References.f_16790_);
-      return this.fixTypeEverywhereTyped("Structure Reference Fix", type, (p_16964_) -> {
-         return p_16964_.update(DSL.remainderFinder(), StructureReferenceCountFix::m_16965_);
+      return this.fixTypeEverywhereTyped("Structure Reference Fix", type, (p_226213_0_) -> {
+         return p_226213_0_.update(DSL.remainderFinder(), StructureReferenceCountFix::m_16965_);
       });
    }
 
    private static <T> Dynamic<T> m_16965_(Dynamic<T> p_16966_) {
-      return p_16966_.update("references", (p_16970_) -> {
-         return p_16970_.createInt(p_16970_.asNumber().map(Number::intValue).result().filter((p_16968_) -> {
-            return p_16968_ > 0;
+      return p_16966_.update("references", (p_226215_0_) -> {
+         return p_226215_0_.createInt(p_226215_0_.asNumber().map(Number::intValue).result().filter((p_226214_0_) -> {
+            return p_226214_0_ > 0;
          }).orElse(1));
       });
    }
