--- a/net/minecraft/util/datafix/fixes/VillagerRebuildLevelAndXpFix.java
+++ b/net/minecraft/util/datafix/fixes/VillagerRebuildLevelAndXpFix.java
@@ -31,20 +_,20 @@
       OpticFinder<?> opticfinder2 = type1.findField("Recipes");
       ListType<?> listtype = (ListType)opticfinder2.type();
       OpticFinder<?> opticfinder3 = listtype.getElement().finder();
-      return this.fixTypeEverywhereTyped("Villager level and xp rebuild", this.getInputSchema().getType(References.f_16786_), (p_17098_) -> {
-         return p_17098_.updateTyped(opticfinder, type, (p_17088_) -> {
-            Dynamic<?> dynamic = p_17088_.get(DSL.remainderFinder());
+      return this.fixTypeEverywhereTyped("Villager level and xp rebuild", this.getInputSchema().getType(References.f_16786_), (p_222996_5_) -> {
+         return p_222996_5_.updateTyped(opticfinder, type, (p_222995_3_) -> {
+            Dynamic<?> dynamic = p_222995_3_.get(DSL.remainderFinder());
             int i = dynamic.get("VillagerData").get("level").asInt(0);
-            Typed<?> typed = p_17088_;
+            Typed<?> typed = p_222995_3_;
             if (i == 0 || i == 1) {
-               int j = p_17088_.getOptionalTyped(opticfinder1).flatMap((p_17107_) -> {
-                  return p_17107_.getOptionalTyped(opticfinder2);
-               }).map((p_17091_) -> {
-                  return p_17091_.getAllTyped(opticfinder3).size();
+               int j = p_222995_3_.getOptionalTyped(opticfinder1).flatMap((p_223002_1_) -> {
+                  return p_223002_1_.getOptionalTyped(opticfinder2);
+               }).map((p_222997_1_) -> {
+                  return p_222997_1_.getAllTyped(opticfinder3).size();
                }).orElse(0);
                i = Mth.m_14045_(j / 2, 1, 5);
                if (i > 1) {
-                  typed = m_17099_(p_17088_, i);
+                  typed = m_17099_(p_222995_3_, i);
                }
             }
 
@@ -59,17 +_,17 @@
    }
 
    private static Typed<?> m_17099_(Typed<?> p_17100_, int p_17101_) {
-      return p_17100_.update(DSL.remainderFinder(), (p_17104_) -> {
-         return p_17104_.update("VillagerData", (p_17113_) -> {
-            return p_17113_.set("level", p_17113_.createInt(p_17101_));
+      return p_17100_.update(DSL.remainderFinder(), (p_222998_1_) -> {
+         return p_222998_1_.update("VillagerData", (p_222999_1_) -> {
+            return p_222999_1_.set("level", p_222999_1_.createInt(p_17101_));
          });
       });
    }
 
    private static Typed<?> m_17108_(Typed<?> p_17109_, int p_17110_) {
       int i = m_17079_(p_17110_);
-      return p_17109_.update(DSL.remainderFinder(), (p_17083_) -> {
-         return p_17083_.set("Xp", p_17083_.createInt(i));
+      return p_17109_.update(DSL.remainderFinder(), (p_223000_1_) -> {
+         return p_223000_1_.set("Xp", p_223000_1_.createInt(i));
       });
    }
 }
