--- a/com/mojang/blaze3d/audio/Channel.java
+++ b/com/mojang/blaze3d/audio/Channel.java
@@ -120,8 +_,8 @@
    }
 
    public void m_83656_(SoundBuffer p_83657_) {
-      p_83657_.m_83800_().ifPresent((p_83676_) -> {
-         AL10.alSourcei(this.f_83642_, 4105, p_83676_);
+      p_83657_.m_83800_().ifPresent((p_216431_1_) -> {
+         AL10.alSourcei(this.f_83642_, 4105, p_216431_1_);
       });
    }
 
@@ -142,8 +_,8 @@
             for(int i = 0; i < p_83653_; ++i) {
                ByteBuffer bytebuffer = this.f_83645_.m_7118_(this.f_83644_);
                if (bytebuffer != null) {
-                  (new SoundBuffer(bytebuffer, this.f_83645_.m_6206_())).m_83802_().ifPresent((p_83669_) -> {
-                     AL10.alSourceQueueBuffers(this.f_83642_, new int[]{p_83669_});
+                  (new SoundBuffer(bytebuffer, this.f_83645_.m_6206_())).m_83802_().ifPresent((p_216424_1_) -> {
+                     AL10.alSourceQueueBuffers(this.f_83642_, new int[]{p_216424_1_});
                   });
                }
             }
