--- a/com/mojang/blaze3d/audio/Library.java
+++ b/com/mojang/blaze3d/audio/Library.java
@@ -157,20 +_,6 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   interface ChannelPool {
-      @Nullable
-      Channel m_5574_();
-
-      boolean m_5658_(Channel p_83712_);
-
-      void m_6471_();
-
-      int m_6492_();
-
-      int m_6500_();
-   }
-
-   @OnlyIn(Dist.CLIENT)
    static class CountingChannelPool implements Library.ChannelPool {
       private final int f_83713_;
       private final Set<Channel> f_83714_ = Sets.newIdentityHashSet();
@@ -215,6 +_,20 @@
       public int m_6500_() {
          return this.f_83714_.size();
       }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   interface ChannelPool {
+      @Nullable
+      Channel m_5574_();
+
+      boolean m_5658_(Channel p_83712_);
+
+      void m_6471_();
+
+      int m_6492_();
+
+      int m_6500_();
    }
 
    @OnlyIn(Dist.CLIENT)
