--- a/net/minecraft/world/damagesource/DamageSource.java
+++ b/net/minecraft/world/damagesource/DamageSource.java
@@ -144,7 +_,7 @@
       return this.f_19329_;
    }
 
-   protected DamageSource(String p_19333_) {
+   public DamageSource(String p_19333_) {
       this.f_19326_ = p_19333_;
    }
 
@@ -158,24 +_,24 @@
       return null;
    }
 
-   protected DamageSource m_19380_() {
+   public DamageSource m_19380_() {
       this.f_19327_ = true;
       this.f_19330_ = 0.0F;
       return this;
    }
 
-   protected DamageSource m_19381_() {
+   public DamageSource m_19381_() {
       this.f_19328_ = true;
       return this;
    }
 
-   protected DamageSource m_19382_() {
+   public DamageSource m_19382_() {
       this.f_19329_ = true;
       this.f_19330_ = 0.0F;
       return this;
    }
 
-   protected DamageSource m_19383_() {
+   public DamageSource m_19383_() {
       this.f_19300_ = true;
       return this;
    }
