--- a/net/minecraft/world/entity/ai/sensing/AdultSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/AdultSensor.java
@@ -15,18 +_,18 @@
    }
 
    protected void m_5578_(ServerLevel p_26617_, AgableMob p_26618_) {
-      p_26618_.m_6274_().m_21952_(MemoryModuleType.f_26365_).ifPresent((p_26634_) -> {
-         this.m_26627_(p_26618_, p_26634_);
+      p_26618_.m_6274_().m_21952_(MemoryModuleType.f_26365_).ifPresent((p_234118_2_) -> {
+         this.m_26627_(p_26618_, p_234118_2_);
       });
    }
 
    private void m_26627_(AgableMob p_26628_, List<LivingEntity> p_26629_) {
-      Optional<AgableMob> optional = p_26629_.stream().filter((p_26626_) -> {
-         return p_26626_.m_6095_() == p_26628_.m_6095_();
-      }).map((p_26631_) -> {
-         return (AgableMob)p_26631_;
-      }).filter((p_26623_) -> {
-         return !p_26623_.m_6162_();
+      Optional<AgableMob> optional = p_26629_.stream().filter((p_234115_1_) -> {
+         return p_234115_1_.m_6095_() == p_26628_.m_6095_();
+      }).map((p_234117_0_) -> {
+         return (AgableMob)p_234117_0_;
+      }).filter((p_234114_0_) -> {
+         return !p_234114_0_.m_6162_();
       }).findFirst();
       p_26628_.m_6274_().m_21886_(MemoryModuleType.f_26331_, optional);
    }
