--- a/net/minecraft/world/entity/ai/sensing/HurtBySensor.java
+++ b/net/minecraft/world/entity/ai/sensing/HurtBySensor.java
@@ -27,8 +_,8 @@
          brain.m_21936_(MemoryModuleType.f_26381_);
       }
 
-      brain.m_21952_(MemoryModuleType.f_26382_).ifPresent((p_26675_) -> {
-         if (!p_26675_.m_6084_() || p_26675_.f_19853_ != p_26670_) {
+      brain.m_21952_(MemoryModuleType.f_26382_).ifPresent((p_234121_2_) -> {
+         if (!p_234121_2_.m_6084_() || p_234121_2_.f_19853_ != p_26670_) {
             brain.m_21936_(MemoryModuleType.f_26382_);
          }
 
