--- a/net/minecraft/world/entity/ai/sensing/PiglinSpecificSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/PiglinSpecificSensor.java
@@ -102,8 +_,8 @@
    }
 
    private static Optional<BlockPos> m_26734_(ServerLevel p_26735_, LivingEntity p_26736_) {
-      return BlockPos.m_121930_(p_26736_.m_20183_(), 8, 4, (p_26733_) -> {
-         return m_26728_(p_26735_, p_26733_);
+      return BlockPos.m_121930_(p_26736_.m_20183_(), 8, 4, (p_234125_1_) -> {
+         return m_26728_(p_26735_, p_234125_1_);
       });
    }
 
