--- a/net/minecraft/world/entity/ai/sensing/VillagerHostilesSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/VillagerHostilesSensor.java
@@ -23,11 +_,11 @@
    }
 
    private Optional<LivingEntity> m_26849_(LivingEntity p_26850_) {
-      return this.m_26858_(p_26850_).flatMap((p_26857_) -> {
-         return p_26857_.stream().filter(this::m_26867_).filter((p_26871_) -> {
-            return this.m_26860_(p_26850_, p_26871_);
-         }).min((p_26865_, p_26866_) -> {
-            return this.m_26851_(p_26850_, p_26865_, p_26866_);
+      return this.m_26858_(p_26850_).flatMap((p_220984_2_) -> {
+         return p_220984_2_.stream().filter(this::m_26867_).filter((p_220985_2_) -> {
+            return this.m_26860_(p_26850_, p_220985_2_);
+         }).min((p_220986_2_, p_220986_3_) -> {
+            return this.m_26851_(p_26850_, p_220986_2_, p_220986_3_);
          });
       });
    }
