--- a/net/minecraft/world/effect/MobEffectCategory.java
+++ b/net/minecraft/world/effect/MobEffectCategory.java
@@ -1,8 +_,6 @@
 package net.minecraft.world.effect;
 
 import net.minecraft.ChatFormatting;
-import net.minecraftforge.api.distmarker.Dist;
-import net.minecraftforge.api.distmarker.OnlyIn;
 
 public enum MobEffectCategory {
    BENEFICIAL(ChatFormatting.BLUE),
@@ -15,7 +_,6 @@
       this.f_19490_ = p_19496_;
    }
 
-   @OnlyIn(Dist.CLIENT)
    public ChatFormatting m_19497_() {
       return this.f_19490_;
    }
