--- a/net/minecraft/world/scores/Objective.java
+++ b/net/minecraft/world/scores/Objective.java
@@ -43,8 +_,8 @@
    }
 
    private Component m_83325_() {
-      return ComponentUtils.m_130748_(this.f_83304_.m_6881_().m_130938_((p_83319_) -> {
-         return p_83319_.m_131144_(new HoverEvent(HoverEvent.Action.f_130831_, new TextComponent(this.f_83302_)));
+      return ComponentUtils.m_130748_(this.f_83304_.m_6881_().m_130938_((p_237497_1_) -> {
+         return p_237497_1_.m_131144_(new HoverEvent(HoverEvent.Action.f_130831_, new TextComponent(this.f_83302_)));
       }));
    }
 
