--- a/net/minecraft/world/scores/Team.java
+++ b/net/minecraft/world/scores/Team.java
@@ -48,10 +_,10 @@
       PUSH_OTHER_TEAMS("pushOtherTeams", 2),
       PUSH_OWN_TEAM("pushOwnTeam", 3);
 
-      private static final Map<String, Team.CollisionRule> f_83545_ = Arrays.stream(values()).collect(Collectors.toMap((p_83559_) -> {
-         return p_83559_.f_83543_;
-      }, (p_83554_) -> {
-         return p_83554_;
+      private static final Map<String, Team.CollisionRule> f_83545_ = Arrays.stream(values()).collect(Collectors.toMap((p_199871_0_) -> {
+         return p_199871_0_.f_83543_;
+      }, (p_199870_0_) -> {
+         return p_199870_0_;
       }));
       public final String f_83543_;
       public final int f_83544_;
@@ -77,10 +_,10 @@
       HIDE_FOR_OTHER_TEAMS("hideForOtherTeams", 2),
       HIDE_FOR_OWN_TEAM("hideForOwnTeam", 3);
 
-      private static final Map<String, Team.Visibility> f_83569_ = Arrays.stream(values()).collect(Collectors.toMap((p_83583_) -> {
-         return p_83583_.f_83567_;
-      }, (p_83578_) -> {
-         return p_83578_;
+      private static final Map<String, Team.Visibility> f_83569_ = Arrays.stream(values()).collect(Collectors.toMap((p_199873_0_) -> {
+         return p_199873_0_.f_83567_;
+      }, (p_199872_0_) -> {
+         return p_199872_0_;
       }));
       public final String f_83567_;
       public final int f_83568_;
