--- a/net/minecraft/world/level/pathfinder/NodeEvaluator.java
+++ b/net/minecraft/world/level/pathfinder/NodeEvaluator.java
@@ -38,7 +_,7 @@
    }
 
    protected Node m_5676_(int p_77325_, int p_77326_, int p_77327_) {
-      return this.f_77314_.computeIfAbsent(Node.m_77295_(p_77325_, p_77326_, p_77327_), (p_77332_) -> {
+      return this.f_77314_.computeIfAbsent(Node.m_77295_(p_77325_, p_77326_, p_77327_), (p_215743_3_) -> {
          return new Node(p_77325_, p_77326_, p_77327_);
       });
    }
