--- a/net/minecraft/world/level/pathfinder/PathFinder.java
+++ b/net/minecraft/world/level/pathfinder/PathFinder.java
@@ -31,8 +_,8 @@
       this.f_77423_.m_77081_();
       this.f_77422_.m_6028_(p_77428_, p_77429_);
       Node node = this.f_77422_.m_7171_();
-      Map<Target, BlockPos> map = p_77430_.stream().collect(Collectors.toMap((p_77448_) -> {
-         return this.f_77422_.m_7568_((double)p_77448_.m_123341_(), (double)p_77448_.m_123342_(), (double)p_77448_.m_123343_());
+      Map<Target, BlockPos> map = p_77430_.stream().collect(Collectors.toMap((p_224782_1_) -> {
+         return this.f_77422_.m_7568_((double)p_224782_1_.m_123341_(), (double)p_224782_1_.m_123342_(), (double)p_224782_1_.m_123343_());
       }, Function.identity()));
       Path path = this.m_77438_(node, map, p_77431_, p_77432_, p_77433_);
       this.f_77422_.m_6802_();
@@ -95,10 +_,10 @@
          }
       }
 
-      Optional<Path> optional = !set2.isEmpty() ? set2.stream().map((p_77454_) -> {
-         return this.m_77434_(p_77454_.m_77508_(), p_77440_.get(p_77454_), true);
-      }).min(Comparator.comparingInt(Path::m_77398_)) : set.stream().map((p_77451_) -> {
-         return this.m_77434_(p_77451_.m_77508_(), p_77440_.get(p_77451_), false);
+      Optional<Path> optional = !set2.isEmpty() ? set2.stream().map((p_224778_2_) -> {
+         return this.m_77434_(p_224778_2_.m_77508_(), p_77440_.get(p_224778_2_), true);
+      }).min(Comparator.comparingInt(Path::m_77398_)) : set.stream().map((p_224777_2_) -> {
+         return this.m_77434_(p_224777_2_.m_77508_(), p_77440_.get(p_224777_2_), false);
       }).min(Comparator.comparingDouble(Path::m_77407_).thenComparingInt(Path::m_77398_));
       return !optional.isPresent() ? null : optional.get();
    }
