--- a/net/minecraft/commands/arguments/selector/EntitySelector.java
+++ b/net/minecraft/commands/arguments/selector/EntitySelector.java
@@ -188,14 +_,14 @@
       Predicate<Entity> predicate = this.f_121114_;
       if (this.f_121117_ != null) {
          AABB aabb = this.f_121117_.m_82383_(p_121145_);
-         predicate = predicate.and((p_121143_) -> {
-            return aabb.m_82381_(p_121143_.m_20191_());
+         predicate = predicate.and((p_197344_1_) -> {
+            return aabb.m_82381_(p_197344_1_.m_20191_());
          });
       }
 
       if (!this.f_121115_.m_55327_()) {
-         predicate = predicate.and((p_121148_) -> {
-            return this.f_121115_.m_55343_(p_121148_.m_20238_(p_121145_));
+         predicate = predicate.and((p_211376_2_) -> {
+            return this.f_121115_.m_55343_(p_211376_2_.m_20238_(p_121145_));
          });
       }
 
