--- a/net/minecraft/server/commands/BanListCommands.java
+++ b/net/minecraft/server/commands/BanListCommands.java
@@ -12,15 +_,15 @@
 
 public class BanListCommands {
    public static void m_136543_(CommandDispatcher<CommandSourceStack> p_136544_) {
-      p_136544_.register(Commands.m_82127_("banlist").requires((p_136548_) -> {
-         return p_136548_.m_6761_(3);
-      }).executes((p_136555_) -> {
-         PlayerList playerlist = p_136555_.getSource().m_81377_().m_6846_();
-         return m_136549_(p_136555_.getSource(), Lists.newArrayList(Iterables.concat(playerlist.m_11295_().m_11395_(), playerlist.m_11299_().m_11395_())));
-      }).then(Commands.m_82127_("ips").executes((p_136553_) -> {
-         return m_136549_(p_136553_.getSource(), p_136553_.getSource().m_81377_().m_6846_().m_11299_().m_11395_());
-      })).then(Commands.m_82127_("players").executes((p_136546_) -> {
-         return m_136549_(p_136546_.getSource(), p_136546_.getSource().m_81377_().m_6846_().m_11295_().m_11395_());
+      p_136544_.register(Commands.m_82127_("banlist").requires((p_198233_0_) -> {
+         return p_198233_0_.m_6761_(3);
+      }).executes((p_198231_0_) -> {
+         PlayerList playerlist = p_198231_0_.getSource().m_81377_().m_6846_();
+         return m_136549_(p_198231_0_.getSource(), Lists.newArrayList(Iterables.concat(playerlist.m_11295_().m_11395_(), playerlist.m_11299_().m_11395_())));
+      }).then(Commands.m_82127_("ips").executes((p_198228_0_) -> {
+         return m_136549_(p_198228_0_.getSource(), p_198228_0_.getSource().m_81377_().m_6846_().m_11299_().m_11395_());
+      })).then(Commands.m_82127_("players").executes((p_198232_0_) -> {
+         return m_136549_(p_198232_0_.getSource(), p_198232_0_.getSource().m_81377_().m_6846_().m_11295_().m_11395_());
       })));
    }
 
