--- a/net/minecraft/server/commands/BanPlayerCommands.java
+++ b/net/minecraft/server/commands/BanPlayerCommands.java
@@ -22,12 +_,12 @@
    private static final SimpleCommandExceptionType f_136556_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.ban.failed"));
 
    public static void m_136558_(CommandDispatcher<CommandSourceStack> p_136559_) {
-      p_136559_.register(Commands.m_82127_("ban").requires((p_136563_) -> {
-         return p_136563_.m_6761_(3);
-      }).then(Commands.m_82129_("targets", GameProfileArgument.m_94584_()).executes((p_136569_) -> {
-         return m_136564_(p_136569_.getSource(), GameProfileArgument.m_94590_(p_136569_, "targets"), (Component)null);
-      }).then(Commands.m_82129_("reason", MessageArgument.m_96832_()).executes((p_136561_) -> {
-         return m_136564_(p_136561_.getSource(), GameProfileArgument.m_94590_(p_136561_, "targets"), MessageArgument.m_96835_(p_136561_, "reason"));
+      p_136559_.register(Commands.m_82127_("ban").requires((p_198238_0_) -> {
+         return p_198238_0_.m_6761_(3);
+      }).then(Commands.m_82129_("targets", GameProfileArgument.m_94584_()).executes((p_198234_0_) -> {
+         return m_136564_(p_198234_0_.getSource(), GameProfileArgument.m_94590_(p_198234_0_, "targets"), (Component)null);
+      }).then(Commands.m_82129_("reason", MessageArgument.m_96832_()).executes((p_198237_0_) -> {
+         return m_136564_(p_198237_0_.getSource(), GameProfileArgument.m_94590_(p_198237_0_, "targets"), MessageArgument.m_96835_(p_198237_0_, "reason"));
       }))));
    }
 
