--- a/net/minecraft/server/commands/BossBarCommands.java
+++ b/net/minecraft/server/commands/BossBarCommands.java
@@ -28,11 +_,11 @@
 import net.minecraft.world.entity.player.Player;
 
 public class BossBarCommands {
-   private static final DynamicCommandExceptionType f_136571_ = new DynamicCommandExceptionType((p_136636_) -> {
-      return new TranslatableComponent("commands.bossbar.create.failed", p_136636_);
+   private static final DynamicCommandExceptionType f_136571_ = new DynamicCommandExceptionType((p_208783_0_) -> {
+      return new TranslatableComponent("commands.bossbar.create.failed", p_208783_0_);
    });
-   private static final DynamicCommandExceptionType f_136572_ = new DynamicCommandExceptionType((p_136623_) -> {
-      return new TranslatableComponent("commands.bossbar.unknown", p_136623_);
+   private static final DynamicCommandExceptionType f_136572_ = new DynamicCommandExceptionType((p_208782_0_) -> {
+      return new TranslatableComponent("commands.bossbar.unknown", p_208782_0_);
    });
    private static final SimpleCommandExceptionType f_136573_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.bossbar.set.players.unchanged"));
    private static final SimpleCommandExceptionType f_136574_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.bossbar.set.name.unchanged"));
@@ -42,63 +_,63 @@
    private static final SimpleCommandExceptionType f_136578_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.bossbar.set.max.unchanged"));
    private static final SimpleCommandExceptionType f_136579_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.bossbar.set.visibility.unchanged.hidden"));
    private static final SimpleCommandExceptionType f_136580_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.bossbar.set.visibility.unchanged.visible"));
-   public static final SuggestionProvider<CommandSourceStack> f_136570_ = (p_136587_, p_136588_) -> {
-      return SharedSuggestionProvider.m_82926_(p_136587_.getSource().m_81377_().m_129901_().m_136292_(), p_136588_);
+   public static final SuggestionProvider<CommandSourceStack> f_136570_ = (p_201404_0_, p_201404_1_) -> {
+      return SharedSuggestionProvider.m_82926_(p_201404_0_.getSource().m_81377_().m_129901_().m_136292_(), p_201404_1_);
    };
 
    public static void m_136582_(CommandDispatcher<CommandSourceStack> p_136583_) {
-      p_136583_.register(Commands.m_82127_("bossbar").requires((p_136627_) -> {
-         return p_136627_.m_6761_(2);
-      }).then(Commands.m_82127_("add").then(Commands.m_82129_("id", ResourceLocationArgument.m_106984_()).then(Commands.m_82129_("name", ComponentArgument.m_87114_()).executes((p_136693_) -> {
-         return m_136591_(p_136693_.getSource(), ResourceLocationArgument.m_107011_(p_136693_, "id"), ComponentArgument.m_87117_(p_136693_, "name"));
-      })))).then(Commands.m_82127_("remove").then(Commands.m_82129_("id", ResourceLocationArgument.m_106984_()).suggests(f_136570_).executes((p_136691_) -> {
-         return m_136649_(p_136691_.getSource(), m_136584_(p_136691_));
-      }))).then(Commands.m_82127_("list").executes((p_136689_) -> {
-         return m_136589_(p_136689_.getSource());
-      })).then(Commands.m_82127_("set").then(Commands.m_82129_("id", ResourceLocationArgument.m_106984_()).suggests(f_136570_).then(Commands.m_82127_("name").then(Commands.m_82129_("name", ComponentArgument.m_87114_()).executes((p_136687_) -> {
-         return m_136614_(p_136687_.getSource(), m_136584_(p_136687_), ComponentArgument.m_87117_(p_136687_, "name"));
-      }))).then(Commands.m_82127_("color").then(Commands.m_82127_("pink").executes((p_136685_) -> {
-         return m_136602_(p_136685_.getSource(), m_136584_(p_136685_), BossEvent.BossBarColor.PINK);
-      })).then(Commands.m_82127_("blue").executes((p_136683_) -> {
-         return m_136602_(p_136683_.getSource(), m_136584_(p_136683_), BossEvent.BossBarColor.BLUE);
-      })).then(Commands.m_82127_("red").executes((p_136681_) -> {
-         return m_136602_(p_136681_.getSource(), m_136584_(p_136681_), BossEvent.BossBarColor.RED);
-      })).then(Commands.m_82127_("green").executes((p_136679_) -> {
-         return m_136602_(p_136679_.getSource(), m_136584_(p_136679_), BossEvent.BossBarColor.GREEN);
-      })).then(Commands.m_82127_("yellow").executes((p_136677_) -> {
-         return m_136602_(p_136677_.getSource(), m_136584_(p_136677_), BossEvent.BossBarColor.YELLOW);
-      })).then(Commands.m_82127_("purple").executes((p_136675_) -> {
-         return m_136602_(p_136675_.getSource(), m_136584_(p_136675_), BossEvent.BossBarColor.PURPLE);
-      })).then(Commands.m_82127_("white").executes((p_136673_) -> {
-         return m_136602_(p_136673_.getSource(), m_136584_(p_136673_), BossEvent.BossBarColor.WHITE);
-      }))).then(Commands.m_82127_("style").then(Commands.m_82127_("progress").executes((p_136671_) -> {
-         return m_136606_(p_136671_.getSource(), m_136584_(p_136671_), BossEvent.BossBarOverlay.PROGRESS);
-      })).then(Commands.m_82127_("notched_6").executes((p_136669_) -> {
-         return m_136606_(p_136669_.getSource(), m_136584_(p_136669_), BossEvent.BossBarOverlay.NOTCHED_6);
-      })).then(Commands.m_82127_("notched_10").executes((p_136667_) -> {
-         return m_136606_(p_136667_.getSource(), m_136584_(p_136667_), BossEvent.BossBarOverlay.NOTCHED_10);
-      })).then(Commands.m_82127_("notched_12").executes((p_136665_) -> {
-         return m_136606_(p_136665_.getSource(), m_136584_(p_136665_), BossEvent.BossBarOverlay.NOTCHED_12);
-      })).then(Commands.m_82127_("notched_20").executes((p_136663_) -> {
-         return m_136606_(p_136663_.getSource(), m_136584_(p_136663_), BossEvent.BossBarOverlay.NOTCHED_20);
-      }))).then(Commands.m_82127_("value").then(Commands.m_82129_("value", IntegerArgumentType.integer(0)).executes((p_136661_) -> {
-         return m_136598_(p_136661_.getSource(), m_136584_(p_136661_), IntegerArgumentType.getInteger(p_136661_, "value"));
-      }))).then(Commands.m_82127_("max").then(Commands.m_82129_("max", IntegerArgumentType.integer(1)).executes((p_136659_) -> {
-         return m_136631_(p_136659_.getSource(), m_136584_(p_136659_), IntegerArgumentType.getInteger(p_136659_, "max"));
-      }))).then(Commands.m_82127_("visible").then(Commands.m_82129_("visible", BoolArgumentType.bool()).executes((p_136657_) -> {
-         return m_136618_(p_136657_.getSource(), m_136584_(p_136657_), BoolArgumentType.getBool(p_136657_, "visible"));
-      }))).then(Commands.m_82127_("players").executes((p_136655_) -> {
-         return m_136610_(p_136655_.getSource(), m_136584_(p_136655_), Collections.emptyList());
-      }).then(Commands.m_82129_("targets", EntityArgument.m_91470_()).executes((p_136653_) -> {
-         return m_136610_(p_136653_.getSource(), m_136584_(p_136653_), EntityArgument.m_91471_(p_136653_, "targets"));
-      }))))).then(Commands.m_82127_("get").then(Commands.m_82129_("id", ResourceLocationArgument.m_106984_()).suggests(f_136570_).then(Commands.m_82127_("value").executes((p_136648_) -> {
-         return m_136595_(p_136648_.getSource(), m_136584_(p_136648_));
-      })).then(Commands.m_82127_("max").executes((p_136643_) -> {
-         return m_136628_(p_136643_.getSource(), m_136584_(p_136643_));
-      })).then(Commands.m_82127_("visible").executes((p_136638_) -> {
-         return m_136639_(p_136638_.getSource(), m_136584_(p_136638_));
-      })).then(Commands.m_82127_("players").executes((p_136625_) -> {
-         return m_136644_(p_136625_.getSource(), m_136584_(p_136625_));
+      p_136583_.register(Commands.m_82127_("bossbar").requires((p_201423_0_) -> {
+         return p_201423_0_.m_6761_(2);
+      }).then(Commands.m_82127_("add").then(Commands.m_82129_("id", ResourceLocationArgument.m_106984_()).then(Commands.m_82129_("name", ComponentArgument.m_87114_()).executes((p_201426_0_) -> {
+         return m_136591_(p_201426_0_.getSource(), ResourceLocationArgument.m_107011_(p_201426_0_, "id"), ComponentArgument.m_87117_(p_201426_0_, "name"));
+      })))).then(Commands.m_82127_("remove").then(Commands.m_82129_("id", ResourceLocationArgument.m_106984_()).suggests(f_136570_).executes((p_201429_0_) -> {
+         return m_136649_(p_201429_0_.getSource(), m_136584_(p_201429_0_));
+      }))).then(Commands.m_82127_("list").executes((p_201396_0_) -> {
+         return m_136589_(p_201396_0_.getSource());
+      })).then(Commands.m_82127_("set").then(Commands.m_82129_("id", ResourceLocationArgument.m_106984_()).suggests(f_136570_).then(Commands.m_82127_("name").then(Commands.m_82129_("name", ComponentArgument.m_87114_()).executes((p_201401_0_) -> {
+         return m_136614_(p_201401_0_.getSource(), m_136584_(p_201401_0_), ComponentArgument.m_87117_(p_201401_0_, "name"));
+      }))).then(Commands.m_82127_("color").then(Commands.m_82127_("pink").executes((p_201409_0_) -> {
+         return m_136602_(p_201409_0_.getSource(), m_136584_(p_201409_0_), BossEvent.BossBarColor.PINK);
+      })).then(Commands.m_82127_("blue").executes((p_201422_0_) -> {
+         return m_136602_(p_201422_0_.getSource(), m_136584_(p_201422_0_), BossEvent.BossBarColor.BLUE);
+      })).then(Commands.m_82127_("red").executes((p_201417_0_) -> {
+         return m_136602_(p_201417_0_.getSource(), m_136584_(p_201417_0_), BossEvent.BossBarColor.RED);
+      })).then(Commands.m_82127_("green").executes((p_201424_0_) -> {
+         return m_136602_(p_201424_0_.getSource(), m_136584_(p_201424_0_), BossEvent.BossBarColor.GREEN);
+      })).then(Commands.m_82127_("yellow").executes((p_201393_0_) -> {
+         return m_136602_(p_201393_0_.getSource(), m_136584_(p_201393_0_), BossEvent.BossBarColor.YELLOW);
+      })).then(Commands.m_82127_("purple").executes((p_201391_0_) -> {
+         return m_136602_(p_201391_0_.getSource(), m_136584_(p_201391_0_), BossEvent.BossBarColor.PURPLE);
+      })).then(Commands.m_82127_("white").executes((p_201406_0_) -> {
+         return m_136602_(p_201406_0_.getSource(), m_136584_(p_201406_0_), BossEvent.BossBarColor.WHITE);
+      }))).then(Commands.m_82127_("style").then(Commands.m_82127_("progress").executes((p_201399_0_) -> {
+         return m_136606_(p_201399_0_.getSource(), m_136584_(p_201399_0_), BossEvent.BossBarOverlay.PROGRESS);
+      })).then(Commands.m_82127_("notched_6").executes((p_201419_0_) -> {
+         return m_136606_(p_201419_0_.getSource(), m_136584_(p_201419_0_), BossEvent.BossBarOverlay.NOTCHED_6);
+      })).then(Commands.m_82127_("notched_10").executes((p_201412_0_) -> {
+         return m_136606_(p_201412_0_.getSource(), m_136584_(p_201412_0_), BossEvent.BossBarOverlay.NOTCHED_10);
+      })).then(Commands.m_82127_("notched_12").executes((p_201421_0_) -> {
+         return m_136606_(p_201421_0_.getSource(), m_136584_(p_201421_0_), BossEvent.BossBarOverlay.NOTCHED_12);
+      })).then(Commands.m_82127_("notched_20").executes((p_201403_0_) -> {
+         return m_136606_(p_201403_0_.getSource(), m_136584_(p_201403_0_), BossEvent.BossBarOverlay.NOTCHED_20);
+      }))).then(Commands.m_82127_("value").then(Commands.m_82129_("value", IntegerArgumentType.integer(0)).executes((p_201408_0_) -> {
+         return m_136598_(p_201408_0_.getSource(), m_136584_(p_201408_0_), IntegerArgumentType.getInteger(p_201408_0_, "value"));
+      }))).then(Commands.m_82127_("max").then(Commands.m_82129_("max", IntegerArgumentType.integer(1)).executes((p_201395_0_) -> {
+         return m_136631_(p_201395_0_.getSource(), m_136584_(p_201395_0_), IntegerArgumentType.getInteger(p_201395_0_, "max"));
+      }))).then(Commands.m_82127_("visible").then(Commands.m_82129_("visible", BoolArgumentType.bool()).executes((p_201427_0_) -> {
+         return m_136618_(p_201427_0_.getSource(), m_136584_(p_201427_0_), BoolArgumentType.getBool(p_201427_0_, "visible"));
+      }))).then(Commands.m_82127_("players").executes((p_201430_0_) -> {
+         return m_136610_(p_201430_0_.getSource(), m_136584_(p_201430_0_), Collections.emptyList());
+      }).then(Commands.m_82129_("targets", EntityArgument.m_91470_()).executes((p_201411_0_) -> {
+         return m_136610_(p_201411_0_.getSource(), m_136584_(p_201411_0_), EntityArgument.m_91471_(p_201411_0_, "targets"));
+      }))))).then(Commands.m_82127_("get").then(Commands.m_82129_("id", ResourceLocationArgument.m_106984_()).suggests(f_136570_).then(Commands.m_82127_("value").executes((p_201418_0_) -> {
+         return m_136595_(p_201418_0_.getSource(), m_136584_(p_201418_0_));
+      })).then(Commands.m_82127_("max").executes((p_201398_0_) -> {
+         return m_136628_(p_201398_0_.getSource(), m_136584_(p_201398_0_));
+      })).then(Commands.m_82127_("visible").executes((p_201392_0_) -> {
+         return m_136639_(p_201392_0_.getSource(), m_136584_(p_201392_0_));
+      })).then(Commands.m_82127_("players").executes((p_201388_0_) -> {
+         return m_136644_(p_201388_0_.getSource(), m_136584_(p_201388_0_));
       })))));
    }
 
