--- a/net/minecraft/server/commands/EmoteCommands.java
+++ b/net/minecraft/server/commands/EmoteCommands.java
@@ -16,26 +_,26 @@
 
 public class EmoteCommands {
    public static void m_136985_(CommandDispatcher<CommandSourceStack> p_136986_) {
-      p_136986_.register(Commands.m_82127_("me").then(Commands.m_82129_("action", StringArgumentType.greedyString()).executes((p_136988_) -> {
-         String s = StringArgumentType.getString(p_136988_, "action");
-         Entity entity = p_136988_.getSource().m_81373_();
-         MinecraftServer minecraftserver = p_136988_.getSource().m_81377_();
+      p_136986_.register(Commands.m_82127_("me").then(Commands.m_82129_("action", StringArgumentType.greedyString()).executes((p_198365_0_) -> {
+         String s = StringArgumentType.getString(p_198365_0_, "action");
+         Entity entity = p_198365_0_.getSource().m_81373_();
+         MinecraftServer minecraftserver = p_198365_0_.getSource().m_81377_();
          if (entity != null) {
             if (entity instanceof ServerPlayer) {
                TextFilter textfilter = ((ServerPlayer)entity).m_8967_();
                if (textfilter != null) {
-                  textfilter.m_6770_(s).thenAcceptAsync((p_137001_) -> {
-                     p_137001_.ifPresent((p_136996_) -> {
-                        minecraftserver.m_6846_().m_11264_(m_136989_(p_136988_, p_136996_), ChatType.CHAT, entity.m_20148_());
+                  textfilter.m_6770_(s).thenAcceptAsync((p_244713_3_) -> {
+                     p_244713_3_.ifPresent((p_244712_3_) -> {
+                        minecraftserver.m_6846_().m_11264_(m_136989_(p_198365_0_, p_244712_3_), ChatType.CHAT, entity.m_20148_());
                      });
                   }, minecraftserver);
                   return 1;
                }
             }
 
-            minecraftserver.m_6846_().m_11264_(m_136989_(p_136988_, s), ChatType.CHAT, entity.m_20148_());
+            minecraftserver.m_6846_().m_11264_(m_136989_(p_198365_0_, s), ChatType.CHAT, entity.m_20148_());
          } else {
-            minecraftserver.m_6846_().m_11264_(m_136989_(p_136988_, s), ChatType.SYSTEM, Util.f_137441_);
+            minecraftserver.m_6846_().m_11264_(m_136989_(p_198365_0_, s), ChatType.SYSTEM, Util.f_137441_);
          }
 
          return 1;
