--- a/net/minecraft/server/commands/FillCommand.java
+++ b/net/minecraft/server/commands/FillCommand.java
@@ -27,31 +_,31 @@
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 
 public class FillCommand {
-   private static final Dynamic2CommandExceptionType f_137372_ = new Dynamic2CommandExceptionType((p_137392_, p_137393_) -> {
-      return new TranslatableComponent("commands.fill.toobig", p_137392_, p_137393_);
+   private static final Dynamic2CommandExceptionType f_137372_ = new Dynamic2CommandExceptionType((p_208897_0_, p_208897_1_) -> {
+      return new TranslatableComponent("commands.fill.toobig", p_208897_0_, p_208897_1_);
    });
    private static final BlockInput f_137373_ = new BlockInput(Blocks.f_50016_.m_49966_(), Collections.emptySet(), (CompoundTag)null);
    private static final SimpleCommandExceptionType f_137374_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.fill.failed"));
 
    public static void m_137379_(CommandDispatcher<CommandSourceStack> p_137380_) {
-      p_137380_.register(Commands.m_82127_("fill").requires((p_137384_) -> {
-         return p_137384_.m_6761_(2);
-      }).then(Commands.m_82129_("from", BlockPosArgument.m_118239_()).then(Commands.m_82129_("to", BlockPosArgument.m_118239_()).then(Commands.m_82129_("block", BlockStateArgument.m_116120_()).executes((p_137405_) -> {
-         return m_137385_(p_137405_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_137405_, "from"), BlockPosArgument.m_118242_(p_137405_, "to")), BlockStateArgument.m_116123_(p_137405_, "block"), FillCommand.Mode.REPLACE, (Predicate<BlockInWorld>)null);
-      }).then(Commands.m_82127_("replace").executes((p_137403_) -> {
-         return m_137385_(p_137403_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_137403_, "from"), BlockPosArgument.m_118242_(p_137403_, "to")), BlockStateArgument.m_116123_(p_137403_, "block"), FillCommand.Mode.REPLACE, (Predicate<BlockInWorld>)null);
-      }).then(Commands.m_82129_("filter", BlockPredicateArgument.m_115570_()).executes((p_137401_) -> {
-         return m_137385_(p_137401_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_137401_, "from"), BlockPosArgument.m_118242_(p_137401_, "to")), BlockStateArgument.m_116123_(p_137401_, "block"), FillCommand.Mode.REPLACE, BlockPredicateArgument.m_115573_(p_137401_, "filter"));
-      }))).then(Commands.m_82127_("keep").executes((p_137399_) -> {
-         return m_137385_(p_137399_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_137399_, "from"), BlockPosArgument.m_118242_(p_137399_, "to")), BlockStateArgument.m_116123_(p_137399_, "block"), FillCommand.Mode.REPLACE, (p_137378_) -> {
-            return p_137378_.m_61175_().m_46859_(p_137378_.m_61176_());
+      p_137380_.register(Commands.m_82127_("fill").requires((p_198471_0_) -> {
+         return p_198471_0_.m_6761_(2);
+      }).then(Commands.m_82129_("from", BlockPosArgument.m_118239_()).then(Commands.m_82129_("to", BlockPosArgument.m_118239_()).then(Commands.m_82129_("block", BlockStateArgument.m_116120_()).executes((p_198472_0_) -> {
+         return m_137385_(p_198472_0_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_198472_0_, "from"), BlockPosArgument.m_118242_(p_198472_0_, "to")), BlockStateArgument.m_116123_(p_198472_0_, "block"), FillCommand.Mode.REPLACE, (Predicate<BlockInWorld>)null);
+      }).then(Commands.m_82127_("replace").executes((p_198464_0_) -> {
+         return m_137385_(p_198464_0_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_198464_0_, "from"), BlockPosArgument.m_118242_(p_198464_0_, "to")), BlockStateArgument.m_116123_(p_198464_0_, "block"), FillCommand.Mode.REPLACE, (Predicate<BlockInWorld>)null);
+      }).then(Commands.m_82129_("filter", BlockPredicateArgument.m_115570_()).executes((p_198466_0_) -> {
+         return m_137385_(p_198466_0_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_198466_0_, "from"), BlockPosArgument.m_118242_(p_198466_0_, "to")), BlockStateArgument.m_116123_(p_198466_0_, "block"), FillCommand.Mode.REPLACE, BlockPredicateArgument.m_115573_(p_198466_0_, "filter"));
+      }))).then(Commands.m_82127_("keep").executes((p_198462_0_) -> {
+         return m_137385_(p_198462_0_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_198462_0_, "from"), BlockPosArgument.m_118242_(p_198462_0_, "to")), BlockStateArgument.m_116123_(p_198462_0_, "block"), FillCommand.Mode.REPLACE, (p_198469_0_) -> {
+            return p_198469_0_.m_61175_().m_46859_(p_198469_0_.m_61176_());
          });
-      })).then(Commands.m_82127_("outline").executes((p_137397_) -> {
-         return m_137385_(p_137397_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_137397_, "from"), BlockPosArgument.m_118242_(p_137397_, "to")), BlockStateArgument.m_116123_(p_137397_, "block"), FillCommand.Mode.OUTLINE, (Predicate<BlockInWorld>)null);
-      })).then(Commands.m_82127_("hollow").executes((p_137395_) -> {
-         return m_137385_(p_137395_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_137395_, "from"), BlockPosArgument.m_118242_(p_137395_, "to")), BlockStateArgument.m_116123_(p_137395_, "block"), FillCommand.Mode.HOLLOW, (Predicate<BlockInWorld>)null);
-      })).then(Commands.m_82127_("destroy").executes((p_137382_) -> {
-         return m_137385_(p_137382_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_137382_, "from"), BlockPosArgument.m_118242_(p_137382_, "to")), BlockStateArgument.m_116123_(p_137382_, "block"), FillCommand.Mode.DESTROY, (Predicate<BlockInWorld>)null);
+      })).then(Commands.m_82127_("outline").executes((p_198467_0_) -> {
+         return m_137385_(p_198467_0_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_198467_0_, "from"), BlockPosArgument.m_118242_(p_198467_0_, "to")), BlockStateArgument.m_116123_(p_198467_0_, "block"), FillCommand.Mode.OUTLINE, (Predicate<BlockInWorld>)null);
+      })).then(Commands.m_82127_("hollow").executes((p_198461_0_) -> {
+         return m_137385_(p_198461_0_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_198461_0_, "from"), BlockPosArgument.m_118242_(p_198461_0_, "to")), BlockStateArgument.m_116123_(p_198461_0_, "block"), FillCommand.Mode.HOLLOW, (Predicate<BlockInWorld>)null);
+      })).then(Commands.m_82127_("destroy").executes((p_198468_0_) -> {
+         return m_137385_(p_198468_0_.getSource(), new BoundingBox(BlockPosArgument.m_118242_(p_198468_0_, "from"), BlockPosArgument.m_118242_(p_198468_0_, "to")), BlockStateArgument.m_116123_(p_198468_0_, "block"), FillCommand.Mode.DESTROY, (Predicate<BlockInWorld>)null);
       }))))));
    }
 
@@ -93,18 +_,18 @@
    }
 
    static enum Mode {
-      REPLACE((p_137433_, p_137434_, p_137435_, p_137436_) -> {
-         return p_137435_;
-      }),
-      OUTLINE((p_137428_, p_137429_, p_137430_, p_137431_) -> {
-         return p_137429_.m_123341_() != p_137428_.f_70988_ && p_137429_.m_123341_() != p_137428_.f_70991_ && p_137429_.m_123342_() != p_137428_.f_70989_ && p_137429_.m_123342_() != p_137428_.f_70992_ && p_137429_.m_123343_() != p_137428_.f_70990_ && p_137429_.m_123343_() != p_137428_.f_70993_ ? null : p_137430_;
-      }),
-      HOLLOW((p_137423_, p_137424_, p_137425_, p_137426_) -> {
-         return p_137424_.m_123341_() != p_137423_.f_70988_ && p_137424_.m_123341_() != p_137423_.f_70991_ && p_137424_.m_123342_() != p_137423_.f_70989_ && p_137424_.m_123342_() != p_137423_.f_70992_ && p_137424_.m_123343_() != p_137423_.f_70990_ && p_137424_.m_123343_() != p_137423_.f_70993_ ? FillCommand.f_137373_ : p_137425_;
-      }),
-      DESTROY((p_137418_, p_137419_, p_137420_, p_137421_) -> {
-         p_137421_.m_46961_(p_137419_, true);
-         return p_137420_;
+      REPLACE((p_198450_0_, p_198450_1_, p_198450_2_, p_198450_3_) -> {
+         return p_198450_2_;
+      }),
+      OUTLINE((p_198454_0_, p_198454_1_, p_198454_2_, p_198454_3_) -> {
+         return p_198454_1_.m_123341_() != p_198454_0_.f_70988_ && p_198454_1_.m_123341_() != p_198454_0_.f_70991_ && p_198454_1_.m_123342_() != p_198454_0_.f_70989_ && p_198454_1_.m_123342_() != p_198454_0_.f_70992_ && p_198454_1_.m_123343_() != p_198454_0_.f_70990_ && p_198454_1_.m_123343_() != p_198454_0_.f_70993_ ? null : p_198454_2_;
+      }),
+      HOLLOW((p_198453_0_, p_198453_1_, p_198453_2_, p_198453_3_) -> {
+         return p_198453_1_.m_123341_() != p_198453_0_.f_70988_ && p_198453_1_.m_123341_() != p_198453_0_.f_70991_ && p_198453_1_.m_123342_() != p_198453_0_.f_70989_ && p_198453_1_.m_123342_() != p_198453_0_.f_70992_ && p_198453_1_.m_123343_() != p_198453_0_.f_70990_ && p_198453_1_.m_123343_() != p_198453_0_.f_70993_ ? FillCommand.f_137373_ : p_198453_2_;
+      }),
+      DESTROY((p_198452_0_, p_198452_1_, p_198452_2_, p_198452_3_) -> {
+         p_198452_3_.m_46961_(p_198452_1_, true);
+         return p_198452_2_;
       });
 
       public final SetBlockCommand.Filter f_137410_;
