--- a/net/minecraft/server/commands/GameModeCommand.java
+++ b/net/minecraft/server/commands/GameModeCommand.java
@@ -17,16 +_,16 @@
 
 public class GameModeCommand {
    public static void m_137729_(CommandDispatcher<CommandSourceStack> p_137730_) {
-      LiteralArgumentBuilder<CommandSourceStack> literalargumentbuilder = Commands.m_82127_("gamemode").requires((p_137736_) -> {
-         return p_137736_.m_6761_(2);
+      LiteralArgumentBuilder<CommandSourceStack> literalargumentbuilder = Commands.m_82127_("gamemode").requires((p_198485_0_) -> {
+         return p_198485_0_.m_6761_(2);
       });
 
       for(GameType gametype : GameType.values()) {
          if (gametype != GameType.NOT_SET) {
-            literalargumentbuilder.then(Commands.m_82127_(gametype.m_46405_()).executes((p_137743_) -> {
-               return m_137731_(p_137743_, Collections.singleton(p_137743_.getSource().m_81375_()), gametype);
-            }).then(Commands.m_82129_("target", EntityArgument.m_91470_()).executes((p_137728_) -> {
-               return m_137731_(p_137728_, EntityArgument.m_91477_(p_137728_, "target"), gametype);
+            literalargumentbuilder.then(Commands.m_82127_(gametype.m_46405_()).executes((p_198483_1_) -> {
+               return m_137731_(p_198483_1_, Collections.singleton(p_198483_1_.getSource().m_81375_()), gametype);
+            }).then(Commands.m_82129_("target", EntityArgument.m_91470_()).executes((p_198486_1_) -> {
+               return m_137731_(p_198486_1_, EntityArgument.m_91477_(p_198486_1_, "target"), gametype);
             })));
          }
       }
