--- a/net/minecraft/server/commands/KillCommand.java
+++ b/net/minecraft/server/commands/KillCommand.java
@@ -11,12 +_,12 @@
 
 public class KillCommand {
    public static void m_137807_(CommandDispatcher<CommandSourceStack> p_137808_) {
-      p_137808_.register(Commands.m_82127_("kill").requires((p_137812_) -> {
-         return p_137812_.m_6761_(2);
-      }).executes((p_137817_) -> {
-         return m_137813_(p_137817_.getSource(), ImmutableList.of(p_137817_.getSource().m_81374_()));
-      }).then(Commands.m_82129_("targets", EntityArgument.m_91460_()).executes((p_137810_) -> {
-         return m_137813_(p_137810_.getSource(), EntityArgument.m_91461_(p_137810_, "targets"));
+      p_137808_.register(Commands.m_82127_("kill").requires((p_198521_0_) -> {
+         return p_198521_0_.m_6761_(2);
+      }).executes((p_198520_0_) -> {
+         return m_137813_(p_198520_0_.getSource(), ImmutableList.of(p_198520_0_.getSource().m_81374_()));
+      }).then(Commands.m_82129_("targets", EntityArgument.m_91460_()).executes((p_229810_0_) -> {
+         return m_137813_(p_229810_0_.getSource(), EntityArgument.m_91461_(p_229810_0_, "targets"));
       })));
    }
 
