--- a/net/minecraft/server/commands/ListPlayersCommand.java
+++ b/net/minecraft/server/commands/ListPlayersCommand.java
@@ -14,10 +_,10 @@
 
 public class ListPlayersCommand {
    public static void m_137820_(CommandDispatcher<CommandSourceStack> p_137821_) {
-      p_137821_.register(Commands.m_82127_("list").executes((p_137830_) -> {
-         return m_137824_(p_137830_.getSource());
-      }).then(Commands.m_82127_("uuids").executes((p_137823_) -> {
-         return m_137831_(p_137823_.getSource());
+      p_137821_.register(Commands.m_82127_("list").executes((p_198523_0_) -> {
+         return m_137824_(p_198523_0_.getSource());
+      }).then(Commands.m_82127_("uuids").executes((p_208202_0_) -> {
+         return m_137831_(p_208202_0_.getSource());
       })));
    }
 
@@ -26,8 +_,8 @@
    }
 
    private static int m_137831_(CommandSourceStack p_137832_) {
-      return m_137826_(p_137832_, (p_137819_) -> {
-         return new TranslatableComponent("commands.list.nameAndId", p_137819_.m_7755_(), p_137819_.m_36316_().getId());
+      return m_137826_(p_137832_, (p_244373_0_) -> {
+         return new TranslatableComponent("commands.list.nameAndId", p_244373_0_.m_7755_(), p_244373_0_.m_36316_().getId());
       });
    }
 
