--- a/net/minecraft/server/commands/LootCommand.java
+++ b/net/minecraft/server/commands/LootCommand.java
@@ -43,60 +_,60 @@
 import net.minecraft.world.phys.Vec3;
 
 public class LootCommand {
-   public static final SuggestionProvider<CommandSourceStack> f_137877_ = (p_137910_, p_137911_) -> {
-      LootTables loottables = p_137910_.getSource().m_81377_().m_129898_();
-      return SharedSuggestionProvider.m_82926_(loottables.m_79195_(), p_137911_);
+   public static final SuggestionProvider<CommandSourceStack> f_137877_ = (p_218873_0_, p_218873_1_) -> {
+      LootTables loottables = p_218873_0_.getSource().m_81377_().m_129898_();
+      return SharedSuggestionProvider.m_82926_(loottables.m_79195_(), p_218873_1_);
    };
-   private static final DynamicCommandExceptionType f_137878_ = new DynamicCommandExceptionType((p_137999_) -> {
-      return new TranslatableComponent("commands.drop.no_held_items", p_137999_);
+   private static final DynamicCommandExceptionType f_137878_ = new DynamicCommandExceptionType((p_218896_0_) -> {
+      return new TranslatableComponent("commands.drop.no_held_items", p_218896_0_);
    });
-   private static final DynamicCommandExceptionType f_137879_ = new DynamicCommandExceptionType((p_137977_) -> {
-      return new TranslatableComponent("commands.drop.no_loot_table", p_137977_);
+   private static final DynamicCommandExceptionType f_137879_ = new DynamicCommandExceptionType((p_218889_0_) -> {
+      return new TranslatableComponent("commands.drop.no_loot_table", p_218889_0_);
    });
 
    public static void m_137897_(CommandDispatcher<CommandSourceStack> p_137898_) {
-      p_137898_.register(m_137902_(Commands.m_82127_("loot").requires((p_137937_) -> {
-         return p_137937_.m_6761_(2);
-      }), (p_137900_, p_137901_) -> {
-         return p_137900_.then(Commands.m_82127_("fish").then(Commands.m_82129_("loot_table", ResourceLocationArgument.m_106984_()).suggests(f_137877_).then(Commands.m_82129_("pos", BlockPosArgument.m_118239_()).executes((p_138046_) -> {
-            return m_137926_(p_138046_, ResourceLocationArgument.m_107011_(p_138046_, "loot_table"), BlockPosArgument.m_118242_(p_138046_, "pos"), ItemStack.f_41583_, p_137901_);
-         }).then(Commands.m_82129_("tool", ItemArgument.m_120960_()).executes((p_138043_) -> {
-            return m_137926_(p_138043_, ResourceLocationArgument.m_107011_(p_138043_, "loot_table"), BlockPosArgument.m_118242_(p_138043_, "pos"), ItemArgument.m_120963_(p_138043_, "tool").m_120980_(1, false), p_137901_);
-         })).then(Commands.m_82127_("mainhand").executes((p_138040_) -> {
-            return m_137926_(p_138040_, ResourceLocationArgument.m_107011_(p_138040_, "loot_table"), BlockPosArgument.m_118242_(p_138040_, "pos"), m_137938_(p_138040_.getSource(), EquipmentSlot.MAINHAND), p_137901_);
-         })).then(Commands.m_82127_("offhand").executes((p_138037_) -> {
-            return m_137926_(p_138037_, ResourceLocationArgument.m_107011_(p_138037_, "loot_table"), BlockPosArgument.m_118242_(p_138037_, "pos"), m_137938_(p_138037_.getSource(), EquipmentSlot.OFFHAND), p_137901_);
-         }))))).then(Commands.m_82127_("loot").then(Commands.m_82129_("loot_table", ResourceLocationArgument.m_106984_()).suggests(f_137877_).executes((p_138030_) -> {
-            return m_137932_(p_138030_, ResourceLocationArgument.m_107011_(p_138030_, "loot_table"), p_137901_);
-         }))).then(Commands.m_82127_("kill").then(Commands.m_82129_("target", EntityArgument.m_91449_()).executes((p_138023_) -> {
-            return m_137905_(p_138023_, EntityArgument.m_91452_(p_138023_, "target"), p_137901_);
-         }))).then(Commands.m_82127_("mine").then(Commands.m_82129_("pos", BlockPosArgument.m_118239_()).executes((p_138016_) -> {
-            return m_137912_(p_138016_, BlockPosArgument.m_118242_(p_138016_, "pos"), ItemStack.f_41583_, p_137901_);
-         }).then(Commands.m_82129_("tool", ItemArgument.m_120960_()).executes((p_138009_) -> {
-            return m_137912_(p_138009_, BlockPosArgument.m_118242_(p_138009_, "pos"), ItemArgument.m_120963_(p_138009_, "tool").m_120980_(1, false), p_137901_);
-         })).then(Commands.m_82127_("mainhand").executes((p_138002_) -> {
-            return m_137912_(p_138002_, BlockPosArgument.m_118242_(p_138002_, "pos"), m_137938_(p_138002_.getSource(), EquipmentSlot.MAINHAND), p_137901_);
-         })).then(Commands.m_82127_("offhand").executes((p_137990_) -> {
-            return m_137912_(p_137990_, BlockPosArgument.m_118242_(p_137990_, "pos"), m_137938_(p_137990_.getSource(), EquipmentSlot.OFFHAND), p_137901_);
+      p_137898_.register(m_137902_(Commands.m_82127_("loot").requires((p_218903_0_) -> {
+         return p_218903_0_.m_6761_(2);
+      }), (p_218880_0_, p_218880_1_) -> {
+         return p_218880_0_.then(Commands.m_82127_("fish").then(Commands.m_82129_("loot_table", ResourceLocationArgument.m_106984_()).suggests(f_137877_).then(Commands.m_82129_("pos", BlockPosArgument.m_118239_()).executes((p_218899_1_) -> {
+            return m_137926_(p_218899_1_, ResourceLocationArgument.m_107011_(p_218899_1_, "loot_table"), BlockPosArgument.m_118242_(p_218899_1_, "pos"), ItemStack.f_41583_, p_218880_1_);
+         }).then(Commands.m_82129_("tool", ItemArgument.m_120960_()).executes((p_218874_1_) -> {
+            return m_137926_(p_218874_1_, ResourceLocationArgument.m_107011_(p_218874_1_, "loot_table"), BlockPosArgument.m_118242_(p_218874_1_, "pos"), ItemArgument.m_120963_(p_218874_1_, "tool").m_120980_(1, false), p_218880_1_);
+         })).then(Commands.m_82127_("mainhand").executes((p_218892_1_) -> {
+            return m_137926_(p_218892_1_, ResourceLocationArgument.m_107011_(p_218892_1_, "loot_table"), BlockPosArgument.m_118242_(p_218892_1_, "pos"), m_137938_(p_218892_1_.getSource(), EquipmentSlot.MAINHAND), p_218880_1_);
+         })).then(Commands.m_82127_("offhand").executes((p_218898_1_) -> {
+            return m_137926_(p_218898_1_, ResourceLocationArgument.m_107011_(p_218898_1_, "loot_table"), BlockPosArgument.m_118242_(p_218898_1_, "pos"), m_137938_(p_218898_1_.getSource(), EquipmentSlot.OFFHAND), p_218880_1_);
+         }))))).then(Commands.m_82127_("loot").then(Commands.m_82129_("loot_table", ResourceLocationArgument.m_106984_()).suggests(f_137877_).executes((p_218861_1_) -> {
+            return m_137932_(p_218861_1_, ResourceLocationArgument.m_107011_(p_218861_1_, "loot_table"), p_218880_1_);
+         }))).then(Commands.m_82127_("kill").then(Commands.m_82129_("target", EntityArgument.m_91449_()).executes((p_218891_1_) -> {
+            return m_137905_(p_218891_1_, EntityArgument.m_91452_(p_218891_1_, "target"), p_218880_1_);
+         }))).then(Commands.m_82127_("mine").then(Commands.m_82129_("pos", BlockPosArgument.m_118239_()).executes((p_218897_1_) -> {
+            return m_137912_(p_218897_1_, BlockPosArgument.m_118242_(p_218897_1_, "pos"), ItemStack.f_41583_, p_218880_1_);
+         }).then(Commands.m_82129_("tool", ItemArgument.m_120960_()).executes((p_218878_1_) -> {
+            return m_137912_(p_218878_1_, BlockPosArgument.m_118242_(p_218878_1_, "pos"), ItemArgument.m_120963_(p_218878_1_, "tool").m_120980_(1, false), p_218880_1_);
+         })).then(Commands.m_82127_("mainhand").executes((p_218895_1_) -> {
+            return m_137912_(p_218895_1_, BlockPosArgument.m_118242_(p_218895_1_, "pos"), m_137938_(p_218895_1_.getSource(), EquipmentSlot.MAINHAND), p_218880_1_);
+         })).then(Commands.m_82127_("offhand").executes((p_218888_1_) -> {
+            return m_137912_(p_218888_1_, BlockPosArgument.m_118242_(p_218888_1_, "pos"), m_137938_(p_218888_1_.getSource(), EquipmentSlot.OFFHAND), p_218880_1_);
          }))));
       }));
    }
 
    private static <T extends ArgumentBuilder<CommandSourceStack, T>> T m_137902_(T p_137903_, LootCommand.TailProvider p_137904_) {
-      return p_137903_.then(Commands.m_82127_("replace").then(Commands.m_82127_("entity").then(Commands.m_82129_("entities", EntityArgument.m_91460_()).then(p_137904_.m_138053_(Commands.m_82129_("slot", SlotArgument.m_111276_()), (p_138032_, p_138033_, p_138034_) -> {
-         return m_137978_(EntityArgument.m_91461_(p_138032_, "entities"), SlotArgument.m_111279_(p_138032_, "slot"), p_138033_.size(), p_138033_, p_138034_);
-      }).then(p_137904_.m_138053_(Commands.m_82129_("count", IntegerArgumentType.integer(0)), (p_138025_, p_138026_, p_138027_) -> {
-         return m_137978_(EntityArgument.m_91461_(p_138025_, "entities"), SlotArgument.m_111279_(p_138025_, "slot"), IntegerArgumentType.getInteger(p_138025_, "count"), p_138026_, p_138027_);
-      }))))).then(Commands.m_82127_("block").then(Commands.m_82129_("targetPos", BlockPosArgument.m_118239_()).then(p_137904_.m_138053_(Commands.m_82129_("slot", SlotArgument.m_111276_()), (p_138018_, p_138019_, p_138020_) -> {
-         return m_137953_((CommandSourceStack)p_138018_.getSource(), BlockPosArgument.m_118242_(p_138018_, "targetPos"), SlotArgument.m_111279_(p_138018_, "slot"), p_138019_.size(), p_138019_, p_138020_);
-      }).then(p_137904_.m_138053_(Commands.m_82129_("count", IntegerArgumentType.integer(0)), (p_138011_, p_138012_, p_138013_) -> {
-         return m_137953_((CommandSourceStack)p_138011_.getSource(), BlockPosArgument.m_118242_(p_138011_, "targetPos"), IntegerArgumentType.getInteger(p_138011_, "slot"), IntegerArgumentType.getInteger(p_138011_, "count"), p_138012_, p_138013_);
-      })))))).then(Commands.m_82127_("insert").then(p_137904_.m_138053_(Commands.m_82129_("targetPos", BlockPosArgument.m_118239_()), (p_138004_, p_138005_, p_138006_) -> {
-         return m_137960_((CommandSourceStack)p_138004_.getSource(), BlockPosArgument.m_118242_(p_138004_, "targetPos"), p_138005_, p_138006_);
-      }))).then(Commands.m_82127_("give").then(p_137904_.m_138053_(Commands.m_82129_("players", EntityArgument.m_91470_()), (p_137992_, p_137993_, p_137994_) -> {
-         return m_137984_(EntityArgument.m_91477_(p_137992_, "players"), p_137993_, p_137994_);
-      }))).then(Commands.m_82127_("spawn").then(p_137904_.m_138053_(Commands.m_82129_("targetPos", Vec3Argument.m_120841_()), (p_137918_, p_137919_, p_137920_) -> {
-         return m_137945_((CommandSourceStack)p_137918_.getSource(), Vec3Argument.m_120844_(p_137918_, "targetPos"), p_137919_, p_137920_);
+      return p_137903_.then(Commands.m_82127_("replace").then(Commands.m_82127_("entity").then(Commands.m_82129_("entities", EntityArgument.m_91460_()).then(p_137904_.m_138053_(Commands.m_82129_("slot", SlotArgument.m_111276_()), (p_218866_0_, p_218866_1_, p_218866_2_) -> {
+         return m_137978_(EntityArgument.m_91461_(p_218866_0_, "entities"), SlotArgument.m_111279_(p_218866_0_, "slot"), p_218866_1_.size(), p_218866_1_, p_218866_2_);
+      }).then(p_137904_.m_138053_(Commands.m_82129_("count", IntegerArgumentType.integer(0)), (p_218884_0_, p_218884_1_, p_218884_2_) -> {
+         return m_137978_(EntityArgument.m_91461_(p_218884_0_, "entities"), SlotArgument.m_111279_(p_218884_0_, "slot"), IntegerArgumentType.getInteger(p_218884_0_, "count"), p_218884_1_, p_218884_2_);
+      }))))).then(Commands.m_82127_("block").then(Commands.m_82129_("targetPos", BlockPosArgument.m_118239_()).then(p_137904_.m_138053_(Commands.m_82129_("slot", SlotArgument.m_111276_()), (p_218864_0_, p_218864_1_, p_218864_2_) -> {
+         return m_137953_(p_218864_0_.getSource(), BlockPosArgument.m_118242_(p_218864_0_, "targetPos"), SlotArgument.m_111279_(p_218864_0_, "slot"), p_218864_1_.size(), p_218864_1_, p_218864_2_);
+      }).then(p_137904_.m_138053_(Commands.m_82129_("count", IntegerArgumentType.integer(0)), (p_218870_0_, p_218870_1_, p_218870_2_) -> {
+         return m_137953_(p_218870_0_.getSource(), BlockPosArgument.m_118242_(p_218870_0_, "targetPos"), IntegerArgumentType.getInteger(p_218870_0_, "slot"), IntegerArgumentType.getInteger(p_218870_0_, "count"), p_218870_1_, p_218870_2_);
+      })))))).then(Commands.m_82127_("insert").then(p_137904_.m_138053_(Commands.m_82129_("targetPos", BlockPosArgument.m_118239_()), (p_218885_0_, p_218885_1_, p_218885_2_) -> {
+         return m_137960_(p_218885_0_.getSource(), BlockPosArgument.m_118242_(p_218885_0_, "targetPos"), p_218885_1_, p_218885_2_);
+      }))).then(Commands.m_82127_("give").then(p_137904_.m_138053_(Commands.m_82129_("players", EntityArgument.m_91470_()), (p_218867_0_, p_218867_1_, p_218867_2_) -> {
+         return m_137984_(EntityArgument.m_91477_(p_218867_0_, "players"), p_218867_1_, p_218867_2_);
+      }))).then(Commands.m_82127_("spawn").then(p_137904_.m_138053_(Commands.m_82129_("targetPos", Vec3Argument.m_120841_()), (p_218877_0_, p_218877_1_, p_218877_2_) -> {
+         return m_137945_(p_218877_0_.getSource(), Vec3Argument.m_120844_(p_218877_0_, "targetPos"), p_218877_1_, p_218877_2_);
       })));
    }
 
@@ -220,8 +_,8 @@
 
    private static int m_137945_(CommandSourceStack p_137946_, Vec3 p_137947_, List<ItemStack> p_137948_, LootCommand.Callback p_137949_) throws CommandSyntaxException {
       ServerLevel serverlevel = p_137946_.m_81372_();
-      p_137948_.forEach((p_137884_) -> {
-         ItemEntity itementity = new ItemEntity(serverlevel, p_137947_.f_82479_, p_137947_.f_82480_, p_137947_.f_82481_, p_137884_.m_41777_());
+      p_137948_.forEach((p_218882_2_) -> {
+         ItemEntity itementity = new ItemEntity(serverlevel, p_137947_.f_82479_, p_137947_.f_82480_, p_137947_.f_82481_, p_218882_2_.m_41777_());
          itementity.m_32060_();
          serverlevel.m_7967_(itementity);
       });
@@ -265,8 +_,8 @@
       BlockEntity blockentity = serverlevel.m_7702_(p_137914_);
       LootContext.Builder lootcontext$builder = (new LootContext.Builder(serverlevel)).m_78972_(LootContextParams.f_81460_, Vec3.m_82512_(p_137914_)).m_78972_(LootContextParams.f_81461_, blockstate).m_78984_(LootContextParams.f_81462_, blockentity).m_78984_(LootContextParams.f_81455_, commandsourcestack.m_81373_()).m_78972_(LootContextParams.f_81463_, p_137915_);
       List<ItemStack> list = blockstate.m_60724_(lootcontext$builder);
-      return p_137916_.m_138049_(p_137913_, list, (p_137944_) -> {
-         m_137968_(commandsourcestack, p_137944_, blockstate.m_60734_().m_60589_());
+      return p_137916_.m_138049_(p_137913_, list, (p_218893_2_) -> {
+         m_137968_(commandsourcestack, p_218893_2_, blockstate.m_60734_().m_60589_());
       });
    }
 
@@ -289,8 +_,8 @@
          lootcontext$builder.m_78972_(LootContextParams.f_81460_, commandsourcestack.m_81371_());
          LootTable loottable = commandsourcestack.m_81377_().m_129898_().m_79217_(resourcelocation);
          List<ItemStack> list = loottable.m_79129_(lootcontext$builder.m_78975_(LootContextParamSets.f_81415_));
-         return p_137908_.m_138049_(p_137906_, list, (p_137975_) -> {
-            m_137968_(commandsourcestack, p_137975_, resourcelocation);
+         return p_137908_.m_138049_(p_137906_, list, (p_218863_2_) -> {
+            m_137968_(commandsourcestack, p_218863_2_, resourcelocation);
          });
       }
    }
@@ -311,9 +_,14 @@
       CommandSourceStack commandsourcestack = p_137922_.getSource();
       LootTable loottable = commandsourcestack.m_81377_().m_129898_().m_79217_(p_137923_);
       List<ItemStack> list = loottable.m_79129_(p_137924_);
-      return p_137925_.m_138049_(p_137922_, list, (p_137997_) -> {
-         m_137965_(commandsourcestack, p_137997_);
+      return p_137925_.m_138049_(p_137922_, list, (p_218902_1_) -> {
+         m_137965_(commandsourcestack, p_218902_1_);
       });
+   }
+
+   @FunctionalInterface
+   interface TailProvider {
+      ArgumentBuilder<CommandSourceStack, ?> m_138053_(ArgumentBuilder<CommandSourceStack, ?> p_138054_, LootCommand.DropConsumer p_138055_);
    }
 
    @FunctionalInterface
@@ -324,10 +_,5 @@
    @FunctionalInterface
    interface DropConsumer {
       int m_138049_(CommandContext<CommandSourceStack> p_138050_, List<ItemStack> p_138051_, LootCommand.Callback p_138052_) throws CommandSyntaxException;
-   }
-
-   @FunctionalInterface
-   interface TailProvider {
-      ArgumentBuilder<CommandSourceStack, ?> m_138053_(ArgumentBuilder<CommandSourceStack, ?> p_138054_, LootCommand.DropConsumer p_138055_);
    }
 }
