--- a/net/minecraft/server/commands/RecipeCommand.java
+++ b/net/minecraft/server/commands/RecipeCommand.java
@@ -19,16 +_,16 @@
    private static final SimpleCommandExceptionType f_138198_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.recipe.take.failed"));
 
    public static void m_138200_(CommandDispatcher<CommandSourceStack> p_138201_) {
-      p_138201_.register(Commands.m_82127_("recipe").requires((p_138205_) -> {
-         return p_138205_.m_6761_(2);
-      }).then(Commands.m_82127_("give").then(Commands.m_82129_("targets", EntityArgument.m_91470_()).then(Commands.m_82129_("recipe", ResourceLocationArgument.m_106984_()).suggests(SuggestionProviders.f_121642_).executes((p_138219_) -> {
-         return m_138206_(p_138219_.getSource(), EntityArgument.m_91477_(p_138219_, "targets"), Collections.singleton(ResourceLocationArgument.m_106994_(p_138219_, "recipe")));
-      })).then(Commands.m_82127_("*").executes((p_138217_) -> {
-         return m_138206_(p_138217_.getSource(), EntityArgument.m_91477_(p_138217_, "targets"), p_138217_.getSource().m_81377_().m_129894_().m_44051_());
-      })))).then(Commands.m_82127_("take").then(Commands.m_82129_("targets", EntityArgument.m_91470_()).then(Commands.m_82129_("recipe", ResourceLocationArgument.m_106984_()).suggests(SuggestionProviders.f_121642_).executes((p_138211_) -> {
-         return m_138212_(p_138211_.getSource(), EntityArgument.m_91477_(p_138211_, "targets"), Collections.singleton(ResourceLocationArgument.m_106994_(p_138211_, "recipe")));
-      })).then(Commands.m_82127_("*").executes((p_138203_) -> {
-         return m_138212_(p_138203_.getSource(), EntityArgument.m_91477_(p_138203_, "targets"), p_138203_.getSource().m_81377_().m_129894_().m_44051_());
+      p_138201_.register(Commands.m_82127_("recipe").requires((p_198593_0_) -> {
+         return p_198593_0_.m_6761_(2);
+      }).then(Commands.m_82127_("give").then(Commands.m_82129_("targets", EntityArgument.m_91470_()).then(Commands.m_82129_("recipe", ResourceLocationArgument.m_106984_()).suggests(SuggestionProviders.f_121642_).executes((p_198588_0_) -> {
+         return m_138206_(p_198588_0_.getSource(), EntityArgument.m_91477_(p_198588_0_, "targets"), Collections.singleton(ResourceLocationArgument.m_106994_(p_198588_0_, "recipe")));
+      })).then(Commands.m_82127_("*").executes((p_198591_0_) -> {
+         return m_138206_(p_198591_0_.getSource(), EntityArgument.m_91477_(p_198591_0_, "targets"), p_198591_0_.getSource().m_81377_().m_129894_().m_44051_());
+      })))).then(Commands.m_82127_("take").then(Commands.m_82129_("targets", EntityArgument.m_91470_()).then(Commands.m_82129_("recipe", ResourceLocationArgument.m_106984_()).suggests(SuggestionProviders.f_121642_).executes((p_198587_0_) -> {
+         return m_138212_(p_198587_0_.getSource(), EntityArgument.m_91477_(p_198587_0_, "targets"), Collections.singleton(ResourceLocationArgument.m_106994_(p_198587_0_, "recipe")));
+      })).then(Commands.m_82127_("*").executes((p_198592_0_) -> {
+         return m_138212_(p_198592_0_.getSource(), EntityArgument.m_91477_(p_198592_0_, "targets"), p_198592_0_.getSource().m_81377_().m_129894_().m_44051_());
       })))));
    }
 
