--- a/net/minecraft/server/commands/SaveAllCommand.java
+++ b/net/minecraft/server/commands/SaveAllCommand.java
@@ -12,12 +_,12 @@
    private static final SimpleCommandExceptionType f_138269_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.save.failed"));
 
    public static void m_138271_(CommandDispatcher<CommandSourceStack> p_138272_) {
-      p_138272_.register(Commands.m_82127_("save-all").requires((p_138276_) -> {
-         return p_138276_.m_6761_(4);
-      }).executes((p_138281_) -> {
-         return m_138277_(p_138281_.getSource(), false);
-      }).then(Commands.m_82127_("flush").executes((p_138274_) -> {
-         return m_138277_(p_138274_.getSource(), true);
+      p_138272_.register(Commands.m_82127_("save-all").requires((p_198615_0_) -> {
+         return p_198615_0_.m_6761_(4);
+      }).executes((p_198610_0_) -> {
+         return m_138277_(p_198610_0_.getSource(), false);
+      }).then(Commands.m_82127_("flush").executes((p_198613_0_) -> {
+         return m_138277_(p_198613_0_.getSource(), true);
       })));
    }
 
