--- a/net/minecraft/server/commands/SaveOnCommand.java
+++ b/net/minecraft/server/commands/SaveOnCommand.java
@@ -11,10 +_,10 @@
    private static final SimpleCommandExceptionType f_138290_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.save.alreadyOn"));
 
    public static void m_138292_(CommandDispatcher<CommandSourceStack> p_138293_) {
-      p_138293_.register(Commands.m_82127_("save-on").requires((p_138297_) -> {
-         return p_138297_.m_6761_(4);
-      }).executes((p_138295_) -> {
-         CommandSourceStack commandsourcestack = p_138295_.getSource();
+      p_138293_.register(Commands.m_82127_("save-on").requires((p_198623_0_) -> {
+         return p_198623_0_.m_6761_(4);
+      }).executes((p_198622_0_) -> {
+         CommandSourceStack commandsourcestack = p_198622_0_.getSource();
          boolean flag = false;
 
          for(ServerLevel serverlevel : commandsourcestack.m_81377_().m_129785_()) {
