--- a/net/minecraft/server/commands/SaveOffCommand.java
+++ b/net/minecraft/server/commands/SaveOffCommand.java
@@ -11,10 +_,10 @@
    private static final SimpleCommandExceptionType f_138282_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.save.alreadyOff"));
 
    public static void m_138284_(CommandDispatcher<CommandSourceStack> p_138285_) {
-      p_138285_.register(Commands.m_82127_("save-off").requires((p_138289_) -> {
-         return p_138289_.m_6761_(4);
-      }).executes((p_138287_) -> {
-         CommandSourceStack commandsourcestack = p_138287_.getSource();
+      p_138285_.register(Commands.m_82127_("save-off").requires((p_198619_0_) -> {
+         return p_198619_0_.m_6761_(4);
+      }).executes((p_198618_0_) -> {
+         CommandSourceStack commandsourcestack = p_198618_0_.getSource();
          boolean flag = false;
 
          for(ServerLevel serverlevel : commandsourcestack.m_81377_().m_129785_()) {
