--- a/net/minecraft/server/commands/SayCommand.java
+++ b/net/minecraft/server/commands/SayCommand.java
@@ -12,16 +_,16 @@
 
 public class SayCommand {
    public static void m_138409_(CommandDispatcher<CommandSourceStack> p_138410_) {
-      p_138410_.register(Commands.m_82127_("say").requires((p_138414_) -> {
-         return p_138414_.m_6761_(2);
-      }).then(Commands.m_82129_("message", MessageArgument.m_96832_()).executes((p_138412_) -> {
-         Component component = MessageArgument.m_96835_(p_138412_, "message");
-         TranslatableComponent translatablecomponent = new TranslatableComponent("chat.type.announcement", p_138412_.getSource().m_81357_(), component);
-         Entity entity = p_138412_.getSource().m_81373_();
+      p_138410_.register(Commands.m_82127_("say").requires((p_198627_0_) -> {
+         return p_198627_0_.m_6761_(2);
+      }).then(Commands.m_82129_("message", MessageArgument.m_96832_()).executes((p_198626_0_) -> {
+         Component component = MessageArgument.m_96835_(p_198626_0_, "message");
+         TranslatableComponent translatablecomponent = new TranslatableComponent("chat.type.announcement", p_198626_0_.getSource().m_81357_(), component);
+         Entity entity = p_198626_0_.getSource().m_81373_();
          if (entity != null) {
-            p_138412_.getSource().m_81377_().m_6846_().m_11264_(translatablecomponent, ChatType.CHAT, entity.m_20148_());
+            p_198626_0_.getSource().m_81377_().m_6846_().m_11264_(translatablecomponent, ChatType.CHAT, entity.m_20148_());
          } else {
-            p_138412_.getSource().m_81377_().m_6846_().m_11264_(translatablecomponent, ChatType.SYSTEM, Util.f_137441_);
+            p_198626_0_.getSource().m_81377_().m_6846_().m_11264_(translatablecomponent, ChatType.SYSTEM, Util.f_137441_);
          }
 
          return 1;
