--- a/net/minecraft/server/commands/SpectateCommand.java
+++ b/net/minecraft/server/commands/SpectateCommand.java
@@ -15,19 +_,19 @@
 
 public class SpectateCommand {
    private static final SimpleCommandExceptionType f_138674_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.spectate.self"));
-   private static final DynamicCommandExceptionType f_138675_ = new DynamicCommandExceptionType((p_138688_) -> {
-      return new TranslatableComponent("commands.spectate.not_spectator", p_138688_);
+   private static final DynamicCommandExceptionType f_138675_ = new DynamicCommandExceptionType((p_229830_0_) -> {
+      return new TranslatableComponent("commands.spectate.not_spectator", p_229830_0_);
    });
 
    public static void m_138677_(CommandDispatcher<CommandSourceStack> p_138678_) {
-      p_138678_.register(Commands.m_82127_("spectate").requires((p_138682_) -> {
-         return p_138682_.m_6761_(2);
-      }).executes((p_138692_) -> {
-         return m_138683_(p_138692_.getSource(), (Entity)null, p_138692_.getSource().m_81375_());
-      }).then(Commands.m_82129_("target", EntityArgument.m_91449_()).executes((p_138690_) -> {
-         return m_138683_(p_138690_.getSource(), EntityArgument.m_91452_(p_138690_, "target"), p_138690_.getSource().m_81375_());
-      }).then(Commands.m_82129_("player", EntityArgument.m_91466_()).executes((p_138680_) -> {
-         return m_138683_(p_138680_.getSource(), EntityArgument.m_91452_(p_138680_, "target"), EntityArgument.m_91474_(p_138680_, "player"));
+      p_138678_.register(Commands.m_82127_("spectate").requires((p_229828_0_) -> {
+         return p_229828_0_.m_6761_(2);
+      }).executes((p_229832_0_) -> {
+         return m_138683_(p_229832_0_.getSource(), (Entity)null, p_229832_0_.getSource().m_81375_());
+      }).then(Commands.m_82129_("target", EntityArgument.m_91449_()).executes((p_229831_0_) -> {
+         return m_138683_(p_229831_0_.getSource(), EntityArgument.m_91452_(p_229831_0_, "target"), p_229831_0_.getSource().m_81375_());
+      }).then(Commands.m_82129_("player", EntityArgument.m_91466_()).executes((p_229827_0_) -> {
+         return m_138683_(p_229827_0_.getSource(), EntityArgument.m_91452_(p_229827_0_, "target"), EntityArgument.m_91474_(p_229827_0_, "player"));
       }))));
    }
 
