--- a/net/minecraft/server/commands/StopCommand.java
+++ b/net/minecraft/server/commands/StopCommand.java
@@ -7,11 +_,11 @@
 
 public class StopCommand {
    public static void m_138785_(CommandDispatcher<CommandSourceStack> p_138786_) {
-      p_138786_.register(Commands.m_82127_("stop").requires((p_138790_) -> {
-         return p_138790_.m_6761_(4);
-      }).executes((p_138788_) -> {
-         p_138788_.getSource().m_81354_(new TranslatableComponent("commands.stop.stopping"), true);
-         p_138788_.getSource().m_81377_().m_7570_(false);
+      p_138786_.register(Commands.m_82127_("stop").requires((p_198727_0_) -> {
+         return p_198727_0_.m_6761_(4);
+      }).executes((p_198726_0_) -> {
+         p_198726_0_.getSource().m_81354_(new TranslatableComponent("commands.stop.stopping"), true);
+         p_198726_0_.getSource().m_81377_().m_7570_(false);
          return 1;
       }));
    }
