--- a/net/minecraft/server/commands/TeamCommand.java
+++ b/net/minecraft/server/commands/TeamCommand.java
@@ -26,8 +_,8 @@
 
 public class TeamCommand {
    private static final SimpleCommandExceptionType f_138862_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.team.add.duplicate"));
-   private static final DynamicCommandExceptionType f_138863_ = new DynamicCommandExceptionType((p_138921_) -> {
-      return new TranslatableComponent("commands.team.add.longName", p_138921_);
+   private static final DynamicCommandExceptionType f_138863_ = new DynamicCommandExceptionType((p_208916_0_) -> {
+      return new TranslatableComponent("commands.team.add.longName", p_208916_0_);
    });
    private static final SimpleCommandExceptionType f_138864_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.team.empty.unchanged"));
    private static final SimpleCommandExceptionType f_138865_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.team.option.name.unchanged"));
@@ -41,62 +_,62 @@
    private static final SimpleCommandExceptionType f_138873_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.team.option.collisionRule.unchanged"));
 
    public static void m_138877_(CommandDispatcher<CommandSourceStack> p_138878_) {
-      p_138878_.register(Commands.m_82127_("team").requires((p_138925_) -> {
-         return p_138925_.m_6761_(2);
-      }).then(Commands.m_82127_("list").executes((p_138876_) -> {
-         return m_138881_(p_138876_.getSource());
-      }).then(Commands.m_82129_("team", TeamArgument.m_112088_()).executes((p_138995_) -> {
-         return m_138943_(p_138995_.getSource(), TeamArgument.m_112091_(p_138995_, "team"));
-      }))).then(Commands.m_82127_("add").then(Commands.m_82129_("team", StringArgumentType.word()).executes((p_138993_) -> {
-         return m_138910_(p_138993_.getSource(), StringArgumentType.getString(p_138993_, "team"));
-      }).then(Commands.m_82129_("displayName", ComponentArgument.m_87114_()).executes((p_138991_) -> {
-         return m_138913_(p_138991_.getSource(), StringArgumentType.getString(p_138991_, "team"), ComponentArgument.m_87117_(p_138991_, "displayName"));
-      })))).then(Commands.m_82127_("remove").then(Commands.m_82129_("team", TeamArgument.m_112088_()).executes((p_138989_) -> {
-         return m_138926_(p_138989_.getSource(), TeamArgument.m_112091_(p_138989_, "team"));
-      }))).then(Commands.m_82127_("empty").then(Commands.m_82129_("team", TeamArgument.m_112088_()).executes((p_138987_) -> {
-         return m_138883_(p_138987_.getSource(), TeamArgument.m_112091_(p_138987_, "team"));
-      }))).then(Commands.m_82127_("join").then(Commands.m_82129_("team", TeamArgument.m_112088_()).executes((p_138985_) -> {
-         return m_138894_(p_138985_.getSource(), TeamArgument.m_112091_(p_138985_, "team"), Collections.singleton(p_138985_.getSource().m_81374_().m_6302_()));
-      }).then(Commands.m_82129_("members", ScoreHolderArgument.m_108239_()).suggests(ScoreHolderArgument.f_108210_).executes((p_138983_) -> {
-         return m_138894_(p_138983_.getSource(), TeamArgument.m_112091_(p_138983_, "team"), ScoreHolderArgument.m_108246_(p_138983_, "members"));
-      })))).then(Commands.m_82127_("leave").then(Commands.m_82129_("members", ScoreHolderArgument.m_108239_()).suggests(ScoreHolderArgument.f_108210_).executes((p_138981_) -> {
-         return m_138917_(p_138981_.getSource(), ScoreHolderArgument.m_108246_(p_138981_, "members"));
-      }))).then(Commands.m_82127_("modify").then(Commands.m_82129_("team", TeamArgument.m_112088_()).then(Commands.m_82127_("displayName").then(Commands.m_82129_("displayName", ComponentArgument.m_87114_()).executes((p_138979_) -> {
-         return m_138902_(p_138979_.getSource(), TeamArgument.m_112091_(p_138979_, "team"), ComponentArgument.m_87117_(p_138979_, "displayName"));
-      }))).then(Commands.m_82127_("color").then(Commands.m_82129_("value", ColorArgument.m_85463_()).executes((p_138977_) -> {
-         return m_138898_(p_138977_.getSource(), TeamArgument.m_112091_(p_138977_, "team"), ColorArgument.m_85466_(p_138977_, "value"));
-      }))).then(Commands.m_82127_("friendlyFire").then(Commands.m_82129_("allowed", BoolArgumentType.bool()).executes((p_138975_) -> {
-         return m_138937_(p_138975_.getSource(), TeamArgument.m_112091_(p_138975_, "team"), BoolArgumentType.getBool(p_138975_, "allowed"));
-      }))).then(Commands.m_82127_("seeFriendlyInvisibles").then(Commands.m_82129_("allowed", BoolArgumentType.bool()).executes((p_138973_) -> {
-         return m_138906_(p_138973_.getSource(), TeamArgument.m_112091_(p_138973_, "team"), BoolArgumentType.getBool(p_138973_, "allowed"));
-      }))).then(Commands.m_82127_("nametagVisibility").then(Commands.m_82127_("never").executes((p_138971_) -> {
-         return m_138890_(p_138971_.getSource(), TeamArgument.m_112091_(p_138971_, "team"), Team.Visibility.NEVER);
-      })).then(Commands.m_82127_("hideForOtherTeams").executes((p_138969_) -> {
-         return m_138890_(p_138969_.getSource(), TeamArgument.m_112091_(p_138969_, "team"), Team.Visibility.HIDE_FOR_OTHER_TEAMS);
-      })).then(Commands.m_82127_("hideForOwnTeam").executes((p_138967_) -> {
-         return m_138890_(p_138967_.getSource(), TeamArgument.m_112091_(p_138967_, "team"), Team.Visibility.HIDE_FOR_OWN_TEAM);
-      })).then(Commands.m_82127_("always").executes((p_138965_) -> {
-         return m_138890_(p_138965_.getSource(), TeamArgument.m_112091_(p_138965_, "team"), Team.Visibility.ALWAYS);
-      }))).then(Commands.m_82127_("deathMessageVisibility").then(Commands.m_82127_("never").executes((p_138963_) -> {
-         return m_138929_(p_138963_.getSource(), TeamArgument.m_112091_(p_138963_, "team"), Team.Visibility.NEVER);
-      })).then(Commands.m_82127_("hideForOtherTeams").executes((p_138961_) -> {
-         return m_138929_(p_138961_.getSource(), TeamArgument.m_112091_(p_138961_, "team"), Team.Visibility.HIDE_FOR_OTHER_TEAMS);
-      })).then(Commands.m_82127_("hideForOwnTeam").executes((p_138959_) -> {
-         return m_138929_(p_138959_.getSource(), TeamArgument.m_112091_(p_138959_, "team"), Team.Visibility.HIDE_FOR_OWN_TEAM);
-      })).then(Commands.m_82127_("always").executes((p_138957_) -> {
-         return m_138929_(p_138957_.getSource(), TeamArgument.m_112091_(p_138957_, "team"), Team.Visibility.ALWAYS);
-      }))).then(Commands.m_82127_("collisionRule").then(Commands.m_82127_("never").executes((p_138955_) -> {
-         return m_138886_(p_138955_.getSource(), TeamArgument.m_112091_(p_138955_, "team"), Team.CollisionRule.NEVER);
-      })).then(Commands.m_82127_("pushOwnTeam").executes((p_138953_) -> {
-         return m_138886_(p_138953_.getSource(), TeamArgument.m_112091_(p_138953_, "team"), Team.CollisionRule.PUSH_OWN_TEAM);
-      })).then(Commands.m_82127_("pushOtherTeams").executes((p_138951_) -> {
-         return m_138886_(p_138951_.getSource(), TeamArgument.m_112091_(p_138951_, "team"), Team.CollisionRule.PUSH_OTHER_TEAMS);
-      })).then(Commands.m_82127_("always").executes((p_138942_) -> {
-         return m_138886_(p_138942_.getSource(), TeamArgument.m_112091_(p_138942_, "team"), Team.CollisionRule.ALWAYS);
-      }))).then(Commands.m_82127_("prefix").then(Commands.m_82129_("prefix", ComponentArgument.m_87114_()).executes((p_138923_) -> {
-         return m_138933_(p_138923_.getSource(), TeamArgument.m_112091_(p_138923_, "team"), ComponentArgument.m_87117_(p_138923_, "prefix"));
-      }))).then(Commands.m_82127_("suffix").then(Commands.m_82129_("suffix", ComponentArgument.m_87114_()).executes((p_138880_) -> {
-         return m_138946_(p_138880_.getSource(), TeamArgument.m_112091_(p_138880_, "team"), ComponentArgument.m_87117_(p_138880_, "suffix"));
+      p_138878_.register(Commands.m_82127_("team").requires((p_198780_0_) -> {
+         return p_198780_0_.m_6761_(2);
+      }).then(Commands.m_82127_("list").executes((p_198760_0_) -> {
+         return m_138881_(p_198760_0_.getSource());
+      }).then(Commands.m_82129_("team", TeamArgument.m_112088_()).executes((p_198763_0_) -> {
+         return m_138943_(p_198763_0_.getSource(), TeamArgument.m_112091_(p_198763_0_, "team"));
+      }))).then(Commands.m_82127_("add").then(Commands.m_82129_("team", StringArgumentType.word()).executes((p_198767_0_) -> {
+         return m_138910_(p_198767_0_.getSource(), StringArgumentType.getString(p_198767_0_, "team"));
+      }).then(Commands.m_82129_("displayName", ComponentArgument.m_87114_()).executes((p_198779_0_) -> {
+         return m_138913_(p_198779_0_.getSource(), StringArgumentType.getString(p_198779_0_, "team"), ComponentArgument.m_87117_(p_198779_0_, "displayName"));
+      })))).then(Commands.m_82127_("remove").then(Commands.m_82129_("team", TeamArgument.m_112088_()).executes((p_198773_0_) -> {
+         return m_138926_(p_198773_0_.getSource(), TeamArgument.m_112091_(p_198773_0_, "team"));
+      }))).then(Commands.m_82127_("empty").then(Commands.m_82129_("team", TeamArgument.m_112088_()).executes((p_198785_0_) -> {
+         return m_138883_(p_198785_0_.getSource(), TeamArgument.m_112091_(p_198785_0_, "team"));
+      }))).then(Commands.m_82127_("join").then(Commands.m_82129_("team", TeamArgument.m_112088_()).executes((p_198758_0_) -> {
+         return m_138894_(p_198758_0_.getSource(), TeamArgument.m_112091_(p_198758_0_, "team"), Collections.singleton(p_198758_0_.getSource().m_81374_().m_6302_()));
+      }).then(Commands.m_82129_("members", ScoreHolderArgument.m_108239_()).suggests(ScoreHolderArgument.f_108210_).executes((p_198755_0_) -> {
+         return m_138894_(p_198755_0_.getSource(), TeamArgument.m_112091_(p_198755_0_, "team"), ScoreHolderArgument.m_108246_(p_198755_0_, "members"));
+      })))).then(Commands.m_82127_("leave").then(Commands.m_82129_("members", ScoreHolderArgument.m_108239_()).suggests(ScoreHolderArgument.f_108210_).executes((p_198765_0_) -> {
+         return m_138917_(p_198765_0_.getSource(), ScoreHolderArgument.m_108246_(p_198765_0_, "members"));
+      }))).then(Commands.m_82127_("modify").then(Commands.m_82129_("team", TeamArgument.m_112088_()).then(Commands.m_82127_("displayName").then(Commands.m_82129_("displayName", ComponentArgument.m_87114_()).executes((p_211919_0_) -> {
+         return m_138902_(p_211919_0_.getSource(), TeamArgument.m_112091_(p_211919_0_, "team"), ComponentArgument.m_87117_(p_211919_0_, "displayName"));
+      }))).then(Commands.m_82127_("color").then(Commands.m_82129_("value", ColorArgument.m_85463_()).executes((p_198762_0_) -> {
+         return m_138898_(p_198762_0_.getSource(), TeamArgument.m_112091_(p_198762_0_, "team"), ColorArgument.m_85466_(p_198762_0_, "value"));
+      }))).then(Commands.m_82127_("friendlyFire").then(Commands.m_82129_("allowed", BoolArgumentType.bool()).executes((p_198775_0_) -> {
+         return m_138937_(p_198775_0_.getSource(), TeamArgument.m_112091_(p_198775_0_, "team"), BoolArgumentType.getBool(p_198775_0_, "allowed"));
+      }))).then(Commands.m_82127_("seeFriendlyInvisibles").then(Commands.m_82129_("allowed", BoolArgumentType.bool()).executes((p_198770_0_) -> {
+         return m_138906_(p_198770_0_.getSource(), TeamArgument.m_112091_(p_198770_0_, "team"), BoolArgumentType.getBool(p_198770_0_, "allowed"));
+      }))).then(Commands.m_82127_("nametagVisibility").then(Commands.m_82127_("never").executes((p_198778_0_) -> {
+         return m_138890_(p_198778_0_.getSource(), TeamArgument.m_112091_(p_198778_0_, "team"), Team.Visibility.NEVER);
+      })).then(Commands.m_82127_("hideForOtherTeams").executes((p_198764_0_) -> {
+         return m_138890_(p_198764_0_.getSource(), TeamArgument.m_112091_(p_198764_0_, "team"), Team.Visibility.HIDE_FOR_OTHER_TEAMS);
+      })).then(Commands.m_82127_("hideForOwnTeam").executes((p_198766_0_) -> {
+         return m_138890_(p_198766_0_.getSource(), TeamArgument.m_112091_(p_198766_0_, "team"), Team.Visibility.HIDE_FOR_OWN_TEAM);
+      })).then(Commands.m_82127_("always").executes((p_198759_0_) -> {
+         return m_138890_(p_198759_0_.getSource(), TeamArgument.m_112091_(p_198759_0_, "team"), Team.Visibility.ALWAYS);
+      }))).then(Commands.m_82127_("deathMessageVisibility").then(Commands.m_82127_("never").executes((p_198789_0_) -> {
+         return m_138929_(p_198789_0_.getSource(), TeamArgument.m_112091_(p_198789_0_, "team"), Team.Visibility.NEVER);
+      })).then(Commands.m_82127_("hideForOtherTeams").executes((p_198791_0_) -> {
+         return m_138929_(p_198791_0_.getSource(), TeamArgument.m_112091_(p_198791_0_, "team"), Team.Visibility.HIDE_FOR_OTHER_TEAMS);
+      })).then(Commands.m_82127_("hideForOwnTeam").executes((p_198769_0_) -> {
+         return m_138929_(p_198769_0_.getSource(), TeamArgument.m_112091_(p_198769_0_, "team"), Team.Visibility.HIDE_FOR_OWN_TEAM);
+      })).then(Commands.m_82127_("always").executes((p_198774_0_) -> {
+         return m_138929_(p_198774_0_.getSource(), TeamArgument.m_112091_(p_198774_0_, "team"), Team.Visibility.ALWAYS);
+      }))).then(Commands.m_82127_("collisionRule").then(Commands.m_82127_("never").executes((p_198761_0_) -> {
+         return m_138886_(p_198761_0_.getSource(), TeamArgument.m_112091_(p_198761_0_, "team"), Team.CollisionRule.NEVER);
+      })).then(Commands.m_82127_("pushOwnTeam").executes((p_198756_0_) -> {
+         return m_138886_(p_198756_0_.getSource(), TeamArgument.m_112091_(p_198756_0_, "team"), Team.CollisionRule.PUSH_OWN_TEAM);
+      })).then(Commands.m_82127_("pushOtherTeams").executes((p_198754_0_) -> {
+         return m_138886_(p_198754_0_.getSource(), TeamArgument.m_112091_(p_198754_0_, "team"), Team.CollisionRule.PUSH_OTHER_TEAMS);
+      })).then(Commands.m_82127_("always").executes((p_198790_0_) -> {
+         return m_138886_(p_198790_0_.getSource(), TeamArgument.m_112091_(p_198790_0_, "team"), Team.CollisionRule.ALWAYS);
+      }))).then(Commands.m_82127_("prefix").then(Commands.m_82129_("prefix", ComponentArgument.m_87114_()).executes((p_207514_0_) -> {
+         return m_138933_(p_207514_0_.getSource(), TeamArgument.m_112091_(p_207514_0_, "team"), ComponentArgument.m_87117_(p_207514_0_, "prefix"));
+      }))).then(Commands.m_82127_("suffix").then(Commands.m_82129_("suffix", ComponentArgument.m_87114_()).executes((p_207516_0_) -> {
+         return m_138946_(p_207516_0_.getSource(), TeamArgument.m_112091_(p_207516_0_, "team"), ComponentArgument.m_87117_(p_207516_0_, "suffix"));
       }))))));
    }
 
