--- a/net/minecraft/server/commands/WeatherCommand.java
+++ b/net/minecraft/server/commands/WeatherCommand.java
@@ -8,20 +_,20 @@
 
 public class WeatherCommand {
    public static void m_139166_(CommandDispatcher<CommandSourceStack> p_139167_) {
-      p_139167_.register(Commands.m_82127_("weather").requires((p_139171_) -> {
-         return p_139171_.m_6761_(2);
-      }).then(Commands.m_82127_("clear").executes((p_139190_) -> {
-         return m_139172_(p_139190_.getSource(), 6000);
-      }).then(Commands.m_82129_("duration", IntegerArgumentType.integer(0, 1000000)).executes((p_139188_) -> {
-         return m_139172_(p_139188_.getSource(), IntegerArgumentType.getInteger(p_139188_, "duration") * 20);
-      }))).then(Commands.m_82127_("rain").executes((p_139186_) -> {
-         return m_139177_(p_139186_.getSource(), 6000);
-      }).then(Commands.m_82129_("duration", IntegerArgumentType.integer(0, 1000000)).executes((p_139181_) -> {
-         return m_139177_(p_139181_.getSource(), IntegerArgumentType.getInteger(p_139181_, "duration") * 20);
-      }))).then(Commands.m_82127_("thunder").executes((p_139176_) -> {
-         return m_139182_(p_139176_.getSource(), 6000);
-      }).then(Commands.m_82129_("duration", IntegerArgumentType.integer(0, 1000000)).executes((p_139169_) -> {
-         return m_139182_(p_139169_.getSource(), IntegerArgumentType.getInteger(p_139169_, "duration") * 20);
+      p_139167_.register(Commands.m_82127_("weather").requires((p_198868_0_) -> {
+         return p_198868_0_.m_6761_(2);
+      }).then(Commands.m_82127_("clear").executes((p_198861_0_) -> {
+         return m_139172_(p_198861_0_.getSource(), 6000);
+      }).then(Commands.m_82129_("duration", IntegerArgumentType.integer(0, 1000000)).executes((p_198864_0_) -> {
+         return m_139172_(p_198864_0_.getSource(), IntegerArgumentType.getInteger(p_198864_0_, "duration") * 20);
+      }))).then(Commands.m_82127_("rain").executes((p_198860_0_) -> {
+         return m_139177_(p_198860_0_.getSource(), 6000);
+      }).then(Commands.m_82129_("duration", IntegerArgumentType.integer(0, 1000000)).executes((p_198866_0_) -> {
+         return m_139177_(p_198866_0_.getSource(), IntegerArgumentType.getInteger(p_198866_0_, "duration") * 20);
+      }))).then(Commands.m_82127_("thunder").executes((p_198859_0_) -> {
+         return m_139182_(p_198859_0_.getSource(), 6000);
+      }).then(Commands.m_82129_("duration", IntegerArgumentType.integer(0, 1000000)).executes((p_198867_0_) -> {
+         return m_139182_(p_198867_0_.getSource(), IntegerArgumentType.getInteger(p_198867_0_, "duration") * 20);
       }))));
    }
 
