--- a/net/minecraft/server/commands/WorldBorderCommand.java
+++ b/net/minecraft/server/commands/WorldBorderCommand.java
@@ -25,28 +_,28 @@
    private static final SimpleCommandExceptionType f_139244_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.worldborder.damage.amount.failed"));
 
    public static void m_139246_(CommandDispatcher<CommandSourceStack> p_139247_) {
-      p_139247_.register(Commands.m_82127_("worldborder").requires((p_139268_) -> {
-         return p_139268_.m_6761_(2);
-      }).then(Commands.m_82127_("add").then(Commands.m_82129_("distance", FloatArgumentType.floatArg(-6.0E7F, 6.0E7F)).executes((p_139290_) -> {
-         return m_139252_(p_139290_.getSource(), p_139290_.getSource().m_81372_().m_6857_().m_61959_() + (double)FloatArgumentType.getFloat(p_139290_, "distance"), 0L);
-      }).then(Commands.m_82129_("time", IntegerArgumentType.integer(0)).executes((p_139288_) -> {
-         return m_139252_(p_139288_.getSource(), p_139288_.getSource().m_81372_().m_6857_().m_61959_() + (double)FloatArgumentType.getFloat(p_139288_, "distance"), p_139288_.getSource().m_81372_().m_6857_().m_61960_() + (long)IntegerArgumentType.getInteger(p_139288_, "time") * 1000L);
-      })))).then(Commands.m_82127_("set").then(Commands.m_82129_("distance", FloatArgumentType.floatArg(-6.0E7F, 6.0E7F)).executes((p_139286_) -> {
-         return m_139252_(p_139286_.getSource(), (double)FloatArgumentType.getFloat(p_139286_, "distance"), 0L);
-      }).then(Commands.m_82129_("time", IntegerArgumentType.integer(0)).executes((p_139284_) -> {
-         return m_139252_(p_139284_.getSource(), (double)FloatArgumentType.getFloat(p_139284_, "distance"), (long)IntegerArgumentType.getInteger(p_139284_, "time") * 1000L);
-      })))).then(Commands.m_82127_("center").then(Commands.m_82129_("pos", Vec2Argument.m_120822_()).executes((p_139282_) -> {
-         return m_139262_(p_139282_.getSource(), Vec2Argument.m_120825_(p_139282_, "pos"));
-      }))).then(Commands.m_82127_("damage").then(Commands.m_82127_("amount").then(Commands.m_82129_("damagePerBlock", FloatArgumentType.floatArg(0.0F)).executes((p_139280_) -> {
-         return m_139269_(p_139280_.getSource(), FloatArgumentType.getFloat(p_139280_, "damagePerBlock"));
-      }))).then(Commands.m_82127_("buffer").then(Commands.m_82129_("distance", FloatArgumentType.floatArg(0.0F)).executes((p_139278_) -> {
-         return m_139256_(p_139278_.getSource(), FloatArgumentType.getFloat(p_139278_, "distance"));
-      })))).then(Commands.m_82127_("get").executes((p_139276_) -> {
-         return m_139250_(p_139276_.getSource());
-      })).then(Commands.m_82127_("warning").then(Commands.m_82127_("distance").then(Commands.m_82129_("distance", IntegerArgumentType.integer(0)).executes((p_139266_) -> {
-         return m_139272_(p_139266_.getSource(), IntegerArgumentType.getInteger(p_139266_, "distance"));
-      }))).then(Commands.m_82127_("time").then(Commands.m_82129_("time", IntegerArgumentType.integer(0)).executes((p_139249_) -> {
-         return m_139259_(p_139249_.getSource(), IntegerArgumentType.getInteger(p_139249_, "time"));
+      p_139247_.register(Commands.m_82127_("worldborder").requires((p_198903_0_) -> {
+         return p_198903_0_.m_6761_(2);
+      }).then(Commands.m_82127_("add").then(Commands.m_82129_("distance", FloatArgumentType.floatArg(-6.0E7F, 6.0E7F)).executes((p_198908_0_) -> {
+         return m_139252_(p_198908_0_.getSource(), p_198908_0_.getSource().m_81372_().m_6857_().m_61959_() + (double)FloatArgumentType.getFloat(p_198908_0_, "distance"), 0L);
+      }).then(Commands.m_82129_("time", IntegerArgumentType.integer(0)).executes((p_198901_0_) -> {
+         return m_139252_(p_198901_0_.getSource(), p_198901_0_.getSource().m_81372_().m_6857_().m_61959_() + (double)FloatArgumentType.getFloat(p_198901_0_, "distance"), p_198901_0_.getSource().m_81372_().m_6857_().m_61960_() + (long)IntegerArgumentType.getInteger(p_198901_0_, "time") * 1000L);
+      })))).then(Commands.m_82127_("set").then(Commands.m_82129_("distance", FloatArgumentType.floatArg(-6.0E7F, 6.0E7F)).executes((p_198906_0_) -> {
+         return m_139252_(p_198906_0_.getSource(), (double)FloatArgumentType.getFloat(p_198906_0_, "distance"), 0L);
+      }).then(Commands.m_82129_("time", IntegerArgumentType.integer(0)).executes((p_198909_0_) -> {
+         return m_139252_(p_198909_0_.getSource(), (double)FloatArgumentType.getFloat(p_198909_0_, "distance"), (long)IntegerArgumentType.getInteger(p_198909_0_, "time") * 1000L);
+      })))).then(Commands.m_82127_("center").then(Commands.m_82129_("pos", Vec2Argument.m_120822_()).executes((p_198893_0_) -> {
+         return m_139262_(p_198893_0_.getSource(), Vec2Argument.m_120825_(p_198893_0_, "pos"));
+      }))).then(Commands.m_82127_("damage").then(Commands.m_82127_("amount").then(Commands.m_82129_("damagePerBlock", FloatArgumentType.floatArg(0.0F)).executes((p_198897_0_) -> {
+         return m_139269_(p_198897_0_.getSource(), FloatArgumentType.getFloat(p_198897_0_, "damagePerBlock"));
+      }))).then(Commands.m_82127_("buffer").then(Commands.m_82129_("distance", FloatArgumentType.floatArg(0.0F)).executes((p_198905_0_) -> {
+         return m_139256_(p_198905_0_.getSource(), FloatArgumentType.getFloat(p_198905_0_, "distance"));
+      })))).then(Commands.m_82127_("get").executes((p_198900_0_) -> {
+         return m_139250_(p_198900_0_.getSource());
+      })).then(Commands.m_82127_("warning").then(Commands.m_82127_("distance").then(Commands.m_82129_("distance", IntegerArgumentType.integer(0)).executes((p_198892_0_) -> {
+         return m_139272_(p_198892_0_.getSource(), IntegerArgumentType.getInteger(p_198892_0_, "distance"));
+      }))).then(Commands.m_82127_("time").then(Commands.m_82129_("time", IntegerArgumentType.integer(0)).executes((p_198907_0_) -> {
+         return m_139259_(p_198907_0_.getSource(), IntegerArgumentType.getInteger(p_198907_0_, "time"));
       })))));
    }
 
