--- a/net/minecraft/world/level/levelgen/feature/ConfiguredFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/ConfiguredFeature.java
@@ -18,8 +_,8 @@
 import org.apache.logging.log4j.Logger;
 
 public class ConfiguredFeature<FC extends FeatureConfiguration, F extends Feature<FC>> implements Decoratable<ConfiguredFeature<?, ?>> {
-   public static final Codec<ConfiguredFeature<?, ?>> f_65373_ = Registry.f_122839_.dispatch((p_65391_) -> {
-      return p_65391_.f_65377_;
+   public static final Codec<ConfiguredFeature<?, ?>> f_65373_ = Registry.f_122839_.dispatch((p_236266_0_) -> {
+      return p_236266_0_.f_65377_;
    }, Feature::m_65787_);
    public static final Codec<Supplier<ConfiguredFeature<?, ?>>> f_65374_ = RegistryFileCodec.m_135589_(Registry.f_122881_, f_65373_);
    public static final Codec<List<Supplier<ConfiguredFeature<?, ?>>>> f_65375_ = RegistryFileCodec.m_135600_(Registry.f_122881_, f_65373_);
