--- a/net/minecraft/world/level/levelgen/feature/ConfiguredStructureFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/ConfiguredStructureFeature.java
@@ -17,8 +_,8 @@
 import net.minecraft.world.level.levelgen.structure.templatesystem.StructureManager;
 
 public class ConfiguredStructureFeature<FC extends FeatureConfiguration, F extends StructureFeature<FC>> {
-   public static final Codec<ConfiguredStructureFeature<?, ?>> f_65400_ = Registry.f_122841_.dispatch((p_65410_) -> {
-      return p_65410_.f_65403_;
+   public static final Codec<ConfiguredStructureFeature<?, ?>> f_65400_ = Registry.f_122841_.dispatch((p_236271_0_) -> {
+      return p_236271_0_.f_65403_;
    }, StructureFeature::m_67097_);
    public static final Codec<Supplier<ConfiguredStructureFeature<?, ?>>> f_65401_ = RegistryFileCodec.m_135589_(Registry.f_122882_, f_65400_);
    public static final Codec<List<Supplier<ConfiguredStructureFeature<?, ?>>>> f_65402_ = RegistryFileCodec.m_135600_(Registry.f_122882_, f_65400_);
