--- a/net/minecraft/world/level/levelgen/feature/DecoratedFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/DecoratedFeature.java
@@ -17,8 +_,8 @@
 
    public boolean m_6957_(WorldGenLevel p_65497_, ChunkGenerator p_65498_, Random p_65499_, BlockPos p_65500_, DecoratedFeatureConfiguration p_65501_) {
       MutableBoolean mutableboolean = new MutableBoolean();
-      p_65501_.f_67578_.m_70480_(new DecorationContext(p_65497_, p_65498_), p_65499_, p_65500_).forEach((p_65514_) -> {
-         if (p_65501_.f_67577_.get().m_65385_(p_65497_, p_65498_, p_65499_, p_65514_)) {
+      p_65501_.f_67578_.m_70480_(new DecorationContext(p_65497_, p_65498_), p_65499_, p_65500_).forEach((p_242772_5_) -> {
+         if (p_65501_.f_67577_.get().m_65385_(p_65497_, p_65498_, p_65499_, p_242772_5_)) {
             mutableboolean.setTrue();
          }
 
