--- a/net/minecraft/world/level/levelgen/feature/Feature.java
+++ b/net/minecraft/world/level/levelgen/feature/Feature.java
@@ -8,7 +_,6 @@
 import net.minecraft.world.level.LevelWriter;
 import net.minecraft.world.level.WorldGenLevel;
 import net.minecraft.world.level.block.Block;
-import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.block.state.BlockBehaviour;
 import net.minecraft.world.level.block.state.BlockState;
 import net.minecraft.world.level.chunk.ChunkGenerator;
@@ -37,7 +_,7 @@
 import net.minecraft.world.level.levelgen.feature.configurations.SpringConfiguration;
 import net.minecraft.world.level.levelgen.feature.configurations.TreeConfiguration;
 
-public abstract class Feature<FC extends FeatureConfiguration> {
+public abstract class Feature<FC extends FeatureConfiguration> extends net.minecraftforge.registries.ForgeRegistryEntry<Feature<?>> {
    public static final Feature<NoneFeatureConfiguration> f_65759_ = m_65807_("no_op", new NoOpFeature(NoneFeatureConfiguration.f_67815_));
    public static final Feature<TreeConfiguration> f_65760_ = m_65807_("tree", new TreeFeature(TreeConfiguration.f_68184_));
    public static final AbstractFlowerFeature<RandomPatchConfiguration> f_65761_ = m_65807_("flower", new DefaultFlowerFeature(RandomPatchConfiguration.f_67902_));
@@ -119,11 +_,11 @@
    public abstract boolean m_6957_(WorldGenLevel p_65794_, ChunkGenerator p_65795_, Random p_65796_, BlockPos p_65797_, FC p_65798_);
 
    protected static boolean m_65799_(Block p_65800_) {
-      return p_65800_ == Blocks.f_50069_ || p_65800_ == Blocks.f_50122_ || p_65800_ == Blocks.f_50228_ || p_65800_ == Blocks.f_50334_;
+      return net.minecraftforge.common.Tags.Blocks.STONE.m_8110_(p_65800_);
    }
 
    public static boolean m_65813_(Block p_65814_) {
-      return p_65814_ == Blocks.f_50493_ || p_65814_ == Blocks.f_50440_ || p_65814_ == Blocks.f_50599_ || p_65814_ == Blocks.f_50546_ || p_65814_ == Blocks.f_50195_;
+      return net.minecraftforge.common.Tags.Blocks.DIRT.m_8110_(p_65814_);
    }
 
    public static boolean m_65788_(LevelSimulatedReader p_65789_, BlockPos p_65790_) {
