--- a/net/minecraft/world/level/levelgen/feature/HugeFungusFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/HugeFungusFeature.java
@@ -50,9 +_,9 @@
    }
 
    private static boolean m_65923_(LevelAccessor p_65924_, BlockPos p_65925_, boolean p_65926_) {
-      return p_65924_.m_7433_(p_65925_, (p_65966_) -> {
-         Material material = p_65966_.m_60767_();
-         return p_65966_.m_60767_().m_76336_() || p_65926_ && material == Material.f_76300_;
+      return p_65924_.m_7433_(p_65925_, (p_236320_1_) -> {
+         Material material = p_236320_1_.m_60767_();
+         return p_236320_1_.m_60767_().m_76336_() || p_65926_ && material == Material.f_76300_;
       });
    }
 
