--- a/net/minecraft/world/level/levelgen/feature/HugeBrownMushroomFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/HugeBrownMushroomFeature.java
@@ -25,7 +_,7 @@
             boolean flag5 = flag2 || flag3;
             if (!flag4 || !flag5) {
                p_65890_.m_122154_(p_65888_, j, p_65889_, k);
-               if (!p_65886_.m_8055_(p_65890_).m_60804_(p_65886_, p_65890_)) {
+               if (p_65886_.m_8055_(p_65890_).canBeReplacedByLeaves(p_65886_, p_65890_)) {
                   boolean flag6 = flag || flag5 && j == 1 - i;
                   boolean flag7 = flag1 || flag5 && j == i - 1;
                   boolean flag8 = flag2 || flag4 && k == 1 - i;
