--- a/net/minecraft/world/level/levelgen/feature/JigsawFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/JigsawFeature.java
@@ -26,8 +_,8 @@
    }
 
    public StructureFeature.StructureStartFactory<JigsawConfiguration> m_6258_() {
-      return (p_66158_, p_66159_, p_66160_, p_66161_, p_66162_, p_66163_) -> {
-         return new JigsawFeature.FeatureStart(this, p_66159_, p_66160_, p_66161_, p_66162_, p_66163_);
+      return (p_242778_1_, p_242778_2_, p_242778_3_, p_242778_4_, p_242778_5_, p_242778_6_) -> {
+         return new JigsawFeature.FeatureStart(this, p_242778_2_, p_242778_3_, p_242778_4_, p_242778_5_, p_242778_6_);
       };
    }
 
