--- a/net/minecraft/world/level/levelgen/feature/NetherFortressFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/NetherFortressFeature.java
@@ -33,7 +_,8 @@
       return NetherFortressFeature.NetherBridgeStart::new;
    }
 
-   public List<MobSpawnSettings.SpawnerData> m_7963_() {
+   @Override
+   public List<MobSpawnSettings.SpawnerData> getDefaultSpawnList() {
       return f_66381_;
    }
 
