--- a/net/minecraft/world/level/levelgen/feature/RuinedPortalFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/RuinedPortalFeature.java
@@ -67,8 +_,8 @@
       }
 
       List<BlockPos> list1 = ImmutableList.of(new BlockPos(p_66702_.f_70988_, 0, p_66702_.f_70990_), new BlockPos(p_66702_.f_70991_, 0, p_66702_.f_70990_), new BlockPos(p_66702_.f_70988_, 0, p_66702_.f_70993_), new BlockPos(p_66702_.f_70991_, 0, p_66702_.f_70993_));
-      List<BlockGetter> list = list1.stream().map((p_66672_) -> {
-         return p_66697_.m_6295_(p_66672_.m_123341_(), p_66672_.m_123343_());
+      List<BlockGetter> list = list1.stream().map((p_236333_1_) -> {
+         return p_66697_.m_6295_(p_236333_1_.m_123341_(), p_236333_1_.m_123343_());
       }).collect(Collectors.toList());
       Heightmap.Types heightmap$types = p_66698_ == RuinedPortalPiece.VerticalPlacement.ON_OCEAN_FLOOR ? Heightmap.Types.OCEAN_FLOOR_WG : Heightmap.Types.WORLD_SURFACE_WG;
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
@@ -100,6 +_,38 @@
       return p_66693_ < p_66694_ ? m_66676_(p_66692_, p_66693_, p_66694_) : p_66694_;
    }
 
+   public static enum Type implements StringRepresentable {
+      STANDARD("standard"),
+      DESERT("desert"),
+      JUNGLE("jungle"),
+      SWAMP("swamp"),
+      MOUNTAIN("mountain"),
+      OCEAN("ocean"),
+      NETHER("nether");
+
+      public static final Codec<RuinedPortalFeature.Type> f_66735_ = StringRepresentable.m_14350_(RuinedPortalFeature.Type::values, RuinedPortalFeature.Type::m_66747_);
+      private static final Map<String, RuinedPortalFeature.Type> f_66736_ = Arrays.stream(values()).collect(Collectors.toMap(RuinedPortalFeature.Type::m_66749_, (p_236345_0_) -> {
+         return p_236345_0_;
+      }));
+      private final String f_66737_;
+
+      private Type(String p_66743_) {
+         this.f_66737_ = p_66743_;
+      }
+
+      public String m_66749_() {
+         return this.f_66737_;
+      }
+
+      public static RuinedPortalFeature.Type m_66747_(String p_66748_) {
+         return f_66736_.get(p_66748_);
+      }
+
+      public String m_7912_() {
+         return this.f_66737_;
+      }
+   }
+
    public static class FeatureStart extends StructureStart<RuinedPortalConfiguration> {
       protected FeatureStart(StructureFeature<RuinedPortalConfiguration> p_66706_, int p_66707_, int p_66708_, BoundingBox p_66709_, int p_66710_, long p_66711_) {
          super(p_66706_, p_66707_, p_66708_, p_66709_, p_66710_, p_66711_);
@@ -167,38 +_,6 @@
 
          this.f_73562_.add(new RuinedPortalPiece(blockpos2, ruinedportalpiece$verticalplacement, ruinedportalpiece$properties, resourcelocation, structuretemplate, rotation, mirror, blockpos));
          this.m_5876_();
-      }
-   }
-
-   public static enum Type implements StringRepresentable {
-      STANDARD("standard"),
-      DESERT("desert"),
-      JUNGLE("jungle"),
-      SWAMP("swamp"),
-      MOUNTAIN("mountain"),
-      OCEAN("ocean"),
-      NETHER("nether");
-
-      public static final Codec<RuinedPortalFeature.Type> f_66735_ = StringRepresentable.m_14350_(RuinedPortalFeature.Type::values, RuinedPortalFeature.Type::m_66747_);
-      private static final Map<String, RuinedPortalFeature.Type> f_66736_ = Arrays.stream(values()).collect(Collectors.toMap(RuinedPortalFeature.Type::m_66749_, (p_66746_) -> {
-         return p_66746_;
-      }));
-      private final String f_66737_;
-
-      private Type(String p_66743_) {
-         this.f_66737_ = p_66743_;
-      }
-
-      public String m_66749_() {
-         return this.f_66737_;
-      }
-
-      public static RuinedPortalFeature.Type m_66747_(String p_66748_) {
-         return f_66736_.get(p_66748_);
-      }
-
-      public String m_7912_() {
-         return this.f_66737_;
       }
    }
 }
