--- a/net/minecraft/client/particle/DripParticle.java
+++ b/net/minecraft/client/particle/DripParticle.java
@@ -67,20 +_,6 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   static class CoolingDripHangParticle extends DripParticle.DripHangParticle {
-      private CoolingDripHangParticle(ClientLevel p_106068_, double p_106069_, double p_106070_, double p_106071_, Fluid p_106072_, ParticleOptions p_106073_) {
-         super(p_106068_, p_106069_, p_106070_, p_106071_, p_106072_, p_106073_);
-      }
-
-      protected void m_5956_() {
-         this.f_107227_ = 1.0F;
-         this.f_107228_ = 16.0F / (float)(40 - this.f_107225_ + 16);
-         this.f_107229_ = 4.0F / (float)(40 - this.f_107225_ + 8);
-         super.m_5956_();
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
    static class DripHangParticle extends DripParticle {
       private final ParticleOptions f_106083_;
 
@@ -107,80 +_,6 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   static class DripLandParticle extends DripParticle {
-      private DripLandParticle(ClientLevel p_106102_, double p_106103_, double p_106104_, double p_106105_, Fluid p_106106_) {
-         super(p_106102_, p_106103_, p_106104_, p_106105_, p_106106_);
-         this.f_107225_ = (int)(16.0D / (Math.random() * 0.8D + 0.2D));
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   static class FallAndLandParticle extends DripParticle.FallingParticle {
-      protected final ParticleOptions f_106114_;
-
-      private FallAndLandParticle(ClientLevel p_106116_, double p_106117_, double p_106118_, double p_106119_, Fluid p_106120_, ParticleOptions p_106121_) {
-         super(p_106116_, p_106117_, p_106118_, p_106119_, p_106120_);
-         this.f_106114_ = p_106121_;
-      }
-
-      protected void m_5949_() {
-         if (this.f_107218_) {
-            this.m_107274_();
-            this.f_107208_.m_7106_(this.f_106114_, this.f_107212_, this.f_107213_, this.f_107214_, 0.0D, 0.0D, 0.0D);
-         }
-
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   static class FallingParticle extends DripParticle {
-      private FallingParticle(ClientLevel p_106132_, double p_106133_, double p_106134_, double p_106135_, Fluid p_106136_) {
-         super(p_106132_, p_106133_, p_106134_, p_106135_, p_106136_);
-         this.f_107225_ = (int)(64.0D / (Math.random() * 0.8D + 0.2D));
-      }
-
-      protected void m_5949_() {
-         if (this.f_107218_) {
-            this.m_107274_();
-         }
-
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   static class HoneyFallAndLandParticle extends DripParticle.FallAndLandParticle {
-      private HoneyFallAndLandParticle(ClientLevel p_106146_, double p_106147_, double p_106148_, double p_106149_, Fluid p_106150_, ParticleOptions p_106151_) {
-         super(p_106146_, p_106147_, p_106148_, p_106149_, p_106150_, p_106151_);
-      }
-
-      protected void m_5949_() {
-         if (this.f_107218_) {
-            this.m_107274_();
-            this.f_107208_.m_7106_(this.f_106114_, this.f_107212_, this.f_107213_, this.f_107214_, 0.0D, 0.0D, 0.0D);
-            this.f_107208_.m_7785_(this.f_107212_ + 0.5D, this.f_107213_, this.f_107214_ + 0.5D, SoundEvents.f_11694_, SoundSource.BLOCKS, 0.3F + this.f_107208_.f_46441_.nextFloat() * 2.0F / 3.0F, 1.0F, false);
-         }
-
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   public static class HoneyFallProvider implements ParticleProvider<SimpleParticleType> {
-      protected final SpriteSet f_106161_;
-
-      public HoneyFallProvider(SpriteSet p_106163_) {
-         this.f_106161_ = p_106163_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_106174_, ClientLevel p_106175_, double p_106176_, double p_106177_, double p_106178_, double p_106179_, double p_106180_, double p_106181_) {
-         DripParticle dripparticle = new DripParticle.HoneyFallAndLandParticle(p_106175_, p_106176_, p_106177_, p_106178_, Fluids.f_76191_, ParticleTypes.f_123781_);
-         dripparticle.f_107226_ = 0.01F;
-         dripparticle.m_107253_(0.582F, 0.448F, 0.082F);
-         dripparticle.m_108335_(this.f_106161_);
-         return dripparticle;
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
    public static class HoneyHangProvider implements ParticleProvider<SimpleParticleType> {
       protected final SpriteSet f_106182_;
 
@@ -199,35 +_,16 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   public static class HoneyLandProvider implements ParticleProvider<SimpleParticleType> {
-      protected final SpriteSet f_106203_;
-
-      public HoneyLandProvider(SpriteSet p_106205_) {
-         this.f_106203_ = p_106205_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_106216_, ClientLevel p_106217_, double p_106218_, double p_106219_, double p_106220_, double p_106221_, double p_106222_, double p_106223_) {
-         DripParticle dripparticle = new DripParticle.DripLandParticle(p_106217_, p_106218_, p_106219_, p_106220_, Fluids.f_76191_);
-         dripparticle.f_107225_ = (int)(128.0D / (Math.random() * 0.8D + 0.2D));
-         dripparticle.m_107253_(0.522F, 0.408F, 0.082F);
-         dripparticle.m_108335_(this.f_106203_);
-         return dripparticle;
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   public static class LavaFallProvider implements ParticleProvider<SimpleParticleType> {
-      protected final SpriteSet f_106224_;
-
-      public LavaFallProvider(SpriteSet p_106226_) {
-         this.f_106224_ = p_106226_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_106237_, ClientLevel p_106238_, double p_106239_, double p_106240_, double p_106241_, double p_106242_, double p_106243_, double p_106244_) {
-         DripParticle dripparticle = new DripParticle.FallAndLandParticle(p_106238_, p_106239_, p_106240_, p_106241_, Fluids.f_76195_, ParticleTypes.f_123802_);
-         dripparticle.m_107253_(1.0F, 0.2857143F, 0.083333336F);
-         dripparticle.m_108335_(this.f_106224_);
-         return dripparticle;
+   static class CoolingDripHangParticle extends DripParticle.DripHangParticle {
+      private CoolingDripHangParticle(ClientLevel p_106068_, double p_106069_, double p_106070_, double p_106071_, Fluid p_106072_, ParticleOptions p_106073_) {
+         super(p_106068_, p_106069_, p_106070_, p_106071_, p_106072_, p_106073_);
+      }
+
+      protected void m_5956_() {
+         this.f_107227_ = 1.0F;
+         this.f_107228_ = 16.0F / (float)(40 - this.f_107225_ + 16);
+         this.f_107229_ = 4.0F / (float)(40 - this.f_107225_ + 8);
+         super.m_5956_();
       }
    }
 
@@ -247,18 +_,104 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   public static class LavaLandProvider implements ParticleProvider<SimpleParticleType> {
-      protected final SpriteSet f_106266_;
-
-      public LavaLandProvider(SpriteSet p_106268_) {
-         this.f_106266_ = p_106268_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_106279_, ClientLevel p_106280_, double p_106281_, double p_106282_, double p_106283_, double p_106284_, double p_106285_, double p_106286_) {
-         DripParticle dripparticle = new DripParticle.DripLandParticle(p_106280_, p_106281_, p_106282_, p_106283_, Fluids.f_76195_);
+   public static class ObsidianTearHangProvider implements ParticleProvider<SimpleParticleType> {
+      protected final SpriteSet f_106329_;
+
+      public ObsidianTearHangProvider(SpriteSet p_106331_) {
+         this.f_106329_ = p_106331_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_106342_, ClientLevel p_106343_, double p_106344_, double p_106345_, double p_106346_, double p_106347_, double p_106348_, double p_106349_) {
+         DripParticle.DripHangParticle dripparticle$driphangparticle = new DripParticle.DripHangParticle(p_106343_, p_106344_, p_106345_, p_106346_, Fluids.f_76191_, ParticleTypes.f_123787_);
+         dripparticle$driphangparticle.f_106048_ = true;
+         dripparticle$driphangparticle.f_107226_ *= 0.01F;
+         dripparticle$driphangparticle.f_107225_ = 100;
+         dripparticle$driphangparticle.m_107253_(0.51171875F, 0.03125F, 0.890625F);
+         dripparticle$driphangparticle.m_108335_(this.f_106329_);
+         return dripparticle$driphangparticle;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public static class WaterHangProvider implements ParticleProvider<SimpleParticleType> {
+      protected final SpriteSet f_106392_;
+
+      public WaterHangProvider(SpriteSet p_106394_) {
+         this.f_106392_ = p_106394_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_106405_, ClientLevel p_106406_, double p_106407_, double p_106408_, double p_106409_, double p_106410_, double p_106411_, double p_106412_) {
+         DripParticle dripparticle = new DripParticle.DripHangParticle(p_106406_, p_106407_, p_106408_, p_106409_, Fluids.f_76193_, ParticleTypes.f_123804_);
+         dripparticle.m_107253_(0.2F, 0.3F, 1.0F);
+         dripparticle.m_108335_(this.f_106392_);
+         return dripparticle;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public static class HoneyFallProvider implements ParticleProvider<SimpleParticleType> {
+      protected final SpriteSet f_106161_;
+
+      public HoneyFallProvider(SpriteSet p_106163_) {
+         this.f_106161_ = p_106163_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_106174_, ClientLevel p_106175_, double p_106176_, double p_106177_, double p_106178_, double p_106179_, double p_106180_, double p_106181_) {
+         DripParticle dripparticle = new DripParticle.HoneyFallAndLandParticle(p_106175_, p_106176_, p_106177_, p_106178_, Fluids.f_76191_, ParticleTypes.f_123781_);
+         dripparticle.f_107226_ = 0.01F;
+         dripparticle.m_107253_(0.582F, 0.448F, 0.082F);
+         dripparticle.m_108335_(this.f_106161_);
+         return dripparticle;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   static class HoneyFallAndLandParticle extends DripParticle.FallAndLandParticle {
+      private HoneyFallAndLandParticle(ClientLevel p_106146_, double p_106147_, double p_106148_, double p_106149_, Fluid p_106150_, ParticleOptions p_106151_) {
+         super(p_106146_, p_106147_, p_106148_, p_106149_, p_106150_, p_106151_);
+      }
+
+      protected void m_5949_() {
+         if (this.f_107218_) {
+            this.m_107274_();
+            this.f_107208_.m_7106_(this.f_106114_, this.f_107212_, this.f_107213_, this.f_107214_, 0.0D, 0.0D, 0.0D);
+            this.f_107208_.m_7785_(this.f_107212_ + 0.5D, this.f_107213_, this.f_107214_ + 0.5D, SoundEvents.f_11694_, SoundSource.BLOCKS, 0.3F + this.f_107208_.f_46441_.nextFloat() * 2.0F / 3.0F, 1.0F, false);
+         }
+
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public static class LavaFallProvider implements ParticleProvider<SimpleParticleType> {
+      protected final SpriteSet f_106224_;
+
+      public LavaFallProvider(SpriteSet p_106226_) {
+         this.f_106224_ = p_106226_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_106237_, ClientLevel p_106238_, double p_106239_, double p_106240_, double p_106241_, double p_106242_, double p_106243_, double p_106244_) {
+         DripParticle dripparticle = new DripParticle.FallAndLandParticle(p_106238_, p_106239_, p_106240_, p_106241_, Fluids.f_76195_, ParticleTypes.f_123802_);
          dripparticle.m_107253_(1.0F, 0.2857143F, 0.083333336F);
-         dripparticle.m_108335_(this.f_106266_);
+         dripparticle.m_108335_(this.f_106224_);
          return dripparticle;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   static class FallAndLandParticle extends DripParticle.FallingParticle {
+      protected final ParticleOptions f_106114_;
+
+      private FallAndLandParticle(ClientLevel p_106116_, double p_106117_, double p_106118_, double p_106119_, Fluid p_106120_, ParticleOptions p_106121_) {
+         super(p_106116_, p_106117_, p_106118_, p_106119_, p_106120_);
+         this.f_106114_ = p_106121_;
+      }
+
+      protected void m_5949_() {
+         if (this.f_107218_) {
+            this.m_107274_();
+            this.f_107208_.m_7106_(this.f_106114_, this.f_107212_, this.f_107213_, this.f_107214_, 0.0D, 0.0D, 0.0D);
+         }
+
       }
    }
 
@@ -281,6 +_,21 @@
    }
 
    @OnlyIn(Dist.CLIENT)
+   static class FallingParticle extends DripParticle {
+      private FallingParticle(ClientLevel p_106132_, double p_106133_, double p_106134_, double p_106135_, Fluid p_106136_) {
+         super(p_106132_, p_106133_, p_106134_, p_106135_, p_106136_);
+         this.f_107225_ = (int)(64.0D / (Math.random() * 0.8D + 0.2D));
+      }
+
+      protected void m_5949_() {
+         if (this.f_107218_) {
+            this.m_107274_();
+         }
+
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
    public static class ObsidianTearFallProvider implements ParticleProvider<SimpleParticleType> {
       protected final SpriteSet f_106308_;
 
@@ -299,21 +_,59 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   public static class ObsidianTearHangProvider implements ParticleProvider<SimpleParticleType> {
-      protected final SpriteSet f_106329_;
-
-      public ObsidianTearHangProvider(SpriteSet p_106331_) {
-         this.f_106329_ = p_106331_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_106342_, ClientLevel p_106343_, double p_106344_, double p_106345_, double p_106346_, double p_106347_, double p_106348_, double p_106349_) {
-         DripParticle.DripHangParticle dripparticle$driphangparticle = new DripParticle.DripHangParticle(p_106343_, p_106344_, p_106345_, p_106346_, Fluids.f_76191_, ParticleTypes.f_123787_);
-         dripparticle$driphangparticle.f_106048_ = true;
-         dripparticle$driphangparticle.f_107226_ *= 0.01F;
-         dripparticle$driphangparticle.f_107225_ = 100;
-         dripparticle$driphangparticle.m_107253_(0.51171875F, 0.03125F, 0.890625F);
-         dripparticle$driphangparticle.m_108335_(this.f_106329_);
-         return dripparticle$driphangparticle;
+   public static class WaterFallProvider implements ParticleProvider<SimpleParticleType> {
+      protected final SpriteSet f_106371_;
+
+      public WaterFallProvider(SpriteSet p_106373_) {
+         this.f_106371_ = p_106373_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_106384_, ClientLevel p_106385_, double p_106386_, double p_106387_, double p_106388_, double p_106389_, double p_106390_, double p_106391_) {
+         DripParticle dripparticle = new DripParticle.FallAndLandParticle(p_106385_, p_106386_, p_106387_, p_106388_, Fluids.f_76193_, ParticleTypes.f_123769_);
+         dripparticle.m_107253_(0.2F, 0.3F, 1.0F);
+         dripparticle.m_108335_(this.f_106371_);
+         return dripparticle;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   static class DripLandParticle extends DripParticle {
+      private DripLandParticle(ClientLevel p_106102_, double p_106103_, double p_106104_, double p_106105_, Fluid p_106106_) {
+         super(p_106102_, p_106103_, p_106104_, p_106105_, p_106106_);
+         this.f_107225_ = (int)(16.0D / (Math.random() * 0.8D + 0.2D));
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public static class HoneyLandProvider implements ParticleProvider<SimpleParticleType> {
+      protected final SpriteSet f_106203_;
+
+      public HoneyLandProvider(SpriteSet p_106205_) {
+         this.f_106203_ = p_106205_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_106216_, ClientLevel p_106217_, double p_106218_, double p_106219_, double p_106220_, double p_106221_, double p_106222_, double p_106223_) {
+         DripParticle dripparticle = new DripParticle.DripLandParticle(p_106217_, p_106218_, p_106219_, p_106220_, Fluids.f_76191_);
+         dripparticle.f_107225_ = (int)(128.0D / (Math.random() * 0.8D + 0.2D));
+         dripparticle.m_107253_(0.522F, 0.408F, 0.082F);
+         dripparticle.m_108335_(this.f_106203_);
+         return dripparticle;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public static class LavaLandProvider implements ParticleProvider<SimpleParticleType> {
+      protected final SpriteSet f_106266_;
+
+      public LavaLandProvider(SpriteSet p_106268_) {
+         this.f_106266_ = p_106268_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_106279_, ClientLevel p_106280_, double p_106281_, double p_106282_, double p_106283_, double p_106284_, double p_106285_, double p_106286_) {
+         DripParticle dripparticle = new DripParticle.DripLandParticle(p_106280_, p_106281_, p_106282_, p_106283_, Fluids.f_76195_);
+         dripparticle.m_107253_(1.0F, 0.2857143F, 0.083333336F);
+         dripparticle.m_108335_(this.f_106266_);
+         return dripparticle;
       }
    }
 
@@ -331,38 +_,6 @@
          dripparticle.f_107225_ = (int)(28.0D / (Math.random() * 0.8D + 0.2D));
          dripparticle.m_107253_(0.51171875F, 0.03125F, 0.890625F);
          dripparticle.m_108335_(this.f_106350_);
-         return dripparticle;
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   public static class WaterFallProvider implements ParticleProvider<SimpleParticleType> {
-      protected final SpriteSet f_106371_;
-
-      public WaterFallProvider(SpriteSet p_106373_) {
-         this.f_106371_ = p_106373_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_106384_, ClientLevel p_106385_, double p_106386_, double p_106387_, double p_106388_, double p_106389_, double p_106390_, double p_106391_) {
-         DripParticle dripparticle = new DripParticle.FallAndLandParticle(p_106385_, p_106386_, p_106387_, p_106388_, Fluids.f_76193_, ParticleTypes.f_123769_);
-         dripparticle.m_107253_(0.2F, 0.3F, 1.0F);
-         dripparticle.m_108335_(this.f_106371_);
-         return dripparticle;
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   public static class WaterHangProvider implements ParticleProvider<SimpleParticleType> {
-      protected final SpriteSet f_106392_;
-
-      public WaterHangProvider(SpriteSet p_106394_) {
-         this.f_106392_ = p_106394_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_106405_, ClientLevel p_106406_, double p_106407_, double p_106408_, double p_106409_, double p_106410_, double p_106411_, double p_106412_) {
-         DripParticle dripparticle = new DripParticle.DripHangParticle(p_106406_, p_106407_, p_106408_, p_106409_, Fluids.f_76193_, ParticleTypes.f_123804_);
-         dripparticle.m_107253_(0.2F, 0.3F, 1.0F);
-         dripparticle.m_108335_(this.f_106392_);
          return dripparticle;
       }
    }
