--- a/net/minecraft/client/particle/FireworkParticles.java
+++ b/net/minecraft/client/particle/FireworkParticles.java
@@ -22,21 +_,6 @@
 @OnlyIn(Dist.CLIENT)
 public class FireworkParticles {
    @OnlyIn(Dist.CLIENT)
-   public static class FlashProvider implements ParticleProvider<SimpleParticleType> {
-      private final SpriteSet f_106655_;
-
-      public FlashProvider(SpriteSet p_106657_) {
-         this.f_106655_ = p_106657_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_106668_, ClientLevel p_106669_, double p_106670_, double p_106671_, double p_106672_, double p_106673_, double p_106674_, double p_106675_) {
-         FireworkParticles.OverlayParticle fireworkparticles$overlayparticle = new FireworkParticles.OverlayParticle(p_106669_, p_106670_, p_106671_, p_106672_);
-         fireworkparticles$overlayparticle.m_108335_(this.f_106655_);
-         return fireworkparticles$overlayparticle;
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
    public static class OverlayParticle extends TextureSheetParticle {
       private OverlayParticle(ClientLevel p_106677_, double p_106678_, double p_106679_, double p_106680_) {
          super(p_106677_, p_106678_, p_106679_, p_106680_);
@@ -54,6 +_,21 @@
 
       public float m_5902_(float p_106693_) {
          return 7.1F * Mth.m_14031_(((float)this.f_107224_ + p_106693_ - 1.0F) * 0.25F * (float)Math.PI);
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public static class FlashProvider implements ParticleProvider<SimpleParticleType> {
+      private final SpriteSet f_106655_;
+
+      public FlashProvider(SpriteSet p_106657_) {
+         this.f_106655_ = p_106657_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_106668_, ClientLevel p_106669_, double p_106670_, double p_106671_, double p_106672_, double p_106673_, double p_106674_, double p_106675_) {
+         FireworkParticles.OverlayParticle fireworkparticles$overlayparticle = new FireworkParticles.OverlayParticle(p_106669_, p_106670_, p_106671_, p_106672_);
+         fireworkparticles$overlayparticle.m_108335_(this.f_106655_);
+         return fireworkparticles$overlayparticle;
       }
    }
 
