--- a/net/minecraft/client/particle/ParticleDescription.java
+++ b/net/minecraft/client/particle/ParticleDescription.java
@@ -29,8 +_,8 @@
       JsonArray jsonarray = GsonHelper.m_13832_(p_107286_, "textures", (JsonArray)null);
       List<ResourceLocation> list;
       if (jsonarray != null) {
-         list = Streams.stream(jsonarray).map((p_107284_) -> {
-            return GsonHelper.m_13805_(p_107284_, "texture");
+         list = Streams.stream(jsonarray).map((p_217597_0_) -> {
+            return GsonHelper.m_13805_(p_217597_0_, "texture");
          }).map(ResourceLocation::new).collect(ImmutableList.toImmutableList());
       } else {
          list = null;
