--- a/net/minecraft/client/particle/Particle.java
+++ b/net/minecraft/client/particle/Particle.java
@@ -208,4 +_,12 @@
    public void m_107259_(AABB p_107260_) {
       this.f_107207_ = p_107260_;
    }
+
+   /**
+    * Forge added method that controls if a particle should be culled to it's bounding box.
+    * Default behaviour is culling enabled
+    */
+   public boolean shouldCull() {
+      return true;
+   }
 }
