--- a/net/minecraft/client/particle/SpellParticle.java
+++ b/net/minecraft/client/particle/SpellParticle.java
@@ -73,6 +_,19 @@
    }
 
    @OnlyIn(Dist.CLIENT)
+   public static class Provider implements ParticleProvider<SimpleParticleType> {
+      private final SpriteSet f_107845_;
+
+      public Provider(SpriteSet p_107847_) {
+         this.f_107845_ = p_107847_;
+      }
+
+      public Particle m_6966_(SimpleParticleType p_107858_, ClientLevel p_107859_, double p_107860_, double p_107861_, double p_107862_, double p_107863_, double p_107864_, double p_107865_) {
+         return new SpellParticle(p_107859_, p_107860_, p_107861_, p_107862_, p_107863_, p_107864_, p_107865_, this.f_107845_);
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
    public static class InstantProvider implements ParticleProvider<SimpleParticleType> {
       private final SpriteSet f_107803_;
 
@@ -97,19 +_,6 @@
          Particle particle = new SpellParticle(p_107838_, p_107839_, p_107840_, p_107841_, p_107842_, p_107843_, p_107844_, this.f_107824_);
          particle.m_107253_((float)p_107842_, (float)p_107843_, (float)p_107844_);
          return particle;
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   public static class Provider implements ParticleProvider<SimpleParticleType> {
-      private final SpriteSet f_107845_;
-
-      public Provider(SpriteSet p_107847_) {
-         this.f_107845_ = p_107847_;
-      }
-
-      public Particle m_6966_(SimpleParticleType p_107858_, ClientLevel p_107859_, double p_107860_, double p_107861_, double p_107862_, double p_107863_, double p_107864_, double p_107865_) {
-         return new SpellParticle(p_107859_, p_107860_, p_107861_, p_107862_, p_107863_, p_107864_, p_107865_, this.f_107845_);
       }
    }
 
