--- a/net/minecraft/tags/Tag.java
+++ b/net/minecraft/tags/Tag.java
@@ -46,6 +_,7 @@
 
    public static class Builder {
       private final List<Tag.BuilderEntry> f_13302_ = Lists.newArrayList();
+      private boolean replace = false;
 
       public static Tag.Builder m_13304_() {
          return new Tag.Builder();
@@ -68,6 +_,15 @@
          return this.m_13307_(new Tag.TagEntry(p_13336_), p_13337_);
       }
 
+      public Tag.Builder replace(boolean value) {
+         this.replace = value;
+         return this;
+      }
+
+      public Tag.Builder replace() {
+         return replace(true);
+      }
+
       public <T> Optional<Tag<T>> m_13320_(Function<ResourceLocation, Tag<T>> p_13321_, Function<ResourceLocation, T> p_13322_) {
          ImmutableSet.Builder<T> builder = ImmutableSet.builder();
 
@@ -86,7 +_,7 @@
 
       public <T> Stream<Tag.BuilderEntry> m_13331_(Function<ResourceLocation, Tag<T>> p_13332_, Function<ResourceLocation, T> p_13333_) {
          return this.m_13330_().filter((p_13326_) -> {
-            return !p_13326_.m_13347_().m_7657_(p_13332_, p_13333_, (p_13316_) -> {
+            return !p_13326_.m_13347_().m_7657_(p_13332_, p_13333_, (p_232957_0_) -> {
             });
          });
       }
@@ -103,6 +_,7 @@
             this.f_13302_.clear();
          }
 
+         net.minecraftforge.common.ForgeHooks.deserializeTagAdditions(list, p_13313_, f_13302_);
          list.forEach((p_13319_) -> {
             this.f_13302_.add(new Tag.BuilderEntry(p_13319_, p_13314_));
          });
@@ -138,7 +_,7 @@
             tag$builderentry.m_13347_().m_6383_(jsonarray);
          }
 
-         jsonobject.addProperty("replace", false);
+         jsonobject.addProperty("replace", replace);
          jsonobject.add("values", jsonarray);
          return jsonobject;
       }
@@ -186,6 +_,7 @@
       public String toString() {
          return this.f_13349_.toString();
       }
+      @Override public boolean equals(Object o) { return o == this || (o instanceof Tag.ElementEntry && java.util.Objects.equals(this.f_13349_, ((Tag.ElementEntry) o).f_13349_)); }
    }
 
    public interface Entry {
@@ -278,5 +_,7 @@
       public String toString() {
          return "#" + this.f_13383_;
       }
+      @Override public boolean equals(Object o) { return o == this || (o instanceof Tag.TagEntry && java.util.Objects.equals(this.f_13383_, ((Tag.TagEntry) o).f_13383_)); }
+      public ResourceLocation getId() { return f_13383_; }
    }
 }
