--- a/net/minecraft/tags/TagLoader.java
+++ b/net/minecraft/tags/TagLoader.java
@@ -45,8 +_,8 @@
       return CompletableFuture.supplyAsync(() -> {
          Map<ResourceLocation, Tag.Builder> map = Maps.newHashMap();
 
-         for(ResourceLocation resourcelocation : p_13459_.m_6540_(this.f_13449_, (p_13462_) -> {
-            return p_13462_.endsWith(".json");
+         for(ResourceLocation resourcelocation : p_13459_.m_6540_(this.f_13449_, (p_242225_0_) -> {
+            return p_242225_0_.endsWith(".json");
          })) {
             String s = resourcelocation.m_135815_();
             ResourceLocation resourcelocation1 = new ResourceLocation(resourcelocation.m_135827_(), s.substring(this.f_13449_.length() + 1, s.length() - f_13447_));
@@ -61,7 +_,7 @@
                      if (jsonobject == null) {
                         f_13445_.error("Couldn't load {} tag list {} from {} in data pack {} as it is empty or null", this.f_13450_, resourcelocation1, resourcelocation, resource.m_7816_());
                      } else {
-                        map.computeIfAbsent(resourcelocation1, (p_13473_) -> {
+                        map.computeIfAbsent(resourcelocation1, (p_242229_0_) -> {
                            return Tag.Builder.m_13304_();
                         }).m_13312_(jsonobject, resource.m_7816_());
                      }
@@ -83,8 +_,8 @@
    public TagCollection<T> m_13463_(Map<ResourceLocation, Tag.Builder> p_13464_) {
       Map<ResourceLocation, Tag<T>> map = Maps.newHashMap();
       Function<ResourceLocation, Tag<T>> function = map::get;
-      Function<ResourceLocation, T> function1 = (p_13471_) -> {
-         return this.f_13448_.apply(p_13471_).orElse((T)null);
+      Function<ResourceLocation, T> function1 = (p_242228_1_) -> {
+         return this.f_13448_.apply(p_242228_1_).orElse((T)null);
       };
 
       while(!p_13464_.isEmpty()) {
@@ -106,8 +_,8 @@
          }
       }
 
-      p_13464_.forEach((p_13468_, p_13469_) -> {
-         f_13445_.error("Couldn't load {} tag {} as it is missing following references: {}", this.f_13450_, p_13468_, p_13469_.m_13331_(function, function1).map(Objects::toString).collect(Collectors.joining(",")));
+      p_13464_.forEach((p_242227_3_, p_242227_4_) -> {
+         f_13445_.error("Couldn't load {} tag {} as it is missing following references: {}", this.f_13450_, p_242227_3_, p_242227_4_.m_13331_(function, function1).map(Objects::toString).collect(Collectors.joining(",")));
       });
       return TagCollection.m_13396_(map);
    }
