--- a/net/minecraft/client/gui/screens/debug/GameModeSwitcherScreen.java
+++ b/net/minecraft/client/gui/screens/debug/GameModeSwitcherScreen.java
@@ -75,8 +_,8 @@
          m_93133_(p_97557_, i, j, 0.0F, 0.0F, 125, 75, 128, 128);
          p_97557_.m_85849_();
          super.m_6305_(p_97557_, p_97558_, p_97559_, p_97560_);
-         this.f_97545_.ifPresent((p_97563_) -> {
-            m_93215_(p_97557_, this.f_96547_, p_97563_.m_97597_(), this.f_96543_ / 2, this.f_96544_ / 2 - 30 - 20, -1);
+         this.f_97545_.ifPresent((p_238712_2_) -> {
+            m_93215_(p_97557_, this.f_96547_, p_238712_2_.m_97597_(), this.f_96543_ / 2, this.f_96544_ / 2 - 30 - 20, -1);
          });
          m_93215_(p_97557_, this.f_96547_, f_97543_, this.f_96543_ / 2, this.f_96544_ / 2 + 5, 16777215);
          if (!this.f_97548_) {
@@ -89,8 +_,8 @@
 
          for(GameModeSwitcherScreen.GameModeSlot gamemodeswitcherscreen$gamemodeslot : this.f_97549_) {
             gamemodeswitcherscreen$gamemodeslot.m_6305_(p_97557_, p_97558_, p_97559_, p_97560_);
-            this.f_97545_.ifPresent((p_97569_) -> {
-               gamemodeswitcherscreen$gamemodeslot.m_97643_(p_97569_ == gamemodeswitcherscreen$gamemodeslot.f_97623_);
+            this.f_97545_.ifPresent((p_238714_1_) -> {
+               gamemodeswitcherscreen$gamemodeslot.m_97643_(p_238714_1_ == gamemodeswitcherscreen$gamemodeslot.f_97623_);
             });
             if (!flag && gamemodeswitcherscreen$gamemodeslot.m_5702_()) {
                this.f_97545_ = Optional.of(gamemodeswitcherscreen$gamemodeslot.f_97623_);
