--- a/net/minecraft/world/item/context/UseOnContext.java
+++ b/net/minecraft/world/item/context/UseOnContext.java
@@ -22,7 +_,7 @@
       this(p_43709_.f_19853_, p_43709_, p_43710_, p_43709_.m_21120_(p_43710_), p_43711_);
    }
 
-   protected UseOnContext(Level p_43713_, @Nullable Player p_43714_, InteractionHand p_43715_, ItemStack p_43716_, BlockHitResult p_43717_) {
+   public UseOnContext(Level p_43713_, @Nullable Player p_43714_, InteractionHand p_43715_, ItemStack p_43716_, BlockHitResult p_43717_) {
       this.f_43703_ = p_43714_;
       this.f_43704_ = p_43715_;
       this.f_43705_ = p_43717_;
