--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/DarkOakFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/DarkOakFoliagePlacer.java
@@ -11,8 +_,8 @@
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 
 public class DarkOakFoliagePlacer extends FoliagePlacer {
-   public static final Codec<DarkOakFoliagePlacer> f_68455_ = RecordCodecBuilder.create((p_68473_) -> {
-      return m_68573_(p_68473_).apply(p_68473_, DarkOakFoliagePlacer::new);
+   public static final Codec<DarkOakFoliagePlacer> f_68455_ = RecordCodecBuilder.create((p_236746_0_) -> {
+      return m_68573_(p_236746_0_).apply(p_236746_0_, DarkOakFoliagePlacer::new);
    });
 
    public DarkOakFoliagePlacer(UniformInt p_68458_, UniformInt p_68459_) {
