--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/FancyFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/FancyFoliagePlacer.java
@@ -12,8 +_,8 @@
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 
 public class FancyFoliagePlacer extends BlobFoliagePlacer {
-   public static final Codec<FancyFoliagePlacer> f_68492_ = RecordCodecBuilder.create((p_68518_) -> {
-      return m_68413_(p_68518_).apply(p_68518_, FancyFoliagePlacer::new);
+   public static final Codec<FancyFoliagePlacer> f_68492_ = RecordCodecBuilder.create((p_236748_0_) -> {
+      return m_68413_(p_236748_0_).apply(p_236748_0_, FancyFoliagePlacer::new);
    });
 
    public FancyFoliagePlacer(UniformInt p_68495_, UniformInt p_68496_, int p_68497_) {
