--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/MegaJungleFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/MegaJungleFoliagePlacer.java
@@ -11,10 +_,10 @@
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 
 public class MegaJungleFoliagePlacer extends FoliagePlacer {
-   public static final Codec<MegaJungleFoliagePlacer> f_68608_ = RecordCodecBuilder.create((p_68630_) -> {
-      return m_68573_(p_68630_).and(Codec.intRange(0, 16).fieldOf("height").forGetter((p_68628_) -> {
-         return p_68628_.f_68609_;
-      })).apply(p_68630_, MegaJungleFoliagePlacer::new);
+   public static final Codec<MegaJungleFoliagePlacer> f_68608_ = RecordCodecBuilder.create((p_236776_0_) -> {
+      return m_68573_(p_236776_0_).and(Codec.intRange(0, 16).fieldOf("height").forGetter((p_236777_0_) -> {
+         return p_236777_0_.f_68609_;
+      })).apply(p_236776_0_, MegaJungleFoliagePlacer::new);
    });
    protected final int f_68609_;
 
