--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/MegaPineFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/MegaPineFoliagePlacer.java
@@ -12,10 +_,10 @@
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 
 public class MegaPineFoliagePlacer extends FoliagePlacer {
-   public static final Codec<MegaPineFoliagePlacer> f_68642_ = RecordCodecBuilder.create((p_68664_) -> {
-      return m_68573_(p_68664_).and(UniformInt.m_14432_(0, 16, 8).fieldOf("crown_height").forGetter((p_68662_) -> {
-         return p_68662_.f_68643_;
-      })).apply(p_68664_, MegaPineFoliagePlacer::new);
+   public static final Codec<MegaPineFoliagePlacer> f_68642_ = RecordCodecBuilder.create((p_242832_0_) -> {
+      return m_68573_(p_242832_0_).and(UniformInt.m_14432_(0, 16, 8).fieldOf("crown_height").forGetter((p_242831_0_) -> {
+         return p_242831_0_.f_68643_;
+      })).apply(p_242832_0_, MegaPineFoliagePlacer::new);
    });
    private final UniformInt f_68643_;
 
