--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/SpruceFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/SpruceFoliagePlacer.java
@@ -11,10 +_,10 @@
 import net.minecraft.world.level.levelgen.structure.BoundingBox;
 
 public class SpruceFoliagePlacer extends FoliagePlacer {
-   public static final Codec<SpruceFoliagePlacer> f_68713_ = RecordCodecBuilder.create((p_68735_) -> {
-      return m_68573_(p_68735_).and(UniformInt.m_14432_(0, 16, 8).fieldOf("trunk_height").forGetter((p_68733_) -> {
-         return p_68733_.f_68714_;
-      })).apply(p_68735_, SpruceFoliagePlacer::new);
+   public static final Codec<SpruceFoliagePlacer> f_68713_ = RecordCodecBuilder.create((p_242836_0_) -> {
+      return m_68573_(p_242836_0_).and(UniformInt.m_14432_(0, 16, 8).fieldOf("trunk_height").forGetter((p_242835_0_) -> {
+         return p_242835_0_.f_68714_;
+      })).apply(p_242836_0_, SpruceFoliagePlacer::new);
    });
    private final UniformInt f_68714_;
 
