--- a/net/minecraft/world/level/storage/loot/functions/ApplyBonusCount.java
+++ b/net/minecraft/world/level/storage/loot/functions/ApplyBonusCount.java
@@ -51,26 +_,26 @@
    }
 
    public static LootItemConditionalFunction.Builder<?> m_79917_(Enchantment p_79918_, float p_79919_, int p_79920_) {
-      return m_80683_((p_79928_) -> {
-         return new ApplyBonusCount(p_79928_, p_79918_, new ApplyBonusCount.BinomialWithBonusCount(p_79920_, p_79919_));
+      return m_80683_((p_215864_3_) -> {
+         return new ApplyBonusCount(p_215864_3_, p_79918_, new ApplyBonusCount.BinomialWithBonusCount(p_79920_, p_79919_));
       });
    }
 
    public static LootItemConditionalFunction.Builder<?> m_79915_(Enchantment p_79916_) {
-      return m_80683_((p_79943_) -> {
-         return new ApplyBonusCount(p_79943_, p_79916_, new ApplyBonusCount.OreDrops());
+      return m_80683_((p_215866_1_) -> {
+         return new ApplyBonusCount(p_215866_1_, p_79916_, new ApplyBonusCount.OreDrops());
       });
    }
 
    public static LootItemConditionalFunction.Builder<?> m_79939_(Enchantment p_79940_) {
-      return m_80683_((p_79935_) -> {
-         return new ApplyBonusCount(p_79935_, p_79940_, new ApplyBonusCount.UniformBonusCount(1));
+      return m_80683_((p_215872_1_) -> {
+         return new ApplyBonusCount(p_215872_1_, p_79940_, new ApplyBonusCount.UniformBonusCount(1));
       });
    }
 
    public static LootItemConditionalFunction.Builder<?> m_79921_(Enchantment p_79922_, int p_79923_) {
-      return m_80683_((p_79932_) -> {
-         return new ApplyBonusCount(p_79932_, p_79922_, new ApplyBonusCount.UniformBonusCount(p_79923_));
+      return m_80683_((p_215868_2_) -> {
+         return new ApplyBonusCount(p_215868_2_, p_79922_, new ApplyBonusCount.UniformBonusCount(p_79923_));
       });
    }
 
