--- a/net/minecraft/world/level/storage/loot/functions/CopyBlockState.java
+++ b/net/minecraft/world/level/storage/loot/functions/CopyBlockState.java
@@ -51,8 +_,8 @@
             compoundtag.m_128365_("BlockStateTag", compoundtag1);
          }
 
-         this.f_80048_.stream().filter(blockstate::m_61138_).forEach((p_80072_) -> {
-            compoundtag1.m_128359_(p_80072_.m_61708_(), m_80064_(blockstate, p_80072_));
+         this.f_80048_.stream().filter(blockstate::m_61138_).forEach((p_227548_2_) -> {
+            compoundtag1.m_128359_(p_227548_2_.m_61708_(), m_80064_(blockstate, p_227548_2_));
          });
       }
 
@@ -99,8 +_,8 @@
          super.m_6170_(p_80097_, p_80098_, p_80099_);
          p_80097_.addProperty("block", Registry.f_122824_.m_7981_(p_80098_.f_80047_).toString());
          JsonArray jsonarray = new JsonArray();
-         p_80098_.f_80048_.forEach((p_80091_) -> {
-            jsonarray.add(p_80091_.m_61708_());
+         p_80098_.f_80048_.forEach((p_227553_1_) -> {
+            jsonarray.add(p_227553_1_.m_61708_());
          });
          p_80097_.add("properties", jsonarray);
       }
@@ -114,8 +_,8 @@
          Set<Property<?>> set = Sets.newHashSet();
          JsonArray jsonarray = GsonHelper.m_13832_(p_80093_, "properties", (JsonArray)null);
          if (jsonarray != null) {
-            jsonarray.forEach((p_80111_) -> {
-               set.add(statedefinition.m_61081_(GsonHelper.m_13805_(p_80111_, "property")));
+            jsonarray.forEach((p_227554_2_) -> {
+               set.add(statedefinition.m_61081_(GsonHelper.m_13805_(p_227554_2_, "property")));
             });
          }
 
