--- a/net/minecraft/world/level/storage/loot/functions/LimitCount.java
+++ b/net/minecraft/world/level/storage/loot/functions/LimitCount.java
@@ -28,8 +_,8 @@
    }
 
    public static LootItemConditionalFunction.Builder<?> m_80646_(IntLimiter p_80647_) {
-      return m_80683_((p_80650_) -> {
-         return new LimitCount(p_80650_, p_80647_);
+      return m_80683_((p_215912_1_) -> {
+         return new LimitCount(p_215912_1_, p_80647_);
       });
    }
 
