--- a/net/minecraft/world/level/storage/loot/functions/LootItemConditionalFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/LootItemConditionalFunction.java
@@ -10,6 +_,7 @@
 import net.minecraft.util.GsonHelper;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.storage.loot.LootContext;
+import net.minecraft.world.level.storage.loot.Serializer;
 import net.minecraft.world.level.storage.loot.ValidationContext;
 import net.minecraft.world.level.storage.loot.predicates.ConditionUserBuilder;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
@@ -63,22 +_,6 @@
       }
    }
 
-   static final class DummyBuilder extends LootItemConditionalFunction.Builder<LootItemConditionalFunction.DummyBuilder> {
-      private final Function<LootItemCondition[], LootItemFunction> f_80700_;
-
-      public DummyBuilder(Function<LootItemCondition[], LootItemFunction> p_80702_) {
-         this.f_80700_ = p_80702_;
-      }
-
-      protected LootItemConditionalFunction.DummyBuilder m_6477_() {
-         return this;
-      }
-
-      public LootItemFunction m_7453_() {
-         return this.f_80700_.apply(this.m_80699_());
-      }
-   }
-
    public abstract static class Serializer<T extends LootItemConditionalFunction> implements net.minecraft.world.level.storage.loot.Serializer<T> {
       public void m_6170_(JsonObject p_80711_, T p_80712_, JsonSerializationContext p_80713_) {
          if (!ArrayUtils.isEmpty((Object[])p_80712_.f_80676_)) {
@@ -93,5 +_,21 @@
       }
 
       public abstract T m_6821_(JsonObject p_80721_, JsonDeserializationContext p_80722_, LootItemCondition[] p_80723_);
+   }
+
+   static final class DummyBuilder extends LootItemConditionalFunction.Builder<LootItemConditionalFunction.DummyBuilder> {
+      private final Function<LootItemCondition[], LootItemFunction> f_80700_;
+
+      public DummyBuilder(Function<LootItemCondition[], LootItemFunction> p_80702_) {
+         this.f_80700_ = p_80702_;
+      }
+
+      protected LootItemConditionalFunction.DummyBuilder m_6477_() {
+         return this;
+      }
+
+      public LootItemFunction m_7453_() {
+         return this.f_80700_.apply(this.m_80699_());
+      }
    }
 }
