--- a/net/minecraft/world/level/storage/loot/functions/LootItemFunctions.java
+++ b/net/minecraft/world/level/storage/loot/functions/LootItemFunctions.java
@@ -9,8 +_,8 @@
 import net.minecraft.world.level.storage.loot.Serializer;
 
 public class LootItemFunctions {
-   public static final BiFunction<ItemStack, LootContext, ItemStack> f_80735_ = (p_80760_, p_80761_) -> {
-      return p_80760_;
+   public static final BiFunction<ItemStack, LootContext, ItemStack> f_80735_ = (p_216240_0_, p_216240_1_) -> {
+      return p_216240_0_;
    };
    public static final LootItemFunctionType f_80736_ = m_80762_("set_count", new SetItemCountFunction.Serializer());
    public static final LootItemFunctionType f_80737_ = m_80762_("enchant_with_levels", new EnchantWithLevelsFunction.Serializer());
@@ -51,16 +_,16 @@
       case 2:
          BiFunction<ItemStack, LootContext, ItemStack> bifunction = p_80771_[0];
          BiFunction<ItemStack, LootContext, ItemStack> bifunction1 = p_80771_[1];
-         return (p_80768_, p_80769_) -> {
-            return bifunction1.apply(bifunction.apply(p_80768_, p_80769_), p_80769_);
+         return (p_216239_2_, p_216239_3_) -> {
+            return bifunction1.apply(bifunction.apply(p_216239_2_, p_216239_3_), p_216239_3_);
          };
       default:
-         return (p_80774_, p_80775_) -> {
+         return (p_216238_1_, p_216238_2_) -> {
             for(BiFunction<ItemStack, LootContext, ItemStack> bifunction2 : p_80771_) {
-               p_80774_ = bifunction2.apply(p_80774_, p_80775_);
+               p_216238_1_ = bifunction2.apply(p_216238_1_, p_216238_2_);
             }
 
-            return p_80774_;
+            return p_216238_1_;
          };
       }
    }
