--- a/net/minecraft/world/level/storage/loot/functions/SetContainerContents.java
+++ b/net/minecraft/world/level/storage/loot/functions/SetContainerContents.java
@@ -35,9 +_,9 @@
          return p_80911_;
       } else {
          NonNullList<ItemStack> nonnulllist = NonNullList.m_122779_();
-         this.f_80902_.forEach((p_80916_) -> {
-            p_80916_.m_6562_(p_80912_, (p_80924_) -> {
-               p_80924_.m_6941_(LootTable.m_79142_(nonnulllist::add), p_80912_);
+         this.f_80902_.forEach((p_215921_2_) -> {
+            p_215921_2_.m_6562_(p_80912_, (p_215922_2_) -> {
+               p_215922_2_.m_6941_(LootTable.m_79142_(nonnulllist::add), p_80912_);
             });
          });
          CompoundTag compoundtag = new CompoundTag();
