--- a/net/minecraft/world/level/storage/loot/functions/SetNbtFunction.java
+++ b/net/minecraft/world/level/storage/loot/functions/SetNbtFunction.java
@@ -30,8 +_,8 @@
    }
 
    public static LootItemConditionalFunction.Builder<?> m_81187_(CompoundTag p_81188_) {
-      return m_80683_((p_81191_) -> {
-         return new SetNbtFunction(p_81191_, p_81188_);
+      return m_80683_((p_215951_1_) -> {
+         return new SetNbtFunction(p_215951_1_, p_81188_);
       });
    }
 
