--- a/net/minecraft/world/level/block/state/predicate/BlockStatePredicate.java
+++ b/net/minecraft/world/level/block/state/predicate/BlockStatePredicate.java
@@ -11,7 +_,7 @@
 import net.minecraft.world.level.block.state.properties.Property;
 
 public class BlockStatePredicate implements Predicate<BlockState> {
-   public static final Predicate<BlockState> f_61281_ = (p_61299_) -> {
+   public static final Predicate<BlockState> f_61281_ = (p_201026_0_) -> {
       return true;
    };
    private final StateDefinition<Block, BlockState> f_61282_;
