--- a/net/minecraft/world/inventory/ContainerLevelAccess.java
+++ b/net/minecraft/world/inventory/ContainerLevelAccess.java
@@ -28,8 +_,8 @@
    }
 
    default void m_39292_(BiConsumer<Level, BlockPos> p_39293_) {
-      this.m_6721_((p_39296_, p_39297_) -> {
-         p_39293_.accept(p_39296_, p_39297_);
+      this.m_6721_((p_221487_1_, p_221487_2_) -> {
+         p_39293_.accept(p_221487_1_, p_221487_2_);
          return Optional.empty();
       });
    }
