--- a/net/minecraft/world/inventory/ItemCombinerMenu.java
+++ b/net/minecraft/world/inventory/ItemCombinerMenu.java
@@ -69,14 +_,14 @@
 
    public void m_6877_(Player p_39790_) {
       super.m_6877_(p_39790_);
-      this.f_39770_.m_39292_((p_39796_, p_39797_) -> {
-         this.m_38877_(p_39790_, p_39796_, this.f_39769_);
+      this.f_39770_.m_39292_((p_234647_2_, p_234647_3_) -> {
+         this.m_38877_(p_39790_, p_234647_2_, this.f_39769_);
       });
    }
 
    public boolean m_6875_(Player p_39780_) {
-      return this.f_39770_.m_39299_((p_39785_, p_39786_) -> {
-         return !this.m_8039_(p_39785_.m_8055_(p_39786_)) ? false : p_39780_.m_20275_((double)p_39786_.m_123341_() + 0.5D, (double)p_39786_.m_123342_() + 0.5D, (double)p_39786_.m_123343_() + 0.5D) <= 64.0D;
+      return this.f_39770_.m_39299_((p_234646_2_, p_234646_3_) -> {
+         return !this.m_8039_(p_234646_2_.m_8055_(p_234646_3_)) ? false : p_39780_.m_20275_((double)p_234646_3_.m_123341_() + 0.5D, (double)p_234646_3_.m_123342_() + 0.5D, (double)p_234646_3_.m_123343_() + 0.5D) <= 64.0D;
       }, true);
    }
 
