--- a/net/minecraft/world/inventory/LoomMenu.java
+++ b/net/minecraft/world/inventory/LoomMenu.java
@@ -76,10 +_,10 @@
                LoomMenu.this.f_39846_.m_6422_(0);
             }
 
-            p_39861_.m_39292_((p_39952_, p_39953_) -> {
-               long l = p_39952_.m_46467_();
+            p_39861_.m_39292_((p_216951_1_, p_216951_2_) -> {
+               long l = p_216951_1_.m_46467_();
                if (LoomMenu.this.f_39852_ != l) {
-                  p_39952_.m_5594_((Player)null, p_39953_, SoundEvents.f_12492_, SoundSource.BLOCKS, 1.0F, 1.0F);
+                  p_216951_1_.m_5594_((Player)null, p_216951_2_, SoundEvents.f_12492_, SoundSource.BLOCKS, 1.0F, 1.0F);
                   LoomMenu.this.f_39852_ = l;
                }
 
@@ -203,7 +_,7 @@
 
    public void m_6877_(Player p_39881_) {
       super.m_6877_(p_39881_);
-      this.f_39845_.m_39292_((p_39871_, p_39872_) -> {
+      this.f_39845_.m_39292_((p_217028_2_, p_217028_3_) -> {
          this.m_38877_(p_39881_, p_39881_.f_19853_, this.f_39853_);
       });
    }
