--- a/net/minecraft/world/inventory/RecipeBookMenu.java
+++ b/net/minecraft/world/inventory/RecipeBookMenu.java
@@ -32,6 +_,10 @@
    @OnlyIn(Dist.CLIENT)
    public abstract int m_6653_();
 
+   public java.util.List<net.minecraft.client.RecipeBookCategories> getRecipeBookCategories() {
+      return net.minecraft.client.RecipeBookCategories.m_92269_(this.m_5867_());
+   }
+
    @OnlyIn(Dist.CLIENT)
    public abstract RecipeBookType m_5867_();
 }
