--- a/net/minecraft/world/inventory/StonecutterMenu.java
+++ b/net/minecraft/world/inventory/StonecutterMenu.java
@@ -59,10 +_,10 @@
                StonecutterMenu.this.m_40342_();
             }
 
-            p_40299_.m_39292_((p_40364_, p_40365_) -> {
-               long l = p_40364_.m_46467_();
+            p_40299_.m_39292_((p_216954_1_, p_216954_2_) -> {
+               long l = p_216954_1_.m_46467_();
                if (StonecutterMenu.this.f_40290_ != l) {
-                  p_40364_.m_5594_((Player)null, p_40365_, SoundEvents.f_12494_, SoundSource.BLOCKS, 1.0F, 1.0F);
+                  p_216954_1_.m_5594_((Player)null, p_216954_2_, SoundEvents.f_12494_, SoundSource.BLOCKS, 1.0F, 1.0F);
                   StonecutterMenu.this.f_40290_ = l;
                }
 
@@ -214,7 +_,7 @@
    public void m_6877_(Player p_40326_) {
       super.m_6877_(p_40326_);
       this.f_40292_.m_8016_(1);
-      this.f_40285_.m_39292_((p_40313_, p_40314_) -> {
+      this.f_40285_.m_39292_((p_217079_2_, p_217079_3_) -> {
          this.m_38877_(p_40326_, p_40326_.f_19853_, this.f_40284_);
       });
    }
