--- a/net/minecraft/util/monitoring/jmx/MinecraftServerStatistics.java
+++ b/net/minecraft/util/monitoring/jmx/MinecraftServerStatistics.java
@@ -10,7 +_,6 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import javax.annotation.Nullable;
-import javax.management.Attribute;
 import javax.management.AttributeList;
 import javax.management.DynamicMBean;
 import javax.management.InstanceAlreadyExistsException;
@@ -31,16 +_,16 @@
    private static final Logger f_18314_ = LogManager.getLogger();
    private final MinecraftServer f_18315_;
    private final MBeanInfo f_18316_;
-   private final Map<String, MinecraftServerStatistics.AttributeDescription> f_18317_ = Stream.of(new MinecraftServerStatistics.AttributeDescription("tickTimes", this::m_18330_, "Historical tick times (ms)", long[].class), new MinecraftServerStatistics.AttributeDescription("averageTickTime", this::m_18321_, "Current average tick time (ms)", Long.TYPE)).collect(Collectors.toMap((p_18332_) -> {
-      return p_18332_.f_18346_;
+   private final Map<String, MinecraftServerStatistics.AttributeDescription> f_18317_ = Stream.of(new MinecraftServerStatistics.AttributeDescription("tickTimes", this::m_18330_, "Historical tick times (ms)", long[].class), new MinecraftServerStatistics.AttributeDescription("averageTickTime", this::m_18321_, "Current average tick time (ms)", Long.TYPE)).collect(Collectors.toMap((p_233492_0_) -> {
+      return p_233492_0_.f_18346_;
    }, Function.identity()));
 
    private MinecraftServerStatistics(MinecraftServer p_18320_) {
       this.f_18315_ = p_18320_;
-      MBeanAttributeInfo[] ambeanattributeinfo = this.f_18317_.values().stream().map((p_18327_) -> {
-         return p_18327_.m_18361_();
-      }).toArray((p_18323_) -> {
-         return new MBeanAttributeInfo[p_18323_];
+      MBeanAttributeInfo[] ambeanattributeinfo = this.f_18317_.values().stream().map((p_233489_0_) -> {
+         return p_233489_0_.m_18361_();
+      }).toArray((p_233487_0_) -> {
+         return new MBeanAttributeInfo[p_233487_0_];
       });
       this.f_18316_ = new MBeanInfo(MinecraftServerStatistics.class.getSimpleName(), "metrics for dedicated server", ambeanattributeinfo, (MBeanConstructorInfo[])null, (MBeanOperationInfo[])null, new MBeanNotificationInfo[0]);
    }
@@ -68,12 +_,12 @@
       return minecraftserverstatistics$attributedescription == null ? null : minecraftserverstatistics$attributedescription.f_18347_.get();
    }
 
-   public void setAttribute(Attribute p_18343_) {
+   public void setAttribute(javax.management.Attribute p_18343_) {
    }
 
    public AttributeList getAttributes(String[] p_18336_) {
-      List<Attribute> list = Arrays.stream(p_18336_).map(this.f_18317_::get).filter(Objects::nonNull).map((p_18325_) -> {
-         return new Attribute(p_18325_.f_18346_, p_18325_.f_18347_.get());
+      List<javax.management.Attribute> list = Arrays.stream(p_18336_).map(this.f_18317_::get).filter(Objects::nonNull).map((p_233488_0_) -> {
+         return new javax.management.Attribute(p_233488_0_.f_18346_, p_233488_0_.f_18347_.get());
       }).collect(Collectors.toList());
       return new AttributeList(list);
    }
