--- a/net/minecraft/world/level/levelgen/carver/ConfiguredWorldCarver.java
+++ b/net/minecraft/world/level/levelgen/carver/ConfiguredWorldCarver.java
@@ -13,8 +_,8 @@
 import net.minecraft.world.level.chunk.ChunkAccess;
 
 public class ConfiguredWorldCarver<WC extends CarverConfiguration> {
-   public static final Codec<ConfiguredWorldCarver<?>> f_64846_ = Registry.f_122837_.dispatch((p_64867_) -> {
-      return p_64867_.f_64849_;
+   public static final Codec<ConfiguredWorldCarver<?>> f_64846_ = Registry.f_122837_.dispatch((p_236236_0_) -> {
+      return p_236236_0_.f_64849_;
    }, WorldCarver::m_65072_);
    public static final Codec<Supplier<ConfiguredWorldCarver<?>>> f_64847_ = RegistryFileCodec.m_135589_(Registry.f_122880_, f_64846_);
    public static final Codec<List<Supplier<ConfiguredWorldCarver<?>>>> f_64848_ = RegistryFileCodec.m_135600_(Registry.f_122880_, f_64846_);
