--- a/net/minecraft/world/level/levelgen/feature/treedecorators/TrunkVineDecorator.java
+++ b/net/minecraft/world/level/levelgen/feature/treedecorators/TrunkVineDecorator.java
@@ -19,30 +_,30 @@
    }
 
    public void m_6218_(WorldGenLevel p_70061_, Random p_70062_, List<BlockPos> p_70063_, List<BlockPos> p_70064_, Set<BlockPos> p_70065_, BoundingBox p_70066_) {
-      p_70063_.forEach((p_70072_) -> {
+      p_70063_.forEach((p_236880_5_) -> {
          if (p_70062_.nextInt(3) > 0) {
-            BlockPos blockpos = p_70072_.m_122024_();
+            BlockPos blockpos = p_236880_5_.m_122024_();
             if (Feature.m_65810_(p_70061_, blockpos)) {
                this.m_70030_(p_70061_, blockpos, VineBlock.f_57835_, p_70065_, p_70066_);
             }
          }
 
          if (p_70062_.nextInt(3) > 0) {
-            BlockPos blockpos1 = p_70072_.m_122029_();
+            BlockPos blockpos1 = p_236880_5_.m_122029_();
             if (Feature.m_65810_(p_70061_, blockpos1)) {
                this.m_70030_(p_70061_, blockpos1, VineBlock.f_57837_, p_70065_, p_70066_);
             }
          }
 
          if (p_70062_.nextInt(3) > 0) {
-            BlockPos blockpos2 = p_70072_.m_122012_();
+            BlockPos blockpos2 = p_236880_5_.m_122012_();
             if (Feature.m_65810_(p_70061_, blockpos2)) {
                this.m_70030_(p_70061_, blockpos2, VineBlock.f_57836_, p_70065_, p_70066_);
             }
          }
 
          if (p_70062_.nextInt(3) > 0) {
-            BlockPos blockpos3 = p_70072_.m_122019_();
+            BlockPos blockpos3 = p_236880_5_.m_122019_();
             if (Feature.m_65810_(p_70061_, blockpos3)) {
                this.m_70030_(p_70061_, blockpos3, VineBlock.f_57834_, p_70065_, p_70066_);
             }
