--- a/net/minecraft/client/resources/sounds/BiomeAmbientSoundsHandler.java
+++ b/net/minecraft/client/resources/sounds/BiomeAmbientSoundsHandler.java
@@ -49,34 +_,34 @@
          this.f_119634_ = biome.m_47564_();
          this.f_119635_ = biome.m_47565_();
          this.f_119633_.values().forEach(BiomeAmbientSoundsHandler.LoopSoundInstance::m_119659_);
-         biome.m_47563_().ifPresent((p_119653_) -> {
-            BiomeAmbientSoundsHandler.LoopSoundInstance biomeambientsoundshandler$loopsoundinstance = this.f_119633_.compute(biome, (p_119645_, p_119646_) -> {
-               if (p_119646_ == null) {
-                  p_119646_ = new BiomeAmbientSoundsHandler.LoopSoundInstance(p_119653_);
-                  this.f_119630_.m_120367_(p_119646_);
+         biome.m_47563_().ifPresent((p_239522_2_) -> {
+            BiomeAmbientSoundsHandler.LoopSoundInstance biomeambientsoundshandler$loopsoundinstance = this.f_119633_.compute(biome, (p_239519_2_, p_239519_3_) -> {
+               if (p_239519_3_ == null) {
+                  p_239519_3_ = new BiomeAmbientSoundsHandler.LoopSoundInstance(p_239522_2_);
+                  this.f_119630_.m_120367_(p_239519_3_);
                }
 
-               p_119646_.m_119660_();
-               return p_119646_;
+               p_239519_3_.m_119660_();
+               return p_239519_3_;
             });
          });
       }
 
-      this.f_119635_.ifPresent((p_119648_) -> {
-         if (this.f_119632_.nextDouble() < p_119648_.m_47383_()) {
-            this.f_119630_.m_120367_(SimpleSoundInstance.m_119759_(p_119648_.m_47378_()));
+      this.f_119635_.ifPresent((p_239520_1_) -> {
+         if (this.f_119632_.nextDouble() < p_239520_1_.m_47383_()) {
+            this.f_119630_.m_120367_(SimpleSoundInstance.m_119759_(p_239520_1_.m_47378_()));
          }
 
       });
-      this.f_119634_.ifPresent((p_119650_) -> {
+      this.f_119634_.ifPresent((p_239521_1_) -> {
          Level level = this.f_119629_.f_19853_;
-         int i = p_119650_.m_47406_() * 2 + 1;
-         BlockPos blockpos = new BlockPos(this.f_119629_.m_20185_() + (double)this.f_119632_.nextInt(i) - (double)p_119650_.m_47406_(), this.f_119629_.m_20188_() + (double)this.f_119632_.nextInt(i) - (double)p_119650_.m_47406_(), this.f_119629_.m_20189_() + (double)this.f_119632_.nextInt(i) - (double)p_119650_.m_47406_());
+         int i = p_239521_1_.m_47406_() * 2 + 1;
+         BlockPos blockpos = new BlockPos(this.f_119629_.m_20185_() + (double)this.f_119632_.nextInt(i) - (double)p_239521_1_.m_47406_(), this.f_119629_.m_20188_() + (double)this.f_119632_.nextInt(i) - (double)p_239521_1_.m_47406_(), this.f_119629_.m_20189_() + (double)this.f_119632_.nextInt(i) - (double)p_239521_1_.m_47406_());
          int j = level.m_45517_(LightLayer.SKY, blockpos);
          if (j > 0) {
             this.f_119636_ -= (float)j / (float)level.m_7469_() * 0.001F;
          } else {
-            this.f_119636_ -= (float)(level.m_45517_(LightLayer.BLOCK, blockpos) - 1) / (float)p_119650_.m_47403_();
+            this.f_119636_ -= (float)(level.m_45517_(LightLayer.BLOCK, blockpos) - 1) / (float)p_239521_1_.m_47403_();
          }
 
          if (this.f_119636_ >= 1.0F) {
@@ -87,8 +_,8 @@
             double d4 = d1 - this.f_119629_.m_20188_();
             double d5 = d2 - this.f_119629_.m_20189_();
             double d6 = (double)Mth.m_14003_(d3 * d3 + d4 * d4 + d5 * d5);
-            double d7 = d6 + p_119650_.m_47409_();
-            SimpleSoundInstance simplesoundinstance = SimpleSoundInstance.m_119761_(p_119650_.m_47398_(), this.f_119629_.m_20185_() + d3 / d6 * d7, this.f_119629_.m_20188_() + d4 / d6 * d7, this.f_119629_.m_20189_() + d5 / d6 * d7);
+            double d7 = d6 + p_239521_1_.m_47409_();
+            SimpleSoundInstance simplesoundinstance = SimpleSoundInstance.m_119761_(p_239521_1_.m_47398_(), this.f_119629_.m_20185_() + d3 / d6 * d7, this.f_119629_.m_20188_() + d4 / d6 * d7, this.f_119629_.m_20189_() + d5 / d6 * d7);
             this.f_119630_.m_120367_(simplesoundinstance);
             this.f_119636_ = 0.0F;
          } else {
