--- a/net/minecraft/client/resources/sounds/BubbleColumnAmbientSoundHandler.java
+++ b/net/minecraft/client/resources/sounds/BubbleColumnAmbientSoundHandler.java
@@ -21,8 +_,8 @@
 
    public void m_7551_() {
       Level level = this.f_119662_.f_19853_;
-      BlockState blockstate = level.m_46847_(this.f_119662_.m_20191_().m_82377_(0.0D, (double)-0.4F, 0.0D).m_82406_(0.001D)).filter((p_119669_) -> {
-         return p_119669_.m_60713_(Blocks.f_50628_);
+      BlockState blockstate = level.m_46847_(this.f_119662_.m_20191_().m_82377_(0.0D, (double)-0.4F, 0.0D).m_82406_(0.001D)).filter((p_239528_0_) -> {
+         return p_239528_0_.m_60713_(Blocks.f_50628_);
       }).findFirst().orElse((BlockState)null);
       if (blockstate != null) {
          if (!this.f_119663_ && !this.f_119664_ && blockstate.m_60713_(Blocks.f_50628_) && !this.f_119662_.m_5833_()) {
