--- a/net/minecraft/gametest/framework/GameTestBatchRunner.java
+++ b/net/minecraft/gametest/framework/GameTestBatchRunner.java
@@ -33,16 +_,16 @@
       this.f_127552_ = p_127566_;
       this.f_127553_ = p_127567_;
       this.f_127554_ = p_127568_;
-      p_127563_.forEach((p_127575_) -> {
+      p_127563_.forEach((p_240539_3_) -> {
          Collection<GameTestInfo> collection = Lists.newArrayList();
 
-         for(TestFunction testfunction : p_127575_.m_127549_()) {
+         for(TestFunction testfunction : p_240539_3_.m_127549_()) {
             GameTestInfo gametestinfo = new GameTestInfo(testfunction, p_127565_, p_127566_);
             collection.add(gametestinfo);
             this.f_127555_.add(gametestinfo);
          }
 
-         this.f_127557_.add(Pair.of(p_127575_, collection));
+         this.f_127557_.add(Pair.of(p_240539_3_, collection));
       });
    }
 
@@ -65,8 +_,8 @@
          gametestbatch.m_127547_(this.f_127552_);
          String s = gametestbatch.m_127546_();
          f_127550_.info("Running test batch '" + s + "' (" + collection.size() + " tests)...");
-         collection.forEach((p_127585_) -> {
-            this.f_127558_.m_127809_(p_127585_);
+         collection.forEach((p_229483_1_) -> {
+            this.f_127558_.m_127809_(p_229483_1_);
             this.f_127558_.m_127811_(new GameTestListener() {
                public void m_8073_(GameTestInfo p_127590_) {
                }
@@ -75,8 +_,8 @@
                   GameTestBatchRunner.this.m_127581_(p_127592_);
                }
             });
-            BlockPos blockpos = this.f_127556_.get(p_127585_);
-            GameTestRunner.m_127742_(p_127585_, blockpos, this.f_127553_);
+            BlockPos blockpos = this.f_127556_.get(p_229483_1_);
+            GameTestRunner.m_127742_(p_229483_1_, blockpos, this.f_127553_);
          });
       }
    }
