--- a/net/minecraft/gametest/framework/GameTestAssertPosException.java
+++ b/net/minecraft/gametest/framework/GameTestAssertPosException.java
@@ -24,7 +_,7 @@
    }
 
    private GameTestAssertPosException() {
-      super((String)null);
-      throw new RuntimeException("Synthetic constructor do not call");
+      super("Synthetic constructor added by MCP, do not call");
+      throw new RuntimeException("Synthetic constructor added by MCP, do not call");
    }
 }
