--- a/net/minecraft/gametest/framework/GameTestInfo.java
+++ b/net/minecraft/gametest/framework/GameTestInfo.java
@@ -74,16 +_,16 @@
                if (this.f_127603_.isEmpty()) {
                   this.m_127622_(new GameTestTimeoutException("Didn't succeed or fail within " + this.f_127598_.m_128079_() + " ticks"));
                } else {
-                  this.f_127603_.forEach((p_127632_) -> {
-                     p_127632_.m_127779_(this.f_127606_);
+                  this.f_127603_.forEach((p_229509_1_) -> {
+                     p_229509_1_.m_127779_(this.f_127606_);
                   });
                   if (this.f_127611_ == null) {
                      this.m_127622_(new GameTestTimeoutException("No sequences finished"));
                   }
                }
             } else {
-               this.f_127603_.forEach((p_127627_) -> {
-                  p_127627_.m_127777_(this.f_127606_);
+               this.f_127603_.forEach((p_229505_1_) -> {
+                  p_229505_1_.m_127777_(this.f_127606_);
                });
             }
 
@@ -145,8 +_,8 @@
    public void m_127622_(Throwable p_127623_) {
       this.m_127650_();
       this.f_127611_ = p_127623_;
-      this.f_127601_.forEach((p_127635_) -> {
-         p_127635_.m_8066_(this);
+      this.f_127601_.forEach((p_229511_1_) -> {
+         p_229511_1_.m_8066_(this);
       });
    }
 
@@ -168,8 +_,8 @@
       this.m_127617_(structureblockentity.m_58899_());
       structureblockentity.m_59868_(this.m_127633_());
       StructureUtils.m_127875_(this.f_127599_, new BlockPos(1, 0, -1), this.m_127646_(), this.f_127600_);
-      this.f_127601_.forEach((p_127630_) -> {
-         p_127630_.m_8073_(this);
+      this.f_127601_.forEach((p_229508_1_) -> {
+         p_229508_1_.m_8073_(this);
       });
    }
 
