--- a/net/minecraft/gametest/framework/MultipleTestTracker.java
+++ b/net/minecraft/gametest/framework/MultipleTestTracker.java
@@ -24,8 +_,8 @@
 
    public void m_127811_(GameTestListener p_127812_) {
       this.f_127799_.add(p_127812_);
-      this.f_127798_.forEach((p_127815_) -> {
-         p_127815_.m_127624_(p_127812_);
+      this.f_127798_.forEach((p_240559_1_) -> {
+         p_240559_1_.m_127624_(p_127812_);
       });
    }
 
@@ -71,13 +_,13 @@
    public String m_127822_() {
       StringBuffer stringbuffer = new StringBuffer();
       stringbuffer.append('[');
-      this.f_127798_.forEach((p_127806_) -> {
-         if (!p_127806_.m_127640_()) {
+      this.f_127798_.forEach((p_229582_1_) -> {
+         if (!p_229582_1_.m_127640_()) {
             stringbuffer.append(' ');
-         } else if (p_127806_.m_127638_()) {
+         } else if (p_229582_1_.m_127638_()) {
             stringbuffer.append('+');
-         } else if (p_127806_.m_127639_()) {
-            stringbuffer.append((char)(p_127806_.m_127643_() ? 'X' : 'x'));
+         } else if (p_229582_1_.m_127639_()) {
+            stringbuffer.append((char)(p_229582_1_.m_127643_() ? 'X' : 'x'));
          } else {
             stringbuffer.append('_');
          }
