--- a/net/minecraft/gametest/framework/TestCommand.java
+++ b/net/minecraft/gametest/framework/TestCommand.java
@@ -48,54 +_,54 @@
 
 public class TestCommand {
    public static void m_127946_(CommandDispatcher<CommandSourceStack> p_127947_) {
-      p_127947_.register(Commands.m_82127_("test").then(Commands.m_82127_("runthis").executes((p_128057_) -> {
-         return m_127950_(p_128057_.getSource());
-      })).then(Commands.m_82127_("runthese").executes((p_128055_) -> {
-         return m_128001_(p_128055_.getSource());
-      })).then(Commands.m_82127_("runfailed").executes((p_128053_) -> {
-         return m_127982_(p_128053_.getSource(), false, 0, 8);
-      }).then(Commands.m_82129_("onlyRequiredTests", BoolArgumentType.bool()).executes((p_128051_) -> {
-         return m_127982_(p_128051_.getSource(), BoolArgumentType.getBool(p_128051_, "onlyRequiredTests"), 0, 8);
-      }).then(Commands.m_82129_("rotationSteps", IntegerArgumentType.integer()).executes((p_128049_) -> {
-         return m_127982_(p_128049_.getSource(), BoolArgumentType.getBool(p_128049_, "onlyRequiredTests"), IntegerArgumentType.getInteger(p_128049_, "rotationSteps"), 8);
-      }).then(Commands.m_82129_("testsPerRow", IntegerArgumentType.integer()).executes((p_128047_) -> {
-         return m_127982_(p_128047_.getSource(), BoolArgumentType.getBool(p_128047_, "onlyRequiredTests"), IntegerArgumentType.getInteger(p_128047_, "rotationSteps"), IntegerArgumentType.getInteger(p_128047_, "testsPerRow"));
-      }))))).then(Commands.m_82127_("run").then(Commands.m_82129_("testName", TestFunctionArgument.m_128088_()).executes((p_128045_) -> {
-         return m_127978_(p_128045_.getSource(), TestFunctionArgument.m_128091_(p_128045_, "testName"), 0);
-      }).then(Commands.m_82129_("rotationSteps", IntegerArgumentType.integer()).executes((p_128043_) -> {
-         return m_127978_(p_128043_.getSource(), TestFunctionArgument.m_128091_(p_128043_, "testName"), IntegerArgumentType.getInteger(p_128043_, "rotationSteps"));
-      })))).then(Commands.m_82127_("runall").executes((p_128041_) -> {
-         return m_127955_(p_128041_.getSource(), 0, 8);
-      }).then(Commands.m_82129_("testClassName", TestClassNameArgument.m_127917_()).executes((p_128039_) -> {
-         return m_127962_(p_128039_.getSource(), TestClassNameArgument.m_127920_(p_128039_, "testClassName"), 0, 8);
-      }).then(Commands.m_82129_("rotationSteps", IntegerArgumentType.integer()).executes((p_128037_) -> {
-         return m_127962_(p_128037_.getSource(), TestClassNameArgument.m_127920_(p_128037_, "testClassName"), IntegerArgumentType.getInteger(p_128037_, "rotationSteps"), 8);
-      }).then(Commands.m_82129_("testsPerRow", IntegerArgumentType.integer()).executes((p_128035_) -> {
-         return m_127962_(p_128035_.getSource(), TestClassNameArgument.m_127920_(p_128035_, "testClassName"), IntegerArgumentType.getInteger(p_128035_, "rotationSteps"), IntegerArgumentType.getInteger(p_128035_, "testsPerRow"));
-      })))).then(Commands.m_82129_("rotationSteps", IntegerArgumentType.integer()).executes((p_128033_) -> {
-         return m_127955_(p_128033_.getSource(), IntegerArgumentType.getInteger(p_128033_, "rotationSteps"), 8);
-      }).then(Commands.m_82129_("testsPerRow", IntegerArgumentType.integer()).executes((p_128031_) -> {
-         return m_127955_(p_128031_.getSource(), IntegerArgumentType.getInteger(p_128031_, "rotationSteps"), IntegerArgumentType.getInteger(p_128031_, "testsPerRow"));
-      })))).then(Commands.m_82127_("export").then(Commands.m_82129_("testName", StringArgumentType.word()).executes((p_128029_) -> {
-         return m_128010_(p_128029_.getSource(), StringArgumentType.getString(p_128029_, "testName"));
-      }))).then(Commands.m_82127_("exportthis").executes((p_128027_) -> {
-         return m_128008_(p_128027_.getSource());
-      })).then(Commands.m_82127_("import").then(Commands.m_82129_("testName", StringArgumentType.word()).executes((p_128025_) -> {
-         return m_128015_(p_128025_.getSource(), StringArgumentType.getString(p_128025_, "testName"));
-      }))).then(Commands.m_82127_("pos").executes((p_128023_) -> {
-         return m_127959_(p_128023_.getSource(), "pos");
-      }).then(Commands.m_82129_("var", StringArgumentType.word()).executes((p_128021_) -> {
-         return m_127959_(p_128021_.getSource(), StringArgumentType.getString(p_128021_, "var"));
-      }))).then(Commands.m_82127_("create").then(Commands.m_82129_("testName", StringArgumentType.word()).executes((p_128019_) -> {
-         return m_127967_(p_128019_.getSource(), StringArgumentType.getString(p_128019_, "testName"), 5, 5, 5);
-      }).then(Commands.m_82129_("width", IntegerArgumentType.integer()).executes((p_128014_) -> {
-         return m_127967_(p_128014_.getSource(), StringArgumentType.getString(p_128014_, "testName"), IntegerArgumentType.getInteger(p_128014_, "width"), IntegerArgumentType.getInteger(p_128014_, "width"), IntegerArgumentType.getInteger(p_128014_, "width"));
-      }).then(Commands.m_82129_("height", IntegerArgumentType.integer()).then(Commands.m_82129_("depth", IntegerArgumentType.integer()).executes((p_128007_) -> {
-         return m_127967_(p_128007_.getSource(), StringArgumentType.getString(p_128007_, "testName"), IntegerArgumentType.getInteger(p_128007_, "width"), IntegerArgumentType.getInteger(p_128007_, "height"), IntegerArgumentType.getInteger(p_128007_, "depth"));
-      })))))).then(Commands.m_82127_("clearall").executes((p_128000_) -> {
-         return m_127952_(p_128000_.getSource(), 200);
-      }).then(Commands.m_82129_("radius", IntegerArgumentType.integer()).executes((p_127949_) -> {
-         return m_127952_(p_127949_.getSource(), IntegerArgumentType.getInteger(p_127949_, "radius"));
+      p_127947_.register(Commands.m_82127_("test").then(Commands.m_82127_("runthis").executes((p_229647_0_) -> {
+         return m_127950_(p_229647_0_.getSource());
+      })).then(Commands.m_82127_("runthese").executes((p_229646_0_) -> {
+         return m_128001_(p_229646_0_.getSource());
+      })).then(Commands.m_82127_("runfailed").executes((p_240582_0_) -> {
+         return m_127982_(p_240582_0_.getSource(), false, 0, 8);
+      }).then(Commands.m_82129_("onlyRequiredTests", BoolArgumentType.bool()).executes((p_240585_0_) -> {
+         return m_127982_(p_240585_0_.getSource(), BoolArgumentType.getBool(p_240585_0_, "onlyRequiredTests"), 0, 8);
+      }).then(Commands.m_82129_("rotationSteps", IntegerArgumentType.integer()).executes((p_240588_0_) -> {
+         return m_127982_(p_240588_0_.getSource(), BoolArgumentType.getBool(p_240588_0_, "onlyRequiredTests"), IntegerArgumentType.getInteger(p_240588_0_, "rotationSteps"), 8);
+      }).then(Commands.m_82129_("testsPerRow", IntegerArgumentType.integer()).executes((p_240586_0_) -> {
+         return m_127982_(p_240586_0_.getSource(), BoolArgumentType.getBool(p_240586_0_, "onlyRequiredTests"), IntegerArgumentType.getInteger(p_240586_0_, "rotationSteps"), IntegerArgumentType.getInteger(p_240586_0_, "testsPerRow"));
+      }))))).then(Commands.m_82127_("run").then(Commands.m_82129_("testName", TestFunctionArgument.m_128088_()).executes((p_229645_0_) -> {
+         return m_127978_(p_229645_0_.getSource(), TestFunctionArgument.m_128091_(p_229645_0_, "testName"), 0);
+      }).then(Commands.m_82129_("rotationSteps", IntegerArgumentType.integer()).executes((p_240584_0_) -> {
+         return m_127978_(p_240584_0_.getSource(), TestFunctionArgument.m_128091_(p_240584_0_, "testName"), IntegerArgumentType.getInteger(p_240584_0_, "rotationSteps"));
+      })))).then(Commands.m_82127_("runall").executes((p_229644_0_) -> {
+         return m_127955_(p_229644_0_.getSource(), 0, 8);
+      }).then(Commands.m_82129_("testClassName", TestClassNameArgument.m_127917_()).executes((p_229643_0_) -> {
+         return m_127962_(p_229643_0_.getSource(), TestClassNameArgument.m_127920_(p_229643_0_, "testClassName"), 0, 8);
+      }).then(Commands.m_82129_("rotationSteps", IntegerArgumentType.integer()).executes((p_240580_0_) -> {
+         return m_127962_(p_240580_0_.getSource(), TestClassNameArgument.m_127920_(p_240580_0_, "testClassName"), IntegerArgumentType.getInteger(p_240580_0_, "rotationSteps"), 8);
+      }).then(Commands.m_82129_("testsPerRow", IntegerArgumentType.integer()).executes((p_240579_0_) -> {
+         return m_127962_(p_240579_0_.getSource(), TestClassNameArgument.m_127920_(p_240579_0_, "testClassName"), IntegerArgumentType.getInteger(p_240579_0_, "rotationSteps"), IntegerArgumentType.getInteger(p_240579_0_, "testsPerRow"));
+      })))).then(Commands.m_82129_("rotationSteps", IntegerArgumentType.integer()).executes((p_240569_0_) -> {
+         return m_127955_(p_240569_0_.getSource(), IntegerArgumentType.getInteger(p_240569_0_, "rotationSteps"), 8);
+      }).then(Commands.m_82129_("testsPerRow", IntegerArgumentType.integer()).executes((p_218527_0_) -> {
+         return m_127955_(p_218527_0_.getSource(), IntegerArgumentType.getInteger(p_218527_0_, "rotationSteps"), IntegerArgumentType.getInteger(p_218527_0_, "testsPerRow"));
+      })))).then(Commands.m_82127_("export").then(Commands.m_82129_("testName", StringArgumentType.word()).executes((p_229642_0_) -> {
+         return m_128010_(p_229642_0_.getSource(), StringArgumentType.getString(p_229642_0_, "testName"));
+      }))).then(Commands.m_82127_("exportthis").executes((p_240587_0_) -> {
+         return m_128008_(p_240587_0_.getSource());
+      })).then(Commands.m_82127_("import").then(Commands.m_82129_("testName", StringArgumentType.word()).executes((p_229641_0_) -> {
+         return m_128015_(p_229641_0_.getSource(), StringArgumentType.getString(p_229641_0_, "testName"));
+      }))).then(Commands.m_82127_("pos").executes((p_229640_0_) -> {
+         return m_127959_(p_229640_0_.getSource(), "pos");
+      }).then(Commands.m_82129_("var", StringArgumentType.word()).executes((p_229639_0_) -> {
+         return m_127959_(p_229639_0_.getSource(), StringArgumentType.getString(p_229639_0_, "var"));
+      }))).then(Commands.m_82127_("create").then(Commands.m_82129_("testName", StringArgumentType.word()).executes((p_229637_0_) -> {
+         return m_127967_(p_229637_0_.getSource(), StringArgumentType.getString(p_229637_0_, "testName"), 5, 5, 5);
+      }).then(Commands.m_82129_("width", IntegerArgumentType.integer()).executes((p_229635_0_) -> {
+         return m_127967_(p_229635_0_.getSource(), StringArgumentType.getString(p_229635_0_, "testName"), IntegerArgumentType.getInteger(p_229635_0_, "width"), IntegerArgumentType.getInteger(p_229635_0_, "width"), IntegerArgumentType.getInteger(p_229635_0_, "width"));
+      }).then(Commands.m_82129_("height", IntegerArgumentType.integer()).then(Commands.m_82129_("depth", IntegerArgumentType.integer()).executes((p_229632_0_) -> {
+         return m_127967_(p_229632_0_.getSource(), StringArgumentType.getString(p_229632_0_, "testName"), IntegerArgumentType.getInteger(p_229632_0_, "width"), IntegerArgumentType.getInteger(p_229632_0_, "height"), IntegerArgumentType.getInteger(p_229632_0_, "depth"));
+      })))))).then(Commands.m_82127_("clearall").executes((p_229628_0_) -> {
+         return m_127952_(p_229628_0_.getSource(), 200);
+      }).then(Commands.m_82129_("radius", IntegerArgumentType.integer()).executes((p_229614_0_) -> {
+         return m_127952_(p_229614_0_.getSource(), IntegerArgumentType.getInteger(p_229614_0_, "radius"));
       }))));
    }
 
@@ -171,8 +_,8 @@
          GameTestRunner.m_127685_(serverlevel);
          m_128003_(p_128002_, "Running " + collection.size() + " tests...");
          MultipleTestTracker multipletesttracker = new MultipleTestTracker();
-         collection.forEach((p_127943_) -> {
-            m_127929_(serverlevel, p_127943_, multipletesttracker);
+         collection.forEach((p_229626_2_) -> {
+            m_127929_(serverlevel, p_229626_2_, multipletesttracker);
          });
          return 1;
       }
@@ -284,8 +_,8 @@
       Collection<GameTestInfo> collection = GameTestRunner.m_127752_(p_127975_, blockpos1, rotation, serverlevel, GameTestTicker.f_127783_, p_127977_);
       MultipleTestTracker multipletesttracker = new MultipleTestTracker(collection);
       multipletesttracker.m_127811_(new TestCommand.TestSummaryDisplayer(serverlevel, multipletesttracker));
-      multipletesttracker.m_127807_((p_127992_) -> {
-         GameTestRegistry.m_127664_(p_127992_.m_127648_());
+      multipletesttracker.m_127807_((p_240576_0_) -> {
+         GameTestRegistry.m_127664_(p_240576_0_.m_127648_());
       });
    }
 
@@ -353,10 +_,10 @@
    }
 
    private static void m_127933_(ServerLevel p_127934_, String p_127935_, ChatFormatting p_127936_) {
-      p_127934_.m_8795_((p_127945_) -> {
+      p_127934_.m_8795_((p_229627_0_) -> {
          return true;
-      }).forEach((p_127990_) -> {
-         p_127990_.m_6352_(new TextComponent(p_127936_ + p_127935_), Util.f_137441_);
+      }).forEach((p_229621_2_) -> {
+         p_229621_2_.m_6352_(new TextComponent(p_127936_ + p_127935_), Util.f_137441_);
       });
    }
 
