--- a/net/minecraft/advancements/CriteriaTriggers.java
+++ b/net/minecraft/advancements/CriteriaTriggers.java
@@ -88,7 +_,7 @@
    public static final ItemPickedUpByEntityTrigger f_10564_ = m_10595_(new ItemPickedUpByEntityTrigger());
    public static final PlayerInteractTrigger f_10565_ = m_10595_(new PlayerInteractTrigger());
 
-   private static <T extends CriterionTrigger<?>> T m_10595_(T p_10596_) {
+   public static <T extends CriterionTrigger<?>> T m_10595_(T p_10596_) {
       if (f_10566_.containsKey(p_10596_.m_7295_())) {
          throw new IllegalArgumentException("Duplicate criterion id " + p_10596_.m_7295_());
       } else {
@@ -99,7 +_,7 @@
 
    @Nullable
    public static <T extends CriterionTriggerInstance> CriterionTrigger<T> m_10597_(ResourceLocation p_10598_) {
-      return (CriterionTrigger<T>)f_10566_.get(p_10598_);
+      return (CriterionTrigger<T>) f_10566_.get(p_10598_);
    }
 
    public static Iterable<? extends CriterionTrigger<?>> m_10594_() {
