--- a/net/minecraft/client/gui/font/providers/GlyphProviderBuilderType.java
+++ b/net/minecraft/client/gui/font/providers/GlyphProviderBuilderType.java
@@ -14,9 +_,9 @@
    TTF("ttf", TrueTypeGlyphProviderBuilder::m_95499_),
    LEGACY_UNICODE("legacy_unicode", LegacyUnicodeBitmapsProvider.Builder::m_95452_);
 
-   private static final Map<String, GlyphProviderBuilderType> f_95403_ = Util.m_137469_(Maps.newHashMap(), (p_95418_) -> {
+   private static final Map<String, GlyphProviderBuilderType> f_95403_ = Util.m_137469_(Maps.newHashMap(), (p_211639_0_) -> {
       for(GlyphProviderBuilderType glyphproviderbuildertype : values()) {
-         p_95418_.put(glyphproviderbuildertype.f_95404_, glyphproviderbuildertype);
+         p_211639_0_.put(glyphproviderbuildertype.f_95404_, glyphproviderbuildertype);
       }
 
    });
