--- a/net/minecraft/world/level/ChunkTickList.java
+++ b/net/minecraft/world/level/ChunkTickList.java
@@ -14,8 +_,8 @@
    private final Function<T, ResourceLocation> f_45632_;
 
    public ChunkTickList(Function<T, ResourceLocation> p_45637_, List<TickNextTickData<T>> p_45638_, long p_45639_) {
-      this(p_45637_, p_45638_.stream().map((p_45642_) -> {
-         return new ChunkTickList.ScheduledTick<>(p_45642_.m_47340_(), p_45642_.f_47323_, (int)(p_45642_.f_47324_ - p_45639_), p_45642_.f_47325_);
+      this(p_45637_, p_45638_.stream().map((p_234854_2_) -> {
+         return new ChunkTickList.ScheduledTick<>(p_234854_2_.m_47340_(), p_234854_2_.f_47323_, (int)(p_234854_2_.f_47324_ - p_45639_), p_234854_2_.f_47325_);
       }).collect(Collectors.toList()));
    }
 
@@ -69,8 +_,8 @@
    }
 
    public void m_45643_(TickList<T> p_45644_) {
-      this.f_45631_.forEach((p_45647_) -> {
-         p_45644_.m_7663_(p_45647_.f_45664_, p_45647_.f_45667_, p_45647_.f_45665_, p_45647_.f_45666_);
+      this.f_45631_.forEach((p_234856_1_) -> {
+         p_45644_.m_7663_(p_234856_1_.f_45664_, p_234856_1_.f_45667_, p_234856_1_.f_45665_, p_234856_1_.f_45666_);
       });
    }
 
