--- a/net/minecraft/world/level/CollisionSpliterator.java
+++ b/net/minecraft/world/level/CollisionSpliterator.java
@@ -31,7 +_,7 @@
    private final BiPredicate<BlockState, BlockPos> f_45796_;
 
    public CollisionSpliterator(CollisionGetter p_45798_, @Nullable Entity p_45799_, AABB p_45800_) {
-      this(p_45798_, p_45799_, p_45800_, (p_45810_, p_45811_) -> {
+      this(p_45798_, p_45799_, p_45800_, (p_241459_0_, p_241459_1_) -> {
          return true;
       });
    }
