--- a/net/minecraft/world/level/CollisionGetter.java
+++ b/net/minecraft/world/level/CollisionGetter.java
@@ -36,19 +_,19 @@
    }
 
    default boolean m_45772_(AABB p_45773_) {
-      return this.m_45768_((Entity)null, p_45773_, (p_45780_) -> {
+      return this.m_45768_((Entity)null, p_45773_, (p_234866_0_) -> {
          return true;
       });
    }
 
    default boolean m_45786_(Entity p_45787_) {
-      return this.m_45768_(p_45787_, p_45787_.m_20191_(), (p_45760_) -> {
+      return this.m_45768_(p_45787_, p_45787_.m_20191_(), (p_234864_0_) -> {
          return true;
       });
    }
 
    default boolean m_45756_(Entity p_45757_, AABB p_45758_) {
-      return this.m_45768_(p_45757_, p_45758_, (p_45745_) -> {
+      return this.m_45768_(p_45757_, p_45758_, (p_234863_0_) -> {
          return true;
       });
    }
