--- a/net/minecraft/world/level/EntityGetter.java
+++ b/net/minecraft/world/level/EntityGetter.java
@@ -58,13 +_,13 @@
          return Stream.empty();
       } else {
          AABB aabb = p_46006_.m_82400_(1.0E-7D);
-         return this.m_6249_(p_46005_, aabb, p_46007_.and((p_45962_) -> {
-            if (p_45962_.m_20191_().m_82381_(aabb)) {
+         return this.m_6249_(p_46005_, aabb, p_46007_.and((p_234892_2_) -> {
+            if (p_234892_2_.m_20191_().m_82381_(aabb)) {
                if (p_46005_ == null) {
-                  if (p_45962_.m_5829_()) {
+                  if (p_234892_2_.m_5829_()) {
                      return true;
                   }
-               } else if (p_46005_.m_7337_(p_45962_)) {
+               } else if (p_46005_.m_7337_(p_234892_2_)) {
                   return true;
                }
             }
