--- a/net/minecraft/world/level/GameRules.java
+++ b/net/minecraft/world/level/GameRules.java
@@ -29,8 +_,8 @@
 
 public class GameRules {
    private static final Logger f_46128_ = LogManager.getLogger();
-   private static final Map<GameRules.Key<?>, GameRules.Type<?>> f_46129_ = Maps.newTreeMap(Comparator.comparing((p_46218_) -> {
-      return p_46218_.f_46323_;
+   private static final Map<GameRules.Key<?>, GameRules.Type<?>> f_46129_ = Maps.newTreeMap(Comparator.comparing((p_223597_0_) -> {
+      return p_223597_0_.f_46323_;
    }));
    public static final GameRules.Key<GameRules.BooleanValue> f_46131_ = m_46189_("doFireTick", GameRules.Category.UPDATES, GameRules.BooleanValue.m_46250_(true));
    public static final GameRules.Key<GameRules.BooleanValue> f_46132_ = m_46189_("mobGriefing", GameRules.Category.MOBS, GameRules.BooleanValue.m_46250_(true));
@@ -46,10 +_,10 @@
    public static final GameRules.Key<GameRules.BooleanValue> f_46142_ = m_46189_("showDeathMessages", GameRules.Category.CHAT, GameRules.BooleanValue.m_46250_(true));
    public static final GameRules.Key<GameRules.IntegerValue> f_46143_ = m_46189_("randomTickSpeed", GameRules.Category.UPDATES, GameRules.IntegerValue.m_46312_(3));
    public static final GameRules.Key<GameRules.BooleanValue> f_46144_ = m_46189_("sendCommandFeedback", GameRules.Category.CHAT, GameRules.BooleanValue.m_46250_(true));
-   public static final GameRules.Key<GameRules.BooleanValue> f_46145_ = m_46189_("reducedDebugInfo", GameRules.Category.MISC, GameRules.BooleanValue.m_46252_(false, (p_46212_, p_46213_) -> {
-      byte b0 = (byte)(p_46213_.m_46223_() ? 22 : 23);
+   public static final GameRules.Key<GameRules.BooleanValue> f_46145_ = m_46189_("reducedDebugInfo", GameRules.Category.MISC, GameRules.BooleanValue.m_46252_(false, (p_223589_0_, p_223589_1_) -> {
+      byte b0 = (byte)(p_223589_1_.m_46223_() ? 22 : 23);
 
-      for(ServerPlayer serverplayer : p_46212_.m_6846_().m_11314_()) {
+      for(ServerPlayer serverplayer : p_223589_0_.m_6846_().m_11314_()) {
          serverplayer.f_8906_.m_9829_(new ClientboundEntityEventPacket(serverplayer, b0));
       }
 
@@ -64,9 +_,9 @@
    public static final GameRules.Key<GameRules.BooleanValue> f_46153_ = m_46189_("announceAdvancements", GameRules.Category.CHAT, GameRules.BooleanValue.m_46250_(true));
    public static final GameRules.Key<GameRules.BooleanValue> f_46154_ = m_46189_("disableRaids", GameRules.Category.MOBS, GameRules.BooleanValue.m_46250_(false));
    public static final GameRules.Key<GameRules.BooleanValue> f_46155_ = m_46189_("doInsomnia", GameRules.Category.SPAWNING, GameRules.BooleanValue.m_46250_(true));
-   public static final GameRules.Key<GameRules.BooleanValue> f_46156_ = m_46189_("doImmediateRespawn", GameRules.Category.PLAYER, GameRules.BooleanValue.m_46252_(false, (p_46200_, p_46201_) -> {
-      for(ServerPlayer serverplayer : p_46200_.m_6846_().m_11314_()) {
-         serverplayer.f_8906_.m_9829_(new ClientboundGameEventPacket(ClientboundGameEventPacket.f_132164_, p_46201_.m_46223_() ? 1.0F : 0.0F));
+   public static final GameRules.Key<GameRules.BooleanValue> f_46156_ = m_46189_("doImmediateRespawn", GameRules.Category.PLAYER, GameRules.BooleanValue.m_46252_(false, (p_226686_0_, p_226686_1_) -> {
+      for(ServerPlayer serverplayer : p_226686_0_.m_6846_().m_11314_()) {
+         serverplayer.f_8906_.m_9829_(new ClientboundGameEventPacket(ClientboundGameEventPacket.f_132164_, p_226686_1_.m_46223_() ? 1.0F : 0.0F));
       }
 
    }));
@@ -79,7 +_,7 @@
    public static final GameRules.Key<GameRules.BooleanValue> f_46127_ = m_46189_("universalAnger", GameRules.Category.MOBS, GameRules.BooleanValue.m_46250_(false));
    private final Map<GameRules.Key<?>, GameRules.Value<?>> f_46130_;
 
-   private static <T extends GameRules.Value<T>> GameRules.Key<T> m_46189_(String p_46190_, GameRules.Category p_46191_, GameRules.Type<T> p_46192_) {
+   public static <T extends GameRules.Value<T>> GameRules.Key<T> m_46189_(String p_46190_, GameRules.Category p_46191_, GameRules.Type<T> p_46192_) {
       GameRules.Key<T> key = new GameRules.Key<>(p_46190_, p_46191_);
       GameRules.Type<?> type = f_46129_.put(key, p_46192_);
       if (type != null) {
@@ -95,8 +_,8 @@
    }
 
    public GameRules() {
-      this.f_46130_ = f_46129_.entrySet().stream().collect(ImmutableMap.toImmutableMap(Entry::getKey, (p_46210_) -> {
-         return p_46210_.getValue().m_46352_();
+      this.f_46130_ = f_46129_.entrySet().stream().collect(ImmutableMap.toImmutableMap(Entry::getKey, (p_226684_0_) -> {
+         return p_226684_0_.getValue().m_46352_();
       }));
    }
 
@@ -110,39 +_,39 @@
 
    public CompoundTag m_46163_() {
       CompoundTag compoundtag = new CompoundTag();
-      this.f_46130_.forEach((p_46197_, p_46198_) -> {
-         compoundtag.m_128359_(p_46197_.f_46323_, p_46198_.m_5831_());
+      this.f_46130_.forEach((p_226688_1_, p_226688_2_) -> {
+         compoundtag.m_128359_(p_226688_1_.f_46323_, p_226688_2_.m_5831_());
       });
       return compoundtag;
    }
 
    private void m_46183_(DynamicLike<?> p_46184_) {
-      this.f_46130_.forEach((p_46187_, p_46188_) -> {
-         p_46184_.get(p_46187_.f_46323_).asString().result().ifPresent(p_46188_::m_7377_);
+      this.f_46130_.forEach((p_234902_1_, p_234902_2_) -> {
+         p_46184_.get(p_234902_1_.f_46323_).asString().result().ifPresent(p_234902_2_::m_7377_);
       });
    }
 
    public GameRules m_46202_() {
-      return new GameRules(this.f_46130_.entrySet().stream().collect(ImmutableMap.toImmutableMap(Entry::getKey, (p_46194_) -> {
-         return p_46194_.getValue().m_5590_();
+      return new GameRules(this.f_46130_.entrySet().stream().collect(ImmutableMap.toImmutableMap(Entry::getKey, (p_234904_0_) -> {
+         return p_234904_0_.getValue().m_5590_();
       })));
    }
 
    public static void m_46164_(GameRules.GameRuleTypeVisitor p_46165_) {
-      f_46129_.forEach((p_46205_, p_46206_) -> {
-         m_46166_(p_46165_, p_46205_, p_46206_);
+      f_46129_.forEach((p_234906_1_, p_234906_2_) -> {
+         m_46166_(p_46165_, p_234906_1_, p_234906_2_);
       });
    }
 
    private static <T extends GameRules.Value<T>> void m_46166_(GameRules.GameRuleTypeVisitor p_46167_, GameRules.Key<?> p_46168_, GameRules.Type<?> p_46169_) {
       p_46167_.m_6889_((GameRules.Key)p_46168_, p_46169_);
-      p_46169_.m_46353_(p_46167_, (GameRules.Key)p_46168_);
+      p_46169_.m_46353_(p_46167_,(GameRules.Key) p_46168_);
    }
 
    @OnlyIn(Dist.CLIENT)
    public void m_46176_(GameRules p_46177_, @Nullable MinecraftServer p_46178_) {
-      p_46177_.f_46130_.keySet().forEach((p_46182_) -> {
-         this.m_46172_(p_46182_, p_46177_, p_46178_);
+      p_46177_.f_46130_.keySet().forEach((p_234900_3_) -> {
+         this.m_46172_(p_234900_3_, p_46177_, p_46178_);
       });
    }
 
@@ -164,13 +_,13 @@
       private boolean f_46219_;
 
       private static GameRules.Type<GameRules.BooleanValue> m_46252_(boolean p_46253_, BiConsumer<MinecraftServer, GameRules.BooleanValue> p_46254_) {
-         return new GameRules.Type<>(BoolArgumentType::bool, (p_46242_) -> {
-            return new GameRules.BooleanValue(p_46242_, p_46253_);
+         return new GameRules.Type<>(BoolArgumentType::bool, (p_223574_1_) -> {
+            return new GameRules.BooleanValue(p_223574_1_, p_46253_);
          }, p_46254_, GameRules.GameRuleTypeVisitor::m_6891_);
       }
 
       private static GameRules.Type<GameRules.BooleanValue> m_46250_(boolean p_46251_) {
-         return m_46252_(p_46251_, (p_46236_, p_46237_) -> {
+         return m_46252_(p_46251_, (p_223569_0_, p_223569_1_) -> {
          });
       }
 
@@ -240,6 +_,10 @@
       }
    }
 
+   interface VisitorCaller<T extends GameRules.Value<T>> {
+      void m_46374_(GameRules.GameRuleTypeVisitor p_46375_, GameRules.Key<T> p_46376_, GameRules.Type<T> p_46377_);
+   }
+
    public interface GameRuleTypeVisitor {
       default <T extends GameRules.Value<T>> void m_6889_(GameRules.Key<T> p_46278_, GameRules.Type<T> p_46279_) {
       }
@@ -255,13 +_,13 @@
       private int f_46284_;
 
       private static GameRules.Type<GameRules.IntegerValue> m_46294_(int p_46295_, BiConsumer<MinecraftServer, GameRules.IntegerValue> p_46296_) {
-         return new GameRules.Type<>(IntegerArgumentType::integer, (p_46293_) -> {
-            return new GameRules.IntegerValue(p_46293_, p_46295_);
+         return new GameRules.Type<>(IntegerArgumentType::integer, (p_223565_1_) -> {
+            return new GameRules.IntegerValue(p_223565_1_, p_46295_);
          }, p_46296_, GameRules.GameRuleTypeVisitor::m_6894_);
       }
 
       private static GameRules.Type<GameRules.IntegerValue> m_46312_(int p_46313_) {
-         return m_46294_(p_46313_, (p_46309_, p_46310_) -> {
+         return m_46294_(p_46313_, (p_223561_0_, p_223561_1_) -> {
          });
       }
 
@@ -429,9 +_,5 @@
 
       @OnlyIn(Dist.CLIENT)
       public abstract void m_5614_(T p_46363_, @Nullable MinecraftServer p_46364_);
-   }
-
-   interface VisitorCaller<T extends GameRules.Value<T>> {
-      void m_46374_(GameRules.GameRuleTypeVisitor p_46375_, GameRules.Key<T> p_46376_, GameRules.Type<T> p_46377_);
    }
 }
