--- a/net/minecraft/world/level/NaturalSpawner.java
+++ b/net/minecraft/world/level/NaturalSpawner.java
@@ -76,7 +_,7 @@
          } while(mob.m_21532_() || mob.m_8023_());
 
          final Entity entity_f = entity;
-         MobCategory mobcategory = entity.m_6095_().m_20674_();
+         MobCategory mobcategory = entity.getClassification(true);
          if (mobcategory != MobCategory.MISC) {
             BlockPos blockpos = entity.m_20183_();
             long i = ChunkPos.m_45589_(blockpos.m_123341_() >> 4, blockpos.m_123343_() >> 4);
@@ -163,13 +_,15 @@
                         }
 
                         mob.m_7678_(d0, (double)i, d1, p_47040_.f_46441_.nextFloat() * 360.0F, 0.0F);
-                        if (m_46991_(p_47040_, mob, d2)) {
+                        int canSpawn = net.minecraftforge.common.ForgeHooks.canEntitySpawn(mob, p_47040_, d0, i, d1, null, MobSpawnType.NATURAL);
+                        if (canSpawn != -1 && (canSpawn == 1 || m_46991_(p_47040_, mob, d2))) {
+                           if (!net.minecraftforge.event.ForgeEventFactory.doSpecialSpawn(mob, p_47040_, (float)d0, (float)i, (float)d1, null, MobSpawnType.NATURAL))
                            spawngroupdata = mob.m_6518_(p_47040_, p_47040_.m_6436_(mob.m_20183_()), MobSpawnType.NATURAL, spawngroupdata, (CompoundTag)null);
                            ++j;
                            ++l1;
                            p_47040_.m_47205_(mob);
                            p_47044_.m_47100_(mob, p_47041_);
-                           if (j >= mob.m_5792_()) {
+                           if (j >= net.minecraftforge.event.ForgeEventFactory.getMaxSpawnPackSize(mob)) {
                               return;
                            }
 
@@ -247,6 +_,7 @@
          return null;
       } else {
          List<MobSpawnSettings.SpawnerData> list = m_47010_(p_47018_, p_47019_, p_47020_, p_47021_, p_47023_, biome);
+         list = net.minecraftforge.event.ForgeEventFactory.getPotentialSpawns(p_47018_, p_47021_, p_47023_, list);
          return list.isEmpty() ? null : WeighedRandom.m_14475_(p_47022_, list);
       }
    }
@@ -286,6 +_,13 @@
       if (p_47052_ == SpawnPlacements.Type.NO_RESTRICTIONS) {
          return true;
       } else if (p_47055_ != null && p_47053_.m_6857_().m_61937_(p_47054_)) {
+         return p_47052_.canSpawnAt(p_47053_, p_47054_, p_47055_);
+      }
+      return false;
+   }
+
+   public static boolean canSpawnAtBody(SpawnPlacements.Type p_47052_, LevelReader p_47053_, BlockPos p_47054_, @Nullable EntityType<?> p_47055_) {
+      {
          BlockState blockstate = p_47053_.m_8055_(p_47054_);
          FluidState fluidstate = p_47053_.m_6425_(p_47054_);
          BlockPos blockpos = p_47054_.m_7494_();
@@ -298,14 +_,12 @@
          case ON_GROUND:
          default:
             BlockState blockstate1 = p_47053_.m_8055_(blockpos1);
-            if (!blockstate1.m_60643_(p_47053_, blockpos1, p_47055_)) {
+            if (!blockstate1.canCreatureSpawn(p_47053_, blockpos1, p_47052_, p_47055_)) {
                return false;
             } else {
                return m_47056_(p_47053_, p_47054_, blockstate, fluidstate, p_47055_) && m_47056_(p_47053_, blockpos, p_47053_.m_8055_(blockpos), p_47053_.m_6425_(blockpos), p_47055_);
             }
          }
-      } else {
-         return false;
       }
    }
 
@@ -349,6 +_,7 @@
                      entity.m_7678_(d0, (double)blockpos.m_123342_(), d1, p_47084_.nextFloat() * 360.0F, 0.0F);
                      if (entity instanceof Mob) {
                         Mob mob = (Mob)entity;
+                        if (net.minecraftforge.common.ForgeHooks.canEntitySpawn(mob, p_47080_, d0, blockpos.m_123342_(), d1, null, MobSpawnType.CHUNK_GENERATION) == -1) continue;
                         if (mob.m_5545_(p_47080_, MobSpawnType.CHUNK_GENERATION) && mob.m_6914_(p_47080_)) {
                            spawngroupdata = mob.m_6518_(p_47080_, p_47080_.m_6436_(mob.m_20183_()), MobSpawnType.CHUNK_GENERATION, spawngroupdata, (CompoundTag)null);
                            p_47080_.m_47205_(mob);
