--- a/net/minecraft/world/level/TickNextTickData.java
+++ b/net/minecraft/world/level/TickNextTickData.java
@@ -37,12 +_,12 @@
    }
 
    public static <T> Comparator<TickNextTickData<T>> m_47337_() {
-      return Comparator.<TickNextTickData<T>>comparingLong((p_47344_) -> {
-         return p_47344_.f_47324_;
-      }).thenComparing((p_47342_) -> {
-         return p_47342_.f_47325_;
-      }).thenComparingLong((p_47339_) -> {
-         return p_47339_.f_47328_;
+      return Comparator.<TickNextTickData<T>>comparingLong((p_226710_0_) -> {
+         return p_226710_0_.f_47324_;
+      }).thenComparing((p_226709_0_) -> {
+         return p_226709_0_.f_47325_;
+      }).thenComparingLong((p_226708_0_) -> {
+         return p_226708_0_.f_47328_;
       });
    }
 
