--- a/net/minecraft/world/level/storage/CommandStorage.java
+++ b/net/minecraft/world/level/storage/CommandStorage.java
@@ -39,8 +_,8 @@
    }
 
    public Stream<ResourceLocation> m_78036_() {
-      return this.f_78032_.entrySet().stream().flatMap((p_78043_) -> {
-         return p_78043_.getValue().m_78072_(p_78043_.getKey());
+      return this.f_78032_.entrySet().stream().flatMap((p_227487_0_) -> {
+         return p_227487_0_.getValue().m_78072_(p_227487_0_.getKey());
       });
    }
 
@@ -66,8 +_,8 @@
 
       public CompoundTag m_7176_(CompoundTag p_78075_) {
          CompoundTag compoundtag = new CompoundTag();
-         this.f_78055_.forEach((p_78070_, p_78071_) -> {
-            compoundtag.m_128365_(p_78070_, p_78071_.m_6426_());
+         this.f_78055_.forEach((p_227496_1_, p_227496_2_) -> {
+            compoundtag.m_128365_(p_227496_1_, p_227496_2_.m_6426_());
          });
          p_78075_.m_128365_("contents", compoundtag);
          return p_78075_;
@@ -89,8 +_,8 @@
       }
 
       public Stream<ResourceLocation> m_78072_(String p_78073_) {
-         return this.f_78055_.keySet().stream().map((p_78062_) -> {
-            return new ResourceLocation(p_78073_, p_78062_);
+         return this.f_78055_.keySet().stream().map((p_227494_1_) -> {
+            return new ResourceLocation(p_78073_, p_227494_1_);
          });
       }
    }
