--- a/net/minecraft/world/level/storage/LevelSummary.java
+++ b/net/minecraft/world/level/storage/LevelSummary.java
@@ -8,7 +_,6 @@
 import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.network.chat.TextComponent;
 import net.minecraft.network.chat.TranslatableComponent;
-import net.minecraft.util.StringUtil;
 import net.minecraft.world.level.GameType;
 import net.minecraft.world.level.LevelSettings;
 import net.minecraftforge.api.distmarker.Dist;
@@ -85,7 +_,7 @@
 
    @OnlyIn(Dist.CLIENT)
    public MutableComponent m_78370_() {
-      return (MutableComponent)(StringUtil.m_14408_(this.f_78345_.m_78393_()) ? new TranslatableComponent("selectWorld.versionUnknown") : new TextComponent(this.f_78345_.m_78393_()));
+      return (MutableComponent)(net.minecraft.util.StringUtil.m_14408_(this.f_78345_.m_78393_()) ? new TranslatableComponent("selectWorld.versionUnknown") : new TextComponent(this.f_78345_.m_78393_()));
    }
 
    public LevelVersion m_78371_() {
