--- a/net/minecraft/world/level/storage/McRegionUpgrader.java
+++ b/net/minecraft/world/level/storage/McRegionUpgrader.java
@@ -147,8 +_,8 @@
 
    private static void m_78422_(File p_78423_, Collection<File> p_78424_) {
       File file1 = new File(p_78423_, "region");
-      File[] afile = file1.listFiles((p_78420_, p_78421_) -> {
-         return p_78421_.endsWith(".mcr");
+      File[] afile = file1.listFiles((p_215791_0_, p_215791_1_) -> {
+         return p_215791_1_.endsWith(".mcr");
       });
       if (afile != null) {
          Collections.addAll(p_78424_, afile);
