--- a/net/minecraft/world/entity/ai/village/poi/PoiRecord.java
+++ b/net/minecraft/world/entity/ai/village/poi/PoiRecord.java
@@ -13,14 +_,14 @@
    private final Runnable f_27230_;
 
    public static Codec<PoiRecord> m_27242_(Runnable p_27243_) {
-      return RecordCodecBuilder.create((p_27246_) -> {
-         return p_27246_.group(BlockPos.f_121852_.fieldOf("pos").forGetter((p_27252_) -> {
-            return p_27252_.f_27227_;
-         }), Registry.f_122870_.fieldOf("type").forGetter((p_27249_) -> {
-            return p_27249_.f_27228_;
-         }), Codec.INT.fieldOf("free_tickets").orElse(0).forGetter((p_27241_) -> {
-            return p_27241_.f_27229_;
-         }), RecordCodecBuilder.point(p_27243_)).apply(p_27246_, PoiRecord::new);
+      return RecordCodecBuilder.create((p_234151_1_) -> {
+         return p_234151_1_.group(BlockPos.f_121852_.fieldOf("pos").forGetter((p_234153_0_) -> {
+            return p_234153_0_.f_27227_;
+         }), Registry.f_122870_.fieldOf("type").forGetter((p_234152_0_) -> {
+            return p_234152_0_.f_27228_;
+         }), Codec.INT.fieldOf("free_tickets").orElse(0).forGetter((p_234149_0_) -> {
+            return p_234149_0_.f_27229_;
+         }), RecordCodecBuilder.point(p_27243_)).apply(p_234151_1_, PoiRecord::new);
       });
    }
 
