--- a/net/minecraft/world/entity/ai/village/poi/PoiSection.java
+++ b/net/minecraft/world/entity/ai/village/poi/PoiSection.java
@@ -30,12 +_,12 @@
    private boolean f_27264_;
 
    public static Codec<PoiSection> m_27295_(Runnable p_27296_) {
-      return RecordCodecBuilder.<PoiSection>create((p_27299_) -> {
-         return p_27299_.group(RecordCodecBuilder.point(p_27296_), Codec.BOOL.optionalFieldOf("Valid", Boolean.valueOf(false)).forGetter((p_27312_) -> {
-            return p_27312_.f_27264_;
-         }), PoiRecord.m_27242_(p_27296_).listOf().fieldOf("Records").forGetter((p_27276_) -> {
-            return ImmutableList.copyOf(p_27276_.f_27261_.values());
-         })).apply(p_27299_, PoiSection::new);
+      return RecordCodecBuilder.<PoiSection>create((p_234159_1_) -> {
+         return p_234159_1_.group(RecordCodecBuilder.point(p_27296_), Codec.BOOL.optionalFieldOf("Valid", Boolean.valueOf(false)).forGetter((p_234162_0_) -> {
+            return p_234162_0_.f_27264_;
+         }), PoiRecord.m_27242_(p_27296_).listOf().fieldOf("Records").forGetter((p_234154_0_) -> {
+            return ImmutableList.copyOf(p_234154_0_.f_27261_.values());
+         })).apply(p_234159_1_, PoiSection::new);
       }).orElseGet(Util.m_137489_("Failed to read POI section: ", f_27260_::error), () -> {
          return new PoiSection(p_27296_, false, ImmutableList.of());
       });
@@ -52,10 +_,10 @@
    }
 
    public Stream<PoiRecord> m_27304_(Predicate<PoiType> p_27305_, PoiManager.Occupancy p_27306_) {
-      return this.f_27262_.entrySet().stream().filter((p_27309_) -> {
-         return p_27305_.test(p_27309_.getKey());
-      }).flatMap((p_27301_) -> {
-         return p_27301_.getValue().stream();
+      return this.f_27262_.entrySet().stream().filter((p_234161_1_) -> {
+         return p_27305_.test(p_234161_1_.getKey());
+      }).flatMap((p_234160_0_) -> {
+         return p_234160_0_.getValue().stream();
       }).filter(p_27306_.m_27221_());
    }
 
@@ -90,7 +_,7 @@
       }
 
       this.f_27261_.put(short1, p_27274_);
-      this.f_27262_.computeIfAbsent(poitype, (p_27278_) -> {
+      this.f_27262_.computeIfAbsent(poitype, (p_234155_0_) -> {
          return Sets.newHashSet();
       }).add(p_27274_);
       return true;
@@ -134,10 +_,10 @@
       if (!this.f_27264_) {
          Short2ObjectMap<PoiRecord> short2objectmap = new Short2ObjectOpenHashMap<>(this.f_27261_);
          this.m_27310_();
-         p_27303_.accept((p_27293_, p_27294_) -> {
-            short short1 = SectionPos.m_123218_(p_27293_);
-            PoiRecord poirecord = short2objectmap.computeIfAbsent(short1, (p_27287_) -> {
-               return new PoiRecord(p_27293_, p_27294_, this.f_27263_);
+         p_27303_.accept((p_234157_2_, p_234157_3_) -> {
+            short short1 = SectionPos.m_123218_(p_234157_2_);
+            PoiRecord poirecord = short2objectmap.computeIfAbsent(short1, (p_234156_3_) -> {
+               return new PoiRecord(p_234157_2_, p_234157_3_, this.f_27263_);
             });
             this.m_27273_(poirecord);
          });
