--- a/net/minecraft/data/models/blockstates/MultiPartGenerator.java
+++ b/net/minecraft/data/models/blockstates/MultiPartGenerator.java
@@ -51,8 +_,8 @@
 
    public JsonElement get() {
       StateDefinition<Block, BlockState> statedefinition = this.f_125199_.m_49965_();
-      this.f_125200_.forEach((p_125208_) -> {
-         p_125208_.m_5848_(statedefinition);
+      this.f_125200_.forEach((p_240107_1_) -> {
+         p_240107_1_.m_5848_(statedefinition);
       });
       JsonArray jsonarray = new JsonArray();
       this.f_125200_.stream().map(MultiPartGenerator.Entry::get).forEach(jsonarray::add);
