--- a/net/minecraft/data/models/blockstates/PropertyDispatch.java
+++ b/net/minecraft/data/models/blockstates/PropertyDispatch.java
@@ -32,13 +_,13 @@
       Stream<Selector> stream = Stream.of(Selector.m_125485_());
 
       for(Property<?> property : list) {
-         stream = stream.flatMap((p_125316_) -> {
-            return property.m_61702_().map(p_125316_::m_125486_);
+         stream = stream.flatMap((p_240138_1_) -> {
+            return property.m_61702_().map(p_240138_1_::m_125486_);
          });
       }
 
-      List<Selector> list1 = stream.filter((p_125318_) -> {
-         return !this.f_125291_.containsKey(p_125318_);
+      List<Selector> list1 = stream.filter((p_240139_1_) -> {
+         return !this.f_125291_.containsKey(p_240139_1_);
       }).collect(Collectors.toList());
       if (!list1.isEmpty()) {
          throw new IllegalStateException("Missing definition for properties: " + list1);
@@ -67,6 +_,69 @@
       return new PropertyDispatch.C5<>(p_125309_, p_125310_, p_125311_, p_125312_, p_125313_);
    }
 
+   public static class C5<T1 extends Comparable<T1>, T2 extends Comparable<T2>, T3 extends Comparable<T3>, T4 extends Comparable<T4>, T5 extends Comparable<T5>> extends PropertyDispatch {
+      private final Property<T1> f_125442_;
+      private final Property<T2> f_125443_;
+      private final Property<T3> f_125444_;
+      private final Property<T4> f_125445_;
+      private final Property<T5> f_125446_;
+
+      private C5(Property<T1> p_125448_, Property<T2> p_125449_, Property<T3> p_125450_, Property<T4> p_125451_, Property<T5> p_125452_) {
+         this.f_125442_ = p_125448_;
+         this.f_125443_ = p_125449_;
+         this.f_125444_ = p_125450_;
+         this.f_125445_ = p_125451_;
+         this.f_125446_ = p_125452_;
+      }
+
+      public List<Property<?>> m_7336_() {
+         return ImmutableList.of(this.f_125442_, this.f_125443_, this.f_125444_, this.f_125445_, this.f_125446_);
+      }
+
+      public PropertyDispatch.C5<T1, T2, T3, T4, T5> m_125467_(T1 p_125468_, T2 p_125469_, T3 p_125470_, T4 p_125471_, T5 p_125472_, List<Variant> p_125473_) {
+         Selector selector = Selector.m_125490_(this.f_125442_.m_61699_(p_125468_), this.f_125443_.m_61699_(p_125469_), this.f_125444_.m_61699_(p_125470_), this.f_125445_.m_61699_(p_125471_), this.f_125446_.m_61699_(p_125472_));
+         this.m_125319_(selector, p_125473_);
+         return this;
+      }
+
+      public PropertyDispatch.C5<T1, T2, T3, T4, T5> m_125460_(T1 p_125461_, T2 p_125462_, T3 p_125463_, T4 p_125464_, T5 p_125465_, Variant p_125466_) {
+         return this.m_125467_(p_125461_, p_125462_, p_125463_, p_125464_, p_125465_, Collections.singletonList(p_125466_));
+      }
+   }
+
+   public static class C4<T1 extends Comparable<T1>, T2 extends Comparable<T2>, T3 extends Comparable<T3>, T4 extends Comparable<T4>> extends PropertyDispatch {
+      private final Property<T1> f_125414_;
+      private final Property<T2> f_125415_;
+      private final Property<T3> f_125416_;
+      private final Property<T4> f_125417_;
+
+      private C4(Property<T1> p_125419_, Property<T2> p_125420_, Property<T3> p_125421_, Property<T4> p_125422_) {
+         this.f_125414_ = p_125419_;
+         this.f_125415_ = p_125420_;
+         this.f_125416_ = p_125421_;
+         this.f_125417_ = p_125422_;
+      }
+
+      public List<Property<?>> m_7336_() {
+         return ImmutableList.of(this.f_125414_, this.f_125415_, this.f_125416_, this.f_125417_);
+      }
+
+      public PropertyDispatch.C4<T1, T2, T3, T4> m_125435_(T1 p_125436_, T2 p_125437_, T3 p_125438_, T4 p_125439_, List<Variant> p_125440_) {
+         Selector selector = Selector.m_125490_(this.f_125414_.m_61699_(p_125436_), this.f_125415_.m_61699_(p_125437_), this.f_125416_.m_61699_(p_125438_), this.f_125417_.m_61699_(p_125439_));
+         this.m_125319_(selector, p_125440_);
+         return this;
+      }
+
+      public PropertyDispatch.C4<T1, T2, T3, T4> m_125429_(T1 p_125430_, T2 p_125431_, T3 p_125432_, T4 p_125433_, Variant p_125434_) {
+         return this.m_125435_(p_125430_, p_125431_, p_125432_, p_125433_, Collections.singletonList(p_125434_));
+      }
+   }
+
+   @FunctionalInterface
+   public interface TriFunction<P1, P2, P3, R> {
+      R m_125475_(P1 p_125476_, P2 p_125477_, P3 p_125478_);
+   }
+
    public static class C1<T1 extends Comparable<T1>> extends PropertyDispatch {
       private final Property<T1> f_125323_;
 
@@ -89,50 +_,8 @@
       }
 
       public PropertyDispatch m_125335_(Function<T1, Variant> p_125336_) {
-         this.f_125323_.m_6908_().forEach((p_125340_) -> {
-            this.m_125329_(p_125340_, p_125336_.apply(p_125340_));
-         });
-         return this;
-      }
-   }
-
-   public static class C2<T1 extends Comparable<T1>, T2 extends Comparable<T2>> extends PropertyDispatch {
-      private final Property<T1> f_125341_;
-      private final Property<T2> f_125342_;
-
-      private C2(Property<T1> p_125344_, Property<T2> p_125345_) {
-         this.f_125341_ = p_125344_;
-         this.f_125342_ = p_125345_;
-      }
-
-      public List<Property<?>> m_7336_() {
-         return ImmutableList.of(this.f_125341_, this.f_125342_);
-      }
-
-      public PropertyDispatch.C2<T1, T2> m_125354_(T1 p_125355_, T2 p_125356_, List<Variant> p_125357_) {
-         Selector selector = Selector.m_125490_(this.f_125341_.m_61699_(p_125355_), this.f_125342_.m_61699_(p_125356_));
-         this.m_125319_(selector, p_125357_);
-         return this;
-      }
-
-      public PropertyDispatch.C2<T1, T2> m_125350_(T1 p_125351_, T2 p_125352_, Variant p_125353_) {
-         return this.m_125354_(p_125351_, p_125352_, Collections.singletonList(p_125353_));
-      }
-
-      public PropertyDispatch m_125362_(BiFunction<T1, T2, Variant> p_125363_) {
-         this.f_125341_.m_6908_().forEach((p_125376_) -> {
-            this.f_125342_.m_6908_().forEach((p_125371_) -> {
-               this.m_125350_((T1)p_125376_, p_125371_, p_125363_.apply((T1)p_125376_, p_125371_));
-            });
-         });
-         return this;
-      }
-
-      public PropertyDispatch m_125372_(BiFunction<T1, T2, List<Variant>> p_125373_) {
-         this.f_125341_.m_6908_().forEach((p_125366_) -> {
-            this.f_125342_.m_6908_().forEach((p_125361_) -> {
-               this.m_125354_((T1)p_125366_, p_125361_, p_125373_.apply((T1)p_125366_, p_125361_));
-            });
+         this.f_125323_.m_6908_().forEach((p_240146_2_) -> {
+            this.m_125329_(p_240146_2_, p_125336_.apply(p_240146_2_));
          });
          return this;
       }
@@ -164,10 +_,10 @@
       }
 
       public PropertyDispatch m_125389_(PropertyDispatch.TriFunction<T1, T2, T3, Variant> p_125390_) {
-         this.f_125377_.m_6908_().forEach((p_125404_) -> {
-            this.f_125378_.m_6908_().forEach((p_125408_) -> {
-               this.f_125379_.m_6908_().forEach((p_125413_) -> {
-                  this.m_125391_((T1)p_125404_, (T2)p_125408_, p_125413_, p_125390_.m_125475_((T1)p_125404_, (T2)p_125408_, p_125413_));
+         this.f_125377_.m_6908_().forEach((p_240163_2_) -> {
+            this.f_125378_.m_6908_().forEach((p_240164_3_) -> {
+               this.f_125379_.m_6908_().forEach((p_240165_4_) -> {
+                  this.m_125391_((T1)p_240163_2_, (T2)p_240164_3_, p_240165_4_, p_125390_.m_125475_((T1)p_240163_2_, (T2)p_240164_3_, p_240165_4_));
                });
             });
          });
@@ -175,66 +_,45 @@
       }
    }
 
-   public static class C4<T1 extends Comparable<T1>, T2 extends Comparable<T2>, T3 extends Comparable<T3>, T4 extends Comparable<T4>> extends PropertyDispatch {
-      private final Property<T1> f_125414_;
-      private final Property<T2> f_125415_;
-      private final Property<T3> f_125416_;
-      private final Property<T4> f_125417_;
-
-      private C4(Property<T1> p_125419_, Property<T2> p_125420_, Property<T3> p_125421_, Property<T4> p_125422_) {
-         this.f_125414_ = p_125419_;
-         this.f_125415_ = p_125420_;
-         this.f_125416_ = p_125421_;
-         this.f_125417_ = p_125422_;
-      }
-
-      public List<Property<?>> m_7336_() {
-         return ImmutableList.of(this.f_125414_, this.f_125415_, this.f_125416_, this.f_125417_);
-      }
-
-      public PropertyDispatch.C4<T1, T2, T3, T4> m_125435_(T1 p_125436_, T2 p_125437_, T3 p_125438_, T4 p_125439_, List<Variant> p_125440_) {
-         Selector selector = Selector.m_125490_(this.f_125414_.m_61699_(p_125436_), this.f_125415_.m_61699_(p_125437_), this.f_125416_.m_61699_(p_125438_), this.f_125417_.m_61699_(p_125439_));
-         this.m_125319_(selector, p_125440_);
-         return this;
-      }
-
-      public PropertyDispatch.C4<T1, T2, T3, T4> m_125429_(T1 p_125430_, T2 p_125431_, T3 p_125432_, T4 p_125433_, Variant p_125434_) {
-         return this.m_125435_(p_125430_, p_125431_, p_125432_, p_125433_, Collections.singletonList(p_125434_));
-      }
-   }
-
-   public static class C5<T1 extends Comparable<T1>, T2 extends Comparable<T2>, T3 extends Comparable<T3>, T4 extends Comparable<T4>, T5 extends Comparable<T5>> extends PropertyDispatch {
-      private final Property<T1> f_125442_;
-      private final Property<T2> f_125443_;
-      private final Property<T3> f_125444_;
-      private final Property<T4> f_125445_;
-      private final Property<T5> f_125446_;
-
-      private C5(Property<T1> p_125448_, Property<T2> p_125449_, Property<T3> p_125450_, Property<T4> p_125451_, Property<T5> p_125452_) {
-         this.f_125442_ = p_125448_;
-         this.f_125443_ = p_125449_;
-         this.f_125444_ = p_125450_;
-         this.f_125445_ = p_125451_;
-         this.f_125446_ = p_125452_;
-      }
-
-      public List<Property<?>> m_7336_() {
-         return ImmutableList.of(this.f_125442_, this.f_125443_, this.f_125444_, this.f_125445_, this.f_125446_);
-      }
-
-      public PropertyDispatch.C5<T1, T2, T3, T4, T5> m_125467_(T1 p_125468_, T2 p_125469_, T3 p_125470_, T4 p_125471_, T5 p_125472_, List<Variant> p_125473_) {
-         Selector selector = Selector.m_125490_(this.f_125442_.m_61699_(p_125468_), this.f_125443_.m_61699_(p_125469_), this.f_125444_.m_61699_(p_125470_), this.f_125445_.m_61699_(p_125471_), this.f_125446_.m_61699_(p_125472_));
-         this.m_125319_(selector, p_125473_);
-         return this;
-      }
-
-      public PropertyDispatch.C5<T1, T2, T3, T4, T5> m_125460_(T1 p_125461_, T2 p_125462_, T3 p_125463_, T4 p_125464_, T5 p_125465_, Variant p_125466_) {
-         return this.m_125467_(p_125461_, p_125462_, p_125463_, p_125464_, p_125465_, Collections.singletonList(p_125466_));
-      }
-   }
-
-   @FunctionalInterface
-   public interface TriFunction<P1, P2, P3, R> {
-      R m_125475_(P1 p_125476_, P2 p_125477_, P3 p_125478_);
+   public static class C2<T1 extends Comparable<T1>, T2 extends Comparable<T2>> extends PropertyDispatch {
+      private final Property<T1> f_125341_;
+      private final Property<T2> f_125342_;
+
+      private C2(Property<T1> p_125344_, Property<T2> p_125345_) {
+         this.f_125341_ = p_125344_;
+         this.f_125342_ = p_125345_;
+      }
+
+      public List<Property<?>> m_7336_() {
+         return ImmutableList.of(this.f_125341_, this.f_125342_);
+      }
+
+      public PropertyDispatch.C2<T1, T2> m_125354_(T1 p_125355_, T2 p_125356_, List<Variant> p_125357_) {
+         Selector selector = Selector.m_125490_(this.f_125341_.m_61699_(p_125355_), this.f_125342_.m_61699_(p_125356_));
+         this.m_125319_(selector, p_125357_);
+         return this;
+      }
+
+      public PropertyDispatch.C2<T1, T2> m_125350_(T1 p_125351_, T2 p_125352_, Variant p_125353_) {
+         return this.m_125354_(p_125351_, p_125352_, Collections.singletonList(p_125353_));
+      }
+
+      public PropertyDispatch m_125362_(BiFunction<T1, T2, Variant> p_125363_) {
+         this.f_125341_.m_6908_().forEach((p_240156_2_) -> {
+            this.f_125342_.m_6908_().forEach((p_240154_3_) -> {
+               this.m_125350_((T1)p_240156_2_, p_240154_3_, p_125363_.apply((T1)p_240156_2_, p_240154_3_));
+            });
+         });
+         return this;
+      }
+
+      public PropertyDispatch m_125372_(BiFunction<T1, T2, List<Variant>> p_125373_) {
+         this.f_125341_.m_6908_().forEach((p_240153_2_) -> {
+            this.f_125342_.m_6908_().forEach((p_240151_3_) -> {
+               this.m_125354_((T1)p_240153_2_, p_240151_3_, p_125373_.apply((T1)p_240153_2_, p_240151_3_));
+            });
+         });
+         return this;
+      }
    }
 }
