--- a/net/minecraft/data/models/blockstates/Variant.java
+++ b/net/minecraft/data/models/blockstates/Variant.java
@@ -33,8 +_,8 @@
 
    public JsonElement get() {
       JsonObject jsonobject = new JsonObject();
-      this.f_125499_.values().forEach((p_125507_) -> {
-         p_125507_.m_125563_(jsonobject);
+      this.f_125499_.values().forEach((p_240196_1_) -> {
+         p_240196_1_.m_125563_(jsonobject);
       });
       return jsonobject;
    }
@@ -44,8 +_,8 @@
          return p_125515_.get(0).get();
       } else {
          JsonArray jsonarray = new JsonArray();
-         p_125515_.forEach((p_125504_) -> {
-            jsonarray.add(p_125504_.get());
+         p_125515_.forEach((p_240195_1_) -> {
+            jsonarray.add(p_240195_1_.get());
          });
          return jsonarray;
       }
