--- a/net/minecraft/client/renderer/block/BlockRenderDispatcher.java
+++ b/net/minecraft/client/renderer/block/BlockRenderDispatcher.java
@@ -35,7 +_,7 @@
    public BlockRenderDispatcher(BlockModelShaper p_110905_, BlockColors p_110906_) {
       this.f_110899_ = p_110905_;
       this.f_110903_ = p_110906_;
-      this.f_110900_ = new ModelBlockRenderer(this.f_110903_);
+      this.f_110900_ = new net.minecraftforge.client.model.pipeline.ForgeBlockModelRenderer(this.f_110903_);
       this.f_110901_ = new LiquidBlockRenderer();
    }
 
@@ -43,22 +_,30 @@
       return this.f_110899_;
    }
 
+   @Deprecated //Forge: Model parameter
    public void m_110918_(BlockState p_110919_, BlockPos p_110920_, BlockAndTintGetter p_110921_, PoseStack p_110922_, VertexConsumer p_110923_) {
-      if (p_110919_.m_60799_() == RenderShape.MODEL) {
-         BakedModel bakedmodel = this.f_110899_.m_110893_(p_110919_);
-         long i = p_110919_.m_60726_(p_110920_);
-         this.f_110900_.m_111047_(p_110921_, bakedmodel, p_110919_, p_110920_, p_110922_, p_110923_, true, this.f_110902_, i, OverlayTexture.f_118083_);
+       renderBlockDamage(p_110919_,p_110920_, p_110921_, p_110922_, p_110923_, net.minecraftforge.client.model.data.EmptyModelData.INSTANCE);
+   }
+   public void renderBlockDamage(BlockState blockStateIn, BlockPos posIn, BlockAndTintGetter lightReaderIn, PoseStack matrixStackIn, VertexConsumer vertexBuilderIn, net.minecraftforge.client.model.data.IModelData modelData) {
+      if (blockStateIn.m_60799_() == RenderShape.MODEL) {
+         BakedModel bakedmodel = this.f_110899_.m_110893_(blockStateIn);
+         long i = blockStateIn.m_60726_(posIn);
+         this.f_110900_.renderModel(lightReaderIn, bakedmodel, blockStateIn, posIn, matrixStackIn, vertexBuilderIn, true, this.f_110902_, i, OverlayTexture.f_118083_, modelData);
       }
    }
 
+   @Deprecated //Forge: Model parameter
    public boolean m_110924_(BlockState p_110925_, BlockPos p_110926_, BlockAndTintGetter p_110927_, PoseStack p_110928_, VertexConsumer p_110929_, boolean p_110930_, Random p_110931_) {
+       return renderModel(p_110925_, p_110926_, p_110927_, p_110928_, p_110929_, p_110930_, p_110931_, net.minecraftforge.client.model.data.EmptyModelData.INSTANCE);
+   }
+   public boolean renderModel(BlockState blockStateIn, BlockPos posIn, BlockAndTintGetter lightReaderIn, PoseStack matrixStackIn, VertexConsumer vertexBuilderIn, boolean checkSides, Random rand, net.minecraftforge.client.model.data.IModelData modelData) {
       try {
-         RenderShape rendershape = p_110925_.m_60799_();
-         return rendershape != RenderShape.MODEL ? false : this.f_110900_.m_111047_(p_110927_, this.m_110910_(p_110925_), p_110925_, p_110926_, p_110928_, p_110929_, p_110930_, p_110931_, p_110925_.m_60726_(p_110926_), OverlayTexture.f_118083_);
+         RenderShape rendershape = blockStateIn.m_60799_();
+         return rendershape != RenderShape.MODEL ? false : this.f_110900_.renderModel(lightReaderIn, this.m_110910_(blockStateIn), blockStateIn, posIn, matrixStackIn, vertexBuilderIn, checkSides, rand, blockStateIn.m_60726_(posIn), OverlayTexture.f_118083_, modelData);
       } catch (Throwable throwable) {
          CrashReport crashreport = CrashReport.m_127521_(throwable, "Tesselating block in world");
          CrashReportCategory crashreportcategory = crashreport.m_127514_("Block being tesselated");
-         CrashReportCategory.m_128170_(crashreportcategory, p_110926_, p_110925_);
+         CrashReportCategory.m_128170_(crashreportcategory, posIn, blockStateIn);
          throw new ReportedException(crashreport);
       }
    }
@@ -82,7 +_,11 @@
       return this.f_110899_.m_110893_(p_110911_);
    }
 
+   @Deprecated //Forge: Model parameter
    public void m_110912_(BlockState p_110913_, PoseStack p_110914_, MultiBufferSource p_110915_, int p_110916_, int p_110917_) {
+      renderBlock(p_110913_, p_110914_, p_110915_, p_110916_, p_110917_, net.minecraftforge.client.model.data.EmptyModelData.INSTANCE);
+   }
+   public void renderBlock(BlockState p_110913_, PoseStack p_110914_, MultiBufferSource p_110915_, int p_110916_, int p_110917_, net.minecraftforge.client.model.data.IModelData modelData) {
       RenderShape rendershape = p_110913_.m_60799_();
       if (rendershape != RenderShape.INVISIBLE) {
          switch(rendershape) {
@@ -92,10 +_,11 @@
             float f = (float)(i >> 16 & 255) / 255.0F;
             float f1 = (float)(i >> 8 & 255) / 255.0F;
             float f2 = (float)(i & 255) / 255.0F;
-            this.f_110900_.m_111067_(p_110914_.m_85850_(), p_110915_.m_6299_(ItemBlockRenderTypes.m_109284_(p_110913_, false)), p_110913_, bakedmodel, f, f1, f2, p_110916_, p_110917_);
+            this.f_110900_.renderModel(p_110914_.m_85850_(), p_110915_.m_6299_(ItemBlockRenderTypes.m_109284_(p_110913_, false)), p_110913_, bakedmodel, f, f1, f2, p_110916_, p_110917_, modelData);
             break;
          case ENTITYBLOCK_ANIMATED:
-            BlockEntityWithoutLevelRenderer.f_108814_.m_108829_(new ItemStack(p_110913_.m_60734_()), ItemTransforms.TransformType.NONE, p_110914_, p_110915_, p_110916_, p_110917_);
+            ItemStack stack = new ItemStack(p_110913_.m_60734_());
+            stack.m_41720_().getItemStackTileEntityRenderer().m_108829_(stack, ItemTransforms.TransformType.NONE, p_110914_, p_110915_, p_110916_, p_110917_);
          }
 
       }
@@ -103,5 +_,10 @@
 
    public void m_6213_(ResourceManager p_110909_) {
       this.f_110901_.m_110944_();
+   }
+
+   @Override
+   public net.minecraftforge.resource.IResourceType getResourceType() {
+      return net.minecraftforge.resource.VanillaResourceType.MODELS;
    }
 }
