--- a/net/minecraft/data/models/model/ModelTemplate.java
+++ b/net/minecraft/data/models/model/ModelTemplate.java
@@ -41,13 +_,13 @@
       Map<TextureSlot, ResourceLocation> map = this.m_125608_(p_125614_);
       p_125615_.accept(p_125613_, () -> {
          JsonObject jsonobject = new JsonObject();
-         this.f_125585_.ifPresent((p_125607_) -> {
-            jsonobject.addProperty("parent", p_125607_.toString());
+         this.f_125585_.ifPresent((p_240231_1_) -> {
+            jsonobject.addProperty("parent", p_240231_1_.toString());
          });
          if (!map.isEmpty()) {
             JsonObject jsonobject1 = new JsonObject();
-            map.forEach((p_125603_, p_125604_) -> {
-               jsonobject1.addProperty(p_125603_.m_125897_(), p_125604_.toString());
+            map.forEach((p_240230_1_, p_240230_2_) -> {
+               jsonobject1.addProperty(p_240230_1_.m_125897_(), p_240230_2_.toString());
             });
             jsonobject.add("textures", jsonobject1);
          }
