--- a/net/minecraft/data/models/model/TexturedModel.java
+++ b/net/minecraft/data/models/model/TexturedModel.java
@@ -60,8 +_,8 @@
    }
 
    private static TexturedModel.Provider m_125942_(Function<Block, TextureMapping> p_125943_, ModelTemplate p_125944_) {
-      return (p_125948_) -> {
-         return new TexturedModel(p_125943_.apply(p_125948_), p_125944_);
+      return (p_240462_2_) -> {
+         return new TexturedModel(p_125943_.apply(p_240462_2_), p_125944_);
       };
    }
 
@@ -82,8 +_,8 @@
       }
 
       default TexturedModel.Provider m_125959_(Consumer<TextureMapping> p_125960_) {
-         return (p_125963_) -> {
-            return this.m_125964_(p_125963_).m_125940_(p_125960_);
+         return (p_240468_2_) -> {
+            return this.m_125964_(p_240468_2_).m_125940_(p_125960_);
          };
       }
    }
