--- a/net/minecraft/world/level/dimension/LevelStem.java
+++ b/net/minecraft/world/level/dimension/LevelStem.java
@@ -21,8 +_,8 @@
 import net.minecraft.world.level.levelgen.NoiseGeneratorSettings;
 
 public final class LevelStem {
-   public static final Codec<LevelStem> f_63970_ = RecordCodecBuilder.create((p_63986_) -> {
-      return p_63986_.group(DimensionType.f_63853_.fieldOf("type").forGetter(LevelStem::m_63981_), ChunkGenerator.f_62136_.fieldOf("generator").forGetter(LevelStem::m_63990_)).apply(p_63986_, p_63986_.stable(LevelStem::new));
+   public static final Codec<LevelStem> f_63970_ = RecordCodecBuilder.create((p_236061_0_) -> {
+      return p_236061_0_.group(DimensionType.f_63853_.fieldOf("type").forGetter(LevelStem::m_63981_), ChunkGenerator.f_62136_.fieldOf("generator").forGetter(LevelStem::m_63990_)).apply(p_236061_0_, p_236061_0_.stable(LevelStem::new));
    });
    public static final ResourceKey<LevelStem> f_63971_ = ResourceKey.m_135785_(Registry.f_122820_, new ResourceLocation("overworld"));
    public static final ResourceKey<LevelStem> f_63972_ = ResourceKey.m_135785_(Registry.f_122820_, new ResourceLocation("the_nether"));
