--- a/net/minecraft/commands/arguments/item/FunctionArgument.java
+++ b/net/minecraft/commands/arguments/item/FunctionArgument.java
@@ -18,11 +_,11 @@
 
 public class FunctionArgument implements ArgumentType<FunctionArgument.Result> {
    private static final Collection<String> f_120902_ = Arrays.asList("foo", "foo:bar", "#foo");
-   private static final DynamicCommandExceptionType f_120903_ = new DynamicCommandExceptionType((p_120927_) -> {
-      return new TranslatableComponent("arguments.function.tag.unknown", p_120927_);
+   private static final DynamicCommandExceptionType f_120903_ = new DynamicCommandExceptionType((p_208691_0_) -> {
+      return new TranslatableComponent("arguments.function.tag.unknown", p_208691_0_);
    });
-   private static final DynamicCommandExceptionType f_120904_ = new DynamicCommandExceptionType((p_120917_) -> {
-      return new TranslatableComponent("arguments.function.unknown", p_120917_);
+   private static final DynamicCommandExceptionType f_120904_ = new DynamicCommandExceptionType((p_208694_0_) -> {
+      return new TranslatableComponent("arguments.function.unknown", p_208694_0_);
    });
 
    public static FunctionArgument m_120907_() {
