--- a/net/minecraft/commands/arguments/item/ItemParser.java
+++ b/net/minecraft/commands/arguments/item/ItemParser.java
@@ -23,11 +_,11 @@
 
 public class ItemParser {
    public static final SimpleCommandExceptionType f_120991_ = new SimpleCommandExceptionType(new TranslatableComponent("argument.item.tag.disallowed"));
-   public static final DynamicCommandExceptionType f_120992_ = new DynamicCommandExceptionType((p_121013_) -> {
-      return new TranslatableComponent("argument.item.id.invalid", p_121013_);
+   public static final DynamicCommandExceptionType f_120992_ = new DynamicCommandExceptionType((p_208696_0_) -> {
+      return new TranslatableComponent("argument.item.id.invalid", p_208696_0_);
    });
-   private static final BiFunction<SuggestionsBuilder, TagCollection<Item>, CompletableFuture<Suggestions>> f_120993_ = (p_121028_, p_121029_) -> {
-      return p_121028_.buildFuture();
+   private static final BiFunction<SuggestionsBuilder, TagCollection<Item>, CompletableFuture<Suggestions>> f_120993_ = (p_239571_0_, p_239571_1_) -> {
+      return p_239571_0_.buildFuture();
    };
    private final StringReader f_120994_;
    private final boolean f_120995_;
