--- a/net/minecraft/commands/arguments/item/ItemInput.java
+++ b/net/minecraft/commands/arguments/item/ItemInput.java
@@ -12,8 +_,8 @@
 import net.minecraft.world.item.ItemStack;
 
 public class ItemInput implements Predicate<ItemStack> {
-   private static final Dynamic2CommandExceptionType f_120972_ = new Dynamic2CommandExceptionType((p_120986_, p_120987_) -> {
-      return new TranslatableComponent("arguments.item.overstacked", p_120986_, p_120987_);
+   private static final Dynamic2CommandExceptionType f_120972_ = new Dynamic2CommandExceptionType((p_208695_0_, p_208695_1_) -> {
+      return new TranslatableComponent("arguments.item.overstacked", p_208695_0_, p_208695_1_);
    });
    private final Item f_120973_;
    @Nullable
