--- a/net/minecraft/world/level/levelgen/placement/CarvingMaskDecorator.java
+++ b/net/minecraft/world/level/levelgen/placement/CarvingMaskDecorator.java
@@ -16,12 +_,12 @@
    public Stream<BlockPos> m_7887_(DecorationContext p_70432_, Random p_70433_, CarvingMaskDecoratorConfiguration p_70434_, BlockPos p_70435_) {
       ChunkPos chunkpos = new ChunkPos(p_70435_);
       BitSet bitset = p_70432_.m_70587_(chunkpos, p_70434_.f_70442_);
-      return IntStream.range(0, bitset.length()).filter((p_70440_) -> {
-         return bitset.get(p_70440_) && p_70433_.nextFloat() < p_70434_.f_70443_;
-      }).mapToObj((p_70425_) -> {
-         int i = p_70425_ & 15;
-         int j = p_70425_ >> 4 & 15;
-         int k = p_70425_ >> 8;
+      return IntStream.range(0, bitset.length()).filter((p_215067_3_) -> {
+         return bitset.get(p_215067_3_) && p_70433_.nextFloat() < p_70434_.f_70443_;
+      }).mapToObj((p_215068_1_) -> {
+         int i = p_215068_1_ & 15;
+         int j = p_215068_1_ >> 4 & 15;
+         int k = p_215068_1_ >> 8;
          return new BlockPos(chunkpos.m_45604_() + i, k, chunkpos.m_45605_() + j);
       });
    }
