--- a/net/minecraft/world/level/levelgen/placement/ConfiguredDecorator.java
+++ b/net/minecraft/world/level/levelgen/placement/ConfiguredDecorator.java
@@ -9,8 +_,8 @@
 import net.minecraft.world.level.levelgen.feature.configurations.DecoratorConfiguration;
 
 public class ConfiguredDecorator<DC extends DecoratorConfiguration> implements Decoratable<ConfiguredDecorator<?>> {
-   public static final Codec<ConfiguredDecorator<?>> f_70471_ = Registry.f_122845_.dispatch("type", (p_70488_) -> {
-      return p_70488_.f_70472_;
+   public static final Codec<ConfiguredDecorator<?>> f_70471_ = Registry.f_122845_.dispatch("type", (p_236954_0_) -> {
+      return p_236954_0_.f_70472_;
    }, FeatureDecorator::m_70710_);
    private final FeatureDecorator<DC> f_70472_;
    private final DC f_70473_;
