--- a/net/minecraft/world/level/levelgen/placement/DecoratedDecorator.java
+++ b/net/minecraft/world/level/levelgen/placement/DecoratedDecorator.java
@@ -11,8 +_,8 @@
    }
 
    public Stream<BlockPos> m_7887_(DecorationContext p_70566_, Random p_70567_, DecoratedDecoratorConfiguration p_70568_, BlockPos p_70569_) {
-      return p_70568_.m_70577_().m_70480_(p_70566_, p_70567_, p_70569_).flatMap((p_70559_) -> {
-         return p_70568_.m_70580_().m_70480_(p_70566_, p_70567_, p_70559_);
+      return p_70568_.m_70577_().m_70480_(p_70566_, p_70567_, p_70569_).flatMap((p_242882_3_) -> {
+         return p_70568_.m_70580_().m_70480_(p_70566_, p_70567_, p_242882_3_);
       });
    }
 }
