--- a/net/minecraft/world/level/levelgen/placement/EmeraldPlacementDecorator.java
+++ b/net/minecraft/world/level/levelgen/placement/EmeraldPlacementDecorator.java
@@ -14,7 +_,7 @@
 
    public Stream<BlockPos> m_6193_(Random p_70630_, NoneDecoratorConfiguration p_70631_, BlockPos p_70632_) {
       int i = 3 + p_70630_.nextInt(6);
-      return IntStream.range(0, i).mapToObj((p_70636_) -> {
+      return IntStream.range(0, i).mapToObj((p_215060_2_) -> {
          int j = p_70630_.nextInt(16) + p_70632_.m_123341_();
          int k = p_70630_.nextInt(16) + p_70632_.m_123343_();
          int l = p_70630_.nextInt(28) + 4;
