--- a/net/minecraft/world/level/levelgen/placement/FeatureDecorator.java
+++ b/net/minecraft/world/level/levelgen/placement/FeatureDecorator.java
@@ -15,7 +_,7 @@
 import net.minecraft.world.level.levelgen.placement.nether.GlowstoneDecorator;
 import net.minecraft.world.level.levelgen.placement.nether.MagmaDecorator;
 
-public abstract class FeatureDecorator<DC extends DecoratorConfiguration> {
+public abstract class FeatureDecorator<DC extends DecoratorConfiguration> extends net.minecraftforge.registries.ForgeRegistryEntry<FeatureDecorator<?>> {
    public static final FeatureDecorator<NoneDecoratorConfiguration> f_70681_ = m_70717_("nope", new NopePlacementDecorator(NoneDecoratorConfiguration.f_67810_));
    public static final FeatureDecorator<ChanceDecoratorConfiguration> f_70682_ = m_70717_("chance", new ChanceDecorator(ChanceDecoratorConfiguration.f_70464_));
    public static final FeatureDecorator<CountConfiguration> f_70683_ = m_70717_("count", new CountDecorator(CountConfiguration.f_67568_));
