--- a/net/minecraft/world/level/levelgen/placement/FrequencyWithExtraChanceDecoratorConfiguration.java
+++ b/net/minecraft/world/level/levelgen/placement/FrequencyWithExtraChanceDecoratorConfiguration.java
@@ -5,14 +_,14 @@
 import net.minecraft.world.level.levelgen.feature.configurations.DecoratorConfiguration;
 
 public class FrequencyWithExtraChanceDecoratorConfiguration implements DecoratorConfiguration {
-   public static final Codec<FrequencyWithExtraChanceDecoratorConfiguration> f_70723_ = RecordCodecBuilder.create((p_70733_) -> {
-      return p_70733_.group(Codec.INT.fieldOf("count").forGetter((p_70739_) -> {
-         return p_70739_.f_70724_;
-      }), Codec.FLOAT.fieldOf("extra_chance").forGetter((p_70737_) -> {
-         return p_70737_.f_70725_;
-      }), Codec.INT.fieldOf("extra_count").forGetter((p_70735_) -> {
-         return p_70735_.f_70726_;
-      })).apply(p_70733_, FrequencyWithExtraChanceDecoratorConfiguration::new);
+   public static final Codec<FrequencyWithExtraChanceDecoratorConfiguration> f_70723_ = RecordCodecBuilder.create((p_236974_0_) -> {
+      return p_236974_0_.group(Codec.INT.fieldOf("count").forGetter((p_236977_0_) -> {
+         return p_236977_0_.f_70724_;
+      }), Codec.FLOAT.fieldOf("extra_chance").forGetter((p_236976_0_) -> {
+         return p_236976_0_.f_70725_;
+      }), Codec.INT.fieldOf("extra_count").forGetter((p_236975_0_) -> {
+         return p_236975_0_.f_70726_;
+      })).apply(p_236974_0_, FrequencyWithExtraChanceDecoratorConfiguration::new);
    });
    public final int f_70724_;
    public final float f_70725_;
