--- a/net/minecraft/world/level/levelgen/placement/NoiseBasedDecorator.java
+++ b/net/minecraft/world/level/levelgen/placement/NoiseBasedDecorator.java
@@ -15,7 +_,7 @@
    public Stream<BlockPos> m_6193_(Random p_70803_, NoiseCountFactorDecoratorConfiguration p_70804_, BlockPos p_70805_) {
       double d0 = Biome.f_47433_.m_75449_((double)p_70805_.m_123341_() / p_70804_.f_70808_, (double)p_70805_.m_123343_() / p_70804_.f_70808_, false);
       int i = (int)Math.ceil((d0 + p_70804_.f_70809_) * (double)p_70804_.f_70807_);
-      return IntStream.range(0, i).mapToObj((p_70797_) -> {
+      return IntStream.range(0, i).mapToObj((p_242913_1_) -> {
          return p_70805_;
       });
    }
