--- a/net/minecraft/util/thread/BlockableEventLoop.java
+++ b/net/minecraft/util/thread/BlockableEventLoop.java
@@ -49,7 +_,7 @@
       return this.m_5660_() ? CompletableFuture.supplyAsync(p_18692_, this) : CompletableFuture.completedFuture(p_18692_.get());
    }
 
-   private CompletableFuture<Void> m_18689_(Runnable p_18690_) {
+   public CompletableFuture<Void> m_18689_(Runnable p_18690_) {
       return CompletableFuture.supplyAsync(() -> {
          p_18690_.run();
          return null;
