--- a/net/minecraft/world/item/trading/Merchant.java
+++ b/net/minecraft/world/item/trading/Merchant.java
@@ -42,8 +_,8 @@
    }
 
    default void m_45301_(Player p_45302_, Component p_45303_, int p_45304_) {
-      OptionalInt optionalint = p_45302_.m_5893_(new SimpleMenuProvider((p_45298_, p_45299_, p_45300_) -> {
-         return new MerchantMenu(p_45298_, p_45299_, this);
+      OptionalInt optionalint = p_45302_.m_5893_(new SimpleMenuProvider((p_213701_1_, p_213701_2_, p_213701_3_) -> {
+         return new MerchantMenu(p_213701_1_, p_213701_2_, this);
       }, p_45303_));
       if (optionalint.isPresent()) {
          MerchantOffers merchantoffers = this.m_6616_();
