--- a/net/minecraft/client/renderer/entity/layers/HorseMarkingLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/HorseMarkingLayer.java
@@ -18,12 +_,12 @@
 
 @OnlyIn(Dist.CLIENT)
 public class HorseMarkingLayer extends RenderLayer<Horse, HorseModel<Horse>> {
-   private static final Map<Markings, ResourceLocation> f_117042_ = Util.m_137469_(Maps.newEnumMap(Markings.class), (p_117069_) -> {
-      p_117069_.put(Markings.NONE, (ResourceLocation)null);
-      p_117069_.put(Markings.WHITE, new ResourceLocation("textures/entity/horse/horse_markings_white.png"));
-      p_117069_.put(Markings.WHITE_FIELD, new ResourceLocation("textures/entity/horse/horse_markings_whitefield.png"));
-      p_117069_.put(Markings.WHITE_DOTS, new ResourceLocation("textures/entity/horse/horse_markings_whitedots.png"));
-      p_117069_.put(Markings.BLACK_DOTS, new ResourceLocation("textures/entity/horse/horse_markings_blackdots.png"));
+   private static final Map<Markings, ResourceLocation> f_117042_ = Util.m_137469_(Maps.newEnumMap(Markings.class), (p_239406_0_) -> {
+      p_239406_0_.put(Markings.NONE, (ResourceLocation)null);
+      p_239406_0_.put(Markings.WHITE, new ResourceLocation("textures/entity/horse/horse_markings_white.png"));
+      p_239406_0_.put(Markings.WHITE_FIELD, new ResourceLocation("textures/entity/horse/horse_markings_whitefield.png"));
+      p_239406_0_.put(Markings.WHITE_DOTS, new ResourceLocation("textures/entity/horse/horse_markings_whitedots.png"));
+      p_239406_0_.put(Markings.BLACK_DOTS, new ResourceLocation("textures/entity/horse/horse_markings_blackdots.png"));
    });
 
    public HorseMarkingLayer(RenderLayerParent<Horse, HorseModel<Horse>> p_117045_) {
