--- a/net/minecraft/world/level/timers/TimerQueue.java
+++ b/net/minecraft/world/level/timers/TimerQueue.java
@@ -25,10 +_,10 @@
    private final Table<String, Long, TimerQueue.Event<T>> f_82244_ = HashBasedTable.create();
 
    private static <T> Comparator<TimerQueue.Event<T>> m_82270_() {
-      return Comparator.<TimerQueue.Event<T>>comparingLong((p_82272_) -> {
-         return p_82272_.f_82273_;
-      }).thenComparing((p_82269_) -> {
-         return p_82269_.f_82274_;
+      return Comparator.<TimerQueue.Event<T>>comparingLong((p_227578_0_) -> {
+         return p_227578_0_.f_82273_;
+      }).thenComparing((p_227577_0_) -> {
+         return p_227577_0_.f_82274_;
       });
    }
 
@@ -37,11 +_,11 @@
       this.f_82242_.clear();
       this.f_82244_.clear();
       this.f_82243_ = UnsignedLong.ZERO;
-      p_82250_.forEach((p_82253_) -> {
-         if (!(p_82253_.getValue() instanceof CompoundTag)) {
-            f_82240_.warn("Invalid format of events: {}", (Object)p_82253_);
+      p_82250_.forEach((p_237478_1_) -> {
+         if (!(p_237478_1_.getValue() instanceof CompoundTag)) {
+            f_82240_.warn("Invalid format of events: {}", (Object)p_237478_1_);
          } else {
-            this.m_82265_((CompoundTag)p_82253_.getValue());
+            this.m_82265_((CompoundTag)p_237478_1_.getValue());
          }
       });
    }
