--- a/net/minecraft/world/entity/raid/Raider.java
+++ b/net/minecraft/world/entity/raid/Raider.java
@@ -46,8 +_,8 @@
 
 public abstract class Raider extends PatrollingMonster {
    protected static final EntityDataAccessor<Boolean> f_37835_ = SynchedEntityData.m_135353_(Raider.class, EntityDataSerializers.f_135035_);
-   private static final Predicate<ItemEntity> f_37831_ = (p_37872_) -> {
-      return !p_37872_.m_32063_() && p_37872_.m_6084_() && ItemStack.m_41728_(p_37872_.m_32055_(), Raid.m_37779_());
+   private static final Predicate<ItemEntity> f_37831_ = (p_213647_0_) -> {
+      return !p_213647_0_.m_32063_() && p_213647_0_.m_6084_() && ItemStack.m_41728_(p_213647_0_.m_32055_(), Raid.m_37779_());
    };
    @Nullable
    protected Raid f_37836_;
@@ -278,6 +_,42 @@
 
    public abstract SoundEvent m_7930_();
 
+   public class RaiderCelebration extends Goal {
+      private final Raider f_37921_;
+
+      RaiderCelebration(Raider p_37924_) {
+         this.f_37921_ = p_37924_;
+         this.m_7021_(EnumSet.of(Goal.Flag.MOVE));
+      }
+
+      public boolean m_8036_() {
+         Raid raid = this.f_37921_.m_37885_();
+         return this.f_37921_.m_6084_() && this.f_37921_.m_5448_() == null && raid != null && raid.m_37768_();
+      }
+
+      public void m_8056_() {
+         this.f_37921_.m_37899_(true);
+         super.m_8056_();
+      }
+
+      public void m_8041_() {
+         this.f_37921_.m_37899_(false);
+         super.m_8041_();
+      }
+
+      public void m_8037_() {
+         if (!this.f_37921_.m_20067_() && this.f_37921_.f_19796_.nextInt(100) == 0) {
+            Raider.this.m_5496_(Raider.this.m_7930_(), Raider.this.m_6121_(), Raider.this.m_6100_());
+         }
+
+         if (!this.f_37921_.m_20159_() && this.f_37921_.f_19796_.nextInt(50) == 0) {
+            this.f_37921_.m_21569_().m_24901_();
+         }
+
+         super.m_8037_();
+      }
+   }
+
    public class HoldGroundAttackGoal extends Goal {
       private final Raider f_37903_;
       private final float f_37904_;
@@ -335,78 +_,6 @@
       }
    }
 
-   public class ObtainRaidLeaderBannerGoal<T extends Raider> extends Goal {
-      private final T f_37914_;
-
-      public ObtainRaidLeaderBannerGoal(T p_37917_) {
-         this.f_37914_ = p_37917_;
-         this.m_7021_(EnumSet.of(Goal.Flag.MOVE));
-      }
-
-      public boolean m_8036_() {
-         Raid raid = this.f_37914_.m_37885_();
-         if (this.f_37914_.m_37886_() && !this.f_37914_.m_37885_().m_37706_() && this.f_37914_.m_7490_() && !ItemStack.m_41728_(this.f_37914_.m_6844_(EquipmentSlot.HEAD), Raid.m_37779_())) {
-            Raider raider = raid.m_37750_(this.f_37914_.m_37887_());
-            if (raider == null || !raider.m_6084_()) {
-               List<ItemEntity> list = this.f_37914_.f_19853_.m_6443_(ItemEntity.class, this.f_37914_.m_20191_().m_82377_(16.0D, 8.0D, 16.0D), Raider.f_37831_);
-               if (!list.isEmpty()) {
-                  return this.f_37914_.m_21573_().m_5624_(list.get(0), (double)1.15F);
-               }
-            }
-
-            return false;
-         } else {
-            return false;
-         }
-      }
-
-      public void m_8037_() {
-         if (this.f_37914_.m_21573_().m_26567_().m_123306_(this.f_37914_.m_20182_(), 1.414D)) {
-            List<ItemEntity> list = this.f_37914_.f_19853_.m_6443_(ItemEntity.class, this.f_37914_.m_20191_().m_82377_(4.0D, 4.0D, 4.0D), Raider.f_37831_);
-            if (!list.isEmpty()) {
-               this.f_37914_.m_7581_(list.get(0));
-            }
-         }
-
-      }
-   }
-
-   public class RaiderCelebration extends Goal {
-      private final Raider f_37921_;
-
-      RaiderCelebration(Raider p_37924_) {
-         this.f_37921_ = p_37924_;
-         this.m_7021_(EnumSet.of(Goal.Flag.MOVE));
-      }
-
-      public boolean m_8036_() {
-         Raid raid = this.f_37921_.m_37885_();
-         return this.f_37921_.m_6084_() && this.f_37921_.m_5448_() == null && raid != null && raid.m_37768_();
-      }
-
-      public void m_8056_() {
-         this.f_37921_.m_37899_(true);
-         super.m_8056_();
-      }
-
-      public void m_8041_() {
-         this.f_37921_.m_37899_(false);
-         super.m_8041_();
-      }
-
-      public void m_8037_() {
-         if (!this.f_37921_.m_20067_() && this.f_37921_.f_19796_.nextInt(100) == 0) {
-            Raider.this.m_5496_(Raider.this.m_7930_(), Raider.this.m_6121_(), Raider.this.m_6100_());
-         }
-
-         if (!this.f_37921_.m_20159_() && this.f_37921_.f_19796_.nextInt(50) == 0) {
-            this.f_37921_.m_21569_().m_24901_();
-         }
-
-         super.m_8037_();
-      }
-   }
-
    static class RaiderMoveThroughVillageGoal extends Goal {
       private final Raider f_37929_;
       private final double f_37930_;
@@ -434,8 +_,8 @@
       private boolean m_37949_() {
          ServerLevel serverlevel = (ServerLevel)this.f_37929_.f_19853_;
          BlockPos blockpos = this.f_37929_.m_20183_();
-         Optional<BlockPos> optional = serverlevel.m_8904_().m_27126_((p_37941_) -> {
-            return p_37941_ == PoiType.f_27346_;
+         Optional<BlockPos> optional = serverlevel.m_8904_().m_27126_((p_220859_0_) -> {
+            return p_220859_0_ == PoiType.f_27346_;
          }, this::m_37942_, PoiManager.Occupancy.ANY, blockpos, 48, this.f_37929_.f_19796_);
          if (!optional.isPresent()) {
             return false;
@@ -498,6 +_,42 @@
       private void m_37950_() {
          if (this.f_37932_.size() > 2) {
             this.f_37932_.remove(0);
+         }
+
+      }
+   }
+
+   public class ObtainRaidLeaderBannerGoal<T extends Raider> extends Goal {
+      private final T f_37914_;
+
+      public ObtainRaidLeaderBannerGoal(T p_37917_) {
+         this.f_37914_ = p_37917_;
+         this.m_7021_(EnumSet.of(Goal.Flag.MOVE));
+      }
+
+      public boolean m_8036_() {
+         Raid raid = this.f_37914_.m_37885_();
+         if (this.f_37914_.m_37886_() && !this.f_37914_.m_37885_().m_37706_() && this.f_37914_.m_7490_() && !ItemStack.m_41728_(this.f_37914_.m_6844_(EquipmentSlot.HEAD), Raid.m_37779_())) {
+            Raider raider = raid.m_37750_(this.f_37914_.m_37887_());
+            if (raider == null || !raider.m_6084_()) {
+               List<ItemEntity> list = this.f_37914_.f_19853_.m_6443_(ItemEntity.class, this.f_37914_.m_20191_().m_82377_(16.0D, 8.0D, 16.0D), Raider.f_37831_);
+               if (!list.isEmpty()) {
+                  return this.f_37914_.m_21573_().m_5624_(list.get(0), (double)1.15F);
+               }
+            }
+
+            return false;
+         } else {
+            return false;
+         }
+      }
+
+      public void m_8037_() {
+         if (this.f_37914_.m_21573_().m_26567_().m_123306_(this.f_37914_.m_20182_(), 1.414D)) {
+            List<ItemEntity> list = this.f_37914_.f_19853_.m_6443_(ItemEntity.class, this.f_37914_.m_20191_().m_82377_(4.0D, 4.0D, 4.0D), Raider.f_37831_);
+            if (!list.isEmpty()) {
+               this.f_37914_.m_7581_(list.get(0));
+            }
          }
 
       }
