--- a/net/minecraft/server/commands/data/BlockDataAccessor.java
+++ b/net/minecraft/server/commands/data/BlockDataAccessor.java
@@ -20,10 +_,10 @@
 
 public class BlockDataAccessor implements DataAccessor {
    private static final SimpleCommandExceptionType f_139292_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.data.block.invalid"));
-   public static final Function<String, DataCommands.DataProvider> f_139291_ = (p_139305_) -> {
+   public static final Function<String, DataCommands.DataProvider> f_139291_ = (p_218923_0_) -> {
       return new DataCommands.DataProvider() {
          public DataAccessor m_7018_(CommandContext<CommandSourceStack> p_139319_) throws CommandSyntaxException {
-            BlockPos blockpos = BlockPosArgument.m_118242_(p_139319_, p_139305_ + "Pos");
+            BlockPos blockpos = BlockPosArgument.m_118242_(p_139319_, p_218923_0_ + "Pos");
             BlockEntity blockentity = ((CommandSourceStack)p_139319_.getSource()).m_81372_().m_7702_(blockpos);
             if (blockentity == null) {
                throw BlockDataAccessor.f_139292_.create();
@@ -33,7 +_,7 @@
          }
 
          public ArgumentBuilder<CommandSourceStack, ?> m_7621_(ArgumentBuilder<CommandSourceStack, ?> p_139316_, Function<ArgumentBuilder<CommandSourceStack, ?>, ArgumentBuilder<CommandSourceStack, ?>> p_139317_) {
-            return p_139316_.then(Commands.m_82127_("block").then((ArgumentBuilder)p_139317_.apply(Commands.m_82129_(p_139305_ + "Pos", BlockPosArgument.m_118239_()))));
+            return p_139316_.then(Commands.m_82127_("block").then((ArgumentBuilder)p_139317_.apply(Commands.m_82129_(p_218923_0_ + "Pos", BlockPosArgument.m_118239_()))));
          }
       };
    };
