--- a/net/minecraft/server/commands/data/DataCommands.java
+++ b/net/minecraft/server/commands/data/DataCommands.java
@@ -33,64 +_,64 @@
 
 public class DataCommands {
    private static final SimpleCommandExceptionType f_139352_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.data.merge.failed"));
-   private static final DynamicCommandExceptionType f_139353_ = new DynamicCommandExceptionType((p_139491_) -> {
-      return new TranslatableComponent("commands.data.get.invalid", p_139491_);
+   private static final DynamicCommandExceptionType f_139353_ = new DynamicCommandExceptionType((p_208922_0_) -> {
+      return new TranslatableComponent("commands.data.get.invalid", p_208922_0_);
    });
-   private static final DynamicCommandExceptionType f_139354_ = new DynamicCommandExceptionType((p_139481_) -> {
-      return new TranslatableComponent("commands.data.get.unknown", p_139481_);
+   private static final DynamicCommandExceptionType f_139354_ = new DynamicCommandExceptionType((p_208919_0_) -> {
+      return new TranslatableComponent("commands.data.get.unknown", p_208919_0_);
    });
    private static final SimpleCommandExceptionType f_139355_ = new SimpleCommandExceptionType(new TranslatableComponent("commands.data.get.multiple"));
-   private static final DynamicCommandExceptionType f_139356_ = new DynamicCommandExceptionType((p_139468_) -> {
-      return new TranslatableComponent("commands.data.modify.expected_list", p_139468_);
-   });
-   private static final DynamicCommandExceptionType f_139357_ = new DynamicCommandExceptionType((p_139448_) -> {
-      return new TranslatableComponent("commands.data.modify.expected_object", p_139448_);
-   });
-   private static final DynamicCommandExceptionType f_139358_ = new DynamicCommandExceptionType((p_139402_) -> {
-      return new TranslatableComponent("commands.data.modify.invalid_index", p_139402_);
+   private static final DynamicCommandExceptionType f_139356_ = new DynamicCommandExceptionType((p_218931_0_) -> {
+      return new TranslatableComponent("commands.data.modify.expected_list", p_218931_0_);
+   });
+   private static final DynamicCommandExceptionType f_139357_ = new DynamicCommandExceptionType((p_218948_0_) -> {
+      return new TranslatableComponent("commands.data.modify.expected_object", p_218948_0_);
+   });
+   private static final DynamicCommandExceptionType f_139358_ = new DynamicCommandExceptionType((p_218943_0_) -> {
+      return new TranslatableComponent("commands.data.modify.invalid_index", p_218943_0_);
    });
    public static final List<Function<String, DataCommands.DataProvider>> f_139349_ = ImmutableList.of(EntityDataAccessor.f_139505_, BlockDataAccessor.f_139291_, StorageDataAccessor.f_139531_);
-   public static final List<DataCommands.DataProvider> f_139350_ = f_139349_.stream().map((p_139450_) -> {
-      return p_139450_.apply("target");
+   public static final List<DataCommands.DataProvider> f_139350_ = f_139349_.stream().map((p_218925_0_) -> {
+      return p_218925_0_.apply("target");
    }).collect(ImmutableList.toImmutableList());
-   public static final List<DataCommands.DataProvider> f_139351_ = f_139349_.stream().map((p_139410_) -> {
-      return p_139410_.apply("source");
+   public static final List<DataCommands.DataProvider> f_139351_ = f_139349_.stream().map((p_218947_0_) -> {
+      return p_218947_0_.apply("source");
    }).collect(ImmutableList.toImmutableList());
 
    public static void m_139365_(CommandDispatcher<CommandSourceStack> p_139366_) {
-      LiteralArgumentBuilder<CommandSourceStack> literalargumentbuilder = Commands.m_82127_("data").requires((p_139381_) -> {
-         return p_139381_.m_6761_(2);
+      LiteralArgumentBuilder<CommandSourceStack> literalargumentbuilder = Commands.m_82127_("data").requires((p_198939_0_) -> {
+         return p_198939_0_.m_6761_(2);
       });
 
       for(DataCommands.DataProvider datacommands$dataprovider : f_139350_) {
-         literalargumentbuilder.then(datacommands$dataprovider.m_7621_(Commands.m_82127_("merge"), (p_139471_) -> {
-            return p_139471_.then(Commands.m_82129_("nbt", CompoundTagArgument.m_87657_()).executes((p_139494_) -> {
-               return m_139394_(p_139494_.getSource(), datacommands$dataprovider.m_7018_(p_139494_), CompoundTagArgument.m_87660_(p_139494_, "nbt"));
+         literalargumentbuilder.then(datacommands$dataprovider.m_7621_(Commands.m_82127_("merge"), (p_198943_1_) -> {
+            return p_198943_1_.then(Commands.m_82129_("nbt", CompoundTagArgument.m_87657_()).executes((p_198936_1_) -> {
+               return m_139394_(p_198936_1_.getSource(), datacommands$dataprovider.m_7018_(p_198936_1_), CompoundTagArgument.m_87660_(p_198936_1_, "nbt"));
             }));
-         })).then(datacommands$dataprovider.m_7621_(Commands.m_82127_("get"), (p_139453_) -> {
-            return p_139453_.executes((p_139484_) -> {
-               return m_139382_(p_139484_.getSource(), datacommands$dataprovider.m_7018_(p_139484_));
-            }).then(Commands.m_82129_("path", NbtPathArgument.m_99487_()).executes((p_139474_) -> {
-               return m_139443_(p_139474_.getSource(), datacommands$dataprovider.m_7018_(p_139474_), NbtPathArgument.m_99498_(p_139474_, "path"));
-            }).then(Commands.m_82129_("scale", DoubleArgumentType.doubleArg()).executes((p_139456_) -> {
-               return m_139389_(p_139456_.getSource(), datacommands$dataprovider.m_7018_(p_139456_), NbtPathArgument.m_99498_(p_139456_, "path"), DoubleArgumentType.getDouble(p_139456_, "scale"));
+         })).then(datacommands$dataprovider.m_7621_(Commands.m_82127_("get"), (p_198940_1_) -> {
+            return p_198940_1_.executes((p_198944_1_) -> {
+               return m_139382_(p_198944_1_.getSource(), datacommands$dataprovider.m_7018_(p_198944_1_));
+            }).then(Commands.m_82129_("path", NbtPathArgument.m_99487_()).executes((p_198945_1_) -> {
+               return m_139443_(p_198945_1_.getSource(), datacommands$dataprovider.m_7018_(p_198945_1_), NbtPathArgument.m_99498_(p_198945_1_, "path"));
+            }).then(Commands.m_82129_("scale", DoubleArgumentType.doubleArg()).executes((p_198935_1_) -> {
+               return m_139389_(p_198935_1_.getSource(), datacommands$dataprovider.m_7018_(p_198935_1_), NbtPathArgument.m_99498_(p_198935_1_, "path"), DoubleArgumentType.getDouble(p_198935_1_, "scale"));
             })));
-         })).then(datacommands$dataprovider.m_7621_(Commands.m_82127_("remove"), (p_139413_) -> {
-            return p_139413_.then(Commands.m_82129_("path", NbtPathArgument.m_99487_()).executes((p_139416_) -> {
-               return m_139385_(p_139416_.getSource(), datacommands$dataprovider.m_7018_(p_139416_), NbtPathArgument.m_99498_(p_139416_, "path"));
+         })).then(datacommands$dataprovider.m_7621_(Commands.m_82127_("remove"), (p_198934_1_) -> {
+            return p_198934_1_.then(Commands.m_82129_("path", NbtPathArgument.m_99487_()).executes((p_198941_1_) -> {
+               return m_139385_(p_198941_1_.getSource(), datacommands$dataprovider.m_7018_(p_198941_1_), NbtPathArgument.m_99498_(p_198941_1_, "path"));
             }));
-         })).then(m_139403_((p_139368_, p_139369_) -> {
-            p_139368_.then(Commands.m_82127_("insert").then(Commands.m_82129_("index", IntegerArgumentType.integer()).then(p_139369_.m_139500_((p_139486_, p_139487_, p_139488_, p_139489_) -> {
-               int i = IntegerArgumentType.getInteger(p_139486_, "index");
-               return m_139360_(i, p_139487_, p_139488_, p_139489_);
-            })))).then(Commands.m_82127_("prepend").then(p_139369_.m_139500_((p_139476_, p_139477_, p_139478_, p_139479_) -> {
-               return m_139360_(0, p_139477_, p_139478_, p_139479_);
-            }))).then(Commands.m_82127_("append").then(p_139369_.m_139500_((p_139463_, p_139464_, p_139465_, p_139466_) -> {
-               return m_139360_(-1, p_139464_, p_139465_, p_139466_);
-            }))).then(Commands.m_82127_("set").then(p_139369_.m_139500_((p_139439_, p_139440_, p_139441_, p_139442_) -> {
-               return p_139441_.m_99645_(p_139440_, Iterables.getLast(p_139442_)::m_6426_);
-            }))).then(Commands.m_82127_("merge").then(p_139369_.m_139500_((p_139371_, p_139372_, p_139373_, p_139374_) -> {
-               Collection<Tag> collection = p_139373_.m_99640_(p_139372_, CompoundTag::new);
+         })).then(m_139403_((p_218924_0_, p_218924_1_) -> {
+            p_218924_0_.then(Commands.m_82127_("insert").then(Commands.m_82129_("index", IntegerArgumentType.integer()).then(p_218924_1_.m_139500_((p_218930_0_, p_218930_1_, p_218930_2_, p_218930_3_) -> {
+               int i = IntegerArgumentType.getInteger(p_218930_0_, "index");
+               return m_139360_(i, p_218930_1_, p_218930_2_, p_218930_3_);
+            })))).then(Commands.m_82127_("prepend").then(p_218924_1_.m_139500_((p_218932_0_, p_218932_1_, p_218932_2_, p_218932_3_) -> {
+               return m_139360_(0, p_218932_1_, p_218932_2_, p_218932_3_);
+            }))).then(Commands.m_82127_("append").then(p_218924_1_.m_139500_((p_218941_0_, p_218941_1_, p_218941_2_, p_218941_3_) -> {
+               return m_139360_(-1, p_218941_1_, p_218941_2_, p_218941_3_);
+            }))).then(Commands.m_82127_("set").then(p_218924_1_.m_139500_((p_218954_0_, p_218954_1_, p_218954_2_, p_218954_3_) -> {
+               return p_218954_2_.m_99645_(p_218954_1_, Iterables.getLast(p_218954_3_)::m_6426_);
+            }))).then(Commands.m_82127_("merge").then(p_218924_1_.m_139500_((p_218927_0_, p_218927_1_, p_218927_2_, p_218927_3_) -> {
+               Collection<Tag> collection = p_218927_2_.m_99640_(p_218927_1_, CompoundTag::new);
                int i = 0;
 
                for(Tag tag : collection) {
@@ -101,7 +_,7 @@
                   CompoundTag compoundtag = (CompoundTag)tag;
                   CompoundTag compoundtag1 = compoundtag.m_6426_();
 
-                  for(Tag tag1 : p_139374_) {
+                  for(Tag tag1 : p_218927_3_) {
                      if (!(tag1 instanceof CompoundTag)) {
                         throw f_139357_.create(tag1);
                      }
@@ -154,32 +_,32 @@
       LiteralArgumentBuilder<CommandSourceStack> literalargumentbuilder = Commands.m_82127_("modify");
 
       for(DataCommands.DataProvider datacommands$dataprovider : f_139350_) {
-         datacommands$dataprovider.m_7621_(literalargumentbuilder, (p_139408_) -> {
+         datacommands$dataprovider.m_7621_(literalargumentbuilder, (p_218940_2_) -> {
             ArgumentBuilder<CommandSourceStack, ?> argumentbuilder = Commands.m_82129_("targetPath", NbtPathArgument.m_99487_());
 
             for(DataCommands.DataProvider datacommands$dataprovider1 : f_139351_) {
-               p_139404_.accept(argumentbuilder, (p_139427_) -> {
-                  return datacommands$dataprovider1.m_7621_(Commands.m_82127_("from"), (p_139432_) -> {
-                     return p_139432_.executes((p_139461_) -> {
-                        List<Tag> list = Collections.singletonList(datacommands$dataprovider1.m_7018_(p_139461_).m_6184_());
-                        return m_139375_(p_139461_, datacommands$dataprovider, p_139427_, list);
-                     }).then(Commands.m_82129_("sourcePath", NbtPathArgument.m_99487_()).executes((p_139437_) -> {
-                        DataAccessor dataaccessor = datacommands$dataprovider1.m_7018_(p_139437_);
-                        NbtPathArgument.NbtPath nbtpathargument$nbtpath = NbtPathArgument.m_99498_(p_139437_, "sourcePath");
+               p_139404_.accept(argumentbuilder, (p_218934_2_) -> {
+                  return datacommands$dataprovider1.m_7621_(Commands.m_82127_("from"), (p_218929_3_) -> {
+                     return p_218929_3_.executes((p_218937_3_) -> {
+                        List<Tag> list = Collections.singletonList(datacommands$dataprovider1.m_7018_(p_218937_3_).m_6184_());
+                        return m_139375_(p_218937_3_, datacommands$dataprovider, p_218934_2_, list);
+                     }).then(Commands.m_82129_("sourcePath", NbtPathArgument.m_99487_()).executes((p_218936_3_) -> {
+                        DataAccessor dataaccessor = datacommands$dataprovider1.m_7018_(p_218936_3_);
+                        NbtPathArgument.NbtPath nbtpathargument$nbtpath = NbtPathArgument.m_99498_(p_218936_3_, "sourcePath");
                         List<Tag> list = nbtpathargument$nbtpath.m_99638_(dataaccessor.m_6184_());
-                        return m_139375_(p_139437_, datacommands$dataprovider, p_139427_, list);
+                        return m_139375_(p_218936_3_, datacommands$dataprovider, p_218934_2_, list);
                      }));
                   });
                });
             }
 
-            p_139404_.accept(argumentbuilder, (p_139419_) -> {
-               return Commands.m_82127_("value").then(Commands.m_82129_("value", NbtTagArgument.m_100659_()).executes((p_139423_) -> {
-                  List<Tag> list = Collections.singletonList(NbtTagArgument.m_100662_(p_139423_, "value"));
-                  return m_139375_(p_139423_, datacommands$dataprovider, p_139419_, list);
+            p_139404_.accept(argumentbuilder, (p_218949_1_) -> {
+               return Commands.m_82127_("value").then(Commands.m_82129_("value", NbtTagArgument.m_100659_()).executes((p_218952_2_) -> {
+                  List<Tag> list = Collections.singletonList(NbtTagArgument.m_100662_(p_218952_2_, "value"));
+                  return m_139375_(p_218952_2_, datacommands$dataprovider, p_218949_1_, list);
                }));
             });
-            return p_139408_.then(argumentbuilder);
+            return p_218940_2_.then(argumentbuilder);
          });
       }
 
@@ -272,17 +_,17 @@
       }
    }
 
-   interface DataManipulator {
-      int m_139495_(CommandContext<CommandSourceStack> p_139496_, CompoundTag p_139497_, NbtPathArgument.NbtPath p_139498_, List<Tag> p_139499_) throws CommandSyntaxException;
-   }
-
-   interface DataManipulatorDecorator {
-      ArgumentBuilder<CommandSourceStack, ?> m_139500_(DataCommands.DataManipulator p_139501_);
-   }
-
    public interface DataProvider {
       DataAccessor m_7018_(CommandContext<CommandSourceStack> p_139504_) throws CommandSyntaxException;
 
       ArgumentBuilder<CommandSourceStack, ?> m_7621_(ArgumentBuilder<CommandSourceStack, ?> p_139502_, Function<ArgumentBuilder<CommandSourceStack, ?>, ArgumentBuilder<CommandSourceStack, ?>> p_139503_);
+   }
+
+   interface DataManipulatorDecorator {
+      ArgumentBuilder<CommandSourceStack, ?> m_139500_(DataCommands.DataManipulator p_139501_);
+   }
+
+   interface DataManipulator {
+      int m_139495_(CommandContext<CommandSourceStack> p_139496_, CompoundTag p_139497_, NbtPathArgument.NbtPath p_139498_, List<Tag> p_139499_) throws CommandSyntaxException;
    }
 }
