--- a/net/minecraft/server/commands/data/StorageDataAccessor.java
+++ b/net/minecraft/server/commands/data/StorageDataAccessor.java
@@ -18,17 +_,17 @@
 import net.minecraft.world.level.storage.CommandStorage;
 
 public class StorageDataAccessor implements DataAccessor {
-   private static final SuggestionProvider<CommandSourceStack> f_139532_ = (p_139547_, p_139548_) -> {
-      return SharedSuggestionProvider.m_82957_(m_139560_(p_139547_).m_78036_(), p_139548_);
+   private static final SuggestionProvider<CommandSourceStack> f_139532_ = (p_229838_0_, p_229838_1_) -> {
+      return SharedSuggestionProvider.m_82957_(m_139560_(p_229838_0_).m_78036_(), p_229838_1_);
    };
-   public static final Function<String, DataCommands.DataProvider> f_139531_ = (p_139554_) -> {
+   public static final Function<String, DataCommands.DataProvider> f_139531_ = (p_229839_0_) -> {
       return new DataCommands.DataProvider() {
          public DataAccessor m_7018_(CommandContext<CommandSourceStack> p_139570_) {
-            return new StorageDataAccessor(StorageDataAccessor.m_139560_(p_139570_), ResourceLocationArgument.m_107011_(p_139570_, p_139554_));
+            return new StorageDataAccessor(StorageDataAccessor.m_139560_(p_139570_), ResourceLocationArgument.m_107011_(p_139570_, p_229839_0_));
          }
 
          public ArgumentBuilder<CommandSourceStack, ?> m_7621_(ArgumentBuilder<CommandSourceStack, ?> p_139567_, Function<ArgumentBuilder<CommandSourceStack, ?>, ArgumentBuilder<CommandSourceStack, ?>> p_139568_) {
-            return p_139567_.then(Commands.m_82127_("storage").then((ArgumentBuilder)p_139568_.apply(Commands.m_82129_(p_139554_, ResourceLocationArgument.m_106984_()).suggests(StorageDataAccessor.f_139532_))));
+            return p_139567_.then(Commands.m_82127_("storage").then((ArgumentBuilder)p_139568_.apply(Commands.m_82129_(p_229839_0_, ResourceLocationArgument.m_106984_()).suggests(StorageDataAccessor.f_139532_))));
          }
       };
    };
