--- a/net/minecraft/client/renderer/DimensionSpecialEffects.java
+++ b/net/minecraft/client/renderer/DimensionSpecialEffects.java
@@ -13,12 +_,12 @@
 
 @OnlyIn(Dist.CLIENT)
 public abstract class DimensionSpecialEffects {
-   private static final Object2ObjectMap<ResourceLocation, DimensionSpecialEffects> f_108857_ = Util.m_137469_(new Object2ObjectArrayMap<>(), (p_108881_) -> {
+   private static final Object2ObjectMap<ResourceLocation, DimensionSpecialEffects> f_108857_ = Util.m_137469_(new Object2ObjectArrayMap<>(), (p_239214_0_) -> {
       DimensionSpecialEffects.OverworldEffects dimensionspecialeffects$overworldeffects = new DimensionSpecialEffects.OverworldEffects();
-      p_108881_.defaultReturnValue(dimensionspecialeffects$overworldeffects);
-      p_108881_.put(DimensionType.f_63840_, dimensionspecialeffects$overworldeffects);
-      p_108881_.put(DimensionType.f_63841_, new DimensionSpecialEffects.NetherEffects());
-      p_108881_.put(DimensionType.f_63842_, new DimensionSpecialEffects.EndEffects());
+      p_239214_0_.defaultReturnValue(dimensionspecialeffects$overworldeffects);
+      p_239214_0_.put(DimensionType.f_63840_, dimensionspecialeffects$overworldeffects);
+      p_239214_0_.put(DimensionType.f_63841_, new DimensionSpecialEffects.NetherEffects());
+      p_239214_0_.put(DimensionType.f_63842_, new DimensionSpecialEffects.EndEffects());
    });
    private final float[] f_108858_ = new float[4];
    private final float f_108859_;
@@ -26,6 +_,10 @@
    private final DimensionSpecialEffects.SkyType f_108861_;
    private final boolean f_108862_;
    private final boolean f_108863_;
+   private net.minecraftforge.client.IWeatherRenderHandler weatherRenderHandler = null;
+   private net.minecraftforge.client.IWeatherParticleRenderHandler weatherParticleRenderHandler = null;
+   private net.minecraftforge.client.ISkyRenderHandler skyRenderHandler = null;
+   private net.minecraftforge.client.ICloudRenderHandler cloudRenderHandler = null;
 
    public DimensionSpecialEffects(float p_108866_, boolean p_108867_, DimensionSpecialEffects.SkyType p_108868_, boolean p_108869_, boolean p_108870_) {
       this.f_108859_ = p_108866_;
@@ -82,6 +_,35 @@
       return this.f_108863_;
    }
 
+   public void setWeatherRenderHandler(net.minecraftforge.client.IWeatherRenderHandler weatherRenderHandler) {
+      this.weatherRenderHandler = weatherRenderHandler;
+   }
+   public void setWeatherParticleRenderHandler(net.minecraftforge.client.IWeatherParticleRenderHandler weatherParticleRenderHandler) {
+      this.weatherParticleRenderHandler = weatherParticleRenderHandler;
+   }
+   public void setSkyRenderHandler(net.minecraftforge.client.ISkyRenderHandler skyRenderHandler) {
+      this.skyRenderHandler = skyRenderHandler;
+   }
+   public void setCloudRenderHandler(net.minecraftforge.client.ICloudRenderHandler cloudRenderHandler) {
+      this.cloudRenderHandler = cloudRenderHandler;
+   }
+   @Nullable
+   public net.minecraftforge.client.ICloudRenderHandler getCloudRenderHandler() {
+      return cloudRenderHandler;
+   }
+   @Nullable
+   public net.minecraftforge.client.IWeatherRenderHandler getWeatherRenderHandler() {
+      return weatherRenderHandler;
+   }
+   @Nullable
+   public net.minecraftforge.client.IWeatherParticleRenderHandler getWeatherParticleRenderHandler() {
+      return weatherParticleRenderHandler;
+   }
+   @Nullable
+   public net.minecraftforge.client.ISkyRenderHandler getSkyRenderHandler() {
+      return skyRenderHandler;
+   }
+
    @OnlyIn(Dist.CLIENT)
    public static class EndEffects extends DimensionSpecialEffects {
       public EndEffects() {
@@ -103,6 +_,13 @@
    }
 
    @OnlyIn(Dist.CLIENT)
+   public static enum SkyType {
+      NONE,
+      NORMAL,
+      END;
+   }
+
+   @OnlyIn(Dist.CLIENT)
    public static class NetherEffects extends DimensionSpecialEffects {
       public NetherEffects() {
          super(Float.NaN, true, DimensionSpecialEffects.SkyType.NONE, false, true);
@@ -130,12 +_,5 @@
       public boolean m_5781_(int p_108905_, int p_108906_) {
          return false;
       }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   public static enum SkyType {
-      NONE,
-      NORMAL,
-      END;
    }
 }
