--- a/net/minecraft/client/renderer/RenderType.java
+++ b/net/minecraft/client/renderer/RenderType.java
@@ -218,11 +_,11 @@
    }
 
    public static RenderType m_110497_(ResourceLocation p_110498_) {
-      return m_110423_("text", DefaultVertexFormat.f_85820_, 7, 256, false, true, RenderType.CompositeState.m_110628_().m_110681_(new RenderStateShard.TextureStateShard(p_110498_, false, false)).m_110659_(f_110141_).m_110685_(f_110139_).m_110671_(f_110152_).m_110691_(false));
+      return net.minecraftforge.client.ForgeRenderTypes.getText(p_110498_);
    }
 
    public static RenderType m_110500_(ResourceLocation p_110501_) {
-      return m_110423_("text_see_through", DefaultVertexFormat.f_85820_, 7, 256, false, true, RenderType.CompositeState.m_110628_().m_110681_(new RenderStateShard.TextureStateShard(p_110501_, false, false)).m_110659_(f_110141_).m_110685_(f_110139_).m_110671_(f_110152_).m_110663_(f_110111_).m_110687_(f_110115_).m_110691_(false));
+      return net.minecraftforge.client.ForgeRenderTypes.getTextSeeThrough(p_110501_);
    }
 
    public static RenderType m_110502_() {
