--- a/net/minecraft/client/renderer/RenderType.java
+++ b/net/minecraft/client/renderer/RenderType.java
@@ -323,66 +_,19 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   static final class CompositeRenderType extends RenderType {
-      private static final ObjectOpenCustomHashSet<RenderType.CompositeRenderType> f_110510_ = new ObjectOpenCustomHashSet<>(RenderType.CompositeRenderType.EqualsStrategy.INSTANCE);
-      private final RenderType.CompositeState f_110511_;
-      private final int f_110512_;
-      private final Optional<RenderType> f_110513_;
-      private final boolean f_110514_;
-
-      private CompositeRenderType(String p_110517_, VertexFormat p_110518_, int p_110519_, int p_110520_, boolean p_110521_, boolean p_110522_, RenderType.CompositeState p_110523_) {
-         super(p_110517_, p_110518_, p_110519_, p_110520_, p_110521_, p_110522_, () -> {
-            p_110523_.f_110592_.forEach(RenderStateShard::m_110185_);
-         }, () -> {
-            p_110523_.f_110592_.forEach(RenderStateShard::m_110188_);
-         });
-         this.f_110511_ = p_110523_;
-         this.f_110513_ = p_110523_.f_110591_ == RenderType.OutlineProperty.AFFECTS_OUTLINE ? p_110523_.f_110576_.m_110340_().map((p_110530_) -> {
-            return m_110440_(p_110530_, p_110523_.f_110582_);
-         }) : Optional.empty();
-         this.f_110514_ = p_110523_.f_110591_ == RenderType.OutlineProperty.IS_OUTLINE;
-         this.f_110512_ = Objects.hash(super.hashCode(), p_110523_);
-      }
-
-      private static RenderType.CompositeRenderType m_110543_(String p_110544_, VertexFormat p_110545_, int p_110546_, int p_110547_, boolean p_110548_, boolean p_110549_, RenderType.CompositeState p_110550_) {
-         return f_110510_.addOrGet(new RenderType.CompositeRenderType(p_110544_, p_110545_, p_110546_, p_110547_, p_110548_, p_110549_, p_110550_));
-      }
-
-      public Optional<RenderType> m_7280_() {
-         return this.f_110513_;
-      }
-
-      public boolean m_5492_() {
-         return this.f_110514_;
-      }
-
-      public boolean equals(@Nullable Object p_110552_) {
-         return this == p_110552_;
-      }
-
-      public int hashCode() {
-         return this.f_110512_;
+   static enum OutlineProperty {
+      NONE("none"),
+      IS_OUTLINE("is_outline"),
+      AFFECTS_OUTLINE("affects_outline");
+
+      private final String f_110696_;
+
+      private OutlineProperty(String p_110702_) {
+         this.f_110696_ = p_110702_;
       }
 
       public String toString() {
-         return "RenderType[" + this.f_110511_ + ']';
-      }
-
-      @OnlyIn(Dist.CLIENT)
-      static enum EqualsStrategy implements Strategy<RenderType.CompositeRenderType> {
-         INSTANCE;
-
-         public int hashCode(@Nullable RenderType.CompositeRenderType p_110564_) {
-            return p_110564_ == null ? 0 : p_110564_.f_110512_;
-         }
-
-         public boolean equals(@Nullable RenderType.CompositeRenderType p_110566_, @Nullable RenderType.CompositeRenderType p_110567_) {
-            if (p_110566_ == p_110567_) {
-               return true;
-            } else {
-               return p_110566_ != null && p_110567_ != null ? Objects.equals(p_110566_.f_110511_, p_110567_.f_110511_) : false;
-            }
-         }
+         return this.f_110696_;
       }
    }
 
@@ -556,19 +_,66 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   static enum OutlineProperty {
-      NONE("none"),
-      IS_OUTLINE("is_outline"),
-      AFFECTS_OUTLINE("affects_outline");
-
-      private final String f_110696_;
-
-      private OutlineProperty(String p_110702_) {
-         this.f_110696_ = p_110702_;
+   static final class CompositeRenderType extends RenderType {
+      private static final ObjectOpenCustomHashSet<RenderType.CompositeRenderType> f_110510_ = new ObjectOpenCustomHashSet<>(RenderType.CompositeRenderType.EqualsStrategy.INSTANCE);
+      private final RenderType.CompositeState f_110511_;
+      private final int f_110512_;
+      private final Optional<RenderType> f_110513_;
+      private final boolean f_110514_;
+
+      private CompositeRenderType(String p_110517_, VertexFormat p_110518_, int p_110519_, int p_110520_, boolean p_110521_, boolean p_110522_, RenderType.CompositeState p_110523_) {
+         super(p_110517_, p_110518_, p_110519_, p_110520_, p_110521_, p_110522_, () -> {
+            p_110523_.f_110592_.forEach(RenderStateShard::m_110185_);
+         }, () -> {
+            p_110523_.f_110592_.forEach(RenderStateShard::m_110188_);
+         });
+         this.f_110511_ = p_110523_;
+         this.f_110513_ = p_110523_.f_110591_ == RenderType.OutlineProperty.AFFECTS_OUTLINE ? p_110523_.f_110576_.m_110340_().map((p_239275_1_) -> {
+            return m_110440_(p_239275_1_, p_110523_.f_110582_);
+         }) : Optional.empty();
+         this.f_110514_ = p_110523_.f_110591_ == RenderType.OutlineProperty.IS_OUTLINE;
+         this.f_110512_ = Objects.hash(super.hashCode(), p_110523_);
+      }
+
+      private static RenderType.CompositeRenderType m_110543_(String p_110544_, VertexFormat p_110545_, int p_110546_, int p_110547_, boolean p_110548_, boolean p_110549_, RenderType.CompositeState p_110550_) {
+         return f_110510_.addOrGet(new RenderType.CompositeRenderType(p_110544_, p_110545_, p_110546_, p_110547_, p_110548_, p_110549_, p_110550_));
+      }
+
+      public Optional<RenderType> m_7280_() {
+         return this.f_110513_;
+      }
+
+      public boolean m_5492_() {
+         return this.f_110514_;
+      }
+
+      public boolean equals(@Nullable Object p_110552_) {
+         return this == p_110552_;
+      }
+
+      public int hashCode() {
+         return this.f_110512_;
       }
 
       public String toString() {
-         return this.f_110696_;
+         return "RenderType[" + this.f_110511_ + ']';
+      }
+
+      @OnlyIn(Dist.CLIENT)
+      static enum EqualsStrategy implements Strategy<RenderType.CompositeRenderType> {
+         INSTANCE;
+
+         public int hashCode(@Nullable RenderType.CompositeRenderType p_110564_) {
+            return p_110564_ == null ? 0 : p_110564_.f_110512_;
+         }
+
+         public boolean equals(@Nullable RenderType.CompositeRenderType p_110566_, @Nullable RenderType.CompositeRenderType p_110567_) {
+            if (p_110566_ == p_110567_) {
+               return true;
+            } else {
+               return p_110566_ != null && p_110567_ != null ? Objects.equals(p_110566_.f_110511_, p_110567_.f_110511_) : false;
+            }
+         }
       }
    }
 }
