--- a/net/minecraft/client/renderer/RenderStateShard.java
+++ b/net/minecraft/client/renderer/RenderStateShard.java
@@ -504,13 +_,6 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   public static class OutputStateShard extends RenderStateShard {
-      public OutputStateShard(String p_110300_, Runnable p_110301_, Runnable p_110302_) {
-         super(p_110300_, p_110301_, p_110302_);
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
    public static class OverlayStateShard extends RenderStateShard.BooleanStateShard {
       public OverlayStateShard(boolean p_110304_) {
          super("overlay", () -> {
@@ -599,6 +_,13 @@
 
       public String toString() {
          return this.f_110133_ + '[' + (this.f_110318_ ? "smooth" : "flat") + ']';
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
+   public static class OutputStateShard extends RenderStateShard {
+      public OutputStateShard(String p_110300_, Runnable p_110301_, Runnable p_110302_) {
+         super(p_110300_, p_110301_, p_110302_);
       }
    }
 
