--- a/net/minecraft/world/phys/Vec3.java
+++ b/net/minecraft/world/phys/Vec3.java
@@ -111,7 +_,6 @@
       return this.m_82542_(p_82491_, p_82491_, p_82491_);
    }
 
-   @OnlyIn(Dist.CLIENT)
    public Vec3 m_82548_() {
       return this.m_82490_(-1.0D);
    }
@@ -190,12 +_,10 @@
       return new Vec3(d0, d1, d2);
    }
 
-   @OnlyIn(Dist.CLIENT)
    public static Vec3 m_82503_(Vec2 p_82504_) {
       return m_82498_(p_82504_.f_82470_, p_82504_.f_82471_);
    }
 
-   @OnlyIn(Dist.CLIENT)
    public static Vec3 m_82498_(float p_82499_, float p_82500_) {
       float f = Mth.m_14089_(-p_82500_ * ((float)Math.PI / 180F) - (float)Math.PI);
       float f1 = Mth.m_14031_(-p_82500_ * ((float)Math.PI / 180F) - (float)Math.PI);
