--- a/net/minecraft/data/advancements/AdvancementProvider.java
+++ b/net/minecraft/data/advancements/AdvancementProvider.java
@@ -30,14 +_,14 @@
    public void m_6865_(HashCache p_123969_) throws IOException {
       Path path = this.f_123962_.m_123916_();
       Set<ResourceLocation> set = Sets.newHashSet();
-      Consumer<Advancement> consumer = (p_123977_) -> {
-         if (!set.add(p_123977_.m_138327_())) {
-            throw new IllegalStateException("Duplicate advancement " + p_123977_.m_138327_());
+      Consumer<Advancement> consumer = (p_204017_3_) -> {
+         if (!set.add(p_204017_3_.m_138327_())) {
+            throw new IllegalStateException("Duplicate advancement " + p_204017_3_.m_138327_());
          } else {
-            Path path1 = m_123970_(path, p_123977_);
+            Path path1 = m_123970_(path, p_204017_3_);
 
             try {
-               DataProvider.m_123920_(f_123961_, p_123969_, p_123977_.m_138313_().m_138400_(), path1);
+               DataProvider.m_123920_(f_123961_, p_123969_, p_204017_3_.m_138313_().m_138400_(), path1);
             } catch (IOException ioexception) {
                f_123960_.error("Couldn't save advancement {}", path1, ioexception);
             }
