--- a/net/minecraft/data/advancements/HusbandryAdvancements.java
+++ b/net/minecraft/data/advancements/HusbandryAdvancements.java
@@ -86,8 +_,8 @@
    }
 
    private Advancement.Builder m_124013_(Advancement.Builder p_124014_) {
-      Cat.f_28102_.forEach((p_124003_, p_124004_) -> {
-         p_124014_.m_138386_(p_124004_.m_135815_(), TameAnimalTrigger.TriggerInstance.m_68848_(EntityPredicate.Builder.m_36633_().m_36660_(p_124004_).m_36662_()));
+      Cat.f_28102_.forEach((p_218461_1_, p_218461_2_) -> {
+         p_124014_.m_138386_(p_218461_2_.m_135815_(), TameAnimalTrigger.TriggerInstance.m_68848_(EntityPredicate.Builder.m_36633_().m_36660_(p_218461_2_).m_36662_()));
       });
       return p_124014_;
    }
