--- a/net/minecraft/server/packs/AbstractPackResources.java
+++ b/net/minecraft/server/packs/AbstractPackResources.java
@@ -19,7 +_,7 @@
 
 public abstract class AbstractPackResources implements PackResources {
    private static final Logger f_10204_ = LogManager.getLogger();
-   protected final File f_10203_;
+   public final File f_10203_;
 
    public AbstractPackResources(File p_10207_) {
       this.f_10203_ = p_10207_;
