--- a/net/minecraft/util/datafix/schemas/V1470.java
+++ b/net/minecraft/util/datafix/schemas/V1470.java
@@ -28,7 +_,7 @@
       m_17705_(p_17710_, map, "minecraft:phantom");
       m_17705_(p_17710_, map, "minecraft:dolphin");
       m_17705_(p_17710_, map, "minecraft:drowned");
-      p_17710_.register(map, "minecraft:trident", (p_17704_) -> {
+      p_17710_.register(map, "minecraft:trident", (p_206561_1_) -> {
          return DSL.optionalFields("inBlockState", References.f_16783_.in(p_17710_));
       });
       return map;
