--- a/net/minecraft/util/datafix/schemas/V1800.java
+++ b/net/minecraft/util/datafix/schemas/V1800.java
@@ -21,7 +_,7 @@
    public Map<String, Supplier<TypeTemplate>> registerEntities(Schema p_17744_) {
       Map<String, Supplier<TypeTemplate>> map = super.registerEntities(p_17744_);
       m_17739_(p_17744_, map, "minecraft:panda");
-      p_17744_.register(map, "minecraft:pillager", (p_17738_) -> {
+      p_17744_.register(map, "minecraft:pillager", (p_219875_1_) -> {
          return DSL.optionalFields("Inventory", DSL.list(References.f_16782_.in(p_17744_)), V100.m_17330_(p_17744_));
       });
       return map;
