--- a/net/minecraft/util/datafix/schemas/V705.java
+++ b/net/minecraft/util/datafix/schemas/V705.java
@@ -38,23 +_,23 @@
       Map<String, Supplier<TypeTemplate>> map = Maps.newHashMap();
       p_18148_.registerSimple(map, "minecraft:area_effect_cloud");
       m_18082_(p_18148_, map, "minecraft:armor_stand");
-      p_18148_.register(map, "minecraft:arrow", (p_18164_) -> {
+      p_18148_.register(map, "minecraft:arrow", (p_206582_1_) -> {
          return DSL.optionalFields("inTile", References.f_16787_.in(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:bat");
       m_18082_(p_18148_, map, "minecraft:blaze");
       p_18148_.registerSimple(map, "minecraft:boat");
       m_18082_(p_18148_, map, "minecraft:cave_spider");
-      p_18148_.register(map, "minecraft:chest_minecart", (p_18161_) -> {
+      p_18148_.register(map, "minecraft:chest_minecart", (p_206574_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18148_), "Items", DSL.list(References.f_16782_.in(p_18148_)));
       });
       m_18082_(p_18148_, map, "minecraft:chicken");
-      p_18148_.register(map, "minecraft:commandblock_minecart", (p_18158_) -> {
+      p_18148_.register(map, "minecraft:commandblock_minecart", (p_206575_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:cow");
       m_18082_(p_18148_, map, "minecraft:creeper");
-      p_18148_.register(map, "minecraft:donkey", (p_18155_) -> {
+      p_18148_.register(map, "minecraft:donkey", (p_206594_1_) -> {
          return DSL.optionalFields("Items", DSL.list(References.f_16782_.in(p_18148_)), "SaddleItem", References.f_16782_.in(p_18148_), V100.m_17330_(p_18148_));
       });
       p_18148_.registerSimple(map, "minecraft:dragon_fireball");
@@ -62,45 +_,45 @@
       m_18082_(p_18148_, map, "minecraft:elder_guardian");
       p_18148_.registerSimple(map, "minecraft:ender_crystal");
       m_18082_(p_18148_, map, "minecraft:ender_dragon");
-      p_18148_.register(map, "minecraft:enderman", (p_18146_) -> {
+      p_18148_.register(map, "minecraft:enderman", (p_206567_1_) -> {
          return DSL.optionalFields("carried", References.f_16787_.in(p_18148_), V100.m_17330_(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:endermite");
       m_18093_(p_18148_, map, "minecraft:ender_pearl");
       p_18148_.registerSimple(map, "minecraft:eye_of_ender_signal");
-      p_18148_.register(map, "minecraft:falling_block", (p_18143_) -> {
+      p_18148_.register(map, "minecraft:falling_block", (p_206586_1_) -> {
          return DSL.optionalFields("Block", References.f_16787_.in(p_18148_), "TileEntityData", References.f_16781_.in(p_18148_));
       });
       m_18093_(p_18148_, map, "minecraft:fireball");
-      p_18148_.register(map, "minecraft:fireworks_rocket", (p_18140_) -> {
+      p_18148_.register(map, "minecraft:fireworks_rocket", (p_206588_1_) -> {
          return DSL.optionalFields("FireworksItem", References.f_16782_.in(p_18148_));
       });
-      p_18148_.register(map, "minecraft:furnace_minecart", (p_18137_) -> {
+      p_18148_.register(map, "minecraft:furnace_minecart", (p_206570_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:ghast");
       m_18082_(p_18148_, map, "minecraft:giant");
       m_18082_(p_18148_, map, "minecraft:guardian");
-      p_18148_.register(map, "minecraft:hopper_minecart", (p_18134_) -> {
+      p_18148_.register(map, "minecraft:hopper_minecart", (p_206584_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18148_), "Items", DSL.list(References.f_16782_.in(p_18148_)));
       });
-      p_18148_.register(map, "minecraft:horse", (p_18131_) -> {
+      p_18148_.register(map, "minecraft:horse", (p_206595_1_) -> {
          return DSL.optionalFields("ArmorItem", References.f_16782_.in(p_18148_), "SaddleItem", References.f_16782_.in(p_18148_), V100.m_17330_(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:husk");
-      p_18148_.register(map, "minecraft:item", (p_18128_) -> {
+      p_18148_.register(map, "minecraft:item", (p_206578_1_) -> {
          return DSL.optionalFields("Item", References.f_16782_.in(p_18148_));
       });
-      p_18148_.register(map, "minecraft:item_frame", (p_18125_) -> {
+      p_18148_.register(map, "minecraft:item_frame", (p_206587_1_) -> {
          return DSL.optionalFields("Item", References.f_16782_.in(p_18148_));
       });
       p_18148_.registerSimple(map, "minecraft:leash_knot");
       m_18082_(p_18148_, map, "minecraft:magma_cube");
-      p_18148_.register(map, "minecraft:minecart", (p_18122_) -> {
+      p_18148_.register(map, "minecraft:minecart", (p_206568_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:mooshroom");
-      p_18148_.register(map, "minecraft:mule", (p_18119_) -> {
+      p_18148_.register(map, "minecraft:mule", (p_206579_1_) -> {
          return DSL.optionalFields("Items", DSL.list(References.f_16782_.in(p_18148_)), "SaddleItem", References.f_16782_.in(p_18148_), V100.m_17330_(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:ocelot");
@@ -108,7 +_,7 @@
       p_18148_.registerSimple(map, "minecraft:parrot");
       m_18082_(p_18148_, map, "minecraft:pig");
       m_18082_(p_18148_, map, "minecraft:polar_bear");
-      p_18148_.register(map, "minecraft:potion", (p_18116_) -> {
+      p_18148_.register(map, "minecraft:potion", (p_206573_1_) -> {
          return DSL.optionalFields("Potion", References.f_16782_.in(p_18148_), "inTile", References.f_16787_.in(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:rabbit");
@@ -117,27 +_,27 @@
       p_18148_.registerSimple(map, "minecraft:shulker_bullet");
       m_18082_(p_18148_, map, "minecraft:silverfish");
       m_18082_(p_18148_, map, "minecraft:skeleton");
-      p_18148_.register(map, "minecraft:skeleton_horse", (p_18113_) -> {
+      p_18148_.register(map, "minecraft:skeleton_horse", (p_206592_1_) -> {
          return DSL.optionalFields("SaddleItem", References.f_16782_.in(p_18148_), V100.m_17330_(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:slime");
       m_18093_(p_18148_, map, "minecraft:small_fireball");
       m_18093_(p_18148_, map, "minecraft:snowball");
       m_18082_(p_18148_, map, "minecraft:snowman");
-      p_18148_.register(map, "minecraft:spawner_minecart", (p_18110_) -> {
+      p_18148_.register(map, "minecraft:spawner_minecart", (p_206583_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18148_), References.f_16789_.in(p_18148_));
       });
-      p_18148_.register(map, "minecraft:spectral_arrow", (p_18107_) -> {
+      p_18148_.register(map, "minecraft:spectral_arrow", (p_206571_1_) -> {
          return DSL.optionalFields("inTile", References.f_16787_.in(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:spider");
       m_18082_(p_18148_, map, "minecraft:squid");
       m_18082_(p_18148_, map, "minecraft:stray");
       p_18148_.registerSimple(map, "minecraft:tnt");
-      p_18148_.register(map, "minecraft:tnt_minecart", (p_18104_) -> {
+      p_18148_.register(map, "minecraft:tnt_minecart", (p_206591_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18148_));
       });
-      p_18148_.register(map, "minecraft:villager", (p_18101_) -> {
+      p_18148_.register(map, "minecraft:villager", (p_206580_1_) -> {
          return DSL.optionalFields("Inventory", DSL.list(References.f_16782_.in(p_18148_)), "Offers", DSL.optionalFields("Recipes", DSL.list(DSL.optionalFields("buy", References.f_16782_.in(p_18148_), "buyB", References.f_16782_.in(p_18148_), "sell", References.f_16782_.in(p_18148_)))), V100.m_17330_(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:villager_golem");
@@ -149,7 +_,7 @@
       m_18093_(p_18148_, map, "minecraft:xp_bottle");
       p_18148_.registerSimple(map, "minecraft:xp_orb");
       m_18082_(p_18148_, map, "minecraft:zombie");
-      p_18148_.register(map, "minecraft:zombie_horse", (p_18092_) -> {
+      p_18148_.register(map, "minecraft:zombie_horse", (p_206569_1_) -> {
          return DSL.optionalFields("SaddleItem", References.f_16782_.in(p_18148_), V100.m_17330_(p_18148_));
       });
       m_18082_(p_18148_, map, "minecraft:zombie_pigman");
@@ -157,7 +_,7 @@
       p_18148_.registerSimple(map, "minecraft:evocation_fangs");
       m_18082_(p_18148_, map, "minecraft:evocation_illager");
       p_18148_.registerSimple(map, "minecraft:illusion_illager");
-      p_18148_.register(map, "minecraft:llama", (p_18081_) -> {
+      p_18148_.register(map, "minecraft:llama", (p_209329_1_) -> {
          return DSL.optionalFields("Items", DSL.list(References.f_16782_.in(p_18148_)), "SaddleItem", References.f_16782_.in(p_18148_), "DecorItem", References.f_16782_.in(p_18148_), V100.m_17330_(p_18148_));
       });
       p_18148_.registerSimple(map, "minecraft:llama_spit");
