--- a/net/minecraft/util/datafix/schemas/V99.java
+++ b/net/minecraft/util/datafix/schemas/V99.java
@@ -17,41 +_,41 @@
 
 public class V99 extends Schema {
    private static final Logger f_18181_ = LogManager.getLogger();
-   private static final Map<String, String> f_18182_ = DataFixUtils.make(Maps.newHashMap(), (p_18215_) -> {
-      p_18215_.put("minecraft:furnace", "Furnace");
-      p_18215_.put("minecraft:lit_furnace", "Furnace");
-      p_18215_.put("minecraft:chest", "Chest");
-      p_18215_.put("minecraft:trapped_chest", "Chest");
-      p_18215_.put("minecraft:ender_chest", "EnderChest");
-      p_18215_.put("minecraft:jukebox", "RecordPlayer");
-      p_18215_.put("minecraft:dispenser", "Trap");
-      p_18215_.put("minecraft:dropper", "Dropper");
-      p_18215_.put("minecraft:sign", "Sign");
-      p_18215_.put("minecraft:mob_spawner", "MobSpawner");
-      p_18215_.put("minecraft:noteblock", "Music");
-      p_18215_.put("minecraft:brewing_stand", "Cauldron");
-      p_18215_.put("minecraft:enhanting_table", "EnchantTable");
-      p_18215_.put("minecraft:command_block", "CommandBlock");
-      p_18215_.put("minecraft:beacon", "Beacon");
-      p_18215_.put("minecraft:skull", "Skull");
-      p_18215_.put("minecraft:daylight_detector", "DLDetector");
-      p_18215_.put("minecraft:hopper", "Hopper");
-      p_18215_.put("minecraft:banner", "Banner");
-      p_18215_.put("minecraft:flower_pot", "FlowerPot");
-      p_18215_.put("minecraft:repeating_command_block", "CommandBlock");
-      p_18215_.put("minecraft:chain_command_block", "CommandBlock");
-      p_18215_.put("minecraft:standing_sign", "Sign");
-      p_18215_.put("minecraft:wall_sign", "Sign");
-      p_18215_.put("minecraft:piston_head", "Piston");
-      p_18215_.put("minecraft:daylight_detector_inverted", "DLDetector");
-      p_18215_.put("minecraft:unpowered_comparator", "Comparator");
-      p_18215_.put("minecraft:powered_comparator", "Comparator");
-      p_18215_.put("minecraft:wall_banner", "Banner");
-      p_18215_.put("minecraft:standing_banner", "Banner");
-      p_18215_.put("minecraft:structure_block", "Structure");
-      p_18215_.put("minecraft:end_portal", "Airportal");
-      p_18215_.put("minecraft:end_gateway", "EndGateway");
-      p_18215_.put("minecraft:shield", "Banner");
+   private static final Map<String, String> f_18182_ = DataFixUtils.make(Maps.newHashMap(), (p_209320_0_) -> {
+      p_209320_0_.put("minecraft:furnace", "Furnace");
+      p_209320_0_.put("minecraft:lit_furnace", "Furnace");
+      p_209320_0_.put("minecraft:chest", "Chest");
+      p_209320_0_.put("minecraft:trapped_chest", "Chest");
+      p_209320_0_.put("minecraft:ender_chest", "EnderChest");
+      p_209320_0_.put("minecraft:jukebox", "RecordPlayer");
+      p_209320_0_.put("minecraft:dispenser", "Trap");
+      p_209320_0_.put("minecraft:dropper", "Dropper");
+      p_209320_0_.put("minecraft:sign", "Sign");
+      p_209320_0_.put("minecraft:mob_spawner", "MobSpawner");
+      p_209320_0_.put("minecraft:noteblock", "Music");
+      p_209320_0_.put("minecraft:brewing_stand", "Cauldron");
+      p_209320_0_.put("minecraft:enhanting_table", "EnchantTable");
+      p_209320_0_.put("minecraft:command_block", "CommandBlock");
+      p_209320_0_.put("minecraft:beacon", "Beacon");
+      p_209320_0_.put("minecraft:skull", "Skull");
+      p_209320_0_.put("minecraft:daylight_detector", "DLDetector");
+      p_209320_0_.put("minecraft:hopper", "Hopper");
+      p_209320_0_.put("minecraft:banner", "Banner");
+      p_209320_0_.put("minecraft:flower_pot", "FlowerPot");
+      p_209320_0_.put("minecraft:repeating_command_block", "CommandBlock");
+      p_209320_0_.put("minecraft:chain_command_block", "CommandBlock");
+      p_209320_0_.put("minecraft:standing_sign", "Sign");
+      p_209320_0_.put("minecraft:wall_sign", "Sign");
+      p_209320_0_.put("minecraft:piston_head", "Piston");
+      p_209320_0_.put("minecraft:daylight_detector_inverted", "DLDetector");
+      p_209320_0_.put("minecraft:unpowered_comparator", "Comparator");
+      p_209320_0_.put("minecraft:powered_comparator", "Comparator");
+      p_209320_0_.put("minecraft:wall_banner", "Banner");
+      p_209320_0_.put("minecraft:standing_banner", "Banner");
+      p_209320_0_.put("minecraft:structure_block", "Structure");
+      p_209320_0_.put("minecraft:end_portal", "Airportal");
+      p_209320_0_.put("minecraft:end_gateway", "EndGateway");
+      p_209320_0_.put("minecraft:shield", "Banner");
    });
    protected static final HookFunction f_18180_ = new HookFunction() {
       public <T> T apply(DynamicOps<T> p_18312_, T p_18313_) {
@@ -93,20 +_,20 @@
 
    public Map<String, Supplier<TypeTemplate>> registerEntities(Schema p_18305_) {
       Map<String, Supplier<TypeTemplate>> map = Maps.newHashMap();
-      p_18305_.register(map, "Item", (p_18301_) -> {
+      p_18305_.register(map, "Item", (p_206678_1_) -> {
          return DSL.optionalFields("Item", References.f_16782_.in(p_18305_));
       });
       p_18305_.registerSimple(map, "XPOrb");
       m_18224_(p_18305_, map, "ThrownEgg");
       p_18305_.registerSimple(map, "LeashKnot");
       p_18305_.registerSimple(map, "Painting");
-      p_18305_.register(map, "Arrow", (p_18298_) -> {
-         return DSL.optionalFields("inTile", References.f_16787_.in(p_18305_));
-      });
-      p_18305_.register(map, "TippedArrow", (p_18295_) -> {
-         return DSL.optionalFields("inTile", References.f_16787_.in(p_18305_));
-      });
-      p_18305_.register(map, "SpectralArrow", (p_18292_) -> {
+      p_18305_.register(map, "Arrow", (p_206682_1_) -> {
+         return DSL.optionalFields("inTile", References.f_16787_.in(p_18305_));
+      });
+      p_18305_.register(map, "TippedArrow", (p_206655_1_) -> {
+         return DSL.optionalFields("inTile", References.f_16787_.in(p_18305_));
+      });
+      p_18305_.register(map, "SpectralArrow", (p_206671_1_) -> {
          return DSL.optionalFields("inTile", References.f_16787_.in(p_18305_));
       });
       m_18224_(p_18305_, map, "Snowball");
@@ -114,19 +_,19 @@
       m_18224_(p_18305_, map, "SmallFireball");
       m_18224_(p_18305_, map, "ThrownEnderpearl");
       p_18305_.registerSimple(map, "EyeOfEnderSignal");
-      p_18305_.register(map, "ThrownPotion", (p_18289_) -> {
+      p_18305_.register(map, "ThrownPotion", (p_206688_1_) -> {
          return DSL.optionalFields("inTile", References.f_16787_.in(p_18305_), "Potion", References.f_16782_.in(p_18305_));
       });
       m_18224_(p_18305_, map, "ThrownExpBottle");
-      p_18305_.register(map, "ItemFrame", (p_18284_) -> {
+      p_18305_.register(map, "ItemFrame", (p_206661_1_) -> {
          return DSL.optionalFields("Item", References.f_16782_.in(p_18305_));
       });
       m_18224_(p_18305_, map, "WitherSkull");
       p_18305_.registerSimple(map, "PrimedTnt");
-      p_18305_.register(map, "FallingSand", (p_18279_) -> {
+      p_18305_.register(map, "FallingSand", (p_206679_1_) -> {
          return DSL.optionalFields("Block", References.f_16787_.in(p_18305_), "TileEntityData", References.f_16781_.in(p_18305_));
       });
-      p_18305_.register(map, "FireworksRocketEntity", (p_18274_) -> {
+      p_18305_.register(map, "FireworksRocketEntity", (p_206651_1_) -> {
          return DSL.optionalFields("FireworksItem", References.f_16782_.in(p_18305_));
       });
       p_18305_.registerSimple(map, "Boat");
@@ -134,7 +_,7 @@
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18305_), "Items", DSL.list(References.f_16782_.in(p_18305_)));
       });
       m_18236_(p_18305_, map, "MinecartRideable");
-      p_18305_.register(map, "MinecartChest", (p_18269_) -> {
+      p_18305_.register(map, "MinecartChest", (p_206663_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18305_), "Items", DSL.list(References.f_16782_.in(p_18305_)));
       });
       m_18236_(p_18305_, map, "MinecartFurnace");
@@ -142,7 +_,7 @@
       p_18305_.register(map, "MinecartSpawner", () -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18305_), References.f_16789_.in(p_18305_));
       });
-      p_18305_.register(map, "MinecartHopper", (p_18264_) -> {
+      p_18305_.register(map, "MinecartHopper", (p_210752_1_) -> {
          return DSL.optionalFields("DisplayTile", References.f_16787_.in(p_18305_), "Items", DSL.list(References.f_16782_.in(p_18305_)));
       });
       m_18236_(p_18305_, map, "MinecartCommandBlock");
@@ -155,7 +_,7 @@
       m_18193_(p_18305_, map, "Slime");
       m_18193_(p_18305_, map, "Ghast");
       m_18193_(p_18305_, map, "PigZombie");
-      p_18305_.register(map, "Enderman", (p_18259_) -> {
+      p_18305_.register(map, "Enderman", (p_206686_1_) -> {
          return DSL.optionalFields("carried", References.f_16787_.in(p_18305_), m_18188_(p_18305_));
       });
       m_18193_(p_18305_, map, "CaveSpider");
@@ -178,11 +_,11 @@
       m_18193_(p_18305_, map, "SnowMan");
       m_18193_(p_18305_, map, "Ozelot");
       m_18193_(p_18305_, map, "VillagerGolem");
-      p_18305_.register(map, "EntityHorse", (p_18254_) -> {
+      p_18305_.register(map, "EntityHorse", (p_206670_1_) -> {
          return DSL.optionalFields("Items", DSL.list(References.f_16782_.in(p_18305_)), "ArmorItem", References.f_16782_.in(p_18305_), "SaddleItem", References.f_16782_.in(p_18305_), m_18188_(p_18305_));
       });
       m_18193_(p_18305_, map, "Rabbit");
-      p_18305_.register(map, "Villager", (p_18245_) -> {
+      p_18305_.register(map, "Villager", (p_206656_1_) -> {
          return DSL.optionalFields("Inventory", DSL.list(References.f_16782_.in(p_18305_)), "Offers", DSL.optionalFields("Recipes", DSL.list(DSL.optionalFields("buy", References.f_16782_.in(p_18305_), "buyB", References.f_16782_.in(p_18305_), "sell", References.f_16782_.in(p_18305_)))), m_18188_(p_18305_));
       });
       p_18305_.registerSimple(map, "EnderCrystal");
@@ -197,13 +_,13 @@
       m_18246_(p_18303_, map, "Furnace");
       m_18246_(p_18303_, map, "Chest");
       p_18303_.registerSimple(map, "EnderChest");
-      p_18303_.register(map, "RecordPlayer", (p_18235_) -> {
+      p_18303_.register(map, "RecordPlayer", (p_206684_1_) -> {
          return DSL.optionalFields("RecordItem", References.f_16782_.in(p_18303_));
       });
       m_18246_(p_18303_, map, "Trap");
       m_18246_(p_18303_, map, "Dropper");
       p_18303_.registerSimple(map, "Sign");
-      p_18303_.register(map, "MobSpawner", (p_18223_) -> {
+      p_18303_.register(map, "MobSpawner", (p_206667_1_) -> {
          return References.f_16789_.in(p_18303_);
       });
       p_18303_.registerSimple(map, "Music");
@@ -217,7 +_,7 @@
       p_18303_.registerSimple(map, "DLDetector");
       m_18246_(p_18303_, map, "Hopper");
       p_18303_.registerSimple(map, "Comparator");
-      p_18303_.register(map, "FlowerPot", (p_18192_) -> {
+      p_18303_.register(map, "FlowerPot", (p_206653_1_) -> {
          return DSL.optionalFields("Item", DSL.or(DSL.constType(DSL.intType()), References.f_16788_.in(p_18303_)));
       });
       p_18303_.registerSimple(map, "Banner");
@@ -269,19 +_,19 @@
    }
 
    protected static <T> T m_18205_(Dynamic<T> p_18206_, Map<String, String> p_18207_, String p_18208_) {
-      return p_18206_.update("tag", (p_18213_) -> {
-         return p_18213_.update("BlockEntityTag", (p_18204_) -> {
+      return p_18206_.update("tag", (p_209868_3_) -> {
+         return p_209868_3_.update("BlockEntityTag", (p_209870_2_) -> {
             String s = p_18206_.get("id").asString("");
             String s1 = p_18207_.get(NamespacedSchema.m_17311_(s));
             if (s1 == null) {
                f_18181_.warn("Unable to resolve BlockEntity for ItemStack: {}", (Object)s);
-               return p_18204_;
+               return p_209870_2_;
             } else {
-               return p_18204_.set("id", p_18206_.createString(s1));
+               return p_209870_2_.set("id", p_18206_.createString(s1));
             }
-         }).update("EntityTag", (p_18200_) -> {
+         }).update("EntityTag", (p_209866_2_) -> {
             String s = p_18206_.get("id").asString("");
-            return Objects.equals(NamespacedSchema.m_17311_(s), "minecraft:armor_stand") ? p_18200_.set("id", p_18206_.createString(p_18208_)) : p_18200_;
+            return Objects.equals(NamespacedSchema.m_17311_(s), "minecraft:armor_stand") ? p_209866_2_.set("id", p_18206_.createString(p_18208_)) : p_209866_2_;
          });
       }).getValue();
    }
