--- a/net/minecraft/locale/Language.java
+++ b/net/minecraft/locale/Language.java
@@ -43,7 +_,8 @@
          f_128101_.error("Couldn't read strings from /assets/minecraft/lang/en_us.json", (Throwable)ioexception);
       }
 
-      final Map<String, String> map = builder.build();
+      final Map<String, String> map = new java.util.HashMap<>(builder.build());
+      net.minecraftforge.fml.server.LanguageHook.captureLanguageMap(map);
       return new Language() {
          public String m_6834_(String p_128127_) {
             return map.getOrDefault(p_128127_, p_128127_);
@@ -60,12 +_,17 @@
 
          @OnlyIn(Dist.CLIENT)
          public FormattedCharSequence m_5536_(FormattedText p_128129_) {
-            return (p_128132_) -> {
-               return p_128129_.m_7451_((p_128124_, p_128125_) -> {
-                  return StringDecomposer.m_14346_(p_128125_, p_128124_, p_128132_) ? Optional.empty() : FormattedText.f_130759_;
+            return (p_244262_1_) -> {
+               return p_128129_.m_7451_((p_244261_1_, p_244261_2_) -> {
+                  return StringDecomposer.m_14346_(p_244261_2_, p_244261_1_, p_244262_1_) ? Optional.empty() : FormattedText.f_130759_;
                }, Style.f_131099_).isPresent();
             };
          }
+         
+         @Override
+         public Map<String, String> getLanguageData() {
+            return map;
+         }
       };
    }
 
@@ -87,6 +_,9 @@
    public static void m_128114_(Language p_128115_) {
       f_128104_ = p_128115_;
    }
+   
+   // FORGE START
+   public Map<String, String> getLanguageData() { return ImmutableMap.of(); }
 
    public abstract String m_6834_(String p_128111_);
 
