--- a/com/mojang/blaze3d/platform/GlStateManager.java
+++ b/com/mojang/blaze3d/platform/GlStateManager.java
@@ -569,10 +_,17 @@
       GL13.glClientActiveTexture(p_84521_);
    }
 
+   /* Stores the last values sent into glMultiTexCoord2f */
+   public static float lastBrightnessX = 0.0f;
+   public static float lastBrightnessY = 0.0f;
    @Deprecated
    public static void m_84153_(int p_84154_, float p_84155_, float p_84156_) {
       RenderSystem.m_69393_(RenderSystem::m_69586_);
       GL13.glMultiTexCoord2f(p_84154_, p_84155_, p_84156_);
+      if (p_84154_ == GL13.GL_TEXTURE1) {
+          lastBrightnessX = p_84155_;
+          lastBrightnessY = p_84156_;
+       }
    }
 
    public static void m_84388_(int p_84389_, int p_84390_, int p_84391_, int p_84392_) {
