--- a/com/mojang/blaze3d/platform/InputConstants.java
+++ b/com/mojang/blaze3d/platform/InputConstants.java
@@ -76,7 +_,7 @@
 
    public static boolean m_84826_() {
       try {
-         return f_84823_ != null && (boolean)f_84823_.invokeExact();
+         return f_84823_ != null && (boolean) f_84823_.invokeExact();
       } catch (Throwable throwable) {
          throw new RuntimeException(throwable);
       }
@@ -173,16 +_,16 @@
 
    @OnlyIn(Dist.CLIENT)
    public static enum Type {
-      KEYSYM("key.keyboard", (p_84914_, p_84915_) -> {
-         String s = GLFW.glfwGetKeyName(p_84914_, -1);
-         return (Component)(s != null ? new TextComponent(s) : new TranslatableComponent(p_84915_));
-      }),
-      SCANCODE("scancode", (p_84911_, p_84912_) -> {
-         String s = GLFW.glfwGetKeyName(-1, p_84911_);
-         return (Component)(s != null ? new TextComponent(s) : new TranslatableComponent(p_84912_));
-      }),
-      MOUSE("key.mouse", (p_84904_, p_84905_) -> {
-         return Language.m_128107_().m_6722_(p_84905_) ? new TranslatableComponent(p_84905_) : new TranslatableComponent("key.mouse", p_84904_ + 1);
+      KEYSYM("key.keyboard", (p_237528_0_, p_237528_1_) -> {
+         String s = GLFW.glfwGetKeyName(p_237528_0_, -1);
+         return (Component)(s != null ? new TextComponent(s) : new TranslatableComponent(p_237528_1_));
+      }),
+      SCANCODE("scancode", (p_237527_0_, p_237527_1_) -> {
+         String s = GLFW.glfwGetKeyName(-1, p_237527_0_);
+         return (Component)(s != null ? new TextComponent(s) : new TranslatableComponent(p_237527_1_));
+      }),
+      MOUSE("key.mouse", (p_237524_0_, p_237524_1_) -> {
+         return Language.m_128107_().m_6722_(p_237524_1_) ? new TranslatableComponent(p_237524_1_) : new TranslatableComponent("key.mouse", p_237524_0_ + 1);
       });
 
       private final Int2ObjectMap<InputConstants.Key> f_84885_ = new Int2ObjectOpenHashMap<>();
@@ -200,14 +_,14 @@
       }
 
       public InputConstants.Key m_84895_(int p_84896_) {
-         return this.f_84885_.computeIfAbsent(p_84896_, (p_84907_) -> {
-            int i = p_84907_;
+         return this.f_84885_.computeIfAbsent(p_84896_, (p_237525_1_) -> {
+            int i = p_237525_1_;
             if (this == MOUSE) {
-               i = p_84907_ + 1;
+               i = p_237525_1_ + 1;
             }
 
             String s = this.f_84886_ + "." + i;
-            return new InputConstants.Key(s, this, p_84907_);
+            return new InputConstants.Key(s, this, p_237525_1_);
          });
       }
 
