--- a/net/minecraft/resources/RegistryReadOps.java
+++ b/net/minecraft/resources/RegistryReadOps.java
@@ -198,6 +_,7 @@
                ) {
                   JsonParser jsonparser = new JsonParser();
                   JsonElement jsonelement = jsonparser.parse(reader);
+                  if (jsonelement!= null) jsonelement.getAsJsonObject().addProperty("forge:registry_name", p_135727_.m_135782_().toString());
                   return p_135728_.parse(p_135725_, jsonelement).map((p_135730_) -> {
                      return Pair.of(p_135730_, OptionalInt.empty());
                   });
@@ -239,6 +_,7 @@
 
          public <E> DataResult<Pair<E, OptionalInt>> m_8082_(DynamicOps<JsonElement> p_135741_, ResourceKey<? extends Registry<E>> p_135742_, ResourceKey<E> p_135743_, Decoder<E> p_135744_) {
             JsonElement jsonelement = this.f_135736_.get(p_135743_);
+            if (jsonelement!= null) jsonelement.getAsJsonObject().addProperty("forge:registry_name", p_135743_.m_135782_().toString());
             return jsonelement == null ? DataResult.error("Unknown element: " + p_135743_) : p_135744_.parse(p_135741_, jsonelement).setLifecycle(this.f_135738_.get(p_135743_)).map((p_135756_) -> {
                return Pair.of(p_135756_, OptionalInt.of(this.f_135737_.getInt(p_135743_)));
             });
