--- a/net/minecraft/resources/RegistryDataPackCodec.java
+++ b/net/minecraft/resources/RegistryDataPackCodec.java
@@ -29,9 +_,9 @@
 
    public <T> DataResult<Pair<MappedRegistry<E>, T>> decode(DynamicOps<T> p_135563_, T p_135564_) {
       DataResult<Pair<MappedRegistry<E>, T>> dataresult = this.f_135541_.decode(p_135563_, p_135564_);
-      return p_135563_ instanceof RegistryReadOps ? dataresult.flatMap((p_135553_) -> {
-         return ((RegistryReadOps)p_135563_).m_135662_(p_135553_.getFirst(), this.f_135542_, this.f_135543_).map((p_135550_) -> {
-            return Pair.of(p_135550_, (T)p_135553_.getSecond());
+      return p_135563_ instanceof RegistryReadOps ? dataresult.flatMap((p_240862_2_) -> {
+         return ((RegistryReadOps)p_135563_).m_135662_(p_240862_2_.getFirst(), this.f_135542_, this.f_135543_).map((p_240861_1_) -> {
+            return Pair.of(p_240861_1_, (T)p_240862_2_.getSecond());
          });
       }) : dataresult;
    }
