--- a/net/minecraft/world/level/chunk/ImposterProtoChunk.java
+++ b/net/minecraft/world/level/chunk/ImposterProtoChunk.java
@@ -164,14 +_,14 @@
    }
 
    public ProtoTickList<Block> m_5782_() {
-      return new ProtoTickList<>((p_62694_) -> {
-         return p_62694_.m_49966_().m_60795_();
+      return new ProtoTickList<>((p_209219_0_) -> {
+         return p_209219_0_.m_49966_().m_60795_();
       }, this.m_7697_());
    }
 
    public ProtoTickList<Fluid> m_5783_() {
-      return new ProtoTickList<>((p_62717_) -> {
-         return p_62717_ == Fluids.f_76191_;
+      return new ProtoTickList<>((p_209218_0_) -> {
+         return p_209218_0_ == Fluids.f_76191_;
       }, this.m_7697_());
    }
 
