--- a/net/minecraft/client/gui/components/AbstractSelectionList.java
+++ b/net/minecraft/client/gui/components/AbstractSelectionList.java
@@ -7,7 +_,6 @@
 import com.mojang.blaze3d.vertex.DefaultVertexFormat;
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.mojang.blaze3d.vertex.Tesselator;
-import java.util.AbstractList;
 import java.util.Collection;
 import java.util.List;
 import java.util.Objects;
@@ -370,7 +_,7 @@
    }
 
    protected void m_6778_(AbstractSelectionList.SelectionDirection p_93463_) {
-      this.m_93464_(p_93463_, (p_93510_) -> {
+      this.m_93464_(p_93463_, (p_241573_0_) -> {
          return true;
       });
    }
@@ -492,6 +_,13 @@
       p_93506_.f_93521_ = this;
    }
 
+   public int getWidth() { return this.f_93388_; }
+   public int getHeight() { return this.f_93389_; }
+   public int getTop() { return this.f_93390_; }
+   public int getBottom() { return this.f_93391_; }
+   public int getLeft() { return this.f_93393_; }
+   public int getRight() { return this.f_93392_; }
+
    @OnlyIn(Dist.CLIENT)
    public abstract static class Entry<E extends AbstractSelectionList.Entry<E>> implements GuiEventListener {
       @Deprecated
@@ -511,7 +_,7 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   class TrackedList extends AbstractList<E> {
+   class TrackedList extends java.util.AbstractList<E> {
       private final List<E> f_93550_ = Lists.newArrayList();
 
       private TrackedList() {
