--- a/net/minecraft/client/gui/components/Button.java
+++ b/net/minecraft/client/gui/components/Button.java
@@ -7,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class Button extends AbstractButton {
-   public static final Button.OnTooltip f_93716_ = (p_93740_, p_93741_, p_93742_, p_93743_) -> {
+   public static final Button.OnTooltip f_93716_ = (p_238488_0_, p_238488_1_, p_238488_2_, p_238488_3_) -> {
    };
    protected final Button.OnPress f_93717_;
    protected final Button.OnTooltip f_93718_;
