--- a/net/minecraft/client/gui/components/ChatComponent.java
+++ b/net/minecraft/client/gui/components/ChatComponent.java
@@ -271,11 +_,11 @@
    }
 
    private void m_93803_(int p_93804_) {
-      this.f_93761_.removeIf((p_93807_) -> {
-         return p_93807_.m_90795_() == p_93804_;
+      this.f_93761_.removeIf((p_243251_1_) -> {
+         return p_243251_1_.m_90795_() == p_93804_;
       });
-      this.f_93760_.removeIf((p_93779_) -> {
-         return p_93779_.m_90795_() == p_93804_;
+      this.f_93760_.removeIf((p_243250_1_) -> {
+         return p_243250_1_.m_90795_() == p_93804_;
       });
    }
 
