--- a/net/minecraft/client/gui/components/CommandSuggestions.java
+++ b/net/minecraft/client/gui/components/CommandSuggestions.java
@@ -13,7 +_,6 @@
 import com.mojang.brigadier.context.SuggestionContext;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
 import com.mojang.brigadier.suggestion.Suggestion;
-import com.mojang.brigadier.suggestion.Suggestions;
 import com.mojang.brigadier.suggestion.SuggestionsBuilder;
 import com.mojang.brigadier.tree.CommandNode;
 import com.mojang.brigadier.tree.LiteralCommandNode;
@@ -67,7 +_,7 @@
    private int f_93862_;
    private int f_93863_;
    private ParseResults<SharedSuggestionProvider> f_93864_;
-   private CompletableFuture<Suggestions> f_93865_;
+   private CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> f_93865_;
    private CommandSuggestions.SuggestionsList f_93866_;
    private boolean f_93867_;
    private boolean f_93868_;
@@ -115,7 +_,7 @@
 
    public void m_93930_(boolean p_93931_) {
       if (this.f_93865_ != null && this.f_93865_.isDone()) {
-         Suggestions suggestions = this.f_93865_.join();
+         com.mojang.brigadier.suggestion.Suggestions suggestions = this.f_93865_.join();
          if (!suggestions.isEmpty()) {
             int i = 0;
 
@@ -131,7 +_,7 @@
 
    }
 
-   private List<Suggestion> m_93898_(Suggestions p_93899_) {
+   private List<Suggestion> m_93898_(com.mojang.brigadier.suggestion.Suggestions p_93899_) {
       String s = this.f_93853_.m_94155_().substring(0, this.f_93853_.m_94207_());
       int i = m_93912_(s);
       String s1 = s.substring(i).toLowerCase(Locale.ROOT);
