--- a/net/minecraft/client/gui/components/ComponentRenderUtils.java
+++ b/net/minecraft/client/gui/components/ComponentRenderUtils.java
@@ -24,14 +_,14 @@
 
    public static List<FormattedCharSequence> m_94005_(FormattedText p_94006_, int p_94007_, Font p_94008_) {
       ComponentCollector componentcollector = new ComponentCollector();
-      p_94006_.m_7451_((p_93997_, p_93998_) -> {
-         componentcollector.m_90675_(FormattedText.m_130762_(m_93999_(p_93998_), p_93997_));
+      p_94006_.m_7451_((p_238503_1_, p_238503_2_) -> {
+         componentcollector.m_90675_(FormattedText.m_130762_(m_93999_(p_238503_2_), p_238503_1_));
          return Optional.empty();
       }, Style.f_131099_);
       List<FormattedCharSequence> list = Lists.newArrayList();
-      p_94008_.m_92865_().m_92393_(componentcollector.m_90677_(), p_94007_, Style.f_131099_, (p_94003_, p_94004_) -> {
-         FormattedCharSequence formattedcharsequence = Language.m_128107_().m_5536_(p_94003_);
-         list.add(p_94004_ ? FormattedCharSequence.m_13696_(f_93993_, formattedcharsequence) : formattedcharsequence);
+      p_94008_.m_92865_().m_92393_(componentcollector.m_90677_(), p_94007_, Style.f_131099_, (p_243256_1_, p_243256_2_) -> {
+         FormattedCharSequence formattedcharsequence = Language.m_128107_().m_5536_(p_243256_1_);
+         list.add(p_243256_2_ ? FormattedCharSequence.m_13696_(f_93993_, formattedcharsequence) : formattedcharsequence);
       });
       return (List<FormattedCharSequence>)(list.isEmpty() ? Lists.newArrayList(FormattedCharSequence.f_13691_) : list);
    }
