--- a/net/minecraft/client/gui/components/DebugScreenOverlay.java
+++ b/net/minecraft/client/gui/components/DebugScreenOverlay.java
@@ -64,18 +_,18 @@
 
 @OnlyIn(Dist.CLIENT)
 public class DebugScreenOverlay extends GuiComponent {
-   private static final Map<Heightmap.Types, String> f_94029_ = Util.m_137469_(new EnumMap<>(Heightmap.Types.class), (p_94070_) -> {
-      p_94070_.put(Heightmap.Types.WORLD_SURFACE_WG, "SW");
-      p_94070_.put(Heightmap.Types.WORLD_SURFACE, "S");
-      p_94070_.put(Heightmap.Types.OCEAN_FLOOR_WG, "OW");
-      p_94070_.put(Heightmap.Types.OCEAN_FLOOR, "O");
-      p_94070_.put(Heightmap.Types.MOTION_BLOCKING, "M");
-      p_94070_.put(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, "ML");
+   private static final Map<Heightmap.Types, String> f_94029_ = Util.m_137469_(new EnumMap<>(Heightmap.Types.class), (p_212918_0_) -> {
+      p_212918_0_.put(Heightmap.Types.WORLD_SURFACE_WG, "SW");
+      p_212918_0_.put(Heightmap.Types.WORLD_SURFACE, "S");
+      p_212918_0_.put(Heightmap.Types.OCEAN_FLOOR_WG, "OW");
+      p_212918_0_.put(Heightmap.Types.OCEAN_FLOOR, "O");
+      p_212918_0_.put(Heightmap.Types.MOTION_BLOCKING, "M");
+      p_212918_0_.put(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, "ML");
    });
    private final Minecraft f_94030_;
    private final Font f_94031_;
-   private HitResult f_94032_;
-   private HitResult f_94033_;
+   protected HitResult f_94032_;
+   protected HitResult f_94033_;
    @Nullable
    private ChunkPos f_94034_;
    @Nullable
@@ -276,8 +_,8 @@
             if (naturalspawner$spawnstate != null) {
                Object2IntMap<MobCategory> object2intmap = naturalspawner$spawnstate.m_47148_();
                int l = naturalspawner$spawnstate.m_47126_();
-               list.add("SC: " + l + ", " + (String)Stream.of(MobCategory.values()).map((p_94068_) -> {
-                  return Character.toUpperCase(p_94068_.m_21607_().charAt(0)) + ": " + object2intmap.getInt(p_94068_);
+               list.add("SC: " + l + ", " + (String)Stream.of(MobCategory.values()).map((p_238511_1_) -> {
+                  return Character.toUpperCase(p_238511_1_.m_21607_().charAt(0)) + ": " + object2intmap.getInt(p_238511_1_);
                }).collect(Collectors.joining(", ")));
             } else {
                list.add("SC: N/A");
@@ -307,8 +_,8 @@
    }
 
    private Level m_94083_() {
-      return DataFixUtils.orElse(Optional.ofNullable(this.f_94030_.m_91092_()).flatMap((p_94065_) -> {
-         return Optional.ofNullable(p_94065_.m_129880_(this.f_94030_.f_91073_.m_46472_()));
+      return DataFixUtils.orElse(Optional.ofNullable(this.f_94030_.m_91092_()).flatMap((p_212917_1_) -> {
+         return Optional.ofNullable(p_212917_1_.m_129880_(this.f_94030_.f_91073_.m_46472_()));
       }), this.f_94030_.f_91073_);
    }
 
@@ -317,10 +_,10 @@
       if (this.f_94036_ == null) {
          ServerLevel serverlevel = this.m_94081_();
          if (serverlevel != null) {
-            this.f_94036_ = serverlevel.m_7726_().m_8431_(this.f_94034_.f_45578_, this.f_94034_.f_45579_, ChunkStatus.f_62326_, false).thenApply((p_94055_) -> {
-               return p_94055_.map((p_94053_) -> {
-                  return (LevelChunk)p_94053_;
-               }, (p_94074_) -> {
+            this.f_94036_ = serverlevel.m_7726_().m_8431_(this.f_94034_.f_45578_, this.f_94034_.f_45579_, ChunkStatus.f_62326_, false).thenApply((p_222802_0_) -> {
+               return p_222802_0_.map((p_222803_0_) -> {
+                  return (LevelChunk)p_222803_0_;
+               }, (p_222801_0_) -> {
                   return null;
                });
             });
@@ -362,7 +_,7 @@
                list.add(this.m_94071_(entry));
             }
 
-            for(ResourceLocation resourcelocation : this.f_94030_.m_91403_().m_105148_().m_5883_().m_13394_(blockstate.m_60734_())) {
+            for(ResourceLocation resourcelocation : blockstate.m_60734_().getTags()) {
                list.add("#" + resourcelocation);
             }
          }
@@ -378,7 +_,7 @@
                list.add(this.m_94071_(entry1));
             }
 
-            for(ResourceLocation resourcelocation1 : this.f_94030_.m_91403_().m_105148_().m_5879_().m_13394_(fluidstate.m_76152_())) {
+            for(ResourceLocation resourcelocation1 : fluidstate.m_76152_().getTags()) {
                list.add("#" + resourcelocation1);
             }
          }
@@ -388,6 +_,7 @@
             list.add("");
             list.add(ChatFormatting.UNDERLINE + "Targeted Entity");
             list.add(String.valueOf((Object)Registry.f_122826_.m_7981_(entity.m_6095_())));
+            entity.m_6095_().getTags().forEach(t -> list.add("#" + t));
          }
 
          return list;
