--- a/net/minecraft/client/gui/components/EditBox.java
+++ b/net/minecraft/client/gui/components/EditBox.java
@@ -45,8 +_,8 @@
    private String f_94088_;
    private Consumer<String> f_94089_;
    private Predicate<String> f_94090_ = Objects::nonNull;
-   private BiFunction<String, Integer, FormattedCharSequence> f_94091_ = (p_94147_, p_94148_) -> {
-      return FormattedCharSequence.m_13714_(p_94147_, Style.f_131099_);
+   private BiFunction<String, Integer, FormattedCharSequence> f_94091_ = (p_195610_0_, p_195610_1_) -> {
+      return FormattedCharSequence.m_13714_(p_195610_0_, Style.f_131099_);
    };
 
    public EditBox(Font p_94114_, int p_94115_, int p_94116_, int p_94117_, int p_94118_, Component p_94119_) {
