--- a/net/minecraft/client/gui/components/MultiLineLabel.java
+++ b/net/minecraft/client/gui/components/MultiLineLabel.java
@@ -36,20 +_,20 @@
    };
 
    static MultiLineLabel m_94341_(Font p_94342_, FormattedText p_94343_, int p_94344_) {
-      return m_94361_(p_94342_, p_94342_.m_92923_(p_94343_, p_94344_).stream().map((p_94374_) -> {
-         return new MultiLineLabel.TextWithWidth(p_94374_, p_94342_.m_92724_(p_94374_));
+      return m_94361_(p_94342_, p_94342_.m_92923_(p_94343_, p_94344_).stream().map((p_243264_1_) -> {
+         return new MultiLineLabel.TextWithWidth(p_243264_1_, p_94342_.m_92724_(p_243264_1_));
       }).collect(ImmutableList.toImmutableList()));
    }
 
    static MultiLineLabel m_94345_(Font p_94346_, FormattedText p_94347_, int p_94348_, int p_94349_) {
-      return m_94361_(p_94346_, p_94346_.m_92923_(p_94347_, p_94348_).stream().limit((long)p_94349_).map((p_94371_) -> {
-         return new MultiLineLabel.TextWithWidth(p_94371_, p_94346_.m_92724_(p_94371_));
+      return m_94361_(p_94346_, p_94346_.m_92923_(p_94347_, p_94348_).stream().limit((long)p_94349_).map((p_243263_1_) -> {
+         return new MultiLineLabel.TextWithWidth(p_243263_1_, p_94346_.m_92724_(p_243263_1_));
       }).collect(ImmutableList.toImmutableList()));
    }
 
    static MultiLineLabel m_94350_(Font p_94351_, Component... p_94352_) {
-      return m_94361_(p_94351_, Arrays.stream(p_94352_).map(Component::m_7532_).map((p_94360_) -> {
-         return new MultiLineLabel.TextWithWidth(p_94360_, p_94351_.m_92724_(p_94360_));
+      return m_94361_(p_94351_, Arrays.stream(p_94352_).map(Component::m_7532_).map((p_243261_1_) -> {
+         return new MultiLineLabel.TextWithWidth(p_243261_1_, p_94351_.m_92724_(p_243261_1_));
       }).collect(ImmutableList.toImmutableList()));
    }
 
