--- a/net/minecraft/world/entity/ai/util/RandomPos.java
+++ b/net/minecraft/world/entity/ai/util/RandomPos.java
@@ -114,8 +_,8 @@
             BlockPos blockpos3 = new BlockPos((double)j + p_26908_.m_20185_(), (double)k + p_26908_.m_20186_(), (double)l + p_26908_.m_20189_());
             if (blockpos3.m_123342_() >= 0 && blockpos3.m_123342_() <= p_26908_.f_19853_.m_45546_() && (!flag || p_26908_.m_21444_(blockpos3)) && (!p_26919_ || pathnavigation.m_6342_(blockpos3))) {
                if (p_26916_) {
-                  blockpos3 = m_26947_(blockpos3, random.nextInt(p_26917_ + 1) + p_26918_, p_26908_.f_19853_.m_45546_(), (p_26946_) -> {
-                     return p_26908_.f_19853_.m_8055_(p_26946_).m_60767_().m_76333_();
+                  blockpos3 = m_26947_(blockpos3, random.nextInt(p_26917_ + 1) + p_26918_, p_26908_.f_19853_.m_45546_(), (p_226341_1_) -> {
+                     return p_26908_.f_19853_.m_8055_(p_226341_1_).m_60767_().m_76333_();
                   });
                }
 
