--- a/net/minecraft/world/entity/EntitySelector.java
+++ b/net/minecraft/world/entity/EntitySelector.java
@@ -12,37 +_,37 @@
 public final class EntitySelector {
    public static final Predicate<Entity> f_20402_ = Entity::m_6084_;
    public static final Predicate<LivingEntity> f_20403_ = LivingEntity::m_6084_;
-   public static final Predicate<Entity> f_20404_ = (p_20442_) -> {
-      return p_20442_.m_6084_() && !p_20442_.m_20160_() && !p_20442_.m_20159_();
-   };
-   public static final Predicate<Entity> f_20405_ = (p_20440_) -> {
-      return p_20440_ instanceof Container && p_20440_.m_6084_();
-   };
-   public static final Predicate<Entity> f_20406_ = (p_20438_) -> {
-      return !(p_20438_ instanceof Player) || !p_20438_.m_5833_() && !((Player)p_20438_).m_7500_();
-   };
-   public static final Predicate<Entity> f_20407_ = (p_20436_) -> {
-      return !(p_20436_ instanceof Player) || !p_20436_.m_5833_() && !((Player)p_20436_).m_7500_() && p_20436_.f_19853_.m_46791_() != Difficulty.PEACEFUL;
-   };
-   public static final Predicate<Entity> f_20408_ = (p_20434_) -> {
-      return !p_20434_.m_5833_();
+   public static final Predicate<Entity> f_20404_ = (p_200821_0_) -> {
+      return p_200821_0_.m_6084_() && !p_200821_0_.m_20160_() && !p_200821_0_.m_20159_();
+   };
+   public static final Predicate<Entity> f_20405_ = (p_200822_0_) -> {
+      return p_200822_0_ instanceof Container && p_200822_0_.m_6084_();
+   };
+   public static final Predicate<Entity> f_20406_ = (p_200824_0_) -> {
+      return !(p_200824_0_ instanceof Player) || !p_200824_0_.m_5833_() && !((Player)p_200824_0_).m_7500_();
+   };
+   public static final Predicate<Entity> f_20407_ = (p_200818_0_) -> {
+      return !(p_200818_0_ instanceof Player) || !p_200818_0_.m_5833_() && !((Player)p_200818_0_).m_7500_() && p_200818_0_.f_19853_.m_46791_() != Difficulty.PEACEFUL;
+   };
+   public static final Predicate<Entity> f_20408_ = (p_233587_0_) -> {
+      return !p_233587_0_.m_5833_();
    };
 
    public static Predicate<Entity> m_20410_(double p_20411_, double p_20412_, double p_20413_, double p_20414_) {
       double d0 = p_20414_ * p_20414_;
-      return (p_20420_) -> {
-         return p_20420_ != null && p_20420_.m_20275_(p_20411_, p_20412_, p_20413_) <= d0;
+      return (p_233584_8_) -> {
+         return p_233584_8_ != null && p_233584_8_.m_20275_(p_20411_, p_20412_, p_20413_) <= d0;
       };
    }
 
    public static Predicate<Entity> m_20421_(Entity p_20422_) {
       Team team = p_20422_.m_5647_();
       Team.CollisionRule team$collisionrule = team == null ? Team.CollisionRule.ALWAYS : team.m_7156_();
-      return (Predicate<Entity>)(team$collisionrule == Team.CollisionRule.NEVER ? Predicates.alwaysFalse() : f_20408_.and((p_20430_) -> {
-         if (!p_20430_.m_6094_()) {
+      return (Predicate<Entity>)(team$collisionrule == Team.CollisionRule.NEVER ? Predicates.alwaysFalse() : f_20408_.and((p_233586_3_) -> {
+         if (!p_233586_3_.m_6094_()) {
             return false;
-         } else if (!p_20422_.f_19853_.f_46443_ || p_20430_ instanceof Player && ((Player)p_20430_).m_7578_()) {
-            Team team1 = p_20430_.m_5647_();
+         } else if (!p_20422_.f_19853_.f_46443_ || p_233586_3_ instanceof Player && ((Player)p_233586_3_).m_7578_()) {
+            Team team1 = p_233586_3_.m_5647_();
             Team.CollisionRule team$collisionrule1 = team1 == null ? Team.CollisionRule.ALWAYS : team1.m_7156_();
             if (team$collisionrule1 == Team.CollisionRule.NEVER) {
                return false;
@@ -61,11 +_,11 @@
    }
 
    public static Predicate<Entity> m_20431_(Entity p_20432_) {
-      return (p_20425_) -> {
+      return (p_233585_1_) -> {
          while(true) {
-            if (p_20425_.m_20159_()) {
-               p_20425_ = p_20425_.m_20202_();
-               if (p_20425_ != p_20432_) {
+            if (p_233585_1_.m_20159_()) {
+               p_233585_1_ = p_233585_1_.m_20202_();
+               if (p_233585_1_ != p_20432_) {
                   continue;
                }
 
