--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -59,7 +_,6 @@
 import net.minecraft.world.item.SpawnEggItem;
 import net.minecraft.world.item.SwordItem;
 import net.minecraft.world.item.enchantment.EnchantmentHelper;
-import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.LevelAccessor;
 import net.minecraft.world.level.LevelReader;
@@ -81,8 +_,8 @@
    protected JumpControl f_21343_;
    private final BodyRotationControl f_21361_;
    protected PathNavigation f_21344_;
-   protected final GoalSelector f_21345_;
-   protected final GoalSelector f_21346_;
+   public final GoalSelector f_21345_;
+   public final GoalSelector f_21346_;
    private LivingEntity f_21362_;
    private final Sensing f_21349_;
    private final NonNullList<ItemStack> f_21350_ = NonNullList.m_122780_(2, ItemStack.f_41583_);
@@ -196,6 +_,7 @@
 
    public void m_6710_(@Nullable LivingEntity p_21544_) {
       this.f_21362_ = p_21544_;
+      net.minecraftforge.common.ForgeHooks.onLivingSetAttackTarget(this, p_21544_);
    }
 
    public boolean m_6549_(EntityType<?> p_21399_) {
@@ -484,7 +_,7 @@
    public void m_8107_() {
       super.m_8107_();
       this.f_19853_.m_46473_().m_6180_("looting");
-      if (!this.f_19853_.f_46443_ && this.m_21531_() && this.m_6084_() && !this.f_20890_ && this.f_19853_.m_46469_().m_46207_(GameRules.f_46132_)) {
+      if (!this.f_19853_.f_46443_ && this.m_21531_() && this.m_6084_() && !this.f_20890_ && net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(this.f_19853_, this)) {
          for(ItemEntity itementity : this.f_19853_.m_45976_(ItemEntity.class, this.m_20191_().m_82377_(1.0D, 0.0D, 1.0D))) {
             if (!itementity.f_19866_ && !itementity.m_32055_().m_41619_() && !itementity.m_32063_() && this.m_7243_(itementity.m_32055_())) {
                this.m_7581_(itementity);
@@ -599,10 +_,10 @@
    public boolean m_21477_(ItemStack p_21478_, ItemStack p_21479_) {
       if (p_21478_.m_41773_() >= p_21479_.m_41773_() && (!p_21478_.m_41782_() || p_21479_.m_41782_())) {
          if (p_21478_.m_41782_() && p_21479_.m_41782_()) {
-            return p_21478_.m_41783_().m_128431_().stream().anyMatch((p_21513_) -> {
-               return !p_21513_.equals("Damage");
-            }) && !p_21479_.m_41783_().m_128431_().stream().anyMatch((p_21503_) -> {
-               return !p_21503_.equals("Damage");
+            return p_21478_.m_41783_().m_128431_().stream().anyMatch((p_233664_0_) -> {
+               return !p_233664_0_.equals("Damage");
+            }) && !p_21479_.m_41783_().m_128431_().stream().anyMatch((p_233662_0_) -> {
+               return !p_233662_0_.equals("Damage");
             });
          } else {
             return false;
@@ -637,6 +_,14 @@
          this.m_6091_();
       } else if (!this.m_21532_() && !this.m_8023_()) {
          Entity entity = this.f_19853_.m_45930_(this, -1.0D);
+         net.minecraftforge.eventbus.api.Event.Result result = net.minecraftforge.event.ForgeEventFactory.canEntityDespawn(this);
+         if (result == net.minecraftforge.eventbus.api.Event.Result.DENY) {
+            f_20891_ = 0;
+            entity = null;
+         } else if (result == net.minecraftforge.eventbus.api.Event.Result.ALLOW) {
+            this.m_6091_();
+            entity = null;
+         }
          if (entity != null) {
             double d0 = entity.m_20280_(this);
             int i = this.m_6095_().m_20674_().m_21611_();
@@ -877,6 +_,8 @@
    }
 
    public static EquipmentSlot m_21548_(ItemStack p_21549_) {
+      final EquipmentSlot slot = p_21549_.getEquipmentSlot();
+      if (slot != null) return slot; // FORGE: Allow modders to set a non-default equipment slot for a stack; e.g. a non-armor chestplate-slot item
       Item item = p_21549_.m_41720_();
       if (item != Blocks.f_50143_.m_5456_() && (!(item instanceof BlockItem) || !(((BlockItem)item).m_40614_() instanceof AbstractSkullBlock))) {
          if (item instanceof ArmorItem) {
@@ -884,7 +_,7 @@
          } else if (item == Items.f_42741_) {
             return EquipmentSlot.CHEST;
          } else {
-            return item == Items.f_42740_ ? EquipmentSlot.OFFHAND : EquipmentSlot.MAINHAND;
+            return p_21549_.isShield(null) ? EquipmentSlot.OFFHAND : EquipmentSlot.MAINHAND;
          }
       } else {
          return EquipmentSlot.HEAD;
@@ -1057,8 +_,8 @@
             if (this.f_19853_ instanceof ServerLevel) {
                SpawnEggItem spawneggitem = (SpawnEggItem)itemstack.m_41720_();
                Optional<Mob> optional = spawneggitem.m_43215_(p_21500_, this, (EntityType)this.m_6095_(), (ServerLevel)this.f_19853_, this.m_20182_(), itemstack);
-               optional.ifPresent((p_21476_) -> {
-                  this.m_5502_(p_21500_, p_21476_);
+               optional.ifPresent((p_233658_2_) -> {
+                  this.m_5502_(p_21500_, p_233658_2_);
                });
                return optional.isPresent() ? InteractionResult.SUCCESS : InteractionResult.PASS;
             } else {
