--- a/net/minecraft/client/gui/screens/achievement/StatsScreen.java
+++ b/net/minecraft/client/gui/screens/achievement/StatsScreen.java
@@ -6,7 +_,6 @@
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.mojang.blaze3d.vertex.Tesselator;
 import it.unimi.dsi.fastutil.objects.ObjectArrayList;
-import java.util.Comparator;
 import java.util.List;
 import java.util.Set;
 import javax.annotation.Nullable;
@@ -67,16 +_,16 @@
    }
 
    public void m_96975_() {
-      this.m_7082_(new Button(this.f_96543_ / 2 - 120, this.f_96544_ - 52, 80, 20, new TranslatableComponent("stat.generalButton"), (p_96963_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 120, this.f_96544_ - 52, 80, 20, new TranslatableComponent("stat.generalButton"), (p_213109_1_) -> {
          this.m_96924_(this.f_96898_);
       }));
-      Button button = this.m_7082_(new Button(this.f_96543_ / 2 - 40, this.f_96544_ - 52, 80, 20, new TranslatableComponent("stat.itemsButton"), (p_96959_) -> {
+      Button button = this.m_7082_(new Button(this.f_96543_ / 2 - 40, this.f_96544_ - 52, 80, 20, new TranslatableComponent("stat.itemsButton"), (p_213115_1_) -> {
          this.m_96924_(this.f_96899_);
       }));
-      Button button1 = this.m_7082_(new Button(this.f_96543_ / 2 + 40, this.f_96544_ - 52, 80, 20, new TranslatableComponent("stat.mobsButton"), (p_96949_) -> {
+      Button button1 = this.m_7082_(new Button(this.f_96543_ / 2 + 40, this.f_96544_ - 52, 80, 20, new TranslatableComponent("stat.mobsButton"), (p_213114_1_) -> {
          this.m_96924_(this.f_96900_);
       }));
-      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ - 28, 200, 20, CommonComponents.f_130655_, (p_96923_) -> {
+      this.m_7082_(new Button(this.f_96543_ / 2 - 100, this.f_96544_ - 28, 200, 20, CommonComponents.f_130655_, (p_213113_1_) -> {
          this.f_96541_.m_91152_(this.f_96896_);
       }));
       if (this.f_96899_.m_6702_().isEmpty()) {
@@ -158,8 +_,8 @@
       public GeneralStatisticsList(Minecraft p_96995_) {
          super(p_96995_, StatsScreen.this.f_96543_, StatsScreen.this.f_96544_, 32, StatsScreen.this.f_96544_ - 64, 10);
          ObjectArrayList<Stat<ResourceLocation>> objectarraylist = new ObjectArrayList<>(Stats.f_12988_.iterator());
-         objectarraylist.sort(Comparator.comparing((p_96997_) -> {
-            return I18n.m_118938_(StatsScreen.m_96946_(p_96997_));
+         objectarraylist.sort(java.util.Comparator.comparing((p_238679_0_) -> {
+            return I18n.m_118938_(StatsScreen.m_96946_(p_238679_0_));
          }));
 
          for(Stat<ResourceLocation> stat : objectarraylist) {
@@ -191,13 +_,70 @@
    }
 
    @OnlyIn(Dist.CLIENT)
+   class MobsStatisticsList extends ObjectSelectionList<StatsScreen.MobsStatisticsList.MobRow> {
+      public MobsStatisticsList(Minecraft p_97100_) {
+         super(p_97100_, StatsScreen.this.f_96543_, StatsScreen.this.f_96544_, 32, StatsScreen.this.f_96544_ - 64, 9 * 4);
+
+         for(EntityType<?> entitytype : Registry.f_122826_) {
+            if (StatsScreen.this.f_96901_.m_13015_(Stats.f_12986_.m_12902_(entitytype)) > 0 || StatsScreen.this.f_96901_.m_13015_(Stats.f_12987_.m_12902_(entitytype)) > 0) {
+               this.m_7085_(new StatsScreen.MobsStatisticsList.MobRow(entitytype));
+            }
+         }
+
+      }
+
+      protected void m_7733_(PoseStack p_97102_) {
+         StatsScreen.this.m_7333_(p_97102_);
+      }
+
+      @OnlyIn(Dist.CLIENT)
+      class MobRow extends ObjectSelectionList.Entry<StatsScreen.MobsStatisticsList.MobRow> {
+         private final EntityType<?> f_97104_;
+         private final Component f_97105_;
+         private final Component f_97106_;
+         private final boolean f_97107_;
+         private final Component f_97108_;
+         private final boolean f_97109_;
+
+         public MobRow(EntityType<?> p_97112_) {
+            this.f_97104_ = p_97112_;
+            this.f_97105_ = p_97112_.m_20676_();
+            int i = StatsScreen.this.f_96901_.m_13015_(Stats.f_12986_.m_12902_(p_97112_));
+            if (i == 0) {
+               this.f_97106_ = new TranslatableComponent("stat_type.minecraft.killed.none", this.f_97105_);
+               this.f_97107_ = false;
+            } else {
+               this.f_97106_ = new TranslatableComponent("stat_type.minecraft.killed", i, this.f_97105_);
+               this.f_97107_ = true;
+            }
+
+            int j = StatsScreen.this.f_96901_.m_13015_(Stats.f_12987_.m_12902_(p_97112_));
+            if (j == 0) {
+               this.f_97108_ = new TranslatableComponent("stat_type.minecraft.killed_by.none", this.f_97105_);
+               this.f_97109_ = false;
+            } else {
+               this.f_97108_ = new TranslatableComponent("stat_type.minecraft.killed_by", this.f_97105_, j);
+               this.f_97109_ = true;
+            }
+
+         }
+
+         public void m_6311_(PoseStack p_97114_, int p_97115_, int p_97116_, int p_97117_, int p_97118_, int p_97119_, int p_97120_, int p_97121_, boolean p_97122_, float p_97123_) {
+            GuiComponent.m_93243_(p_97114_, StatsScreen.this.f_96547_, this.f_97105_, p_97117_ + 2, p_97116_ + 1, 16777215);
+            GuiComponent.m_93243_(p_97114_, StatsScreen.this.f_96547_, this.f_97106_, p_97117_ + 2 + 10, p_97116_ + 1 + 9, this.f_97107_ ? 9474192 : 6316128);
+            GuiComponent.m_93243_(p_97114_, StatsScreen.this.f_96547_, this.f_97108_, p_97117_ + 2 + 10, p_97116_ + 1 + 9 * 2, this.f_97109_ ? 9474192 : 6316128);
+         }
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
    class ItemStatisticsList extends ObjectSelectionList<StatsScreen.ItemStatisticsList.ItemRow> {
       protected final List<StatType<Block>> f_97021_;
       protected final List<StatType<Item>> f_97022_;
       private final int[] f_97029_ = new int[]{3, 4, 1, 2, 5, 6};
       protected int f_97023_ = -1;
       protected final List<Item> f_97024_;
-      protected final Comparator<Item> f_97025_ = new StatsScreen.ItemStatisticsList.ItemComparator();
+      protected final java.util.Comparator<Item> f_97025_ = new StatsScreen.ItemStatisticsList.ItemComparator();
       @Nullable
       protected StatType<?> f_97026_;
       protected int f_97027_;
@@ -374,7 +_,7 @@
       }
 
       @OnlyIn(Dist.CLIENT)
-      class ItemComparator implements Comparator<Item> {
+      class ItemComparator implements java.util.Comparator<Item> {
          private ItemComparator() {
          }
 
@@ -427,63 +_,6 @@
          protected void m_97091_(PoseStack p_97092_, @Nullable Stat<?> p_97093_, int p_97094_, int p_97095_, boolean p_97096_) {
             String s = p_97093_ == null ? "-" : p_97093_.m_12860_(StatsScreen.this.f_96901_.m_13015_(p_97093_));
             GuiComponent.m_93236_(p_97092_, StatsScreen.this.f_96547_, s, p_97094_ - StatsScreen.this.f_96547_.m_92895_(s), p_97095_ + 5, p_97096_ ? 16777215 : 9474192);
-         }
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   class MobsStatisticsList extends ObjectSelectionList<StatsScreen.MobsStatisticsList.MobRow> {
-      public MobsStatisticsList(Minecraft p_97100_) {
-         super(p_97100_, StatsScreen.this.f_96543_, StatsScreen.this.f_96544_, 32, StatsScreen.this.f_96544_ - 64, 9 * 4);
-
-         for(EntityType<?> entitytype : Registry.f_122826_) {
-            if (StatsScreen.this.f_96901_.m_13015_(Stats.f_12986_.m_12902_(entitytype)) > 0 || StatsScreen.this.f_96901_.m_13015_(Stats.f_12987_.m_12902_(entitytype)) > 0) {
-               this.m_7085_(new StatsScreen.MobsStatisticsList.MobRow(entitytype));
-            }
-         }
-
-      }
-
-      protected void m_7733_(PoseStack p_97102_) {
-         StatsScreen.this.m_7333_(p_97102_);
-      }
-
-      @OnlyIn(Dist.CLIENT)
-      class MobRow extends ObjectSelectionList.Entry<StatsScreen.MobsStatisticsList.MobRow> {
-         private final EntityType<?> f_97104_;
-         private final Component f_97105_;
-         private final Component f_97106_;
-         private final boolean f_97107_;
-         private final Component f_97108_;
-         private final boolean f_97109_;
-
-         public MobRow(EntityType<?> p_97112_) {
-            this.f_97104_ = p_97112_;
-            this.f_97105_ = p_97112_.m_20676_();
-            int i = StatsScreen.this.f_96901_.m_13015_(Stats.f_12986_.m_12902_(p_97112_));
-            if (i == 0) {
-               this.f_97106_ = new TranslatableComponent("stat_type.minecraft.killed.none", this.f_97105_);
-               this.f_97107_ = false;
-            } else {
-               this.f_97106_ = new TranslatableComponent("stat_type.minecraft.killed", i, this.f_97105_);
-               this.f_97107_ = true;
-            }
-
-            int j = StatsScreen.this.f_96901_.m_13015_(Stats.f_12987_.m_12902_(p_97112_));
-            if (j == 0) {
-               this.f_97108_ = new TranslatableComponent("stat_type.minecraft.killed_by.none", this.f_97105_);
-               this.f_97109_ = false;
-            } else {
-               this.f_97108_ = new TranslatableComponent("stat_type.minecraft.killed_by", this.f_97105_, j);
-               this.f_97109_ = true;
-            }
-
-         }
-
-         public void m_6311_(PoseStack p_97114_, int p_97115_, int p_97116_, int p_97117_, int p_97118_, int p_97119_, int p_97120_, int p_97121_, boolean p_97122_, float p_97123_) {
-            GuiComponent.m_93243_(p_97114_, StatsScreen.this.f_96547_, this.f_97105_, p_97117_ + 2, p_97116_ + 1, 16777215);
-            GuiComponent.m_93243_(p_97114_, StatsScreen.this.f_96547_, this.f_97106_, p_97117_ + 2 + 10, p_97116_ + 1 + 9, this.f_97107_ ? 9474192 : 6316128);
-            GuiComponent.m_93243_(p_97114_, StatsScreen.this.f_96547_, this.f_97108_, p_97117_ + 2 + 10, p_97116_ + 1 + 9 * 2, this.f_97109_ ? 9474192 : 6316128);
          }
       }
    }
