--- a/net/minecraft/client/model/geom/ModelPart.java
+++ b/net/minecraft/client/model/geom/ModelPart.java
@@ -244,6 +_,27 @@
    }
 
    @OnlyIn(Dist.CLIENT)
+   static class Vertex {
+      public final Vector3f f_104371_;
+      public final float f_104372_;
+      public final float f_104373_;
+
+      public Vertex(float p_104375_, float p_104376_, float p_104377_, float p_104378_, float p_104379_) {
+         this(new Vector3f(p_104375_, p_104376_, p_104377_), p_104378_, p_104379_);
+      }
+
+      public ModelPart.Vertex m_104384_(float p_104385_, float p_104386_) {
+         return new ModelPart.Vertex(this.f_104371_, p_104385_, p_104386_);
+      }
+
+      public Vertex(Vector3f p_104381_, float p_104382_, float p_104383_) {
+         this.f_104371_ = p_104381_;
+         this.f_104372_ = p_104382_;
+         this.f_104373_ = p_104383_;
+      }
+   }
+
+   @OnlyIn(Dist.CLIENT)
    static class Polygon {
       public final ModelPart.Vertex[] f_104359_;
       public final Vector3f f_104360_;
@@ -271,27 +_,6 @@
             this.f_104360_.m_122263_(-1.0F, 1.0F, 1.0F);
          }
 
-      }
-   }
-
-   @OnlyIn(Dist.CLIENT)
-   static class Vertex {
-      public final Vector3f f_104371_;
-      public final float f_104372_;
-      public final float f_104373_;
-
-      public Vertex(float p_104375_, float p_104376_, float p_104377_, float p_104378_, float p_104379_) {
-         this(new Vector3f(p_104375_, p_104376_, p_104377_), p_104378_, p_104379_);
-      }
-
-      public ModelPart.Vertex m_104384_(float p_104385_, float p_104386_) {
-         return new ModelPart.Vertex(this.f_104371_, p_104385_, p_104386_);
-      }
-
-      public Vertex(Vector3f p_104381_, float p_104382_, float p_104383_) {
-         this.f_104371_ = p_104381_;
-         this.f_104372_ = p_104382_;
-         this.f_104373_ = p_104383_;
       }
    }
 }
