--- a/net/minecraft/world/level/levelgen/structure/templatesystem/BlockMatchTest.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/BlockMatchTest.java
@@ -7,8 +_,8 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class BlockMatchTest extends RuleTest {
-   public static final Codec<BlockMatchTest> f_74063_ = Registry.f_122824_.fieldOf("block").xmap(BlockMatchTest::new, (p_74073_) -> {
-      return p_74073_.f_74064_;
+   public static final Codec<BlockMatchTest> f_74063_ = Registry.f_122824_.fieldOf("block").xmap(BlockMatchTest::new, (p_237076_0_) -> {
+      return p_237076_0_.f_74064_;
    }).codec();
    private final Block f_74064_;
 
