--- a/net/minecraft/world/level/levelgen/structure/templatesystem/BlockRotProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/BlockRotProcessor.java
@@ -7,8 +_,8 @@
 import net.minecraft.world.level.LevelReader;
 
 public class BlockRotProcessor extends StructureProcessor {
-   public static final Codec<BlockRotProcessor> f_74074_ = Codec.FLOAT.fieldOf("integrity").orElse(1.0F).xmap(BlockRotProcessor::new, (p_74088_) -> {
-      return p_74088_.f_74075_;
+   public static final Codec<BlockRotProcessor> f_74074_ = Codec.FLOAT.fieldOf("integrity").orElse(1.0F).xmap(BlockRotProcessor::new, (p_237078_0_) -> {
+      return p_237078_0_.f_74075_;
    }).codec();
    private final float f_74075_;
 
