--- a/net/minecraft/world/level/levelgen/structure/templatesystem/GravityProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/GravityProcessor.java
@@ -9,12 +_,12 @@
 import net.minecraft.world.level.levelgen.Heightmap;
 
 public class GravityProcessor extends StructureProcessor {
-   public static final Codec<GravityProcessor> f_74100_ = RecordCodecBuilder.create((p_74116_) -> {
-      return p_74116_.group(Heightmap.Types.f_64274_.fieldOf("heightmap").orElse(Heightmap.Types.WORLD_SURFACE_WG).forGetter((p_74120_) -> {
-         return p_74120_.f_74101_;
-      }), Codec.INT.fieldOf("offset").orElse(0).forGetter((p_74118_) -> {
-         return p_74118_.f_74102_;
-      })).apply(p_74116_, GravityProcessor::new);
+   public static final Codec<GravityProcessor> f_74100_ = RecordCodecBuilder.create((p_237082_0_) -> {
+      return p_237082_0_.group(Heightmap.Types.f_64274_.fieldOf("heightmap").orElse(Heightmap.Types.WORLD_SURFACE_WG).forGetter((p_237084_0_) -> {
+         return p_237084_0_.f_74101_;
+      }), Codec.INT.fieldOf("offset").orElse(0).forGetter((p_237083_0_) -> {
+         return p_237083_0_.f_74102_;
+      })).apply(p_237082_0_, GravityProcessor::new);
    });
    private final Heightmap.Types f_74101_;
    private final int f_74102_;
