--- a/net/minecraft/world/level/levelgen/structure/templatesystem/ProcessorRule.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/ProcessorRule.java
@@ -10,18 +_,18 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class ProcessorRule {
-   public static final Codec<ProcessorRule> f_74215_ = RecordCodecBuilder.create((p_74246_) -> {
-      return p_74246_.group(RuleTest.f_74307_.fieldOf("input_predicate").forGetter((p_74257_) -> {
-         return p_74257_.f_74216_;
-      }), RuleTest.f_74307_.fieldOf("location_predicate").forGetter((p_74255_) -> {
-         return p_74255_.f_74217_;
-      }), PosRuleTest.f_74198_.optionalFieldOf("position_predicate", PosAlwaysTrueTest.f_74188_).forGetter((p_74253_) -> {
-         return p_74253_.f_74218_;
-      }), BlockState.f_61039_.fieldOf("output_state").forGetter((p_74251_) -> {
-         return p_74251_.f_74219_;
-      }), CompoundTag.f_128325_.optionalFieldOf("output_nbt").forGetter((p_74248_) -> {
-         return Optional.ofNullable(p_74248_.f_74220_);
-      })).apply(p_74246_, ProcessorRule::new);
+   public static final Codec<ProcessorRule> f_74215_ = RecordCodecBuilder.create((p_237111_0_) -> {
+      return p_237111_0_.group(RuleTest.f_74307_.fieldOf("input_predicate").forGetter((p_237116_0_) -> {
+         return p_237116_0_.f_74216_;
+      }), RuleTest.f_74307_.fieldOf("location_predicate").forGetter((p_237115_0_) -> {
+         return p_237115_0_.f_74217_;
+      }), PosRuleTest.f_74198_.optionalFieldOf("position_predicate", PosAlwaysTrueTest.f_74188_).forGetter((p_237114_0_) -> {
+         return p_237114_0_.f_74218_;
+      }), BlockState.f_61039_.fieldOf("output_state").forGetter((p_237113_0_) -> {
+         return p_237113_0_.f_74219_;
+      }), CompoundTag.f_128325_.optionalFieldOf("output_nbt").forGetter((p_237112_0_) -> {
+         return Optional.ofNullable(p_237112_0_.f_74220_);
+      })).apply(p_237111_0_, ProcessorRule::new);
    });
    private final RuleTest f_74216_;
    private final RuleTest f_74217_;
