--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructureManager.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructureManager.java
@@ -52,9 +_,9 @@
 
    @Nullable
    public StructureTemplate m_74346_(ResourceLocation p_74347_) {
-      return this.f_74326_.computeIfAbsent(p_74347_, (p_74360_) -> {
-         StructureTemplate structuretemplate = this.m_74357_(p_74360_);
-         return structuretemplate != null ? structuretemplate : this.m_74355_(p_74360_);
+      return this.f_74326_.computeIfAbsent(p_74347_, (p_209204_1_) -> {
+         StructureTemplate structuretemplate = this.m_74357_(p_209204_1_);
+         return structuretemplate != null ? structuretemplate : this.m_74355_(p_209204_1_);
       });
    }
 
