--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
@@ -120,12 +_,12 @@
    }
 
    private static List<StructureTemplate.StructureBlockInfo> m_74614_(List<StructureTemplate.StructureBlockInfo> p_74615_, List<StructureTemplate.StructureBlockInfo> p_74616_, List<StructureTemplate.StructureBlockInfo> p_74617_) {
-      Comparator<StructureTemplate.StructureBlockInfo> comparator = Comparator.<StructureTemplate.StructureBlockInfo>comparingInt((p_74641_) -> {
-         return p_74641_.f_74675_.m_123342_();
-      }).thenComparingInt((p_74637_) -> {
-         return p_74637_.f_74675_.m_123341_();
-      }).thenComparingInt((p_74572_) -> {
-         return p_74572_.f_74675_.m_123343_();
+      Comparator<StructureTemplate.StructureBlockInfo> comparator = Comparator.<StructureTemplate.StructureBlockInfo>comparingInt((p_237154_0_) -> {
+         return p_237154_0_.f_74675_.m_123342_();
+      }).thenComparingInt((p_237153_0_) -> {
+         return p_237153_0_.f_74675_.m_123341_();
+      }).thenComparingInt((p_237148_0_) -> {
+         return p_237148_0_.f_74675_.m_123343_();
       });
       p_74615_.sort(comparator);
       p_74617_.sort(comparator);
@@ -138,8 +_,8 @@
    }
 
    private void m_74500_(Level p_74501_, BlockPos p_74502_, BlockPos p_74503_) {
-      List<Entity> list = p_74501_.m_6443_(Entity.class, new AABB(p_74502_, p_74503_), (p_74499_) -> {
-         return !(p_74499_ instanceof Player);
+      List<Entity> list = p_74501_.m_6443_(Entity.class, new AABB(p_74502_, p_74503_), (p_237142_0_) -> {
+         return !(p_237142_0_ instanceof Player);
       });
       this.f_74483_.clear();
 
@@ -190,6 +_,10 @@
       return m_74593_(p_74565_, p_74564_.m_74401_(), p_74564_.m_74404_(), p_74564_.m_74407_());
    }
 
+   public static Vec3 transformedVec3d(StructurePlaceSettings placementIn, Vec3 pos) {
+      return m_74578_(pos, placementIn.m_74401_(), placementIn.m_74404_(), placementIn.m_74407_());
+   }
+
    public void m_74531_(ServerLevelAccessor p_74532_, BlockPos p_74533_, StructurePlaceSettings p_74534_, Random p_74535_) {
       p_74534_.m_74412_();
       this.m_74628_(p_74532_, p_74533_, p_74534_, p_74535_);
@@ -215,7 +_,7 @@
             int i1 = Integer.MIN_VALUE;
             int j1 = Integer.MIN_VALUE;
 
-            for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : m_74517_(p_74537_, p_74538_, p_74539_, p_74540_, list)) {
+            for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : processBlockInfos(p_74537_, p_74538_, p_74539_, p_74540_, list, this)) {
                BlockPos blockpos = structuretemplate$structureblockinfo.f_74675_;
                if (boundingbox == null || boundingbox.m_71051_(blockpos)) {
                   FluidState fluidstate = p_74540_.m_74413_() ? p_74537_.m_6425_(blockpos) : null;
@@ -330,7 +_,7 @@
             }
 
             if (!p_74540_.m_74408_()) {
-               this.m_74523_(p_74537_, p_74538_, p_74540_.m_74401_(), p_74540_.m_74404_(), p_74540_.m_74407_(), boundingbox, p_74540_.m_74414_());
+               this.addEntitiesToWorld(p_74537_, p_74538_, p_74540_);
             }
 
             return true;
@@ -341,17 +_,17 @@
    }
 
    public static void m_74510_(LevelAccessor p_74511_, int p_74512_, DiscreteVoxelShape p_74513_, int p_74514_, int p_74515_, int p_74516_) {
-      p_74513_.m_82810_((p_74494_, p_74495_, p_74496_, p_74497_) -> {
-         BlockPos blockpos = new BlockPos(p_74514_ + p_74495_, p_74515_ + p_74496_, p_74516_ + p_74497_);
-         BlockPos blockpos1 = blockpos.m_121945_(p_74494_);
+      p_74513_.m_82810_((p_237141_5_, p_237141_6_, p_237141_7_, p_237141_8_) -> {
+         BlockPos blockpos = new BlockPos(p_74514_ + p_237141_6_, p_74515_ + p_237141_7_, p_74516_ + p_237141_8_);
+         BlockPos blockpos1 = blockpos.m_121945_(p_237141_5_);
          BlockState blockstate = p_74511_.m_8055_(blockpos);
          BlockState blockstate1 = p_74511_.m_8055_(blockpos1);
-         BlockState blockstate2 = blockstate.m_60728_(p_74494_, blockstate1, p_74511_, blockpos, blockpos1);
+         BlockState blockstate2 = blockstate.m_60728_(p_237141_5_, blockstate1, p_74511_, blockpos, blockpos1);
          if (blockstate != blockstate2) {
             p_74511_.m_7731_(blockpos, blockstate2, p_74512_ & -2);
          }
 
-         BlockState blockstate3 = blockstate1.m_60728_(p_74494_.m_122424_(), blockstate2, p_74511_, blockpos1, blockpos);
+         BlockState blockstate3 = blockstate1.m_60728_(p_237141_5_.m_122424_(), blockstate2, p_74511_, blockpos1, blockpos);
          if (blockstate1 != blockstate3) {
             p_74511_.m_7731_(blockpos1, blockstate3, p_74512_ & -2);
          }
@@ -359,14 +_,19 @@
       });
    }
 
+   @Deprecated //Use Forge version
    public static List<StructureTemplate.StructureBlockInfo> m_74517_(LevelAccessor p_74518_, BlockPos p_74519_, BlockPos p_74520_, StructurePlaceSettings p_74521_, List<StructureTemplate.StructureBlockInfo> p_74522_) {
+      return processBlockInfos(p_74518_, p_74519_, p_74520_, p_74521_, p_74522_, null);
+   }
+
+   public static List<StructureTemplate.StructureBlockInfo> processBlockInfos(LevelAccessor p_237145_0_, BlockPos p_237145_1_, BlockPos p_237145_2_, StructurePlaceSettings p_237145_3_, List<StructureTemplate.StructureBlockInfo> p_237145_4_, @Nullable StructureTemplate template) {
       List<StructureTemplate.StructureBlockInfo> list = Lists.newArrayList();
 
-      for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : p_74522_) {
-         BlockPos blockpos = m_74563_(p_74521_, structuretemplate$structureblockinfo.f_74675_).m_121955_(p_74519_);
+      for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : p_237145_4_) {
+         BlockPos blockpos = m_74563_(p_237145_3_, structuretemplate$structureblockinfo.f_74675_).m_121955_(p_237145_1_);
          StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo1 = new StructureTemplate.StructureBlockInfo(blockpos, structuretemplate$structureblockinfo.f_74676_, structuretemplate$structureblockinfo.f_74677_ != null ? structuretemplate$structureblockinfo.f_74677_.m_6426_() : null);
 
-         for(Iterator<StructureProcessor> iterator = p_74521_.m_74411_().iterator(); structuretemplate$structureblockinfo1 != null && iterator.hasNext(); structuretemplate$structureblockinfo1 = iterator.next().m_7382_(p_74518_, p_74519_, p_74520_, structuretemplate$structureblockinfo, structuretemplate$structureblockinfo1, p_74521_)) {
+         for(Iterator<StructureProcessor> iterator = p_237145_3_.m_74411_().iterator(); structuretemplate$structureblockinfo1 != null && iterator.hasNext(); structuretemplate$structureblockinfo1 = iterator.next().process(p_237145_0_, p_237145_1_, p_237145_2_, structuretemplate$structureblockinfo, structuretemplate$structureblockinfo1, p_237145_3_, template)) {
          }
 
          if (structuretemplate$structureblockinfo1 != null) {
@@ -377,28 +_,45 @@
       return list;
    }
 
-   private void m_74523_(ServerLevelAccessor p_74524_, BlockPos p_74525_, Mirror p_74526_, Rotation p_74527_, BlockPos p_74528_, @Nullable BoundingBox p_74529_, boolean p_74530_) {
-      for(StructureTemplate.StructureEntityInfo structuretemplate$structureentityinfo : this.f_74483_) {
-         BlockPos blockpos = m_74593_(structuretemplate$structureentityinfo.f_74684_, p_74526_, p_74527_, p_74528_).m_121955_(p_74525_);
-         if (p_74529_ == null || p_74529_.m_71051_(blockpos)) {
+   public static List<StructureTemplate.StructureEntityInfo> processEntityInfos(@Nullable StructureTemplate template, LevelAccessor p_215387_0_, BlockPos p_215387_1_, StructurePlaceSettings p_215387_2_, List<StructureTemplate.StructureEntityInfo> p_215387_3_) {
+      List<StructureTemplate.StructureEntityInfo> list = Lists.newArrayList();
+      for(StructureTemplate.StructureEntityInfo entityInfo : p_215387_3_) {
+         Vec3 pos = transformedVec3d(p_215387_2_, entityInfo.f_74683_).m_82549_(Vec3.m_82528_(p_215387_1_));
+         BlockPos blockpos = m_74563_(p_215387_2_, entityInfo.f_74684_).m_121955_(p_215387_1_);
+         StructureTemplate.StructureEntityInfo info = new StructureTemplate.StructureEntityInfo(pos, blockpos, entityInfo.f_74685_);
+         for (StructureProcessor proc : p_215387_2_.m_74411_()) {
+            info = proc.processEntity(p_215387_0_, p_215387_1_, entityInfo, info, p_215387_2_, template);
+            if (info == null)
+               break;
+         }
+         if (info != null)
+            list.add(info);
+      }
+      return list;
+   }
+
+   private void addEntitiesToWorld(ServerLevelAccessor p_237143_1_, BlockPos p_237143_2_, StructurePlaceSettings placementIn) {
+      for(StructureTemplate.StructureEntityInfo structuretemplate$structureentityinfo : processEntityInfos(this, p_237143_1_, p_237143_2_, placementIn, this.f_74483_)) {
+         BlockPos blockpos = m_74593_(structuretemplate$structureentityinfo.f_74684_, placementIn.m_74401_(), placementIn.m_74404_(), placementIn.m_74407_()).m_121955_(p_237143_2_);
+         blockpos = structuretemplate$structureentityinfo.f_74684_; // FORGE: Position will have already been transformed by processEntityInfos
+         if (placementIn.m_74409_() == null || placementIn.m_74409_().m_71051_(blockpos)) {
             CompoundTag compoundtag = structuretemplate$structureentityinfo.f_74685_.m_6426_();
-            Vec3 vec3 = m_74578_(structuretemplate$structureentityinfo.f_74683_, p_74526_, p_74527_, p_74528_);
-            Vec3 vec31 = vec3.m_82520_((double)p_74525_.m_123341_(), (double)p_74525_.m_123342_(), (double)p_74525_.m_123343_());
+            Vec3 vec31 = structuretemplate$structureentityinfo.f_74683_; // FORGE: Position will have already been transformed by processEntityInfos
             ListTag listtag = new ListTag();
             listtag.add(DoubleTag.m_128500_(vec31.f_82479_));
             listtag.add(DoubleTag.m_128500_(vec31.f_82480_));
             listtag.add(DoubleTag.m_128500_(vec31.f_82481_));
             compoundtag.m_128365_("Pos", listtag);
             compoundtag.m_128473_("UUID");
-            m_74543_(p_74524_, compoundtag).ifPresent((p_74553_) -> {
-               float f = p_74553_.m_6961_(p_74526_);
-               f = f + (p_74553_.f_19857_ - p_74553_.m_7890_(p_74527_));
-               p_74553_.m_7678_(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, f, p_74553_.f_19858_);
-               if (p_74530_ && p_74553_ instanceof Mob) {
-                  ((Mob)p_74553_).m_6518_(p_74524_, p_74524_.m_6436_(new BlockPos(vec31)), MobSpawnType.STRUCTURE, (SpawnGroupData)null, compoundtag);
+            m_74543_(p_237143_1_, compoundtag).ifPresent((p_242927_6_) -> {
+               float f = p_242927_6_.m_6961_(placementIn.m_74401_());
+               f = f + (p_242927_6_.f_19857_ - p_242927_6_.m_7890_(placementIn.m_74404_()));
+               p_242927_6_.m_7678_(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, f, p_242927_6_.f_19858_);
+               if (placementIn.m_74414_() && p_242927_6_ instanceof Mob) {
+                  ((Mob)p_242927_6_).m_6518_(p_237143_1_, p_237143_1_.m_6436_(new BlockPos(vec31)), MobSpawnType.STRUCTURE, (SpawnGroupData)null, compoundtag);
                }
 
-               p_74524_.m_47205_(p_74553_);
+               p_237143_1_.m_47205_(p_242927_6_);
             });
          }
       }
@@ -726,27 +_,6 @@
       return listtag;
    }
 
-   public static final class Palette {
-      private final List<StructureTemplate.StructureBlockInfo> f_74645_;
-      private final Map<Block, List<StructureTemplate.StructureBlockInfo>> f_74646_ = Maps.newHashMap();
-
-      private Palette(List<StructureTemplate.StructureBlockInfo> p_74648_) {
-         this.f_74645_ = p_74648_;
-      }
-
-      public List<StructureTemplate.StructureBlockInfo> m_74652_() {
-         return this.f_74645_;
-      }
-
-      public List<StructureTemplate.StructureBlockInfo> m_74653_(Block p_74654_) {
-         return this.f_74646_.computeIfAbsent(p_74654_, (p_74659_) -> {
-            return this.f_74645_.stream().filter((p_74657_) -> {
-               return p_74657_.f_74676_.m_60713_(p_74659_);
-            }).collect(Collectors.toList());
-         });
-      }
-   }
-
    static class SimplePalette implements Iterable<BlockState> {
       public static final BlockState f_74660_ = Blocks.f_50016_.m_49966_();
       private final IdMapper<BlockState> f_74661_ = new IdMapper<>(16);
@@ -805,6 +_,27 @@
          this.f_74683_ = p_74687_;
          this.f_74684_ = p_74688_;
          this.f_74685_ = p_74689_;
+      }
+   }
+
+   public static final class Palette {
+      private final List<StructureTemplate.StructureBlockInfo> f_74645_;
+      private final Map<Block, List<StructureTemplate.StructureBlockInfo>> f_74646_ = Maps.newHashMap();
+
+      private Palette(List<StructureTemplate.StructureBlockInfo> p_74648_) {
+         this.f_74645_ = p_74648_;
+      }
+
+      public List<StructureTemplate.StructureBlockInfo> m_74652_() {
+         return this.f_74645_;
+      }
+
+      public List<StructureTemplate.StructureBlockInfo> m_74653_(Block p_74654_) {
+         return this.f_74646_.computeIfAbsent(p_74654_, (p_237160_1_) -> {
+            return this.f_74645_.stream().filter((p_237159_1_) -> {
+               return p_237159_1_.f_74676_.m_60713_(p_237160_1_);
+            }).collect(Collectors.toList());
+         });
       }
    }
 }
