--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructureProcessorType.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructureProcessorType.java
@@ -18,11 +_,11 @@
    StructureProcessorType<LavaSubmergedBlockProcessor> f_74464_ = m_74476_("lava_submerged_block", LavaSubmergedBlockProcessor.f_74134_);
    Codec<StructureProcessor> f_74465_ = Registry.f_122891_.dispatch("processor_type", StructureProcessor::m_6953_, StructureProcessorType::m_74481_);
    Codec<StructureProcessorList> f_74466_ = f_74465_.listOf().xmap(StructureProcessorList::new, StructureProcessorList::m_74425_);
-   Codec<StructureProcessorList> f_74467_ = Codec.either(f_74466_.fieldOf("processors").codec(), f_74466_).xmap((p_74471_) -> {
-      return p_74471_.map((p_74480_) -> {
-         return p_74480_;
-      }, (p_74475_) -> {
-         return p_74475_;
+   Codec<StructureProcessorList> f_74467_ = Codec.either(f_74466_.fieldOf("processors").codec(), f_74466_).xmap((p_242923_0_) -> {
+      return p_242923_0_.map((p_242926_0_) -> {
+         return p_242926_0_;
+      }, (p_242925_0_) -> {
+         return p_242925_0_;
       });
    }, Either::left);
    Codec<Supplier<StructureProcessorList>> f_74468_ = RegistryFileCodec.m_135589_(Registry.f_122883_, f_74467_);
