--- a/net/minecraft/world/level/storage/loot/entries/ComposableEntryContainer.java
+++ b/net/minecraft/world/level/storage/loot/entries/ComposableEntryContainer.java
@@ -6,10 +_,10 @@
 
 @FunctionalInterface
 interface ComposableEntryContainer {
-   ComposableEntryContainer f_79405_ = (p_79418_, p_79419_) -> {
+   ComposableEntryContainer f_79405_ = (p_216134_0_, p_216134_1_) -> {
       return false;
    };
-   ComposableEntryContainer f_79406_ = (p_79409_, p_79410_) -> {
+   ComposableEntryContainer f_79406_ = (p_216136_0_, p_216136_1_) -> {
       return true;
    };
 
@@ -17,15 +_,15 @@
 
    default ComposableEntryContainer m_79411_(ComposableEntryContainer p_79412_) {
       Objects.requireNonNull(p_79412_);
-      return (p_79424_, p_79425_) -> {
-         return this.m_6562_(p_79424_, p_79425_) && p_79412_.m_6562_(p_79424_, p_79425_);
+      return (p_216137_2_, p_216137_3_) -> {
+         return this.m_6562_(p_216137_2_, p_216137_3_) && p_79412_.m_6562_(p_216137_2_, p_216137_3_);
       };
    }
 
    default ComposableEntryContainer m_79420_(ComposableEntryContainer p_79421_) {
       Objects.requireNonNull(p_79421_);
-      return (p_79415_, p_79416_) -> {
-         return this.m_6562_(p_79415_, p_79416_) || p_79421_.m_6562_(p_79415_, p_79416_);
+      return (p_216138_2_, p_216138_3_) -> {
+         return this.m_6562_(p_216138_2_, p_216138_3_) || p_79421_.m_6562_(p_216138_2_, p_216138_3_);
       };
    }
 }
