--- a/net/minecraft/world/level/storage/loot/entries/DynamicLoot.java
+++ b/net/minecraft/world/level/storage/loot/entries/DynamicLoot.java
@@ -28,8 +_,8 @@
    }
 
    public static LootPoolSingletonContainer.Builder<?> m_79483_(ResourceLocation p_79484_) {
-      return m_79687_((p_79487_, p_79488_, p_79489_, p_79490_) -> {
-         return new DynamicLoot(p_79484_, p_79487_, p_79488_, p_79489_, p_79490_);
+      return m_79687_((p_216164_1_, p_216164_2_, p_216164_3_, p_216164_4_) -> {
+         return new DynamicLoot(p_79484_, p_216164_1_, p_216164_2_, p_216164_3_, p_216164_4_);
       });
    }
 
