--- a/net/minecraft/world/level/storage/loot/entries/EntryGroup.java
+++ b/net/minecraft/world/level/storage/loot/entries/EntryGroup.java
@@ -20,15 +_,15 @@
       case 2:
          ComposableEntryContainer composableentrycontainer = p_79559_[0];
          ComposableEntryContainer composableentrycontainer1 = p_79559_[1];
-         return (p_79556_, p_79557_) -> {
-            composableentrycontainer.m_6562_(p_79556_, p_79557_);
-            composableentrycontainer1.m_6562_(p_79556_, p_79557_);
+         return (p_216151_2_, p_216151_3_) -> {
+            composableentrycontainer.m_6562_(p_216151_2_, p_216151_3_);
+            composableentrycontainer1.m_6562_(p_216151_2_, p_216151_3_);
             return true;
          };
       default:
-         return (p_79562_, p_79563_) -> {
+         return (p_216152_1_, p_216152_2_) -> {
             for(ComposableEntryContainer composableentrycontainer2 : p_79559_) {
-               composableentrycontainer2.m_6562_(p_79562_, p_79563_);
+               composableentrycontainer2.m_6562_(p_216152_1_, p_216152_2_);
             }
 
             return true;
