--- a/net/minecraft/world/level/storage/loot/entries/SequentialEntry.java
+++ b/net/minecraft/world/level/storage/loot/entries/SequentialEntry.java
@@ -20,9 +_,9 @@
       case 2:
          return p_79816_[0].m_79411_(p_79816_[1]);
       default:
-         return (p_79819_, p_79820_) -> {
+         return (p_216153_1_, p_216153_2_) -> {
             for(ComposableEntryContainer composableentrycontainer : p_79816_) {
-               if (!composableentrycontainer.m_6562_(p_79819_, p_79820_)) {
+               if (!composableentrycontainer.m_6562_(p_216153_1_, p_216153_2_)) {
                   return false;
                }
             }
