--- a/net/minecraft/client/renderer/texture/TextureAtlasSprite.java
+++ b/net/minecraft/client/renderer/texture/TextureAtlasSprite.java
@@ -18,7 +_,7 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 @OnlyIn(Dist.CLIENT)
-public class TextureAtlasSprite implements AutoCloseable {
+public class TextureAtlasSprite implements AutoCloseable, net.minecraftforge.client.extensions.IForgeTextureAtlasSprite {
    private final TextureAtlas f_118343_;
    private final TextureAtlasSprite.Info f_118344_;
    private final AnimationMetadataSection f_118345_;
@@ -136,6 +_,7 @@
 
    private void m_118375_(int p_118376_, int p_118377_, NativeImage[] p_118378_) {
       for(int i = 0; i < this.f_118342_.length; ++i) {
+         if ((this.f_118344_.f_118423_ >> i <= 0) || (this.f_118344_.f_118424_ >> i <= 0)) break;
          p_118378_[i].m_85003_(i, this.f_118349_ >> i, this.f_118350_ >> i, p_118376_ >> i, p_118377_ >> i, this.f_118344_.f_118423_ >> i, this.f_118344_.f_118424_ >> i, this.f_118342_.length > 1, false);
       }
 
@@ -341,5 +_,10 @@
          }
 
       }
+   }
+
+   // Forge Start
+   public int getPixelRGBA(int frameIndex, int x, int y) {
+      return this.f_118342_[0].m_84985_(x + this.f_118346_[frameIndex] * m_118405_(), y + this.f_118347_[frameIndex] * m_118408_());
    }
 }
