--- a/net/minecraft/util/datafix/DataFixers.java
+++ b/net/minecraft/util/datafix/DataFixers.java
@@ -246,8 +_,8 @@
       Schema schema16 = p_14514_.addSchema(501, V501::new);
       p_14514_.addFixer(new AddNewChoices(schema16, "Add 1.10 entities fix", References.f_16786_));
       Schema schema17 = p_14514_.addSchema(502, f_14508_);
-      p_14514_.addFixer(ItemRenameFix.m_16003_(schema17, "cooked_fished item renamer", (p_14533_) -> {
-         return Objects.equals(NamespacedSchema.m_17311_(p_14533_), "minecraft:cooked_fished") ? "minecraft:cooked_fish" : p_14533_;
+      p_14514_.addFixer(ItemRenameFix.m_16003_(schema17, "cooked_fished item renamer", (p_207111_0_) -> {
+         return Objects.equals(NamespacedSchema.m_17311_(p_207111_0_), "minecraft:cooked_fished") ? "minecraft:cooked_fish" : p_207111_0_;
       }));
       p_14514_.addFixer(new EntityZombieVillagerTypeFix(schema17, false));
       Schema schema18 = p_14514_.addSchema(505, f_14508_);
@@ -329,11 +_,11 @@
       p_14514_.addFixer(new AddNewChoices(schema50, "Add 1.13 entities fix", References.f_16786_));
       Schema schema51 = p_14514_.addSchema(1474, f_14509_);
       p_14514_.addFixer(new ColorlessShulkerEntityFix(schema51, false));
-      p_14514_.addFixer(BlockRenameFix.m_14914_(schema51, "Colorless shulker block fixer", (p_14531_) -> {
-         return Objects.equals(NamespacedSchema.m_17311_(p_14531_), "minecraft:purple_shulker_box") ? "minecraft:shulker_box" : p_14531_;
+      p_14514_.addFixer(BlockRenameFix.m_14914_(schema51, "Colorless shulker block fixer", (p_207106_0_) -> {
+         return Objects.equals(NamespacedSchema.m_17311_(p_207106_0_), "minecraft:purple_shulker_box") ? "minecraft:shulker_box" : p_207106_0_;
       }));
-      p_14514_.addFixer(ItemRenameFix.m_16003_(schema51, "Colorless shulker item fixer", (p_14516_) -> {
-         return Objects.equals(NamespacedSchema.m_17311_(p_14516_), "minecraft:purple_shulker_box") ? "minecraft:shulker_box" : p_14516_;
+      p_14514_.addFixer(ItemRenameFix.m_16003_(schema51, "Colorless shulker item fixer", (p_207101_0_) -> {
+         return Objects.equals(NamespacedSchema.m_17311_(p_207101_0_), "minecraft:purple_shulker_box") ? "minecraft:shulker_box" : p_207101_0_;
       }));
       Schema schema52 = p_14514_.addSchema(1475, f_14509_);
       p_14514_.addFixer(BlockRenameFix.m_14914_(schema52, "Flowing fixer", m_14524_(ImmutableMap.of("minecraft:flowing_water", "minecraft:water", "minecraft:flowing_lava", "minecraft:lava"))));
@@ -534,14 +_,14 @@
    }
 
    private static UnaryOperator<String> m_14524_(Map<String, String> p_14525_) {
-      return (p_14528_) -> {
-         return p_14525_.getOrDefault(p_14528_, p_14528_);
+      return (p_241302_1_) -> {
+         return p_14525_.getOrDefault(p_241302_1_, p_241302_1_);
       };
    }
 
    private static UnaryOperator<String> m_14517_(String p_14518_, String p_14519_) {
-      return (p_14523_) -> {
-         return Objects.equals(p_14523_, p_14518_) ? p_14519_ : p_14523_;
+      return (p_241300_2_) -> {
+         return Objects.equals(p_241300_2_, p_14518_) ? p_14519_ : p_241300_2_;
       };
    }
 }
