--- a/net/minecraft/client/gui/font/glyphs/WhiteGlyph.java
+++ b/net/minecraft/client/gui/font/glyphs/WhiteGlyph.java
@@ -10,7 +_,7 @@
 public enum WhiteGlyph implements RawGlyph {
    INSTANCE;
 
-   private static final NativeImage f_95309_ = Util.m_137469_(new NativeImage(NativeImage.Format.RGBA, 5, 8, false), (p_95319_) -> {
+   private static final NativeImage f_95309_ = Util.m_137469_(new NativeImage(NativeImage.Format.RGBA, 5, 8, false), (p_228173_0_) -> {
       for(int i = 0; i < 8; ++i) {
          for(int j = 0; j < 5; ++j) {
             if (j != 0 && j + 1 != 5 && i != 0 && i + 1 != 8) {
@@ -19,11 +_,11 @@
                boolean flag1 = true;
             }
 
-            p_95319_.m_84988_(j, i, -1);
+            p_228173_0_.m_84988_(j, i, -1);
          }
       }
 
-      p_95319_.m_85123_();
+      p_228173_0_.m_85123_();
    });
 
    public int m_5631_() {
